{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "from pprint import pprint\n",
    "import json\n",
    "import sys\n",
    "sys.path.insert(0, '../genepattern-python/')\n",
    "from gp import functions\n",
    "\n",
    "%load_ext autoreload\n",
    "%autoreload 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Logged into GenePattern server at: https://genepattern.broadinstitute.org/gp.\n",
      "Processing task (0/280): ABSOLUTE ...\n",
      "Processing task (1/280): ABSOLUTE.review ...\n",
      "Processing task (2/280): ABSOLUTE.summarize ...\n",
      "Processing task (3/280): Acgh2Tab ...\n",
      "Processing task (4/280): AddFCSEventIndex ...\n",
      "Processing task (5/280): AddFCSParameter ...\n",
      "Processing task (6/280): AddNoiseToFCS ...\n",
      "Processing task (7/280): AffySTExpressionFileCreator ...\n",
      "Processing task (8/280): ApplyGatingML ...\n",
      "Processing task (9/280): ARACNE ...\n",
      "Processing task (10/280): AreaChange ...\n",
      "Processing task (11/280): Arff2Gct ...\n",
      "Processing task (12/280): ATARiS ...\n"
     ]
    }
   ],
   "source": [
    "functions.login('https://genepattern.broadinstitute.org/gp', 'Simpli', 'Simpli')\n",
    "\n",
    "tasks = []\n",
    "task_json = {\n",
    "    \"library_path\": \"Jumis/tools/genepattern-python\",\n",
    "    \"tasks\": tasks,\n",
    "}\n",
    "\n",
    "tasks.append(\n",
    "    {\n",
    "        \"function_path\": 'gp.functions.login',\n",
    "        \"label\": 'Login GenePattern (Broad Institute)',\n",
    "        \"description\": 'Login into the GenePattern server at the Broad Institure.',\n",
    "        \"required_args\":\n",
    "        [            \n",
    "            {\n",
    "                'name': 'user_name',\n",
    "                'label': 'User name',\n",
    "                \"description\": 'GenePattern user name.',\n",
    "            },\n",
    "            {\n",
    "                'name': 'passcode',\n",
    "                'label': 'Passcode',\n",
    "                \"description\": 'GenePattern passcode.',\n",
    "            },        \n",
    "        ],\n",
    "        \"default_args\":\n",
    "        [\n",
    "            {\n",
    "                'name': 'server_name',\n",
    "                'value': 'https://genepattern.broadinstitute.org/gp',\n",
    "            },\n",
    "        ],\n",
    "    }\n",
    "            )\n",
    "\n",
    "for i, t in enumerate(functions.get_tasks()):\n",
    "    print('Processing task ({}/280): {} ...'.format(i, t))\n",
    "    \n",
    "    t = functions.GPTask(functions.SERVER, t)\n",
    "    t.param_load()\n",
    "    name = t.get_name()\n",
    "\n",
    "    required_args = []\n",
    "    default_args = [{'name': 'task_name', 'value': name}]\n",
    "    optional_args = []\n",
    "    other_information = {}\n",
    "\n",
    "    task_entry = {\n",
    "        \"function_path\": 'gp.functions.execute',\n",
    "        \"label\": name,\n",
    "        \"description\": t.get_description(),\n",
    "        \"required_args\": required_args,\n",
    "        \"default_args\": default_args,\n",
    "        \"optional_args\": optional_args,\n",
    "        \"returns\": [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}],\n",
    "        \"other_information\": other_information,\n",
    "    }\n",
    "\n",
    "    ps = t.get_parameters()\n",
    "    for p in ps:\n",
    "        a = {}\n",
    "        description = p.get_description()\n",
    "        default_value = p.get_default_value()\n",
    "\n",
    "        a['name'] = p.get_name()\n",
    "        a['description'] = description\n",
    "        if default_value:\n",
    "            a['value'] = default_value\n",
    "\n",
    "        type_ = p.get_type()\n",
    "        if p.is_choice_param():\n",
    "            choices = [d['value'] for d in p.get_choices()]\n",
    "            a['description']: '{} (default={}) (choices={}) (type={})'.format(description, default_value, choices, type_,)\n",
    "        else:\n",
    "            a['description']: '{} (default={}) (type={})'.format(description, default_value, type_,)\n",
    "\n",
    "        if default_value:\n",
    "            optional_args.append(a)\n",
    "        else:\n",
    "            required_args.append(a)\n",
    "\n",
    "    other_information['lsid'] = t.get_lsid()\n",
    "    other_information['version'] = t.get_version()\n",
    "\n",
    "    tasks.append(task_entry)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "pprint(tasks)\n",
    "with open('/Users/kateme/Jumis/tools/genepattern-python/functions.json', 'w') as f:\n",
    "    json.dump(task_json, f, indent=4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Logged into GenePattern server at: https://genepattern.broadinstitute.org/gp.\n",
      "<urllib.request.Request object at 0x10c3f0240>\n",
      "b'{\"lsid\": \"urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00020:5.1\", \"params\": [{\"name\": \"input.filename\", \"values\": [\"https://genepattern.broadinstitute.org/gp/users/Simpli/tmp/run7768702568123352155.tmp/animals.gct\"]}], \"tags\": [\"GenePattern Python Client\"]}'\n"
     ]
    }
   ],
   "source": [
    "js = functions.make_task_spec('PreprocessDataset')\n",
    "# js.set_parameter('input.filename',\n",
    "#                  'https://software.broadinstitute.org/cancer/software/genepattern/data/all_aml/all_aml_test.gct')\n",
    "js.set_parameter('input.filename',\n",
    "                 '/Users/kateme/Jumis/data/animals/animals.gct')\n",
    "j = functions.submit_job(js)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "True\n",
      "Filtered all genes.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "print(j.is_finished())\n",
    "for f in j.get_output_files():\n",
    "    print(f.read())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Printing tasks in JSON format ...\n",
      "Loading task-specifying JSONs in directory /Users/kateme/.Simpli/json/ ...\n",
      "Loading task-specifying JSON /Users/kateme/.Simpli/json/default_tasks.json ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task Load JSON: {'library_path': '', 'library_name': 'simpli.default_tasks', 'function_name': 'link_json', 'description': 'No description.', 'required_args': [{'name': 'filepath', 'value': '', 'label': 'Simpli JSON to be Loaded', 'description': 'No description.'}], 'default_args': [], 'optional_args': [], 'returns': []}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task Reset JSONs: {'library_path': '', 'library_name': 'simpli.default_tasks', 'function_name': 'reset_jsons', 'description': 'No description.', 'required_args': [], 'default_args': [], 'optional_args': [], 'returns': []}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task Assign: {'library_path': '', 'library_name': 'simpli.default_tasks', 'function_name': 'just_return', 'description': 'No description.', 'required_args': [{'name': 'value', 'value': '', 'label': 'Value to Assign', 'description': 'No description.'}], 'default_args': [], 'optional_args': [], 'returns': [{'label': 'Variable Name to Assign to', 'description': 'No description.'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task Slice DataFrame by Row: {'library_path': '', 'library_name': 'simpli.default_tasks', 'function_name': 'slice_dataframe', 'description': 'Return a subset of rows of a given DataFrame object.', 'required_args': [{'name': 'dataframe', 'value': '', 'label': 'DataFrame', 'description': 'Target dataframe object.'}, {'name': 'indices', 'value': '', 'label': 'Row Index(ices)', 'description': 'An integer or list of integers separated by commas of the rows to slice. Note that indices are 0 based.'}], 'default_args': [{'name': 'ax', 'value': '0', 'label': 'Ax', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Sliced DataFrame', 'description': 'A slice of the DataFrame including the specified rows.'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task Slice DataFrame by Column: {'library_path': '', 'library_name': 'simpli.default_tasks', 'function_name': 'slice_dataframe', 'description': 'No description.', 'required_args': [{'name': 'dataframe', 'value': '', 'label': 'DataFrame', 'description': 'No description.'}, {'name': 'indices', 'value': '', 'label': 'Column Index(ices)', 'description': 'No description.'}], 'default_args': [{'name': 'ax', 'value': '1', 'label': 'Ax', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Sliced DataFrame', 'description': 'No description.'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task Set Notebook Theme: {'library_path': '', 'library_name': 'simpli.default_tasks', 'function_name': 'set_notebook_theme', 'description': 'No description.', 'required_args': [{'name': 'filepath', 'value': '', 'label': 'CSS File', 'description': 'No description.'}], 'default_args': [], 'optional_args': [], 'returns': []}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task Center Align Outputs: {'library_path': '', 'library_name': 'simpli.default_tasks', 'function_name': 'center_align_output_cells', 'description': 'No description.', 'required_args': [], 'default_args': [], 'optional_args': [], 'returns': []}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task Display Start Banner and Logos: {'library_path': '', 'library_name': 'simpli.default_tasks', 'function_name': 'display_start_banner_and_logos', 'description': 'No description.', 'required_args': [], 'default_args': [], 'optional_args': [], 'returns': []}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task Display End Banner: {'library_path': '', 'library_name': 'simpli.default_tasks', 'function_name': 'display_end_banner', 'description': 'No description.', 'required_args': [], 'default_args': [], 'optional_args': [], 'returns': []}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task Link YouTube Video: {'library_path': '', 'library_name': 'simpli.default_tasks', 'function_name': 'youtube', 'description': 'No description.', 'required_args': [{'name': 'url', 'value': '', 'label': 'YouTube Video URL', 'description': 'No description.'}], 'default_args': [], 'optional_args': [], 'returns': []}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task Toggle Input Cells: {'library_path': '', 'library_name': 'simpli.default_tasks', 'function_name': 'toggle_input_cells', 'description': 'No description.', 'required_args': [], 'default_args': [], 'optional_args': [], 'returns': []}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task Execute HTML: {'library_path': '', 'library_name': 'simpli.default_tasks', 'function_name': 'display_raw_html', 'description': 'No description.', 'required_args': [], 'default_args': [], 'optional_args': [], 'returns': []}.\n",
      "(Getting tasks ...)\n",
      "Setting/updating task {} to be {'Load JSON': {'library_path': '', 'library_name': 'simpli.default_tasks', 'function_name': 'link_json', 'description': 'No description.', 'required_args': [{'name': 'filepath', 'value': '', 'label': 'Simpli JSON to be Loaded', 'description': 'No description.'}], 'default_args': [], 'optional_args': [], 'returns': []}, 'Reset JSONs': {'library_path': '', 'library_name': 'simpli.default_tasks', 'function_name': 'reset_jsons', 'description': 'No description.', 'required_args': [], 'default_args': [], 'optional_args': [], 'returns': []}, 'Assign': {'library_path': '', 'library_name': 'simpli.default_tasks', 'function_name': 'just_return', 'description': 'No description.', 'required_args': [{'name': 'value', 'value': '', 'label': 'Value to Assign', 'description': 'No description.'}], 'default_args': [], 'optional_args': [], 'returns': [{'label': 'Variable Name to Assign to', 'description': 'No description.'}]}, 'Slice DataFrame by Row': {'library_path': '', 'library_name': 'simpli.default_tasks', 'function_name': 'slice_dataframe', 'description': 'Return a subset of rows of a given DataFrame object.', 'required_args': [{'name': 'dataframe', 'value': '', 'label': 'DataFrame', 'description': 'Target dataframe object.'}, {'name': 'indices', 'value': '', 'label': 'Row Index(ices)', 'description': 'An integer or list of integers separated by commas of the rows to slice. Note that indices are 0 based.'}], 'default_args': [{'name': 'ax', 'value': '0', 'label': 'Ax', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Sliced DataFrame', 'description': 'A slice of the DataFrame including the specified rows.'}]}, 'Slice DataFrame by Column': {'library_path': '', 'library_name': 'simpli.default_tasks', 'function_name': 'slice_dataframe', 'description': 'No description.', 'required_args': [{'name': 'dataframe', 'value': '', 'label': 'DataFrame', 'description': 'No description.'}, {'name': 'indices', 'value': '', 'label': 'Column Index(ices)', 'description': 'No description.'}], 'default_args': [{'name': 'ax', 'value': '1', 'label': 'Ax', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Sliced DataFrame', 'description': 'No description.'}]}, 'Set Notebook Theme': {'library_path': '', 'library_name': 'simpli.default_tasks', 'function_name': 'set_notebook_theme', 'description': 'No description.', 'required_args': [{'name': 'filepath', 'value': '', 'label': 'CSS File', 'description': 'No description.'}], 'default_args': [], 'optional_args': [], 'returns': []}, 'Center Align Outputs': {'library_path': '', 'library_name': 'simpli.default_tasks', 'function_name': 'center_align_output_cells', 'description': 'No description.', 'required_args': [], 'default_args': [], 'optional_args': [], 'returns': []}, 'Display Start Banner and Logos': {'library_path': '', 'library_name': 'simpli.default_tasks', 'function_name': 'display_start_banner_and_logos', 'description': 'No description.', 'required_args': [], 'default_args': [], 'optional_args': [], 'returns': []}, 'Display End Banner': {'library_path': '', 'library_name': 'simpli.default_tasks', 'function_name': 'display_end_banner', 'description': 'No description.', 'required_args': [], 'default_args': [], 'optional_args': [], 'returns': []}, 'Link YouTube Video': {'library_path': '', 'library_name': 'simpli.default_tasks', 'function_name': 'youtube', 'description': 'No description.', 'required_args': [{'name': 'url', 'value': '', 'label': 'YouTube Video URL', 'description': 'No description.'}], 'default_args': [], 'optional_args': [], 'returns': []}, 'Toggle Input Cells': {'library_path': '', 'library_name': 'simpli.default_tasks', 'function_name': 'toggle_input_cells', 'description': 'No description.', 'required_args': [], 'default_args': [], 'optional_args': [], 'returns': []}, 'Execute HTML': {'library_path': '', 'library_name': 'simpli.default_tasks', 'function_name': 'display_raw_html', 'description': 'No description.', 'required_args': [], 'default_args': [], 'optional_args': [], 'returns': []}} ...\n",
      "(Getting tasks ...)\n",
      "(Setting tasks ...)\n",
      "Loading task-specifying JSON /Users/kateme/.Simpli/json/functions.json ...\n",
      "\tAssumed that library_path (/Users/kateme/Jumis/tools/genepattern-python) is relative to the user-home directory.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task Login GenePattern (Broad Institute): {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'login', 'description': 'Login into the GenePattern server at the Broad Institure.', 'required_args': [{'name': 'user_name', 'value': '', 'label': 'User name', 'description': 'GenePattern user name.'}, {'name': 'passcode', 'value': '', 'label': 'Passcode', 'description': 'GenePattern passcode.'}], 'default_args': [{'name': 'server_name', 'value': 'https://genepattern.broadinstitute.org/gp', 'label': 'Server Name', 'description': 'No description.'}], 'optional_args': [], 'returns': []}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task ABSOLUTE: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '[**Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] Extracts absolute copy numbers per cancer cell from a mixed DNA population. Use this module for the per-sample processing step in the workflow (usually after HAPSEG).', 'required_args': [{'name': 'seg.dat.file', 'value': '', 'label': 'Seg.Dat.File', 'description': 'A HAPSEG output file'}, {'name': 'primary.disease', 'value': '', 'label': 'Primary.Disease', 'description': 'Primary disease of the sample'}, {'name': 'sample.name', 'value': '', 'label': 'Sample.Name', 'description': 'The name of the sample, included in output plots'}, {'name': 'maf.file', 'value': '', 'label': 'Maf.File', 'description': 'If available, a minor allele frequency file. This specifies the data for somatic point mutations to be used by ABSOLUTE.'}, {'name': 'min.mut.af', 'value': '', 'label': 'Min.Mut.Af', 'description': 'If specified, a minimum mutation allelic fraction. Mutations with lower allelic fractions will be filtered out before analysis. Note that if maf.fn is specified, min.mut.af must also be specified.'}], 'default_args': [{'name': 'task_name', 'value': 'ABSOLUTE', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file.name.base', 'value': '<sample.name>', 'label': 'Output.File.Name.Base', 'description': 'If specified, provides a base filename for all output files. The default value is the sample.name.'}, {'name': 'sigma.p', 'value': '0', 'label': 'Sigma.P', 'description': 'Provisional value of excess sample level variance used for mode search'}, {'name': 'max.sigma.h', 'value': '0.015', 'label': 'Max.Sigma.H', 'description': 'Maximum value of excess sample level variance (Eq. 6)'}, {'name': 'min.ploidy', 'value': '0.95', 'label': 'Min.Ploidy', 'description': 'Minimum ploidy value to consider. Solutions implying lower ploidy values will be discarded'}, {'name': 'max.ploidy', 'value': '10', 'label': 'Max.Ploidy', 'description': 'Maximum ploidy value to consider. Solutions implying greater ploidy values will be discarded'}, {'name': 'platform', 'value': 'SNP_6.0', 'label': 'Platform', 'description': \"The chip type used, supported values are currently 'SNP_250K_STY', 'SNP_6.0' and 'Illumina_WES'\"}, {'name': 'max.as.seg.count', 'value': '1500', 'label': 'Max.As.Seg.Count', 'description': \"Maximum number of allelic segments. Samples with a higher segment count will be flagged as 'failed'\"}, {'name': 'max.neg.genome', 'value': '0.005', 'label': 'Max.Neg.Genome', 'description': 'Maximum genome fraction that may be modeled as non-clonal with copy-ratio below that of clonal homozygous deletion. Solutions implying greater values will be discarded.'}, {'name': 'max.non.clonal', 'value': '0.05', 'label': 'Max.Non.Clonal', 'description': 'Maximum genome fraction that may be modeled as non-clonal (subclonal SCNA). Solutions implying greater values will be discarded.'}, {'name': 'copy.number.type', 'value': 'allelic', 'label': 'Copy.Number.Type', 'description': \"The copy number type to assess, can be one of 'allelic' or 'total'\"}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task ABSOLUTE.review: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '[**Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] Extracts the absolute copy number per cancer cell from a mixed DNA population.  Use this module for the Review and Extract step in the workflow (after ABSOLUTE).', 'required_args': [{'name': 'reviewed.pp.calls.file', 'value': '', 'label': 'Reviewed.Pp.Calls.File', 'description': 'The calls file from a previous run of the ABSOLUTE.summarize module that has been reviewed and annotated by an analyst.'}, {'name': 'analyst.id', 'value': '', 'label': 'Analyst.Id', 'description': 'The user id of the analyst who called the solutions.  This is used for display/reporting purposes only.'}, {'name': 'modes.file', 'value': '', 'label': 'Modes.File', 'description': 'The obj_summary.PP-modes.data.RData file produced by the previous run of the ABSOLUTE.summarize module.'}, {'name': 'collection.name', 'value': '', 'label': 'Collection.Name', 'description': 'A descriptive name for this collection of samples.  Use the same name as in the previous run of the ABSOLUTE.summarize module.  This is used for display/reporting purposes only.'}], 'default_args': [{'name': 'task_name', 'value': 'ABSOLUTE.review', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'copy.number.type', 'value': 'allelic', 'label': 'Copy.Number.Type', 'description': \"The copy number type to assess, can be one of 'allelic' or 'total'\"}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task ABSOLUTE.summarize: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '[**Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] Summarizes the results from multiple ABSOLUTE runs so that an analyst can manually review the solutions.', 'required_args': [{'name': 'collection.name', 'value': '', 'label': 'Collection.Name', 'description': 'A descriptive name for this collection of samples.'}, {'name': 'absolute.files', 'value': '', 'label': 'Absolute.Files', 'description': 'A group of ABSOLUTE RData output files from individual sample runs to be summarized together.'}], 'default_args': [{'name': 'task_name', 'value': 'ABSOLUTE.summarize', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'copy.number.type', 'value': 'allelic', 'label': 'Copy.Number.Type', 'description': \"The copy number type to assess, can be one of 'allelic' or 'total'.\"}, {'name': 'plot.modes', 'value': 'TRUE', 'label': 'Plot.Modes', 'description': 'Set this to FALSE to disable plotting of the purity/ploidy modes.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task Acgh2Tab: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Converts acgh files to a tab-delimited format usable by Genomica ', 'required_args': [{'name': 'acgh.input.file', 'value': '', 'label': 'Acgh.Input.File', 'description': 'Data in aCHG format'}, {'name': 'presentlist.file', 'value': '', 'label': 'Presentlist.File', 'description': 'List of samples with a certain gene signature (from Genomica).'}, {'name': 'absentlist.file', 'value': '', 'label': 'Absentlist.File', 'description': 'List of samples without a certain gene signature (from Genomica).'}], 'default_args': [{'name': 'task_name', 'value': 'Acgh2Tab', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'genelocs.file', 'value': '<libdir>hg18_refseq_genes.txt', 'label': 'Genelocs.File', 'description': 'File giving gene locations (from GenomeBrowser).'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task AddFCSEventIndex: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Adds indexes to events in a Flow Cytometry Standard (FCS) data file.', 'required_args': [{'name': 'Input.data.file', 'value': '', 'label': 'Input.Data.File', 'description': 'An FCS data file to which you want to add indexes.'}, {'name': 'Index.parameter.long.name', 'value': '', 'label': 'Index.Parameter.Long.Name', 'description': 'The long name ($PnS keyword) of the parameter to get the index; the default is blank.'}], 'default_args': [{'name': 'task_name', 'value': 'AddFCSEventIndex', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Start.index', 'value': '1', 'label': 'Start.Index', 'description': 'The index number for the first event in the data set; the default is 1. The index will be incremented by 1 for each of the events in the data set. '}, {'name': 'Index.parameter.short.name', 'value': 'Index', 'label': 'Index.Parameter.Short.Name', 'description': 'The short name ($PnN keyword) of the parameter to get the index; the default is \"Index.\" '}, {'name': 'Output.fcs.file.name', 'value': '<Input.data.file_basename>', 'label': 'Output.Fcs.File.Name', 'description': 'The output file name.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task AddFCSParameter: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Add parameters and their values to a FCS data file', 'required_args': [{'name': 'Input.fcs.data.file', 'value': '', 'label': 'Input.Fcs.Data.File', 'description': 'A FCS data file to add parameters to.'}, {'name': 'Input.csv.file', 'value': '', 'label': 'Input.Csv.File', 'description': 'A CSV file with parameters and values to add.'}], 'default_args': [{'name': 'task_name', 'value': 'AddFCSParameter', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Output.file.name', 'value': '<Input.fcs.data.file_basename>_with_extra_params.fcs', 'label': 'Output.File.Name', 'description': 'The output FCS file name.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task AddNoiseToFCS: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Add noise to specified parameters in an FCS data file.', 'required_args': [{'name': 'Input.FCS.data.file', 'value': '', 'label': 'Input.FCS.Data.File', 'description': 'An FCS data file to add noise to.'}, {'name': 'Parameters.to.add.noise.to', 'value': '', 'label': 'Parameters.To.Add.Noise.To', 'description': 'List of parameters to which the module will add noise. The list may be provided as a list of indexes (e.g., \"1-3,5,7\") or as a list of parameter names (i.e., values of the $PnN keyword, such as \"FSC-H,FL1-H,FL2-A\"). By default, the noise will be added to all parameters but the Time parameter.'}], 'default_args': [{'name': 'task_name', 'value': 'AddNoiseToFCS', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Output.FCS.file.name', 'value': '<Input.FCS.data.file_basename>.with_noise.fcs', 'label': 'Output.FCS.File.Name', 'description': 'The output FCS file name.'}, {'name': 'Noise.level', 'value': '0.5', 'label': 'Noise.Level', 'description': 'Noise level in percent. This is the maximum noise that will be added to an event based on the current event parameter value or on the maximum range of the appropriate parameter. The latter option will be effective if the Uniform noise option is selected.'}, {'name': 'Noise', 'value': 'Proportional', 'label': 'Noise', 'description': 'How to add the noise: Proportional (The maximum amount of noise added to each value will be based on the value itself, e.g., low values will get lower amount of noise.) or Uniform (The maximum amount of noise added to each value is only dependent on the range of the particular parameter).'}, {'name': 'Saturated.events', 'value': 'RemoveSaturated', 'label': 'Saturated.Events', 'description': 'Whether to remove saturated events. An event is considered saturated if one or more of its parameter values are closer than 0.1 percent from the maximum range for that parameter (i.e., value of the appropriate $PnR keyword).'}, {'name': 'Random.Seed', 'value': '1234567', 'label': 'Random.Seed', 'description': 'The seed to use for the random generator. Leave this blank if want to use a randomly selected seed.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task AffySTExpressionFileCreator: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '[Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] Creates a GCT file from a set of CEL files from Affymetrix ST arrays.', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'One or more Affymetrix ST CEL files either uploaded directly, packaged into a ZIP or TAR bundle, or supplied through a directory input.  The CEL files can be in GZ format and the TAR can be in GZ, XZ, or BZ2 format.  The parameter will accept multiple inputs in any of these forms. '}, {'name': 'clm.file', 'value': '', 'label': 'Clm.File', 'description': 'A tab-delimited text file containing one scan, sample, and class per line'}], 'default_args': [{'name': 'task_name', 'value': 'AffySTExpressionFileCreator', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'normalize', 'value': 'yes', 'label': 'Normalize', 'description': 'Whether to normalize data using quantile normalization'}, {'name': 'background.correct', 'value': 'yes', 'label': 'Background.Correct', 'description': 'Whether to perform background correction'}, {'name': 'qc.plot.format', 'value': 'skip', 'label': 'Qc.Plot.Format', 'description': 'Choose a file format for the QC plots (or skip them completely)'}, {'name': 'annotate.rows', 'value': 'no', 'label': 'Annotate.Rows', 'description': 'Whether to annotate rows with the gene symbol and description'}, {'name': 'output.file.base', 'value': '<job_id>.expression.gct', 'label': 'Output.File.Base', 'description': 'The base name of the output file(s). File extensions will be added automatically.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task ApplyGatingML: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Apply a Gating-ML file on an FCS data file (gate and/or transform list mode data) ', 'required_args': [{'name': 'Input.FCS.data.file', 'value': '', 'label': 'Input.FCS.Data.File', 'description': 'An FCS data file to be gated.'}, {'name': 'Input.Gating-ML.file', 'value': '', 'label': 'Input.Gating-ML.File', 'description': 'A Gating-ML file specifying the gates.'}], 'default_args': [{'name': 'task_name', 'value': 'ApplyGatingML', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Output.base.name', 'value': '<Input.FCS.data.file_basename>', 'label': 'Output.Base.Name', 'description': 'Base for output file names'}, {'name': 'Output.figures', 'value': 'none', 'label': 'Output.Figures', 'description': 'Also output figures of the gates'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task ARACNE: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Runs the ARACNE algorithm for reverse engineering cellular networks', 'required_args': [{'name': 'dataset.file', 'value': '', 'label': 'Dataset.File', 'description': 'Dataset file - .res, .gct'}, {'name': 'hub.gene', 'value': '', 'label': 'Hub.Gene', 'description': 'The name of a gene whose network interactions you want to reconstruct'}, {'name': 'hub.genes.file', 'value': '', 'label': 'Hub.Genes.File', 'description': 'A file containing a list of hub genes with one gene per line - .txt'}, {'name': 'transcription.factor.file', 'value': '', 'label': 'Transcription.Factor.File', 'description': 'A file containing a list of genes with one gene per line that encode transcription factors - .txt'}, {'name': 'kernel.width', 'value': '', 'label': 'Kernel.Width', 'description': 'The kernel width (or window width) of the Gaussian Kernel Estimator'}], 'default_args': [{'name': 'task_name', 'value': 'ARACNE', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'mi.threshold', 'value': '0', 'label': 'Mi.Threshold', 'description': 'Threshold for the mutual information (MI) estimate to be considered statistically significant'}, {'name': 'p.value', 'value': '1', 'label': 'P.Value', 'description': 'Threshold for the p-value of a MI estimate to be considered statistically significant'}, {'name': 'dpi.tolerance', 'value': '1', 'label': 'Dpi.Tolerance', 'description': 'The percentage of MI estimation considered as sampling error'}, {'name': 'mean.filter', 'value': '0', 'label': 'Mean.Filter', 'description': 'Filter out non-informative genes whose mean expression value is smaller than mean filter'}, {'name': 'cv.filter', 'value': '0', 'label': 'Cv.Filter', 'description': 'Filter out non-informative genes whose coefficient of variance is smaller than cv filter'}, {'name': 'output.file', 'value': '<dataset.file_basename>.output.adj', 'label': 'Output.File', 'description': 'Output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task AreaChange: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Calculates fraction of area under the spectrum that is attributable to signal (area after noise removal / original area)', 'required_args': [{'name': 'spectrum.filename', 'value': '', 'label': 'Spectrum.Filename', 'description': 'spectrum data file - .csv'}, {'name': 'low.Da', 'value': '', 'label': 'Low.Da', 'description': 'minimum M/Z to include'}, {'name': 'high.Da', 'value': '', 'label': 'High.Da', 'description': 'maximum M/Z to include'}], 'default_args': [{'name': 'task_name', 'value': 'AreaChange', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'quality.threshold', 'value': '0.45', 'label': 'Quality.Threshold', 'description': 'discard spectra with quality < quality threshold'}, {'name': 'window.filter', 'value': '5', 'label': 'Window.Filter', 'description': 'window filter size'}, {'name': 'factor', 'value': '3', 'label': 'Factor', 'description': 'factor to multiply standard deviation by'}, {'name': 'smooth', 'value': 'median', 'label': 'Smooth', 'description': 'averaging function'}, {'name': 'output.file', 'value': '<spectrum.filename_basename>.area.odf', 'label': 'Output.File', 'description': 'output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task Arff2Gct: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Convert an .arff file into a gene pattern .gct / .cls file pair', 'required_args': [{'name': 'arff.file', 'value': '', 'label': 'Arff.File', 'description': 'The dataset in ARFF format.  Multi-label class names are separated by colons (:). - .arff'}], 'default_args': [{'name': 'task_name', 'value': 'Arff2Gct', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<arff.file_basename>', 'label': 'Output.File', 'description': 'Name to give the .gct containing dataset features and .cls containing dataset class labels output files.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task ATARiS: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Runs ATARiS on RNAi reagent-level data.', 'required_args': [{'name': 'gct.file', 'value': '', 'label': 'Gct.File', 'description': 'input GCT file. Unique reagent identifier in first column, gene symbol in second.'}], 'default_args': [{'name': 'task_name', 'value': 'ATARiS', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'identifier', 'value': '<gct.file_basename>', 'label': 'Identifier', 'description': 'text string to identify the current run'}, {'name': 'random.seed', 'value': '12345', 'label': 'Random.Seed', 'description': 'seed for random number generator'}, {'name': 'null.significance', 'value': '0.15', 'label': 'Null.Significance', 'description': 'gene solution significance level (1 - null percentile)'}, {'name': 'min.A.value', 'value': '0.3', 'label': 'Min.A.Value', 'description': 'minimum effect magnitude for RNAi reagent (relative to reagent with maximal effect)'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task AuDIT: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Automated Detection of Inaccurate and Imprecise Transitions in MRM Mass Spectrometry', 'required_args': [{'name': 'data.file', 'value': '', 'label': 'Data.File', 'description': 'input file with pre-processed MRM data in csv format'}], 'default_args': [{'name': 'task_name', 'value': 'AuDIT', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'skyline.export', 'value': 'TRUE', 'label': 'Skyline.Export', 'description': 'if \"yes\", data file is a Skyline csv export including following columns: Sample (usually derived from SampleName, having identical value for all its replicates), PeptideSequence, ReplicateName, FragmentIon, PrecursorCharge, ProductCharge, light Area, heavy Area. When \"no\", input must contain sample, replicate, peptide, transition.id, area, IS.area in that order.'}, {'name': 'pvalue.threshold', 'value': '0.00001', 'label': 'Pvalue.Threshold', 'description': 'p-value threshold above which transition is quantification-worthy. Must be a value between 0 and 1 inclusive.'}, {'name': 'cv.threshold', 'value': '0.2', 'label': 'Cv.Threshold', 'description': 'threshold for coefficient of variation below which transition is quantification-worthy'}, {'name': 'all.pairs', 'value': 'FALSE', 'label': 'All.Pairs', 'description': 'calculate all possible pairs of relative ratios for a transition'}, {'name': 'output.intermediate.results', 'value': 'FALSE', 'label': 'Output.Intermediate.Results', 'description': 'create files with intermediate results'}, {'name': 'output.prefix', 'value': '<data.file_basename>-results', 'label': 'Output.Prefix', 'description': 'file name prefix used for output file(s)'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task BedToGtf: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Converts BED files to GFF or GTF format', 'required_args': [{'name': 'bed.file', 'value': '', 'label': 'Bed.File', 'description': 'A bed file'}], 'default_args': [{'name': 'task_name', 'value': 'BedToGtf', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.format', 'value': 'gtf', 'label': 'Output.Format', 'description': 'Whether to output a gff file or exon only gtf file. The bed file must contain at least 12 columns in order to generate a gtf file.'}, {'name': 'output.prefix', 'value': '<bed.file_basename>', 'label': 'Output.Prefix', 'description': 'The prefix for the output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task Beroukhim.Getz.2007.PNAS.Glioma.GISTIC: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '', 'required_args': [], 'default_args': [{'name': 'task_name', 'value': 'Beroukhim.Getz.2007.PNAS.Glioma.GISTIC', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task BlastTrainTest: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Sequence similarity classification using BLAST', 'required_args': [{'name': 'train.file', 'value': '', 'label': 'Train.File', 'description': 'The training file in .FASTA format. Multi-label class names are separated by colons (:).'}, {'name': 'test.file', 'value': '', 'label': 'Test.File', 'description': 'The testing file in .FASTA format. Multi-label class names are separated by colons (:).'}], 'default_args': [{'name': 'task_name', 'value': 'BlastTrainTest', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'sub.matrix', 'value': 'PAM30', 'label': 'Sub.Matrix', 'description': 'The name of the substitution matrix used by Blast.'}, {'name': 'pred.results.file', 'value': '<test.file_basename>', 'label': 'Pred.Results.File', 'description': 'The name of the output file for prediction results.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task BlastXValidation: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Sequence similarity cross validation prediction using BLAST', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'The input sequence and annotation information in .FASTA format.'}], 'default_args': [{'name': 'task_name', 'value': 'BlastXValidation', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'num.folds', 'value': '2', 'label': 'Num.Folds', 'description': 'The number of folds should be an integer greater than or equal to two.'}, {'name': 'sub.matrix', 'value': 'PAM30', 'label': 'Sub.Matrix', 'description': 'The name of the substitution matrix used by Blast.'}, {'name': 'pred.results.file', 'value': '<input.file_basename>', 'label': 'Pred.Results.File', 'description': 'The name of the output file for prediction results.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task Bowtie.aligner: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Bowtie2 (v. 2.1.0) is an ultrafast and memory-efficient short read aligner.', 'required_args': [{'name': 'prebuilt.bowtie.index', 'value': '', 'label': 'Prebuilt.Bowtie.Index', 'description': 'A prebuilt Bowtie 2 index. Either a prebuilt or a custom Bowtie 2 index must be specified.'}, {'name': 'custom.bowtie.index', 'value': '', 'label': 'Custom.Bowtie.Index', 'description': 'A ZIP archive containing Bowtie 2 index files. Either a prebuilt or a custom Bowtie 2 index must be specified.'}, {'name': 'input.format', 'value': '', 'label': 'Input.Format', 'description': 'The format of the reads input files.'}, {'name': 'reads.pair.1', 'value': '', 'label': 'Reads.Pair.1', 'description': 'Unpaired reads file or first mate for paired reads. A file or zip of files containing reads in FASTA or FASTQ format (can be compressed - ie .gz).'}, {'name': 'reads.pair.2', 'value': '', 'label': 'Reads.Pair.2', 'description': 'Second mate for paired reads. A file or zip of files in FASTA or FASTQ format (can be compressed - ie .gz).'}, {'name': 'quality.value.scale', 'value': '', 'label': 'Quality.Value.Scale', 'description': 'The scale to use for quality values.'}, {'name': 'integer.quality.values', 'value': '', 'label': 'Integer.Quality.Values', 'description': 'Whether the quality values are represented as space-separated ASCII integers (i.e 40 40 30 40 ..).'}, {'name': 'max.reads.to.align', 'value': '', 'label': 'Max.Reads.To.Align', 'description': 'The maximum number of reads to align.'}, {'name': 'mate.orientations', 'value': '', 'label': 'Mate.Orientations', 'description': 'The upstream/downstream mate orientations for a valid paired-end alignment against the forward reference strand.'}], 'default_args': [{'name': 'task_name', 'value': 'Bowtie.aligner', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'alignment.mode', 'value': '--end-to-end', 'label': 'Alignment.Mode', 'description': 'The alignment mode to use.'}, {'name': 'preset.options', 'value': 'sensitive', 'label': 'Preset.Options', 'description': 'A combination of pre-packaged options based on speed and sensitivity/accuracy.'}, {'name': 'min.fragment.length', 'value': '0', 'label': 'Min.Fragment.Length', 'description': 'The minimum fragment length for valid paired-end alignments.'}, {'name': 'max.fragment.length', 'value': '500', 'label': 'Max.Fragment.Length', 'description': 'The maximum fragment length for valid paired-end alignments.'}, {'name': 'random.seed', 'value': '123456', 'label': 'Random.Seed', 'description': 'The seed to use for the pseudo-random number generator.'}, {'name': 'output.prefix', 'value': '<reads.pair.1_basename>_<reads.pair.2_basename>', 'label': 'Output.Prefix', 'description': 'The prefix to use for the output file name.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task Bowtie.indexer: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Builds a Bowtie2 (v. 2.1.0) index from a set of DNA sequences', 'required_args': [{'name': 'fasta.file', 'value': '', 'label': 'Fasta.File', 'description': 'A file or zip of files containing sequences in FASTA format'}], 'default_args': [{'name': 'task_name', 'value': 'Bowtie.indexer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'random.seed', 'value': '123456', 'label': 'Random.Seed', 'description': 'The seed for the random number generator'}, {'name': 'index.name', 'value': '<fasta.file_basename>', 'label': 'Index.Name', 'description': 'The base name of the index files'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task BWA.aln: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '[**Beta Release** Contact gp-help with any issues.  Check stdout.txt and stderr.txt for errors] A fast and accurate short-read alignment tool that allows for mismatches and gaps. Alignments are output in a SAM format file, which provides Phred-scale quality scores for each alignment.  Uses BWA 0.7.4 aln and samse/sampe.', 'required_args': [{'name': 'BWA.index', 'value': '', 'label': 'BWA.Index', 'description': 'A zip file or directory containing BWA index files.'}, {'name': 'reads.pair.1', 'value': '', 'label': 'Reads.Pair.1', 'description': 'Single-end or first paired-end reads file in FASTA, FASTQ, or BAM format.  For paired-end data, this should be the forward (\"*_1\" or \"left\") input file.  Note: the FASTA or FASTQ can be gzipped.'}, {'name': 'reads.pair.2', 'value': '', 'label': 'Reads.Pair.2', 'description': 'The reverse (\"*_2\" or \"right\") input file for paired-end reads in FASTA, FASTQ, or BAM format. Note: the FASTA or FASTQ can be gzipped.'}, {'name': 'bam.mapping', 'value': '', 'label': 'Bam.Mapping', 'description': 'Specifies how to map BAM input. This is only required if input file is in BAM format.'}, {'name': 'max.edit.distance', 'value': '', 'label': 'Max.Edit.Distance', 'description': 'The max edit distance. This specifies a threshold of the maximum number of deletions, insertions, and substitutions needed to transform the reference sequence into the read sequence.'}, {'name': 'max.num.gap', 'value': '', 'label': 'Max.Num.Gap', 'description': 'Maximum number of gap opens. This specifies a threshold of the maximum number of gaps that can be initiated to match a given read to the reference.'}, {'name': 'max.gap.extension', 'value': '', 'label': 'Max.Gap.Extension', 'description': 'Maximum number of gap extensions. This specifies a threshold of the maximum number of bases by which gaps in a read can be extended.'}, {'name': 'max.deletion.length', 'value': '', 'label': 'Max.Deletion.Length', 'description': \"Disallow a long deletion within this many bp of the 3' end.\"}, {'name': 'max.indel.length', 'value': '', 'label': 'Max.Indel.Length', 'description': 'Disallow an indel within this many bp of the ends'}, {'name': 'seed.length', 'value': '', 'label': 'Seed.Length', 'description': 'The set of bases determined by this option in the high-quality (left) end of the read is the seed.'}, {'name': 'max.seed.edit.distance', 'value': '', 'label': 'Max.Seed.Edit.Distance', 'description': 'Maximum edit distance in the seed; that is, the maximum number of changes required to transform the reference sequence of the seed into the read sequence of the seed.'}, {'name': 'mismatch.penalty', 'value': '', 'label': 'Mismatch.Penalty', 'description': 'Mismatch penalty'}, {'name': 'gap.open.penalty', 'value': '', 'label': 'Gap.Open.Penalty', 'description': 'Gap open penalty.  The gap open penalty is the score taken away for the initiation of the gap in sequence. To make the match more significant you can try to make the gap penalty larger.'}, {'name': 'gap.extension.penalty', 'value': '', 'label': 'Gap.Extension.Penalty', 'description': 'Gap extension penalty.  The gap extension penalty is added to the standard gap penalty for each base or residue in the gap.  To reduce long gaps, increase the extension gap penalty.'}, {'name': 'max.best.hits', 'value': '', 'label': 'Max.Best.Hits', 'description': 'Proceed with suboptimal alignments if there are no more than this many equally best hits. This option only affects paired-end mapping.'}, {'name': 'trim.reads', 'value': '', 'label': 'Trim.Reads', 'description': 'Specifies a quality threshold for read trimming'}, {'name': 'barcode.length', 'value': '', 'label': 'Barcode.Length', 'description': \"Length of barcode starting from the 5' end\"}, {'name': 'max.insert.size', 'value': '', 'label': 'Max.Insert.Size', 'description': 'Specifies the maximum insert size for a read pair to be considered to be mapped properly'}, {'name': 'max.occurrences', 'value': '', 'label': 'Max.Occurrences', 'description': 'Specifies the maximum occurrences of a read for pairing'}, {'name': 'max.alignments', 'value': '', 'label': 'Max.Alignments', 'description': 'Maximum number of alignments to output in the XA tag for reads paired properly'}, {'name': 'max.dc.alignments', 'value': '', 'label': 'Max.Dc.Alignments', 'description': 'Maximum number of alignments to output in the XA tag for disconcordant read pairs (excluding singletons)'}], 'default_args': [{'name': 'task_name', 'value': 'BWA.aln', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'iterative.search', 'value': 'no', 'label': 'Iterative.Search', 'description': 'Whether to disable iterative search'}, {'name': 'Illumina.1.3.format', 'value': 'no', 'label': 'Illumina.1.3.Format', 'description': 'The input is in the Illumina 1.3+ read format'}, {'name': 'output.prefix', 'value': '<reads.pair.1_basename>', 'label': 'Output.Prefix', 'description': 'Prefix to use for output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task BWA.bwasw: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '[**Beta Release** Contact gp-help with any issues.  Check stdout.txt and stderr.txt for errors] Aligns long sequences (>200 bp) to a sequence database using BWA 0.7.4 bwasw.', 'required_args': [{'name': 'BWA.index', 'value': '', 'label': 'BWA.Index', 'description': 'A zip file or directory containing BWA index files.'}, {'name': 'read.file', 'value': '', 'label': 'Read.File', 'description': 'Single-end or first paired-end reads file in FASTA or FASTQ format.  For paired-end data, this should be the forward (\"*_1\" or \"left\") input file.'}, {'name': 'mate.read.file', 'value': '', 'label': 'Mate.Read.File', 'description': 'The reverse (\"*_2\" or \"right\") reads file for paired-end reads in FASTA or FASTQ format.'}, {'name': 'match.score', 'value': '', 'label': 'Match.Score', 'description': 'The score of the match'}, {'name': 'mismatch.penalty', 'value': '', 'label': 'Mismatch.Penalty', 'description': 'Specifies the mismatch penalty.'}, {'name': 'gap.open.penalty', 'value': '', 'label': 'Gap.Open.Penalty', 'description': 'Gap open penalty.  The gap open penalty is the score taken away for the initiation of the gap in sequence. To make the match more significant you can try to make the gap penalty larger.'}, {'name': 'gap.extension.penalty', 'value': '', 'label': 'Gap.Extension.Penalty', 'description': 'Gap extension penalty.  The gap extension penalty is added to the standard gap penalty for each base or residue in the gap.  To reduce long gaps, increase the extension gap penalty. '}, {'name': 'band.width', 'value': '', 'label': 'Band.Width', 'description': 'band width in the banded alignment'}, {'name': 'min.score.threshold', 'value': '', 'label': 'Min.Score.Threshold', 'description': 'Minimum score threshold divided by the match score'}, {'name': 'threshold.coefficient', 'value': '', 'label': 'Threshold.Coefficient', 'description': 'Coefficient for threshold adjustment according to query length'}, {'name': 'z.best.heuristics', 'value': '', 'label': 'Z.Best.Heuristics', 'description': 'Z-best heuristics. Specifying a higher number increases accuracy at the cost of speed'}, {'name': 'max.sa.interval.size', 'value': '', 'label': 'Max.Sa.Interval.Size', 'description': 'Maximum SA interval size for initiating a seed. Specifying a higher number increases accuracy at the cost of speed.'}, {'name': 'min.num.seeds', 'value': '', 'label': 'Min.Num.Seeds', 'description': 'The minimum number of seeds contained in the best alignment from the forward-forward alignment process that allows the algorithm to skip performing the reverse alignment. BWA.bwasw tends to be faster and more accurate if the alignment is supported by more seeds.'}], 'default_args': [{'name': 'task_name', 'value': 'BWA.bwasw', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'num.threads', 'value': '1', 'label': 'Num.Threads', 'description': 'Number of threads'}, {'name': 'output.prefix', 'value': '<read.file_basename>', 'label': 'Output.Prefix', 'description': 'Prefix to use for the output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task BWA.indexer: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '[**Beta Release** Contact gp-help with any issues.  Check stdout.txt and stderr.txt for errors] Creates a BWT index from a sequence FASTA file for use with the BWA 0.7.4 aligners.', 'required_args': [{'name': 'fasta.file', 'value': '', 'label': 'Fasta.File', 'description': 'A sequence fasta file'}, {'name': 'algorithm', 'value': '', 'label': 'Algorithm', 'description': 'The algorithm to use for constructing the BWT index'}], 'default_args': [{'name': 'task_name', 'value': 'BWA.indexer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.prefix', 'value': '<fasta.file_basename>', 'label': 'Output.Prefix', 'description': 'A prefix for the output file name.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task CaArray2ImportViewer: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Imports data files from CaArray 2.4.1 and creates gct or cls files ', 'required_args': [{'name': 'experiment', 'value': '', 'label': 'Experiment', 'description': 'Title or identifier of the caArray experiment to import data from.  https://array.nci.nih.gov/caarray/home.action'}], 'default_args': [{'name': 'task_name', 'value': 'CaArray2ImportViewer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'url', 'value': 'http://array.nci.nih.gov:8080', 'label': 'Url', 'description': 'URL for caArray server.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task CART: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Classification and Regression Tree', 'required_args': [{'name': 'train.data.filename', 'value': '', 'label': 'Train.Data.Filename', 'description': 'The training data file - .gct, .res, ignored if a saved model (saved.model.filename) is used'}, {'name': 'train.cls.filename', 'value': '', 'label': 'Train.Cls.Filename', 'description': 'The training class file - .cls, ignored if a saved model (saved.model.filename) is used'}, {'name': 'saved.model.filename', 'value': '', 'label': 'Saved.Model.Filename', 'description': 'input CART model'}, {'name': 'test.data.filename', 'value': '', 'label': 'Test.Data.Filename', 'description': 'The test data file - .gct, .res'}, {'name': 'test.cls.filename', 'value': '', 'label': 'Test.Cls.Filename', 'description': 'The test class file - .cls'}], 'default_args': [{'name': 'task_name', 'value': 'CART', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'model.output.file', 'value': '<train.data.filename_basename>.cart.model', 'label': 'Model.Output.File', 'description': 'name of output CART model file  '}, {'name': 'pred.results.file', 'value': '<test.data.filename_basename>.pred.odf', 'label': 'Pred.Results.File', 'description': 'The name of the output file for prediction results'}, {'name': 'tree.output.file', 'value': '<test.data.filename_basename>.tree.pdf', 'label': 'Tree.Output.File', 'description': 'The name of the file containing a plot of the classification tree'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task CARTXValidation: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Classification and Regression Tree Cross-Validation', 'required_args': [{'name': 'data.filename', 'value': '', 'label': 'Data.Filename', 'description': 'The data file - .gct, .res'}, {'name': 'cls.filename', 'value': '', 'label': 'Cls.Filename', 'description': 'The class file - .cls '}], 'default_args': [{'name': 'task_name', 'value': 'CARTXValidation', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'prediction.results.file', 'value': '<data.filename_basename>.pred.odf', 'label': 'Prediction.Results.File', 'description': 'The name of the output file for prediction results'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task CBS: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Segments DNA copy number data into regions of estimated equal copy number using circular binary segmentation (CBS). ', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'Copy number file - .cn'}, {'name': 'sample.name', 'value': '', 'label': 'Sample.Name', 'description': 'The name of the sample to use. Leave blank to use the first sample in the input file.'}], 'default_args': [{'name': 'task_name', 'value': 'CBS', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'permutations', 'value': '10000', 'label': 'Permutations', 'description': 'Number of permutations used for p-value computation'}, {'name': 'significance.level', 'value': '0.01', 'label': 'Significance.Level', 'description': 'Significance levels for the test to accept change-points'}, {'name': 'output.prefix', 'value': '<input.file_basename>', 'label': 'Output.Prefix', 'description': 'The prefix to use for the output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task ChIPSeq.CreateHeatmap: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Generates a heatmap based on the ChIP-Seq signal extracted from a BAM file, according to the chromosomal locations specified in the BED file used in ExtractSignal.  This is an experimental module built for the Broad CEGS workshop of July 29-30, 2013.', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'Input file in TAB format, generated by ChipSeq.ExtractSignal'}], 'default_args': [{'name': 'task_name', 'value': 'ChIPSeq.CreateHeatmap', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'normalization', 'value': 'z-score', 'label': 'Normalization', 'description': 'Choose a normalization method'}, {'name': 'percent.start', 'value': '0.45', 'label': 'Percent.Start', 'description': 'percent of the bed region to start clustering at'}, {'name': 'percent.end', 'value': '0.55', 'label': 'Percent.End', 'description': 'percent of the bed region to end clustering at '}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task ChIPSeq.CreatePlots: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Generates a scatter plot and correlation matrix based on the ChIP-Seq signal extracted from a BAM file, according to the chromosomal locations specified in the BED file used in ExtractSignal.  This is an experimental module built for the Broad CEGS workshop of July 29-30, 2013.', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'Input file in TAB format, generated by ChipSeq.ExtractSignal'}], 'default_args': [{'name': 'task_name', 'value': 'ChIPSeq.CreatePlots', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task ChIPSeq.ExtractSignal: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Segment the read from a BAM file into bins by chromosome and extract the signal values to a tab delimited file for subsequent visualization.  This is an experimental module built for the Broad CEGS workshop of July 29-30, 2013.', 'required_args': [{'name': 'bam.file', 'value': '', 'label': 'Bam.File', 'description': 'Input file in BAM format.'}, {'name': 'chromosome.sizes.file', 'value': '', 'label': 'Chromosome.Sizes.File', 'description': \"Specify a chromosome sizes file.  Chromosome names must start with 'chr'.\"}, {'name': 'bin.size', 'value': '', 'label': 'Bin.Size', 'description': 'The desired size of the bins.  Defaults to 25 if no value is supplied.'}, {'name': 'whole.cell.extract.control.file', 'value': '', 'label': 'Whole.Cell.Extract.Control.File', 'description': 'A bam file containing Whole Cell Extract control data to use in normalizing the bin values.'}, {'name': 'region.bed.file', 'value': '', 'label': 'Region.Bed.File', 'description': 'The BED file containing regions of interest.'}, {'name': 'region.bin.size', 'value': '', 'label': 'Region.Bin.Size', 'description': 'Bin the bed regions into sub-regions of this size (in bps).  Must be a multiple of the bin size such that an integer number of bin file bins greater than zero fits into a single region bin. (E.g. One could have bin files of 25 bp, extend bed regions by 100 bp, then have region bins of 50 bp for a total of four region bins.). Defaults to bin.size if no value is supplied.'}, {'name': 'extend.by', 'value': '', 'label': 'Extend.By', 'description': 'The number of basepairs to extend a region from its center.  Total extended region size will be twice this value.  This option is most useful if one wants to analyze  regions of the same length and number of bins.  To simply use the bin boundaries nearest the BED coordinates (i.e. analyze variable numbers of bins per BED line), do not specify this parameter.'}], 'default_args': [{'name': 'task_name', 'value': 'ChIPSeq.ExtractSignal', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'assembly.name', 'value': 'hg19', 'label': 'Assembly.Name', 'description': 'Specify an assembly (i.e. hg19, hg18, etc.).'}, {'name': 'normalize', 'value': 'yes', 'label': 'Normalize', 'description': 'Normalize each bin value by scaling it to 20000000 reads.'}, {'name': 'aggregation.function', 'value': 'avg', 'label': 'Aggregation.Function', 'description': 'Function to run on the bin values over the regions of interest.'}, {'name': 'output.filename', 'value': '<bam.file_basename>.<region.bed.file_basename>.tab', 'label': 'Output.Filename', 'description': 'The name of the output file.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task ClassNeighbors: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Select genes that most closely resemble a profile', 'required_args': [{'name': 'data.filename', 'value': '', 'label': 'Data.Filename', 'description': 'Dataset file - .res, .gct, .odf'}, {'name': 'class.filename', 'value': '', 'label': 'Class.Filename', 'description': 'class file - .cls, .odf'}], 'default_args': [{'name': 'task_name', 'value': 'ClassNeighbors', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'marker.gene.list.file', 'value': '<data.filename_basename>.genelist', 'label': 'Marker.Gene.List.File', 'description': 'output filename for marker gene list results - .odf file'}, {'name': 'marker.data.set.file', 'value': '<data.filename_basename>.neighbors', 'label': 'Marker.Data.Set.File', 'description': 'output filename for marker data - .gct file'}, {'name': 'num.neighbors', 'value': '50', 'label': 'Num.Neighbors', 'description': 'number of neighbors to find'}, {'name': 'num.permutations', 'value': '100', 'label': 'Num.Permutations', 'description': 'number of permutations in permutation test'}, {'name': 'user.pval', 'value': '0.5', 'label': 'User.Pval', 'description': 'user-set significance value for permutation test'}, {'name': 'mean.or.median', 'value': 'mean', 'label': 'Mean.Or.Median', 'description': 'use mean or median for feature selection'}, {'name': 'ttest.or.snr', 'value': 't-test', 'label': 'Ttest.Or.Snr', 'description': 'use signal-to-noise or t-test to select neighbors'}, {'name': 'filter.data', 'value': 'no', 'label': 'Filter.Data', 'description': 'if no, values below will be ignored'}, {'name': 'min.threshold', 'value': '10', 'label': 'Min.Threshold', 'description': 'minimum threshold for data'}, {'name': 'max.threshold', 'value': '16000', 'label': 'Max.Threshold', 'description': 'maximum threshold for data'}, {'name': 'min.fold.diff', 'value': '5', 'label': 'Min.Fold.Diff', 'description': 'minimum fold difference for filtering genes'}, {'name': 'min.abs.diff', 'value': '50', 'label': 'Min.Abs.Diff', 'description': 'minimum absolute difference for filtering genes'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task ClsFileCreator: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'A tool to create a class label (CLS) file.', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'A dataset - .gct'}], 'default_args': [{'name': 'task_name', 'value': 'ClsFileCreator', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task CNToXCN: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '', 'required_args': [{'name': 'cn.file', 'value': '', 'label': 'Cn.File', 'description': 'Input CN file (a tab-delimited file format that contains SNP copy numbers)'}], 'default_args': [{'name': 'task_name', 'value': 'CNToXCN', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.prefix', 'value': '<cn.file_basename>', 'label': 'Output.Prefix', 'description': 'Prefix for the output XCN file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task CoClustering: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Co-clustering of RNA and global proteome data', 'required_args': [], 'default_args': [{'name': 'task_name', 'value': 'CoClustering', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'min.correlation', 'value': '0.4', 'label': 'Min.Correlation', 'description': 'Minimum correlation between RNA and proteome for gene to be included in clustering'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task CollapseDataset: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Collapses all probe set values for a gene into a single vector of values', 'required_args': [{'name': 'dataset.file', 'value': '', 'label': 'Dataset.File', 'description': 'Dataset file - .gct, .res'}, {'name': 'chip.platform', 'value': '', 'label': 'Chip.Platform', 'description': 'The chip platform'}], 'default_args': [{'name': 'task_name', 'value': 'CollapseDataset', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'collapse.mode', 'value': 'Maximum', 'label': 'Collapse.Mode', 'description': 'Collapse mode for probe sets => 1 gene'}, {'name': 'output.file.name', 'value': '<dataset.file_basename>.collapsed', 'label': 'Output.File.Name', 'description': 'The output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task ComBat: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Batch Correction', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'input file - .res, .gct'}, {'name': 'sample.info.file', 'value': '', 'label': 'Sample.Info.File', 'description': 'A file containing information about the samples. The sample info file must contain the three columns: Array, Sample, and Batch. Any additional columns are used as covariates in the analysis.(.txt file)'}, {'name': 'absent.calls.filter', 'value': '', 'label': 'Absent.Calls.Filter', 'description': 'The filter to apply to genes with absent calls (a value between 0 and 1). Filters genes with absent calls in 1-(absent calls filter) of the samples. Ignored if input file is not a .res file.\\t'}], 'default_args': [{'name': 'task_name', 'value': 'ComBat', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'covariate.columns', 'value': 'all', 'label': 'Covariate.Columns', 'description': \"The covariate columns to use in analysis. This is either set to all, none, or a list specifying one or more covariate columns from sample info file to use in the analysis (i.e, 4, 5, 7, ...). If this is set to 'all' then all the covariate columns are used. If this is set to 'none' then none of the covariate columns are used.\"}, {'name': 'create.prior.plots', 'value': 'T', 'label': 'Create.Prior.Plots', 'description': 'Whether to generate prior probability distribution plots. Ignored if non-parametric method is used.'}, {'name': 'prior.method', 'value': 'T', 'label': 'Prior.Method', 'description': 'Empirical Bayes priors distribution estimation method to use. Either a parametric or non-parametric method can be applied. If unsure of which method to use try the parametric method first since it is faster.'}, {'name': 'output.file', 'value': '<input.file_basename>.combat.<input.file_extension>', 'label': 'Output.File', 'description': 'Name of batch adjusted output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task CombineOdf: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Combine two prediction results files into a single, weighted, multi-label prediction results file', 'required_args': [{'name': 'pred1.file', 'value': '', 'label': 'Pred1.File', 'description': 'The first prediction file in .odf format.'}, {'name': 'pred2.file', 'value': '', 'label': 'Pred2.File', 'description': 'The second prediction file in .odf format.'}], 'default_args': [{'name': 'task_name', 'value': 'CombineOdf', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'pred1.file.weight', 'value': '0.5', 'label': 'Pred1.File.Weight', 'description': 'Weight assigned to classifier that produced first prediction file.'}, {'name': 'pred.file', 'value': '<pred1.file_basename>_<pred2.file_basename>', 'label': 'Pred.File', 'description': 'The name of the prediction file in modified .odf format.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task ComparativeMarkerSelection: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Identify differentially expressed genes that can discriminate between distinct classes of samples.', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'The input file - .res, .gct. Note that if your data is log transformed, you will need to set the \"log transformed data\" parameter below to \"yes\".'}, {'name': 'cls.file', 'value': '', 'label': 'Cls.File', 'description': 'The class file - .cls'}, {'name': 'confounding.variable.cls.file', 'value': '', 'label': 'Confounding.Variable.Cls.File', 'description': 'The class file containing the confounding variable.'}, {'name': 'min.std', 'value': '', 'label': 'Min.Std', 'description': 'The minimum standard deviation if test statistic includes min std option'}], 'default_args': [{'name': 'task_name', 'value': 'ComparativeMarkerSelection', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'test.direction', 'value': '2', 'label': 'Test.Direction', 'description': 'The test to perform'}, {'name': 'test.statistic', 'value': '0', 'label': 'Test.Statistic', 'description': 'The statistic to use'}, {'name': 'number.of.permutations', 'value': '10000', 'label': 'Number.Of.Permutations', 'description': 'The number of permutations to perform (use 0 to calculate asymptotic p-values using the standard independent two-sample t-test)'}, {'name': 'log.transformed.data', 'value': 'false', 'label': 'Log.Transformed.Data', 'description': 'Whether the input data has been log transformed'}, {'name': 'complete', 'value': 'false', 'label': 'Complete', 'description': 'Whether to perform all possible permutations'}, {'name': 'balanced', 'value': 'false', 'label': 'Balanced', 'description': 'Whether to perform balanced permutations'}, {'name': 'random.seed', 'value': '779948241', 'label': 'Random.Seed', 'description': 'The seed of the random number generator used to produce permutations'}, {'name': 'smooth.p.values', 'value': 'true', 'label': 'Smooth.P.Values', 'description': 'Whether to smooth p-values'}, {'name': 'phenotype.test', 'value': 'one versus all', 'label': 'Phenotype.Test', 'description': 'Tests to perform when cls file has more than two classes (Note: not fully corrected for multiple hypothesis testing)'}, {'name': 'output.filename', 'value': '<input.file_basename>.comp.marker.odf', 'label': 'Output.Filename', 'description': 'The name of the output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task ComparativeMarkerSelectionViewer: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Views the results from ComparativeMarkerSelection', 'required_args': [{'name': 'comparative.marker.selection.filename', 'value': '', 'label': 'Comparative.Marker.Selection.Filename', 'description': 'The output from ComparativeMarkerSelection'}, {'name': 'dataset.filename', 'value': '', 'label': 'Dataset.Filename', 'description': 'The dataset file used as input to ComparativeMarkerSelection'}], 'default_args': [{'name': 'task_name', 'value': 'ComparativeMarkerSelectionViewer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task CompareSpectra: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Compares two spectra to determine similarity', 'required_args': [{'name': 'spectrum1.filename', 'value': '', 'label': 'Spectrum1.Filename', 'description': 'Spectrum file -.csv'}, {'name': 'spectrum2.filename', 'value': '', 'label': 'Spectrum2.Filename', 'description': 'Spectrum file - .csv'}, {'name': 'low.Da', 'value': '', 'label': 'Low.Da', 'description': 'Minimum M/Z to include'}, {'name': 'high.Da', 'value': '', 'label': 'High.Da', 'description': 'Maximum M/Z to include'}], 'default_args': [{'name': 'task_name', 'value': 'CompareSpectra', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<spectrum1.filename_basename>_<spectrum2.filename_basename>_compare.odf', 'label': 'Output.File', 'description': 'Output file name - .odf'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task CompensateFCS: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Compensates an FCS data file', 'required_args': [{'name': 'FCS.data.file', 'value': '', 'label': 'FCS.Data.File', 'description': 'A FCS data file to be compensated'}, {'name': 'Spillover.matrix.file', 'value': '', 'label': 'Spillover.Matrix.File', 'description': 'A text file with spillover matrix. Use this parameter if there is no spillover matrix in the input FCS file or to use a spillover matrix different from the matrix in the input FCS file. The value from the FCS data file will be used if this parameter is not specified.'}], 'default_args': [{'name': 'task_name', 'value': 'CompensateFCS', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Output.file.name', 'value': '<FCS.data.file_basename>.compensated', 'label': 'Output.File.Name', 'description': 'The name of the output file. The extension will match the file format specified in Output file format.'}, {'name': 'Output.file.format', 'value': 'FCS', 'label': 'Output.File.Format', 'description': 'The file format in which to save the compensated data.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task ConcatenateFilelist: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Concatenate all of the items in the input filelist into a single large file.', 'required_args': [{'name': 'filelist', 'value': '', 'label': 'Filelist', 'description': 'a text file containing a list of file paths'}, {'name': 'num.header.lines', 'value': '', 'label': 'Num.Header.Lines', 'description': 'specify number of header lines so that they are output only once'}, {'name': 'output.filename', 'value': '', 'label': 'Output.Filename', 'description': 'The name of the output file'}], 'default_args': [{'name': 'task_name', 'value': 'ConcatenateFilelist', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task ConsensusClustering: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Resampling-based clustering method ', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'The data to be clustered (.gct, .res, .odf)'}, {'name': 'cluster.by', 'value': '', 'label': 'Cluster.By', 'description': 'Whether to cluster by rows/genes or columns/experiments'}, {'name': 'create.heat.map', 'value': '', 'label': 'Create.Heat.Map', 'description': 'Whether to create heatmaps (one for each cluster number)'}], 'default_args': [{'name': 'task_name', 'value': 'ConsensusClustering', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'kmax', 'value': '5', 'label': 'Kmax', 'description': 'Try K=2,3,...,kmax clusters (must be > 1)'}, {'name': 'resampling.iterations', 'value': '20', 'label': 'Resampling.Iterations', 'description': 'Number of resampling iterations'}, {'name': 'seed.value', 'value': '12345', 'label': 'Seed.Value', 'description': 'Random number generator seed'}, {'name': 'clustering.algorithm', 'value': 'HIERARCHICAL', 'label': 'Clustering.Algorithm', 'description': 'Type of clustering algorithm'}, {'name': 'distance.measure', 'value': 'EUCLIDEAN', 'label': 'Distance.Measure', 'description': 'Distance measure'}, {'name': 'resample', 'value': 'subsample', 'label': 'Resample', 'description': \"resampling scheme (one of 'subsample[ratio]', 'features[nfeat]', 'nosampling'). The default is subsampling with a proportion of 0.80. NOTE: The square brackets should not be included (i.e subsample0.7, features210).\"}, {'name': 'merge.type', 'value': 'average', 'label': 'Merge.Type', 'description': 'Ignored when algorithm other than hierarchical selected'}, {'name': 'descent.iterations', 'value': '2000', 'label': 'Descent.Iterations', 'description': 'Number of SOM/NMF iterations'}, {'name': 'output.stub', 'value': '<input.filename_basename>', 'label': 'Output.Stub', 'description': 'Stub pre-pended to all the output file names'}, {'name': 'normalize.type', 'value': '-n1', 'label': 'Normalize.Type', 'description': 'row-wise, column-wise, both'}, {'name': 'normalization.iterations', 'value': '0', 'label': 'Normalization.Iterations', 'description': 'number of row/column normalization iterations (supercedes normalize.type)'}, {'name': 'heat.map.size', 'value': '2', 'label': 'Heat.Map.Size', 'description': \"point size of a consensus matrix's heat map (between 1 and 20)\"}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task ConstellationMap: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'A downstream visualization and analysis tool for gene set enrichment results with interactive web visualizer.', 'required_args': [{'name': 'input.gct.file', 'value': '', 'label': 'Input.Gct.File', 'description': 'Input gene set enrichment file. Expected to be output from ssGSEAProjection module.'}, {'name': 'input.cls.file', 'value': '', 'label': 'Input.Cls.File', 'description': 'Phenotype labels. Input cls file describing phenotypes of gct file.'}, {'name': 'gene.sets.file', 'value': '', 'label': 'Gene.Sets.File', 'description': 'Gene set file. Must match gene sets listed in input gct file. '}, {'name': 'top.n', 'value': '', 'label': 'Top.N', 'description': 'The number of gene sets to be plotted.'}, {'name': 'target.class', 'value': '', 'label': 'Target.Class', 'description': 'The phenotypic category of interest.'}], 'default_args': [{'name': 'task_name', 'value': 'ConstellationMap', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'gene.sets.database', 'value': 'c1.all.v5.1.symbols.gmt', 'label': 'Gene.Sets.Database', 'description': 'Gene sets database from GSEA website.'}, {'name': 'direction', 'value': 'positive', 'label': 'Direction', 'description': 'The direction of correlation, as measured by mutual information, to be investigated.'}, {'name': 'image.format', 'value': 'PNG', 'label': 'Image.Format', 'description': 'The image file type of outputted plots, either bitmap (PNG) or vector graphics (PDF).'}, {'name': 'jaccard.threshold', 'value': '0.1', 'label': 'Jaccard.Threshold', 'description': 'The Jaccard index threshold above which connective edges between gene sets will be plotted.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task ConvertLineEndings: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': \"Converts line endings to the host operating system's format.\", 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'The input file (any non-binary file format)'}], 'default_args': [{'name': 'task_name', 'value': 'ConvertLineEndings', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<input.filename_basename>.cvt.<input.filename_extension>', 'label': 'Output.File', 'description': 'The output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task ConvertToMAGEML: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Converts a gct, res, or odf dataset file to a MAGE-ML file', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'The gct, res, or odf file to convert'}], 'default_args': [{'name': 'task_name', 'value': 'ConvertToMAGEML', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file.base.name', 'value': '<input.filename_basename>', 'label': 'Output.File.Base.Name', 'description': 'The base name of the zip output file or the MAGE-ML file'}, {'name': 'create.zip.file', 'value': 'yes', 'label': 'Create.Zip.File', 'description': 'Whether to zip the MAGE-ML files'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task ConvertToMAGETAB: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'A module to export data from GenePattern in MAGE-TAB format', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'A dataset file  - .gct, .res'}, {'name': 'cls.file', 'value': '', 'label': 'Cls.File', 'description': 'A class file  - .cls'}, {'name': 'intensity.type.label', 'value': '', 'label': 'Intensity.Type.Label', 'description': 'label that specifies the type of intensity values in data matrix file '}], 'default_args': [{'name': 'task_name', 'value': 'ConvertToMAGETAB', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'sample.reference.label', 'value': 'Hybridization REF', 'label': 'Sample.Reference.Label', 'description': 'reference label for samples in the data matrix file'}, {'name': 'feature.reference.label', 'value': 'CompositeElement REF', 'label': 'Feature.Reference.Label', 'description': 'reference label for features in the data matrix file'}, {'name': 'output.prefix', 'value': '<input.file_basename>', 'label': 'Output.Prefix', 'description': 'prefix to use for output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task COPA: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Performs Cancer Outlier Profile Analysis (COPA)', 'required_args': [{'name': 'dataset.file', 'value': '', 'label': 'Dataset.File', 'description': 'The dataset file - .res, .gct'}, {'name': 'sample.info.file', 'value': '', 'label': 'Sample.Info.File', 'description': 'sample info file - .txt'}], 'default_args': [{'name': 'task_name', 'value': 'COPA', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'outlier.cutoff', 'value': '5', 'label': 'Outlier.Cutoff', 'description': \"The cutoff to determine 'outlier' status.\"}, {'name': 'pre-filter.percentile', 'value': '0.95', 'label': 'Pre-Filter.Percentile', 'description': 'The percentile to use for pre-filtering the data. All genes with a number of outlier samples less than the top (default 95th) percentile will be removed from further consideration. This value must be between 0 and 1 exclusive.'}, {'name': 'compute.large', 'value': 'no', 'label': 'Compute.Large', 'description': 'Whether to continue analysis when the number of genes remaining after pre-filtering is greater than 1000'}, {'name': 'max.overlap', 'value': '0', 'label': 'Max.Overlap', 'description': \"The maximum number of samples that can be considered 'outliers' when comparing two genes. The default is 0, indicating that there can be no overlap.\"}, {'name': 'normal.outliers', 'value': '0', 'label': 'Normal.Outliers', 'description': \"The number of normal samples that can be considered 'outliers'. The default is 0, meaning that no normals may be outliers.\"}, {'name': 'gene.pairs.to.plot', 'value': '1-5', 'label': 'Gene.Pairs.To.Plot', 'description': 'A range of gene pairs to plot. (e.g., 2-5) based on the gene pair rank. The ranks are determined by the number of outliers. The more outliers a gene pair has the greater the ranking. Specify none if no gene pair plots should be generated.'}, {'name': 'output.file', 'value': '<dataset.file_basename>', 'label': 'Output.File', 'description': 'The name of the output files'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task CopyNumberDivideByNormals: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Determines the copy number of a target SNP', 'required_args': [{'name': 'snp.file', 'value': '', 'label': 'Snp.File', 'description': 'SNP file containing both normal and target samples.'}, {'name': 'sample.info.file', 'value': '', 'label': 'Sample.Info.File', 'description': 'dChip format sample info file containing Ploidy(numeric) and Sample columns'}], 'default_args': [{'name': 'task_name', 'value': 'CopyNumberDivideByNormals', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'include.genotype.calls', 'value': 'cn', 'label': 'Include.Genotype.Calls', 'description': 'Whether to include genotype calls from the input snp file if they exist.'}, {'name': 'output.file.name', 'value': '<snp.file_basename>', 'label': 'Output.File.Name', 'description': 'Output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task CopyNumberInferencePipeline: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '[**Available on public server only**] Generates segmented copy number calls from raw tumor and normal SNP6 CEL files.', 'required_args': [{'name': 'CopyNumberInferencePipeline.Part1of21.SNPFileCreator_SNP62.cel.files', 'value': '', 'label': 'CopyNumberInferencePipeline.Part1Of21.SNPFileCreator SNP62.Cel.Files', 'description': 'Zip of SNP 6.0 CEL files'}, {'name': 'CopyNumberInferencePipeline.Part1of21.ConvertLineEndings1.input.filename', 'value': '', 'label': 'CopyNumberInferencePipeline.Part1Of21.ConvertLineEndings1.Input.Filename', 'description': 'The input file (any non-binary file format)'}], 'default_args': [{'name': 'task_name', 'value': 'CopyNumberInferencePipeline', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'CopyNumberInferencePipeline.Part1of21.TangentNormalization9.output.files', 'value': 'nocnv', 'label': 'CopyNumberInferencePipeline.Part1Of21.TangentNormalization9.Output.Files', 'description': 'The types of files to output. Whether to output files with CNVs included, removed, or both.'}, {'name': 'CopyNumberInferencePipeline.Part2of2.Wrapper4.CopyNumberInferencePipeline.Part2of21.InsertGenomicLocations3.genome.annotation', 'value': 'hg19', 'label': 'CopyNumberInferencePipeline.Part2Of2.Wrapper4.CopyNumberInferencePipeline.Part2Of21.InsertGenomicLocations3.Genome.Annotation', 'description': 'The internally provided genome annotation to insert into the input file.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task CopyNumberNoise: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Measures noise, calculates QC metrics on copynumber data.', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'The quality scores computed by the CopyNumberInference module. '}], 'default_args': [{'name': 'task_name', 'value': 'CopyNumberNoise', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<input.file_basename>.noise.txt', 'label': 'Output.File', 'description': 'The name of the output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task copyOfReorderGCT: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Reorder a GCT to be more human readable by sorting on a property such as the Gene Symbol, mean expression, sum and variance', 'required_args': [{'name': 'gct.file', 'value': '', 'label': 'Gct.File', 'description': 'A GCT file to be reordered'}], 'default_args': [{'name': 'task_name', 'value': 'copyOfReorderGCT', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'sort.feature', 'value': 'geneSymbol', 'label': 'Sort.Feature', 'description': 'The feature to sort on'}, {'name': 'order.as', 'value': 'a', 'label': 'Order.As', 'description': 'sort in Ascending or Descending order'}, {'name': 'output.file', 'value': '<gct.file_basename>_reord.gct', 'label': 'Output.File', 'description': 'Name of the reordered GCT file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task CoxRegression: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Cox regression using clinical data and .cls file', 'required_args': [{'name': 'input.surv.data.filename', 'value': '', 'label': 'Input.Surv.Data.Filename', 'description': 'input clinical data. Case ID in 1st column (.txt file)'}, {'name': 'input.cls.filename', 'value': '', 'label': 'Input.Cls.Filename', 'description': 'input class information (.cls file)'}], 'default_args': [{'name': 'task_name', 'value': 'CoxRegression', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': 'CoxRegression_result', 'label': 'Output.File', 'description': ''}, {'name': 'time', 'value': 'time', 'label': 'Time', 'description': 'Column name for survival time'}, {'name': 'status', 'value': 'status', 'label': 'Status', 'description': 'Column name for censor (0: censored/alive, 1: event/death)'}, {'name': 'variable.continuous', 'value': 'NA', 'label': 'Variable.Continuous', 'description': '(e.g. age,labdata1,...)'}, {'name': 'variable.category', 'value': 'NA', 'label': 'Variable.Category', 'description': '(e.g. sex,cls,grade,...)'}, {'name': 'variable.interaction.terms', 'value': 'NA', 'label': 'Variable.Interaction.Terms', 'description': '(e.g. sex:cls,sex:grade,...)'}, {'name': 'strata', 'value': 'NA', 'label': 'Strata', 'description': 'Fit is done for each strata'}, {'name': 'input.subgroup', 'value': 'NA', 'label': 'Input.Subgroup', 'description': '(e.g. cls:2) *only one subgroup can be specified'}, {'name': 'variable.selection', 'value': 'none', 'label': 'Variable.Selection', 'description': ''}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task CreateSymlinks: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Creates symlinks to the input files in the job results directory. This is intended to be a helper module for creating scatter-gather pipelines.  Note that this module is not supported on Windows', 'required_args': [{'name': 'input.files', 'value': '', 'label': 'Input.Files', 'description': 'One or more input files'}], 'default_args': [{'name': 'task_name', 'value': 'CreateSymlinks', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task CsvToFcs: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Converts a Comma separated values (CSV) file to a Flow Cytometry Standard (FCS) file.', 'required_args': [{'name': 'Input.CSV.data.file', 'value': '', 'label': 'Input.CSV.Data.File', 'description': 'A CSV file to be converted to FCS.'}], 'default_args': [{'name': 'task_name', 'value': 'CsvToFcs', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Output.FCS.file.name', 'value': '<Input.CSV.data.file_basename>.fcs', 'label': 'Output.FCS.File.Name', 'description': 'The name of the output FCS file.'}, {'name': 'Range', 'value': 'auto', 'label': 'Range', 'description': 'Value of the $PnR keyword for all parameters. By default, this tool will try to set the best settings based on the data in the file; use \"auto\", \"auto-exact\", \"BD\", or an integer value.'}, {'name': 'Data.type', 'value': 'auto', 'label': 'Data.Type', 'description': 'Value of the $DATATYPE keyword. By default, this tool will try to set the best settings based on the data in the input CSV file.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task Cuffcompare: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Cufflinks 2.0.2 - Cuffcompare - Analyzes the transcribed fragments in an assembly', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'One or more GTF file outputs from Cufflinks.'}, {'name': 'reference.GTF', 'value': '', 'label': 'Reference.GTF', 'description': 'A reference annotation GTF'}, {'name': 'reference.genome.file', 'value': '', 'label': 'Reference.Genome.File', 'description': 'Fasta file or zip of fasta files against which your reads were aligned'}, {'name': 'additional.cuffcompare.options', 'value': '', 'label': 'Additional.Cuffcompare.Options', 'description': 'Additional options to be passed along to the Cuffcompare program at the command line.  This parameter gives you a means to specify otherwise unavailable Cuffcompare options and switches not supported by the module; check the Cufflinks manual for details.  Recommended for experts only; use this at your own discretion.'}], 'default_args': [{'name': 'task_name', 'value': 'Cuffcompare', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.prefix', 'value': '<input.file_basename>', 'label': 'Output.Prefix', 'description': 'A prefix for the module output'}, {'name': 'exclude.transcripts', 'value': 'true', 'label': 'Exclude.Transcripts', 'description': 'Whether to ignore reference transcripts that are not overlapped by any transcript in the input files.  Does not apply if no reference GTF is provided.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task Cuffdiff: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Cufflinks 2.0.2 - Finds significant changes in transcript expression, splicing, and promoter use', 'required_args': [{'name': 'aligned.files', 'value': '', 'label': 'Aligned.Files', 'description': 'A set of aligned files grouped by condition'}, {'name': 'GTF.file', 'value': '', 'label': 'GTF.File', 'description': 'A transcript annotation file in GTF/GFF format produced by cufflinks, cuffcompare, cuffmerge, or other source (such as a reference annotation GTF)'}, {'name': 'frag.bias.correct', 'value': '', 'label': 'Frag.Bias.Correct', 'description': 'Reference (FASTA/FA) for bias detection and correction algorithm'}, {'name': 'mask.file', 'value': '', 'label': 'Mask.File', 'description': 'This file tells Cuffdiff to ignore all reads that could have come from transcripts in this GTF/GFF file. It is recommended that annotated rRNA, mitochondrial transcripts, and other abundant transcripts you want to ignore in your analysis be included in this file.'}, {'name': 'additional.cuffdiff.options', 'value': '', 'label': 'Additional.Cuffdiff.Options', 'description': 'Additional options to be passed along to the Cuffdiff program at the command line.  This parameter gives you a means to specify otherwise unavailable Cuffdiff options and switches not supported by the module; check the Cufflinks manual for details.  Recommended for experts only; use this at your own discretion.'}], 'default_args': [{'name': 'task_name', 'value': 'Cuffdiff', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'time.series', 'value': 'false', 'label': 'Time.Series', 'description': 'Analyze the provided samples as a time series, rather than testing for differences between all pairs of samples.'}, {'name': 'normalization.method', 'value': 'geometric', 'label': 'Normalization.Method', 'description': 'The normalization method to be used. Choices are geometric mean (the default), upper quartile, or standard FPKM normalization. See the documentation for a discussion of these methods.'}, {'name': 'FPKM.scaling', 'value': 'compatible-hits', 'label': 'FPKM.Scaling', 'description': 'Controls how Cuffdiff includes mapped fragments toward the number used in the FPKM denominator.  Use compatible-hits (the default) to count only those fragments compatible with some reference transcript.  Use total-hits to count all fragments, even those not compatible with a reference transcript.'}, {'name': 'multi.read.correct', 'value': 'true', 'label': 'Multi.Read.Correct', 'description': \"Instructs Cuffdiff to do use its two-stage read weighting algorithm to more accurately distribute a multi-mapped read's count contribution across the multiple loci the read mapped to.\"}, {'name': 'min.alignment.count', 'value': '10', 'label': 'Min.Alignment.Count', 'description': \"The minimum number of alignments in a locus needed to conduct significance testing on changes in that locus observed between samples.  If no testing is performed, changes in the locus are deemed not significant, and the locus's observed changes do not contribute to correction for multiple testing.\"}, {'name': 'FDR', 'value': '.05', 'label': 'Fdr', 'description': 'The allowed false discovery rate.'}, {'name': 'library.type', 'value': 'inferred', 'label': 'Library.Type', 'description': 'The library type used to generate reads. The default is inferred, meaning that no library type information is passed.'}, {'name': 'skip.diff.exp', 'value': 'false', 'label': 'Skip.Diff.Exp', 'description': 'Tells Cuffdiff to perform quantification and normalization only and to skip its differential expression calculations which are computationally expensive.  The default is to compute differential expression.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task Cufflinks: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Cufflinks 2.0.2 - Assembles transcripts, estimates abundances, and tests for differential expression and regulation in RNA-seq samples', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'Input file in SAM or BAM format'}, {'name': 'transfrag.label', 'value': '', 'label': 'Transfrag.Label', 'description': 'A label for the transfrags in the output files'}, {'name': 'GTF', 'value': '', 'label': 'Gtf', 'description': 'Reference annotation (GFF/GTF file) for isoform expression estimates'}, {'name': 'GTF.guide', 'value': '', 'label': 'GTF.Guide', 'description': 'Reference annotation (GFF/GTF file) to guide RABT assembly'}, {'name': 'mask.file', 'value': '', 'label': 'Mask.File', 'description': 'A GTF file specifying transcripts to be ignored'}, {'name': 'frag.bias.correct', 'value': '', 'label': 'Frag.Bias.Correct', 'description': 'Reference (FASTA/FA) for bias detection and correction algorithm'}, {'name': 'additional.cufflinks.options', 'value': '', 'label': 'Additional.Cufflinks.Options', 'description': 'Additional options to be passed along to the Cufflinks program at the command line.  This parameter gives you a means to specify otherwise unavailable Cufflinks options and switches not supported by the module; check the Cufflinks manual for details.  Recommended for experts only; use this at your own discretion.'}], 'default_args': [{'name': 'task_name', 'value': 'Cufflinks', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'multi.read.correct', 'value': 'false', 'label': 'Multi.Read.Correct', 'description': 'Whether to do an initial estimation procedure to more accurately weight reads mapping to multiple locations in the genome'}, {'name': 'library.type', 'value': 'inferred', 'label': 'Library.Type', 'description': 'The library type used to generate reads.'}, {'name': 'min.frags.per.transfrag', 'value': '10', 'label': 'Min.Frags.Per.Transfrag', 'description': 'Assembled transfrags supported by fewer than this many aligned RNA-Seq fragments are not reported.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task CufflinksCuffmergePipeline: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Creates multiple transcript assemblies using Cufflinks and then merges these assemblies into one', 'required_args': [{'name': 'CreateSymlinks1.input.files', 'value': '', 'label': 'CreateSymlinks1.Input.Files', 'description': 'One or more input files'}, {'name': 'CufflinksWrapperPipeline2.Cufflinks1.GTF.guide', 'value': '', 'label': 'CufflinksWrapperPipeline2.Cufflinks1.GTF.Guide', 'description': 'Reference annotation (GFF/GTF file) to guide RABT assembly'}, {'name': 'CufflinksWrapperPipeline2.Cufflinks1.mask.file', 'value': '', 'label': 'CufflinksWrapperPipeline2.Cufflinks1.Mask.File', 'description': 'A GTF file specifying transcripts to be ignored'}, {'name': 'CufflinksWrapperPipeline2.Cufflinks1.additional.cufflinks.options', 'value': '', 'label': 'CufflinksWrapperPipeline2.Cufflinks1.Additional.Cufflinks.Options', 'description': 'Additional options to be passed along to the Cufflinks program at the command line.  This parameter gives you a means to specify otherwise unavailable Cufflinks options and switches not supported by the module; check the Cufflinks manual for details.  Recommended for experts only; use this at your own discretion.'}, {'name': 'Cufflinks.cuffmerge4.reference.GTF', 'value': '', 'label': 'Cufflinks.Cuffmerge4.Reference.GTF', 'description': 'An optional reference annotation GTF. The input assemblies are merged together with the reference GTF and included in the final output. Cuffmerge will use this to attach gene names and other metadata to the merged catalog.'}, {'name': 'Cufflinks.cuffmerge4.genome.file', 'value': '', 'label': 'Cufflinks.Cuffmerge4.Genome.File', 'description': 'A file containing the genomic DNA sequences for the reference.  This should be a multi-FASTA file with all contigs present.'}], 'default_args': [{'name': 'task_name', 'value': 'CufflinksCuffmergePipeline', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'CufflinksWrapperPipeline2.Cufflinks1.library.type', 'value': 'inferred', 'label': 'CufflinksWrapperPipeline2.Cufflinks1.Library.Type', 'description': 'The library type used to generate reads.'}, {'name': 'CufflinksWrapperPipeline2.Cufflinks1.min.frags.per.transfrag', 'value': '10', 'label': 'CufflinksWrapperPipeline2.Cufflinks1.Min.Frags.Per.Transfrag', 'description': 'Assembled transfrags supported by fewer than this many aligned RNA-Seq fragments are not reported.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task Cuffmerge: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Cufflinks 2.0.2: Cuffmerge - Merge several Cufflinks assemblies', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'GTF files to be merged'}, {'name': 'reference.GTF', 'value': '', 'label': 'Reference.GTF', 'description': 'An optional reference annotation GTF. The input assemblies are merged together with the reference GTF and included in the final output. Cuffmerge will use this to attach gene names and other metadata to the merged catalog.'}, {'name': 'genome.file', 'value': '', 'label': 'Genome.File', 'description': 'A file containing the genomic DNA sequences for the reference.  This should be a multi-FASTA file with all contigs present.'}], 'default_args': [{'name': 'task_name', 'value': 'Cuffmerge', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task CummeRbund.GeneSetReport: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '[Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] Cuffdiff visualization package providing plots based on a user-provided list of genes.', 'required_args': [{'name': 'cuffdiff.input', 'value': '', 'label': 'Cuffdiff.Input', 'description': 'A Cuffdiff job, a previous CummeRbund job, or a cuffData.db file from a previous CummeRbund job.'}, {'name': 'geneset.file', 'value': '', 'label': 'Geneset.File', 'description': 'A text file containing the list of genes for this report, one per line.  Each of these can be represented by a gene symbol (short name), gene ID, isoform_id, tss_group_id, or cds_id.'}, {'name': 'selected.conditions', 'value': '', 'label': 'Selected.Conditions', 'description': 'Specifies the conditions (samples) to be used in the plots.  This should be a comma-separated list of conditions, using the same names as in the upstream Cuffdiff job; leave this blank to use all conditions.  If blank, no pairwise comparison plots will be generated.'}], 'default_args': [{'name': 'task_name', 'value': 'CummeRbund.GeneSetReport', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.format', 'value': 'pdf', 'label': 'Output.Format', 'description': 'The output file format.'}, {'name': 'feature.level', 'value': 'genes', 'label': 'Feature.Level', 'description': 'Feature level for the report.'}, {'name': 'report.as.aggregate', 'value': 'yes', 'label': 'Report.As.Aggregate', 'description': 'Controls whether reporting should be done for individual replicates or aggregate condition/sample values.  The default is to use aggregate sample values.  Note that the Volcano plot always shows aggregate samples.'}, {'name': 'log.transform', 'value': 'yes', 'label': 'Log.Transform', 'description': 'Whether or not to log transform the FPKM values.  Note that the FPKM values are always log transformed for the Volcano plots and that the Scatter plot will use a log2 rather than log10 transformation.  For the Expression Plot and Expression Barplot, this directs that the y-axis will be drawn on a log10 scale.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task CummeRbund.QcReport: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '[Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] Cuffdiff visualization package providing plots and tables related to QC and Global Stats.', 'required_args': [{'name': 'cuffdiff.input', 'value': '', 'label': 'Cuffdiff.Input', 'description': 'A Cuffdiff job, a previous CummeRbund job, or a cuffData.db file from a previous CummeRbund job.'}], 'default_args': [{'name': 'task_name', 'value': 'CummeRbund.QcReport', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.format', 'value': 'pdf', 'label': 'Output.Format', 'description': 'The output file format.'}, {'name': 'feature.level', 'value': 'genes', 'label': 'Feature.Level', 'description': 'Feature level for the report'}, {'name': 'report.as.aggregate', 'value': 'yes', 'label': 'Report.As.Aggregate', 'description': 'Controls whether reporting should be done for individual replicates or aggregate condition/sample values.  The default is to use aggregate sample values.  Controls whether reporting should be done for individual replicates or aggregate condition/sample values.  The default is to use aggregate sample values.  Note that the Dispersion, FPKM.SCV, and MA plots always show aggregate samples.'}, {'name': 'log.transform', 'value': 'yes', 'label': 'Log.Transform', 'description': 'Whether or not to log transform FPKM values.  Note that the FPKM values are always log transformed for the FPKM.SCV plot.  Log transformation does not apply to the Dispersion, MDS, and PCA plots.'}, {'name': 'pca.x', 'value': 'PC2', 'label': 'Pca.X', 'description': 'Indicates which principal component is to be presented on the x-axis of the PCA plot.  Must differ from pca.y.'}, {'name': 'pca.y', 'value': 'PC3', 'label': 'Pca.Y', 'description': 'Indicates which principal component is to be presented on the y-axis of the PCA plot.  Must differ from pca.x.'}, {'name': 'attempt.to.merge.names', 'value': 'no', 'label': 'Attempt.To.Merge.Names', 'description': 'Should the module attempt to merge gene names into the text reports?  Depending on the GTF file used in the Cuffdiff run, the text reports may instead contain tracking ID references, particularly if a Cuffmerge merged.gtf file was used.  Note that this setting does not affect the plots.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task CummeRbund.SelectedConditionsReport: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '[**Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] Cuffdiff visualization package providing plots based on a user-provided list of conditions.', 'required_args': [{'name': 'cuffdiff.input', 'value': '', 'label': 'Cuffdiff.Input', 'description': 'A Cuffdiff job, a previous CummeRbund job, or a cuffData.db file from a previous CummeRbund job.'}, {'name': 'selected.conditions', 'value': '', 'label': 'Selected.Conditions', 'description': 'Specifies the conditions (samples) to be used in the plots.  This should be a comma-separated list of conditions, using the same names as in the upstream Cuffdiff job; leave this blank to use all conditions.  If blank, no pairwise comparison plots will be generated.'}], 'default_args': [{'name': 'task_name', 'value': 'CummeRbund.SelectedConditionsReport', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.format', 'value': 'pdf', 'label': 'Output.Format', 'description': 'The output file format.'}, {'name': 'feature.level', 'value': 'genes', 'label': 'Feature.Level', 'description': 'Feature level for the report.'}, {'name': 'report.as.aggregate', 'value': 'yes', 'label': 'Report.As.Aggregate', 'description': 'Controls whether reporting should be done for individual replicates or aggregate condition/sample values.  The default is to use aggregate sample values.  Note that the Volcano plot always shows aggregate samples.'}, {'name': 'log.transform', 'value': 'yes', 'label': 'Log.Transform', 'description': 'Whether or not to log transform the FPKM values.  Note that the FPKM values are always log transformed for the Volcano plots and that the Scatter plot will use a log2 rather than log10 transformation.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task CummeRbund.SelectedGeneReport: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '[**Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] Cuffdiff visualization package providing plots based on a single user-specified gene.', 'required_args': [{'name': 'cuffdiff.input', 'value': '', 'label': 'Cuffdiff.Input', 'description': 'A Cuffdiff job, a previous CummeRbund job, or a cuffData.db file from a previous CummeRbund job.'}, {'name': 'feature.id', 'value': '', 'label': 'Feature.Id', 'description': 'The gene or feature of interest.  This can be a gene symbol (short name), gene ID, isoform_id, tss_group_id, or cds_id.'}, {'name': 'selected.conditions', 'value': '', 'label': 'Selected.Conditions', 'description': 'Specifies the conditions (samples) to be used in the plots.  This should be a comma-separated list of conditions, using the same names as in the upstream Cuffdiff job; leave this blank to use all conditions.  The Similarity plots will always operate across all conditions.'}], 'default_args': [{'name': 'task_name', 'value': 'CummeRbund.SelectedGeneReport', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'find.similar', 'value': '10', 'label': 'Find.Similar', 'description': 'Optionally, find and plot the top genes (up to this count) with an expression profile most similar to the given gene of interest.  If blank, this will be skipped.    '}, {'name': 'output.format', 'value': 'pdf', 'label': 'Output.Format', 'description': 'The output file format.'}, {'name': 'feature.level', 'value': 'genes', 'label': 'Feature.Level', 'description': 'Feature level for the report.  Note that the Similarity plots will always be generated at the \"genes\" level.'}, {'name': 'report.as.aggregate', 'value': 'yes', 'label': 'Report.As.Aggregate', 'description': 'Controls whether reporting should be done for individual replicates or aggregate condition/sample values.  The default is to use aggregate sample values.  Note that the Similarity plots always show aggregate samples.'}, {'name': 'log.transform', 'value': 'yes', 'label': 'Log.Transform', 'description': 'Whether or not to log transform the FPKM values.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task CytoscapeViewer: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Visualizes a gene network', 'required_args': [{'name': 'dataset.file', 'value': '', 'label': 'Dataset.File', 'description': 'Dataset file - .res, .gct'}, {'name': 'adjacency.matrix.file', 'value': '', 'label': 'Adjacency.Matrix.File', 'description': 'Adjacency matrix - .adj'}], 'default_args': [{'name': 'task_name', 'value': 'CytoscapeViewer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task DAPPLE: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'DAPPLE', 'required_args': [{'name': 'Input.File', 'value': '', 'label': 'Input.File', 'description': 'File containing a list of SNPs, genes or regions. Each input should be on a new line.'}, {'name': 'Keyword', 'value': '', 'label': 'Keyword', 'description': ''}], 'default_args': [{'name': 'task_name', 'value': 'DAPPLE', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Permute.Length', 'value': '0', 'label': 'Permute.Length', 'description': 'To increase speed, this is adaptive permutation (see README)'}, {'name': 'Plot', 'value': 'true', 'label': 'Plot', 'description': ''}, {'name': 'nearest.gene', 'value': 'true', 'label': 'Nearest.Gene', 'description': 'Do you want to use the gene nearest to the SNP (rather than grabbing all genes in the LD wingspan)? Only relevant for SNP inputs.'}, {'name': 'genome.build', 'value': '18', 'label': 'Genome.Build', 'description': 'Which set of SNPs should we use? hg18, hg19, 1KG?'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task DeIdentifyFCS: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'DeIdentifyFCS an FCS data file; remove keywords from a list or matching a regular expression; useful to anonymize FCS data files and/or to remove specific (e.g., clinical) information ', 'required_args': [{'name': 'Input.FCS.data.file', 'value': '', 'label': 'Input.FCS.Data.File', 'description': 'An FCS data file to be de-identified.'}, {'name': 'Remove.keywords', 'value': '', 'label': 'Remove.Keywords', 'description': \"Comma-separated list of keyword names (escape comma by '\\\\' if part of a keyword name) or regular expression to match keywords to be removed.\"}], 'default_args': [{'name': 'task_name', 'value': 'DeIdentifyFCS', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Is.regular.expression.used', 'value': 'No', 'label': 'Is.Regular.Expression.Used', 'description': \"Is a regular expression used in the list of keywords to remove? If not (default) a simple comma-separated list of keyword names is expected. In this list, comma may be escaped by '\\\\' if part of a keyword name, e.g., this \\\\, is in the keyword.\"}, {'name': 'Output.FCS.file.name', 'value': '<Input.FCS.data.file_basename>.deidentified.<Input.FCS.data.file_extension>', 'label': 'Output.FCS.File.Name', 'description': 'Output FCS file name.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task DependencyBiomarkerDotplot: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Creates a dotplot of Achilles gene dependency scores. The dots are colored based on the biomarker status of the cell line (i.e. mutation in gene or no mutation in gene). This module was created for use in the Education Outreach high school computational curriculum. ', 'required_args': [{'name': 'Achilles.Gene.Dependency.Data.File', 'value': '', 'label': 'Achilles.Gene.Dependency.Data.File', 'description': 'Format is a csv file, with cell lines in rows and genes in columns. '}, {'name': 'Biomarker.Data.File', 'value': '', 'label': 'Biomarker.Data.File', 'description': 'Format is a csv file, with cell lines in rows and genes in columns. '}, {'name': 'Gene.1', 'value': '', 'label': 'Gene.1', 'description': 'The gene name should match exactly to the column names in the data files and are case-sensitive. \\n'}, {'name': 'Gene.#2', 'value': '', 'label': 'Gene.#2', 'description': 'The gene name should match exactly to the column names in the data files and are case-sensitive.'}, {'name': 'Gene.#3', 'value': '', 'label': 'Gene.#3', 'description': 'The gene name should match exactly to the column names in the data files and are case-sensitive.'}, {'name': 'Gene.#4', 'value': '', 'label': 'Gene.#4', 'description': 'The gene name should match exactly to the column names in the data files and are case-sensitive.'}, {'name': 'Gene.#5', 'value': '', 'label': 'Gene.#5', 'description': 'The gene name should match exactly to the column names in the data files and are case-sensitive.'}], 'default_args': [{'name': 'task_name', 'value': 'DependencyBiomarkerDotplot', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task DESeq2: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'A Differential Expression workflow for RNA-Seq data based on the DESeq2 package from Bioconductor ', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'A GCT file containing raw RNA-Seq counts, such as is produced by MergeHTSeqCounts'}, {'name': 'cls.file', 'value': '', 'label': 'Cls.File', 'description': 'A categorical CLS file specifying the phenotype classes for the samples in the GCT file.  This should contain exactly two classes with the control specified first.'}, {'name': 'confounding.variable.cls.file', 'value': '', 'label': 'Confounding.Variable.Cls.File', 'description': 'A categorical CLS file specifying an additional confounding variable, mapped to the input file samples.  Use this for a two-factor comparison.'}], 'default_args': [{'name': 'task_name', 'value': 'DESeq2', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file.base', 'value': '<input.file_basename>', 'label': 'Output.File.Base', 'description': 'The base name of the output file(s). File extensions will be added automatically.'}, {'name': 'qc.plot.format', 'value': 'skip', 'label': 'Qc.Plot.Format', 'description': 'Choose a file format for the QC plots (or skip them completely)'}, {'name': 'fdr.threshold', 'value': '0.1', 'label': 'Fdr.Threshold', 'description': 'FDR threshold to use in filtering before creating the reports of top up-regulated and down-regulated genes '}, {'name': 'top.N.count', 'value': '20', 'label': 'Top.N.Count', 'description': 'Max gene count to use when creating the reports of top up-regulated and down-regulated genes '}, {'name': 'random.seed', 'value': '779948241', 'label': 'Random.Seed', 'description': 'Seed to use for randomization'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task DownloadURL: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Downloads a file from a URL', 'required_args': [{'name': 'URL', 'value': '', 'label': 'Url', 'description': 'The URL to retrieve the file from'}, {'name': 'output.file', 'value': '', 'label': 'Output.File', 'description': 'The name of the output file'}], 'default_args': [{'name': 'task_name', 'value': 'DownloadURL', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task ESPPredictor: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Determine which tryptic peptides will generate the highest signal in ESI-MS', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'List of peptide sequences (one per line, no column header)'}], 'default_args': [{'name': 'task_name', 'value': 'ESPPredictor', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task ExpressionFileCreator: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Creates a RES or GCT file from a set of Affymetrix CEL files.  For IVT arrays only; use AffySTExpressionFileCreator for ST arrays.', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'A zip file of CEL files'}, {'name': 'quantile.normalization', 'value': '', 'label': 'Quantile.Normalization', 'description': '(GCRMA and RMA only) Whether to normalize data using quantile normalization'}, {'name': 'background.correct', 'value': '', 'label': 'Background.Correct', 'description': '(RMA only) Whether to background correct using RMA background correction'}, {'name': 'value.to.scale.to', 'value': '', 'label': 'Value.To.Scale.To', 'description': '(median/mean scaling only) The value to scale to.'}, {'name': 'clm.file', 'value': '', 'label': 'Clm.File', 'description': 'A tab-delimited text file containing one scan, sample, and class per line'}, {'name': 'cdf.file', 'value': '', 'label': 'Cdf.File', 'description': 'Custom CDF file. Leave blank to use default internally provided CDF file (custom cdf file is not implemented for GCRMA).'}], 'default_args': [{'name': 'task_name', 'value': 'ExpressionFileCreator', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'method', 'value': 'RMA', 'label': 'Method', 'description': 'The method to use. Note that dchip and MAS5 will not work with ST arrays.'}, {'name': 'compute.present.absent.calls', 'value': 'no', 'label': 'Compute.Present.Absent.Calls', 'description': 'Whether to compute Present/Absent calls'}, {'name': 'normalization.method', 'value': 'median scaling', 'label': 'Normalization.Method', 'description': '(MAS5 only) The normalization method to apply after expression values are computed. The column having the median of the means is used as the reference unless the parameter value to scale to is given.'}, {'name': 'annotate.probes', 'value': 'yes', 'label': 'Annotate.Probes', 'description': 'Whether to annotate probes with the gene symbol and description'}, {'name': 'output.file', 'value': '<input.file_basename>', 'label': 'Output.File', 'description': 'The base name of the output file(s)'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task ExprToGct: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Converts an EXPR file to GCT format', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'Input file in EXPR format'}], 'default_args': [{'name': 'task_name', 'value': 'ExprToGct', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.prefix', 'value': '<input.file_basename>', 'label': 'Output.Prefix', 'description': 'A label that will be used to name the GCT output file.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task ExtractColumnNames: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Lists the sample descriptors from a .res file.', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'Input filename - .res'}], 'default_args': [{'name': 'task_name', 'value': 'ExtractColumnNames', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output', 'value': '<input.filename_basename>', 'label': 'Output', 'description': 'Name of the output file (a .txt extension will be appended)'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task ExtractComparativeMarkerResults: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Creates a derived dataset and feature list file from the results of ComparativeMarkerSelection', 'required_args': [{'name': 'comparative.marker.selection.filename', 'value': '', 'label': 'Comparative.Marker.Selection.Filename', 'description': 'The results from ComparativeMarkerSelection - .odf'}, {'name': 'dataset.filename', 'value': '', 'label': 'Dataset.Filename', 'description': 'The dataset file used to select markers - .gct, .res, Dataset'}, {'name': 'statistic', 'value': '', 'label': 'Statistic', 'description': 'The statistic to filter features on'}, {'name': 'min', 'value': '', 'label': 'Min', 'description': 'Select features with statistic >= min'}, {'name': 'max', 'value': '', 'label': 'Max', 'description': 'Select features with statistic <= max'}, {'name': 'number.of.neighbors', 'value': '', 'label': 'Number.Of.Neighbors', 'description': 'Number of neighbors to select by score in each direction'}], 'default_args': [{'name': 'task_name', 'value': 'ExtractComparativeMarkerResults', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'base.output.name', 'value': '<comparative.marker.selection.filename_basename>.filt', 'label': 'Base.Output.Name', 'description': 'The base name for the output files'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task ExtractFCSDataset: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Extract one or more FCS datasets from an FCS data file', 'required_args': [{'name': 'Input.data.file', 'value': '', 'label': 'Input.Data.File', 'description': 'The input FCS data file; one or more data sets will be extracted from this file.'}, {'name': 'Output.file.name', 'value': '', 'label': 'Output.File.Name', 'description': 'The name for the output file. Uses the input data file name (for example, name.fcs) as the default, and adds _datasetIndex to the name (e.g., name_datasetIndex.fcs).'}, {'name': 'Dataset.indexes', 'value': '', 'label': 'Dataset.Indexes', 'description': 'A list of data set indexes to be extracted from the data file (e.g., 1-3, 5, 8). The first data set has the index 1, the second the index 2, etc. All data sets will be extracted if no indexes are provided.'}], 'default_args': [{'name': 'task_name', 'value': 'ExtractFCSDataset', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task ExtractFCSKeywords: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Extracts keyword(s) value(s) from a Flow Cytometry Standard (FCS) file.', 'required_args': [{'name': 'Input.FCS.data.file', 'value': '', 'label': 'Input.FCS.Data.File', 'description': 'The input FCS data file from which the module will extract keyword/value pairs.'}], 'default_args': [{'name': 'task_name', 'value': 'ExtractFCSKeywords', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Keywords.output.mode', 'value': 'CSV', 'label': 'Keywords.Output.Mode', 'description': 'The file format to be used to store the extracted keywords/value pairs.'}, {'name': 'Keywords.output.file.name', 'value': '<Input.FCS.data.file_basename>', 'label': 'Keywords.Output.File.Name', 'description': 'The name for the file in which the keyword/value pairs are stored.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task ExtractFCSParameters: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Extract specified parameters (dimensions) from an FCS data file.', 'required_args': [{'name': 'Input.FCS.data.file', 'value': '', 'label': 'Input.FCS.Data.File', 'description': 'The input FCS data file from which to extract parameters.'}, {'name': 'Parameters.to.extract', 'value': '', 'label': 'Parameters.To.Extract', 'description': 'A list of parameters to extract; parameters may be specified by their indexes (e.g., 1-3, 5, 8) or by their short names (e.g., FSC-H, SSC-H, FL1-A, FL3-H).'}, {'name': 'Output.file.name', 'value': '', 'label': 'Output.File.Name', 'description': 'The output file name. The default uses the input file name as the base.'}], 'default_args': [{'name': 'task_name', 'value': 'ExtractFCSParameters', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task ExtractRowNames: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Extracts the row names from a .res, .gct, or .odf file.', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'input file - .res, .gct, .odf, file from which the row names are extracted'}], 'default_args': [{'name': 'task_name', 'value': 'ExtractRowNames', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output', 'value': '<input.filename_basename>.row.names.txt', 'label': 'Output', 'description': 'output filename containing the row names - .txt'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task FastQC: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '[**Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] Provides quality control metrics on raw sequence data', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'RNA-seq reads file in FASTQ (bz2 and gz compressed files are supported), SAM, or BAM format.'}, {'name': 'input.format', 'value': '', 'label': 'Input.Format', 'description': 'Bypasses the normal sequence file format detection and forces the program to use the specified format.'}, {'name': 'contaminant.file', 'value': '', 'label': 'Contaminant.File', 'description': 'FastQC screens overrepresented sequences (> 0.1% of total) against a set of possible contaminants consisting of all the primers and adapters used by the various sequencing platforms.  The \"Overrepresented sequences\" report generated by FastQC will indicate when an overrepresented sequence corresponds to a known contaminant.  This optional contaminant file provides a means of adding more sequences to FastQC\\'s built-in list of known contaminants. See module documentation for a description and example of this file\\'s format.'}], 'default_args': [{'name': 'task_name', 'value': 'FastQC', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'kmer.size', 'value': '5', 'label': 'Kmer.Size', 'description': 'The length of Kmer to look for in the Kmer content module. The specified Kmer length must be between 2 and 10. Default length is 5.'}, {'name': 'extract.output', 'value': 'true', 'label': 'Extract.Output', 'description': 'Whether to output an uncompressed version of the report. Set this to yes to view the report directly from within GenePattern.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task FCMFeatureExtraction: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Calculation of features from clustered flow cytometry data.', 'required_args': [{'name': 'Input.CSV.files', 'value': '', 'label': 'Input.CSV.Files', 'description': 'The set of CSV files with clustering results (of a single data file). May be a .zip file, a directory, or a text file listing all input file paths (one file per line).'}], 'default_args': [{'name': 'task_name', 'value': 'FCMFeatureExtraction', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Calculate.number.of.cells', 'value': 'TRUE', 'label': 'Calculate.Number.Of.Cells', 'description': 'Calculate and output the number of cells in each sub-population?'}, {'name': 'Calculate.percentage.of.cells', 'value': 'TRUE', 'label': 'Calculate.Percentage.Of.Cells', 'description': 'Calculate and output the percentage (proportion) of cells in each sub-population?'}, {'name': 'Calculate.means', 'value': 'TRUE', 'label': 'Calculate.Means', 'description': 'Calculate mean parameter values (e.g. Mean Fluorescence Intensities) for all parameters of each sub-population?'}, {'name': 'Output.CSV.file.name', 'value': 'features.csv', 'label': 'Output.CSV.File.Name', 'description': 'The name of the output CSV file.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task FCMSinglePanelQC: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Single panel flow cytometry data quality control', 'required_args': [{'name': 'Input.FCS.data.files', 'value': '', 'label': 'Input.FCS.Data.Files', 'description': 'The set of input FCS files to be tested.'}], 'default_args': [{'name': 'task_name', 'value': 'FCMSinglePanelQC', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Perform.cell.number.test', 'value': 'TRUE', 'label': 'Perform.Cell.Number.Test', 'description': 'Run the cell number control test.'}, {'name': 'Perform.boundary.events.test', 'value': 'TRUE', 'label': 'Perform.Boundary.Events.Test', 'description': 'Run the boundary events test.'}, {'name': 'Perform.timeflow.test', 'value': 'TRUE', 'label': 'Perform.Timeflow.Test', 'description': 'Run the timeflow test, i.e., check whether the flow rate was steady and uninterrupted.'}, {'name': 'Perform.timeline.test', 'value': 'TRUE', 'label': 'Perform.Timeline.Test', 'description': 'Run the timeline test, i.e., detect unexpected non-random acquisition of fluorescence intensities.'}, {'name': 'Zip.output', 'value': 'TRUE', 'label': 'Zip.Output', 'description': 'Zip the resulting report together?'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task FCSClean: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Clean up an FCS file by marking events affected by problems during data acquisition', 'required_args': [{'name': 'Input.FCS', 'value': '', 'label': 'Input.FCS', 'description': 'The FCS file to be cleaned up, i.e., bad events/cells will be marked by a high value of the added Good_vs_Bad channel.'}, {'name': 'Markers', 'value': '', 'label': 'Markers', 'description': 'A comma-separated list of dimensions (flow cytometry parameters/channels) to be checked. The module accepts both, a list of marker names as well as a list of parameter indexes. All dimensions but scatters and time will be used if the Markers parameter is not provided.'}], 'default_args': [{'name': 'task_name', 'value': 'FCSClean', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Output.FCS', 'value': '<Input.FCS_basename>.cleaned.fcs', 'label': 'Output.FCS', 'description': 'The name of the output FCS file.'}, {'name': 'Bin.Size', 'value': '0.01', 'label': 'Bin.Size', 'description': 'The size of a bin as proportion of cells in the FCS file, typically between 0.01 and 0.2.'}, {'name': 'Cell.Cutoff', 'value': '500', 'label': 'Cell.Cutoff', 'description': 'The number of cells to be used for a cut off, typically between 300 and 800, see documentation for more details.'}, {'name': 'Cutoff', 'value': 'median', 'label': 'Cutoff', 'description': 'The method of how cutoff is performed by the algorithm. Typically, a median works well. Alternatively, you may specify a value from the [0,1] interval as a cutoff percentile, or a value greater than 1, which will be assumed as a direct specification of the cutoff threshold.'}, {'name': 'Max.Fc', 'value': '1.3', 'label': 'Max.Fc', 'description': 'Relative increase from mean norm of presumed good data below which data is still presumed to be good.'}, {'name': 'Announce', 'value': 'TRUE', 'label': 'Announce', 'description': 'Shall we print information about identifying vs. not identifying any bad events to the standard output?'}, {'name': 'Diagnostic', 'value': 'FALSE', 'label': 'Diagnostic', 'description': 'Shall we include a diagnostics plot?'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task FCSNormalization: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Per-channel basis normalization of flow cytometry data.', 'required_args': [{'name': 'Input.FCS.files', 'value': '', 'label': 'Input.FCS.Files', 'description': 'The set of input FCS files to be normalized.'}, {'name': 'Channels.to.normalize', 'value': '', 'label': 'Channels.To.Normalize', 'description': 'Comma-separated list of parameter short names to normalize.'}], 'default_args': [{'name': 'task_name', 'value': 'FCSNormalization', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Apply.ASinH', 'value': 'TRUE', 'label': 'Apply.ASinH', 'description': 'Apply the asinh transformation prior normalization?'}, {'name': 'Create.report', 'value': 'TRUE', 'label': 'Create.Report', 'description': 'Create additional histogram channel report?'}, {'name': 'Landmarks.number', 'value': '1', 'label': 'Landmarks.Number', 'description': 'Number of landmarks to be used for normalization.'}, {'name': 'Zip.output', 'value': 'TRUE', 'label': 'Zip.Output', 'description': 'Zip the resulting normalized FCS data files together?'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task FcsToCsv: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Converts a Flow Cytometry Standard (FCS) file to a Comma Separated Values (CSV) file.', 'required_args': [{'name': 'Input.FCS.data.file', 'value': '', 'label': 'Input.FCS.Data.File', 'description': 'An FCS data file to be converted to CSV.'}], 'default_args': [{'name': 'task_name', 'value': 'FcsToCsv', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Output.CSV.file.name', 'value': '<Input.FCS.data.file_basename>.csv', 'label': 'Output.CSV.File.Name', 'description': 'The name of the output CSV file.'}, {'name': 'Use.full.names', 'value': 'TRUE', 'label': 'Use.Full.Names', 'description': 'Whether to try to extract full parameter names from the FCS file.'}, {'name': 'Output.keywords.file.name', 'value': '<Input.FCS.data.file_basename>', 'label': 'Output.Keywords.File.Name', 'description': 'The file name to store keyword value pairs extracted from the FCS file.'}, {'name': 'Output.keywords.mode', 'value': 'CSV', 'label': 'Output.Keywords.Mode', 'description': 'How to store keywords.'}, {'name': 'Channel.to.scale.conversion', 'value': 'TRUE', 'label': 'Channel.To.Scale.Conversion', 'description': 'Whether to convert log channel values to linear scale.'}, {'name': 'Precision', 'value': 'FALSE', 'label': 'Precision', 'description': 'Whether to use regular (32bit, recommended) or double precision.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task FeatureSummaryViewer: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Feature Summary Viewer', 'required_args': [{'name': 'feature.filename', 'value': '', 'label': 'Feature.Filename', 'description': 'odf file with or without features to summarize'}], 'default_args': [{'name': 'task_name', 'value': 'FeatureSummaryViewer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task FileSplitter: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Splits a file into equal-sized chunks, given the number of lines per output file', 'required_args': [{'name': 'inputFile', 'value': '', 'label': 'InputFile', 'description': 'The input file, to be split into chunks'}, {'name': 'numFiles', 'value': '', 'label': 'NumFiles', 'description': \"[Integer, >=1], The maximum number of output files to split the file into. When both 'numFiles' and 'numLines' are set, treat this as the maximum number of output files to create, even if the files have more than the requested number of lines\"}, {'name': 'numLines', 'value': '', 'label': 'NumLines', 'description': '[Integer, >=1], The number of lines in each output file.'}], 'default_args': [{'name': 'task_name', 'value': 'FileSplitter', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task FilterFilelist: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'a helper module which generates a filtered filelist from the input filelist and some filter parameters.', 'required_args': [{'name': 'input.filelist', 'value': '', 'label': 'Input.Filelist', 'description': 'the initial input filelist, a file containing a list of filepaths'}, {'name': 'includePattern', 'value': '', 'label': 'IncludePattern', 'description': 'an optional list of filename patterns to match, e.g. *.cls,*.gct'}, {'name': 'errorOnEmptySet', 'value': '', 'label': 'ErrorOnEmptySet', 'description': 'Whether the module should output stderr if the output result is empty'}], 'default_args': [{'name': 'task_name', 'value': 'FilterFilelist', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'type', 'value': 'f', 'label': 'Type', 'description': 'Include files, directories, or both files and directories in the output filelist'}, {'name': 'output.filename', 'value': '<input.filelist_basename>.filtered.<input.filelist_extension>', 'label': 'Output.Filename', 'description': 'the name of the output file, if none specified, write to stdout'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task FilterLowshRNAs: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Removes shRNAs with low read counts in original DNA plasmid pool. Outputs a GCT file containing filtered shRNA data and a GCT file containing the mean read count for each shRNA in the original DNA plasmid pool. ', 'required_args': [{'name': 'GCT.Files', 'value': '', 'label': 'GCT.Files', 'description': 'Please enter one or more GCT files.'}, {'name': 'File.Prefix', 'value': '', 'label': 'File.Prefix', 'description': ''}], 'default_args': [{'name': 'task_name', 'value': 'FilterLowshRNAs', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task Find.differentially.expressed.genes.in.RNASeq.data: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'This pipeline is used in the GenomeSpace recipe, \"Find differentially expressed genes in RNA-Seq data\". The recipe can be found at: http://www.genomespace.org/support/guides/recipes/sections/find-differentially-expressed-genes-in-rna-seq-data.', 'required_args': [{'name': 'ConvertLineEndings1.input.filename', 'value': '', 'label': 'ConvertLineEndings1.Input.Filename', 'description': 'The input file (any non-binary file format)'}, {'name': 'TopHat2.reads.pair.1', 'value': '', 'label': 'TopHat2.Reads.Pair.1', 'description': 'Unpaired reads file or first mate for paired reads. One or more files containing reads in FASTA or FASTQ format (bz2 and gz compressed files are supported).'}, {'name': 'TopHat4.reads.pair.1', 'value': '', 'label': 'TopHat4.Reads.Pair.1', 'description': 'Unpaired reads file or first mate for paired reads. One or more files containing reads in FASTA or FASTQ format (bz2 and gz compressed files are supported).'}], 'default_args': [{'name': 'task_name', 'value': 'Find.differentially.expressed.genes.in.RNASeq.data', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task Find.subnetworks.of.differentially.expressed.genes.and.identify.associated.biological.functions: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'This pipeline is used in the GenomeSpace recipe, \"Find subnetworks of differentially expressed genes and identify associated biological functions\". The recipe can be found at: http://www.genomespace.org/support/guides/recipes/sections/find-subnetworks-of-differentially-expressed-genes-and-identify-associated-biological-functions.', 'required_args': [{'name': 'PreprocessDataset1.input.filename', 'value': '', 'label': 'PreprocessDataset1.Input.Filename', 'description': 'Input filename - .res, .gct, .odf'}, {'name': 'ComparativeMarkerSelection2.cls.file', 'value': '', 'label': 'ComparativeMarkerSelection2.Cls.File', 'description': 'The class file - .cls'}], 'default_args': [{'name': 'task_name', 'value': 'Find.subnetworks.of.differentially.expressed.genes.and.identify.associated.biological.functions', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task FLAMEChooseOptimalClusterNumber: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Determines the optimal number of clusters for each sample, using user-specified criterion', 'required_args': [{'name': 'mixture.model', 'value': '', 'label': 'Mixture.Model', 'description': 'A .zip file containing mixture modeling results from the FLAMEMixtureModel module'}], 'default_args': [{'name': 'task_name', 'value': 'FLAMEChooseOptimalClusterNumber', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'method', 'value': 'SWR', 'label': 'Method', 'description': 'Criterion used for determining optimal number of clusters'}, {'name': 'seed', 'value': '123456', 'label': 'Seed', 'description': 'An integer; the seed fixes random computational processes during computation. This does not need to be changed under most circumstances.'}, {'name': 'output.prefix', 'value': '<mixture.model_basename>', 'label': 'Output.Prefix', 'description': 'A prefix for output files'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task FLAMEContourDataGenerator: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Generates data for drawing FLAME contour plots', 'required_args': [{'name': 'membership.file', 'value': '', 'label': 'Membership.File', 'description': 'A membership.txt or member.txt file output from one of the FLAME modules. '}, {'name': 'parameters.file', 'value': '', 'label': 'Parameters.File', 'description': 'A parameters.txt file output from one of the FLAME modules. '}, {'name': 'channels.to.plot', 'value': '', 'label': 'Channels.To.Plot', 'description': 'A comma separated list of the index or names of three channels/antibodies to plot'}], 'default_args': [{'name': 'task_name', 'value': 'FLAMEContourDataGenerator', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.prefix', 'value': '<parameters.file_basename>', 'label': 'Output.Prefix', 'description': 'A prefix for output files'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task FLAMEContourViewer.Pipeline: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Pipeline which runs the FLAMECountorDataGenerator and the FLAMEViewer', 'required_args': [{'name': 'FLAMEContourDataGenerator1.membership.file', 'value': '', 'label': 'FLAMEContourDataGenerator1.Membership.File', 'description': 'A membership.txt or member.txt file output from one of the FLAME modules.'}, {'name': 'FLAMEContourDataGenerator1.parameters.file', 'value': '', 'label': 'FLAMEContourDataGenerator1.Parameters.File', 'description': 'A parameters.txt file output from one of the FLAME modules.'}, {'name': 'FLAMEContourDataGenerator1.channels.to.plot', 'value': '', 'label': 'FLAMEContourDataGenerator1.Channels.To.Plot', 'description': 'A comma separated list of the index or names of three channels/antibodies to plot'}], 'default_args': [{'name': 'task_name', 'value': 'FLAMEContourViewer.Pipeline', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'FLAMEContourDataGenerator1.output.prefix', 'value': 'flame_contour', 'label': 'FLAMEContourDataGenerator1.Output.Prefix', 'description': 'A prefix for output files'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task FLAMEMetacluster: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Matches populations/clusters across all samples in the dataset, assembles a table of all parameters in all samples', 'required_args': [{'name': 'optimal.g.mixture.model', 'value': '', 'label': 'Optimal.G.Mixture.Model', 'description': 'A .zip file containing the optimal mixture modeling result of each sample from the output of the FLAMEChooseOptimalClusterNumber module.'}, {'name': 'sample.class', 'value': '', 'label': 'Sample.Class', 'description': ''}], 'default_args': [{'name': 'task_name', 'value': 'FLAMEMetacluster', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'estimate.mode', 'value': 'F', 'label': 'Estimate.Mode', 'description': 'Used only for skew distributions. Whether to estimate the mode for each cluster. Must be the same value used in FLAMEMixtureModel.'}, {'name': 'estimation.increment', 'value': '1', 'label': 'Estimation.Increment', 'description': 'Used only for skew distributions and when estimate mode is set to yes. The smaller the increment, the more accurate the estimation, but the slower the estimation step. Must be the same value used in FLAMEMixtureModel.'}, {'name': 'output.intermediate.results', 'value': 'F', 'label': 'Output.Intermediate.Results', 'description': 'Choose whether to output the intermediate metaclustering results, such as within-class matching results.'}, {'name': 'output.prefix', 'value': '<optimal.g.mixture.model_basename>', 'label': 'Output.Prefix', 'description': 'A prefix for output files.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task FLAMEMixtureModel: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Clusters and estimates population parameters, using preprocessed flow cytometric data', 'required_args': [{'name': 'preprocessed.data', 'value': '', 'label': 'Preprocessed.Data', 'description': 'A .zip file containing preprocessed flow sample files in tab-delimited .txt format.'}, {'name': 'channels.to.cluster', 'value': '', 'label': 'Channels.To.Cluster', 'description': 'A comma-separated list of either channel numbers (e.g., 1, 2, 3, 7) or channel names (e.g., CD4, CD45RA, ZAP70); these denote which antibodies in the (preprocessed) data sample to be used for clustering.'}], 'default_args': [{'name': 'task_name', 'value': 'FLAMEMixtureModel', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'g.min', 'value': '3', 'label': 'G.Min', 'description': 'An integer; minimal number of clusters to be fit for each sample.'}, {'name': 'g.max', 'value': '10', 'label': 'G.Max', 'description': 'An integer; maximal number of clusters to be fit for each sample. g max should be greater or equal to g min.'}, {'name': 'density', 'value': 'skewt', 'label': 'Density', 'description': 'Density distribution to be used for clustering'}, {'name': 'estimate.mode', 'value': 'F', 'label': 'Estimate.Mode', 'description': 'Used only for skew distributions. Whether to estimate the mode for each cluster.  Must be the same value used in FLAMEMixtureModel.'}, {'name': 'estimation.increment', 'value': '1', 'label': 'Estimation.Increment', 'description': 'Used only for skew distributions and when estimate mode is set to yes. The smaller the increment, the more accurate the estimation, but the slower the estimation step. Must be the same value used in FLAMEMixtureModel.'}, {'name': 'seed', 'value': '123456', 'label': 'Seed', 'description': 'An integer; the seed fixes random computational processes during computation. This does not need to be changed under most circumstances.'}, {'name': 'output.prefix', 'value': '<preprocessed.data_basename>', 'label': 'Output.Prefix', 'description': 'A prefix for output files.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task FLAMEPreprocess: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Performs preprocessing of .fcs or .txt flow cytometric data files', 'required_args': [{'name': 'dataset', 'value': '', 'label': 'Dataset', 'description': 'A .zip file containing flow files in .txt or .fcs format. (The files should be all .txt or all .fcs, but not both.)'}, {'name': 'channels', 'value': '', 'label': 'Channels', 'description': 'A comma-separated list of channel numbers indicating which columns (antibodies) to keep (e.g., 1, 2, 3, 7).'}, {'name': 'channel.names', 'value': '', 'label': 'Channel.Names', 'description': 'A comma-separated list of channel/antibody names that corresponds to the channel numbers (e.g., FSC, SSC, CD4, CD45RA).'}], 'default_args': [{'name': 'task_name', 'value': 'FLAMEPreprocess', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'file.type', 'value': 'fcs', 'label': 'File.Type', 'description': 'File type of input sample data.'}, {'name': 'data.scale', 'value': '262144', 'label': 'Data.Scale', 'description': 'The original data scale. 18-bit (default) or 4-decade.'}, {'name': 'remove.dead', 'value': 'F', 'label': 'Remove.Dead', 'description': 'Whether to attempt removing dead cells from the samples.'}, {'name': 'scatter.channels', 'value': '1,2', 'label': 'Scatter.Channels', 'description': 'A comma-separated list of scatter channel numbers (e.g., 1, 2, 3).'}, {'name': 'transformation', 'value': 'logicle', 'label': 'Transformation', 'description': 'The transformation to apply'}, {'name': 'logicle.cofactor', 'value': '3', 'label': 'Logicle.Cofactor', 'description': 'A cofactor that tunes the logicle transformation. The cofactor is usually greater than 0 and less than 10.'}, {'name': 'arsinh.cofactor', 'value': '250', 'label': 'Arsinh.Cofactor', 'description': 'A cofactor that tunes the arsinh transformation. The cofactor is 0 usually between 0 exclusively and a few thousand.'}, {'name': 'output.prefix', 'value': '<dataset_basename>', 'label': 'Output.Prefix', 'description': 'A prefix for output files'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task FLAMEPreviewTransformation: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Evaluates data transformations based on one representative sample', 'required_args': [{'name': 'data.file', 'value': '', 'label': 'Data.File', 'description': 'A flow file in .txt or .fcs format for one sample.'}, {'name': 'channels', 'value': '', 'label': 'Channels', 'description': 'A comma-separated list of channel numbers indicating which columns (antibodies) to keep (e.g., 1, 2, 3, 7).'}, {'name': 'channel.names', 'value': '', 'label': 'Channel.Names', 'description': 'A comma-separated list of channel or antibody names that corresponds to the channel numbers (e.g., FSC, SSC, CD4, CD45RA).'}], 'default_args': [{'name': 'task_name', 'value': 'FLAMEPreviewTransformation', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'file.type', 'value': 'fcs', 'label': 'File.Type', 'description': 'File type of input sample data.'}, {'name': 'data.scale', 'value': '262144', 'label': 'Data.Scale', 'description': 'The original data scale. 18-bit (default) or 4-decade.'}, {'name': 'scatter.channels', 'value': '1,2', 'label': 'Scatter.Channels', 'description': 'A comma-separated list of scatter channel numbers (e.g., 1, 2, 3).'}, {'name': 'transformation', 'value': 'logicle', 'label': 'Transformation', 'description': 'The transformation(s) to apply'}, {'name': 'logicle.low', 'value': '0.5', 'label': 'Logicle.Low', 'description': 'The cofactor that tunes the logicle transformation is usually between 0 and 10 exclusively. Specify the lower limit of the cofactor range to examine.'}, {'name': 'logicle.high', 'value': '5', 'label': 'Logicle.High', 'description': 'The cofactor that tunes the logicle transformation is usually greater than 0 and less than 10. Specify the upper limit of the cofactor range to examine.'}, {'name': 'arsinh.low', 'value': '1', 'label': 'Arsinh.Low', 'description': 'The cofactor that tunes the arsinh transformation is usually between 0 and a few thousand. Specify the lower limit of the cofactor range to examine.'}, {'name': 'arsinh.high', 'value': '1000', 'label': 'Arsinh.High', 'description': 'The cofactor that tunes the arsinh transformation is usually between 0 and a few thousand. Specify the upper limit of the cofactor range to examine.'}, {'name': 'output.prefix', 'value': '<data.file_basename>', 'label': 'Output.Prefix', 'description': 'A prefix for output files'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task FLAMEViewer: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Visualizes results from the FLAME method', 'required_args': [{'name': 'membership.file', 'value': '', 'label': 'Membership.File', 'description': 'A membership.txt file output from the FLAMEContourDataGenerator module. This membership file must match with the contour plot file.'}, {'name': 'contour.plot.file', 'value': '', 'label': 'Contour.Plot.File', 'description': 'The contour plot output file from the FLAMEContourDataGenerator module. This contour plot file must match with the membership file.'}], 'default_args': [{'name': 'task_name', 'value': 'FLAMEViewer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task FlowClustClassifyFCS: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Classify flow cytometry data using the FlowClust algorithm.', 'required_args': [{'name': 'Input.FCS.data.file', 'value': '', 'label': 'Input.FCS.Data.File', 'description': 'The FCS file to be clustered.'}, {'name': 'K', 'value': '', 'label': 'K', 'description': 'The number of populations that the algorithm shall be looking for. If you do not know the K value, you can use the SuggestNumberOfPopulations module to estimate it.'}, {'name': 'Channels', 'value': '', 'label': 'Channels', 'description': 'A comma-separated list of parameter short names ($PnN values) that shall be used to classify the populations. All parameters will be used by default, i.e., if no value for the Channels parameter is provided. The Time parameter is excluded automatically.'}], 'default_args': [{'name': 'task_name', 'value': 'FlowClustClassifyFCS', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'CSV.file.name', 'value': '<Input.FCS.data.file_basename>', 'label': 'CSV.File.Name', 'description': 'The base file name to split the results into CSV files, each containing a single population.'}, {'name': 'CLS.file.name', 'value': '<Input.FCS.data.file_basename>.cls', 'label': 'CLS.File.Name', 'description': 'The name of the CLS output file, which will store the population number for each event.'}, {'name': 'Quantile', 'value': '0.9', 'label': 'Quantile', 'description': 'Outlier quantile. Events outside the specified quantile region will be classified as outliers and be assigned to class 0.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task FlowFingerprinting: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'takes a flow set and generates a description of the multivariate probability distribution function of its flow cytometry data in the form of a fingerprint', 'required_args': [{'name': 'Input.FCS.data.files', 'value': '', 'label': 'Input.FCS.Data.Files', 'description': 'The FCS file(s) to be fingerprinted.'}, {'name': 'Model.FCS.data.file', 'value': '', 'label': 'Model.FCS.Data.File', 'description': 'Training data (FCS file) for the underlying model. If none, the input FCS data is used for creating a default model.'}, {'name': 'Model.parameters', 'value': '', 'label': 'Model.Parameters', 'description': 'A comma-separated list of dimensions/parameters/channels that shall be used for the underlying model. Use either parameter names or indexes. All dimensions except for Time will be used if this parameter is not provided.'}, {'name': 'Model.number.of.recursions', 'value': '', 'label': 'Model.Number.Of.Recursions', 'description': 'Number of times the FCS training data will be subdivided.'}, {'name': 'Model.sample.size', 'value': '', 'label': 'Model.Sample.Size', 'description': 'Used to specify the per flowFrame sample size of the data to use in model generation. If NULL, all of the data in x is used. Setting this to a smaller number will speed up processing, at the cost of accuracy.'}, {'name': 'Workflow.parameter.file', 'value': '', 'label': 'Workflow.Parameter.File', 'description': 'A CSV file containing parameters if module is used in pipeline.'}], 'default_args': [{'name': 'task_name', 'value': 'FlowFingerprinting', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Model.dequantize', 'value': '0', 'label': 'Model.Dequantize', 'description': 'Option to make all events in the training set unique.'}, {'name': 'Output.QC.screening.fingerprint.stack', 'value': '0', 'label': 'Output.QC.Screening.Fingerprint.Stack', 'description': 'Graphical output in form of \"stack\" plot.'}, {'name': 'Output.QC.screening.gate.consistency', 'value': '0', 'label': 'Output.QC.Screening.Gate.Consistency', 'description': 'Graphical output in form of \"qc\" plot.'}, {'name': 'Output.QC.screening.plate', 'value': '1', 'label': 'Output.QC.Screening.Plate', 'description': 'Graphical output in form of \"plate\" plot.'}, {'name': 'Output.csv.file', 'value': 'fingerprints.csv', 'label': 'Output.Csv.File', 'description': 'Name for output CSV file.'}, {'name': 'Output.file.transformation', 'value': '0', 'label': 'Output.File.Transformation', 'description': 'Transformation used on data for CSV output file. \"raw\" is the number of events in each bin, \"normalized\" is the ratio of the number of events divided by the expected number (the expected number of events is calculated by dividing the total number of events in the instance used to make the fingerprint by the number of features in the fingerprint). \"log2norm\" is the log2 of the normalized values.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task FlowMeansCluster: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Cluster flow cytometry data using the FlowMeans algorithm.', 'required_args': [{'name': 'Input.FCS.data.file', 'value': '', 'label': 'Input.FCS.Data.File', 'description': 'The FCS file to be clustered.'}, {'name': 'Dimensions', 'value': '', 'label': 'Dimensions', 'description': 'A comma-separated list of dimensions (parameters/channels) that shall be used for clustering. Use either parameter names or indexes. All dimensions except for Time will be used if this parameter is not provided.'}, {'name': 'Dimensions.to.transform', 'value': '', 'label': 'Dimensions.To.Transform', 'description': 'A comma-separated list of dimensions (parameters/channels) that shall transformed as specified by previous parameter. This will be ignored if no transformation is specified above. If this parameter is not provided and transformation is specified above, the algorithm will use heuristics to identify parameters that shall be transformed. These heuristics are based on how parameters are stored in the FCS file, their resolution and their name. Again, you can use either parameter names or parameter indexes to specify dimensions to transform.'}, {'name': 'Maximum.number.of.clusters', 'value': '', 'label': 'Maximum.Number.Of.Clusters', 'description': 'Maximum number of clusters that FlowMeans will try. This value will be estimated automatically if the value of this parameter is not provided.'}, {'name': 'Number.of.clusters', 'value': '', 'label': 'Number.Of.Clusters', 'description': 'Number of clusters that FlowMeans shall search for. This parameter cannot be combined with the \"Maximum number of clusters\". Also, the number of clusters will be estimated automatically if this parameter is not provided.'}, {'name': 'Max.points.for.covariance', 'value': '', 'label': 'Max.Points.For.Covariance', 'description': 'The maximum number of data points (events) used for calculating the covariance. All data points will be used if this value is not provided.'}, {'name': 'Max.points.for.kernel.density', 'value': '', 'label': 'Max.Points.For.Kernel.Density', 'description': 'The maximum number of data points (events) used for counting the modes using kernel density estimation. All data points will be used if this value is not provided.'}, {'name': 'Seed', 'value': '', 'label': 'Seed', 'description': 'Seed value for the random generator; provide any consistent value to obtain reproducible results.'}], 'default_args': [{'name': 'task_name', 'value': 'FlowMeansCluster', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Transformation', 'value': 'ASinH', 'label': 'Transformation', 'description': 'Which transformation to apply on the data prior clustering. Fluorescence channels are usually better visualized and clustered using a transformation; the ASinH transformation produces good results on most data.'}, {'name': 'Maximum.number.of.iterations', 'value': '50', 'label': 'Maximum.Number.Of.Iterations', 'description': 'The maximum number of iterations allowed within the K-means component of the clustering.'}, {'name': 'Number.of.initialization.sets', 'value': '10', 'label': 'Number.Of.Initialization.Sets', 'description': 'The number of random sets used for initialization.'}, {'name': 'Distance.used', 'value': 'TRUE', 'label': 'Distance.Used', 'description': 'The distance being used within the FlowMeans algorithm.'}, {'name': 'Standardize.data', 'value': 'TRUE', 'label': 'Standardize.Data', 'description': 'Should each data dimension be standardized (i.e, transformed to a [0,1] interval) before clustering. This usually does not harm and is important if data in dimensions used for clustering covers significantly different ranges for different dimensions.   '}, {'name': 'Distance.matrix.update', 'value': 'Mahalanobis', 'label': 'Distance.Matrix.Update', 'description': 'How to update the distance matrix at each merging iteration. Note that updating using Mahalanobis distance is not possible if Euclidean distance is set in the \"Distance used\" above.  '}, {'name': 'Transform.residuals', 'value': 'FALSE', 'label': 'Transform.Residuals', 'description': 'Indicate whether or not to transform residuals to orthogonal distance.   '}, {'name': 'Add.noise', 'value': 'TRUE', 'label': 'Add.Noise', 'description': 'Should a little bit of uniform noise be added to the data to prevent singularity issues?'}, {'name': 'Draw.plots', 'value': 'TRUE', 'label': 'Draw.Plots', 'description': 'Should plots of clustered data be created for quick visual inspection of the results?'}, {'name': 'Print.debug.info', 'value': 'FALSE', 'label': 'Print.Debug.Info', 'description': 'Should additional debugging information be provided (e.g., this may be useful to debug cases where clustering does not work properly for certain data files).'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task FlowMergeCluster: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Cluster flow cytometry data using FlowMerge.', 'required_args': [{'name': 'Input.FCS.data.file', 'value': '', 'label': 'Input.FCS.Data.File', 'description': 'The FCS file to be clustered.'}, {'name': 'Dimensions', 'value': '', 'label': 'Dimensions', 'description': 'A comma-separated list of dimensions (parameters/channels) that shall be used for clustering. Use either parameter names or indexes. All dimensions except for Time will be used if this parameter is not provided.'}, {'name': 'Dimensions.to.transform', 'value': '', 'label': 'Dimensions.To.Transform', 'description': 'A comma-separated list of dimensions (parameters/channels) that shall transformed as specified by previous parameter. This will be ignored if no transformation is specified above. If this parameter is not provided and transformation is specified above, the algorithm will use heuristics to identify parameters that shall be transformed. These heuristics are based on how parameters are stored in the FCS file, their resolution and their name. Again, you can use either parameter names or parameter indexes to specify dimensions to transform.'}], 'default_args': [{'name': 'task_name', 'value': 'FlowMergeCluster', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Transformation', 'value': 'ASinH', 'label': 'Transformation', 'description': 'Which transformation to apply on the data prior clustering. Fluorescence channels are usually better visualized and clustered using a transformation; the ASinH transformation produces good results on most data.'}, {'name': 'Range.for.number.of.clusters', 'value': '1-10', 'label': 'Range.For.Number.Of.Clusters', 'description': 'The range for the number of subpopulations (clusters) that FlowMerge shall search for. FlowMerge will try to pick the best number of clusters from the specified range, which shall be provided in the min-max format, where both, mix and max are integers and min is smaller than max, e.g., 1-10.'}, {'name': 'Estimate.degrees.of.freedom', 'value': '0', 'label': 'Estimate.Degrees.Of.Freedom', 'description': 'An indication whether to estimate the degrees of freedom used for the t distribution when modelling data.'}, {'name': 'Degrees.of.freedom', 'value': '4', 'label': 'Degrees.Of.Freedom', 'description': 'The degrees of freedom used for the t distribution when modelling data. Provided value will be ignored if estimation is requested by the Estimate degrees of freedom parameter. Gaussian distribution will be used if Degrees of freedom are not provided and estimation is not requested.'}, {'name': 'Number.of.computing.nodes', 'value': '1', 'label': 'Number.Of.Computing.Nodes', 'description': \"How many nodes (e.g., processors, cores) to use if you wish to run the analysis in a parallel mode? Enter '1' if you wish do NOT want to use the parallel mode. Enter a number higher than 1 if your server/cluster has multiple computers/processors/cores and you want to utilize several of these for FlowMerge clustering.\"}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task Fpkm_trackingToGct: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Takes a Cufflinks FPKM_tracking file and converts it to GCT format.', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'Input file in FPKM_TRACKING format'}], 'default_args': [{'name': 'task_name', 'value': 'Fpkm_trackingToGct', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'row.labels', 'value': 'Tracking ID', 'label': 'Row.Labels', 'description': 'Select row labels to be written to GCT file'}, {'name': 'row.descriptions', 'value': 'Locus', 'label': 'Row.Descriptions', 'description': 'Select row descriptions to be written to GCT file'}, {'name': 'filter.rows', 'value': 'yes', 'label': 'Filter.Rows', 'description': 'Whether to filter rows with invalid row names (ie., \"-\")'}, {'name': 'output.prefix', 'value': '<input.file_basename>', 'label': 'Output.Prefix', 'description': 'A label that will be used to name the GCT output file.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task FPmatching: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Matches Fluidigm fingerprint report to reference set of CCLE SNP6.0 array data.\\nTakes old Fluidigm software report or new PED style report (saved as .csv).\\nAlso backwards compatible with Sequenom reports (Combine 100k and AffySNP6.0 reports together into one file first).', 'required_args': [{'name': 'platereport', 'value': '', 'label': 'Platereport', 'description': 'Should be .csv format'}, {'name': 'platemapping', 'value': '', 'label': 'Platemapping', 'description': 'Should be .csv format'}], 'default_args': [{'name': 'task_name', 'value': 'FPmatching', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'format', 'value': 'new', 'label': 'Format', 'description': 'New PED style report OR Old Fluidigm software format OR Sequenom style report (expects combined 100k and SNP6.0 file)'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task Gct2Arff: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Description: Convert a gene pattern .gct / .cls file pair into an .arff file', 'required_args': [{'name': 'gct.file', 'value': '', 'label': 'Gct.File', 'description': 'The dataset features - .gct'}, {'name': 'cls.file', 'value': '', 'label': 'Cls.File', 'description': 'The dataset class labels - .cls'}], 'default_args': [{'name': 'task_name', 'value': 'Gct2Arff', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<gct.file_basename>', 'label': 'Output.File', 'description': 'The output file base name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task GctToPcl: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Convert .gct to .pcl', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'GCT file to convert'}], 'default_args': [{'name': 'task_name', 'value': 'GctToPcl', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.filename', 'value': '<input.filename_basename>.pcl', 'label': 'Output.Filename', 'description': 'PCL file that will be produced'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task GENE_E: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Launches GENE-E with input file(s)', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'Dataset to be viewed in GENE-E'}, {'name': 'sample.information.or.class.file', 'value': '', 'label': 'Sample.Information.Or.Class.File', 'description': 'The cls or sample info file'}], 'default_args': [{'name': 'task_name', 'value': 'GENE_E', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task GeneListSignificanceViewer: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Views the results of marker analysis', 'required_args': [{'name': 'inputfilename', 'value': '', 'label': 'Inputfilename', 'description': 'odf format file from marker analysis (GeneNeighbors or ClassNeighbors)'}], 'default_args': [{'name': 'task_name', 'value': 'GeneListSignificanceViewer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task GeneNeighbors: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Select genes that most closely resemble a continuous profile (e.g., another gene)', 'required_args': [{'name': 'data.filename', 'value': '', 'label': 'Data.Filename', 'description': 'data file - .gct, .res, .odf'}, {'name': 'gene.accession', 'value': '', 'label': 'Gene.Accession', 'description': 'reference gene accession from data file to find neighbors for'}], 'default_args': [{'name': 'task_name', 'value': 'GeneNeighbors', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'num.neighbors', 'value': '50', 'label': 'Num.Neighbors', 'description': 'number of neighbors to find'}, {'name': 'marker.list.file', 'value': '<data.filename_basename>.markerlist', 'label': 'Marker.List.File', 'description': 'output filename for analysis results - .odf file'}, {'name': 'marker.dataset.file', 'value': '<data.filename_basename>.markerdata', 'label': 'Marker.Dataset.File', 'description': 'output filename raw data for selected markers - .gct file'}, {'name': 'distance.metric', 'value': '3', 'label': 'Distance.Metric', 'description': 'continuous metric for finding neighbors'}, {'name': 'filter.data', 'value': 'no', 'label': 'Filter.Data', 'description': 'if no, values below will be ignored'}, {'name': 'min.threshold', 'value': '20', 'label': 'Min.Threshold', 'description': 'minimum threshold for data'}, {'name': 'max.threshold', 'value': '20000', 'label': 'Max.Threshold', 'description': 'maximum threshold for data'}, {'name': 'min.fold.diff', 'value': '5', 'label': 'Min.Fold.Diff', 'description': 'minimum fold difference for filtering genes'}, {'name': 'min.abs.diff', 'value': '50', 'label': 'Min.Abs.Diff', 'description': 'minimum absolute difference for filtering genes'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task GenePatternDocumentExtractor: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Extracts GenePattern pipelines and other embedded GenePattern data from Word 2007 documents created with the Microsoft Word Add-In for the GenePattern Reproducible Research Document.', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'The file you want to extract the data from. Note that this must be a Word 2007 (.docx) file -- Files saved as Word 2003 or earlier (2003) may be compatible with the Add-In, but will not work with the extractor.'}], 'default_args': [{'name': 'task_name', 'value': 'GenePatternDocumentExtractor', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task GEOImporter: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Imports data from the Gene Expression Omnibus (GEO)', 'required_args': [{'name': 'GEO.accession', 'value': '', 'label': 'GEO.Accession', 'description': 'GEO Series record (GSExxx) or a GEO Dataset (GDSxxx). Either a GEO accession or a GEO SOFT file must be given.'}, {'name': 'GEO.SOFT.file', 'value': '', 'label': 'GEO.SOFT.File', 'description': 'A previously downloaded GEO SOFT file or its gzipped representation. Either a GEO SOFT file or a GEO accession must be given.'}, {'name': 'ftp.proxy.server', 'value': '', 'label': 'Ftp.Proxy.Server', 'description': 'FTP Proxy server'}, {'name': 'ftp.proxy.username', 'value': '', 'label': 'Ftp.Proxy.Username', 'description': 'FTP Proxy username'}, {'name': 'ftp.proxy.password', 'value': '', 'label': 'Ftp.Proxy.Password', 'description': 'FTP Proxy password'}, {'name': 'http.proxy.server', 'value': '', 'label': 'Http.Proxy.Server', 'description': 'HTTP Proxy server'}, {'name': 'http.proxy.username', 'value': '', 'label': 'Http.Proxy.Username', 'description': 'HTTP Proxy username'}, {'name': 'http.proxy.password', 'value': '', 'label': 'Http.Proxy.Password', 'description': 'HTTP Proxy password'}], 'default_args': [{'name': 'task_name', 'value': 'GEOImporter', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'data.column.name', 'value': 'VALUE', 'label': 'Data.Column.Name', 'description': 'The name of the column that contains the expression values'}, {'name': 'output.filename', 'value': '<GEO.accession><GEO.SOFT.file_basename>', 'label': 'Output.Filename', 'description': 'The name of the output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task GetDatasetInSilico: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'downloads a compressed .tgz file from the Insilico servers and extract it', 'required_args': [{'name': 'DownloadURL1.URL', 'value': '', 'label': 'DownloadURL1.URL', 'description': 'The URL to retrieve the file from'}], 'default_args': [{'name': 'task_name', 'value': 'GetDatasetInSilico', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task GISTIC_2.0: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Genomic Identification of Significant Targets in Cancer (version 2.0.22)', 'required_args': [{'name': 'refgene.file', 'value': '', 'label': 'Refgene.File', 'description': 'The reference genome file input provides cytoband and gene location information.'}, {'name': 'seg.file', 'value': '', 'label': 'Seg.File', 'description': 'The segmentation file contains the segmented data for all the samples identified by GLAD, CBS, or some other segmentation algorithm. (See GLAD file format in the Genepattern file formats documentation.) It is a six column, tab-delimited file with an optional first line identifying the columns. Positions are in base pair units.'}, {'name': 'markers.file', 'value': '', 'label': 'Markers.File', 'description': 'The markers file identifies the marker names and positions of the markers in the original dataset (before segmentation). It is a three column, tab-delimited file with an optional header. If not already, markers are sorted by genomic position.'}, {'name': 'array.list.file', 'value': '', 'label': 'Array.List.File', 'description': 'The array list file is an optional file identifying the subset of samples to be used in the analysis. It is a one column file with an optional header. The sample identifiers listed in the array list file must match the sample names given in the segmentation file.'}, {'name': 'cnv.file', 'value': '', 'label': 'Cnv.File', 'description': 'Specifies germline CNV (or other highly variant) regions to exclude from significance analysis. There are two forms for the cnv file: one allows CNVs to be identified by marker names in a single column; the other identifies excluded regions using their genomic location.'}], 'default_args': [{'name': 'task_name', 'value': 'GISTIC_2.0', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'gene.gistic', 'value': '1', 'label': 'Gene.Gistic', 'description': 'Flag indicating that the gene GISTIC algorithm should be used to calculate the significant regions of deletion using gene locations instead of a marker locations.'}, {'name': 'amplifications.threshold', 'value': '0.1', 'label': 'Amplifications.Threshold', 'description': 'Threshold for copy number amplifications. Regions with a log2 ratio above this value are considered amplified.'}, {'name': 'deletions.threshold', 'value': '0.1', 'label': 'Deletions.Threshold', 'description': 'Threshold for copy number deletions. Regions with a log2 ratio below the negative of this value are considered deletions.'}, {'name': 'join.segment.size', 'value': '4', 'label': 'Join.Segment.Size', 'description': 'Smallest number of markers to allow in segments from the segmented data. Segments that contain a number of markers less than or equal to this number are joined to the neighboring segment that is closest in copy number.'}, {'name': 'q-value.threshold', 'value': '0.25', 'label': 'Q-Value.Threshold', 'description': 'Maximum false discovery rate for a region to be considered significant.'}, {'name': 'remove.X', 'value': '1', 'label': 'Remove.X', 'description': 'Flag indicating whether to remove data from the X-chromosome before analysis.'}, {'name': 'cap.val', 'value': '1.5', 'label': 'Cap.Val', 'description': 'Minimum and maximum cap values on analyzed data. Regions with a log2 ratio greater than the cap are set to the cap value; regions with a log2 ratio less than -cap value are set to -cap.'}, {'name': 'confidence.level', 'value': '0.90', 'label': 'Confidence.Level', 'description': 'Confidence level used to calculate the region containing a driver.'}, {'name': 'run.broad.analysis', 'value': '0', 'label': 'Run.Broad.Analysis', 'description': 'Flag indicating whether an additional broad-level analysis should be performed.'}, {'name': 'focal.length.cutoff', 'value': '0.50', 'label': 'Focal.Length.Cutoff', 'description': 'Threshold used to distinguish broad from focal events, given in units of fraction of chromosome arm.'}, {'name': 'max.sample.segs', 'value': '2500', 'label': 'Max.Sample.Segs', 'description': 'Maximum number of segments allowed for a sample in the input data. Samples with more segments than this threshold are excluded from the analysis.'}, {'name': 'arm.peel', 'value': '0', 'label': 'Arm.Peel', 'description': 'Whether to perform arm level peel off. This helps separate peaks which cleans up noise.'}, {'name': 'sample.center', 'value': 'median', 'label': 'Sample.Center', 'description': 'Method for centering each sample prior to the GISTIC analysis.'}, {'name': 'gene.collapse.method', 'value': 'extreme', 'label': 'Gene.Collapse.Method', 'description': 'Method for reducing marker-level copy number data to the gene-level copy number data in the gene tables. Markers contained in the gene are used when available, otherwise the flanking marker or markers are used.'}, {'name': 'output.prefix', 'value': '<seg.file_basename>', 'label': 'Output.Prefix', 'description': 'The prefix for the output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task GISTICPreprocess: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Prepares SNP files for GISTIC', 'required_args': [{'name': 'snp.archive', 'value': '', 'label': 'Snp.Archive', 'description': 'A zipfile containing the input genotypes and signal intensities. These are .cnx files created by SNPFileCreator.'}, {'name': 'sample.info.file', 'value': '', 'label': 'Sample.Info.File', 'description': 'The sample info file is a tab-delimited text file (*.txt) containing information about the arrays.'}, {'name': 'array.list.file', 'value': '', 'label': 'Array.List.File', 'description': 'The array list file identifies the subset of samples to be used in the analysis. It is a one-column file with a header (??array??). The sample identifiers listed in the array list file must match the sample names given in the segmentation file.'}], 'default_args': [{'name': 'task_name', 'value': 'GISTICPreprocess', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'use.paired.normals', 'value': '0', 'label': 'Use.Paired.Normals', 'description': 'When set to 1, normalizes tumors against paired normals. (Default = 0; best results are usually achieved under this default setting)'}, {'name': 'quality.control.tumors', 'value': '1', 'label': 'Quality.Control.Tumors', 'description': 'When set to 1, performs histogram quality control on tumor samples and includes only those tumors with distinct copy number peaks in the output array list to be used in the GISTIC module. (Default = 1).'}, {'name': 'normal.selection.method', 'value': '1', 'label': 'Normal.Selection.Method', 'description': 'Method for selecting normals to use in sample normalization. Allowed values: 0 (use all normals), 1 (use the n normals most closely matched to the tumor??s noise profile). (Default = 1)'}, {'name': 'n.closest.n', 'value': '5', 'label': 'N.Closest.N', 'description': 'Number of normals to choose when using closest_n normalization. (Default = 5)'}, {'name': 'correct.batch.effect', 'value': '0', 'label': 'Correct.Batch.Effect', 'description': 'When set to 1, runs module to correct for marker intensity variations due to batch effect. (Default = 0)'}, {'name': 'batch.correct.minimum.size', 'value': '5', 'label': 'Batch.Correct.Minimum.Size', 'description': 'Minimum number of samples required in a batch for the batch to be eligible for batch correction. (Default = 5)'}, {'name': 'batch.correct.pvalue', 'value': '0.001', 'label': 'Batch.Correct.Pvalue', 'description': 'p-value threshold for batch effect correction. For each marker, signal intensities in each batch are compared to all others using a t-test. If the p-value for this comparison is less than the p-value threshold, the signal intensities in this batch will be brought to the mean of all other batches. (Default = 0.001)'}, {'name': 'verbosity', 'value': '0', 'label': 'Verbosity', 'description': 'Set verbosity level (stand-alone version only). At a verbosity level of 0, no text indicating program status is written to sdout. At a level of 30, the maximum amount of text is written to stdout. (Default = 0)'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task GLAD: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Gain and Loss Analysis of DNA', 'required_args': [{'name': 'copy.number.input.file', 'value': '', 'label': 'Copy.Number.Input.File', 'description': 'cn file that contains sorted SNPs and raw copy number value per probe. (note: data should not be in log2)'}], 'default_args': [{'name': 'task_name', 'value': 'GLAD', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.filename', 'value': '<copy.number.input.file_basename>.glad', 'label': 'Output.Filename', 'description': 'Name of the output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task GlobalAlignment: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'standard global sequence analysis', 'required_args': [{'name': 'subject.fasta.filename', 'value': '', 'label': 'Subject.Fasta.Filename', 'description': 'filename of subject FASTA file'}, {'name': 'match', 'value': '', 'label': 'Match', 'description': '> 0'}, {'name': 'mismatch', 'value': '', 'label': 'Mismatch', 'description': '< 0'}, {'name': 'gapopen', 'value': '', 'label': 'Gapopen', 'description': '< 0'}, {'name': 'gapextend', 'value': '', 'label': 'Gapextend', 'description': '< 0'}, {'name': 'query.filename', 'value': '', 'label': 'Query.Filename', 'description': 'FASTA file'}], 'default_args': [{'name': 'task_name', 'value': 'GlobalAlignment', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<subject.fasta.filename_basename>', 'label': 'Output.File', 'description': 'filename of output'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task Golub.Slonim.1999.Nature.all.aml.pipeline: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'ALL/AML methodology, from Golub and Slonim et al., 1999', 'required_args': [], 'default_args': [{'name': 'task_name', 'value': 'Golub.Slonim.1999.Nature.all.aml.pipeline', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task GSEA: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Gene Set Enrichment Analysis', 'required_args': [{'name': 'expression.dataset', 'value': '', 'label': 'Expression.Dataset', 'description': 'Dataset file - .res, .gct'}, {'name': 'gene.sets.database', 'value': '', 'label': 'Gene.Sets.Database', 'description': 'Gene sets database from GSEA website.'}, {'name': 'gene.sets.database.file', 'value': '', 'label': 'Gene.Sets.Database.File', 'description': 'Gene sets database - .gmt, .gmx, .grp. Upload a gene set if your gene set is not listed as a choice for the gene sets database parameter.'}, {'name': 'phenotype.labels', 'value': '', 'label': 'Phenotype.Labels', 'description': 'Cls file - .cls, must be binary'}, {'name': 'target.profile', 'value': '', 'label': 'Target.Profile', 'description': 'Name of the target phenotype profile.  Only applicable if class file defines continuous labels (one or more phenotype profiles).  Leave blank if class file defines a discrete phenotype.'}, {'name': 'chip.platform.file', 'value': '', 'label': 'Chip.Platform.File', 'description': 'DNA Chip (array) annotation file from GSEA website.  Upload your own chip file if the one corresponding to your DNA Microarray platform is not listed in the drop-down menu.  A chip file is only required if collapse dataset is set to true.'}], 'default_args': [{'name': 'task_name', 'value': 'GSEA', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'number.of.permutations', 'value': '1000', 'label': 'Number.Of.Permutations', 'description': 'Number of permutations to perform'}, {'name': 'collapse.dataset', 'value': 'true', 'label': 'Collapse.Dataset', 'description': 'Select true to have GSEA collapse each probe set in the expression dataset into a single vector for the gene, which gets identified by its gene symbol.'}, {'name': 'permutation.type', 'value': 'phenotype', 'label': 'Permutation.Type', 'description': 'Type of permutations to perform'}, {'name': 'scoring.scheme', 'value': 'weighted', 'label': 'Scoring.Scheme', 'description': 'The statistic used to score hits (gene set members) and misses (non-members)'}, {'name': 'metric.for.ranking.genes', 'value': 'Signal2Noise', 'label': 'Metric.For.Ranking.Genes', 'description': 'Class separation metric - gene markers are ranked using this metric to produce the gene list'}, {'name': 'gene.list.sorting.mode', 'value': 'real', 'label': 'Gene.List.Sorting.Mode', 'description': 'Mode in which scores from the gene list should be considered'}, {'name': 'gene.list.ordering.mode', 'value': 'descending', 'label': 'Gene.List.Ordering.Mode', 'description': 'Direction in which the gene list should be ordered'}, {'name': 'max.gene.set.size', 'value': '500', 'label': 'Max.Gene.Set.Size', 'description': 'Gene sets larger than this are excluded from the analysis'}, {'name': 'min.gene.set.size', 'value': '15', 'label': 'Min.Gene.Set.Size', 'description': 'Gene sets smaller than this are excluded from the analysis'}, {'name': 'collapsing.mode.for.probe.sets.with.more.than.one.match', 'value': 'Max_probe', 'label': 'Collapsing.Mode.For.Probe.Sets.With.More.Than.One.Match', 'description': 'Collapsing mode'}, {'name': 'normalization.mode', 'value': 'meandiv', 'label': 'Normalization.Mode', 'description': 'Normalization mode'}, {'name': 'randomization.mode', 'value': 'no_balance', 'label': 'Randomization.Mode', 'description': 'Type of phenotype randomization (does not apply to gene set permutations)'}, {'name': 'omit.features.with.no.symbol.match', 'value': 'true', 'label': 'Omit.Features.With.No.Symbol.Match', 'description': 'If there is no known gene symbol match for a probe set, omit it from the collapsed dataset'}, {'name': 'make.detailed.gene.set.report', 'value': 'true', 'label': 'Make.Detailed.Gene.Set.Report', 'description': 'Create detailed gene set report (heat map, mountain plot, etc.) for every enriched gene set'}, {'name': 'median.for.class.metrics', 'value': 'false', 'label': 'Median.For.Class.Metrics', 'description': 'Use the median of each class instead of the mean for the class separation metrics'}, {'name': 'number.of.markers', 'value': '100', 'label': 'Number.Of.Markers', 'description': 'Number of markers'}, {'name': 'plot.graphs.for.the.top.sets.of.each.phenotype', 'value': '20', 'label': 'Plot.Graphs.For.The.Top.Sets.Of.Each.Phenotype', 'description': 'Plot GSEA mountain and related plots for the top sets of each phenotype'}, {'name': 'random.seed', 'value': 'timestamp', 'label': 'Random.Seed', 'description': 'Seed to use for randomization'}, {'name': 'save.random.ranked.lists', 'value': 'false', 'label': 'Save.Random.Ranked.Lists', 'description': 'Whether to save random ranked lists'}, {'name': 'output.file.name', 'value': '<expression.dataset_basename>.zip', 'label': 'Output.File.Name', 'description': 'Name of the output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task GSEALeadingEdgeViewer: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Leading Edge Viewer for GSEA results', 'required_args': [{'name': 'zip.file', 'value': '', 'label': 'Zip.File', 'description': 'zip output file from GSEA - .zip. '}], 'default_args': [{'name': 'task_name', 'value': 'GSEALeadingEdgeViewer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task GSEAPreranked: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Runs the gene set enrichment analysis against a user-supplied ranked list of genes.', 'required_args': [{'name': 'ranked.list', 'value': '', 'label': 'Ranked.List', 'description': 'Rank ordered gene list - .rnk'}, {'name': 'gene.sets.database', 'value': '', 'label': 'Gene.Sets.Database', 'description': 'Gene sets database from MSigDB website.'}, {'name': 'gene.sets.database.file', 'value': '', 'label': 'Gene.Sets.Database.File', 'description': 'Gene sets database - .gmt, .gmx, .grp. Upload a gene set if your gene set is not listed as a choice for the gene sets database parameter.'}, {'name': 'chip.platform.file', 'value': '', 'label': 'Chip.Platform.File', 'description': 'DNA Chip (array) annotation file from GSEA website.  Upload your own chip file if the one corresponding to your DNA Microarray platform is not listed in the drop-down menu.  A chip file is only required if collapse dataset is set to true.'}], 'default_args': [{'name': 'task_name', 'value': 'GSEAPreranked', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'number.of.permutations', 'value': '1000', 'label': 'Number.Of.Permutations', 'description': 'Number of permutations to perform'}, {'name': 'collapse.dataset', 'value': 'false', 'label': 'Collapse.Dataset', 'description': 'Select true to have GSEAPreranked collapse each feature set in the ranked list into a single vector for the gene, which gets identified by its gene symbol.'}, {'name': 'scoring.scheme', 'value': 'weighted', 'label': 'Scoring.Scheme', 'description': 'The statistic used to score hits (gene set members) and misses (non-members)'}, {'name': 'max.gene.set.size', 'value': '500', 'label': 'Max.Gene.Set.Size', 'description': 'Gene sets larger than this are excluded from the analysis'}, {'name': 'min.gene.set.size', 'value': '15', 'label': 'Min.Gene.Set.Size', 'description': 'Gene sets smaller than this are excluded from the analysis'}, {'name': 'normalization.mode', 'value': 'meandiv', 'label': 'Normalization.Mode', 'description': 'Normalization mode'}, {'name': 'collapsing.mode', 'value': 'Max_probe', 'label': 'Collapsing.Mode', 'description': 'Collapsing mode'}, {'name': 'omit.features.with.no.symbol.match', 'value': 'true', 'label': 'Omit.Features.With.No.Symbol.Match', 'description': 'If there is no known gene symbol match for a probe set, omit it from the collapsed dataset; only relevant when collapse dataset is set to true'}, {'name': 'make.detailed.gene.set.report', 'value': 'true', 'label': 'Make.Detailed.Gene.Set.Report', 'description': 'Create detailed gene set report (heat map, mountain plot, etc.) for every enriched gene set'}, {'name': 'num.top.sets', 'value': '20', 'label': 'Num.Top.Sets', 'description': 'Plot GSEA mountain and related plots for the top sets of each phenotype'}, {'name': 'random.seed', 'value': 'timestamp', 'label': 'Random.Seed', 'description': 'Seed to use for generating random permutations.'}, {'name': 'output.file.name', 'value': '<ranked.list_basename>.zip', 'label': 'Output.File.Name', 'description': 'Name of the output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task HAPSEG: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '[**Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] A probabilistic method to interpret bi-allelic marker data in cancer samples.', 'required_args': [{'name': 'plate.name', 'value': '', 'label': 'Plate.Name', 'description': 'Name of the sample plate'}, {'name': 'array.name', 'value': '', 'label': 'Array.Name', 'description': 'Name of the chip that was run'}, {'name': 'seg.file', 'value': '', 'label': 'Seg.File', 'description': 'Segmented copy number data file for this sample. If not provided, HAPSEG will segment the data for you.'}, {'name': 'snp.file', 'value': '', 'label': 'Snp.File', 'description': 'SNP intensity file for this sample'}, {'name': 'use.pop', 'value': '', 'label': 'Use.Pop', 'description': 'Population to run with (currently supported values: CEPH, CH, JA, YOR)'}, {'name': 'calls.file', 'value': '', 'label': 'Calls.File', 'description': 'If using a matched normal sample, a Birdseed calls file needs to be supplied'}, {'name': 'mn.sample', 'value': '', 'label': 'Mn.Sample', 'description': 'If using a matched sample, the name of that matched normal sample'}, {'name': 'clusters.file', 'value': '', 'label': 'Clusters.File', 'description': \"If calibrate.data is 'On' the user must supply a Birdseed clusters file\"}, {'name': 'prev.theta.file', 'value': '', 'label': 'Prev.Theta.File', 'description': 'An optional file storing the previous theta values'}], 'default_args': [{'name': 'task_name', 'value': 'HAPSEG', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'out.file.name', 'value': '<plate.name>_<array.name>.segdat.RData', 'label': 'Out.File.Name', 'description': 'The name of the output file.'}, {'name': 'genome.build', 'value': 'ftp://gpftp.broadinstitute.org/module_support_files/HAPSEG_1.1.1/BEAGLE/phasedBGL/hg19/', 'label': 'Genome.Build', 'description': \"Which build of the human genome to use.  Only 'hg18' and 'hg19' are currently supported.  This will select a set of phased BEAGLE files.\"}, {'name': 'platform', 'value': 'SNP_6.0', 'label': 'Platform', 'description': \"The chip type used. The supported values are currently 'SNP_250K_STY' and 'SNP_6.0'\"}, {'name': 'impute.gt', 'value': 'TRUE', 'label': 'Impute.Gt', 'description': \"If TRUE will impute genotypes via BEAGLE. We recommend this be TRUE unless there's a reason to do otherwise\"}, {'name': 'plot.segfit', 'value': 'TRUE', 'label': 'Plot.Segfit', 'description': 'If TRUE will plot JPG images of the segmentations fits'}, {'name': 'merge.small', 'value': 'TRUE', 'label': 'Merge.Small', 'description': 'If TRUE will merge small segments'}, {'name': 'merge.close', 'value': 'TRUE', 'label': 'Merge.Close', 'description': 'if TRUE will merge close segments'}, {'name': 'min.seg.size', 'value': '10', 'label': 'Min.Seg.Size', 'description': 'Minimum segment size'}, {'name': 'normal', 'value': 'FALSE', 'label': 'Normal', 'description': 'If TRUE this sample will be treated as a normal'}, {'name': 'out.p', 'value': '0.05', 'label': 'Out.P', 'description': 'Outlier probability'}, {'name': 'seg.merge.thresh', 'value': '1e-10', 'label': 'Seg.Merge.Thresh', 'description': 'Segmentation threshold'}, {'name': 'use.normal', 'value': 'FALSE', 'label': 'Use.Normal', 'description': 'If TRUE will use a matched normal sample if one is provided'}, {'name': 'drop.x', 'value': 'FALSE', 'label': 'Drop.X', 'description': 'If TRUE will remove the X chromosome from the calculation'}, {'name': 'drop.y', 'value': 'TRUE', 'label': 'Drop.Y', 'description': 'If TRUE will remove the Y chromosome from the calculation'}, {'name': 'calibrate.data', 'value': 'NA', 'label': 'Calibrate.Data', 'description': \"If 'On', will perform a calibration on the input data.  If 'Off', no calibration will be performed. If left at the default value (Inferred), the calibration status will be inferred.\"}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task HeatMapImage: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Creates a heat map from a dataset', 'required_args': [{'name': 'input.dataset', 'value': '', 'label': 'Input.Dataset', 'description': 'The input dataset - .res, .gct, .odf'}, {'name': 'rows.to.highlight', 'value': '', 'label': 'Rows.To.Highlight', 'description': 'Rows to highlight in the heat map'}, {'name': 'row.highlight.color', 'value': '', 'label': 'Row.Highlight.Color', 'description': 'The color to highlight the rows if a rows to highlight file is provided'}, {'name': 'color.palette', 'value': '', 'label': 'Color.Palette', 'description': 'Color palette consisting of one r:g:b triplet or one hex value per line'}], 'default_args': [{'name': 'task_name', 'value': 'HeatMapImage', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output', 'value': '<input.dataset_basename>', 'label': 'Output', 'description': 'The output file name -.jpeg, .png, .tiff, .bmp, .eps'}, {'name': 'output.format', 'value': 'jpeg', 'label': 'Output.Format', 'description': 'The output file format'}, {'name': 'column.size', 'value': '8', 'label': 'Column.Size', 'description': 'The size in pixels of an element along the horizontal axis'}, {'name': 'row.size', 'value': '8', 'label': 'Row.Size', 'description': 'The size in pixels of an element along the vertical axis'}, {'name': 'show.grid', 'value': 'yes', 'label': 'Show.Grid', 'description': 'Whether to show a grid between elements'}, {'name': 'grid.color', 'value': 'black', 'label': 'Grid.Color', 'description': 'The grid color'}, {'name': 'show.row.descriptions', 'value': 'no', 'label': 'Show.Row.Descriptions', 'description': 'Whether to show row descriptions'}, {'name': 'show.row.names', 'value': 'yes', 'label': 'Show.Row.Names', 'description': 'Whether to show row names'}, {'name': 'color.scheme', 'value': 'row normalized', 'label': 'Color.Scheme', 'description': 'The color scheme'}, {'name': 'use.color.gradient', 'value': 'no', 'label': 'Use.Color.Gradient', 'description': 'Whether to use a linear color gradient or discrete colors'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task HeatMapViewer: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'A configurable heat map viewer that provides users with several options for manipulating and visualizing array-based data', 'required_args': [{'name': 'dataset', 'value': '', 'label': 'Dataset', 'description': 'The dataset to display - .gct'}], 'default_args': [{'name': 'task_name', 'value': 'HeatMapViewer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task helloWorld: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Generic hello world module', 'required_args': [], 'default_args': [{'name': 'task_name', 'value': 'helloWorld', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'message', 'value': 'Hello, World!', 'label': 'Message', 'description': ''}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task helloWorld_R2.15_Rscript: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Demo R2.15_Rscript module', 'required_args': [], 'default_args': [{'name': 'task_name', 'value': 'helloWorld_R2.15_Rscript', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task HierarchicalClustering: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Hierarchical Clustering', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'input data file name - .gct, .res, .pcl'}, {'name': 'log.transform', 'value': '', 'label': 'Log.Transform', 'description': 'log-transform the data before clustering'}, {'name': 'row.center', 'value': '', 'label': 'Row.Center', 'description': 'whether to center each row (gene) in the data'}, {'name': 'row.normalize', 'value': '', 'label': 'Row.Normalize', 'description': 'whether to normalize each row (gene) in the data'}, {'name': 'column.center', 'value': '', 'label': 'Column.Center', 'description': 'whether to center each column (sample) in the data'}, {'name': 'column.normalize', 'value': '', 'label': 'Column.Normalize', 'description': 'whether to normalize each column (sample) in the data'}], 'default_args': [{'name': 'task_name', 'value': 'HierarchicalClustering', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'column.distance.measure', 'value': '2', 'label': 'Column.Distance.Measure', 'description': 'distance measure for column (sample) clustering'}, {'name': 'row.distance.measure', 'value': '0', 'label': 'Row.Distance.Measure', 'description': 'distance measure for row (gene) clustering  NOTE: Filtering beforehand is recommended since row clustering is computationally intensive.'}, {'name': 'clustering.method', 'value': 'a', 'label': 'Clustering.Method', 'description': 'hierarchical clustering method to use'}, {'name': 'output.base.name', 'value': '<input.filename_basename>', 'label': 'Output.Base.Name', 'description': 'base name for output files'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task HierarchicalClusteringImage: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Creates an image of the dendrogram generated from HierarchicalClustering', 'required_args': [{'name': 'cdt', 'value': '', 'label': 'Cdt', 'description': 'Clustered data table, contains the original data, but reordered'}, {'name': 'gtr', 'value': '', 'label': 'Gtr', 'description': 'Gene Tree file, records the order in which genes (rows) were joined (at least one of gtr and atr files are required)'}, {'name': 'atr', 'value': '', 'label': 'Atr', 'description': 'Array Tree file, records the order in which samples (columns) were joined (at least one of gtr and atr files are required)'}, {'name': 'rows.to.highlight', 'value': '', 'label': 'Rows.To.Highlight', 'description': 'Rows to highlight in the heat map'}, {'name': 'row.highlight.color', 'value': '', 'label': 'Row.Highlight.Color', 'description': 'The color to highlight the rows if a rows to highlight file is provided'}, {'name': 'color.palette', 'value': '', 'label': 'Color.Palette', 'description': 'Color palette consisting of one r:g:b triplet or one hex value per line'}], 'default_args': [{'name': 'task_name', 'value': 'HierarchicalClusteringImage', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output', 'value': '<cdt_basename>', 'label': 'Output', 'description': 'The output file name - .bmp, .jpeg, .png, .tiff'}, {'name': 'output.format', 'value': 'jpeg', 'label': 'Output.Format', 'description': 'The output file format'}, {'name': 'column.size', 'value': '8', 'label': 'Column.Size', 'description': 'The size in pixels of an element along the horizontal axis'}, {'name': 'row.size', 'value': '8', 'label': 'Row.Size', 'description': 'The size in pixels of an element along the vertical axis'}, {'name': 'show.grid', 'value': 'yes', 'label': 'Show.Grid', 'description': 'Whether to show a grid between elements'}, {'name': 'grid.color', 'value': '0:0:0', 'label': 'Grid.Color', 'description': 'The grid color'}, {'name': 'show.row.descriptions', 'value': 'no', 'label': 'Show.Row.Descriptions', 'description': 'Whether to show row descriptions'}, {'name': 'show.row.names', 'value': 'yes', 'label': 'Show.Row.Names', 'description': 'Whether to show row names'}, {'name': 'color.scheme', 'value': 'row normalized', 'label': 'Color.Scheme', 'description': 'The color scheme'}, {'name': 'use.color.gradient', 'value': 'no', 'label': 'Use.Color.Gradient', 'description': 'Whether to use a linear color gradient or discrete colors'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task HierarchicalClusteringViewer: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Views results of hierarchical clustering.', 'required_args': [{'name': 'cdt.file', 'value': '', 'label': 'Cdt.File', 'description': 'Clustered data table, contains the original data, but reordered'}, {'name': 'gtr.file', 'value': '', 'label': 'Gtr.File', 'description': 'Gene tree file, records the order in which genes (rows) were joined (at least one of gtr file and atr file is required)'}, {'name': 'atr.file', 'value': '', 'label': 'Atr.File', 'description': 'Array tree file, records the order in which samples (columns) were joined (at least one of gtr file and atr file is required)'}], 'default_args': [{'name': 'task_name', 'value': 'HierarchicalClusteringViewer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task Hu68kHu35kAtoU95: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Converts a list of Affymetrix Hu6800/Hu35KsubA probes to the corresponding Affymetrix U95Av2 probes.', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'the .pol, .tag or .grp file to be converted'}], 'default_args': [{'name': 'task_name', 'value': 'Hu68kHu35kAtoU95', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output', 'value': '<input.filename_basename>.U95', 'label': 'Output', 'description': 'name of the output file (same format as input file)'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task Identify.an.up.or.downregulated.pathway.from.expression.data: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'This pipeline is used in the GenomeSpace recipe, \"Identify an up- or down-regulated pathway from expression data\". The recipe can be found at: http://www.genomespace.org/support/guides/recipes/sections/identify-an-up-or-down-regulated-pathway-from-expression-data.', 'required_args': [{'name': 'ConvertLineEndings1.input.filename', 'value': '', 'label': 'ConvertLineEndings1.Input.Filename', 'description': 'The input file (any non-binary file format)'}, {'name': 'ComparativeMarkerSelection2.cls.file', 'value': '', 'label': 'ComparativeMarkerSelection2.Cls.File', 'description': 'The class file - .cls'}], 'default_args': [{'name': 'task_name', 'value': 'Identify.an.up.or.downregulated.pathway.from.expression.data', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task Identify.and.Validate.a.Consensus.Signature.Using.Gene.Expression.Data: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'This pipeline is used in the GenomeSpace recipe, \"Identify and Validate a Consensus Signature Using Gene Expression Data\". The recipe can be found at: http://genomespace.org/support/guides/recipes/sections/identify-and-validate-a-consensus-signature-using-gene-expression-data.', 'required_args': [{'name': 'SelectFeaturesColumns1.input.filename', 'value': '', 'label': 'SelectFeaturesColumns1.Input.Filename', 'description': 'input file - .res, .gct, .snp, .cn, .loh, .odf, or .cls file from which the slice is extracted'}, {'name': 'SelectFeaturesColumns2.input.filename', 'value': '', 'label': 'SelectFeaturesColumns2.Input.Filename', 'description': 'input file - .res, .gct, .snp, .cn, .loh, .odf, or .cls file from which the slice is extracted'}, {'name': 'SelectFeaturesRows3.list.filename', 'value': '', 'label': 'SelectFeaturesRows3.List.Filename', 'description': 'text file containing one identifier per line that specifies the rows to select. '}], 'default_args': [{'name': 'task_name', 'value': 'Identify.and.Validate.a.Consensus.Signature.Using.Gene.Expression.Data', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task Identify.and.visualize.expressed.transcripts.in.RNASeq.data: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'This pipeline is used in the GenomeSpace recipe, \"Identify and visualize expressed transcripts in RNA-Seq data\". The recipe can be found at: http://www.genomespace.org/support/guides/recipes/sections/identify-and-visualize-expressed-transcripts-in-rna-seq-data.', 'required_args': [{'name': 'ConvertLineEndings1.input.filename', 'value': '', 'label': 'ConvertLineEndings1.Input.Filename', 'description': 'The input file (any non-binary file format)'}, {'name': 'TopHat2.reads.pair.1', 'value': '', 'label': 'TopHat2.Reads.Pair.1', 'description': 'Unpaired reads file or first mate for paired reads. One or more files containing reads in FASTA or FASTQ format (bz2 and gz compressed files are supported).'}, {'name': 'TopHat2.reads.pair.2', 'value': '', 'label': 'TopHat2.Reads.Pair.2', 'description': 'Second mate for paired reads. Zero or more files in FASTA or FASTQ format (bz2 and gz compressed files are supported).'}], 'default_args': [{'name': 'task_name', 'value': 'Identify.and.visualize.expressed.transcripts.in.RNASeq.data', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task IGV: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Launches IGV with input file(s)', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'Dataset to be viewed in IGV or IGV session file'}, {'name': 'index.file', 'value': '', 'label': 'Index.File', 'description': 'The index file.'}, {'name': 'genomeId', 'value': '', 'label': 'GenomeId', 'description': 'Choose the genome for your data. IGV will launch with hg19 or the last viewed genome by default.'}, {'name': 'locus', 'value': '', 'label': 'Locus', 'description': 'Provide a locus or gene of interest for your data. IGV will launch on chr1 or the last viewed chromosome by default. (e.g. chr1:2000-3000 or egfr)'}], 'default_args': [{'name': 'task_name', 'value': 'IGV', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task IGV.js: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Launches IGV.js with input file(s)', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'Track dataset to be viewed in IGV'}, {'name': 'index.file', 'value': '', 'label': 'Index.File', 'description': 'The index file (i.e .bai, .idx, or .tbi).'}, {'name': 'genome', 'value': '', 'label': 'Genome', 'description': 'Choose the genome for your data. IGV will launch with hg19.'}, {'name': 'locus', 'value': '', 'label': 'Locus', 'description': 'Provide a locus or gene of interest for your data. IGV will launch on chr1.'}], 'default_args': [{'name': 'task_name', 'value': 'IGV.js', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task IlluminaConcatenator: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Concatenate normalized Illumina probe sets into a single GCT file', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'ZIP file of normalized GCT files output by IlluminaNormalizer'}, {'name': 'output.file.type', 'value': '', 'label': 'Output.File.Type', 'description': 'Whether to concatenate the data or detection scores'}], 'default_args': [{'name': 'task_name', 'value': 'IlluminaConcatenator', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task IlluminaDASLPipeline: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'creates a GenePattern gct file from raw Illumina scan data', 'required_args': [{'name': 'IlluminaScanExtractor1.sample.description.file', 'value': '', 'label': 'IlluminaScanExtractor1.Sample.Description.File', 'description': 'file with one sample per line, one Sentrix Position, DAP#, and Sentrix ID per DAP'}, {'name': 'IlluminaConcatenator3.output.file.type', 'value': '', 'label': 'IlluminaConcatenator3.Output.File.Type', 'description': 'Whether to concatenate the data or detection scores'}], 'default_args': [{'name': 'task_name', 'value': 'IlluminaDASLPipeline', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'IlluminaScanExtractor1.outlier.method', 'value': '> 3 Std. Deviations', 'label': 'IlluminaScanExtractor1.Outlier.Method', 'description': 'outlier detection method (only applies to extraction using bead-level data)'}, {'name': 'IlluminaNormalizer2.normalization.method', 'value': 'CubicSpline', 'label': 'IlluminaNormalizer2.Normalization.Method', 'description': 'correction method. Note that mean background subtraction does not apply when the input is a single GCT file.'}, {'name': 'IlluminaNormalizer2.discard.controls', 'value': 'no', 'label': 'IlluminaNormalizer2.Discard.Controls', 'description': 'discard top 2 and bottom 2 negative controls (applies to mean background subtraction only)'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task IlluminaExpressionFileCreator: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Creates a GCT file from a zip of Illumina IDAT files and an Illumina manifest file', 'required_args': [{'name': 'idat.zip', 'value': '', 'label': 'Idat.Zip', 'description': 'Zip of Illumina IDAT files'}, {'name': 'manifest', 'value': '', 'label': 'Manifest', 'description': 'Illumina manifest file in tab-delimited (TXT) format. A manifest file is required; either selected from this list or specified in custom manifest below.'}, {'name': 'custom.manifest', 'value': '', 'label': 'Custom.Manifest', 'description': 'Other Illumina manifest file in tab-delimited (TXT) format. A manifest file is required; either selected from the list above or specified here.'}, {'name': 'chip', 'value': '', 'label': 'Chip', 'description': 'CHIP file to map probes to genes.'}, {'name': 'custom.chip', 'value': '', 'label': 'Custom.Chip', 'description': 'Other CHIP file to map probes to genes.'}, {'name': 'clm', 'value': '', 'label': 'Clm', 'description': 'CLM file to map file names to sample names.'}], 'default_args': [{'name': 'task_name', 'value': 'IlluminaExpressionFileCreator', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<idat.zip_basename>.gct', 'label': 'Output.File', 'description': 'Output file name'}, {'name': 'background.subtraction.mode', 'value': 'false', 'label': 'Background.Subtraction.Mode', 'description': 'Perform background subtraction.'}, {'name': 'collapse.mode', 'value': 'none', 'label': 'Collapse.Mode', 'description': 'Collapse probes to genes based on the manifest or CHIP file (if provided).'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task IlluminaNormalizer: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Normalize zipped Illumina scans', 'required_args': [{'name': 'zip.or.gct.file', 'value': '', 'label': 'Zip.Or.Gct.File', 'description': 'a zip of GCT files output by IlluminaScanExtractor (6k DASL) or a GCT file (whole-genome DASL)'}], 'default_args': [{'name': 'task_name', 'value': 'IlluminaNormalizer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'normalization.method', 'value': 'CubicSpline', 'label': 'Normalization.Method', 'description': 'correction method. Note that mean background subtraction does not apply when the input is a single GCT file.'}, {'name': 'output.file', 'value': '<zip.or.gct.file_basename>_<normalization.method>', 'label': 'Output.File', 'description': 'output file name'}, {'name': 'discard.controls', 'value': 'no', 'label': 'Discard.Controls', 'description': 'discard top 2 and bottom 2 negative controls (applies to mean background subtraction only)'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task IlluminaScanExtractor: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Extract intensity values from Illumina scans', 'required_args': [{'name': 'sample.description.file', 'value': '', 'label': 'Sample.Description.File', 'description': 'file with one sample per line, one Sentrix Position, DAP#, and Sentrix ID per DAP'}], 'default_args': [{'name': 'task_name', 'value': 'IlluminaScanExtractor', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'outlier.method', 'value': '> 3 Std. Deviations', 'label': 'Outlier.Method', 'description': 'outlier detection method (only applies to extraction using bead-level data)'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task ImmPort_FLOCK_CrossSample: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'ImmPort FLOCK and CrossSample pipeline', 'required_args': [{'name': 'ImmPortFCSConvLogicleTrans1.Input.FCS.file', 'value': '', 'label': 'ImmPortFCSConvLogicleTrans1.Input.FCS.File', 'description': 'An FCS file that shall be transformed with the Logicle transformation, converted and saved as tab-delimited text file.'}, {'name': 'ImmPortColSelection2.Dimension.specification', 'value': '', 'label': 'ImmPortColSelection2.Dimension.Specification', 'description': 'A list of dimensions/columns to extract, specified by their indexes (starting from 1), e.g., 1,4,6. There should not be any white space characters (e.g, spaces, tabs, new lines, etc.) in the parameter value.'}, {'name': 'ImmPortCrossSample4.Input.file', 'value': '', 'label': 'ImmPortCrossSample4.Input.File', 'description': 'A tab-delimited text file with columns corresponding to parameters and rows to events. Such a text file may be generated from an FCS file by the ImmPortFCSConvLogicleTrans module.'}], 'default_args': [{'name': 'task_name', 'value': 'ImmPort_FLOCK_CrossSample', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task ImmPort_FLOCK_Individual_FCS: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'ImmPort FLOCK and Individual FCS pipeline', 'required_args': [{'name': 'ImmPortFCSConvLogicleTrans1.Input.FCS.file', 'value': '', 'label': 'ImmPortFCSConvLogicleTrans1.Input.FCS.File', 'description': 'An FCS file that shall be transformed with the Logicle transformation, converted and saved as tab-delimited text file.'}, {'name': 'ImmPortColSelection2.Dimension.specification', 'value': '', 'label': 'ImmPortColSelection2.Dimension.Specification', 'description': 'A list of dimensions/columns to extract, specified by their indexes (starting from 1), e.g., 1,4,6. There should not be any white space characters (e.g, spaces, tabs, new lines, etc.) in the parameter value.'}], 'default_args': [{'name': 'task_name', 'value': 'ImmPort_FLOCK_Individual_FCS', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task ImmPortColSelection: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Select a set of columns from a txt file', 'required_args': [{'name': 'Input.file', 'value': '', 'label': 'Input.File', 'description': 'A tab-delimited text file with columns corresponding to parameters and rows to events, converted from an fcs file, typically by ImmPortFCSConvLogicleTrans.'}, {'name': 'Dimension.specification', 'value': '', 'label': 'Dimension.Specification', 'description': 'A list of dimensions/columns to extract, specified by their indexes (starting from 1), e.g., 1,4,6. There should not be any white space characters (e.g, spaces, tabs, new lines, etc.) in the parameter value.'}], 'default_args': [{'name': 'task_name', 'value': 'ImmPortColSelection', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task ImmPortCrossSample: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Applying an individual FLOCK result across another FCM sample (.txt) version 1 (1-round Kmeans)', 'required_args': [{'name': 'Input.file', 'value': '', 'label': 'Input.File', 'description': 'A tab-delimited text file with columns corresponding to parameters and rows to events. Such a text file may be generated from an FCS file by the ImmPortFCSConvLogicleTrans module.'}, {'name': 'Centroid.file', 'value': '', 'label': 'Centroid.File', 'description': 'A text-file specifying the position of each of the cell populations. Such a text file is supposed to be generated by the ImmPortFLOCK module (file population_center.txt).'}], 'default_args': [{'name': 'task_name', 'value': 'ImmPortCrossSample', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Number.of.iterations', 'value': '1', 'label': 'Number.Of.Iterations', 'description': 'The number of iterations that the algorithm will use to match events with the closest population; values from the [1, 10] range are allowed.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task ImmPortCSV2TXT: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Convert a .csv file to a .txt file that FLOCK can read', 'required_args': [{'name': 'Input.CSV.data.file', 'value': '', 'label': 'Input.CSV.Data.File', 'description': 'The input CSV file that will be converted to a a tab-delimited text file; the CSV shall contain a header with parameter names.'}], 'default_args': [{'name': 'task_name', 'value': 'ImmPortCSV2TXT', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task ImmPortFCSConvLogicleTrans: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Logicle transformation of an fcs file to .txt file', 'required_args': [{'name': 'Input.FCS.file', 'value': '', 'label': 'Input.FCS.File', 'description': 'An FCS file that shall be transformed with the Logicle transformation, converted and saved as tab-delimited text file.'}], 'default_args': [{'name': 'task_name', 'value': 'ImmPortFCSConvLogicleTrans', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task ImmPortFLOCK: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Automated Gating with ImmPort FLOCK algorithm version 1.0', 'required_args': [{'name': 'Input.file', 'value': '', 'label': 'Input.File', 'description': 'A tab-delimited text file with columns corresponding to parameters and rows to events. The first row is a heading labelling parameters. Such a text file is supposed to be generated from an FCS file by the ImmPortFCSConvLogicleTrans module optionally followed by the ImmPortColSelection module (if applicable to remove unused parameters). Alternatively, this text file may be created by the ImmPortCSV2TXT module if the original data is in the CSV format.'}], 'default_args': [{'name': 'task_name', 'value': 'ImmPortFLOCK', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Number.of.bins', 'value': '0', 'label': 'Number.Of.Bins', 'description': 'The number of bins used to partition each dimension. The default value of 0 means that the algorithm will automatically determine a suitable number of bins. Any value from the [6,29] range may be provided to manually override algorithmic estimations.'}, {'name': 'Density.threshold', 'value': '0', 'label': 'Density.Threshold', 'description': 'The minimal number of events in a bin (in the hyperspace of used markers) needed to mark the bin as a dense bin. The default value of 0 means that the algorithm will automatically determine a suitable density threshold. Any value from the [3,99] range may be provided to manually override algorithmic estimations.'}, {'name': 'Maximum.number.of.populations', 'value': '30', 'label': 'Maximum.Number.Of.Populations', 'description': 'The maximum number of populations identifiable by the clustering algorithm. The default value of 30 works well in most cases but may be overridden by any value from the [5,999] range. Larger values may increase the execution time of the module.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task ImputeMissingValues.KNN: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Imputes missing values', 'required_args': [{'name': 'data.filename', 'value': '', 'label': 'Data.Filename', 'description': 'Data file (missing values are stored as NA) - .gct'}], 'default_args': [{'name': 'task_name', 'value': 'ImputeMissingValues.KNN', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'k', 'value': '10', 'label': 'K', 'description': 'Number of neighbors to be used in the imputation'}, {'name': 'rowmax', 'value': '0.5', 'label': 'Rowmax', 'description': 'The maximum percent missing data allowed in any row'}, {'name': 'colmax', 'value': '0.8', 'label': 'Colmax', 'description': 'The maximum percent missing data allowed in any column'}, {'name': 'output.file', 'value': '<data.filename_basename>.imputed.gct', 'label': 'Output.File', 'description': 'The name of the output file - .gct'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task Install.Proteogenomics.R.Packages: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Installs all packages and dependencies need to run the Protegenomics GenePattern modules', 'required_args': [], 'default_args': [{'name': 'task_name', 'value': 'Install.Proteogenomics.R.Packages', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task IntersectingIdentifiers: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Given 2 tab delimited files (res, gct, tab, odf) caculate which ids in the first intersect with the second. Outputs a file of intersecting, non-intersecting, in the first but not the second and in the second but not the first', 'required_args': [{'name': 'First.file', 'value': '', 'label': 'First.File', 'description': 'Any tab delimited file'}, {'name': 'Second.file', 'value': '', 'label': 'Second.File', 'description': 'Any tab delimited file'}], 'default_args': [{'name': 'task_name', 'value': 'IntersectingIdentifiers', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'First.file.identifier.column.index', 'value': '1', 'label': 'First.File.Identifier.Column.Index', 'description': 'Index of identifiers in first file, staring at 1'}, {'name': 'Second.file.identifier.column.index', 'value': '1', 'label': 'Second.File.Identifier.Column.Index', 'description': 'Index of identifiers in second file, staring at 1'}, {'name': 'Output.file.name.prefix', 'value': '<First.file_basename>', 'label': 'Output.File.Name.Prefix', 'description': 'Name of output file'}, {'name': 'First.file.skip.rows', 'value': '2', 'label': 'First.File.Skip.Rows', 'description': 'Number of rows to skip in the first file to avoid headers'}, {'name': 'Second.file.skip.rows', 'value': '2', 'label': 'Second.File.Skip.Rows', 'description': 'Number of rows to skip in the first file to avoid headers'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task JavaTreeView: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': \"Hierarchical clustering viewer that reads in Eisen's cdt, atr, and gtr files. A detailed description of these formats is available from http://microarray.ccgb.umn.edu/smd/html/MicroArray/help/formats.shtml\", 'required_args': [{'name': 'cdt.filename', 'value': '', 'label': 'Cdt.Filename', 'description': 'Clustered data table, contains the original data, but reordered'}, {'name': 'gtr.filename', 'value': '', 'label': 'Gtr.Filename', 'description': 'Gene Tree file, records the order in which genes (rows) were joined (at least one of gtr filename and atr filename is required)'}, {'name': 'atr.filename', 'value': '', 'label': 'Atr.Filename', 'description': 'Array Tree file, records the order in which samples (columns) were joined (at least one of gtr filename and atr filename is required)'}], 'default_args': [{'name': 'task_name', 'value': 'JavaTreeView', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task KMeansClassifyFCS: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Find clusters in a flow cytometry file with k means clustering', 'required_args': [{'name': 'Input.FCS.data.file', 'value': '', 'label': 'Input.FCS.Data.File', 'description': 'The FCS file'}, {'name': 'K', 'value': '', 'label': 'K', 'description': 'The number of populations you are looking for.'}, {'name': 'Dimensions', 'value': '', 'label': 'Dimensions', 'description': 'The channels to search for populations, separated by commas. Leave blank to search in all channels. E.g. FSC Lin, SS Lin, CD4 Log'}], 'default_args': [{'name': 'task_name', 'value': 'KMeansClassifyFCS', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'CSV.file.name', 'value': '<Input.FCS.data.file_basename>', 'label': 'CSV.File.Name', 'description': 'The base filename to split the results into csv files, each containing one population'}, {'name': 'CLS.file.name', 'value': '<Input.FCS.data.file_basename>', 'label': 'CLS.File.Name', 'description': 'A .CLS filename to store the population number for each data point'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task KMeansClustering: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Module that performs the K-Means Clustering algorithm', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'File containing data to cluster - .res, .gct, .odf'}, {'name': 'number.of.clusters', 'value': '', 'label': 'Number.Of.Clusters', 'description': 'Number of centroids (clusters)'}, {'name': 'cluster.by', 'value': '', 'label': 'Cluster.By', 'description': 'Whether to cluster by rows or columns'}], 'default_args': [{'name': 'task_name', 'value': 'KMeansClustering', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.base.name', 'value': '<input.filename_basename>_KMcluster_output', 'label': 'Output.Base.Name', 'description': 'The base output file name - .gct'}, {'name': 'seed.value', 'value': '12345', 'label': 'Seed.Value', 'description': 'Seed value for random number generator'}, {'name': 'distance.metric', 'value': '0', 'label': 'Distance.Metric', 'description': 'How to compute distance between points'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task KNN: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'K-nearest neighbors classification', 'required_args': [{'name': 'train.filename', 'value': '', 'label': 'Train.Filename', 'description': 'training data file name - .gct, .res, .odf type = Dataset, ignored if a saved model (saved.model.filename) is used'}, {'name': 'train.class.filename', 'value': '', 'label': 'Train.Class.Filename', 'description': 'class file for training data - .cls, ignored if a saved model (saved.model.filename) is used'}, {'name': 'saved.model.filename', 'value': '', 'label': 'Saved.Model.Filename', 'description': 'input KNN model file - .odf type = KNN Prediction Model'}, {'name': 'test.filename', 'value': '', 'label': 'Test.Filename', 'description': 'test data file name - .gct, .res, .odf type = Dataset'}, {'name': 'test.class.filename', 'value': '', 'label': 'Test.Class.Filename', 'description': 'class file for test data - .cls'}, {'name': 'min.std', 'value': '', 'label': 'Min.Std', 'description': 'minimum standard deviation if feature selection statistic includes min std option'}, {'name': 'feature.list.filename', 'value': '', 'label': 'Feature.List.Filename', 'description': 'features to use for prediction'}], 'default_args': [{'name': 'task_name', 'value': 'KNN', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'model.file', 'value': '<train.filename_basename>.model', 'label': 'Model.File', 'description': 'name of output KNN model file  - .odf type = KNN Prediction Model'}, {'name': 'num.features', 'value': '10', 'label': 'Num.Features', 'description': 'number of selected features if feature list filename is not specified'}, {'name': 'feature.selection.statistic', 'value': '0', 'label': 'Feature.Selection.Statistic', 'description': 'statistic to use to perform feature selection'}, {'name': 'num.neighbors', 'value': '3', 'label': 'Num.Neighbors', 'description': 'number of neighbors for KNN'}, {'name': 'weighting.type', 'value': 'distance', 'label': 'Weighting.Type', 'description': 'weighting type for neighbors'}, {'name': 'distance.measure', 'value': '1', 'label': 'Distance.Measure', 'description': 'distance measure'}, {'name': 'pred.results.file', 'value': '<test.filename_basename>.pred.odf', 'label': 'Pred.Results.File', 'description': 'name of output file for prediction results - .odf'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task KNNXValidation: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'KNN classification with Leave-One-Out Cross-Validation', 'required_args': [{'name': 'data.filename', 'value': '', 'label': 'Data.Filename', 'description': 'data file - .gct, .res, .odf type = Dataset'}, {'name': 'class.filename', 'value': '', 'label': 'Class.Filename', 'description': 'class file - .cls'}, {'name': 'min.std', 'value': '', 'label': 'Min.Std', 'description': 'minimum standard deviation if feature selection statistic includes min std option'}], 'default_args': [{'name': 'task_name', 'value': 'KNNXValidation', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'num.features', 'value': '10', 'label': 'Num.Features', 'description': 'number of selected features'}, {'name': 'feature.selection.statistic', 'value': '0', 'label': 'Feature.Selection.Statistic', 'description': 'statistic to use to perform feature selection'}, {'name': 'num.neighbors', 'value': '3', 'label': 'Num.Neighbors', 'description': 'number of neighbors for KNN'}, {'name': 'weighting.type', 'value': 'distance', 'label': 'Weighting.Type', 'description': 'weighting type for neighbors'}, {'name': 'distance.measure', 'value': '1', 'label': 'Distance.Measure', 'description': 'distance measure'}, {'name': 'pred.results.file', 'value': '<data.filename_basename>.pred.odf', 'label': 'Pred.Results.File', 'description': 'prediction results output file name .odf type = Prediction Results'}, {'name': 'feature.summary.file', 'value': '<data.filename_basename>.feat.odf', 'label': 'Feature.Summary.File', 'description': 'feature summary output file name - .odf type = Prediction Features'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task KSscore: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Computes the Kolmogorov-Smirnov score for a set of genes within an ordered list.', 'required_args': [{'name': 'query.filename', 'value': '', 'label': 'Query.Filename', 'description': 'query gene set file containing one identifier per line'}, {'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'file containing an ordered list of genes (.pol format)'}], 'default_args': [{'name': 'task_name', 'value': 'KSscore', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output', 'value': '<query.filename_basename>', 'label': 'Output', 'description': 'output file name (a .pol extension will be appended)'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task LandmarkMatch: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'A proteomics method to propogate identifed peptides across multiple MS runs', 'required_args': [{'name': 'peakList.filename', 'value': '', 'label': 'PeakList.Filename', 'description': 'Zip file containing peak lists'}, {'name': 'retentionTime.filename', 'value': '', 'label': 'RetentionTime.Filename', 'description': 'Zip file containing retention time (in this case from mapquant)'}, {'name': 'sampleInfo.filename', 'value': '', 'label': 'SampleInfo.Filename', 'description': 'CSV file with header: experiment, sample, class'}, {'name': 'globalID.filename', 'value': '', 'label': 'GlobalID.Filename', 'description': 'Global list of identified peptides'}, {'name': 'accurateMass.filename', 'value': '', 'label': 'AccurateMass.Filename', 'description': 'OPTIONAL: Specify the accurate mass table'}, {'name': 'prefitCoef.filename', 'value': '', 'label': 'PrefitCoef.Filename', 'description': 'OPTIONAL: Specify the prefit coefficients in a file'}, {'name': 'Bootstrap', 'value': '', 'label': 'Bootstrap', 'description': 'Use bootstrapping to calculate landmark match statistics'}], 'default_args': [{'name': 'task_name', 'value': 'LandmarkMatch', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'running.on.LSF', 'value': 'no', 'label': 'Running.On.LSF', 'description': 'Whether you are running on an LSF cluster version of GenePattern'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task Life.Cycle.Time.Point.Correlation: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Calculate correlation to life cycle time point of p.falciparum (derisi data)', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'Input data file name. Data in CSV format, with plasmodbID as first column '}], 'default_args': [{'name': 'task_name', 'value': 'Life.Cycle.Time.Point.Correlation', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<input.file_basename>-correlation.csv', 'label': 'Output.File', 'description': 'Name of output file that will contain the resulting correlation table. CSV format.'}, {'name': 'correlation', 'value': 'pearson', 'label': 'Correlation', 'description': 'Type of correlation to calculate'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task ListFiles: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': \"a helper module which outputs a filelist, the list of all files in the input directory, similar to the unix 'ls' command\", 'required_args': [{'name': 'dir', 'value': '', 'label': 'Dir', 'description': 'a directory'}, {'name': 'recursive', 'value': '', 'label': 'Recursive', 'description': ''}, {'name': 'includePattern', 'value': '', 'label': 'IncludePattern', 'description': 'an optional list of filename patterns, e.g. *.cls,*.gct'}, {'name': 'pruneDir', 'value': '', 'label': 'PruneDir', 'description': 'an optional list of directory name patterns to ignore, e.g. *.svn'}, {'name': 'order.by', 'value': '', 'label': 'Order.By', 'description': 'an optional flag for how to order the files'}, {'name': 'type', 'value': '', 'label': 'Type', 'description': ''}, {'name': 'output.filename', 'value': '', 'label': 'Output.Filename', 'description': 'the name of the output file, if none specified, write to stdout'}], 'default_args': [{'name': 'task_name', 'value': 'ListFiles', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task LocatePeaks: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Locates detected peaks in a spectrum.', 'required_args': [{'name': 'peaks.output.filename', 'value': '', 'label': 'Peaks.Output.Filename', 'description': 'The output from the Peaks module'}, {'name': 'locate.peaks.as', 'value': '', 'label': 'Locate.Peaks.As', 'description': 'max-location of maximum filtered value, area-center of area of the peak'}], 'default_args': [{'name': 'task_name', 'value': 'LocatePeaks', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<peaks.output.filename_basename>.locate.odf', 'label': 'Output.File', 'description': 'output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task LogicleTransformFCS: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Logicle transformation of (selected) parameters in a list mode FCS data file.', 'required_args': [{'name': 'Input.fcs.data.file', 'value': '', 'label': 'Input.Fcs.Data.File', 'description': 'FCS data file to transform.'}, {'name': 'Parameters.to.transform', 'value': '', 'label': 'Parameters.To.Transform', 'description': 'Comma-separated list of parameters to transform, parameters are specified by their short names ($PnN values); Optional. By default, all parameters with the \"FL\" prefix in their short name will be transformed.'}, {'name': 'T', 'value': '', 'label': 'T', 'description': 'T > 0; Data value at the top of the scale range to be displayed, optional. The maximum input value will be used if no explicit value for T is provided.'}, {'name': 'W', 'value': '', 'label': 'W', 'description': 'W >= 0; Linearization width in asymptotic decades; optional. The described calculation will be used to set W for each of the transformed FCS parameters if no explicit value for W is provided.'}], 'default_args': [{'name': 'task_name', 'value': 'LogicleTransformFCS', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'M', 'value': '4.5', 'label': 'M', 'description': 'M < 0; Full width of the intended Logicle display in asymptotic decades, optional. The default value is 4.5 (decades). Note also that for the resulting Logicle transformation, Logicle (T) = M.'}, {'name': 'A', 'value': '0', 'label': 'A', 'description': 'A, usually <= 0; Additional negative display range in asymptotic decades; optional. The default value is 0.'}, {'name': 'Use.Fast.Implementation', 'value': 'no', 'label': 'Use.Fast.Implementation', 'description': 'Use a fast implementation of Logicle with less precision; optional.'}, {'name': 'Output.fcs.file.name', 'value': '<Input.fcs.data.file_basename>.logicle_transformed.fcs', 'label': 'Output.Fcs.File.Name', 'description': 'The output FCS file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task LogisticRegression: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Logistic regression using clinical data and .cls file', 'required_args': [{'name': 'input.clin.data.filename', 'value': '', 'label': 'Input.Clin.Data.Filename', 'description': 'input clinical data (.txt file)'}, {'name': 'input.cls.filename', 'value': '', 'label': 'Input.Cls.Filename', 'description': 'input class information (.cls file)'}], 'default_args': [{'name': 'task_name', 'value': 'LogisticRegression', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': 'LogisticRegression_result', 'label': 'Output.File', 'description': 'output file name'}, {'name': 'response.variable', 'value': 'NA', 'label': 'Response.Variable', 'description': ''}, {'name': 'variable.continuous', 'value': 'NA', 'label': 'Variable.Continuous', 'description': 'variable.continuous'}, {'name': 'variable.category', 'value': 'NA', 'label': 'Variable.Category', 'description': '(e.g. sex,cls,grade,...)'}, {'name': 'variable.interaction.terms', 'value': 'NA', 'label': 'Variable.Interaction.Terms', 'description': '(e.g. sex:cls,sex:grade,...)'}, {'name': 'input.subgroup', 'value': 'NA', 'label': 'Input.Subgroup', 'description': '(e.g. cls:2) *only one subgroup can be specified'}, {'name': 'variable.selection', 'value': 'none', 'label': 'Variable.Selection', 'description': ''}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task LogTransformGLAD: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Log base 2 transform last column of GLAD file (for GISTIC)', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'The file to transform the last column of (glad format)'}], 'default_args': [{'name': 'task_name', 'value': 'LogTransformGLAD', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<input.filename_basename>.LogTransformed.glad', 'label': 'Output.File', 'description': 'Output filename'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task LOHPaired: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Computes LOH for paired samples', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'A .snp file that contains genotypes'}, {'name': 'sample.info.filename', 'value': '', 'label': 'Sample.Info.Filename', 'description': 'A sample info file with the paired information'}], 'default_args': [{'name': 'task_name', 'value': 'LOHPaired', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<input.filename_basename>.loh', 'label': 'Output.File', 'description': 'Output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task Lu.Getz.Miska.Nature.June.2005.mouse.lung.pipeline: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Normal/tumor classifier and kNN prediction of mouse lung samples', 'required_args': [], 'default_args': [{'name': 'task_name', 'value': 'Lu.Getz.Miska.Nature.June.2005.mouse.lung.pipeline', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task MAGEMLImportViewer: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'A visualizer to import data in MAGE-ML format into GenePattern', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'The archive file containing the MAGE-ML files - .tar, .tgz, .tar.gz, .zip, .gz'}], 'default_args': [{'name': 'task_name', 'value': 'MAGEMLImportViewer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task MAGETABImportViewer: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'A visualizer to import data in MAGE-TAB format into GenePattern', 'required_args': [{'name': 'sdrf.file', 'value': '', 'label': 'Sdrf.File', 'description': 'The sdrf file - .txt'}, {'name': 'data.files', 'value': '', 'label': 'Data.Files', 'description': 'A zip file containing MAGE-TAB data matrix files - .zip'}], 'default_args': [{'name': 'task_name', 'value': 'MAGETABImportViewer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task MapChipFeaturesGeneral: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Change (map) the features (genes) of a dataset', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'Input filename - .res'}, {'name': 'lookup.table.filename', 'value': '', 'label': 'Lookup.Table.Filename', 'description': 'Name of the look up table file with the feature mapping'}, {'name': 'table.format', 'value': '', 'label': 'Table.Format', 'description': 'Table format'}, {'name': 'direction', 'value': '', 'label': 'Direction', 'description': 'Direction of look up'}], 'default_args': [{'name': 'task_name', 'value': 'MapChipFeaturesGeneral', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.prefix', 'value': '<input.filename_basename>.fea', 'label': 'Output.Prefix', 'description': 'prefix for output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task Marker.Selection: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Marker selection, ranking and classification', 'required_args': [{'name': 'gct.file', 'value': '', 'label': 'Gct.File', 'description': 'Input data file (in gct format)'}, {'name': 'cls.file', 'value': '', 'label': 'Cls.File', 'description': 'Class file (cls format) to match with input data file'}], 'default_args': [{'name': 'task_name', 'value': 'Marker.Selection', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.prefix', 'value': '<gct.file_basename>', 'label': 'Output.Prefix', 'description': 'Prefix for output file names'}, {'name': 'fdr.threshold', 'value': '0.01', 'label': 'Fdr.Threshold', 'description': 'FDR threshold for calling statistically significant markers'}, {'name': 'gct.id.col', 'value': '2', 'label': 'Gct.Id.Col', 'description': 'Column in data file containing (gene or protein) ID'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task MClustClusterLabel: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Model based labeling/matching of clusters (mate-clustering)', 'required_args': [{'name': 'Input.clustering.results', 'value': '', 'label': 'Input.Clustering.Results', 'description': 'The input clustering results (of multiple data files) in CSV files with each clustered population stored in a separate CSV file. All these CSV files shall have the same columns (dimensions/channels) or at least, they all shall include columns specified by the dimensions parameter.'}, {'name': 'No.of.labels', 'value': '', 'label': 'No.Of.Labels', 'description': 'The number of distinct labels (populations) to look for. You may run the MClustClusterLabelBIC module to get the number of labels you should look for.'}, {'name': 'Dimensions', 'value': '', 'label': 'Dimensions', 'description': 'A comma-separated list of dimensions (parameters/channels) that shall be used for meta clustering (clustering of means of clusters). All specified dimensions shall be included in all the input clustering results CSV files. All dimensions (but Time) will be used if this parameter is not provided.'}], 'default_args': [{'name': 'task_name', 'value': 'MClustClusterLabel', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Model', 'value': 'VII', 'label': 'Model', 'description': 'Specific model used for meta clustering; see documentation for details about different models.'}, {'name': 'Uncertainty.threshold', 'value': '65', 'label': 'Uncertainty.Threshold', 'description': 'The threshold (in percentage) for uncertainty when assigning labels. Input a value greater than 50 and lower than 100. Populations where no label can be assigned with probability greater or equal to this uncertainty threshold will be left unassigned in the results.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task MClustClusterLabelBIC: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Bayesian information criterion (BIC) for model based labeling/matching of clusters; helps estimating the number of labels for MClustClusterLabel.', 'required_args': [{'name': 'Input.clustering.results', 'value': '', 'label': 'Input.Clustering.Results', 'description': 'The input clustering results (of multiple data files) in CSV files with each clustered population stored in a separate CSV file. All these CSV files shall have the same columns (dimensions/channels) or at least, they all shall include columns specified by the dimensions parameter.'}, {'name': 'Dimensions', 'value': '', 'label': 'Dimensions', 'description': 'A comma-separated list of dimensions (parameters/channels) that shall be used for meta clustering (clustering of means of clusters). All specified dimensions shall be included in all the input clustering results CSV files. All dimensions (but Time) will be used if this parameter is not provided.'}], 'default_args': [{'name': 'task_name', 'value': 'MClustClusterLabelBIC', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Range', 'value': '1-10', 'label': 'Range', 'description': 'The range of distinct labels (populations, components) to calculate BIC for. Provide the range in the min-max format, where both, mix and max are integers and min is smaller than max.'}, {'name': 'EII', 'value': 'TRUE', 'label': 'Eii', 'description': 'Include the EII model (spherical, equal volume) when calculating BIC.'}, {'name': 'VII', 'value': 'TRUE', 'label': 'Vii', 'description': 'Include the VII model (spherical, unequal volume) when calculating BIC.'}, {'name': 'EEI', 'value': 'FALSE', 'label': 'Eei', 'description': 'Include the EEI model (diagonal, equal volume and shape) when calculating BIC.'}, {'name': 'VEI', 'value': 'FALSE', 'label': 'Vei', 'description': 'Include the VEI model (diagonal, varying volume, equal shape) when calculating BIC.'}, {'name': 'EVI', 'value': 'FALSE', 'label': 'Evi', 'description': 'Include the EVI model (diagonal, equal volume, varying shape) when calculating BIC.'}, {'name': 'VVI', 'value': 'FALSE', 'label': 'Vvi', 'description': 'Include the VVI model (diagonal, varying volume and shape) when calculating BIC.'}, {'name': 'EEE', 'value': 'FALSE', 'label': 'Eee', 'description': 'Include the EEE model (ellipsoidal, equal volume, shape, and orientation) when calculating BIC.'}, {'name': 'EEV', 'value': 'FALSE', 'label': 'Eev', 'description': 'Include the EEV model (ellipsoidal, equal volume and equal shape) when calculating BIC.'}, {'name': 'VEV', 'value': 'FALSE', 'label': 'Vev', 'description': 'Include the VEV model (ellipsoidal, equal shape) when calculating BIC.'}, {'name': 'VVV', 'value': 'FALSE', 'label': 'Vvv', 'description': 'Include the VVV model (ellipsoidal, varying volume, shape, and orientation) when calculating BIC.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task MergeColumns: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Merge datasets by column.', 'required_args': [{'name': 'input.filename1', 'value': '', 'label': 'Input.Filename1', 'description': 'first file to merge'}, {'name': 'input.filename2', 'value': '', 'label': 'Input.Filename2', 'description': 'second file to merge'}, {'name': 'zip.filename', 'value': '', 'label': 'Zip.Filename', 'description': 'zip file containing files to merge. Either a zip file or the input files above must be provided.'}], 'default_args': [{'name': 'task_name', 'value': 'MergeColumns', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output', 'value': '<input.filename1_basename><zip.filename_basename>.merged', 'label': 'Output', 'description': 'name of the merged file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task MergeFCSDataFiles: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Merge multiple FCS data files into a single FCS datasets; includes multimple subsampling options.', 'required_args': [{'name': 'Input.files', 'value': '', 'label': 'Input.Files', 'description': 'The input FCS files to be merged; either a zip file or a pointer to a directory or a text file with FCS file names (full path) with one filename per line.'}, {'name': 'Keep.n.events', 'value': '', 'label': 'Keep.N.Events', 'description': 'Keep approximately n events from each dataset, optional, cannot be combined with subsampling rate.'}], 'default_args': [{'name': 'task_name', 'value': 'MergeFCSDataFiles', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Subsampling.rate', 'value': '100', 'label': 'Subsampling.Rate', 'description': 'Subsampling rate as percentage of events to be included from each data set, value between 0 and 100, optional, default 100%, cannot be combined with keep n events.'}, {'name': 'Random.seed', 'value': '1234567', 'label': 'Random.Seed', 'description': 'The random seed'}, {'name': 'Output.file.name', 'value': 'MergedFCSDataFiles.fcs', 'label': 'Output.File.Name', 'description': 'The output FCS file name.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task MergeHTSeqCounts: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Merges HTSeq read count data files into one file', 'required_args': [{'name': 'input.files', 'value': '', 'label': 'Input.Files', 'description': 'A directory or one or more text files containing HTSeq read counts'}], 'default_args': [{'name': 'task_name', 'value': 'MergeHTSeqCounts', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.prefix', 'value': '<input.files_basename>', 'label': 'Output.Prefix', 'description': 'The base name of the output file. File extensions will be added automatically.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task MergeRows: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Merge datasets by row.', 'required_args': [{'name': 'input.filename1', 'value': '', 'label': 'Input.Filename1', 'description': 'first file to merge'}, {'name': 'input.filename2', 'value': '', 'label': 'Input.Filename2', 'description': 'second file to merge'}, {'name': 'zip.filename', 'value': '', 'label': 'Zip.Filename', 'description': 'zip file containing files to merge. Either a zip file or the input files above must be provided.'}], 'default_args': [{'name': 'task_name', 'value': 'MergeRows', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output', 'value': '<input.filename1_basename><zip.filename_basename>.merged', 'label': 'Output', 'description': 'name of the merged file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task MetageneProjection: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Projects one or more data sets onto the metagene representation of another data set', 'required_args': [{'name': 'model.gct.file', 'value': '', 'label': 'Model.Gct.File', 'description': 'The model .gct data file'}, {'name': 'model.cls.file', 'value': '', 'label': 'Model.Cls.File', 'description': 'The model .cls data file'}, {'name': 'model.preprocessing.file', 'value': '', 'label': 'Model.Preprocessing.File', 'description': 'The pre-processing parameters for the model dataset'}, {'name': 'test.gct.file', 'value': '', 'label': 'Test.Gct.File', 'description': 'The test .gct or zip of .gct data files'}, {'name': 'test.cls.file', 'value': '', 'label': 'Test.Cls.File', 'description': 'The test .cls or zip of .cls data files'}, {'name': 'test.preprocessing.file', 'value': '', 'label': 'Test.Preprocessing.File', 'description': 'The pre-processing parameters for the test dataset'}, {'name': 'phenotype.plotting.colors', 'value': '', 'label': 'Phenotype.Plotting.Colors', 'description': 'Plotting colors for phenotypes'}, {'name': 'phenotype.plotting.symbols', 'value': '', 'label': 'Phenotype.Plotting.Symbols', 'description': 'Plotting symbols for phenotypes'}], 'default_args': [{'name': 'task_name', 'value': 'MetageneProjection', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'num.characters', 'value': '9', 'label': 'Num.Characters', 'description': 'Number of characters to use to identify classes from the .cls files'}, {'name': 'identifier', 'value': '<model.gct.file_basename>.output', 'label': 'Identifier', 'description': 'Prefix to name output files'}, {'name': 'k.projection', 'value': '3', 'label': 'K.Projection', 'description': 'Number of metagenes in projection'}, {'name': 'algorithm', 'value': 'NMF.div', 'label': 'Algorithm', 'description': 'Algorithm for Metagene Projection'}, {'name': 'number.of.iterations', 'value': '2000', 'label': 'Number.Of.Iterations', 'description': 'Number of algorithm iterations'}, {'name': 'seed', 'value': '1234', 'label': 'Seed', 'description': 'Random seed to initialize metagene matrices (Note: results may only be reproducible on the same platform)'}, {'name': 'post.projection.normalization', 'value': 'yes', 'label': 'Post.Projection.Normalization', 'description': 'apply post-projection normalization'}, {'name': 'heatmap.row.norm', 'value': 'no', 'label': 'Heatmap.Row.Norm', 'description': 'row-normalize (standardize) the rows in the heat map'}, {'name': 'heatmap.color.scheme', 'value': 'reddish color map', 'label': 'Heatmap.Color.Scheme', 'description': ''}, {'name': 'confidence.threshold', 'value': '0.3', 'label': 'Confidence.Threshold', 'description': 'Confidence threshold (Brier score) to separate call from no-calls'}, {'name': 'symbol.scaling', 'value': '1', 'label': 'Symbol.Scaling', 'description': 'Graphical scaling for symbols in plots and plot legends'}, {'name': 'kernel', 'value': 'radial', 'label': 'Kernel', 'description': 'Kernel function for SVM'}, {'name': 'cost', 'value': '1', 'label': 'Cost', 'description': 'cost function for SVM'}, {'name': 'gamma', 'value': '0.05', 'label': 'Gamma', 'description': 'Gamma coefficient for radial base function kernel'}, {'name': 'theta', 'value': '0', 'label': 'Theta', 'description': 'Smoothing parameter only used for the NSNMF.div algorithm'}, {'name': 'lambda', 'value': '0', 'label': 'Lambda', 'description': 'Sparse parameter used only for the SNMF algorithm'}, {'name': 'model.set.refinement', 'value': 'yes', 'label': 'Model.Set.Refinement', 'description': 'perform model set refinement'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task MINDY: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Runs the MINDY algorithm for inferring genes that modulate the activity of a transcription factor at post-transcriptional levels ', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'The dataset file - .res, .gct'}, {'name': 'modulator.list', 'value': '', 'label': 'Modulator.List', 'description': 'A list of possible modulator genes to analyze - .txt'}, {'name': 'dpi.transcription.factor.list', 'value': '', 'label': 'Dpi.Transcription.Factor.List', 'description': 'List of transcription factors in dataset - .txt'}, {'name': 'transcription.factor', 'value': '', 'label': 'Transcription.Factor', 'description': 'Transcription factor'}], 'default_args': [{'name': 'task_name', 'value': 'MINDY', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'sample.percentage', 'value': '35.0', 'label': 'Sample.Percentage', 'description': 'Percentage of samples to use in analysis (must be <=50)'}, {'name': 'conditional.mi.threshold', 'value': '0.0', 'label': 'Conditional.Mi.Threshold', 'description': 'Conditional mutual information (MI) score threshold (number between 0 and 1)'}, {'name': 'conditional.mi.p.value', 'value': '1.0', 'label': 'Conditional.Mi.P.Value', 'description': 'Conditional mutual info p-value threshold (ignored if conditonal MI threshold >0)'}, {'name': 'dpi.tolerance', 'value': '1.0', 'label': 'Dpi.Tolerance', 'description': 'The data processing inequality tolerance (number between 0 and 1)'}, {'name': 'unconditional.mi.threshold', 'value': '0.0', 'label': 'Unconditional.Mi.Threshold', 'description': 'Unconditional mutual information (MI) score threshold (number between 0 and 1)'}, {'name': 'unconditional.mi.p.value', 'value': '1.0', 'label': 'Unconditional.Mi.P.Value', 'description': 'Unconditional mutual information p-value threshold (ignored if unconditonal MI threshold >0)'}, {'name': 'output.file', 'value': '<input.file_basename>.mindy', 'label': 'Output.File', 'description': 'Output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task MINDYViewer: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Visualizes the results of MINDY', 'required_args': [{'name': 'dataset.file', 'value': '', 'label': 'Dataset.File', 'description': 'The dataset file - .res, .gct'}, {'name': 'mindy.file', 'value': '', 'label': 'Mindy.File', 'description': 'Output from MINDY- .mindy'}], 'default_args': [{'name': 'task_name', 'value': 'MINDYViewer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task ModEcocTrainTest: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Modified error correcting output code prediction', 'required_args': [{'name': 'train.file', 'value': '', 'label': 'Train.File', 'description': 'The training features and class label information in .arff format'}, {'name': 'test.file', 'value': '', 'label': 'Test.File', 'description': 'The testing features and possible class label information in .arff format'}, {'name': 'saved.model.file', 'value': '', 'label': 'Saved.Model.File', 'description': 'Input modEcocTrainTest model'}], 'default_args': [{'name': 'task_name', 'value': 'ModEcocTrainTest', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'model.output.file', 'value': '<train.file_basename>.model', 'label': 'Model.Output.File', 'description': 'Output modEcocTrainTest model'}, {'name': 'base.classifier', 'value': '0', 'label': 'Base.Classifier', 'description': 'The name of the base classifier to use'}, {'name': 'base.multiplier', 'value': '4', 'label': 'Base.Multiplier', 'description': 'The base classifier multiplier'}, {'name': 'pred.results.output.file', 'value': '<test.file_basename>', 'label': 'Pred.Results.Output.File', 'description': 'The name of the output file for prediction results.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task ModEcocXValidation: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Modified error correcting output code cross validation prediction', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'The input features and class label information in .arff format'}], 'default_args': [{'name': 'task_name', 'value': 'ModEcocXValidation', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'num.folds', 'value': '2', 'label': 'Num.Folds', 'description': 'The number of folds should be an integer greater than or equal to two.'}, {'name': 'base.classifier', 'value': '0', 'label': 'Base.Classifier', 'description': 'The name of the base classifier to use'}, {'name': 'base.multiplier', 'value': '4', 'label': 'Base.Multiplier', 'description': 'The base classifier multiplier'}, {'name': 'pred.results.file', 'value': '<input.file_basename>', 'label': 'Pred.Results.File', 'description': 'The name of the output file for prediction results.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task Multiplot: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Creates customizable plots of expression data-derived data.', 'required_args': [{'name': 'input.file.name', 'value': '', 'label': 'Input.File.Name', 'description': 'File created by MultiplotPreprocess (.zip)'}], 'default_args': [{'name': 'task_name', 'value': 'Multiplot', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'number.of.plots', 'value': '1', 'label': 'Number.Of.Plots', 'description': 'Smaller one-plot visualizer or bigger two-plot'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task MultiplotExtractor: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Creates customizable plots of expression data-derived data.', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'File created by MultiplotPreprocess (.zip)'}], 'default_args': [{'name': 'task_name', 'value': 'MultiplotExtractor', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task MultiplotPreprocess: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Creates derived data from an expression dataset for use in the Multiplot Visualizer module', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'File with data to prepare for plotting (gct)'}, {'name': 'class.filename', 'value': '', 'label': 'Class.Filename', 'description': 'Class file for this data (cls)'}], 'default_args': [{'name': 'task_name', 'value': 'MultiplotPreprocess', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<input.filename_basename>.zip', 'label': 'Output.File', 'description': 'Name of file to create for use in Multiplot visualizer'}, {'name': 'replicateFC', 'value': '0', 'label': 'ReplicateFC', 'description': 'Whether to compute fold changes between each replicate (turning this on can slow your work).'}, {'name': 'randomData', 'value': '0', 'label': 'RandomData', 'description': 'Whether to create random expression data (and fold changes, etc for this data). Turning this on can slow your work'}, {'name': 'outlierElimination', 'value': '0', 'label': 'OutlierElimination', 'description': 'Remove outlier replicate values of genes from processing (see \"Help\" for details)'}, {'name': 'doStats', 'value': '1', 'label': 'DoStats', 'description': 'Perform t-test, Hochberg p-value adjustment, and significance comparison to randomized data'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task MultiplotStudio: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Scatterplot visualization tool for analyzing differential gene expression.', 'required_args': [{'name': 'dataFile', 'value': '', 'label': 'DataFile', 'description': 'GenePattern dataset (*.gct), Multiplot Studio Workspace file (*.msdata) or CSV (see help for format requirements).'}, {'name': 'classFile', 'value': '', 'label': 'ClassFile', 'description': 'Class file (*.cls). Not required, but lack of a class file will result in fewer calculation options. Ignored if dataFile is a Workspace.'}], 'default_args': [{'name': 'task_name', 'value': 'MultiplotStudio', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task MutSigCV: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Identifies genes that are significantly mutated in cancer genomes, using a model with mutational covariates.', 'required_args': [{'name': 'mutation.table.file', 'value': '', 'label': 'Mutation.Table.File', 'description': 'Mutation list in MAF format.'}, {'name': 'coverage.table.file', 'value': '', 'label': 'Coverage.Table.File', 'description': 'File containing the coverage table.  This should be a file with per-sample coverage as produced by MuTect from the samples\\' WIG files.  If you do not have this level of coverage information, you can instead use the territory file for the reference human exome to direct MutSigCV to assume full coverage.  This file is found using \"Add Path or URL\" under \"shared_data/example_files/MutSigCV_1.3/exome_full192.coverage.txt\".'}, {'name': 'mutation.type.dictionary', 'value': '', 'label': 'Mutation.Type.Dictionary', 'description': 'The mutation type dictionary to use for automatic category and effect discovery.  This is necessary only if you are using a MAF file without the columns \"categ\" and \"effect\".  A suggested dictionary can be found using \"Add Path or URL\" under \"shared_data/example_files/MutSigCV_1.3/mutation_type_dictionary_file.txt\".'}, {'name': 'genome.build', 'value': '', 'label': 'Genome.Build', 'description': 'Genome build to use for automatic category and effect discovery.  This is necessary only if you are using a MAF file without the columns \"categ\" and \"effect\".'}], 'default_args': [{'name': 'task_name', 'value': 'MutSigCV', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'covariates.table.file', 'value': 'ftp://ftp.broadinstitute.org/pub/genepattern/example_files/MutSigCV_1.3/gene.covariates.txt', 'label': 'Covariates.Table.File', 'description': 'File containing the covariates table.  The default selection is the covariates file used in the MutSigCV paper published in Nature (see the documentation for further information and a full reference). '}, {'name': 'output.filename.base', 'value': '<mutation.table.file_basename>', 'label': 'Output.Filename.Base', 'description': 'Base name for the output files.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task mzXMLToCsv: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Converts a mzXML file to a zip of csv files', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'input filename - .mzXML'}], 'default_args': [{'name': 'task_name', 'value': 'mzXMLToCsv', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<input.filename_basename>.zip', 'label': 'Output.File', 'description': 'output file containing zip of csv files'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task NearestTemplatePrediction: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Nearest neighbor prediction based on a list of marker genes', 'required_args': [{'name': 'input.exp.filename', 'value': '', 'label': 'Input.Exp.Filename', 'description': 'Gene expression data set (.gct)'}, {'name': 'input.features.filename', 'value': '', 'label': 'Input.Features.Filename', 'description': 'List of marker genes (.txt): Probe ID, Gene name, Class (1,2,...), Weight (optional)'}], 'default_args': [{'name': 'task_name', 'value': 'NearestTemplatePrediction', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.name', 'value': 'NTP', 'label': 'Output.Name', 'description': 'Name for output files'}, {'name': 'distance.selection', 'value': 'cosine', 'label': 'Distance.Selection', 'description': 'Distance metric'}, {'name': 'weight.genes', 'value': 'T', 'label': 'Weight.Genes', 'description': 'Weight genes? (by statistic, fold change, etc. only for 2 classes)'}, {'name': 'num.resamplings', 'value': '1000', 'label': 'Num.Resamplings', 'description': '# resampling to generate null distribution for distance metric'}, {'name': 'GenePattern.output', 'value': 'F', 'label': 'GenePattern.Output', 'description': 'Create .gct and .cls files for GenePattern'}, {'name': 'random.seed', 'value': '7392854', 'label': 'Random.Seed', 'description': 'Random seed'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task NMF: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Non-negative Matrix Factorization', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'Input file in GCT format with input data (V matrix) (GCT format only). e.g. ALL_vs_AML.gct'}], 'default_args': [{'name': 'task_name', 'value': 'NMF', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<input.filename_basename>.log', 'label': 'Output.File', 'description': 'Output log file. e.g. NMF.log'}, {'name': 'nfact', 'value': '4', 'label': 'Nfact', 'description': 'Number of NMF factors. e.g. 2, 4, 10'}, {'name': 'niter', 'value': '100', 'label': 'Niter', 'description': 'Number of NMF iterations. e.g. 100, 250, 500'}, {'name': 'print.dump', 'value': '0', 'label': 'Print.Dump', 'description': 'Print matrices in output for debug'}, {'name': 'cnorm.flag', 'value': '0', 'label': 'Cnorm.Flag', 'description': 'Column normalization flag'}, {'name': 'rnorm.flag', 'value': '0', 'label': 'Rnorm.Flag', 'description': 'Row normalization flag'}, {'name': 'gshift', 'value': '0', 'label': 'Gshift', 'description': 'Global additive shift after normalization. e.g. 1, 4'}, {'name': 'norm.type', 'value': '0', 'label': 'Norm.Type', 'description': 'Type of normalization'}, {'name': 'seed', 'value': '12345', 'label': 'Seed', 'description': 'Random seed used to initialize W and H matrices by the randon number generator. e.g. 4585, 4567, 5980'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task NMFConsensus: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Non-negative Matrix Factorization (NMF) Consensus Clustering', 'required_args': [{'name': 'dataset.filename', 'value': '', 'label': 'Dataset.Filename', 'description': 'Input dataset (gct or res)'}], 'default_args': [{'name': 'task_name', 'value': 'NMFConsensus', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'k.initial', 'value': '2', 'label': 'K.Initial', 'description': 'Initial value of K'}, {'name': 'k.final', 'value': '5', 'label': 'K.Final', 'description': 'Final value of K'}, {'name': 'num.clusterings', 'value': '20', 'label': 'Num.Clusterings', 'description': 'Number of clusterings per value of K'}, {'name': 'max.num.iterations', 'value': '2000', 'label': 'Max.Num.Iterations', 'description': 'Maximum number of NMF iterations per clustering'}, {'name': 'error.function', 'value': 'divergence', 'label': 'Error.Function', 'description': 'Error function'}, {'name': 'random.seed', 'value': '123456789', 'label': 'Random.Seed', 'description': 'Seed for random number generator'}, {'name': 'output.file.prefix', 'value': '<dataset.filename_basename>', 'label': 'Output.File.Prefix', 'description': 'Prefix to append to all output files'}, {'name': 'stop.convergence', 'value': '40', 'label': 'Stop.Convergence', 'description': 'How many \"no change\" checks are needed to stop NMF iterations before max iterations is reached(convergence)'}, {'name': 'stop.frequency', 'value': '10', 'label': 'Stop.Frequency', 'description': 'Frequency (NMF iterations) of \"no change\" checks'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task NMFConsensus_combine_lewu: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Non-negative Matrix Factorization (NMF) Consensus Clustering (combine single k result)', 'required_args': [{'name': 'param.filename', 'value': '', 'label': 'Param.Filename', 'description': 'Input param file '}], 'default_args': [{'name': 'task_name', 'value': 'NMFConsensus_combine_lewu', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task NMFConsensus_sk_lewu: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Non-negative Matrix Factorization (NMF) Consensus Clustering', 'required_args': [{'name': 'param.filename', 'value': '', 'label': 'Param.Filename', 'description': 'Input param file '}], 'default_args': [{'name': 'task_name', 'value': 'NMFConsensus_sk_lewu', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task NMFPartition: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'The module will output a list of parameter files', 'required_args': [{'name': 'dataset.filename', 'value': '', 'label': 'Dataset.Filename', 'description': 'Input dataset (gct or res)'}], 'default_args': [{'name': 'task_name', 'value': 'NMFPartition', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'k.initial', 'value': '2', 'label': 'K.Initial', 'description': 'Initial value of K'}, {'name': 'k.final', 'value': '5', 'label': 'K.Final', 'description': 'Final value of K'}, {'name': 'num.clusterings', 'value': '20', 'label': 'Num.Clusterings', 'description': 'Number of clusterings per value of K'}, {'name': 'max.num.iterations', 'value': '2000', 'label': 'Max.Num.Iterations', 'description': 'Maximum number of NMF iterations per clustering'}, {'name': 'error.function', 'value': 'divergence', 'label': 'Error.Function', 'description': 'Error function'}, {'name': 'random.seed', 'value': '123456789', 'label': 'Random.Seed', 'description': 'Seed for random number generator'}, {'name': 'output.file.prefix', 'value': '<dataset.filename_basename>', 'label': 'Output.File.Prefix', 'description': 'Prefix to append to all output files'}, {'name': 'stop.convergence', 'value': '40', 'label': 'Stop.Convergence', 'description': 'How many \"no change\" checks are needed to stop NMF iterations before max iterations is reached(convergence)'}, {'name': 'stop.frequency', 'value': '10', 'label': 'Stop.Frequency', 'description': 'Frequency (NMF iterations) of \"no change\" checks'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task NMFSingleKPipe: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '', 'required_args': [{'name': 'NMFConsensus_sk_lewu1.param.filename', 'value': '', 'label': 'NMFConsensus Sk Lewu1.Param.Filename', 'description': 'Input param file '}], 'default_args': [{'name': 'task_name', 'value': 'NMFSingleKPipe', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task NormLines: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Normalizes RNAi hairpin scores across cell lines using ZMAD, PMAD, rank or quantile- ignores NaNs', 'required_args': [{'name': 'gct.file', 'value': '', 'label': 'Gct.File', 'description': 'Input gct file to normalize by samples'}], 'default_args': [{'name': 'task_name', 'value': 'NormLines', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'method', 'value': 'quantile', 'label': 'Method', 'description': 'Options are ZMAD, PMAD, quantile'}, {'name': 'plotfile', 'value': 'NormPDF.pdf', 'label': 'Plotfile', 'description': ''}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task ParallelCBS: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Runs CBS algorithm on multiple samples in parallel', 'required_args': [{'name': 'SplitColumns1.input.file', 'value': '', 'label': 'SplitColumns1.Input.File', 'description': 'A tab delimited file'}], 'default_args': [{'name': 'task_name', 'value': 'ParallelCBS', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'ConcatenateFilelist4.output.filename', 'value': '<filelist_basename>.catted.seg.txt', 'label': 'ConcatenateFilelist4.Output.Filename', 'description': 'The name of the output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task ParallelNMFConsensusPipe: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'A parallelized implementation of the NMFConsensus module, implemented as a GenePattern scatter-gather pipeline.', 'required_args': [{'name': 'NMFPartition1.dataset.filename', 'value': '', 'label': 'NMFPartition1.Dataset.Filename', 'description': 'Input dataset (gct or res)'}], 'default_args': [{'name': 'task_name', 'value': 'ParallelNMFConsensusPipe', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'NMFPartition1.k.initial', 'value': '2', 'label': 'NMFPartition1.K.Initial', 'description': 'Initial value of K'}, {'name': 'NMFPartition1.k.final', 'value': '5', 'label': 'NMFPartition1.K.Final', 'description': 'Final value of K'}, {'name': 'NMFPartition1.num.clusterings', 'value': '20', 'label': 'NMFPartition1.Num.Clusterings', 'description': 'Number of clusterings per value of K'}, {'name': 'NMFPartition1.max.num.iterations', 'value': '2000', 'label': 'NMFPartition1.Max.Num.Iterations', 'description': 'Maximum number of NMF iterations per clustering'}, {'name': 'NMFPartition1.error.function', 'value': 'divergence', 'label': 'NMFPartition1.Error.Function', 'description': 'Error function'}, {'name': 'NMFPartition1.random.seed', 'value': '123456789', 'label': 'NMFPartition1.Random.Seed', 'description': 'Seed for random number generator'}, {'name': 'NMFPartition1.output.file.prefix', 'value': '<dataset.filename_basename>', 'label': 'NMFPartition1.Output.File.Prefix', 'description': 'Prefix to append to all output files'}, {'name': 'NMFPartition1.stop.convergence', 'value': '40', 'label': 'NMFPartition1.Stop.Convergence', 'description': 'How many \"no change\" checks are needed to stop NMF iterations before max iterations is reached(convergence)'}, {'name': 'NMFPartition1.stop.frequency', 'value': '10', 'label': 'NMFPartition1.Stop.Frequency', 'description': 'Frequency (NMF iterations) of \"no change\" checks'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task PARIS: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Feature selection by class or against a continuous vector (Probablility Analysis by Ranked Information Score)', 'required_args': [{'name': 'target.profile', 'value': '', 'label': 'Target.Profile', 'description': 'Target profile file (either .cls or .gct).  Used to classify cell line samples into groups, or assign a continuous target vector to the cell line samples.'}, {'name': 'target', 'value': '', 'label': 'Target', 'description': \"Target gene or ID from target.profile file.  Should match a row in the 'Name' or 'Description' column of a .gct file, or match one label in the 2nd row of the .cls file\"}, {'name': 'data', 'value': '', 'label': 'Data', 'description': 'Data (feature) file to rank based on target profile'}, {'name': 'prefix', 'value': '', 'label': 'Prefix', 'description': 'prefix for labeling output files'}, {'name': 'advanced', 'value': '', 'label': 'Advanced', 'description': 'Leave blank for default.  This parameter allows the user to change the underlying default metric.  Options will be listed in the documentation.'}], 'default_args': [{'name': 'task_name', 'value': 'PARIS', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'permtype', 'value': 'standard', 'label': 'Permtype', 'description': 'standard; subclass.stratified (lineage-corrected)'}, {'name': 'type.flag', 'value': 'gene', 'label': 'Type.Flag', 'description': 'perturbagen level data (shRNA) or gene level scores (gene)'}, {'name': 'quality', 'value': 'normal', 'label': 'Quality', 'description': 'controls how many permutations are run; draft will take less time'}, {'name': 'direction', 'value': 'positive', 'label': 'Direction', 'description': 'Direction of feature matching; positive means high values to high values, low to low; negative means high to low values'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task PCA: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Principal Component Analysis (adapted from MeV, http://www.tm4.org/m)', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'input file - .gct, .res, .odf'}], 'default_args': [{'name': 'task_name', 'value': 'PCA', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'cluster.by', 'value': '3', 'label': 'Cluster.By', 'description': 'Cluster by rows or columns'}, {'name': 'output.file', 'value': '<input.filename_basename>', 'label': 'Output.File', 'description': 'stub name of the output files'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task PCAViewer: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Principal Component Analysis Visualizer (adapted from MeV, http://www.tigr.org/software/tm4/mev.html)', 'required_args': [{'name': 'dataset.file', 'value': '', 'label': 'Dataset.File', 'description': 'The dataset - .res, .gct'}, {'name': 's.matrix.file', 'value': '', 'label': 'S.Matrix.File', 'description': 'The s matrix - .odf'}, {'name': 't.matrix.file', 'value': '', 'label': 'T.Matrix.File', 'description': 'The t matrix - .odf'}, {'name': 'u.matrix.file', 'value': '', 'label': 'U.Matrix.File', 'description': 'The u matrix - .odf'}, {'name': 'cls.or.sample.info.file', 'value': '', 'label': 'Cls.Or.Sample.Info.File', 'description': 'A cls or sample info file - .cls, .txt'}, {'name': 'sample.info.class', 'value': '', 'label': 'Sample.Info.Class', 'description': 'The name of the class column to use in the sample info file (required if sample info file provided)'}, {'name': 'featureset.file', 'value': '', 'label': 'Featureset.File', 'description': 'The feature set file - .gmt, .gmx, .grp'}], 'default_args': [{'name': 'task_name', 'value': 'PCAViewer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task PclToGct: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Convert Stanford preclustering file (.pcl) to .gct', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'pcl file to convert'}], 'default_args': [{'name': 'task_name', 'value': 'PclToGct', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.filename', 'value': '<input.filename_basename>.gct', 'label': 'Output.Filename', 'description': 'output gct file'}, {'name': 'weight.handling', 'value': '3', 'label': 'Weight.Handling', 'description': 'How to handle manually edited weights (weights not equal to 1) present in PCL file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task PeakMatch: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Perform peak matching on LC-MS data', 'required_args': [{'name': 'peakList.filename', 'value': '', 'label': 'PeakList.Filename', 'description': 'Zip file containing peak lists'}, {'name': 'sampleInfo.filename', 'value': '', 'label': 'SampleInfo.Filename', 'description': 'CSV file with header: experiment, sample, class'}, {'name': 'LandmarkMatchOutput.filename', 'value': '', 'label': 'LandmarkMatchOutput.Filename', 'description': 'Zip file containing the output from landmark matching.'}, {'name': 'MZ.tolerance', 'value': '', 'label': 'MZ.Tolerance', 'description': 'm/z tolerance (ppm). Used if landmark match output is not supplied. Defaults to 10 ppm.'}, {'name': 'RT.tolerance', 'value': '', 'label': 'RT.Tolerance', 'description': 'retention time tolerance (min). Used if landmark match is not supplied. Defaults to 2 min.'}, {'name': 'numberProcesses', 'value': '', 'label': 'NumberProcesses', 'description': 'Number of processess when running in parallel. Defaults to 1 (sequential).'}], 'default_args': [{'name': 'task_name', 'value': 'PeakMatch', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'outputName', 'value': 'PeakMatchOutput', 'label': 'OutputName', 'description': 'File name prefix for the output files'}, {'name': 'running.on.LSF', 'value': 'no', 'label': 'Running.On.LSF', 'description': 'Whether you are running on an LSF cluster version of GenePattern'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task Peaks: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Determine peaks in the spectrum using a series of digital filters.', 'required_args': [{'name': 'spectrum.filename', 'value': '', 'label': 'Spectrum.Filename', 'description': 'spectrum data file - .csv'}, {'name': 'low.Da', 'value': '', 'label': 'Low.Da', 'description': 'Minimum M/Z to include'}, {'name': 'high.Da', 'value': '', 'label': 'High.Da', 'description': 'Maximum M/Z to include'}], 'default_args': [{'name': 'task_name', 'value': 'Peaks', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'percentile', 'value': '0.65', 'label': 'Percentile', 'description': 'After applying filters, threshold above which peaks are located'}, {'name': 'smoothing.size', 'value': '21', 'label': 'Smoothing.Size', 'description': 'Size of the smoothing filter'}, {'name': 'adaptive.background.correction.strength', 'value': '0.75', 'label': 'Adaptive.Background.Correction.Strength', 'description': 'Strength of the adaptive background correction filter'}, {'name': 'adaptive.background.correction.size', 'value': '21', 'label': 'Adaptive.Background.Correction.Size', 'description': 'Size of the adaptive background correction filter'}, {'name': 'high.pass.filter.strength', 'value': '10', 'label': 'High.Pass.Filter.Strength', 'description': 'Strength of the high pass filter'}, {'name': 'high.pass.filter.factor', 'value': '5', 'label': 'High.Pass.Filter.Factor', 'description': 'Filter factor for the high pass filter'}, {'name': 'output.file', 'value': '<spectrum.filename_basename>.peaks.odf', 'label': 'Output.File', 'description': 'output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task Picard.AddOrReplaceReadGroups: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Replaces all read groups in the input file with a new read group and assigns all reads to this read group in the output', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'Input file (BAM or SAM).'}, {'name': 'read.group.id', 'value': '', 'label': 'Read.Group.Id', 'description': 'Read Group ID'}, {'name': 'read.group.library', 'value': '', 'label': 'Read.Group.Library', 'description': 'DNA preparation library. Note that this field is required by the module Picard.MarkDuplicates. It is used to determine which read groups may contain molecular duplicates, in the case where the same DNA library was sequenced on multiple lanes.'}, {'name': 'read.group.platform', 'value': '', 'label': 'Read.Group.Platform', 'description': 'Read Group platform (e.g., Illumina, SOLID).'}, {'name': 'read.group.platform.unit', 'value': '', 'label': 'Read.Group.Platform.Unit', 'description': 'Read Group platform unit (e.g., run barcode).'}, {'name': 'read.group.sample.name', 'value': '', 'label': 'Read.Group.Sample.Name', 'description': 'Read Group sample name.'}, {'name': 'read.group.sequencing.center.name', 'value': '', 'label': 'Read.Group.Sequencing.Center.Name', 'description': 'Read Group sequencing center name.'}, {'name': 'read.group.description', 'value': '', 'label': 'Read.Group.Description', 'description': 'Read Group description.'}], 'default_args': [{'name': 'task_name', 'value': 'Picard.AddOrReplaceReadGroups', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.prefix', 'value': '<input.file_basename>', 'label': 'Output.Prefix', 'description': 'A label used to name the output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task Picard.BamToSam: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Converts a BAM file to a SAM file', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'A BAM file'}], 'default_args': [{'name': 'task_name', 'value': 'Picard.BamToSam', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.prefix', 'value': '<input.file_basename>', 'label': 'Output.Prefix', 'description': 'A label used to name the output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task Picard.CreateSequenceDictionary: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Reads FASTA or FASTA.GZ files containing reference sequences, and writes them as a SAM file containing a sequence dictionary', 'required_args': [{'name': 'reference.sequence.file', 'value': '', 'label': 'Reference.Sequence.File', 'description': 'Input reference fasta or fasta.gz '}, {'name': 'genome.assembly', 'value': '', 'label': 'Genome.Assembly', 'description': 'The genome assembly to put into AS field of sequence dictionary entry'}, {'name': 'uri', 'value': '', 'label': 'Uri', 'description': 'The uri to put into UR field of sequence dictionary entry'}, {'name': 'num.sequences', 'value': '', 'label': 'Num.Sequences', 'description': 'Stop after writing this many sequences'}], 'default_args': [{'name': 'task_name', 'value': 'Picard.CreateSequenceDictionary', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'truncate.names.at.white.space', 'value': 'no', 'label': 'Truncate.Names.At.White.Space', 'description': 'Make sequence name the first word from the > line in the reference file'}, {'name': 'output.file', 'value': '<reference.sequence.file_basename>.dict', 'label': 'Output.File', 'description': 'The name of the output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task Picard.FastqToSam: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Extracts read sequences and qualities from the input fastq file and writes them into the output file in unaligned SAM or BAM format', 'required_args': [{'name': 'fastq.file', 'value': '', 'label': 'Fastq.File', 'description': 'Input fastq file (optionally gzipped) for single end data, or first read in paired end data'}, {'name': 'fastq.file.2', 'value': '', 'label': 'Fastq.File.2', 'description': 'The fastq file (optionally gzipped) for second read in paired end data'}, {'name': 'read.group.name', 'value': '', 'label': 'Read.Group.Name', 'description': 'Read group name'}, {'name': 'sample.name', 'value': '', 'label': 'Sample.Name', 'description': 'Sample name to insert into the read group header'}, {'name': 'library.name', 'value': '', 'label': 'Library.Name', 'description': 'The library name to insert into the LB attribute in the read group header'}, {'name': 'platform.unit', 'value': '', 'label': 'Platform.Unit', 'description': 'The platform unit (often run_barcode.lane) to insert into the read group header'}, {'name': 'platform', 'value': '', 'label': 'Platform', 'description': 'The platform type (e.g., Illumina, SOLiD) to insert into the read group header'}], 'default_args': [{'name': 'task_name', 'value': 'Picard.FastqToSam', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'quality.format', 'value': 'Standard', 'label': 'Quality.Format', 'description': 'A value describing how the quality values are encoded in the fastq'}, {'name': 'sort.order', 'value': 'coordinate', 'label': 'Sort.Order', 'description': ''}, {'name': 'output.format', 'value': 'bam', 'label': 'Output.Format', 'description': 'Whether the file output format should be SAM or BAM'}, {'name': 'output.prefix', 'value': '<fastq.file_basename>', 'label': 'Output.Prefix', 'description': 'The prefix to use for the output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task Picard.MarkDuplicates: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Examines aligned records in the supplied SAM or BAM file to locate duplicate reads.', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'A SAM or BAM file'}, {'name': 'read.name.regex', 'value': '', 'label': 'Read.Name.Regex', 'description': 'Regular expression that can be used to parse read names in the input file'}], 'default_args': [{'name': 'task_name', 'value': 'Picard.MarkDuplicates', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'remove.duplicates', 'value': 'false', 'label': 'Remove.Duplicates', 'description': 'Whether to remove duplicates'}, {'name': 'max.file.handles', 'value': '8000', 'label': 'Max.File.Handles', 'description': 'Maximum number of file handles to keep open when spilling read ends to disk'}, {'name': 'optical.duplicate.distance', 'value': '100', 'label': 'Optical.Duplicate.Distance', 'description': 'The maximum offset between two duplicate clusters in order to consider them optical duplicates. This should usually be set to some fairly small number (e.g. 5-10 pixels) unless using later versions of the Illumina pipeline that multiply pixel values by 10, in which case 50-100 is more normal'}, {'name': 'sorting.collection.size.ratio', 'value': '0.25', 'label': 'Sorting.Collection.Size.Ratio', 'description': 'This number, plus the maximum RAM available to the JVM, determine the memory footprint used by some of the sorting collections.  If you are running out of memory, try reducing this number'}, {'name': 'output.prefix', 'value': '<input.file_basename>', 'label': 'Output.Prefix', 'description': 'The prefix to use for the output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task Picard.ReorderSam: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Reorders a SAM file or a BAM file to match contig ordering in a provided reference file', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'A SAM or BAM file.'}, {'name': 'reference.file', 'value': '', 'label': 'Reference.File', 'description': 'Reference sequence to which to reorder reads in FASTA format.'}, {'name': 'reference.sequence.dictionary', 'value': '', 'label': 'Reference.Sequence.Dictionary', 'description': 'The sequence dictionary for the reference sequence file'}], 'default_args': [{'name': 'task_name', 'value': 'Picard.ReorderSam', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'allow.partial.overlap', 'value': 'false', 'label': 'Allow.Partial.Overlap', 'description': 'If yes, then allows only a partial overlap of the BAM contigs with the new reference sequence contigs. This requires a corresponding contig in the new reference for each read contig.'}, {'name': 'allow.contig.length.discordance', 'value': 'false', 'label': 'Allow.Contig.Length.Discordance', 'description': 'If yes, then permits mapping from a read contig to a new reference contig with the same name but a different length. Use with extreme caution and a full knowledge of the possible consequences.'}, {'name': 'output.prefix', 'value': '<input.file_basename>', 'label': 'Output.Prefix', 'description': 'The prefix of the output SAM or BAM file.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task Picard.SamToBam: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Converts a SAM file to a BAM file', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'A SAM file'}], 'default_args': [{'name': 'task_name', 'value': 'Picard.SamToBam', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.prefix', 'value': '<input.file_basename>', 'label': 'Output.Prefix', 'description': 'A label used to name the output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task Picard.SamToFastq: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Convert SAM or BAM files to FASTQ format', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'The SAM or BAM file'}, {'name': 'clipping.attribute', 'value': '', 'label': 'Clipping.Attribute', 'description': 'Attribute that stores the position at which the SAM record should be clipped'}, {'name': 'clipping.action.quality.value', 'value': '', 'label': 'Clipping.Action.Quality.Value', 'description': 'Value to set base qualities to when clipping action is \"change base quality values\"'}, {'name': 'fastq.output.prefix.1', 'value': '', 'label': 'Fastq.Output.Prefix.1', 'description': 'The fastq output file name for single-end fastq or first end of a paired fastq. Only specify when \"per read group\" is \"no\".'}, {'name': 'fastq.output.prefix.2', 'value': '', 'label': 'Fastq.Output.Prefix.2', 'description': 'The second end of a paired fastq. Only specify when \"per read group\" is \"no\".'}], 'default_args': [{'name': 'task_name', 'value': 'Picard.SamToFastq', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'per.read.group', 'value': 'no', 'label': 'Per.Read.Group', 'description': 'Whether to output a fastq file per read group'}, {'name': 're-reverse.bases', 'value': 'no', 'label': 'Re-Reverse.Bases', 'description': 'Whether to re-reverse bases and qualities of reads with negative strand flag set'}, {'name': 'include.non.pf.reads', 'value': 'no', 'label': 'Include.Non.Pf.Reads', 'description': 'Whether to include non-PF reads'}, {'name': 'clipping.action', 'value': 'none', 'label': 'Clipping.Action', 'description': 'Action to take with clipped reads when a clipping attribute is specified'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task Picard.SortSam: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Sorts a SAM or BAM file in a specified order', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'Input file in SAM or BAM format'}, {'name': 'output.format', 'value': '', 'label': 'Output.Format', 'description': 'Whether the file output format should be SAM or BAM'}], 'default_args': [{'name': 'task_name', 'value': 'Picard.SortSam', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'sort.order', 'value': 'coordinate', 'label': 'Sort.Order', 'description': 'Sort order'}, {'name': 'output.prefix', 'value': '<input.file_basename>', 'label': 'Output.Prefix', 'description': 'prefix for the name of output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task PlateQAFCS: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Look for consistency between wells of a single plate by computing summary statistics.', 'required_args': [{'name': 'Input.FCS.files', 'value': '', 'label': 'Input.FCS.Files', 'description': 'FCS files from a single plate.'}], 'default_args': [{'name': 'task_name', 'value': 'PlateQAFCS', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Check.cell.counts', 'value': '1', 'label': 'Check.Cell.Counts', 'description': 'Count cells in each file?'}, {'name': 'Check.medians', 'value': '1', 'label': 'Check.Medians', 'description': 'Report the median measured values?'}, {'name': 'Check.PDF', 'value': '1', 'label': 'Check.PDF', 'description': 'Check the Probability Density Functions (PDF)?'}, {'name': 'Check.ECDF', 'value': '1', 'label': 'Check.ECDF', 'description': 'Check the Empirical Cumulative Distribution Functions (ECDF)?'}, {'name': 'Check.boxplots', 'value': '1', 'label': 'Check.Boxplots', 'description': 'Detect unexpected data distributions (unusual upper/lower quantiles)?'}, {'name': 'Check.timeline', 'value': '1', 'label': 'Check.Timeline', 'description': 'Check for unusual patterns in the acquisition of fluorescence and light scatter measurements over time?'}, {'name': 'Check.timeflow', 'value': '1', 'label': 'Check.Timeflow', 'description': 'Check for disturbances in the flow of cells over time?'}, {'name': 'FSC.channel', 'value': 'FSC-H', 'label': 'FSC.Channel', 'description': 'The name of the forward scatter parameter (short name, value of the $PnN keyword).'}, {'name': 'SSC.channel', 'value': 'SSC-H', 'label': 'SSC.Channel', 'description': 'The name of the side scatter parameter (short name, value of the $PnN keyword).'}, {'name': 'Zip.output', 'value': 'TRUE', 'label': 'Zip.Output', 'description': 'Zip the resulting report together?'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task PlotPeaks: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Plot peaks', 'required_args': [{'name': 'peaks.output.filename', 'value': '', 'label': 'Peaks.Output.Filename', 'description': 'The output from the Peaks module'}], 'default_args': [{'name': 'task_name', 'value': 'PlotPeaks', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<peaks.output.filename_basename>.pdf', 'label': 'Output.File', 'description': 'output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task PredictionResultsViewer: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Visualizes prediction results', 'required_args': [{'name': 'prediction.results.file', 'value': '', 'label': 'Prediction.Results.File', 'description': 'Prediction results - .odf'}], 'default_args': [{'name': 'task_name', 'value': 'PredictionResultsViewer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task Preprocess.and.quality.check.RNASeq.data: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'This pipeline is used in the GenomeSpace recipe, \"Preprocess and quality check RNA-Seq data\". The recipe can be found at: http://www.genomespace.org/support/guides/recipes/sections/preprocess-and-qc-rna-seq-data.', 'required_args': [{'name': 'ConvertLineEndings1.input.filename', 'value': '', 'label': 'ConvertLineEndings1.Input.Filename', 'description': 'The input file (any non-binary file format)'}, {'name': 'TopHat4.GTF.file', 'value': '', 'label': 'TopHat4.GTF.File', 'description': 'A GTF file (v. 2.2 or higher) or GFF3 file containing a list of gene model annotations. TopHat will first extract the transcript sequences and align them to this virtual transcriptome first. Only the reads that do not fully map to the transcriptome will then be mapped on the genome. The reads that did map on the transcriptome will be converted to genomic mappings (spliced as needed) and merged with the novel mappings.'}, {'name': 'TopHat4.reads.pair.1', 'value': '', 'label': 'TopHat4.Reads.Pair.1', 'description': 'Unpaired reads file or first mate for paired reads. One or more files containing reads in FASTA or FASTQ format (bz2 and gz compressed files are supported).'}], 'default_args': [{'name': 'task_name', 'value': 'Preprocess.and.quality.check.RNASeq.data', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task PreprocessDataset: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Performs several preprocessing steps on a res, gct, or odf input file', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'Input filename - .res, .gct, .odf'}, {'name': 'threshold.for.removing.rows', 'value': '', 'label': 'Threshold.For.Removing.Rows', 'description': 'Threshold for removing rows'}, {'name': 'number.of.columns.above.threshold', 'value': '', 'label': 'Number.Of.Columns.Above.Threshold', 'description': 'Remove row if this number of columns not >= given threshold'}], 'default_args': [{'name': 'task_name', 'value': 'PreprocessDataset', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'threshold.and.filter', 'value': '1', 'label': 'Threshold.And.Filter', 'description': 'Apply floor, ceiling and variation filter'}, {'name': 'floor', 'value': '20', 'label': 'Floor', 'description': 'Value for floor threshold'}, {'name': 'ceiling', 'value': '20000', 'label': 'Ceiling', 'description': 'Value for ceiling threshold'}, {'name': 'min.fold.change', 'value': '3', 'label': 'Min.Fold.Change', 'description': 'Minimum fold change for variation filter'}, {'name': 'min.delta', 'value': '100', 'label': 'Min.Delta', 'description': 'Minimum delta for variation filter'}, {'name': 'num.outliers.to.exclude', 'value': '0', 'label': 'Num.Outliers.To.Exclude', 'description': 'Number of outliers per row to ignore when calculating row min and max for variation filter'}, {'name': 'row.normalization', 'value': '0', 'label': 'Row.Normalization', 'description': 'Perform row normalization'}, {'name': 'row.sampling.rate', 'value': '1', 'label': 'Row.Sampling.Rate', 'description': 'Sample rows without replacement to obtain this fraction of the total number of rows'}, {'name': 'log2.transform', 'value': '0', 'label': 'Log2.Transform', 'description': 'Apply log2 transform after all other preprocessing steps'}, {'name': 'output.file.format', 'value': '3', 'label': 'Output.File.Format', 'description': 'Output file format'}, {'name': 'output.file', 'value': '<input.filename_basename>.preprocessed', 'label': 'Output.File', 'description': 'Output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task PreprocessReadCounts: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Preprocess RNA-Seq count data in a GCT file so that it is suitable for use in GenePattern analyses.', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'A GCT file containing raw RNA-Seq counts, such as is produced by MergeHTSeqCounts'}, {'name': 'cls.file', 'value': '', 'label': 'Cls.File', 'description': 'A categorical CLS file specifying the phenotype classes for the samples in the GCT file.  '}], 'default_args': [{'name': 'task_name', 'value': 'PreprocessReadCounts', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<input.file_basename>.preprocessed.gct', 'label': 'Output.File', 'description': 'Output file name'}, {'name': 'expression.value.filter.threshold', 'value': '1', 'label': 'Expression.Value.Filter.Threshold', 'description': 'Threshold to use when filtering CPM expression values; rows are kept only if the values (in CPM) for all columns are greater than this threshold'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task PreviewFCS: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Allows viewing of structural metadata, parameters, and descriptive statistics from a Flow Cytometry Standard (FCS) data file', 'required_args': [{'name': 'Input.FCS.data.file', 'value': '', 'label': 'Input.FCS.Data.File', 'description': 'An input FCS data file to be described.'}], 'default_args': [{'name': 'task_name', 'value': 'PreviewFCS', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Output.file.format', 'value': 'HTML', 'label': 'Output.File.Format', 'description': 'Output file format, either HTML or XML, default HTML.'}, {'name': 'Output.file.name', 'value': '<Input.FCS.data.file_basename>', 'label': 'Output.File.Name', 'description': 'The output file name (optional).'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task Profile.Plots: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Create profile plots for samples', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'Input data file (in gct format)'}, {'name': 'cls.file', 'value': '', 'label': 'Cls.File', 'description': 'File indicating type of samples (sample group) in input.file. Profiles for each group will be plotted in a different color'}], 'default_args': [{'name': 'task_name', 'value': 'Profile.Plots', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<input.file_basename>-profile-plots.pdf', 'label': 'Output.File', 'description': 'Output file name'}, {'name': 'data.type', 'value': 'proteome', 'label': 'Data.Type', 'description': 'Indicate type of input data (proteome, phosphoprotome, etc.) for use in plots'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task Protein.mRNA.Corrlation: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Tabulate and plot correlation between proteome and mRNA', 'required_args': [{'name': 'proteome.file', 'value': '', 'label': 'Proteome.File', 'description': 'Proteome data file in gct format'}, {'name': 'mRNA.file', 'value': '', 'label': 'MRNA.File', 'description': 'mRNA (RNA-seq) data file in gct.format'}], 'default_args': [{'name': 'task_name', 'value': 'Protein.mRNA.Corrlation', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.prefix', 'value': '<proteome.file_basename>', 'label': 'Output.Prefix', 'description': 'Filename prefix for results files'}, {'name': 'fdr.threshold', 'value': '0.01', 'label': 'Fdr.Threshold', 'description': 'FDR threshold for marking statistically significant correlations'}, {'name': 'correlation.type', 'value': 'pearson', 'label': 'Correlation.Type', 'description': 'Correlation type to use for calculating protein-mRNA correlation'}, {'name': 'data.type', 'value': 'proteome', 'label': 'Data.Type', 'description': 'Indicate type of input data (proteome, phosphoprotome, etc.) for use in plots'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task ProteinDatasetCreation: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Extract features from protein .FASTA files for use with standard prediction algorithms', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'The sequence information in .FASTA format'}], 'default_args': [{'name': 'task_name', 'value': 'ProteinDatasetCreation', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'feature.set.1', 'value': '1', 'label': 'Feature.Set.1', 'description': 'Use amino acid sequence length (1 feature)'}, {'name': 'feature.set.2', 'value': '0', 'label': 'Feature.Set.2', 'description': 'Use amino acid composition (20 features)'}, {'name': 'feature.set.3', 'value': '0', 'label': 'Feature.Set.3', 'description': 'Use reduced alphabet composition of the n terminal region (19 features)'}, {'name': 'feature.set.4', 'value': '0', 'label': 'Feature.Set.4', 'description': 'Use reduced alphabet composition of the c terminal region (19 features)'}, {'name': 'feature.set.5', 'value': '0', 'label': 'Feature.Set.5', 'description': 'Use the reduced alphabet for the gap pairs of amino acids (165 features)'}, {'name': 'output.file', 'value': '<input.file_basename>', 'label': 'Output.File', 'description': 'The output files base name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task ProteoArray: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'LC-MS proteomic data processing module', 'required_args': [{'name': 'input.zip.filename', 'value': '', 'label': 'Input.Zip.Filename', 'description': 'zip file of LC-MS runs in mzXML format.'}], 'default_args': [{'name': 'task_name', 'value': 'ProteoArray', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<input.zip.filename_basename>.prot', 'label': 'Output.File', 'description': 'output filename'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task ProteomicsAnalysis: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Runs the proteomics analysis on the set of input spectra', 'required_args': [{'name': 'zip.filename', 'value': '', 'label': 'Zip.Filename', 'description': 'Zip file containing spectra data. Spectra data are stored as M/Z,Intensity'}, {'name': 'peak.list.filename', 'value': '', 'label': 'Peak.List.Filename', 'description': 'file containing one M/Z per line when peaks=input'}, {'name': 'random.seed', 'value': '', 'label': 'Random.Seed', 'description': 'random seed when peaks=random'}, {'name': 'random.n.peaks', 'value': '', 'label': 'Random.N.Peaks', 'description': \"number of random M/Z's selected when peaks=random\"}, {'name': 'low.Da', 'value': '', 'label': 'Low.Da', 'description': 'Minimum M/Z to include'}, {'name': 'high.Da', 'value': '', 'label': 'High.Da', 'description': 'Maximum M/Z to include'}], 'default_args': [{'name': 'task_name', 'value': 'ProteomicsAnalysis', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<zip.filename_basename>', 'label': 'Output.File', 'description': 'The base name for output files'}, {'name': 'quality.threshold', 'value': '0.45', 'label': 'Quality.Threshold', 'description': 'discard spectra with quality < quality threshold'}, {'name': 'fill.na', 'value': 'yes', 'label': 'Fill.Na', 'description': 'whether to fill missing peaks'}, {'name': 'normalize', 'value': 'tan', 'label': 'Normalize', 'description': 'normalization strategy'}, {'name': 'peak.detection.method', 'value': 'detect', 'label': 'Peak.Detection.Method', 'description': 'peak detection method'}, {'name': 'filter.peaks', 'value': 'yes', 'label': 'Filter.Peaks', 'description': 'filter peaks based on peak intensity / noise '}, {'name': 'filter.peaks.factor', 'value': '10', 'label': 'Filter.Peaks.Factor', 'description': 'retains peaks if peak intensity >= filter.peaks.factor * stdev(noise)'}, {'name': 'percentile', 'value': '0.65', 'label': 'Percentile', 'description': 'After applying filters, threshold above which peaks are located'}, {'name': 'smoothing.size', 'value': '21', 'label': 'Smoothing.Size', 'description': 'Size of the smoothing filter'}, {'name': 'adaptive.background.correction.strength', 'value': '0.75', 'label': 'Adaptive.Background.Correction.Strength', 'description': 'Strength of the adaptive background correction filter'}, {'name': 'adaptive.background.correction.size', 'value': '21', 'label': 'Adaptive.Background.Correction.Size', 'description': 'Size of the adaptive background correction filter'}, {'name': 'high.pass.filter.strength', 'value': '10', 'label': 'High.Pass.Filter.Strength', 'description': 'Strength of the high pass filter'}, {'name': 'high.pass.filter.factor', 'value': '5', 'label': 'High.Pass.Filter.Factor', 'description': 'Filter factor for the high pass filter'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task QuaSAR: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'A suite of tools for QC, analysis and visualization of data from MRM-MS experiments including assay configuration, calibration curves and new sample analysis.', 'required_args': [{'name': 'skyline.file', 'value': '', 'label': 'Skyline.File', 'description': 'This is a Skyline export in csv file format. The file has to include the following 10 columns with the exact column names given here (in no particular order): SampleName, PeptideSequence, ProteinName, ReplicateName, PrecursorCharge, ProductCharge, FragmentIon,  AverageMeasuredRetentionTime, light Area, heavy Area. The file may contain many more columns, but it has to include at least these 10 columns'}, {'name': 'concentration.file', 'value': '', 'label': 'Concentration.File', 'description': 'Concentration values given in csv file format. The file has to include the following 5 columns with the exact column names given here (in no particular order): SampleName, SampleID, Analyte Concentration, IS Conc, Concentration Ratio. Notice the single space between Analyte Concentration, IS Conc and Concentration Ratio. The space has to be maintained.'}, {'name': 'title', 'value': '', 'label': 'Title', 'description': 'The title is displayed on each calibration plot'}], 'default_args': [{'name': 'task_name', 'value': 'QuaSAR', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'analyte', 'value': 'light Area', 'label': 'Analyte', 'description': 'User must specify the exact analyte column header name as it is appears in the input CSV file. The default is \"light Area\"'}, {'name': 'standard', 'value': 'heavy Area', 'label': 'Standard', 'description': 'User must specify the exact internal standard column header name as it is appears in the input CSV file. The default is \"heavy Area\"'}, {'name': 'generate.cv.table', 'value': '1', 'label': 'Generate.Cv.Table', 'description': 'If \"yes\" then generate CV (coefficient of variation) table'}, {'name': 'generate.calcurves', 'value': '1', 'label': 'Generate.Calcurves', 'description': 'If \"yes\" then generate calibration curves'}, {'name': 'generate.lodloq.table', 'value': '1', 'label': 'Generate.Lodloq.Table', 'description': 'If \"yes\" then generate LOD/LOQ table'}, {'name': 'generate.peak.area.plots', 'value': '0', 'label': 'Generate.Peak.Area.Plots', 'description': 'If \"yes\" then generate peak area plots'}, {'name': 'use.par', 'value': '0', 'label': 'Use.Par', 'description': 'If \"yes\" then use peak area ratio (PAR) for analysis (instead of concentration)'}, {'name': 'max.calcurve.linear.scale', 'value': '150', 'label': 'Max.Calcurve.Linear.Scale', 'description': 'The maximum value for linear scale in fmols/ul'}, {'name': 'max.calcurve.log.scale', 'value': '150', 'label': 'Max.Calcurve.Log.Scale', 'description': 'The maximum value for log scale in fmols/ul'}, {'name': 'perform.audit', 'value': '1', 'label': 'Perform.Audit', 'description': 'If \"yes\" then perform AuDIT for interference detection.'}, {'name': 'audit.cv.threshold', 'value': '0.2', 'label': 'Audit.Cv.Threshold', 'description': 'For AuDIT the threshold for coefficient of variation below which transition is quantification-worthy. Default is 0.2.'}, {'name': 'output.prefix', 'value': '<skyline.file_basename>', 'label': 'Output.Prefix', 'description': 'Prefix used for all output file names'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task RankNormalize: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Normalize the rows in a data set by rank', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'The dataset to be normalized'}, {'name': 'scale.to.value', 'value': '', 'label': 'Scale.To.Value', 'description': 'Result values will be scaled to this value by multiplication after normalization.  Leave this blank to scale to the number of rows in the data set.'}, {'name': 'threshold', 'value': '', 'label': 'Threshold', 'description': 'Minimum threshold for values in the data set.  Any data set value below this will be increased to the threshold value before normalization.  If more than one of these parameters is specified, the order of precedence is threshold, then ceiling, then shift.'}, {'name': 'ceiling', 'value': '', 'label': 'Ceiling', 'description': 'Maximum ceiling for values in the data set.  Any data set value above this will be decreased to the ceiling value before normalization.  If more than one of these parameters is specified, the order of precedence is threshold, then ceiling, then shift.'}, {'name': 'shift', 'value': '', 'label': 'Shift', 'description': 'Shift all values in the data set.  The shift value will be added to all data set values before normalization.  If more than one of these parameters is specified, the order of precedence is threshold, then ceiling, then shift.'}], 'default_args': [{'name': 'task_name', 'value': 'RankNormalize', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file.name', 'value': '<input.file_basename>.NORM.<input.file_extension>', 'label': 'Output.File.Name', 'description': 'The name to be given to the output file.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task Read_group_trackingToGct: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '[**Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] Converts a Cufflinks read_group_tracking file into GCT format', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'The read_group_tracking file to be converted into GCT format.'}, {'name': 'gtf.file', 'value': '', 'label': 'Gtf.File', 'description': 'An optional GTF file.  If provided, this should be the same file that you provided to Cuffdiff. You can use this to map tracking IDs in the read_group_tracking file to gene symbols for output in the GCT.  Make sure that you set the feature level to match your read_group_tracking file.'}, {'name': 'override.attribute.name.for.lookup', 'value': '', 'label': 'Override.Attribute.Name.For.Lookup', 'description': 'Use this to override the name of the attribute to search against to look up IDs when mapping with a GTF (gene_id, for example). Type in the attribute name exactly as it appears in the GTF.  The retrieval override must also be provided if this parameter is set.'}, {'name': 'override.attribute.name.for.retrieval', 'value': '', 'label': 'Override.Attribute.Name.For.Retrieval', 'description': 'Use this to override the name of the attribute to retrieve on match when mapping with a GTF (gene_name, for example). Type in the attribute name exactly as it appears in the GTF.  The lookup override must also be provided if this parameter is set.'}], 'default_args': [{'name': 'task_name', 'value': 'Read_group_trackingToGct', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file.name', 'value': '<input.file_basename>.gct', 'label': 'Output.File.Name', 'description': 'The name to be given to the GCT output file.'}, {'name': 'expression.value.column', 'value': '3', 'label': 'Expression.Value.Column', 'description': 'The column to use for extracting expression values.'}, {'name': 'screen.out.errors', 'value': 'false', 'label': 'Screen.Out.Errors', 'description': \"Set to 'yes' to exclude from the GCT file any feature having at least one non-OK quantification status across all of the sequenced samples.\"}, {'name': 'output.value.for.error', 'value': 'NA', 'label': 'Output.Value.For.Error', 'description': \"This parameter controls what expression value to write to a GCT file in those cases where a feature's expression estimate carries a non-OK quantification status.  The parameter is only relevant to your final GCT when 'screen out errors' is set to 'no'.  We strongly recommend treating these as missing values (select 'NA' or 'blank').\"}, {'name': 'feature.level.for.symbol.lookup', 'value': 'gene', 'label': 'Feature.Level.For.Symbol.Lookup', 'description': 'Select the feature level of the read_group_tracking file for mapping gene symbols from a GTF. This is ignored if no GTF is provided.'}, {'name': 'choose.gct.column.for.mapping', 'value': 'auto', 'label': 'Choose.Gct.Column.For.Mapping', 'description': 'Use this to explicitly set the GCT column to use when writing symbols retrieved from the GCT. This is automatic by default, meaning that retrieved symbols go into the Name column for gene-level files and into the Description column for other feature levels.'}, {'name': 'report.naming.conflict.details', 'value': 'false', 'label': 'Report.Naming.Conflict.Details', 'description': 'Include the naming conflict details in the summary report, rather than just giving counts of the issues. '}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task ReannotateGCT: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Reannotate a GCT file\\'s \"Description\" column with data from a CHIP file', 'required_args': [{'name': 'gct.file', 'value': '', 'label': 'Gct.File', 'description': 'A GCT file to be annotated'}, {'name': 'chip.file', 'value': '', 'label': 'Chip.File', 'description': 'CHIP file to use for annotations'}], 'default_args': [{'name': 'task_name', 'value': 'ReannotateGCT', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<gct.file_basename>_reannot.gct', 'label': 'Output.File', 'description': 'The name of your re-annotated GCT file. '}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task RemoveCopyNumberOutliers: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Reads rows of a CN file and removes outliers on a sample-by-sample basis using improvised method.', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'A CN file (.cn, .RData, or .txt)'}], 'default_args': [{'name': 'task_name', 'value': 'RemoveCopyNumberOutliers', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'num.probes', 'value': '5', 'label': 'Num.Probes', 'description': 'The number of trailing and leading probes to use to compute the median'}, {'name': 'outlier.probe.replacement', 'value': 'NA', 'label': 'Outlier.Probe.Replacement', 'description': 'The replacement value for the outlier probe'}, {'name': 'multiplicative.threshold', 'value': '4', 'label': 'Multiplicative.Threshold', 'description': 'Multiplicative threshold'}, {'name': 'additive.threshold', 'value': '0.3', 'label': 'Additive.Threshold', 'description': 'Additive threshold'}, {'name': 'output.prefix', 'value': '<input.file_basename>', 'label': 'Output.Prefix', 'description': 'A prefix for the output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task RemoveMissingValues: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Removes features and/or samples containing missing (NA) values', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'Input file (gct, res)'}], 'default_args': [{'name': 'task_name', 'value': 'RemoveMissingValues', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'remove.opt', 'value': '1', 'label': 'Remove.Opt', 'description': 'Remove rows, columns, or both containing missing values'}, {'name': 'output.filename', 'value': '<input.file_basename>.na-rm.gct', 'label': 'Output.Filename', 'description': 'output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task removeSamples: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Remove samples in dataset (.gct), using a list of samples provided', 'required_args': [{'name': 'gct.file', 'value': '', 'label': 'Gct.File', 'description': 'data file to filter'}, {'name': 'removelist', 'value': '', 'label': 'Removelist', 'description': 'list of samples to remove'}, {'name': 'extension', 'value': '', 'label': 'Extension', 'description': 'filename prefix'}], 'default_args': [{'name': 'task_name', 'value': 'removeSamples', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task RemoveSaturatedFCSEvents: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Remove saturated events from an FCS data file', 'required_args': [{'name': 'Input.FCS.data.file', 'value': '', 'label': 'Input.FCS.Data.File', 'description': 'An FCS data file to remove saturated events from.'}], 'default_args': [{'name': 'task_name', 'value': 'RemoveSaturatedFCSEvents', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Output.FCS.file.name', 'value': '<Input.FCS.data.file_basename>.without_saturated.fcs', 'label': 'Output.FCS.File.Name', 'description': 'The output FCS file name.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task RenameFile: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Creates a result file with the contents of the input but with a new name.  Where possible, this is done without copying the file but instead by simply making a link.  While it is possible to run this with a submitted file, the common use will be to change the name of a result file or a previously uploaded file.  Note that the original file will still be present in its original location. ', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'Input file to be renamed'}], 'default_args': [{'name': 'task_name', 'value': 'RenameFile', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.filename', 'value': '<input.file_basename>.cp.<input.file_extension>', 'label': 'Output.Filename', 'description': 'The name to give the output file'}, {'name': 'screen.filename', 'value': 'yes', 'label': 'Screen.Filename', 'description': 'Automatically screen out problematic characters in the filename'}, {'name': 'force.copy', 'value': 'no', 'label': 'Force.Copy', 'description': 'Force a copy to be made rather than a link. '}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task ReorderByClass: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Reorder the samples in an expression dataset and class file by class', 'required_args': [{'name': 'expression.data.filename', 'value': '', 'label': 'Expression.Data.Filename', 'description': 'Expression data - .res, .gct, .odf'}, {'name': 'class.filename', 'value': '', 'label': 'Class.Filename', 'description': 'Class file - .cls'}], 'default_args': [{'name': 'task_name', 'value': 'ReorderByClass', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'expression.data.file.output.name', 'value': '<expression.data.filename_basename>_reordered', 'label': 'Expression.Data.File.Output.Name', 'description': 'Name of expression data output file'}, {'name': 'class.file.output.name', 'value': '<class.filename_basename>_reordered', 'label': 'Class.File.Output.Name', 'description': 'Name of class output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task ReplicatesQC: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Removes cell line replicate samples that fail quality control check of cell line replicate correlations and distribution scores.', 'required_args': [{'name': 'GCT.files', 'value': '', 'label': 'GCT.Files', 'description': 'Please enter at least one and up to three GCT files.'}, {'name': 'extension', 'value': '', 'label': 'Extension', 'description': ''}], 'default_args': [{'name': 'task_name', 'value': 'ReplicatesQC', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task REVEALER: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'REVEALER (Repeated Evaluation of VariablEs conditionAL Entropy and Redundancy) is an analysis method specifically suited to find groups of genomic alterations that match in a complementary way, a predefined functional activation, dependency of drug response target profile. The method starts by considering, if available, already known genomic alterations (seed) that are the known causes or are known or assumed associated with the target. REVEALER starts by finding the genomic alteration that best matches the target profile conditional to the known seed profile using the conditional mutual information. The newly discovered alteration is then merged with the seed to form a new summary feature, and then the process repeats itself finding additional complementary alterations that explain more and more of the target profile.', 'required_args': [], 'default_args': [{'name': 'task_name', 'value': 'REVEALER', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'ds1', 'value': 'https://personal.broadinstitute.org/tamayo/REVEALER/datasets/CTNBB1_transcriptional_reporter.gct', 'label': 'Ds1', 'description': 'Target profile dataset (e.g. https://personal.broadinstitute.org/tamayo/REVEALER/datasets/CTNBB1_transcriptional_reporter.gct)'}, {'name': 'target.name', 'value': 'BETA-CATENIN-REPORTER', 'label': 'Target.Name', 'description': 'Target profile name (e.g. BETA-CATENIN-REPORTER)'}, {'name': 'target.match', 'value': 'positive', 'label': 'Target.Match', 'description': 'Direction of match (e.g. positive)'}, {'name': 'ds2', 'value': 'https://personal.broadinstitute.org/tamayo/REVEALER/datasets/CCLE_MUT_CNA_AMP_DEL_0.70_2fold.MC.gct', 'label': 'Ds2', 'description': 'Features Dataset (e.g. https://personal.broadinstitute.org/tamayo/REVEALER/datasets/CCLE_MUT_CNA_AMP_DEL_0.70_2fold.MC.gct)'}, {'name': 'seed.names', 'value': 'CTNNB1.MC_MUT', 'label': 'Seed.Names', 'description': 'seed(s) names (e.g. CTNNB1.MC_MUT)'}, {'name': 'max.n.iter', 'value': '2', 'label': 'Max.N.Iter', 'description': 'Maximum number of iterations (e.g. 2)'}, {'name': 'n.markers', 'value': '30', 'label': 'N.Markers', 'description': 'Number of top hits to show in heatmap for every iteration (e.g. 30)'}, {'name': 'locs.table.file', 'value': 'https://personal.broadinstitute.org/tamayo/REVEALER/datasets/hgnc_complete_set.Feb_20_2014.v2.txt', 'label': 'Locs.Table.File', 'description': 'Table with chromosomal location for each gene symbol (e.g. https://personal.broadinstitute.org/tamayo/REVEALER/datasets/hgnc_complete_set.Feb_20_2014.v2.txt)'}, {'name': 'count.thres.low', 'value': '3', 'label': 'Count.Thres.Low', 'description': \"Filter out features with less than count.thres.low 1's (e.g. 3)\"}, {'name': 'count.thres.high', 'value': '50', 'label': 'Count.Thres.High', 'description': \"Filter out features with more than count.thres.low 1's (e.g. 50)\"}, {'name': 'pdf.output.file', 'value': 'BCAT_vs_MUT_CNA.pdf', 'label': 'Pdf.Output.File', 'description': 'Output PDF file with results (e.g. BCAT_vs_MUT_CNA.pdf)'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task RLibDemo: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '', 'required_args': [{'name': 'message', 'value': '', 'label': 'Message', 'description': ''}], 'default_args': [{'name': 'task_name', 'value': 'RLibDemo', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task RNASeQC: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '[**Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] Calculates metrics on aligned RNA-seq data.', 'required_args': [{'name': 'bam.files', 'value': '', 'label': 'Bam.Files', 'description': 'An indexed BAM file or zipped set of indexed BAM files to be analyzed. If you are supplying a single BAM file, it should be located in the same folder as its associated index file (BAI). (If SortSam is used to index the BAM file, then the BAM and BAI files are located in the same folder.)'}, {'name': 'sample.info.file', 'value': '', 'label': 'Sample.Info.File', 'description': 'A TXT format sample info file containing a sample ID, sample file name, and notes column in tab-delimited format. - .txt'}, {'name': 'annotation.gtf', 'value': '', 'label': 'Annotation.Gtf', 'description': \"A file containing annotations in GTF format. Either an annotation gtf or annotation gtf file must be provided.Note that the provided hg19 reference sequence requires contigs using in the '1' format.\"}, {'name': 'annotation.gtf.file', 'value': '', 'label': 'Annotation.Gtf.File', 'description': 'A file containing annotations in GTF format. Either an annotation gtf file or annotation gtf must be provided.'}, {'name': 'reference.sequence', 'value': '', 'label': 'Reference.Sequence', 'description': \"A FASTA file containing a reference sequence. The reference sequence provided here has an index (.fai) and a sequence dictionary (.dict). Either a reference sequence or reference sequence file (with index and dictionary) must be provided, but not both. Note that the provided hg19 reference sequence requires contigs using in the '1' format.\"}, {'name': 'reference.sequence.file', 'value': '', 'label': 'Reference.Sequence.File', 'description': 'A FASTA file containing a reference sequence. The reference sequence must have an index (.fai) and a sequence dictionary (.dict) either in the same location or specified in the reference sequence index and reference sequence dictionary parameters. Reference sequence human_assembly19 is available in the documentation. Click on the \"documentation\" link in the upper right hand corner of this page.'}, {'name': 'reference.sequence.index', 'value': '', 'label': 'Reference.Sequence.Index', 'description': 'A file containing the index for the reference sequence - .fai'}, {'name': 'reference.sequence.dictionary', 'value': '', 'label': 'Reference.Sequence.Dictionary', 'description': 'A file containing the dictionary for the reference sequence. - .dict'}, {'name': 'transcript.type.field', 'value': '', 'label': 'Transcript.Type.Field', 'description': 'Specifies the column of the GTF file in which the transcript type is specified. By default, the module looks for an attribute called transcript_type in the annotation gtf.'}, {'name': 'rRNA.interval.file', 'value': '', 'label': 'RRNA.Interval.File', 'description': 'A file containing the genomic coordinates of rRNA. - .list'}, {'name': 'rRNA.reference.file.index', 'value': '', 'label': 'RRNA.Reference.File.Index', 'description': 'zip of index files created by BWA for the supplied rRNA reference file. Required if rRNA reference file is supplied. -.zip'}, {'name': 'transcript.level.metrics', 'value': '', 'label': 'Transcript.Level.Metrics', 'description': 'Whether to calculate transcript-level metrics in addition to sample-level metrics.'}, {'name': 'gc.content.file', 'value': '', 'label': 'Gc.Content.File', 'description': 'A file containing GC content for each of the transcripts. The file must be tab-delimited with 2 columns containing transcript name and GC content. The transcript name must appear in the GTF file.'}, {'name': 'num.downsampling.reads', 'value': '', 'label': 'Num.Downsampling.Reads', 'description': 'Perform downsampling on the given number of reads. It randomly samples the specified number of reads in all experimental samples when calculating metrics.'}, {'name': 'correlation.comparison.file', 'value': '', 'label': 'Correlation.Comparison.File', 'description': 'A GCT file to use to calculate correlation between expression values (only uses first sample if gct file contains more than one sample. Note also that expression values must be log normalized, and the identifiers must match those of the annotation GTF file)'}, {'name': 'gap.length', 'value': '', 'label': 'Gap.Length', 'description': 'Whether to plot the distribution of gap lengths'}, {'name': 'GATK.flags', 'value': '', 'label': 'GATK.Flags', 'description': 'Pass a string of flags directly to the GATK (e.g. \"-DBQ 0\" to set missing base qualities to zero).'}], 'default_args': [{'name': 'task_name', 'value': 'RNASeQC', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'single.end', 'value': 'yes', 'label': 'Single.End', 'description': 'Whether the bam file contains single end reads. Select no for paired-end reads.'}, {'name': 'num.genes', 'value': '1000', 'label': 'Num.Genes', 'description': 'The number of top-expressed genes for which to calculate metrics.'}, {'name': 'num.rRNA.downsampling.reads', 'value': '1000000', 'label': 'Num.RRNA.Downsampling.Reads', 'description': 'Downsamples to calculate rRNA rate more efficiently. Default is 1 million. Set to 0 to disable.'}, {'name': 'transcript.end.length', 'value': '200', 'label': 'Transcript.End.Length', 'description': \"The length of the 3' or 5' end of a transcript.\"}, {'name': 'gc.margin', 'value': '0.375', 'label': 'Gc.Margin', 'description': 'Used in conjunction with the gc content file to specify the percent gc content to use as boundaries. E.g. 0.25 would set a lower cutoff of 25% and an upper cutoff of 75% (default is 0.375).'}, {'name': 'output.prefix', 'value': '<bam.files_basename>', 'label': 'Output.Prefix', 'description': 'The prefix to use for the output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task RNASeQC_CEGS: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '', 'required_args': [{'name': 'Picard.CreateSequenceDictionary1.reference.sequence.file', 'value': '', 'label': 'Picard.CreateSequenceDictionary1.Reference.Sequence.File', 'description': 'Input reference fasta or fasta.gz '}, {'name': 'TopHat2.reads.pair.1', 'value': '', 'label': 'TopHat2.Reads.Pair.1', 'description': 'Unpaired reads file or first mate for paired reads. A file or zip of files containing reads in FASTA or FASTQ format (bz2 and gz compressed files are supported).'}, {'name': 'TopHat2.reads.pair.2', 'value': '', 'label': 'TopHat2.Reads.Pair.2', 'description': 'Second mate for paired reads. A file or zip of files in FASTA or FASTQ format (bz2 and gz compressed files are supported).'}, {'name': 'Picard.ReorderSam3.reference.file', 'value': '', 'label': 'Picard.ReorderSam3.Reference.File', 'description': 'Reference sequence to which to reorder reads in FASTA format.'}, {'name': 'RNASeQC7.gc.content.file', 'value': '', 'label': 'RNASeQC7.Gc.Content.File', 'description': 'A file containing GC content for each of the transcripts. The file must be tab-delimited with 2 columns containing transcript name and GC content. The transcript name must appear in the GTF file.'}], 'default_args': [{'name': 'task_name', 'value': 'RNASeQC_CEGS', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task SamSPECTRALClusterFCS: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Cluster flow cytometry data using the SamSPECTRAL algorithm.', 'required_args': [{'name': 'Input.FCS.data.file', 'value': '', 'label': 'Input.FCS.Data.File', 'description': 'The FCS file to be clustered.'}, {'name': 'Dimensions', 'value': '', 'label': 'Dimensions', 'description': 'A comma-separated list of dimensions (parameters/channels) that shall be used for clustering. Use either parameter names or indexes. All dimensions except for Time will be used if this parameter is not provided.'}, {'name': 'Dimensions.to.transform', 'value': '', 'label': 'Dimensions.To.Transform', 'description': 'A comma-separated list of dimensions (parameters/channels) that shall transformed as specified by previous parameter. This will be ignored if no transformation is specified above. If this parameter is not provided and transformation is specified above, the algorithm will use heuristics to identify parameters that shall be transformed. These heuristics are based on how parameters are stored in the FCS file, their resolution and their name. Again, you can use either parameter names or parameter indexes to specify dimensions to transform.'}, {'name': 'Sigma', 'value': '', 'label': 'Sigma', 'description': 'A scaling parameter that determines the resolution in the spectral clustering stage. By increasing it, more spectral clusters are identified. This can be usefull when small population are aimed. See documentation on more details how Sigma can be adjusted.'}, {'name': 'Separation.factor', 'value': '', 'label': 'Separation.Factor', 'description': 'Threshold that controls to what extend clusters should be combined or kept separate. See documentation on more details how the separation factor can be adjusted.'}, {'name': 'Seed', 'value': '', 'label': 'Seed', 'description': 'Seed value for the random generator; provide any consistent value to obtain reproducible results.'}], 'default_args': [{'name': 'task_name', 'value': 'SamSPECTRALClusterFCS', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Transformation', 'value': 'ASinH', 'label': 'Transformation', 'description': 'Which transformation to apply on the data prior clustering. Fluorescence channels are usually better visualized and clustered using a transformation; the ASinH transformation produces good results on most data.'}, {'name': 'Beta', 'value': '4', 'label': 'Beta', 'description': 'The size of neighborhood that is being searched during the conductance calculation step. The default value of 4 should not be changed unless the input data set contains measurements of more that 100,000. Setting beta to zero or a low value will reduce computational time by applying an approximation to the conductance calculation step. This will allow you to cluster large data sets in reasonable time; however, the quality of the result may be affected. Setting beta to 5 is not recommended since it will increase computational time while not significantly improving clustering results in most cases.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task SAMtools.FastaIndex: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Indexes a reference sequence in FASTA format', 'required_args': [{'name': 'fasta.file', 'value': '', 'label': 'Fasta.File', 'description': 'A sequence in FASTA format.'}], 'default_args': [{'name': 'task_name', 'value': 'SAMtools.FastaIndex', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.prefix', 'value': '<fasta.file_basename>', 'label': 'Output.Prefix', 'description': 'The prefix to use for the output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task Scripture: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Segmentation-based transcript reconstruction algorithm', 'required_args': [{'name': 'alignment.file', 'value': '', 'label': 'Alignment.File', 'description': 'Alignment file in SAM or BAM format, sorted by start position'}, {'name': 'alignment.index.file', 'value': '', 'label': 'Alignment.Index.File', 'description': 'Index file for alignment file, in SAI or BAI format. The index file must have the same root name as the alignment file. (Example: foo.sam and foo.sam.sai OR foo.sai) '}, {'name': 'chromosome.size.file', 'value': '', 'label': 'Chromosome.Size.File', 'description': 'File containing sizes of chromosomes'}, {'name': 'chromosome', 'value': '', 'label': 'Chromosome', 'description': 'Selected chromosome, e.g., chr19'}, {'name': 'chromosome.sequence.file', 'value': '', 'label': 'Chromosome.Sequence.File', 'description': 'Chromosome sequence in FASTA format'}], 'default_args': [{'name': 'task_name', 'value': 'Scripture', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.prefix', 'value': '<alignment.file_basename>', 'label': 'Output.Prefix', 'description': 'A label that will be used to name output files'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task ScripturePipeline: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Segmentation-based transcript reconstruction using single- or paired-end reads', 'required_args': [{'name': 'TopHat1.prebuilt.bowtie.index', 'value': '', 'label': 'TopHat1.Prebuilt.Bowtie.Index', 'description': 'A prebuilt bowtie index. Either a prebuilt or a custom Bowtie index must be specified.'}, {'name': 'TopHat1.custom.bowtie.index', 'value': '', 'label': 'TopHat1.Custom.Bowtie.Index', 'description': 'A ZIP archive containing Bowtie index files. Either a prebuilt or a custom Bowtie index must be specified.'}, {'name': 'TopHat1.reads.pair.1', 'value': '', 'label': 'TopHat1.Reads.Pair.1', 'description': 'Unpaired reads file or first mate for paired reads. A file or zip of files containing reads in FASTA or FASTQ format.'}, {'name': 'TopHat1.reads.pair.2', 'value': '', 'label': 'TopHat1.Reads.Pair.2', 'description': 'Second mate for paired reads. A file or zip of files in FASTA or FASTQ format.'}, {'name': 'TopHat1.mate.inner.dist', 'value': '', 'label': 'TopHat1.Mate.Inner.Dist', 'description': 'The expected mean inner distance between mate pairs.'}, {'name': 'TopHat1.mate.std.dev', 'value': '', 'label': 'TopHat1.Mate.Std.Dev', 'description': 'The standard deviation for the distribution on inner distances between mate pairs.'}, {'name': 'TopHat1.library.type', 'value': '', 'label': 'TopHat1.Library.Type', 'description': 'Library type for strand specific reads.'}, {'name': 'Scripture3.chromosome.size.file', 'value': '', 'label': 'Scripture3.Chromosome.Size.File', 'description': 'File containing sizes of chromosomes'}, {'name': 'Scripture3.chromosome', 'value': '', 'label': 'Scripture3.Chromosome', 'description': 'Selected chromosome, e.g., chr19'}, {'name': 'Scripture3.chromosome.sequence.file', 'value': '', 'label': 'Scripture3.Chromosome.Sequence.File', 'description': 'Chromosome sequence in FASTA format'}], 'default_args': [{'name': 'task_name', 'value': 'ScripturePipeline', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'TopHat1.integer.quals', 'value': 'no', 'label': 'TopHat1.Integer.Quals', 'description': 'Quality values are space-delimited integer values, this becomes default when you specify color option.'}, {'name': 'Scripture3.output.prefix', 'value': '<alignment.file_basename>', 'label': 'Scripture3.Output.Prefix', 'description': 'A label that will be used to name output files'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task ScripturePrealigned: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Segmentation-based transcript reconstruction using pre-aligned reads', 'required_args': [{'name': 'SortSam1.input.file', 'value': '', 'label': 'SortSam1.Input.File', 'description': 'Input file in SAM or BAM format'}, {'name': 'Scripture2.chromosome.size.file', 'value': '', 'label': 'Scripture2.Chromosome.Size.File', 'description': 'File containing sizes of chromosomes'}, {'name': 'Scripture2.chromosome', 'value': '', 'label': 'Scripture2.Chromosome', 'description': 'Selected chromosome, e.g., chr19'}, {'name': 'Scripture2.chromosome.sequence.file', 'value': '', 'label': 'Scripture2.Chromosome.Sequence.File', 'description': 'Chromosome sequence in FASTA format'}], 'default_args': [{'name': 'task_name', 'value': 'ScripturePrealigned', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Scripture2.output.prefix', 'value': '<alignment.file_basename>', 'label': 'Scripture2.Output.Prefix', 'description': 'A label that will be used to name output files'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task SegmentationCount: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Counts the number of segments in a SEG file.', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'A segmentation file - .seg, .txt'}], 'default_args': [{'name': 'task_name', 'value': 'SegmentationCount', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<input.file_basename>.segcount.txt', 'label': 'Output.File', 'description': 'The name of the output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task SelectFeaturesColumns: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': \"Takes a 'column slice' from a .res, .gct, .snp, .cn, .loh, .odf, or .cls file.\", 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'input file - .res, .gct, .snp, .cn, .loh, .odf, or .cls file from which the slice is extracted'}, {'name': 'columns', 'value': '', 'label': 'Columns', 'description': 'columns to select (e.g. 0-3, 5, 9). Either columns or a list file must be provided.'}, {'name': 'list.filename', 'value': '', 'label': 'List.Filename', 'description': 'text file containing one identifier per line that specifies the columns to select. Either columns or a list file must be provided.'}], 'default_args': [{'name': 'task_name', 'value': 'SelectFeaturesColumns', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output', 'value': '<input.filename_basename>.slice', 'label': 'Output', 'description': 'output filename - .res, .gct, .snp, .cn, .loh, .odf, .cls'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task SelectFeaturesRows: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Generate a .res, .gct, .snp, .cn, .loh, or .odf file containing only specified features (i.e. rows).', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'input file - .res, .gct, .snp, .cn, .loh, or .odf file from which the slice is extracted'}, {'name': 'list.filename', 'value': '', 'label': 'List.Filename', 'description': 'text file containing one identifier per line that specifies the rows to select. '}], 'default_args': [{'name': 'task_name', 'value': 'SelectFeaturesRows', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output', 'value': '<list.filename_basename>.slice', 'label': 'Output', 'description': 'output filename - .res, .gct, .snp, .cn, .loh, .odf'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task SelectFileMatrix: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'A helper module which selects a matrix out of a delimited text file.', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'Name of the input file from which to pull the matrix.  Must be a delimited text file.'}, {'name': 'start.row', 'value': '', 'label': 'Start.Row', 'description': 'Starting row of the matrix (line number within the file).  Defaults to the first row.'}, {'name': 'end.row', 'value': '', 'label': 'End.Row', 'description': 'Ending row of the matrix (line number within the file).  Defaults to the last row.'}, {'name': 'start.column', 'value': '', 'label': 'Start.Column', 'description': 'Starting column of the matrix.  Defaults to the first column.'}, {'name': 'end.column', 'value': '', 'label': 'End.Column', 'description': 'Ending column of the matrix. Defaults to the last column.'}, {'name': 'row.select', 'value': '', 'label': 'Row.Select', 'description': 'Select specific rows to include in the matrix (e.g. \"3,5,7-12\").  Note that start/end row parameters may limit this selection.'}, {'name': 'column.select', 'value': '', 'label': 'Column.Select', 'description': 'Select specific columns to include in the matrix (e.g. \"1,2,4-6\").  Note that start/end column parameters may limit this selection'}], 'default_args': [{'name': 'task_name', 'value': 'SelectFileMatrix', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file.base.name', 'value': '<input.file_basename>.select', 'label': 'Output.File.Base.Name', 'description': 'Base name for the output file.  The module will add an appropriate extension, either .csv for a comma-delimted file or .txt for a space- or tab-delimited file).'}, {'name': 'delimiter', 'value': 'TAB', 'label': 'Delimiter', 'description': 'Token naming the character for the output column delimiter. Choices are TAB, COMMA, and SPACE. TAB is the default.'}, {'name': 'output.rejects', 'value': 'false', 'label': 'Output.Rejects', 'description': 'Produce an output file containing the items not meeting the selection criteria.  The file will be named <output.file.base.name>.REJECT.<ext>'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task SetFCSKeywords: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Sets keyword/value(s) in a Flow Cytometry Standard (FCS) file.', 'required_args': [{'name': 'Input.FCS.data.file', 'value': '', 'label': 'Input.FCS.Data.File', 'description': 'The input FCS data file in which to set keyword/value pairs.'}, {'name': 'Keywords.input.file', 'value': '', 'label': 'Keywords.Input.File', 'description': 'The file with keyword/value pairs to add (or change values of existing keywords). The keyword/value pairs can be submitted either as an CSV file with two columns: \"Keyword\" and \"Value\" (heading is optional) or as a text file with a chunk of FCS TEXT segment, e.g., /$P3S/MUC 1 APC/$P4S/Thy1 PE/BD$CUSTOM_VALUE23/12/.'}, {'name': 'Output.fcs.file.name', 'value': '', 'label': 'Output.Fcs.File.Name', 'description': 'The file name for the output FCS data file.'}], 'default_args': [{'name': 'task_name', 'value': 'SetFCSKeywords', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task shRNAcollapseReps: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Collapses replicate samples in a gct data file, based on a sample information file', 'required_args': [{'name': 'gct.file', 'value': '', 'label': 'Gct.File', 'description': 'The input gct data file.'}, {'name': 'sample.info.file', 'value': '', 'label': 'Sample.Info.File', 'description': 'The cell line (sample) information file.'}, {'name': 'prefix', 'value': '', 'label': 'Prefix', 'description': 'The file name prefix.'}], 'default_args': [{'name': 'task_name', 'value': 'shRNAcollapseReps', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task shRNAfoldChange: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Calculate the fold change of each sample in a gct file, based on the appropriate DNA reference value (present in a gct file of the same dimension).', 'required_args': [{'name': 'gct.file', 'value': '', 'label': 'Gct.File', 'description': 'The input gct data file.'}, {'name': 'DNA.reference.file', 'value': '', 'label': 'DNA.Reference.File', 'description': 'The values of the DNA pool reference per sample/shRNA in a gct file (same dimension as input gct file).'}, {'name': 'prefix', 'value': '', 'label': 'Prefix', 'description': 'The file name prefix.'}], 'default_args': [{'name': 'task_name', 'value': 'shRNAfoldChange', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task shRNAmapGenes: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Maps shRNAs in a gct data file to genes based on a mapping file.', 'required_args': [{'name': 'gct.file', 'value': '', 'label': 'Gct.File', 'description': 'The input gct data file.'}, {'name': 'mapping', 'value': '', 'label': 'Mapping', 'description': 'The shRNA to gene mapping file.'}, {'name': 'prefix', 'value': '', 'label': 'Prefix', 'description': 'The file name prefix.'}], 'default_args': [{'name': 'task_name', 'value': 'shRNAmapGenes', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task shRNAremoveOverlap: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Remove shRNAs overlapping in sequence > 3 basepairs from a gct data file', 'required_args': [{'name': 'list', 'value': '', 'label': 'List', 'description': 'The list of shRNAs that are not overlapping.'}, {'name': 'gct.file', 'value': '', 'label': 'Gct.File', 'description': 'The input gct data file.'}, {'name': 'mapping', 'value': '', 'label': 'Mapping', 'description': 'The shRNA to gene mapping file.'}, {'name': 'prefix', 'value': '', 'label': 'Prefix', 'description': 'The file name prefix.'}], 'default_args': [{'name': 'task_name', 'value': 'shRNAremoveOverlap', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task SNPFileCreator: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Process Affymetrix SNP probe-level data into an expression value', 'required_args': [{'name': 'chip.type', 'value': '', 'label': 'Chip.Type', 'description': 'Specify the chip type.'}, {'name': 'network.directory.name', 'value': '', 'label': 'Network.Directory.Name', 'description': 'Networked directory name where the CEL ( and optionally TXT) files reside.'}, {'name': 'zip.filename', 'value': '', 'label': 'Zip.Filename', 'description': 'A zip file containing CEL and optionally TXT files.'}, {'name': 'reference.filename', 'value': '', 'label': 'Reference.Filename', 'description': 'Reference file to use. Leave blank if reference method is median.'}, {'name': 'output.file', 'value': '', 'label': 'Output.File', 'description': 'The name of the output file.'}], 'default_args': [{'name': 'task_name', 'value': 'SNPFileCreator', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'normalization.method', 'value': '2', 'label': 'Normalization.Method', 'description': 'The normalization method to apply.'}, {'name': 'reference.method', 'value': 'median', 'label': 'Reference.Method', 'description': 'Reference method to use in normalization'}, {'name': 'model.method', 'value': '2', 'label': 'Model.Method', 'description': 'The summarization method to apply.'}, {'name': 'allele.specific', 'value': 'no', 'label': 'Allele.Specific', 'description': 'Create Allele Specific probe intensities'}, {'name': 'sort.snp.file', 'value': 'yes', 'label': 'Sort.Snp.File', 'description': 'Sorts output file by Chromosome and location. Filters out probesets without location information'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task SNPFileSorter: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Sorts a .snp file by Chromosome and location', 'required_args': [{'name': 'snp.filename', 'value': '', 'label': 'Snp.Filename', 'description': 'snp file to sort'}], 'default_args': [{'name': 'task_name', 'value': 'SNPFileSorter', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<snp.filename_basename>.sorted.snp', 'label': 'Output.File', 'description': 'output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task SNPMultipleSampleAnalysis: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Determine Regions of Concordant Copy Number Aberrations', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'The input file to analyze - .cn, .xcn, .snp, .txt'}, {'name': 'run.name', 'value': '', 'label': 'Run.Name', 'description': 'The name of the job'}], 'default_args': [{'name': 'task_name', 'value': 'SNPMultipleSampleAnalysis', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'java.max.memory', 'value': '512m', 'label': 'Java.Max.Memory', 'description': ''}, {'name': 'save.dir', 'value': './', 'label': 'Save.Dir', 'description': 'Directory to save to'}, {'name': 'permutations', 'value': '500', 'label': 'Permutations', 'description': 'Number of permutations to use'}, {'name': 'num.tests', 'value': '9', 'label': 'Num.Tests', 'description': 'Number of MSA tests to use'}, {'name': 'num.positions.bin', 'value': '6', 'label': 'Num.Positions.Bin', 'description': 'Number of positions per bin'}, {'name': 'resolution', 'value': '250000', 'label': 'Resolution', 'description': 'The size of the plotted chromosome diagram'}, {'name': 'chromosomes', 'value': 'chr22', 'label': 'Chromosomes', 'description': 'Comma delimited list of chromosomes to analyze'}, {'name': 'save.images', 'value': './', 'label': 'Save.Images', 'description': 'Saves images to the specified directory'}, {'name': 'significance.level', 'value': '.05', 'label': 'Significance.Level', 'description': 'The value at which a region is called significant'}, {'name': 'analyze.by', 'value': 'false', 'label': 'Analyze.By', 'description': ''}, {'name': 'cytoband.filename', 'value': '<libdir>CytoBands.txt', 'label': 'Cytoband.Filename', 'description': 'Cytoband file to use'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task SnpViewer: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Displays a heat map of snp data', 'required_args': [{'name': 'dataset.filename', 'value': '', 'label': 'Dataset.Filename', 'description': 'The dataset to display - .snp, .cn, .xcn'}, {'name': 'sample.info.filename', 'value': '', 'label': 'Sample.Info.Filename', 'description': 'Sample information file'}, {'name': 'sample.list.filename', 'value': '', 'label': 'Sample.List.Filename', 'description': 'Sample list file'}], 'default_args': [{'name': 'task_name', 'value': 'SnpViewer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'max.memory', 'value': '700m', 'label': 'Max.Memory', 'description': 'Maximum memory'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task SOMClustering: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Self-Organizing Maps algorithm', 'required_args': [{'name': 'dataset.filename', 'value': '', 'label': 'Dataset.Filename', 'description': 'Dataset (res, gct, or odf dataset)'}], 'default_args': [{'name': 'task_name', 'value': 'SOMClustering', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.stub', 'value': '<dataset.filename_basename>', 'label': 'Output.Stub', 'description': 'The base output file name - .odf'}, {'name': 'cluster.range', 'value': '2-3', 'label': 'Cluster.Range', 'description': 'Range of values can be entered and the program will automatically run the algorithm for each number of clusters in the range.'}, {'name': 'seed.range', 'value': '42', 'label': 'Seed.Range', 'description': 'The seed for the random number generator, is exposed to allow the user to recreate a given session at a later time, (as opposed to allowing the program to generate a random initial seed which could result in different outcomes even if all the other parameters are identical).'}, {'name': 'iterations', 'value': '50000', 'label': 'Iterations', 'description': 'How many times the algorithm should try to refine the clusters. Initially, this value can be set low for faster exploration, but should be set high (e.g., 50,000) for good convergence.'}, {'name': 'cluster.by', 'value': 'rows', 'label': 'Cluster.By', 'description': 'Whether to cluster by rows or columns'}, {'name': 'som.rows', 'value': '0', 'label': 'Som.Rows', 'description': 'Setting this and som.cols to a non zero value will override cluster.range and the computation will be for the specified geometry'}, {'name': 'som.cols', 'value': '0', 'label': 'Som.Cols', 'description': 'Setting this and som.rows to a non zero value will override cluster.range and the computation will be for the specified geometry'}, {'name': 'initialization', 'value': 'Random_Vectors', 'label': 'Initialization', 'description': 'How to select inital random centroids'}, {'name': 'neighborhood', 'value': 'Bubble', 'label': 'Neighborhood', 'description': 'Neighborhood function determining how centroids near the target centroid are updated'}, {'name': 'alpha.initial', 'value': '0.1', 'label': 'Alpha.Initial', 'description': 'Initial learning weight for centroid updates'}, {'name': 'alpha.final', 'value': '0.005', 'label': 'Alpha.Final', 'description': 'Final learning weight for centroid updates'}, {'name': 'sigma.initial', 'value': '5.0', 'label': 'Sigma.Initial', 'description': 'Initial sigma to determine update neighborhood size around the target centroid'}, {'name': 'sigma.final', 'value': '0.5', 'label': 'Sigma.Final', 'description': 'Final sigma to determine update neighborhood size around the target centroid'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task SOMClusterViewer: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Visualize clusters created with the SOM algorithm', 'required_args': [{'name': 'som.cluster.filename', 'value': '', 'label': 'Som.Cluster.Filename', 'description': 'SOM Properties ODF file'}], 'default_args': [{'name': 'task_name', 'value': 'SOMClusterViewer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task SparseHierarchicalClustering: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Cluster the samples in a data matrix using an adaptively-chosen subset of the features.', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'A GCT file containing data to cluster'}], 'default_args': [{'name': 'task_name', 'value': 'SparseHierarchicalClustering', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'method', 'value': 'average', 'label': 'Method', 'description': 'The type of linkage to use in the hierarchical clustering - \"single\", \"complete\", \"centroid\", or \"average.\"'}, {'name': 'wbound', 'value': '-1', 'label': 'Wbound', 'description': 'The sum of gene weights; this is the tuning parameter for sparse hierarchical clustering. If a non-negative number is given, then this tuning parameter is used. Otherwise, a tuning parameter is selected via a permutation approach. Note: The program runs much more quickly if wbound is specified rather than left at -1.'}, {'name': 'maxnumgenes', 'value': '5000', 'label': 'Maxnumgenes', 'description': 'By default, only the 5000 genes with highest variance will be used (for computational reasons). To use all the genes, set to -1. Warning: using all genes will be slow if there are many genes in the data set.'}, {'name': 'cluster.features', 'value': 'false', 'label': 'Cluster.Features', 'description': 'If a clustering of the genes with non-zero weights is desired, set to TRUE.'}, {'name': 'method.features', 'value': 'average', 'label': 'Method.Features', 'description': 'The type of linkage used to cluster the features (if desired) - \"single\", \"complete\", \"centroid\", or \"average\". Used only if cluster features is set to \"TRUE\"'}, {'name': 'standardize.arrays', 'value': 'true', 'label': 'Standardize.Arrays', 'description': 'Whether or not the arrays should be standardized to have mean zero and standard deviation one before clustering is performed.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task SplitColumns: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Creates a separate file for each column specified from the input file', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'A tab delimited file'}, {'name': 'split.columns', 'value': '', 'label': 'Split.Columns', 'description': 'Specify columns to extract (i.e. 2,4, 5-, 11-17,). Leave blank to split all columns.'}, {'name': 'additional.columns', 'value': '', 'label': 'Additional.Columns', 'description': 'Specify subset of additional columns to include in each (i.e. 2,4, 5-, 11-17) output file.'}], 'default_args': [{'name': 'task_name', 'value': 'SplitColumns', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.prefix', 'value': '<input.file_basename>', 'label': 'Output.Prefix', 'description': 'The prefix to use for the output file name.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task SplitDatasetTrainTest: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Splits a dataset (and cls file) into a number of train and test subsets', 'required_args': [{'name': 'input.dataset.file', 'value': '', 'label': 'Input.Dataset.File', 'description': 'input dataset - .gct, .res'}, {'name': 'cls.file', 'value': '', 'label': 'Cls.File', 'description': 'class template - .cls'}], 'default_args': [{'name': 'task_name', 'value': 'SplitDatasetTrainTest', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'split.method', 'value': 'random', 'label': 'Split.Method', 'description': 'whether to split the data using a percentage split or cross validation approach'}, {'name': 'stratified', 'value': 'T', 'label': 'Stratified', 'description': 'whether to create splits stratified with respect to the class template'}, {'name': 'folds', 'value': '1', 'label': 'Folds', 'description': 'number of train/test splits to generate'}, {'name': 'percentage.split.proportion', 'value': '0.5', 'label': 'Percentage.Split.Proportion', 'description': 'proportion of data to be allocated to train file when split method is percentage split (ignored when split method is cross-validation)'}, {'name': 'seed', 'value': '123456', 'label': 'Seed', 'description': 'random number generator seed'}, {'name': 'output.prefix', 'value': '<input.dataset.file_basename>', 'label': 'Output.Prefix', 'description': 'output saved to output.prefix.{train,test}.n.{gct|res,cls}, where n is the fold index'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task SraToFastQ: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Converts data in SRA format to FastQ using the SRAToolkit version 2.3.5-2.', 'required_args': [{'name': 'accession', 'value': '', 'label': 'Accession', 'description': 'A Sequence Read Archive accession.  Either an accession or an sra.file is required.'}, {'name': 'sra.file', 'value': '', 'label': 'Sra.File', 'description': 'A Sequence Read Archive (.sra) file.  Either an accession or an sra.file is required.'}, {'name': 'paired.end', 'value': '', 'label': 'Paired.End', 'description': 'Whether the reads are paired end'}, {'name': 'quality.score.offset', 'value': '', 'label': 'Quality.Score.Offset', 'description': 'Offset to use for ASCII quality scores. Default is 33 (\"!\").'}, {'name': 'biological.reads.only', 'value': '', 'label': 'Biological.Reads.Only', 'description': 'Whether to include only biological reads and no technical reads in output'}], 'default_args': [{'name': 'task_name', 'value': 'SraToFastQ', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'alignment.status', 'value': 'any', 'label': 'Alignment.Status', 'description': 'The alignment status of the reads to include in output'}, {'name': 'output.compression', 'value': 'none', 'label': 'Output.Compression', 'description': 'Compression to apply to the output files'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task ssGSEAProjection: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Performs single sample GSEA Projection', 'required_args': [{'name': 'input.gct.file', 'value': '', 'label': 'Input.Gct.File', 'description': 'Input gct file.  This module does not allow missing values.'}, {'name': 'output.file.prefix', 'value': '', 'label': 'Output.File.Prefix', 'description': 'Output filename prefix.  If unspecified, is set to <prefix of input.gct.file>.PROJ.'}, {'name': 'gene.sets.database.files', 'value': '', 'label': 'Gene.Sets.Database.Files', 'description': 'Gene sets description file(s) - .gmt, .gmx.  Upload a gene set if your gene set is not listed as a choice for the gene sets database parameter.'}], 'default_args': [{'name': 'task_name', 'value': 'ssGSEAProjection', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'gene.sets.database', 'value': 'c1.all.v5.2.symbols.gmt', 'label': 'Gene.Sets.Database', 'description': 'Gene sets database from GSEA website.'}, {'name': 'gene.symbol.column', 'value': 'Name', 'label': 'Gene.Symbol.Column', 'description': 'Specifies the gct data column containing gene symbols.  In most cases this will be \"Column 1\".'}, {'name': 'gene.set.selection', 'value': 'ALL', 'label': 'Gene.Set.Selection', 'description': 'Comma-separated list of gene set names on which to project expression data. Set to \"ALL\" if projecting to all defined gene sets.'}, {'name': 'sample.normalization.method', 'value': 'rank', 'label': 'Sample.Normalization.Method', 'description': 'Method used to normalize gene expression data'}, {'name': 'weighting.exponent', 'value': '0.75', 'label': 'Weighting.Exponent', 'description': 'Module authors strongly recommend against changing from default.'}, {'name': 'min.gene.set.size', 'value': '10', 'label': 'Min.Gene.Set.Size', 'description': 'Gene sets with overlap smaller than this are excluded from the analysis.'}, {'name': 'combine.mode', 'value': 'combine.add', 'label': 'Combine.Mode', 'description': 'Options for combining enrichment scores for paired *_UP and *_DN gene sets'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task StrandSpecRNAseqEval: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Evaluates strand-specific yeast (S. cerevisiae) RNA-seq libraries', 'required_args': [{'name': 'sam.file', 'value': '', 'label': 'Sam.File', 'description': 'The input SAM file to read and evaluate - .sam'}], 'default_args': [{'name': 'task_name', 'value': 'StrandSpecRNAseqEval', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'average.gene.stats', 'value': '0', 'label': 'Average.Gene.Stats', 'description': 'A flag to indicate whether to run the average gene statistics. This is a long process.'}, {'name': 'flip.strands', 'value': '0', 'label': 'Flip.Strands', 'description': 'A flag to indicate whether the data is flipped, so a forward gene would appear as a reverse one'}, {'name': 'output.prefix', 'value': '<sam.file_basename>', 'label': 'Output.Prefix', 'description': 'A prefix for the output files'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task SubMap: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Maps subclasses between two data sets', 'required_args': [{'name': 'datasetA.file', 'value': '', 'label': 'DatasetA.File', 'description': 'Input dataset A - .gct, should have common gene ID with dataset B'}, {'name': 'datasetB.file', 'value': '', 'label': 'DatasetB.File', 'description': 'Input dataset B - .gct,  should have common gene ID with dataset A'}, {'name': 'classA.file', 'value': '', 'label': 'ClassA.File', 'description': 'Input class label A - .cls, 3rd line should be numeric, +1 if starts from \"0\"'}, {'name': 'classB.file', 'value': '', 'label': 'ClassB.File', 'description': 'Input class label B - .cls, 3rd line should be numeric, +1 if starts from \"0\"'}], 'default_args': [{'name': 'task_name', 'value': 'SubMap', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'num.marker.genes', 'value': '100', 'label': 'Num.Marker.Genes', 'description': 'Number of marker genes to be mapped'}, {'name': 'num.perm', 'value': '100', 'label': 'Num.Perm', 'description': 'Number of random permutations for ES in each cell of SA matrix'}, {'name': 'num.perm.fisher', 'value': '1000', 'label': 'Num.Perm.Fisher', 'description': \"Number of random permutations for Fisher's statistics\"}, {'name': 'weighted.score.type', 'value': '1', 'label': 'Weighted.Score.Type', 'description': 'Weight enrichment by correlation vector (SNR)'}, {'name': 'null.distribution', 'value': 'pool', 'label': 'Null.Distribution', 'description': 'Choice of null distribution'}, {'name': 'p.value.correction', 'value': 'Bonferroni', 'label': 'P.Value.Correction', 'description': 'p-value correction method'}, {'name': 'cluster.rows', 'value': '1', 'label': 'Cluster.Rows', 'description': \"Cluster dataset A's subclass in heatmap of SA matrix\"}, {'name': 'cluster.columns', 'value': '1', 'label': 'Cluster.Columns', 'description': \"Cluster dataset B's subclass in heatmap of SA matrix\"}, {'name': 'nominal.p.value.matrix', 'value': 'T', 'label': 'Nominal.P.Value.Matrix', 'description': 'Create heatmap of each nominal-p matrix'}, {'name': 'create.legend', 'value': 'T', 'label': 'Create.Legend', 'description': 'Create legend for heatmap'}, {'name': 'random.seed', 'value': '47365321', 'label': 'Random.Seed', 'description': 'Random seed'}, {'name': 'output.filename', 'value': 'SubMap', 'label': 'Output.Filename', 'description': 'Name of output files containing heatmap of SA matrices, summary of ES matrix, nominal p-values, and corrected p-values'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task SuggestNumberOfPopulationsFCS: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'The FlowClust module requires that you specify how many populations you are looking for.  This module takes a data driven approach and attempts to discover how many populations are present in an FCS file', 'required_args': [{'name': 'Input.FCS.data.file', 'value': '', 'label': 'Input.FCS.Data.File', 'description': 'The FCS file'}, {'name': 'Dimensions', 'value': '', 'label': 'Dimensions', 'description': 'The channels to search for populations, separated by commas. Leave blank to search in all channels. E.g. FSC Lin, SS Lin, CD4 Log'}, {'name': 'Range', 'value': '', 'label': 'Range', 'description': 'By default, we try between 1 and 30 populations and report the most likely number in the file. As this is a time consuming procedure, you can specify a more limited range. e.g. 1,12'}], 'default_args': [{'name': 'task_name', 'value': 'SuggestNumberOfPopulationsFCS', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task SurvivalCurve: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Draws a survival curve based on a cls file', 'required_args': [{'name': 'input.surv.data.file', 'value': '', 'label': 'Input.Surv.Data.File', 'description': 'input clinical data including survival time & censor status -.txt'}, {'name': 'input.cls.file', 'value': '', 'label': 'Input.Cls.File', 'description': 'Class information file -.cls'}, {'name': 'cls.field', 'value': '', 'label': 'Cls.Field', 'description': 'Name of class field in survival data file (optional)'}, {'name': 'manual.line.type', 'value': '', 'label': 'Manual.Line.Type', 'description': 'e.g. 1,2,3,... (1: solid, 2: dashed, 3: dotted, 4: dot-dashed)'}, {'name': 'manual.line.color', 'value': '', 'label': 'Manual.Line.Color', 'description': 'e.g. black, blue, red, ...'}, {'name': 'max.time', 'value': '', 'label': 'Max.Time', 'description': 'Maximum survival time (in original scale)'}], 'default_args': [{'name': 'task_name', 'value': 'SurvivalCurve', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'time.field', 'value': 'time', 'label': 'Time.Field', 'description': 'Name of survival time field'}, {'name': 'censor.field', 'value': 'status', 'label': 'Censor.Field', 'description': 'Name of survival time field (value should be \"event:1, censor:0\")'}, {'name': 'print.fit.results', 'value': 'F', 'label': 'Print.Fit.Results', 'description': 'Output file name for survival fit results'}, {'name': 'line.type.color.assign', 'value': 'automatic', 'label': 'Line.Type.Color.Assign', 'description': 'Assign line type/color automatically/manually?'}, {'name': 'line.width', 'value': '1', 'label': 'Line.Width', 'description': 'Line width'}, {'name': 'time.conversion', 'value': '1', 'label': 'Time.Conversion', 'description': 'Survival time conversion'}, {'name': 'surv.function.lower', 'value': '0', 'label': 'Surv.Function.Lower', 'description': 'Lower limit of y axis range'}, {'name': 'surv.function.higher', 'value': '1', 'label': 'Surv.Function.Higher', 'description': 'Upper limit of y axis range'}, {'name': 'curve.type', 'value': 'log', 'label': 'Curve.Type', 'description': 'Type of survival curve'}, {'name': 'show.conf.interval', 'value': '0', 'label': 'Show.Conf.Interval', 'description': 'Show confidence inteval?'}, {'name': 'add.legend', 'value': 'T', 'label': 'Add.Legend', 'description': 'Show legend?'}, {'name': 'legend.position', 'value': 'left-bottom', 'label': 'Legend.Position', 'description': 'Position of legend'}, {'name': 'output.filename', 'value': '<input.surv.data.file_basename>', 'label': 'Output.Filename', 'description': 'Name for output files'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task SurvivalDifference: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Tests for survival difference based on cls file', 'required_args': [{'name': 'input.surv.data.file', 'value': '', 'label': 'Input.Surv.Data.File', 'description': 'input clinical data including survival time & censor status  -.txt'}, {'name': 'input.cls.file', 'value': '', 'label': 'Input.Cls.File', 'description': 'class information file (.cls format)'}, {'name': 'cls.field', 'value': '', 'label': 'Cls.Field', 'description': 'Name of class field in survival data file (optional)'}, {'name': 'subgroup', 'value': '', 'label': 'Subgroup', 'description': 'Suggroups (classes) to be compared (e.g., A, C as in 3rd line of cls file)'}], 'default_args': [{'name': 'task_name', 'value': 'SurvivalDifference', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'time.field', 'value': 'time', 'label': 'Time.Field', 'description': 'Name of survival time field'}, {'name': 'censor.field', 'value': 'status', 'label': 'Censor.Field', 'description': 'Name of censor field (value should be \"event:1, no event: 0\")'}, {'name': 'test.type', 'value': '0', 'label': 'Test.Type', 'description': 'Type of test'}, {'name': 'output.filename', 'value': '<input.surv.data.file_basename>', 'label': 'Output.Filename', 'description': 'Output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task SVM: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Classify samples using SVM', 'required_args': [{'name': 'train.data.filename', 'value': '', 'label': 'Train.Data.Filename', 'description': 'The training data file - .gct, .res'}, {'name': 'train.cls.filename', 'value': '', 'label': 'Train.Cls.Filename', 'description': 'The training class file - .cls'}, {'name': 'saved.model.filename', 'value': '', 'label': 'Saved.Model.Filename', 'description': 'A saved model from a previous run of SVM'}, {'name': 'test.data.filename', 'value': '', 'label': 'Test.Data.Filename', 'description': 'The test data file - .gct, .res'}, {'name': 'test.cls.filename', 'value': '', 'label': 'Test.Cls.Filename', 'description': 'The test class file - .cls'}], 'default_args': [{'name': 'task_name', 'value': 'SVM', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'pred.results.output.file', 'value': '<test.data.filename_basename>.pred.odf', 'label': 'Pred.Results.Output.File', 'description': 'The name of the output file for prediction results'}, {'name': 'model.output.file', 'value': '<train.data.filename_basename>.model', 'label': 'Model.Output.File', 'description': 'The name of the file containing the saved model'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task TmapToGct: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Converts a TMAP file to GCT format', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'Input file in TMAP format'}], 'default_args': [{'name': 'task_name', 'value': 'TmapToGct', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'row.labels', 'value': 'Cufflinks gene ID', 'label': 'Row.Labels', 'description': 'Select row labels to be written to GCT file'}, {'name': 'row.descriptions', 'value': 'Cufflinks transcript ID', 'label': 'Row.Descriptions', 'description': 'Select row descriptions to be written to GCT file'}, {'name': 'filter.rows', 'value': 'no', 'label': 'Filter.Rows', 'description': 'Filter rows with invalid row names (ie., \"-\") '}, {'name': 'output.prefix', 'value': '<input.file_basename>', 'label': 'Output.Prefix', 'description': 'A label that will be used to name the GCT output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task TopHat: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '[**Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] TopHat 2.0.11 is a fast splice junction mapper for RNA-Seq reads. It aligns RNA-Seq reads to mammalian-sized genomes using the ultra high-throughput short read aligner Bowtie, and then analyzes the mapping results to identify splice junctions between exons.', 'required_args': [{'name': 'GTF.file', 'value': '', 'label': 'GTF.File', 'description': 'A GTF file (v. 2.2 or higher) or GFF3 file containing a list of gene model annotations. TopHat will first extract the transcript sequences and align them to this virtual transcriptome first. Only the reads that do not fully map to the transcriptome will then be mapped on the genome. The reads that did map on the transcriptome will be converted to genomic mappings (spliced as needed) and merged with the novel mappings.'}, {'name': 'transcriptome.index', 'value': '', 'label': 'Transcriptome.Index', 'description': 'A directory containing a transcriptome index generated from a previous run of TopHat'}, {'name': 'bowtie.index', 'value': '', 'label': 'Bowtie.Index', 'description': 'A zip file or directory containing a Bowtie 2 index.'}, {'name': 'reads.pair.1', 'value': '', 'label': 'Reads.Pair.1', 'description': 'Unpaired reads file or first mate for paired reads. One or more files containing reads in FASTA or FASTQ format (bz2 and gz compressed files are supported).'}, {'name': 'reads.pair.2', 'value': '', 'label': 'Reads.Pair.2', 'description': 'Second mate for paired reads. Zero or more files in FASTA or FASTQ format (bz2 and gz compressed files are supported).'}, {'name': 'mate.std.dev', 'value': '', 'label': 'Mate.Std.Dev', 'description': 'The standard deviation for the distribution of inner distances between mate pairs.'}, {'name': 'library.type', 'value': '', 'label': 'Library.Type', 'description': 'Library type for strand specific reads.'}, {'name': 'Bowtie.preset.options', 'value': '', 'label': 'Bowtie.Preset.Options', 'description': 'A combination of pre-packaged options for Bowtie 2 based on speed and sensitivity/accuracy.'}, {'name': 'max.transcriptome.hits', 'value': '', 'label': 'Max.Transcriptome.Hits', 'description': 'The maximum number of mappings allowed for a read when it is aligned to the virtual transcriptome (provided in the GTF file parameter). Any reads found with more than this number of mappings will be discarded.'}, {'name': 'prefilter.multihits', 'value': '', 'label': 'Prefilter.Multihits', 'description': 'When mapping reads on the virtual transcriptome (provided in the GTF file parameter), some repetitive or low complexity reads that would be discarded in the context of the genome may appear to align to the transcript sequences and thus may end up reported as mapped to those genes only. This option directs TopHat to first align the reads to the whole genome, then exclude such multi-mapped reads.'}, {'name': 'raw.junctions.file', 'value': '', 'label': 'Raw.Junctions.File', 'description': 'A file containing raw junctions. Junctions are specified one per line in a tab-delimited format.'}, {'name': 'find.novel.junctions', 'value': '', 'label': 'Find.Novel.Junctions', 'description': 'If you select no, then the module will only look for junctions indicated in the GTF file supplied in the GTF file parameter. (This parameter is ignored when no GTF file is specified.)'}, {'name': 'min.intron.length', 'value': '', 'label': 'Min.Intron.Length', 'description': 'The minimum intron length. TopHat will ignore donor/acceptor pairs closer than this many bases apart.'}, {'name': 'quality.value.scale', 'value': '', 'label': 'Quality.Value.Scale', 'description': 'Whether to use the Solexa, Phred 33, or  Solexa v. 1.3 (Phred 64) quality value scale.'}, {'name': 'microexon.search', 'value': '', 'label': 'Microexon.Search', 'description': 'Attempts to find alignments incident to microexons. Works only for reads 50bp or longer.'}, {'name': 'fusion.mapping', 'value': '', 'label': 'Fusion.Mapping', 'description': 'Whether to enable fusion mapping'}, {'name': 'read.edit.dist', 'value': '', 'label': 'Read.Edit.Dist', 'description': 'Final read alignments having more than these many edit distance are discarded.'}, {'name': 'read.gap.length', 'value': '', 'label': 'Read.Gap.Length', 'description': 'Final read alignments having more than these many total length of gaps are discarded.'}, {'name': 'additional.tophat.options', 'value': '', 'label': 'Additional.Tophat.Options', 'description': 'Additional options to be passed along to the TopHat program at the command line. This parameter gives you a means to specify otherwise unavailable TopHat options and switches not supported by the module; check the TopHat manual for details. Recommended for experts only; use this at your own discretion.'}], 'default_args': [{'name': 'task_name', 'value': 'TopHat', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'mate.inner.dist', 'value': '50', 'label': 'Mate.Inner.Dist', 'description': 'The expected mean inner distance between mate pairs.'}, {'name': 'transcriptome.only', 'value': 'no', 'label': 'Transcriptome.Only', 'description': 'Whether to align the reads to the virtual transcriptome (provided in the GTF file parameter) and report only those mappings as genomic mappings.'}, {'name': 'min.anchor.length', 'value': '8', 'label': 'Min.Anchor.Length', 'description': 'The anchor length. This value must be at least 3.'}, {'name': 'max.splice.mismatches', 'value': '0', 'label': 'Max.Splice.Mismatches', 'description': 'The maximum number of mismatches that may appear in the \"anchor\" region of a spliced alignment.'}, {'name': 'max.intron.length', 'value': '500000', 'label': 'Max.Intron.Length', 'description': 'The maximum intron length. When searching for junctions ab initio, TopHat will ignore donor/acceptor pairs farther than this many bases apart, except when such a pair is supported by a split segment alignment of a long read.'}, {'name': 'max.insertion.length', 'value': '3', 'label': 'Max.Insertion.Length', 'description': 'The maximum insertion length'}, {'name': 'max.deletion.length', 'value': '3', 'label': 'Max.Deletion.Length', 'description': 'The maximum deletion length'}, {'name': 'max.multihits', 'value': '20', 'label': 'Max.Multihits', 'description': 'The maximum number of times a read can be aligned to the reference genome. If a read is aligned more than this number of times, then TopHat will choose the alignments based on their alignment scores, reporting the alignments with the best alignment scores. If there are more than this number of alignments with the same score for a read, TopHat will randomly report only this many alignments.'}, {'name': 'read.mismatches', 'value': '2', 'label': 'Read.Mismatches', 'description': 'Final read alignments having more than these many mismatches should be discarded.'}, {'name': 'coverage.search', 'value': '--no-coverage-search', 'label': 'Coverage.Search', 'description': 'Enables or disables the coverage based search for junctions. Use when coverage search is disabled by default (such as for reads 75bp or longer), for maximum sensitivity.'}, {'name': 'fusion.anchor.length', 'value': '20', 'label': 'Fusion.Anchor.Length', 'description': 'A \"supporting\" read must map to both sides of a fusion by at least these many bases. Only applies when fusion mapping is set to yes.'}, {'name': 'fusion.read.mismatches', 'value': '2', 'label': 'Fusion.Read.Mismatches', 'description': 'Reads support fusions if they map across fusion with at most these many mismatches. Only applies when fusion mapping is set to yes.'}, {'name': 'output.prefix', 'value': '<job_id>_<reads.pair.1_basename>', 'label': 'Output.Prefix', 'description': 'The prefix to use for the output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task TransposeDataset: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Transpose a Dataset - .res .gct, .odf', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'Dataset - .res .gct, .odf'}], 'default_args': [{'name': 'task_name', 'value': 'TransposeDataset', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file.format', 'value': 'gct', 'label': 'Output.File.Format', 'description': 'The output file format'}, {'name': 'output.file.name', 'value': '<input.filename_basename>.transposed', 'label': 'Output.File.Name', 'description': 'The name of the output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task Trimmomatic: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Provides a variety of options for trimming Illumina FASTQ files of adapter sequences and low-quality reads.', 'required_args': [{'name': 'input.file.1', 'value': '', 'label': 'Input.File.1', 'description': 'The input FASTQ to be trimmed.  For paired-end data, this should be the forward (\"*_1\" or \"left\") input file.'}, {'name': 'input.file.2', 'value': '', 'label': 'Input.File.2', 'description': 'The reverse (\"*_2\" or \"right\") input FASTQ of paired-end data to be trimmed.'}, {'name': 'adapter.clip.sequence.file', 'value': '', 'label': 'Adapter.Clip.Sequence.File', 'description': 'A FASTA file containing the adapter sequences, PCR sequences, etc. to be clipped. This parameter is required to enable adapter clipping. Files are provided for several Illumina pipelines but you can also provide your own; see the documentation for details. Be sure to choose a PE file for paired-end data and an SE file for single-end data.'}, {'name': 'adapter.clip.seed.mismatches', 'value': '', 'label': 'Adapter.Clip.Seed.Mismatches', 'description': 'Specifies the maximum mismatch count which will still allow a full match to be performed. A value of 2 is recommended. This parameter is required to enable adapter clipping.'}, {'name': 'adapter.clip.palindrome.clip.threshold', 'value': '', 'label': 'Adapter.Clip.Palindrome.Clip.Threshold', 'description': \"Specifies how accurate the match between the two 'adapter ligated' reads must be for PE palindrome read alignment. This is the log10 probability against getting a match by random chance; values around 30 or more are recommended. This parameter is required to enable adapter clipping.\"}, {'name': 'adapter.clip.simple.clip.threshold', 'value': '', 'label': 'Adapter.Clip.Simple.Clip.Threshold', 'description': 'Specifies how accurate the match between any adapter etc. sequence must be against a read as a log10 probability against getting a match by random chance; values between 7-15 are recommended. This parameter is required to enable adapter clipping.'}, {'name': 'adapter.clip.min.length', 'value': '', 'label': 'Adapter.Clip.Min.Length', 'description': 'In addition to the alignment score, palindrome mode can verify that a minimum length of adapter has been detected. If unspecified, this defaults to 8 bases, for historical reasons. However, since palindrome mode has a very low false positive rate, this can be safely reduced, even down to 1, to allow shorter adapter fragments to be removed.'}, {'name': 'trim.leading.quality.threshold', 'value': '', 'label': 'Trim.Leading.Quality.Threshold', 'description': 'Remove low quality bases from the beginning. As long as a base has a value below this threshold the base is removed and the next base will be investigated.'}, {'name': 'trim.trailing.quality.threshold', 'value': '', 'label': 'Trim.Trailing.Quality.Threshold', 'description': 'Remove low quality bases from the end. As long as a base has a value below this threshold the base is removed and the next trailing base will be investigated.'}, {'name': 'max.info.target.length', 'value': '', 'label': 'Max.Info.Target.Length', 'description': 'Performs a Max Info adaptive quality trim, balancing the benefits of retaining longer reads against the costs of retaining bases with errors. This parameter specifies the read length which is likely to allow the location of the read within the target sequence to be determined.'}, {'name': 'max.info.strictness', 'value': '', 'label': 'Max.Info.Strictness', 'description': 'This value, which should be set between 0 and 1, specifies the balance between preserving as much read length as possible vs. removal of incorrect bases. A low value of this parameter (<0.2) favors longer reads, while a high value (>0.8) favors read correctness.  Both max.info.target.length and max.info.strictness are required for the Max Info quality trim.'}, {'name': 'sliding.window.size', 'value': '', 'label': 'Sliding.Window.Size', 'description': 'Perform a sliding window trimming, cutting once the average quality within the window falls below a threshold. By considering multiple bases, a single poor quality base will not cause the removal of high quality data later in the read.  This parameter specifies the number of bases to average across.'}, {'name': 'sliding.window.quality.threshold', 'value': '', 'label': 'Sliding.Window.Quality.Threshold', 'description': 'Specifies the average quality required for the sliding window trimming. Both sliding.window.size and sliding.window.quality.threshold are required to enable the sliding window trimming.'}, {'name': 'min.read.length', 'value': '', 'label': 'Min.Read.Length', 'description': 'Remove reads that fall below the specified minimal length.'}, {'name': 'extra.steps', 'value': '', 'label': 'Extra.Steps', 'description': 'Extra steps to be performed after any other processing. These must be specified in exactly the format described in the Trimmomatic manual; see the documentation for details.  This is recommended for advanced users only.'}, {'name': 'convert.phred.scores', 'value': '', 'label': 'Convert.Phred.Scores', 'description': 'Convert phred scores into a particular encoding.  Leave this blank for no conversion.'}], 'default_args': [{'name': 'task_name', 'value': 'Trimmomatic', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.filename.base', 'value': '<input.file.1_basename>', 'label': 'Output.Filename.Base', 'description': 'A base name to be used for the output files.'}, {'name': 'adapter.clip.keep.both.reads', 'value': 'true', 'label': 'Adapter.Clip.Keep.Both.Reads', 'description': 'Controls whether to keep both forward and reverse reads when trimming in palindrome mode. The reverse read is the same as the forward but in reverse complement and so carries no additional information. The default is \"yes\" (retain the reverse read) which is useful when downstream tools cannot handle a combination of paired and unpaired reads.'}, {'name': 'phred.encoding', 'value': 'phred33', 'label': 'Phred.Encoding', 'description': 'Allows you to specify the phred quality encoding.  The default is phred33, which matches modern Illumina pipelines.'}, {'name': 'create.trimlog', 'value': 'false', 'label': 'Create.Trimlog', 'description': 'Create a log of the trimming process.  This gives details on what operations were performed, etc. but can be quite lengthy.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task Two.Component.Normalization: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Normalization for unimodal and bimodal samples using mixture models', 'required_args': [{'name': 'gct.file', 'value': '', 'label': 'Gct.File', 'description': 'Input data file (gct format)'}, {'name': 'cls.file', 'value': '', 'label': 'Cls.File', 'description': 'File indicating type of samples (unimodal or bimodal) in gct.file'}], 'default_args': [{'name': 'task_name', 'value': 'Two.Component.Normalization', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.prefix', 'value': '<gct.file_basename>', 'label': 'Output.Prefix', 'description': 'Filename prefix for results files'}, {'name': 'data.type', 'value': 'proteome', 'label': 'Data.Type', 'description': 'Indicate type of input data (proteome, phosphoprotome, etc.) for use in plots'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task UniquifyLabels: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Makes row and column labels unique', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'input filename - .res, .gct, .odf type=Dataset'}], 'default_args': [{'name': 'task_name', 'value': 'UniquifyLabels', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<input.filename_basename>.unique', 'label': 'Output.File', 'description': 'output file'}, {'name': 'make.row.labels.unique', 'value': 'yes', 'label': 'Make.Row.Labels.Unique', 'description': 'whether to make the row labels unique'}, {'name': 'make.column.labels.unique', 'value': 'yes', 'label': 'Make.Column.Labels.Unique', 'description': 'whether to make the column labels unique'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task untar: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'untars .tgz files', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': ''}], 'default_args': [{'name': 'task_name', 'value': 'untar', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task VennDiagram: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Displays a Venn diagram', 'required_args': [{'name': 'feature.list.one', 'value': '', 'label': 'Feature.List.One', 'description': 'Feature list one - .grp, .txt'}, {'name': 'feature.list.two', 'value': '', 'label': 'Feature.List.Two', 'description': 'Feature list two - .grp, .txt'}, {'name': 'feature.list.three', 'value': '', 'label': 'Feature.List.Three', 'description': 'Feature list three - .grp, .txt'}], 'default_args': [{'name': 'task_name', 'value': 'VennDiagram', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task WeightedVoting: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Weighted Voting classification', 'required_args': [{'name': 'train.filename', 'value': '', 'label': 'Train.Filename', 'description': 'training data file name - .gct, .res, .odf type = Dataset, ignored if a saved model (saved.model.filename) is used'}, {'name': 'train.class.filename', 'value': '', 'label': 'Train.Class.Filename', 'description': 'class file for training data - .cls, ignored if a saved model (saved.model.filename) is used'}, {'name': 'saved.model.filename', 'value': '', 'label': 'Saved.Model.Filename', 'description': 'input Weighted Voting model file - .odf type = Weighted Voting Prediction Model'}, {'name': 'test.filename', 'value': '', 'label': 'Test.Filename', 'description': 'test data file name - .gct, .res, .odf type = Dataset'}, {'name': 'test.class.filename', 'value': '', 'label': 'Test.Class.Filename', 'description': 'class file for test data - .cls'}, {'name': 'min.std', 'value': '', 'label': 'Min.Std', 'description': 'minimum standard deviation if feature selection statistic includes min std option'}, {'name': 'feature.list.filename', 'value': '', 'label': 'Feature.List.Filename', 'description': 'features to use for prediction'}], 'default_args': [{'name': 'task_name', 'value': 'WeightedVoting', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'model.file', 'value': '<train.filename_basename>.model.odf', 'label': 'Model.File', 'description': 'name of output Weighted Voting model file  - .odf type = Weighted Voting Prediction Model'}, {'name': 'num.features', 'value': '10', 'label': 'Num.Features', 'description': 'number of selected features if feature list filename is not specified'}, {'name': 'feature.selection.statistic', 'value': '0', 'label': 'Feature.Selection.Statistic', 'description': 'statistic to use to perform feature selection'}, {'name': 'pred.results.file', 'value': '<test.filename_basename>.pred.odf', 'label': 'Pred.Results.File', 'description': 'name of prediction results output file - .odf type = Prediction Results'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task WeightedVotingXValidation: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Weighted Voting classification with Leave-One-Out Cross-Validation', 'required_args': [{'name': 'data.filename', 'value': '', 'label': 'Data.Filename', 'description': 'Dataset input data file - res, gct, odf'}, {'name': 'class.filename', 'value': '', 'label': 'Class.Filename', 'description': 'Class input file - cls'}, {'name': 'min.std', 'value': '', 'label': 'Min.Std', 'description': 'minimum standard deviation if feature selection statistic includes min std option'}], 'default_args': [{'name': 'task_name', 'value': 'WeightedVotingXValidation', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'num.features', 'value': '10', 'label': 'Num.Features', 'description': 'number of selected features'}, {'name': 'feature.selection.statistic', 'value': '0', 'label': 'Feature.Selection.Statistic', 'description': 'statistic to use to perform feature selection'}, {'name': 'pred.results.file', 'value': '<data.filename_basename>.pred.odf', 'label': 'Pred.Results.File', 'description': 'Prediction results output file name - odf'}, {'name': 'feature.summary.file', 'value': '<data.filename_basename>.feat.odf', 'label': 'Feature.Summary.File', 'description': 'Feature summary output file name - odf'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task XChromosomeCorrect: {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': \"Corrects X Chromosome SNP's for male samples.\", 'required_args': [{'name': 'snp.filename', 'value': '', 'label': 'Snp.Filename', 'description': 'Summarized probe intensity data in .snp format'}, {'name': 'sample.info.filename', 'value': '', 'label': 'Sample.Info.Filename', 'description': 'A Sample information file that contains gender information'}], 'default_args': [{'name': 'task_name', 'value': 'XChromosomeCorrect', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<snp.filename_basename>.xcorrected.snp', 'label': 'Output.File', 'description': 'Output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}.\n",
      "(Getting tasks ...)\n",
      "Setting/updating task {'Load JSON': {'library_path': '', 'library_name': 'simpli.default_tasks', 'function_name': 'link_json', 'description': 'No description.', 'required_args': [{'name': 'filepath', 'value': '', 'label': 'Simpli JSON to be Loaded', 'description': 'No description.'}], 'default_args': [], 'optional_args': [], 'returns': []}, 'Reset JSONs': {'library_path': '', 'library_name': 'simpli.default_tasks', 'function_name': 'reset_jsons', 'description': 'No description.', 'required_args': [], 'default_args': [], 'optional_args': [], 'returns': []}, 'Assign': {'library_path': '', 'library_name': 'simpli.default_tasks', 'function_name': 'just_return', 'description': 'No description.', 'required_args': [{'name': 'value', 'value': '', 'label': 'Value to Assign', 'description': 'No description.'}], 'default_args': [], 'optional_args': [], 'returns': [{'label': 'Variable Name to Assign to', 'description': 'No description.'}]}, 'Slice DataFrame by Row': {'library_path': '', 'library_name': 'simpli.default_tasks', 'function_name': 'slice_dataframe', 'description': 'Return a subset of rows of a given DataFrame object.', 'required_args': [{'name': 'dataframe', 'value': '', 'label': 'DataFrame', 'description': 'Target dataframe object.'}, {'name': 'indices', 'value': '', 'label': 'Row Index(ices)', 'description': 'An integer or list of integers separated by commas of the rows to slice. Note that indices are 0 based.'}], 'default_args': [{'name': 'ax', 'value': '0', 'label': 'Ax', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Sliced DataFrame', 'description': 'A slice of the DataFrame including the specified rows.'}]}, 'Slice DataFrame by Column': {'library_path': '', 'library_name': 'simpli.default_tasks', 'function_name': 'slice_dataframe', 'description': 'No description.', 'required_args': [{'name': 'dataframe', 'value': '', 'label': 'DataFrame', 'description': 'No description.'}, {'name': 'indices', 'value': '', 'label': 'Column Index(ices)', 'description': 'No description.'}], 'default_args': [{'name': 'ax', 'value': '1', 'label': 'Ax', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Sliced DataFrame', 'description': 'No description.'}]}, 'Set Notebook Theme': {'library_path': '', 'library_name': 'simpli.default_tasks', 'function_name': 'set_notebook_theme', 'description': 'No description.', 'required_args': [{'name': 'filepath', 'value': '', 'label': 'CSS File', 'description': 'No description.'}], 'default_args': [], 'optional_args': [], 'returns': []}, 'Center Align Outputs': {'library_path': '', 'library_name': 'simpli.default_tasks', 'function_name': 'center_align_output_cells', 'description': 'No description.', 'required_args': [], 'default_args': [], 'optional_args': [], 'returns': []}, 'Display Start Banner and Logos': {'library_path': '', 'library_name': 'simpli.default_tasks', 'function_name': 'display_start_banner_and_logos', 'description': 'No description.', 'required_args': [], 'default_args': [], 'optional_args': [], 'returns': []}, 'Display End Banner': {'library_path': '', 'library_name': 'simpli.default_tasks', 'function_name': 'display_end_banner', 'description': 'No description.', 'required_args': [], 'default_args': [], 'optional_args': [], 'returns': []}, 'Link YouTube Video': {'library_path': '', 'library_name': 'simpli.default_tasks', 'function_name': 'youtube', 'description': 'No description.', 'required_args': [{'name': 'url', 'value': '', 'label': 'YouTube Video URL', 'description': 'No description.'}], 'default_args': [], 'optional_args': [], 'returns': []}, 'Toggle Input Cells': {'library_path': '', 'library_name': 'simpli.default_tasks', 'function_name': 'toggle_input_cells', 'description': 'No description.', 'required_args': [], 'default_args': [], 'optional_args': [], 'returns': []}, 'Execute HTML': {'library_path': '', 'library_name': 'simpli.default_tasks', 'function_name': 'display_raw_html', 'description': 'No description.', 'required_args': [], 'default_args': [], 'optional_args': [], 'returns': []}} to be {'Login GenePattern (Broad Institute)': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'login', 'description': 'Login into the GenePattern server at the Broad Institure.', 'required_args': [{'name': 'user_name', 'value': '', 'label': 'User name', 'description': 'GenePattern user name.'}, {'name': 'passcode', 'value': '', 'label': 'Passcode', 'description': 'GenePattern passcode.'}], 'default_args': [{'name': 'server_name', 'value': 'https://genepattern.broadinstitute.org/gp', 'label': 'Server Name', 'description': 'No description.'}], 'optional_args': [], 'returns': []}, 'ABSOLUTE': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '[**Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] Extracts absolute copy numbers per cancer cell from a mixed DNA population. Use this module for the per-sample processing step in the workflow (usually after HAPSEG).', 'required_args': [{'name': 'seg.dat.file', 'value': '', 'label': 'Seg.Dat.File', 'description': 'A HAPSEG output file'}, {'name': 'primary.disease', 'value': '', 'label': 'Primary.Disease', 'description': 'Primary disease of the sample'}, {'name': 'sample.name', 'value': '', 'label': 'Sample.Name', 'description': 'The name of the sample, included in output plots'}, {'name': 'maf.file', 'value': '', 'label': 'Maf.File', 'description': 'If available, a minor allele frequency file. This specifies the data for somatic point mutations to be used by ABSOLUTE.'}, {'name': 'min.mut.af', 'value': '', 'label': 'Min.Mut.Af', 'description': 'If specified, a minimum mutation allelic fraction. Mutations with lower allelic fractions will be filtered out before analysis. Note that if maf.fn is specified, min.mut.af must also be specified.'}], 'default_args': [{'name': 'task_name', 'value': 'ABSOLUTE', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file.name.base', 'value': '<sample.name>', 'label': 'Output.File.Name.Base', 'description': 'If specified, provides a base filename for all output files. The default value is the sample.name.'}, {'name': 'sigma.p', 'value': '0', 'label': 'Sigma.P', 'description': 'Provisional value of excess sample level variance used for mode search'}, {'name': 'max.sigma.h', 'value': '0.015', 'label': 'Max.Sigma.H', 'description': 'Maximum value of excess sample level variance (Eq. 6)'}, {'name': 'min.ploidy', 'value': '0.95', 'label': 'Min.Ploidy', 'description': 'Minimum ploidy value to consider. Solutions implying lower ploidy values will be discarded'}, {'name': 'max.ploidy', 'value': '10', 'label': 'Max.Ploidy', 'description': 'Maximum ploidy value to consider. Solutions implying greater ploidy values will be discarded'}, {'name': 'platform', 'value': 'SNP_6.0', 'label': 'Platform', 'description': \"The chip type used, supported values are currently 'SNP_250K_STY', 'SNP_6.0' and 'Illumina_WES'\"}, {'name': 'max.as.seg.count', 'value': '1500', 'label': 'Max.As.Seg.Count', 'description': \"Maximum number of allelic segments. Samples with a higher segment count will be flagged as 'failed'\"}, {'name': 'max.neg.genome', 'value': '0.005', 'label': 'Max.Neg.Genome', 'description': 'Maximum genome fraction that may be modeled as non-clonal with copy-ratio below that of clonal homozygous deletion. Solutions implying greater values will be discarded.'}, {'name': 'max.non.clonal', 'value': '0.05', 'label': 'Max.Non.Clonal', 'description': 'Maximum genome fraction that may be modeled as non-clonal (subclonal SCNA). Solutions implying greater values will be discarded.'}, {'name': 'copy.number.type', 'value': 'allelic', 'label': 'Copy.Number.Type', 'description': \"The copy number type to assess, can be one of 'allelic' or 'total'\"}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ABSOLUTE.review': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '[**Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] Extracts the absolute copy number per cancer cell from a mixed DNA population.  Use this module for the Review and Extract step in the workflow (after ABSOLUTE).', 'required_args': [{'name': 'reviewed.pp.calls.file', 'value': '', 'label': 'Reviewed.Pp.Calls.File', 'description': 'The calls file from a previous run of the ABSOLUTE.summarize module that has been reviewed and annotated by an analyst.'}, {'name': 'analyst.id', 'value': '', 'label': 'Analyst.Id', 'description': 'The user id of the analyst who called the solutions.  This is used for display/reporting purposes only.'}, {'name': 'modes.file', 'value': '', 'label': 'Modes.File', 'description': 'The obj_summary.PP-modes.data.RData file produced by the previous run of the ABSOLUTE.summarize module.'}, {'name': 'collection.name', 'value': '', 'label': 'Collection.Name', 'description': 'A descriptive name for this collection of samples.  Use the same name as in the previous run of the ABSOLUTE.summarize module.  This is used for display/reporting purposes only.'}], 'default_args': [{'name': 'task_name', 'value': 'ABSOLUTE.review', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'copy.number.type', 'value': 'allelic', 'label': 'Copy.Number.Type', 'description': \"The copy number type to assess, can be one of 'allelic' or 'total'\"}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ABSOLUTE.summarize': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '[**Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] Summarizes the results from multiple ABSOLUTE runs so that an analyst can manually review the solutions.', 'required_args': [{'name': 'collection.name', 'value': '', 'label': 'Collection.Name', 'description': 'A descriptive name for this collection of samples.'}, {'name': 'absolute.files', 'value': '', 'label': 'Absolute.Files', 'description': 'A group of ABSOLUTE RData output files from individual sample runs to be summarized together.'}], 'default_args': [{'name': 'task_name', 'value': 'ABSOLUTE.summarize', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'copy.number.type', 'value': 'allelic', 'label': 'Copy.Number.Type', 'description': \"The copy number type to assess, can be one of 'allelic' or 'total'.\"}, {'name': 'plot.modes', 'value': 'TRUE', 'label': 'Plot.Modes', 'description': 'Set this to FALSE to disable plotting of the purity/ploidy modes.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Acgh2Tab': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Converts acgh files to a tab-delimited format usable by Genomica ', 'required_args': [{'name': 'acgh.input.file', 'value': '', 'label': 'Acgh.Input.File', 'description': 'Data in aCHG format'}, {'name': 'presentlist.file', 'value': '', 'label': 'Presentlist.File', 'description': 'List of samples with a certain gene signature (from Genomica).'}, {'name': 'absentlist.file', 'value': '', 'label': 'Absentlist.File', 'description': 'List of samples without a certain gene signature (from Genomica).'}], 'default_args': [{'name': 'task_name', 'value': 'Acgh2Tab', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'genelocs.file', 'value': '<libdir>hg18_refseq_genes.txt', 'label': 'Genelocs.File', 'description': 'File giving gene locations (from GenomeBrowser).'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'AddFCSEventIndex': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Adds indexes to events in a Flow Cytometry Standard (FCS) data file.', 'required_args': [{'name': 'Input.data.file', 'value': '', 'label': 'Input.Data.File', 'description': 'An FCS data file to which you want to add indexes.'}, {'name': 'Index.parameter.long.name', 'value': '', 'label': 'Index.Parameter.Long.Name', 'description': 'The long name ($PnS keyword) of the parameter to get the index; the default is blank.'}], 'default_args': [{'name': 'task_name', 'value': 'AddFCSEventIndex', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Start.index', 'value': '1', 'label': 'Start.Index', 'description': 'The index number for the first event in the data set; the default is 1. The index will be incremented by 1 for each of the events in the data set. '}, {'name': 'Index.parameter.short.name', 'value': 'Index', 'label': 'Index.Parameter.Short.Name', 'description': 'The short name ($PnN keyword) of the parameter to get the index; the default is \"Index.\" '}, {'name': 'Output.fcs.file.name', 'value': '<Input.data.file_basename>', 'label': 'Output.Fcs.File.Name', 'description': 'The output file name.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'AddFCSParameter': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Add parameters and their values to a FCS data file', 'required_args': [{'name': 'Input.fcs.data.file', 'value': '', 'label': 'Input.Fcs.Data.File', 'description': 'A FCS data file to add parameters to.'}, {'name': 'Input.csv.file', 'value': '', 'label': 'Input.Csv.File', 'description': 'A CSV file with parameters and values to add.'}], 'default_args': [{'name': 'task_name', 'value': 'AddFCSParameter', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Output.file.name', 'value': '<Input.fcs.data.file_basename>_with_extra_params.fcs', 'label': 'Output.File.Name', 'description': 'The output FCS file name.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'AddNoiseToFCS': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Add noise to specified parameters in an FCS data file.', 'required_args': [{'name': 'Input.FCS.data.file', 'value': '', 'label': 'Input.FCS.Data.File', 'description': 'An FCS data file to add noise to.'}, {'name': 'Parameters.to.add.noise.to', 'value': '', 'label': 'Parameters.To.Add.Noise.To', 'description': 'List of parameters to which the module will add noise. The list may be provided as a list of indexes (e.g., \"1-3,5,7\") or as a list of parameter names (i.e., values of the $PnN keyword, such as \"FSC-H,FL1-H,FL2-A\"). By default, the noise will be added to all parameters but the Time parameter.'}], 'default_args': [{'name': 'task_name', 'value': 'AddNoiseToFCS', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Output.FCS.file.name', 'value': '<Input.FCS.data.file_basename>.with_noise.fcs', 'label': 'Output.FCS.File.Name', 'description': 'The output FCS file name.'}, {'name': 'Noise.level', 'value': '0.5', 'label': 'Noise.Level', 'description': 'Noise level in percent. This is the maximum noise that will be added to an event based on the current event parameter value or on the maximum range of the appropriate parameter. The latter option will be effective if the Uniform noise option is selected.'}, {'name': 'Noise', 'value': 'Proportional', 'label': 'Noise', 'description': 'How to add the noise: Proportional (The maximum amount of noise added to each value will be based on the value itself, e.g., low values will get lower amount of noise.) or Uniform (The maximum amount of noise added to each value is only dependent on the range of the particular parameter).'}, {'name': 'Saturated.events', 'value': 'RemoveSaturated', 'label': 'Saturated.Events', 'description': 'Whether to remove saturated events. An event is considered saturated if one or more of its parameter values are closer than 0.1 percent from the maximum range for that parameter (i.e., value of the appropriate $PnR keyword).'}, {'name': 'Random.Seed', 'value': '1234567', 'label': 'Random.Seed', 'description': 'The seed to use for the random generator. Leave this blank if want to use a randomly selected seed.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'AffySTExpressionFileCreator': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '[Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] Creates a GCT file from a set of CEL files from Affymetrix ST arrays.', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'One or more Affymetrix ST CEL files either uploaded directly, packaged into a ZIP or TAR bundle, or supplied through a directory input.  The CEL files can be in GZ format and the TAR can be in GZ, XZ, or BZ2 format.  The parameter will accept multiple inputs in any of these forms. '}, {'name': 'clm.file', 'value': '', 'label': 'Clm.File', 'description': 'A tab-delimited text file containing one scan, sample, and class per line'}], 'default_args': [{'name': 'task_name', 'value': 'AffySTExpressionFileCreator', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'normalize', 'value': 'yes', 'label': 'Normalize', 'description': 'Whether to normalize data using quantile normalization'}, {'name': 'background.correct', 'value': 'yes', 'label': 'Background.Correct', 'description': 'Whether to perform background correction'}, {'name': 'qc.plot.format', 'value': 'skip', 'label': 'Qc.Plot.Format', 'description': 'Choose a file format for the QC plots (or skip them completely)'}, {'name': 'annotate.rows', 'value': 'no', 'label': 'Annotate.Rows', 'description': 'Whether to annotate rows with the gene symbol and description'}, {'name': 'output.file.base', 'value': '<job_id>.expression.gct', 'label': 'Output.File.Base', 'description': 'The base name of the output file(s). File extensions will be added automatically.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ApplyGatingML': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Apply a Gating-ML file on an FCS data file (gate and/or transform list mode data) ', 'required_args': [{'name': 'Input.FCS.data.file', 'value': '', 'label': 'Input.FCS.Data.File', 'description': 'An FCS data file to be gated.'}, {'name': 'Input.Gating-ML.file', 'value': '', 'label': 'Input.Gating-ML.File', 'description': 'A Gating-ML file specifying the gates.'}], 'default_args': [{'name': 'task_name', 'value': 'ApplyGatingML', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Output.base.name', 'value': '<Input.FCS.data.file_basename>', 'label': 'Output.Base.Name', 'description': 'Base for output file names'}, {'name': 'Output.figures', 'value': 'none', 'label': 'Output.Figures', 'description': 'Also output figures of the gates'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ARACNE': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Runs the ARACNE algorithm for reverse engineering cellular networks', 'required_args': [{'name': 'dataset.file', 'value': '', 'label': 'Dataset.File', 'description': 'Dataset file - .res, .gct'}, {'name': 'hub.gene', 'value': '', 'label': 'Hub.Gene', 'description': 'The name of a gene whose network interactions you want to reconstruct'}, {'name': 'hub.genes.file', 'value': '', 'label': 'Hub.Genes.File', 'description': 'A file containing a list of hub genes with one gene per line - .txt'}, {'name': 'transcription.factor.file', 'value': '', 'label': 'Transcription.Factor.File', 'description': 'A file containing a list of genes with one gene per line that encode transcription factors - .txt'}, {'name': 'kernel.width', 'value': '', 'label': 'Kernel.Width', 'description': 'The kernel width (or window width) of the Gaussian Kernel Estimator'}], 'default_args': [{'name': 'task_name', 'value': 'ARACNE', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'mi.threshold', 'value': '0', 'label': 'Mi.Threshold', 'description': 'Threshold for the mutual information (MI) estimate to be considered statistically significant'}, {'name': 'p.value', 'value': '1', 'label': 'P.Value', 'description': 'Threshold for the p-value of a MI estimate to be considered statistically significant'}, {'name': 'dpi.tolerance', 'value': '1', 'label': 'Dpi.Tolerance', 'description': 'The percentage of MI estimation considered as sampling error'}, {'name': 'mean.filter', 'value': '0', 'label': 'Mean.Filter', 'description': 'Filter out non-informative genes whose mean expression value is smaller than mean filter'}, {'name': 'cv.filter', 'value': '0', 'label': 'Cv.Filter', 'description': 'Filter out non-informative genes whose coefficient of variance is smaller than cv filter'}, {'name': 'output.file', 'value': '<dataset.file_basename>.output.adj', 'label': 'Output.File', 'description': 'Output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'AreaChange': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Calculates fraction of area under the spectrum that is attributable to signal (area after noise removal / original area)', 'required_args': [{'name': 'spectrum.filename', 'value': '', 'label': 'Spectrum.Filename', 'description': 'spectrum data file - .csv'}, {'name': 'low.Da', 'value': '', 'label': 'Low.Da', 'description': 'minimum M/Z to include'}, {'name': 'high.Da', 'value': '', 'label': 'High.Da', 'description': 'maximum M/Z to include'}], 'default_args': [{'name': 'task_name', 'value': 'AreaChange', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'quality.threshold', 'value': '0.45', 'label': 'Quality.Threshold', 'description': 'discard spectra with quality < quality threshold'}, {'name': 'window.filter', 'value': '5', 'label': 'Window.Filter', 'description': 'window filter size'}, {'name': 'factor', 'value': '3', 'label': 'Factor', 'description': 'factor to multiply standard deviation by'}, {'name': 'smooth', 'value': 'median', 'label': 'Smooth', 'description': 'averaging function'}, {'name': 'output.file', 'value': '<spectrum.filename_basename>.area.odf', 'label': 'Output.File', 'description': 'output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Arff2Gct': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Convert an .arff file into a gene pattern .gct / .cls file pair', 'required_args': [{'name': 'arff.file', 'value': '', 'label': 'Arff.File', 'description': 'The dataset in ARFF format.  Multi-label class names are separated by colons (:). - .arff'}], 'default_args': [{'name': 'task_name', 'value': 'Arff2Gct', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<arff.file_basename>', 'label': 'Output.File', 'description': 'Name to give the .gct containing dataset features and .cls containing dataset class labels output files.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ATARiS': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Runs ATARiS on RNAi reagent-level data.', 'required_args': [{'name': 'gct.file', 'value': '', 'label': 'Gct.File', 'description': 'input GCT file. Unique reagent identifier in first column, gene symbol in second.'}], 'default_args': [{'name': 'task_name', 'value': 'ATARiS', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'identifier', 'value': '<gct.file_basename>', 'label': 'Identifier', 'description': 'text string to identify the current run'}, {'name': 'random.seed', 'value': '12345', 'label': 'Random.Seed', 'description': 'seed for random number generator'}, {'name': 'null.significance', 'value': '0.15', 'label': 'Null.Significance', 'description': 'gene solution significance level (1 - null percentile)'}, {'name': 'min.A.value', 'value': '0.3', 'label': 'Min.A.Value', 'description': 'minimum effect magnitude for RNAi reagent (relative to reagent with maximal effect)'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'AuDIT': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Automated Detection of Inaccurate and Imprecise Transitions in MRM Mass Spectrometry', 'required_args': [{'name': 'data.file', 'value': '', 'label': 'Data.File', 'description': 'input file with pre-processed MRM data in csv format'}], 'default_args': [{'name': 'task_name', 'value': 'AuDIT', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'skyline.export', 'value': 'TRUE', 'label': 'Skyline.Export', 'description': 'if \"yes\", data file is a Skyline csv export including following columns: Sample (usually derived from SampleName, having identical value for all its replicates), PeptideSequence, ReplicateName, FragmentIon, PrecursorCharge, ProductCharge, light Area, heavy Area. When \"no\", input must contain sample, replicate, peptide, transition.id, area, IS.area in that order.'}, {'name': 'pvalue.threshold', 'value': '0.00001', 'label': 'Pvalue.Threshold', 'description': 'p-value threshold above which transition is quantification-worthy. Must be a value between 0 and 1 inclusive.'}, {'name': 'cv.threshold', 'value': '0.2', 'label': 'Cv.Threshold', 'description': 'threshold for coefficient of variation below which transition is quantification-worthy'}, {'name': 'all.pairs', 'value': 'FALSE', 'label': 'All.Pairs', 'description': 'calculate all possible pairs of relative ratios for a transition'}, {'name': 'output.intermediate.results', 'value': 'FALSE', 'label': 'Output.Intermediate.Results', 'description': 'create files with intermediate results'}, {'name': 'output.prefix', 'value': '<data.file_basename>-results', 'label': 'Output.Prefix', 'description': 'file name prefix used for output file(s)'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'BedToGtf': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Converts BED files to GFF or GTF format', 'required_args': [{'name': 'bed.file', 'value': '', 'label': 'Bed.File', 'description': 'A bed file'}], 'default_args': [{'name': 'task_name', 'value': 'BedToGtf', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.format', 'value': 'gtf', 'label': 'Output.Format', 'description': 'Whether to output a gff file or exon only gtf file. The bed file must contain at least 12 columns in order to generate a gtf file.'}, {'name': 'output.prefix', 'value': '<bed.file_basename>', 'label': 'Output.Prefix', 'description': 'The prefix for the output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Beroukhim.Getz.2007.PNAS.Glioma.GISTIC': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '', 'required_args': [], 'default_args': [{'name': 'task_name', 'value': 'Beroukhim.Getz.2007.PNAS.Glioma.GISTIC', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'BlastTrainTest': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Sequence similarity classification using BLAST', 'required_args': [{'name': 'train.file', 'value': '', 'label': 'Train.File', 'description': 'The training file in .FASTA format. Multi-label class names are separated by colons (:).'}, {'name': 'test.file', 'value': '', 'label': 'Test.File', 'description': 'The testing file in .FASTA format. Multi-label class names are separated by colons (:).'}], 'default_args': [{'name': 'task_name', 'value': 'BlastTrainTest', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'sub.matrix', 'value': 'PAM30', 'label': 'Sub.Matrix', 'description': 'The name of the substitution matrix used by Blast.'}, {'name': 'pred.results.file', 'value': '<test.file_basename>', 'label': 'Pred.Results.File', 'description': 'The name of the output file for prediction results.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'BlastXValidation': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Sequence similarity cross validation prediction using BLAST', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'The input sequence and annotation information in .FASTA format.'}], 'default_args': [{'name': 'task_name', 'value': 'BlastXValidation', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'num.folds', 'value': '2', 'label': 'Num.Folds', 'description': 'The number of folds should be an integer greater than or equal to two.'}, {'name': 'sub.matrix', 'value': 'PAM30', 'label': 'Sub.Matrix', 'description': 'The name of the substitution matrix used by Blast.'}, {'name': 'pred.results.file', 'value': '<input.file_basename>', 'label': 'Pred.Results.File', 'description': 'The name of the output file for prediction results.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Bowtie.aligner': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Bowtie2 (v. 2.1.0) is an ultrafast and memory-efficient short read aligner.', 'required_args': [{'name': 'prebuilt.bowtie.index', 'value': '', 'label': 'Prebuilt.Bowtie.Index', 'description': 'A prebuilt Bowtie 2 index. Either a prebuilt or a custom Bowtie 2 index must be specified.'}, {'name': 'custom.bowtie.index', 'value': '', 'label': 'Custom.Bowtie.Index', 'description': 'A ZIP archive containing Bowtie 2 index files. Either a prebuilt or a custom Bowtie 2 index must be specified.'}, {'name': 'input.format', 'value': '', 'label': 'Input.Format', 'description': 'The format of the reads input files.'}, {'name': 'reads.pair.1', 'value': '', 'label': 'Reads.Pair.1', 'description': 'Unpaired reads file or first mate for paired reads. A file or zip of files containing reads in FASTA or FASTQ format (can be compressed - ie .gz).'}, {'name': 'reads.pair.2', 'value': '', 'label': 'Reads.Pair.2', 'description': 'Second mate for paired reads. A file or zip of files in FASTA or FASTQ format (can be compressed - ie .gz).'}, {'name': 'quality.value.scale', 'value': '', 'label': 'Quality.Value.Scale', 'description': 'The scale to use for quality values.'}, {'name': 'integer.quality.values', 'value': '', 'label': 'Integer.Quality.Values', 'description': 'Whether the quality values are represented as space-separated ASCII integers (i.e 40 40 30 40 ..).'}, {'name': 'max.reads.to.align', 'value': '', 'label': 'Max.Reads.To.Align', 'description': 'The maximum number of reads to align.'}, {'name': 'mate.orientations', 'value': '', 'label': 'Mate.Orientations', 'description': 'The upstream/downstream mate orientations for a valid paired-end alignment against the forward reference strand.'}], 'default_args': [{'name': 'task_name', 'value': 'Bowtie.aligner', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'alignment.mode', 'value': '--end-to-end', 'label': 'Alignment.Mode', 'description': 'The alignment mode to use.'}, {'name': 'preset.options', 'value': 'sensitive', 'label': 'Preset.Options', 'description': 'A combination of pre-packaged options based on speed and sensitivity/accuracy.'}, {'name': 'min.fragment.length', 'value': '0', 'label': 'Min.Fragment.Length', 'description': 'The minimum fragment length for valid paired-end alignments.'}, {'name': 'max.fragment.length', 'value': '500', 'label': 'Max.Fragment.Length', 'description': 'The maximum fragment length for valid paired-end alignments.'}, {'name': 'random.seed', 'value': '123456', 'label': 'Random.Seed', 'description': 'The seed to use for the pseudo-random number generator.'}, {'name': 'output.prefix', 'value': '<reads.pair.1_basename>_<reads.pair.2_basename>', 'label': 'Output.Prefix', 'description': 'The prefix to use for the output file name.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Bowtie.indexer': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Builds a Bowtie2 (v. 2.1.0) index from a set of DNA sequences', 'required_args': [{'name': 'fasta.file', 'value': '', 'label': 'Fasta.File', 'description': 'A file or zip of files containing sequences in FASTA format'}], 'default_args': [{'name': 'task_name', 'value': 'Bowtie.indexer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'random.seed', 'value': '123456', 'label': 'Random.Seed', 'description': 'The seed for the random number generator'}, {'name': 'index.name', 'value': '<fasta.file_basename>', 'label': 'Index.Name', 'description': 'The base name of the index files'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'BWA.aln': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '[**Beta Release** Contact gp-help with any issues.  Check stdout.txt and stderr.txt for errors] A fast and accurate short-read alignment tool that allows for mismatches and gaps. Alignments are output in a SAM format file, which provides Phred-scale quality scores for each alignment.  Uses BWA 0.7.4 aln and samse/sampe.', 'required_args': [{'name': 'BWA.index', 'value': '', 'label': 'BWA.Index', 'description': 'A zip file or directory containing BWA index files.'}, {'name': 'reads.pair.1', 'value': '', 'label': 'Reads.Pair.1', 'description': 'Single-end or first paired-end reads file in FASTA, FASTQ, or BAM format.  For paired-end data, this should be the forward (\"*_1\" or \"left\") input file.  Note: the FASTA or FASTQ can be gzipped.'}, {'name': 'reads.pair.2', 'value': '', 'label': 'Reads.Pair.2', 'description': 'The reverse (\"*_2\" or \"right\") input file for paired-end reads in FASTA, FASTQ, or BAM format. Note: the FASTA or FASTQ can be gzipped.'}, {'name': 'bam.mapping', 'value': '', 'label': 'Bam.Mapping', 'description': 'Specifies how to map BAM input. This is only required if input file is in BAM format.'}, {'name': 'max.edit.distance', 'value': '', 'label': 'Max.Edit.Distance', 'description': 'The max edit distance. This specifies a threshold of the maximum number of deletions, insertions, and substitutions needed to transform the reference sequence into the read sequence.'}, {'name': 'max.num.gap', 'value': '', 'label': 'Max.Num.Gap', 'description': 'Maximum number of gap opens. This specifies a threshold of the maximum number of gaps that can be initiated to match a given read to the reference.'}, {'name': 'max.gap.extension', 'value': '', 'label': 'Max.Gap.Extension', 'description': 'Maximum number of gap extensions. This specifies a threshold of the maximum number of bases by which gaps in a read can be extended.'}, {'name': 'max.deletion.length', 'value': '', 'label': 'Max.Deletion.Length', 'description': \"Disallow a long deletion within this many bp of the 3' end.\"}, {'name': 'max.indel.length', 'value': '', 'label': 'Max.Indel.Length', 'description': 'Disallow an indel within this many bp of the ends'}, {'name': 'seed.length', 'value': '', 'label': 'Seed.Length', 'description': 'The set of bases determined by this option in the high-quality (left) end of the read is the seed.'}, {'name': 'max.seed.edit.distance', 'value': '', 'label': 'Max.Seed.Edit.Distance', 'description': 'Maximum edit distance in the seed; that is, the maximum number of changes required to transform the reference sequence of the seed into the read sequence of the seed.'}, {'name': 'mismatch.penalty', 'value': '', 'label': 'Mismatch.Penalty', 'description': 'Mismatch penalty'}, {'name': 'gap.open.penalty', 'value': '', 'label': 'Gap.Open.Penalty', 'description': 'Gap open penalty.  The gap open penalty is the score taken away for the initiation of the gap in sequence. To make the match more significant you can try to make the gap penalty larger.'}, {'name': 'gap.extension.penalty', 'value': '', 'label': 'Gap.Extension.Penalty', 'description': 'Gap extension penalty.  The gap extension penalty is added to the standard gap penalty for each base or residue in the gap.  To reduce long gaps, increase the extension gap penalty.'}, {'name': 'max.best.hits', 'value': '', 'label': 'Max.Best.Hits', 'description': 'Proceed with suboptimal alignments if there are no more than this many equally best hits. This option only affects paired-end mapping.'}, {'name': 'trim.reads', 'value': '', 'label': 'Trim.Reads', 'description': 'Specifies a quality threshold for read trimming'}, {'name': 'barcode.length', 'value': '', 'label': 'Barcode.Length', 'description': \"Length of barcode starting from the 5' end\"}, {'name': 'max.insert.size', 'value': '', 'label': 'Max.Insert.Size', 'description': 'Specifies the maximum insert size for a read pair to be considered to be mapped properly'}, {'name': 'max.occurrences', 'value': '', 'label': 'Max.Occurrences', 'description': 'Specifies the maximum occurrences of a read for pairing'}, {'name': 'max.alignments', 'value': '', 'label': 'Max.Alignments', 'description': 'Maximum number of alignments to output in the XA tag for reads paired properly'}, {'name': 'max.dc.alignments', 'value': '', 'label': 'Max.Dc.Alignments', 'description': 'Maximum number of alignments to output in the XA tag for disconcordant read pairs (excluding singletons)'}], 'default_args': [{'name': 'task_name', 'value': 'BWA.aln', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'iterative.search', 'value': 'no', 'label': 'Iterative.Search', 'description': 'Whether to disable iterative search'}, {'name': 'Illumina.1.3.format', 'value': 'no', 'label': 'Illumina.1.3.Format', 'description': 'The input is in the Illumina 1.3+ read format'}, {'name': 'output.prefix', 'value': '<reads.pair.1_basename>', 'label': 'Output.Prefix', 'description': 'Prefix to use for output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'BWA.bwasw': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '[**Beta Release** Contact gp-help with any issues.  Check stdout.txt and stderr.txt for errors] Aligns long sequences (>200 bp) to a sequence database using BWA 0.7.4 bwasw.', 'required_args': [{'name': 'BWA.index', 'value': '', 'label': 'BWA.Index', 'description': 'A zip file or directory containing BWA index files.'}, {'name': 'read.file', 'value': '', 'label': 'Read.File', 'description': 'Single-end or first paired-end reads file in FASTA or FASTQ format.  For paired-end data, this should be the forward (\"*_1\" or \"left\") input file.'}, {'name': 'mate.read.file', 'value': '', 'label': 'Mate.Read.File', 'description': 'The reverse (\"*_2\" or \"right\") reads file for paired-end reads in FASTA or FASTQ format.'}, {'name': 'match.score', 'value': '', 'label': 'Match.Score', 'description': 'The score of the match'}, {'name': 'mismatch.penalty', 'value': '', 'label': 'Mismatch.Penalty', 'description': 'Specifies the mismatch penalty.'}, {'name': 'gap.open.penalty', 'value': '', 'label': 'Gap.Open.Penalty', 'description': 'Gap open penalty.  The gap open penalty is the score taken away for the initiation of the gap in sequence. To make the match more significant you can try to make the gap penalty larger.'}, {'name': 'gap.extension.penalty', 'value': '', 'label': 'Gap.Extension.Penalty', 'description': 'Gap extension penalty.  The gap extension penalty is added to the standard gap penalty for each base or residue in the gap.  To reduce long gaps, increase the extension gap penalty. '}, {'name': 'band.width', 'value': '', 'label': 'Band.Width', 'description': 'band width in the banded alignment'}, {'name': 'min.score.threshold', 'value': '', 'label': 'Min.Score.Threshold', 'description': 'Minimum score threshold divided by the match score'}, {'name': 'threshold.coefficient', 'value': '', 'label': 'Threshold.Coefficient', 'description': 'Coefficient for threshold adjustment according to query length'}, {'name': 'z.best.heuristics', 'value': '', 'label': 'Z.Best.Heuristics', 'description': 'Z-best heuristics. Specifying a higher number increases accuracy at the cost of speed'}, {'name': 'max.sa.interval.size', 'value': '', 'label': 'Max.Sa.Interval.Size', 'description': 'Maximum SA interval size for initiating a seed. Specifying a higher number increases accuracy at the cost of speed.'}, {'name': 'min.num.seeds', 'value': '', 'label': 'Min.Num.Seeds', 'description': 'The minimum number of seeds contained in the best alignment from the forward-forward alignment process that allows the algorithm to skip performing the reverse alignment. BWA.bwasw tends to be faster and more accurate if the alignment is supported by more seeds.'}], 'default_args': [{'name': 'task_name', 'value': 'BWA.bwasw', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'num.threads', 'value': '1', 'label': 'Num.Threads', 'description': 'Number of threads'}, {'name': 'output.prefix', 'value': '<read.file_basename>', 'label': 'Output.Prefix', 'description': 'Prefix to use for the output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'BWA.indexer': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '[**Beta Release** Contact gp-help with any issues.  Check stdout.txt and stderr.txt for errors] Creates a BWT index from a sequence FASTA file for use with the BWA 0.7.4 aligners.', 'required_args': [{'name': 'fasta.file', 'value': '', 'label': 'Fasta.File', 'description': 'A sequence fasta file'}, {'name': 'algorithm', 'value': '', 'label': 'Algorithm', 'description': 'The algorithm to use for constructing the BWT index'}], 'default_args': [{'name': 'task_name', 'value': 'BWA.indexer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.prefix', 'value': '<fasta.file_basename>', 'label': 'Output.Prefix', 'description': 'A prefix for the output file name.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'CaArray2ImportViewer': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Imports data files from CaArray 2.4.1 and creates gct or cls files ', 'required_args': [{'name': 'experiment', 'value': '', 'label': 'Experiment', 'description': 'Title or identifier of the caArray experiment to import data from.  https://array.nci.nih.gov/caarray/home.action'}], 'default_args': [{'name': 'task_name', 'value': 'CaArray2ImportViewer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'url', 'value': 'http://array.nci.nih.gov:8080', 'label': 'Url', 'description': 'URL for caArray server.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'CART': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Classification and Regression Tree', 'required_args': [{'name': 'train.data.filename', 'value': '', 'label': 'Train.Data.Filename', 'description': 'The training data file - .gct, .res, ignored if a saved model (saved.model.filename) is used'}, {'name': 'train.cls.filename', 'value': '', 'label': 'Train.Cls.Filename', 'description': 'The training class file - .cls, ignored if a saved model (saved.model.filename) is used'}, {'name': 'saved.model.filename', 'value': '', 'label': 'Saved.Model.Filename', 'description': 'input CART model'}, {'name': 'test.data.filename', 'value': '', 'label': 'Test.Data.Filename', 'description': 'The test data file - .gct, .res'}, {'name': 'test.cls.filename', 'value': '', 'label': 'Test.Cls.Filename', 'description': 'The test class file - .cls'}], 'default_args': [{'name': 'task_name', 'value': 'CART', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'model.output.file', 'value': '<train.data.filename_basename>.cart.model', 'label': 'Model.Output.File', 'description': 'name of output CART model file  '}, {'name': 'pred.results.file', 'value': '<test.data.filename_basename>.pred.odf', 'label': 'Pred.Results.File', 'description': 'The name of the output file for prediction results'}, {'name': 'tree.output.file', 'value': '<test.data.filename_basename>.tree.pdf', 'label': 'Tree.Output.File', 'description': 'The name of the file containing a plot of the classification tree'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'CARTXValidation': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Classification and Regression Tree Cross-Validation', 'required_args': [{'name': 'data.filename', 'value': '', 'label': 'Data.Filename', 'description': 'The data file - .gct, .res'}, {'name': 'cls.filename', 'value': '', 'label': 'Cls.Filename', 'description': 'The class file - .cls '}], 'default_args': [{'name': 'task_name', 'value': 'CARTXValidation', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'prediction.results.file', 'value': '<data.filename_basename>.pred.odf', 'label': 'Prediction.Results.File', 'description': 'The name of the output file for prediction results'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'CBS': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Segments DNA copy number data into regions of estimated equal copy number using circular binary segmentation (CBS). ', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'Copy number file - .cn'}, {'name': 'sample.name', 'value': '', 'label': 'Sample.Name', 'description': 'The name of the sample to use. Leave blank to use the first sample in the input file.'}], 'default_args': [{'name': 'task_name', 'value': 'CBS', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'permutations', 'value': '10000', 'label': 'Permutations', 'description': 'Number of permutations used for p-value computation'}, {'name': 'significance.level', 'value': '0.01', 'label': 'Significance.Level', 'description': 'Significance levels for the test to accept change-points'}, {'name': 'output.prefix', 'value': '<input.file_basename>', 'label': 'Output.Prefix', 'description': 'The prefix to use for the output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ChIPSeq.CreateHeatmap': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Generates a heatmap based on the ChIP-Seq signal extracted from a BAM file, according to the chromosomal locations specified in the BED file used in ExtractSignal.  This is an experimental module built for the Broad CEGS workshop of July 29-30, 2013.', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'Input file in TAB format, generated by ChipSeq.ExtractSignal'}], 'default_args': [{'name': 'task_name', 'value': 'ChIPSeq.CreateHeatmap', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'normalization', 'value': 'z-score', 'label': 'Normalization', 'description': 'Choose a normalization method'}, {'name': 'percent.start', 'value': '0.45', 'label': 'Percent.Start', 'description': 'percent of the bed region to start clustering at'}, {'name': 'percent.end', 'value': '0.55', 'label': 'Percent.End', 'description': 'percent of the bed region to end clustering at '}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ChIPSeq.CreatePlots': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Generates a scatter plot and correlation matrix based on the ChIP-Seq signal extracted from a BAM file, according to the chromosomal locations specified in the BED file used in ExtractSignal.  This is an experimental module built for the Broad CEGS workshop of July 29-30, 2013.', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'Input file in TAB format, generated by ChipSeq.ExtractSignal'}], 'default_args': [{'name': 'task_name', 'value': 'ChIPSeq.CreatePlots', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ChIPSeq.ExtractSignal': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Segment the read from a BAM file into bins by chromosome and extract the signal values to a tab delimited file for subsequent visualization.  This is an experimental module built for the Broad CEGS workshop of July 29-30, 2013.', 'required_args': [{'name': 'bam.file', 'value': '', 'label': 'Bam.File', 'description': 'Input file in BAM format.'}, {'name': 'chromosome.sizes.file', 'value': '', 'label': 'Chromosome.Sizes.File', 'description': \"Specify a chromosome sizes file.  Chromosome names must start with 'chr'.\"}, {'name': 'bin.size', 'value': '', 'label': 'Bin.Size', 'description': 'The desired size of the bins.  Defaults to 25 if no value is supplied.'}, {'name': 'whole.cell.extract.control.file', 'value': '', 'label': 'Whole.Cell.Extract.Control.File', 'description': 'A bam file containing Whole Cell Extract control data to use in normalizing the bin values.'}, {'name': 'region.bed.file', 'value': '', 'label': 'Region.Bed.File', 'description': 'The BED file containing regions of interest.'}, {'name': 'region.bin.size', 'value': '', 'label': 'Region.Bin.Size', 'description': 'Bin the bed regions into sub-regions of this size (in bps).  Must be a multiple of the bin size such that an integer number of bin file bins greater than zero fits into a single region bin. (E.g. One could have bin files of 25 bp, extend bed regions by 100 bp, then have region bins of 50 bp for a total of four region bins.). Defaults to bin.size if no value is supplied.'}, {'name': 'extend.by', 'value': '', 'label': 'Extend.By', 'description': 'The number of basepairs to extend a region from its center.  Total extended region size will be twice this value.  This option is most useful if one wants to analyze  regions of the same length and number of bins.  To simply use the bin boundaries nearest the BED coordinates (i.e. analyze variable numbers of bins per BED line), do not specify this parameter.'}], 'default_args': [{'name': 'task_name', 'value': 'ChIPSeq.ExtractSignal', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'assembly.name', 'value': 'hg19', 'label': 'Assembly.Name', 'description': 'Specify an assembly (i.e. hg19, hg18, etc.).'}, {'name': 'normalize', 'value': 'yes', 'label': 'Normalize', 'description': 'Normalize each bin value by scaling it to 20000000 reads.'}, {'name': 'aggregation.function', 'value': 'avg', 'label': 'Aggregation.Function', 'description': 'Function to run on the bin values over the regions of interest.'}, {'name': 'output.filename', 'value': '<bam.file_basename>.<region.bed.file_basename>.tab', 'label': 'Output.Filename', 'description': 'The name of the output file.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ClassNeighbors': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Select genes that most closely resemble a profile', 'required_args': [{'name': 'data.filename', 'value': '', 'label': 'Data.Filename', 'description': 'Dataset file - .res, .gct, .odf'}, {'name': 'class.filename', 'value': '', 'label': 'Class.Filename', 'description': 'class file - .cls, .odf'}], 'default_args': [{'name': 'task_name', 'value': 'ClassNeighbors', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'marker.gene.list.file', 'value': '<data.filename_basename>.genelist', 'label': 'Marker.Gene.List.File', 'description': 'output filename for marker gene list results - .odf file'}, {'name': 'marker.data.set.file', 'value': '<data.filename_basename>.neighbors', 'label': 'Marker.Data.Set.File', 'description': 'output filename for marker data - .gct file'}, {'name': 'num.neighbors', 'value': '50', 'label': 'Num.Neighbors', 'description': 'number of neighbors to find'}, {'name': 'num.permutations', 'value': '100', 'label': 'Num.Permutations', 'description': 'number of permutations in permutation test'}, {'name': 'user.pval', 'value': '0.5', 'label': 'User.Pval', 'description': 'user-set significance value for permutation test'}, {'name': 'mean.or.median', 'value': 'mean', 'label': 'Mean.Or.Median', 'description': 'use mean or median for feature selection'}, {'name': 'ttest.or.snr', 'value': 't-test', 'label': 'Ttest.Or.Snr', 'description': 'use signal-to-noise or t-test to select neighbors'}, {'name': 'filter.data', 'value': 'no', 'label': 'Filter.Data', 'description': 'if no, values below will be ignored'}, {'name': 'min.threshold', 'value': '10', 'label': 'Min.Threshold', 'description': 'minimum threshold for data'}, {'name': 'max.threshold', 'value': '16000', 'label': 'Max.Threshold', 'description': 'maximum threshold for data'}, {'name': 'min.fold.diff', 'value': '5', 'label': 'Min.Fold.Diff', 'description': 'minimum fold difference for filtering genes'}, {'name': 'min.abs.diff', 'value': '50', 'label': 'Min.Abs.Diff', 'description': 'minimum absolute difference for filtering genes'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ClsFileCreator': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'A tool to create a class label (CLS) file.', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'A dataset - .gct'}], 'default_args': [{'name': 'task_name', 'value': 'ClsFileCreator', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'CNToXCN': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '', 'required_args': [{'name': 'cn.file', 'value': '', 'label': 'Cn.File', 'description': 'Input CN file (a tab-delimited file format that contains SNP copy numbers)'}], 'default_args': [{'name': 'task_name', 'value': 'CNToXCN', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.prefix', 'value': '<cn.file_basename>', 'label': 'Output.Prefix', 'description': 'Prefix for the output XCN file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'CoClustering': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Co-clustering of RNA and global proteome data', 'required_args': [], 'default_args': [{'name': 'task_name', 'value': 'CoClustering', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'min.correlation', 'value': '0.4', 'label': 'Min.Correlation', 'description': 'Minimum correlation between RNA and proteome for gene to be included in clustering'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'CollapseDataset': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Collapses all probe set values for a gene into a single vector of values', 'required_args': [{'name': 'dataset.file', 'value': '', 'label': 'Dataset.File', 'description': 'Dataset file - .gct, .res'}, {'name': 'chip.platform', 'value': '', 'label': 'Chip.Platform', 'description': 'The chip platform'}], 'default_args': [{'name': 'task_name', 'value': 'CollapseDataset', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'collapse.mode', 'value': 'Maximum', 'label': 'Collapse.Mode', 'description': 'Collapse mode for probe sets => 1 gene'}, {'name': 'output.file.name', 'value': '<dataset.file_basename>.collapsed', 'label': 'Output.File.Name', 'description': 'The output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ComBat': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Batch Correction', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'input file - .res, .gct'}, {'name': 'sample.info.file', 'value': '', 'label': 'Sample.Info.File', 'description': 'A file containing information about the samples. The sample info file must contain the three columns: Array, Sample, and Batch. Any additional columns are used as covariates in the analysis.(.txt file)'}, {'name': 'absent.calls.filter', 'value': '', 'label': 'Absent.Calls.Filter', 'description': 'The filter to apply to genes with absent calls (a value between 0 and 1). Filters genes with absent calls in 1-(absent calls filter) of the samples. Ignored if input file is not a .res file.\\t'}], 'default_args': [{'name': 'task_name', 'value': 'ComBat', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'covariate.columns', 'value': 'all', 'label': 'Covariate.Columns', 'description': \"The covariate columns to use in analysis. This is either set to all, none, or a list specifying one or more covariate columns from sample info file to use in the analysis (i.e, 4, 5, 7, ...). If this is set to 'all' then all the covariate columns are used. If this is set to 'none' then none of the covariate columns are used.\"}, {'name': 'create.prior.plots', 'value': 'T', 'label': 'Create.Prior.Plots', 'description': 'Whether to generate prior probability distribution plots. Ignored if non-parametric method is used.'}, {'name': 'prior.method', 'value': 'T', 'label': 'Prior.Method', 'description': 'Empirical Bayes priors distribution estimation method to use. Either a parametric or non-parametric method can be applied. If unsure of which method to use try the parametric method first since it is faster.'}, {'name': 'output.file', 'value': '<input.file_basename>.combat.<input.file_extension>', 'label': 'Output.File', 'description': 'Name of batch adjusted output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'CombineOdf': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Combine two prediction results files into a single, weighted, multi-label prediction results file', 'required_args': [{'name': 'pred1.file', 'value': '', 'label': 'Pred1.File', 'description': 'The first prediction file in .odf format.'}, {'name': 'pred2.file', 'value': '', 'label': 'Pred2.File', 'description': 'The second prediction file in .odf format.'}], 'default_args': [{'name': 'task_name', 'value': 'CombineOdf', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'pred1.file.weight', 'value': '0.5', 'label': 'Pred1.File.Weight', 'description': 'Weight assigned to classifier that produced first prediction file.'}, {'name': 'pred.file', 'value': '<pred1.file_basename>_<pred2.file_basename>', 'label': 'Pred.File', 'description': 'The name of the prediction file in modified .odf format.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ComparativeMarkerSelection': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Identify differentially expressed genes that can discriminate between distinct classes of samples.', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'The input file - .res, .gct. Note that if your data is log transformed, you will need to set the \"log transformed data\" parameter below to \"yes\".'}, {'name': 'cls.file', 'value': '', 'label': 'Cls.File', 'description': 'The class file - .cls'}, {'name': 'confounding.variable.cls.file', 'value': '', 'label': 'Confounding.Variable.Cls.File', 'description': 'The class file containing the confounding variable.'}, {'name': 'min.std', 'value': '', 'label': 'Min.Std', 'description': 'The minimum standard deviation if test statistic includes min std option'}], 'default_args': [{'name': 'task_name', 'value': 'ComparativeMarkerSelection', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'test.direction', 'value': '2', 'label': 'Test.Direction', 'description': 'The test to perform'}, {'name': 'test.statistic', 'value': '0', 'label': 'Test.Statistic', 'description': 'The statistic to use'}, {'name': 'number.of.permutations', 'value': '10000', 'label': 'Number.Of.Permutations', 'description': 'The number of permutations to perform (use 0 to calculate asymptotic p-values using the standard independent two-sample t-test)'}, {'name': 'log.transformed.data', 'value': 'false', 'label': 'Log.Transformed.Data', 'description': 'Whether the input data has been log transformed'}, {'name': 'complete', 'value': 'false', 'label': 'Complete', 'description': 'Whether to perform all possible permutations'}, {'name': 'balanced', 'value': 'false', 'label': 'Balanced', 'description': 'Whether to perform balanced permutations'}, {'name': 'random.seed', 'value': '779948241', 'label': 'Random.Seed', 'description': 'The seed of the random number generator used to produce permutations'}, {'name': 'smooth.p.values', 'value': 'true', 'label': 'Smooth.P.Values', 'description': 'Whether to smooth p-values'}, {'name': 'phenotype.test', 'value': 'one versus all', 'label': 'Phenotype.Test', 'description': 'Tests to perform when cls file has more than two classes (Note: not fully corrected for multiple hypothesis testing)'}, {'name': 'output.filename', 'value': '<input.file_basename>.comp.marker.odf', 'label': 'Output.Filename', 'description': 'The name of the output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ComparativeMarkerSelectionViewer': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Views the results from ComparativeMarkerSelection', 'required_args': [{'name': 'comparative.marker.selection.filename', 'value': '', 'label': 'Comparative.Marker.Selection.Filename', 'description': 'The output from ComparativeMarkerSelection'}, {'name': 'dataset.filename', 'value': '', 'label': 'Dataset.Filename', 'description': 'The dataset file used as input to ComparativeMarkerSelection'}], 'default_args': [{'name': 'task_name', 'value': 'ComparativeMarkerSelectionViewer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'CompareSpectra': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Compares two spectra to determine similarity', 'required_args': [{'name': 'spectrum1.filename', 'value': '', 'label': 'Spectrum1.Filename', 'description': 'Spectrum file -.csv'}, {'name': 'spectrum2.filename', 'value': '', 'label': 'Spectrum2.Filename', 'description': 'Spectrum file - .csv'}, {'name': 'low.Da', 'value': '', 'label': 'Low.Da', 'description': 'Minimum M/Z to include'}, {'name': 'high.Da', 'value': '', 'label': 'High.Da', 'description': 'Maximum M/Z to include'}], 'default_args': [{'name': 'task_name', 'value': 'CompareSpectra', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<spectrum1.filename_basename>_<spectrum2.filename_basename>_compare.odf', 'label': 'Output.File', 'description': 'Output file name - .odf'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'CompensateFCS': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Compensates an FCS data file', 'required_args': [{'name': 'FCS.data.file', 'value': '', 'label': 'FCS.Data.File', 'description': 'A FCS data file to be compensated'}, {'name': 'Spillover.matrix.file', 'value': '', 'label': 'Spillover.Matrix.File', 'description': 'A text file with spillover matrix. Use this parameter if there is no spillover matrix in the input FCS file or to use a spillover matrix different from the matrix in the input FCS file. The value from the FCS data file will be used if this parameter is not specified.'}], 'default_args': [{'name': 'task_name', 'value': 'CompensateFCS', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Output.file.name', 'value': '<FCS.data.file_basename>.compensated', 'label': 'Output.File.Name', 'description': 'The name of the output file. The extension will match the file format specified in Output file format.'}, {'name': 'Output.file.format', 'value': 'FCS', 'label': 'Output.File.Format', 'description': 'The file format in which to save the compensated data.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ConcatenateFilelist': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Concatenate all of the items in the input filelist into a single large file.', 'required_args': [{'name': 'filelist', 'value': '', 'label': 'Filelist', 'description': 'a text file containing a list of file paths'}, {'name': 'num.header.lines', 'value': '', 'label': 'Num.Header.Lines', 'description': 'specify number of header lines so that they are output only once'}, {'name': 'output.filename', 'value': '', 'label': 'Output.Filename', 'description': 'The name of the output file'}], 'default_args': [{'name': 'task_name', 'value': 'ConcatenateFilelist', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ConsensusClustering': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Resampling-based clustering method ', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'The data to be clustered (.gct, .res, .odf)'}, {'name': 'cluster.by', 'value': '', 'label': 'Cluster.By', 'description': 'Whether to cluster by rows/genes or columns/experiments'}, {'name': 'create.heat.map', 'value': '', 'label': 'Create.Heat.Map', 'description': 'Whether to create heatmaps (one for each cluster number)'}], 'default_args': [{'name': 'task_name', 'value': 'ConsensusClustering', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'kmax', 'value': '5', 'label': 'Kmax', 'description': 'Try K=2,3,...,kmax clusters (must be > 1)'}, {'name': 'resampling.iterations', 'value': '20', 'label': 'Resampling.Iterations', 'description': 'Number of resampling iterations'}, {'name': 'seed.value', 'value': '12345', 'label': 'Seed.Value', 'description': 'Random number generator seed'}, {'name': 'clustering.algorithm', 'value': 'HIERARCHICAL', 'label': 'Clustering.Algorithm', 'description': 'Type of clustering algorithm'}, {'name': 'distance.measure', 'value': 'EUCLIDEAN', 'label': 'Distance.Measure', 'description': 'Distance measure'}, {'name': 'resample', 'value': 'subsample', 'label': 'Resample', 'description': \"resampling scheme (one of 'subsample[ratio]', 'features[nfeat]', 'nosampling'). The default is subsampling with a proportion of 0.80. NOTE: The square brackets should not be included (i.e subsample0.7, features210).\"}, {'name': 'merge.type', 'value': 'average', 'label': 'Merge.Type', 'description': 'Ignored when algorithm other than hierarchical selected'}, {'name': 'descent.iterations', 'value': '2000', 'label': 'Descent.Iterations', 'description': 'Number of SOM/NMF iterations'}, {'name': 'output.stub', 'value': '<input.filename_basename>', 'label': 'Output.Stub', 'description': 'Stub pre-pended to all the output file names'}, {'name': 'normalize.type', 'value': '-n1', 'label': 'Normalize.Type', 'description': 'row-wise, column-wise, both'}, {'name': 'normalization.iterations', 'value': '0', 'label': 'Normalization.Iterations', 'description': 'number of row/column normalization iterations (supercedes normalize.type)'}, {'name': 'heat.map.size', 'value': '2', 'label': 'Heat.Map.Size', 'description': \"point size of a consensus matrix's heat map (between 1 and 20)\"}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ConstellationMap': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'A downstream visualization and analysis tool for gene set enrichment results with interactive web visualizer.', 'required_args': [{'name': 'input.gct.file', 'value': '', 'label': 'Input.Gct.File', 'description': 'Input gene set enrichment file. Expected to be output from ssGSEAProjection module.'}, {'name': 'input.cls.file', 'value': '', 'label': 'Input.Cls.File', 'description': 'Phenotype labels. Input cls file describing phenotypes of gct file.'}, {'name': 'gene.sets.file', 'value': '', 'label': 'Gene.Sets.File', 'description': 'Gene set file. Must match gene sets listed in input gct file. '}, {'name': 'top.n', 'value': '', 'label': 'Top.N', 'description': 'The number of gene sets to be plotted.'}, {'name': 'target.class', 'value': '', 'label': 'Target.Class', 'description': 'The phenotypic category of interest.'}], 'default_args': [{'name': 'task_name', 'value': 'ConstellationMap', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'gene.sets.database', 'value': 'c1.all.v5.1.symbols.gmt', 'label': 'Gene.Sets.Database', 'description': 'Gene sets database from GSEA website.'}, {'name': 'direction', 'value': 'positive', 'label': 'Direction', 'description': 'The direction of correlation, as measured by mutual information, to be investigated.'}, {'name': 'image.format', 'value': 'PNG', 'label': 'Image.Format', 'description': 'The image file type of outputted plots, either bitmap (PNG) or vector graphics (PDF).'}, {'name': 'jaccard.threshold', 'value': '0.1', 'label': 'Jaccard.Threshold', 'description': 'The Jaccard index threshold above which connective edges between gene sets will be plotted.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ConvertLineEndings': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': \"Converts line endings to the host operating system's format.\", 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'The input file (any non-binary file format)'}], 'default_args': [{'name': 'task_name', 'value': 'ConvertLineEndings', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<input.filename_basename>.cvt.<input.filename_extension>', 'label': 'Output.File', 'description': 'The output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ConvertToMAGEML': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Converts a gct, res, or odf dataset file to a MAGE-ML file', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'The gct, res, or odf file to convert'}], 'default_args': [{'name': 'task_name', 'value': 'ConvertToMAGEML', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file.base.name', 'value': '<input.filename_basename>', 'label': 'Output.File.Base.Name', 'description': 'The base name of the zip output file or the MAGE-ML file'}, {'name': 'create.zip.file', 'value': 'yes', 'label': 'Create.Zip.File', 'description': 'Whether to zip the MAGE-ML files'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ConvertToMAGETAB': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'A module to export data from GenePattern in MAGE-TAB format', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'A dataset file  - .gct, .res'}, {'name': 'cls.file', 'value': '', 'label': 'Cls.File', 'description': 'A class file  - .cls'}, {'name': 'intensity.type.label', 'value': '', 'label': 'Intensity.Type.Label', 'description': 'label that specifies the type of intensity values in data matrix file '}], 'default_args': [{'name': 'task_name', 'value': 'ConvertToMAGETAB', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'sample.reference.label', 'value': 'Hybridization REF', 'label': 'Sample.Reference.Label', 'description': 'reference label for samples in the data matrix file'}, {'name': 'feature.reference.label', 'value': 'CompositeElement REF', 'label': 'Feature.Reference.Label', 'description': 'reference label for features in the data matrix file'}, {'name': 'output.prefix', 'value': '<input.file_basename>', 'label': 'Output.Prefix', 'description': 'prefix to use for output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'COPA': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Performs Cancer Outlier Profile Analysis (COPA)', 'required_args': [{'name': 'dataset.file', 'value': '', 'label': 'Dataset.File', 'description': 'The dataset file - .res, .gct'}, {'name': 'sample.info.file', 'value': '', 'label': 'Sample.Info.File', 'description': 'sample info file - .txt'}], 'default_args': [{'name': 'task_name', 'value': 'COPA', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'outlier.cutoff', 'value': '5', 'label': 'Outlier.Cutoff', 'description': \"The cutoff to determine 'outlier' status.\"}, {'name': 'pre-filter.percentile', 'value': '0.95', 'label': 'Pre-Filter.Percentile', 'description': 'The percentile to use for pre-filtering the data. All genes with a number of outlier samples less than the top (default 95th) percentile will be removed from further consideration. This value must be between 0 and 1 exclusive.'}, {'name': 'compute.large', 'value': 'no', 'label': 'Compute.Large', 'description': 'Whether to continue analysis when the number of genes remaining after pre-filtering is greater than 1000'}, {'name': 'max.overlap', 'value': '0', 'label': 'Max.Overlap', 'description': \"The maximum number of samples that can be considered 'outliers' when comparing two genes. The default is 0, indicating that there can be no overlap.\"}, {'name': 'normal.outliers', 'value': '0', 'label': 'Normal.Outliers', 'description': \"The number of normal samples that can be considered 'outliers'. The default is 0, meaning that no normals may be outliers.\"}, {'name': 'gene.pairs.to.plot', 'value': '1-5', 'label': 'Gene.Pairs.To.Plot', 'description': 'A range of gene pairs to plot. (e.g., 2-5) based on the gene pair rank. The ranks are determined by the number of outliers. The more outliers a gene pair has the greater the ranking. Specify none if no gene pair plots should be generated.'}, {'name': 'output.file', 'value': '<dataset.file_basename>', 'label': 'Output.File', 'description': 'The name of the output files'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'CopyNumberDivideByNormals': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Determines the copy number of a target SNP', 'required_args': [{'name': 'snp.file', 'value': '', 'label': 'Snp.File', 'description': 'SNP file containing both normal and target samples.'}, {'name': 'sample.info.file', 'value': '', 'label': 'Sample.Info.File', 'description': 'dChip format sample info file containing Ploidy(numeric) and Sample columns'}], 'default_args': [{'name': 'task_name', 'value': 'CopyNumberDivideByNormals', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'include.genotype.calls', 'value': 'cn', 'label': 'Include.Genotype.Calls', 'description': 'Whether to include genotype calls from the input snp file if they exist.'}, {'name': 'output.file.name', 'value': '<snp.file_basename>', 'label': 'Output.File.Name', 'description': 'Output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'CopyNumberInferencePipeline': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '[**Available on public server only**] Generates segmented copy number calls from raw tumor and normal SNP6 CEL files.', 'required_args': [{'name': 'CopyNumberInferencePipeline.Part1of21.SNPFileCreator_SNP62.cel.files', 'value': '', 'label': 'CopyNumberInferencePipeline.Part1Of21.SNPFileCreator SNP62.Cel.Files', 'description': 'Zip of SNP 6.0 CEL files'}, {'name': 'CopyNumberInferencePipeline.Part1of21.ConvertLineEndings1.input.filename', 'value': '', 'label': 'CopyNumberInferencePipeline.Part1Of21.ConvertLineEndings1.Input.Filename', 'description': 'The input file (any non-binary file format)'}], 'default_args': [{'name': 'task_name', 'value': 'CopyNumberInferencePipeline', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'CopyNumberInferencePipeline.Part1of21.TangentNormalization9.output.files', 'value': 'nocnv', 'label': 'CopyNumberInferencePipeline.Part1Of21.TangentNormalization9.Output.Files', 'description': 'The types of files to output. Whether to output files with CNVs included, removed, or both.'}, {'name': 'CopyNumberInferencePipeline.Part2of2.Wrapper4.CopyNumberInferencePipeline.Part2of21.InsertGenomicLocations3.genome.annotation', 'value': 'hg19', 'label': 'CopyNumberInferencePipeline.Part2Of2.Wrapper4.CopyNumberInferencePipeline.Part2Of21.InsertGenomicLocations3.Genome.Annotation', 'description': 'The internally provided genome annotation to insert into the input file.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'CopyNumberNoise': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Measures noise, calculates QC metrics on copynumber data.', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'The quality scores computed by the CopyNumberInference module. '}], 'default_args': [{'name': 'task_name', 'value': 'CopyNumberNoise', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<input.file_basename>.noise.txt', 'label': 'Output.File', 'description': 'The name of the output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'copyOfReorderGCT': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Reorder a GCT to be more human readable by sorting on a property such as the Gene Symbol, mean expression, sum and variance', 'required_args': [{'name': 'gct.file', 'value': '', 'label': 'Gct.File', 'description': 'A GCT file to be reordered'}], 'default_args': [{'name': 'task_name', 'value': 'copyOfReorderGCT', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'sort.feature', 'value': 'geneSymbol', 'label': 'Sort.Feature', 'description': 'The feature to sort on'}, {'name': 'order.as', 'value': 'a', 'label': 'Order.As', 'description': 'sort in Ascending or Descending order'}, {'name': 'output.file', 'value': '<gct.file_basename>_reord.gct', 'label': 'Output.File', 'description': 'Name of the reordered GCT file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'CoxRegression': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Cox regression using clinical data and .cls file', 'required_args': [{'name': 'input.surv.data.filename', 'value': '', 'label': 'Input.Surv.Data.Filename', 'description': 'input clinical data. Case ID in 1st column (.txt file)'}, {'name': 'input.cls.filename', 'value': '', 'label': 'Input.Cls.Filename', 'description': 'input class information (.cls file)'}], 'default_args': [{'name': 'task_name', 'value': 'CoxRegression', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': 'CoxRegression_result', 'label': 'Output.File', 'description': ''}, {'name': 'time', 'value': 'time', 'label': 'Time', 'description': 'Column name for survival time'}, {'name': 'status', 'value': 'status', 'label': 'Status', 'description': 'Column name for censor (0: censored/alive, 1: event/death)'}, {'name': 'variable.continuous', 'value': 'NA', 'label': 'Variable.Continuous', 'description': '(e.g. age,labdata1,...)'}, {'name': 'variable.category', 'value': 'NA', 'label': 'Variable.Category', 'description': '(e.g. sex,cls,grade,...)'}, {'name': 'variable.interaction.terms', 'value': 'NA', 'label': 'Variable.Interaction.Terms', 'description': '(e.g. sex:cls,sex:grade,...)'}, {'name': 'strata', 'value': 'NA', 'label': 'Strata', 'description': 'Fit is done for each strata'}, {'name': 'input.subgroup', 'value': 'NA', 'label': 'Input.Subgroup', 'description': '(e.g. cls:2) *only one subgroup can be specified'}, {'name': 'variable.selection', 'value': 'none', 'label': 'Variable.Selection', 'description': ''}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'CreateSymlinks': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Creates symlinks to the input files in the job results directory. This is intended to be a helper module for creating scatter-gather pipelines.  Note that this module is not supported on Windows', 'required_args': [{'name': 'input.files', 'value': '', 'label': 'Input.Files', 'description': 'One or more input files'}], 'default_args': [{'name': 'task_name', 'value': 'CreateSymlinks', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'CsvToFcs': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Converts a Comma separated values (CSV) file to a Flow Cytometry Standard (FCS) file.', 'required_args': [{'name': 'Input.CSV.data.file', 'value': '', 'label': 'Input.CSV.Data.File', 'description': 'A CSV file to be converted to FCS.'}], 'default_args': [{'name': 'task_name', 'value': 'CsvToFcs', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Output.FCS.file.name', 'value': '<Input.CSV.data.file_basename>.fcs', 'label': 'Output.FCS.File.Name', 'description': 'The name of the output FCS file.'}, {'name': 'Range', 'value': 'auto', 'label': 'Range', 'description': 'Value of the $PnR keyword for all parameters. By default, this tool will try to set the best settings based on the data in the file; use \"auto\", \"auto-exact\", \"BD\", or an integer value.'}, {'name': 'Data.type', 'value': 'auto', 'label': 'Data.Type', 'description': 'Value of the $DATATYPE keyword. By default, this tool will try to set the best settings based on the data in the input CSV file.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Cuffcompare': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Cufflinks 2.0.2 - Cuffcompare - Analyzes the transcribed fragments in an assembly', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'One or more GTF file outputs from Cufflinks.'}, {'name': 'reference.GTF', 'value': '', 'label': 'Reference.GTF', 'description': 'A reference annotation GTF'}, {'name': 'reference.genome.file', 'value': '', 'label': 'Reference.Genome.File', 'description': 'Fasta file or zip of fasta files against which your reads were aligned'}, {'name': 'additional.cuffcompare.options', 'value': '', 'label': 'Additional.Cuffcompare.Options', 'description': 'Additional options to be passed along to the Cuffcompare program at the command line.  This parameter gives you a means to specify otherwise unavailable Cuffcompare options and switches not supported by the module; check the Cufflinks manual for details.  Recommended for experts only; use this at your own discretion.'}], 'default_args': [{'name': 'task_name', 'value': 'Cuffcompare', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.prefix', 'value': '<input.file_basename>', 'label': 'Output.Prefix', 'description': 'A prefix for the module output'}, {'name': 'exclude.transcripts', 'value': 'true', 'label': 'Exclude.Transcripts', 'description': 'Whether to ignore reference transcripts that are not overlapped by any transcript in the input files.  Does not apply if no reference GTF is provided.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Cuffdiff': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Cufflinks 2.0.2 - Finds significant changes in transcript expression, splicing, and promoter use', 'required_args': [{'name': 'aligned.files', 'value': '', 'label': 'Aligned.Files', 'description': 'A set of aligned files grouped by condition'}, {'name': 'GTF.file', 'value': '', 'label': 'GTF.File', 'description': 'A transcript annotation file in GTF/GFF format produced by cufflinks, cuffcompare, cuffmerge, or other source (such as a reference annotation GTF)'}, {'name': 'frag.bias.correct', 'value': '', 'label': 'Frag.Bias.Correct', 'description': 'Reference (FASTA/FA) for bias detection and correction algorithm'}, {'name': 'mask.file', 'value': '', 'label': 'Mask.File', 'description': 'This file tells Cuffdiff to ignore all reads that could have come from transcripts in this GTF/GFF file. It is recommended that annotated rRNA, mitochondrial transcripts, and other abundant transcripts you want to ignore in your analysis be included in this file.'}, {'name': 'additional.cuffdiff.options', 'value': '', 'label': 'Additional.Cuffdiff.Options', 'description': 'Additional options to be passed along to the Cuffdiff program at the command line.  This parameter gives you a means to specify otherwise unavailable Cuffdiff options and switches not supported by the module; check the Cufflinks manual for details.  Recommended for experts only; use this at your own discretion.'}], 'default_args': [{'name': 'task_name', 'value': 'Cuffdiff', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'time.series', 'value': 'false', 'label': 'Time.Series', 'description': 'Analyze the provided samples as a time series, rather than testing for differences between all pairs of samples.'}, {'name': 'normalization.method', 'value': 'geometric', 'label': 'Normalization.Method', 'description': 'The normalization method to be used. Choices are geometric mean (the default), upper quartile, or standard FPKM normalization. See the documentation for a discussion of these methods.'}, {'name': 'FPKM.scaling', 'value': 'compatible-hits', 'label': 'FPKM.Scaling', 'description': 'Controls how Cuffdiff includes mapped fragments toward the number used in the FPKM denominator.  Use compatible-hits (the default) to count only those fragments compatible with some reference transcript.  Use total-hits to count all fragments, even those not compatible with a reference transcript.'}, {'name': 'multi.read.correct', 'value': 'true', 'label': 'Multi.Read.Correct', 'description': \"Instructs Cuffdiff to do use its two-stage read weighting algorithm to more accurately distribute a multi-mapped read's count contribution across the multiple loci the read mapped to.\"}, {'name': 'min.alignment.count', 'value': '10', 'label': 'Min.Alignment.Count', 'description': \"The minimum number of alignments in a locus needed to conduct significance testing on changes in that locus observed between samples.  If no testing is performed, changes in the locus are deemed not significant, and the locus's observed changes do not contribute to correction for multiple testing.\"}, {'name': 'FDR', 'value': '.05', 'label': 'Fdr', 'description': 'The allowed false discovery rate.'}, {'name': 'library.type', 'value': 'inferred', 'label': 'Library.Type', 'description': 'The library type used to generate reads. The default is inferred, meaning that no library type information is passed.'}, {'name': 'skip.diff.exp', 'value': 'false', 'label': 'Skip.Diff.Exp', 'description': 'Tells Cuffdiff to perform quantification and normalization only and to skip its differential expression calculations which are computationally expensive.  The default is to compute differential expression.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Cufflinks': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Cufflinks 2.0.2 - Assembles transcripts, estimates abundances, and tests for differential expression and regulation in RNA-seq samples', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'Input file in SAM or BAM format'}, {'name': 'transfrag.label', 'value': '', 'label': 'Transfrag.Label', 'description': 'A label for the transfrags in the output files'}, {'name': 'GTF', 'value': '', 'label': 'Gtf', 'description': 'Reference annotation (GFF/GTF file) for isoform expression estimates'}, {'name': 'GTF.guide', 'value': '', 'label': 'GTF.Guide', 'description': 'Reference annotation (GFF/GTF file) to guide RABT assembly'}, {'name': 'mask.file', 'value': '', 'label': 'Mask.File', 'description': 'A GTF file specifying transcripts to be ignored'}, {'name': 'frag.bias.correct', 'value': '', 'label': 'Frag.Bias.Correct', 'description': 'Reference (FASTA/FA) for bias detection and correction algorithm'}, {'name': 'additional.cufflinks.options', 'value': '', 'label': 'Additional.Cufflinks.Options', 'description': 'Additional options to be passed along to the Cufflinks program at the command line.  This parameter gives you a means to specify otherwise unavailable Cufflinks options and switches not supported by the module; check the Cufflinks manual for details.  Recommended for experts only; use this at your own discretion.'}], 'default_args': [{'name': 'task_name', 'value': 'Cufflinks', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'multi.read.correct', 'value': 'false', 'label': 'Multi.Read.Correct', 'description': 'Whether to do an initial estimation procedure to more accurately weight reads mapping to multiple locations in the genome'}, {'name': 'library.type', 'value': 'inferred', 'label': 'Library.Type', 'description': 'The library type used to generate reads.'}, {'name': 'min.frags.per.transfrag', 'value': '10', 'label': 'Min.Frags.Per.Transfrag', 'description': 'Assembled transfrags supported by fewer than this many aligned RNA-Seq fragments are not reported.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'CufflinksCuffmergePipeline': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Creates multiple transcript assemblies using Cufflinks and then merges these assemblies into one', 'required_args': [{'name': 'CreateSymlinks1.input.files', 'value': '', 'label': 'CreateSymlinks1.Input.Files', 'description': 'One or more input files'}, {'name': 'CufflinksWrapperPipeline2.Cufflinks1.GTF.guide', 'value': '', 'label': 'CufflinksWrapperPipeline2.Cufflinks1.GTF.Guide', 'description': 'Reference annotation (GFF/GTF file) to guide RABT assembly'}, {'name': 'CufflinksWrapperPipeline2.Cufflinks1.mask.file', 'value': '', 'label': 'CufflinksWrapperPipeline2.Cufflinks1.Mask.File', 'description': 'A GTF file specifying transcripts to be ignored'}, {'name': 'CufflinksWrapperPipeline2.Cufflinks1.additional.cufflinks.options', 'value': '', 'label': 'CufflinksWrapperPipeline2.Cufflinks1.Additional.Cufflinks.Options', 'description': 'Additional options to be passed along to the Cufflinks program at the command line.  This parameter gives you a means to specify otherwise unavailable Cufflinks options and switches not supported by the module; check the Cufflinks manual for details.  Recommended for experts only; use this at your own discretion.'}, {'name': 'Cufflinks.cuffmerge4.reference.GTF', 'value': '', 'label': 'Cufflinks.Cuffmerge4.Reference.GTF', 'description': 'An optional reference annotation GTF. The input assemblies are merged together with the reference GTF and included in the final output. Cuffmerge will use this to attach gene names and other metadata to the merged catalog.'}, {'name': 'Cufflinks.cuffmerge4.genome.file', 'value': '', 'label': 'Cufflinks.Cuffmerge4.Genome.File', 'description': 'A file containing the genomic DNA sequences for the reference.  This should be a multi-FASTA file with all contigs present.'}], 'default_args': [{'name': 'task_name', 'value': 'CufflinksCuffmergePipeline', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'CufflinksWrapperPipeline2.Cufflinks1.library.type', 'value': 'inferred', 'label': 'CufflinksWrapperPipeline2.Cufflinks1.Library.Type', 'description': 'The library type used to generate reads.'}, {'name': 'CufflinksWrapperPipeline2.Cufflinks1.min.frags.per.transfrag', 'value': '10', 'label': 'CufflinksWrapperPipeline2.Cufflinks1.Min.Frags.Per.Transfrag', 'description': 'Assembled transfrags supported by fewer than this many aligned RNA-Seq fragments are not reported.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Cuffmerge': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Cufflinks 2.0.2: Cuffmerge - Merge several Cufflinks assemblies', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'GTF files to be merged'}, {'name': 'reference.GTF', 'value': '', 'label': 'Reference.GTF', 'description': 'An optional reference annotation GTF. The input assemblies are merged together with the reference GTF and included in the final output. Cuffmerge will use this to attach gene names and other metadata to the merged catalog.'}, {'name': 'genome.file', 'value': '', 'label': 'Genome.File', 'description': 'A file containing the genomic DNA sequences for the reference.  This should be a multi-FASTA file with all contigs present.'}], 'default_args': [{'name': 'task_name', 'value': 'Cuffmerge', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'CummeRbund.GeneSetReport': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '[Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] Cuffdiff visualization package providing plots based on a user-provided list of genes.', 'required_args': [{'name': 'cuffdiff.input', 'value': '', 'label': 'Cuffdiff.Input', 'description': 'A Cuffdiff job, a previous CummeRbund job, or a cuffData.db file from a previous CummeRbund job.'}, {'name': 'geneset.file', 'value': '', 'label': 'Geneset.File', 'description': 'A text file containing the list of genes for this report, one per line.  Each of these can be represented by a gene symbol (short name), gene ID, isoform_id, tss_group_id, or cds_id.'}, {'name': 'selected.conditions', 'value': '', 'label': 'Selected.Conditions', 'description': 'Specifies the conditions (samples) to be used in the plots.  This should be a comma-separated list of conditions, using the same names as in the upstream Cuffdiff job; leave this blank to use all conditions.  If blank, no pairwise comparison plots will be generated.'}], 'default_args': [{'name': 'task_name', 'value': 'CummeRbund.GeneSetReport', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.format', 'value': 'pdf', 'label': 'Output.Format', 'description': 'The output file format.'}, {'name': 'feature.level', 'value': 'genes', 'label': 'Feature.Level', 'description': 'Feature level for the report.'}, {'name': 'report.as.aggregate', 'value': 'yes', 'label': 'Report.As.Aggregate', 'description': 'Controls whether reporting should be done for individual replicates or aggregate condition/sample values.  The default is to use aggregate sample values.  Note that the Volcano plot always shows aggregate samples.'}, {'name': 'log.transform', 'value': 'yes', 'label': 'Log.Transform', 'description': 'Whether or not to log transform the FPKM values.  Note that the FPKM values are always log transformed for the Volcano plots and that the Scatter plot will use a log2 rather than log10 transformation.  For the Expression Plot and Expression Barplot, this directs that the y-axis will be drawn on a log10 scale.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'CummeRbund.QcReport': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '[Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] Cuffdiff visualization package providing plots and tables related to QC and Global Stats.', 'required_args': [{'name': 'cuffdiff.input', 'value': '', 'label': 'Cuffdiff.Input', 'description': 'A Cuffdiff job, a previous CummeRbund job, or a cuffData.db file from a previous CummeRbund job.'}], 'default_args': [{'name': 'task_name', 'value': 'CummeRbund.QcReport', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.format', 'value': 'pdf', 'label': 'Output.Format', 'description': 'The output file format.'}, {'name': 'feature.level', 'value': 'genes', 'label': 'Feature.Level', 'description': 'Feature level for the report'}, {'name': 'report.as.aggregate', 'value': 'yes', 'label': 'Report.As.Aggregate', 'description': 'Controls whether reporting should be done for individual replicates or aggregate condition/sample values.  The default is to use aggregate sample values.  Controls whether reporting should be done for individual replicates or aggregate condition/sample values.  The default is to use aggregate sample values.  Note that the Dispersion, FPKM.SCV, and MA plots always show aggregate samples.'}, {'name': 'log.transform', 'value': 'yes', 'label': 'Log.Transform', 'description': 'Whether or not to log transform FPKM values.  Note that the FPKM values are always log transformed for the FPKM.SCV plot.  Log transformation does not apply to the Dispersion, MDS, and PCA plots.'}, {'name': 'pca.x', 'value': 'PC2', 'label': 'Pca.X', 'description': 'Indicates which principal component is to be presented on the x-axis of the PCA plot.  Must differ from pca.y.'}, {'name': 'pca.y', 'value': 'PC3', 'label': 'Pca.Y', 'description': 'Indicates which principal component is to be presented on the y-axis of the PCA plot.  Must differ from pca.x.'}, {'name': 'attempt.to.merge.names', 'value': 'no', 'label': 'Attempt.To.Merge.Names', 'description': 'Should the module attempt to merge gene names into the text reports?  Depending on the GTF file used in the Cuffdiff run, the text reports may instead contain tracking ID references, particularly if a Cuffmerge merged.gtf file was used.  Note that this setting does not affect the plots.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'CummeRbund.SelectedConditionsReport': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '[**Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] Cuffdiff visualization package providing plots based on a user-provided list of conditions.', 'required_args': [{'name': 'cuffdiff.input', 'value': '', 'label': 'Cuffdiff.Input', 'description': 'A Cuffdiff job, a previous CummeRbund job, or a cuffData.db file from a previous CummeRbund job.'}, {'name': 'selected.conditions', 'value': '', 'label': 'Selected.Conditions', 'description': 'Specifies the conditions (samples) to be used in the plots.  This should be a comma-separated list of conditions, using the same names as in the upstream Cuffdiff job; leave this blank to use all conditions.  If blank, no pairwise comparison plots will be generated.'}], 'default_args': [{'name': 'task_name', 'value': 'CummeRbund.SelectedConditionsReport', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.format', 'value': 'pdf', 'label': 'Output.Format', 'description': 'The output file format.'}, {'name': 'feature.level', 'value': 'genes', 'label': 'Feature.Level', 'description': 'Feature level for the report.'}, {'name': 'report.as.aggregate', 'value': 'yes', 'label': 'Report.As.Aggregate', 'description': 'Controls whether reporting should be done for individual replicates or aggregate condition/sample values.  The default is to use aggregate sample values.  Note that the Volcano plot always shows aggregate samples.'}, {'name': 'log.transform', 'value': 'yes', 'label': 'Log.Transform', 'description': 'Whether or not to log transform the FPKM values.  Note that the FPKM values are always log transformed for the Volcano plots and that the Scatter plot will use a log2 rather than log10 transformation.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'CummeRbund.SelectedGeneReport': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '[**Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] Cuffdiff visualization package providing plots based on a single user-specified gene.', 'required_args': [{'name': 'cuffdiff.input', 'value': '', 'label': 'Cuffdiff.Input', 'description': 'A Cuffdiff job, a previous CummeRbund job, or a cuffData.db file from a previous CummeRbund job.'}, {'name': 'feature.id', 'value': '', 'label': 'Feature.Id', 'description': 'The gene or feature of interest.  This can be a gene symbol (short name), gene ID, isoform_id, tss_group_id, or cds_id.'}, {'name': 'selected.conditions', 'value': '', 'label': 'Selected.Conditions', 'description': 'Specifies the conditions (samples) to be used in the plots.  This should be a comma-separated list of conditions, using the same names as in the upstream Cuffdiff job; leave this blank to use all conditions.  The Similarity plots will always operate across all conditions.'}], 'default_args': [{'name': 'task_name', 'value': 'CummeRbund.SelectedGeneReport', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'find.similar', 'value': '10', 'label': 'Find.Similar', 'description': 'Optionally, find and plot the top genes (up to this count) with an expression profile most similar to the given gene of interest.  If blank, this will be skipped.    '}, {'name': 'output.format', 'value': 'pdf', 'label': 'Output.Format', 'description': 'The output file format.'}, {'name': 'feature.level', 'value': 'genes', 'label': 'Feature.Level', 'description': 'Feature level for the report.  Note that the Similarity plots will always be generated at the \"genes\" level.'}, {'name': 'report.as.aggregate', 'value': 'yes', 'label': 'Report.As.Aggregate', 'description': 'Controls whether reporting should be done for individual replicates or aggregate condition/sample values.  The default is to use aggregate sample values.  Note that the Similarity plots always show aggregate samples.'}, {'name': 'log.transform', 'value': 'yes', 'label': 'Log.Transform', 'description': 'Whether or not to log transform the FPKM values.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'CytoscapeViewer': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Visualizes a gene network', 'required_args': [{'name': 'dataset.file', 'value': '', 'label': 'Dataset.File', 'description': 'Dataset file - .res, .gct'}, {'name': 'adjacency.matrix.file', 'value': '', 'label': 'Adjacency.Matrix.File', 'description': 'Adjacency matrix - .adj'}], 'default_args': [{'name': 'task_name', 'value': 'CytoscapeViewer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'DAPPLE': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'DAPPLE', 'required_args': [{'name': 'Input.File', 'value': '', 'label': 'Input.File', 'description': 'File containing a list of SNPs, genes or regions. Each input should be on a new line.'}, {'name': 'Keyword', 'value': '', 'label': 'Keyword', 'description': ''}], 'default_args': [{'name': 'task_name', 'value': 'DAPPLE', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Permute.Length', 'value': '0', 'label': 'Permute.Length', 'description': 'To increase speed, this is adaptive permutation (see README)'}, {'name': 'Plot', 'value': 'true', 'label': 'Plot', 'description': ''}, {'name': 'nearest.gene', 'value': 'true', 'label': 'Nearest.Gene', 'description': 'Do you want to use the gene nearest to the SNP (rather than grabbing all genes in the LD wingspan)? Only relevant for SNP inputs.'}, {'name': 'genome.build', 'value': '18', 'label': 'Genome.Build', 'description': 'Which set of SNPs should we use? hg18, hg19, 1KG?'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'DeIdentifyFCS': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'DeIdentifyFCS an FCS data file; remove keywords from a list or matching a regular expression; useful to anonymize FCS data files and/or to remove specific (e.g., clinical) information ', 'required_args': [{'name': 'Input.FCS.data.file', 'value': '', 'label': 'Input.FCS.Data.File', 'description': 'An FCS data file to be de-identified.'}, {'name': 'Remove.keywords', 'value': '', 'label': 'Remove.Keywords', 'description': \"Comma-separated list of keyword names (escape comma by '\\\\' if part of a keyword name) or regular expression to match keywords to be removed.\"}], 'default_args': [{'name': 'task_name', 'value': 'DeIdentifyFCS', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Is.regular.expression.used', 'value': 'No', 'label': 'Is.Regular.Expression.Used', 'description': \"Is a regular expression used in the list of keywords to remove? If not (default) a simple comma-separated list of keyword names is expected. In this list, comma may be escaped by '\\\\' if part of a keyword name, e.g., this \\\\, is in the keyword.\"}, {'name': 'Output.FCS.file.name', 'value': '<Input.FCS.data.file_basename>.deidentified.<Input.FCS.data.file_extension>', 'label': 'Output.FCS.File.Name', 'description': 'Output FCS file name.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'DependencyBiomarkerDotplot': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Creates a dotplot of Achilles gene dependency scores. The dots are colored based on the biomarker status of the cell line (i.e. mutation in gene or no mutation in gene). This module was created for use in the Education Outreach high school computational curriculum. ', 'required_args': [{'name': 'Achilles.Gene.Dependency.Data.File', 'value': '', 'label': 'Achilles.Gene.Dependency.Data.File', 'description': 'Format is a csv file, with cell lines in rows and genes in columns. '}, {'name': 'Biomarker.Data.File', 'value': '', 'label': 'Biomarker.Data.File', 'description': 'Format is a csv file, with cell lines in rows and genes in columns. '}, {'name': 'Gene.1', 'value': '', 'label': 'Gene.1', 'description': 'The gene name should match exactly to the column names in the data files and are case-sensitive. \\n'}, {'name': 'Gene.#2', 'value': '', 'label': 'Gene.#2', 'description': 'The gene name should match exactly to the column names in the data files and are case-sensitive.'}, {'name': 'Gene.#3', 'value': '', 'label': 'Gene.#3', 'description': 'The gene name should match exactly to the column names in the data files and are case-sensitive.'}, {'name': 'Gene.#4', 'value': '', 'label': 'Gene.#4', 'description': 'The gene name should match exactly to the column names in the data files and are case-sensitive.'}, {'name': 'Gene.#5', 'value': '', 'label': 'Gene.#5', 'description': 'The gene name should match exactly to the column names in the data files and are case-sensitive.'}], 'default_args': [{'name': 'task_name', 'value': 'DependencyBiomarkerDotplot', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'DESeq2': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'A Differential Expression workflow for RNA-Seq data based on the DESeq2 package from Bioconductor ', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'A GCT file containing raw RNA-Seq counts, such as is produced by MergeHTSeqCounts'}, {'name': 'cls.file', 'value': '', 'label': 'Cls.File', 'description': 'A categorical CLS file specifying the phenotype classes for the samples in the GCT file.  This should contain exactly two classes with the control specified first.'}, {'name': 'confounding.variable.cls.file', 'value': '', 'label': 'Confounding.Variable.Cls.File', 'description': 'A categorical CLS file specifying an additional confounding variable, mapped to the input file samples.  Use this for a two-factor comparison.'}], 'default_args': [{'name': 'task_name', 'value': 'DESeq2', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file.base', 'value': '<input.file_basename>', 'label': 'Output.File.Base', 'description': 'The base name of the output file(s). File extensions will be added automatically.'}, {'name': 'qc.plot.format', 'value': 'skip', 'label': 'Qc.Plot.Format', 'description': 'Choose a file format for the QC plots (or skip them completely)'}, {'name': 'fdr.threshold', 'value': '0.1', 'label': 'Fdr.Threshold', 'description': 'FDR threshold to use in filtering before creating the reports of top up-regulated and down-regulated genes '}, {'name': 'top.N.count', 'value': '20', 'label': 'Top.N.Count', 'description': 'Max gene count to use when creating the reports of top up-regulated and down-regulated genes '}, {'name': 'random.seed', 'value': '779948241', 'label': 'Random.Seed', 'description': 'Seed to use for randomization'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'DownloadURL': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Downloads a file from a URL', 'required_args': [{'name': 'URL', 'value': '', 'label': 'Url', 'description': 'The URL to retrieve the file from'}, {'name': 'output.file', 'value': '', 'label': 'Output.File', 'description': 'The name of the output file'}], 'default_args': [{'name': 'task_name', 'value': 'DownloadURL', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ESPPredictor': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Determine which tryptic peptides will generate the highest signal in ESI-MS', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'List of peptide sequences (one per line, no column header)'}], 'default_args': [{'name': 'task_name', 'value': 'ESPPredictor', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ExpressionFileCreator': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Creates a RES or GCT file from a set of Affymetrix CEL files.  For IVT arrays only; use AffySTExpressionFileCreator for ST arrays.', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'A zip file of CEL files'}, {'name': 'quantile.normalization', 'value': '', 'label': 'Quantile.Normalization', 'description': '(GCRMA and RMA only) Whether to normalize data using quantile normalization'}, {'name': 'background.correct', 'value': '', 'label': 'Background.Correct', 'description': '(RMA only) Whether to background correct using RMA background correction'}, {'name': 'value.to.scale.to', 'value': '', 'label': 'Value.To.Scale.To', 'description': '(median/mean scaling only) The value to scale to.'}, {'name': 'clm.file', 'value': '', 'label': 'Clm.File', 'description': 'A tab-delimited text file containing one scan, sample, and class per line'}, {'name': 'cdf.file', 'value': '', 'label': 'Cdf.File', 'description': 'Custom CDF file. Leave blank to use default internally provided CDF file (custom cdf file is not implemented for GCRMA).'}], 'default_args': [{'name': 'task_name', 'value': 'ExpressionFileCreator', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'method', 'value': 'RMA', 'label': 'Method', 'description': 'The method to use. Note that dchip and MAS5 will not work with ST arrays.'}, {'name': 'compute.present.absent.calls', 'value': 'no', 'label': 'Compute.Present.Absent.Calls', 'description': 'Whether to compute Present/Absent calls'}, {'name': 'normalization.method', 'value': 'median scaling', 'label': 'Normalization.Method', 'description': '(MAS5 only) The normalization method to apply after expression values are computed. The column having the median of the means is used as the reference unless the parameter value to scale to is given.'}, {'name': 'annotate.probes', 'value': 'yes', 'label': 'Annotate.Probes', 'description': 'Whether to annotate probes with the gene symbol and description'}, {'name': 'output.file', 'value': '<input.file_basename>', 'label': 'Output.File', 'description': 'The base name of the output file(s)'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ExprToGct': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Converts an EXPR file to GCT format', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'Input file in EXPR format'}], 'default_args': [{'name': 'task_name', 'value': 'ExprToGct', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.prefix', 'value': '<input.file_basename>', 'label': 'Output.Prefix', 'description': 'A label that will be used to name the GCT output file.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ExtractColumnNames': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Lists the sample descriptors from a .res file.', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'Input filename - .res'}], 'default_args': [{'name': 'task_name', 'value': 'ExtractColumnNames', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output', 'value': '<input.filename_basename>', 'label': 'Output', 'description': 'Name of the output file (a .txt extension will be appended)'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ExtractComparativeMarkerResults': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Creates a derived dataset and feature list file from the results of ComparativeMarkerSelection', 'required_args': [{'name': 'comparative.marker.selection.filename', 'value': '', 'label': 'Comparative.Marker.Selection.Filename', 'description': 'The results from ComparativeMarkerSelection - .odf'}, {'name': 'dataset.filename', 'value': '', 'label': 'Dataset.Filename', 'description': 'The dataset file used to select markers - .gct, .res, Dataset'}, {'name': 'statistic', 'value': '', 'label': 'Statistic', 'description': 'The statistic to filter features on'}, {'name': 'min', 'value': '', 'label': 'Min', 'description': 'Select features with statistic >= min'}, {'name': 'max', 'value': '', 'label': 'Max', 'description': 'Select features with statistic <= max'}, {'name': 'number.of.neighbors', 'value': '', 'label': 'Number.Of.Neighbors', 'description': 'Number of neighbors to select by score in each direction'}], 'default_args': [{'name': 'task_name', 'value': 'ExtractComparativeMarkerResults', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'base.output.name', 'value': '<comparative.marker.selection.filename_basename>.filt', 'label': 'Base.Output.Name', 'description': 'The base name for the output files'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ExtractFCSDataset': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Extract one or more FCS datasets from an FCS data file', 'required_args': [{'name': 'Input.data.file', 'value': '', 'label': 'Input.Data.File', 'description': 'The input FCS data file; one or more data sets will be extracted from this file.'}, {'name': 'Output.file.name', 'value': '', 'label': 'Output.File.Name', 'description': 'The name for the output file. Uses the input data file name (for example, name.fcs) as the default, and adds _datasetIndex to the name (e.g., name_datasetIndex.fcs).'}, {'name': 'Dataset.indexes', 'value': '', 'label': 'Dataset.Indexes', 'description': 'A list of data set indexes to be extracted from the data file (e.g., 1-3, 5, 8). The first data set has the index 1, the second the index 2, etc. All data sets will be extracted if no indexes are provided.'}], 'default_args': [{'name': 'task_name', 'value': 'ExtractFCSDataset', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ExtractFCSKeywords': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Extracts keyword(s) value(s) from a Flow Cytometry Standard (FCS) file.', 'required_args': [{'name': 'Input.FCS.data.file', 'value': '', 'label': 'Input.FCS.Data.File', 'description': 'The input FCS data file from which the module will extract keyword/value pairs.'}], 'default_args': [{'name': 'task_name', 'value': 'ExtractFCSKeywords', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Keywords.output.mode', 'value': 'CSV', 'label': 'Keywords.Output.Mode', 'description': 'The file format to be used to store the extracted keywords/value pairs.'}, {'name': 'Keywords.output.file.name', 'value': '<Input.FCS.data.file_basename>', 'label': 'Keywords.Output.File.Name', 'description': 'The name for the file in which the keyword/value pairs are stored.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ExtractFCSParameters': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Extract specified parameters (dimensions) from an FCS data file.', 'required_args': [{'name': 'Input.FCS.data.file', 'value': '', 'label': 'Input.FCS.Data.File', 'description': 'The input FCS data file from which to extract parameters.'}, {'name': 'Parameters.to.extract', 'value': '', 'label': 'Parameters.To.Extract', 'description': 'A list of parameters to extract; parameters may be specified by their indexes (e.g., 1-3, 5, 8) or by their short names (e.g., FSC-H, SSC-H, FL1-A, FL3-H).'}, {'name': 'Output.file.name', 'value': '', 'label': 'Output.File.Name', 'description': 'The output file name. The default uses the input file name as the base.'}], 'default_args': [{'name': 'task_name', 'value': 'ExtractFCSParameters', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ExtractRowNames': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Extracts the row names from a .res, .gct, or .odf file.', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'input file - .res, .gct, .odf, file from which the row names are extracted'}], 'default_args': [{'name': 'task_name', 'value': 'ExtractRowNames', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output', 'value': '<input.filename_basename>.row.names.txt', 'label': 'Output', 'description': 'output filename containing the row names - .txt'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'FastQC': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '[**Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] Provides quality control metrics on raw sequence data', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'RNA-seq reads file in FASTQ (bz2 and gz compressed files are supported), SAM, or BAM format.'}, {'name': 'input.format', 'value': '', 'label': 'Input.Format', 'description': 'Bypasses the normal sequence file format detection and forces the program to use the specified format.'}, {'name': 'contaminant.file', 'value': '', 'label': 'Contaminant.File', 'description': 'FastQC screens overrepresented sequences (> 0.1% of total) against a set of possible contaminants consisting of all the primers and adapters used by the various sequencing platforms.  The \"Overrepresented sequences\" report generated by FastQC will indicate when an overrepresented sequence corresponds to a known contaminant.  This optional contaminant file provides a means of adding more sequences to FastQC\\'s built-in list of known contaminants. See module documentation for a description and example of this file\\'s format.'}], 'default_args': [{'name': 'task_name', 'value': 'FastQC', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'kmer.size', 'value': '5', 'label': 'Kmer.Size', 'description': 'The length of Kmer to look for in the Kmer content module. The specified Kmer length must be between 2 and 10. Default length is 5.'}, {'name': 'extract.output', 'value': 'true', 'label': 'Extract.Output', 'description': 'Whether to output an uncompressed version of the report. Set this to yes to view the report directly from within GenePattern.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'FCMFeatureExtraction': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Calculation of features from clustered flow cytometry data.', 'required_args': [{'name': 'Input.CSV.files', 'value': '', 'label': 'Input.CSV.Files', 'description': 'The set of CSV files with clustering results (of a single data file). May be a .zip file, a directory, or a text file listing all input file paths (one file per line).'}], 'default_args': [{'name': 'task_name', 'value': 'FCMFeatureExtraction', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Calculate.number.of.cells', 'value': 'TRUE', 'label': 'Calculate.Number.Of.Cells', 'description': 'Calculate and output the number of cells in each sub-population?'}, {'name': 'Calculate.percentage.of.cells', 'value': 'TRUE', 'label': 'Calculate.Percentage.Of.Cells', 'description': 'Calculate and output the percentage (proportion) of cells in each sub-population?'}, {'name': 'Calculate.means', 'value': 'TRUE', 'label': 'Calculate.Means', 'description': 'Calculate mean parameter values (e.g. Mean Fluorescence Intensities) for all parameters of each sub-population?'}, {'name': 'Output.CSV.file.name', 'value': 'features.csv', 'label': 'Output.CSV.File.Name', 'description': 'The name of the output CSV file.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'FCMSinglePanelQC': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Single panel flow cytometry data quality control', 'required_args': [{'name': 'Input.FCS.data.files', 'value': '', 'label': 'Input.FCS.Data.Files', 'description': 'The set of input FCS files to be tested.'}], 'default_args': [{'name': 'task_name', 'value': 'FCMSinglePanelQC', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Perform.cell.number.test', 'value': 'TRUE', 'label': 'Perform.Cell.Number.Test', 'description': 'Run the cell number control test.'}, {'name': 'Perform.boundary.events.test', 'value': 'TRUE', 'label': 'Perform.Boundary.Events.Test', 'description': 'Run the boundary events test.'}, {'name': 'Perform.timeflow.test', 'value': 'TRUE', 'label': 'Perform.Timeflow.Test', 'description': 'Run the timeflow test, i.e., check whether the flow rate was steady and uninterrupted.'}, {'name': 'Perform.timeline.test', 'value': 'TRUE', 'label': 'Perform.Timeline.Test', 'description': 'Run the timeline test, i.e., detect unexpected non-random acquisition of fluorescence intensities.'}, {'name': 'Zip.output', 'value': 'TRUE', 'label': 'Zip.Output', 'description': 'Zip the resulting report together?'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'FCSClean': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Clean up an FCS file by marking events affected by problems during data acquisition', 'required_args': [{'name': 'Input.FCS', 'value': '', 'label': 'Input.FCS', 'description': 'The FCS file to be cleaned up, i.e., bad events/cells will be marked by a high value of the added Good_vs_Bad channel.'}, {'name': 'Markers', 'value': '', 'label': 'Markers', 'description': 'A comma-separated list of dimensions (flow cytometry parameters/channels) to be checked. The module accepts both, a list of marker names as well as a list of parameter indexes. All dimensions but scatters and time will be used if the Markers parameter is not provided.'}], 'default_args': [{'name': 'task_name', 'value': 'FCSClean', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Output.FCS', 'value': '<Input.FCS_basename>.cleaned.fcs', 'label': 'Output.FCS', 'description': 'The name of the output FCS file.'}, {'name': 'Bin.Size', 'value': '0.01', 'label': 'Bin.Size', 'description': 'The size of a bin as proportion of cells in the FCS file, typically between 0.01 and 0.2.'}, {'name': 'Cell.Cutoff', 'value': '500', 'label': 'Cell.Cutoff', 'description': 'The number of cells to be used for a cut off, typically between 300 and 800, see documentation for more details.'}, {'name': 'Cutoff', 'value': 'median', 'label': 'Cutoff', 'description': 'The method of how cutoff is performed by the algorithm. Typically, a median works well. Alternatively, you may specify a value from the [0,1] interval as a cutoff percentile, or a value greater than 1, which will be assumed as a direct specification of the cutoff threshold.'}, {'name': 'Max.Fc', 'value': '1.3', 'label': 'Max.Fc', 'description': 'Relative increase from mean norm of presumed good data below which data is still presumed to be good.'}, {'name': 'Announce', 'value': 'TRUE', 'label': 'Announce', 'description': 'Shall we print information about identifying vs. not identifying any bad events to the standard output?'}, {'name': 'Diagnostic', 'value': 'FALSE', 'label': 'Diagnostic', 'description': 'Shall we include a diagnostics plot?'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'FCSNormalization': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Per-channel basis normalization of flow cytometry data.', 'required_args': [{'name': 'Input.FCS.files', 'value': '', 'label': 'Input.FCS.Files', 'description': 'The set of input FCS files to be normalized.'}, {'name': 'Channels.to.normalize', 'value': '', 'label': 'Channels.To.Normalize', 'description': 'Comma-separated list of parameter short names to normalize.'}], 'default_args': [{'name': 'task_name', 'value': 'FCSNormalization', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Apply.ASinH', 'value': 'TRUE', 'label': 'Apply.ASinH', 'description': 'Apply the asinh transformation prior normalization?'}, {'name': 'Create.report', 'value': 'TRUE', 'label': 'Create.Report', 'description': 'Create additional histogram channel report?'}, {'name': 'Landmarks.number', 'value': '1', 'label': 'Landmarks.Number', 'description': 'Number of landmarks to be used for normalization.'}, {'name': 'Zip.output', 'value': 'TRUE', 'label': 'Zip.Output', 'description': 'Zip the resulting normalized FCS data files together?'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'FcsToCsv': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Converts a Flow Cytometry Standard (FCS) file to a Comma Separated Values (CSV) file.', 'required_args': [{'name': 'Input.FCS.data.file', 'value': '', 'label': 'Input.FCS.Data.File', 'description': 'An FCS data file to be converted to CSV.'}], 'default_args': [{'name': 'task_name', 'value': 'FcsToCsv', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Output.CSV.file.name', 'value': '<Input.FCS.data.file_basename>.csv', 'label': 'Output.CSV.File.Name', 'description': 'The name of the output CSV file.'}, {'name': 'Use.full.names', 'value': 'TRUE', 'label': 'Use.Full.Names', 'description': 'Whether to try to extract full parameter names from the FCS file.'}, {'name': 'Output.keywords.file.name', 'value': '<Input.FCS.data.file_basename>', 'label': 'Output.Keywords.File.Name', 'description': 'The file name to store keyword value pairs extracted from the FCS file.'}, {'name': 'Output.keywords.mode', 'value': 'CSV', 'label': 'Output.Keywords.Mode', 'description': 'How to store keywords.'}, {'name': 'Channel.to.scale.conversion', 'value': 'TRUE', 'label': 'Channel.To.Scale.Conversion', 'description': 'Whether to convert log channel values to linear scale.'}, {'name': 'Precision', 'value': 'FALSE', 'label': 'Precision', 'description': 'Whether to use regular (32bit, recommended) or double precision.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'FeatureSummaryViewer': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Feature Summary Viewer', 'required_args': [{'name': 'feature.filename', 'value': '', 'label': 'Feature.Filename', 'description': 'odf file with or without features to summarize'}], 'default_args': [{'name': 'task_name', 'value': 'FeatureSummaryViewer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'FileSplitter': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Splits a file into equal-sized chunks, given the number of lines per output file', 'required_args': [{'name': 'inputFile', 'value': '', 'label': 'InputFile', 'description': 'The input file, to be split into chunks'}, {'name': 'numFiles', 'value': '', 'label': 'NumFiles', 'description': \"[Integer, >=1], The maximum number of output files to split the file into. When both 'numFiles' and 'numLines' are set, treat this as the maximum number of output files to create, even if the files have more than the requested number of lines\"}, {'name': 'numLines', 'value': '', 'label': 'NumLines', 'description': '[Integer, >=1], The number of lines in each output file.'}], 'default_args': [{'name': 'task_name', 'value': 'FileSplitter', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'FilterFilelist': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'a helper module which generates a filtered filelist from the input filelist and some filter parameters.', 'required_args': [{'name': 'input.filelist', 'value': '', 'label': 'Input.Filelist', 'description': 'the initial input filelist, a file containing a list of filepaths'}, {'name': 'includePattern', 'value': '', 'label': 'IncludePattern', 'description': 'an optional list of filename patterns to match, e.g. *.cls,*.gct'}, {'name': 'errorOnEmptySet', 'value': '', 'label': 'ErrorOnEmptySet', 'description': 'Whether the module should output stderr if the output result is empty'}], 'default_args': [{'name': 'task_name', 'value': 'FilterFilelist', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'type', 'value': 'f', 'label': 'Type', 'description': 'Include files, directories, or both files and directories in the output filelist'}, {'name': 'output.filename', 'value': '<input.filelist_basename>.filtered.<input.filelist_extension>', 'label': 'Output.Filename', 'description': 'the name of the output file, if none specified, write to stdout'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'FilterLowshRNAs': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Removes shRNAs with low read counts in original DNA plasmid pool. Outputs a GCT file containing filtered shRNA data and a GCT file containing the mean read count for each shRNA in the original DNA plasmid pool. ', 'required_args': [{'name': 'GCT.Files', 'value': '', 'label': 'GCT.Files', 'description': 'Please enter one or more GCT files.'}, {'name': 'File.Prefix', 'value': '', 'label': 'File.Prefix', 'description': ''}], 'default_args': [{'name': 'task_name', 'value': 'FilterLowshRNAs', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Find.differentially.expressed.genes.in.RNASeq.data': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'This pipeline is used in the GenomeSpace recipe, \"Find differentially expressed genes in RNA-Seq data\". The recipe can be found at: http://www.genomespace.org/support/guides/recipes/sections/find-differentially-expressed-genes-in-rna-seq-data.', 'required_args': [{'name': 'ConvertLineEndings1.input.filename', 'value': '', 'label': 'ConvertLineEndings1.Input.Filename', 'description': 'The input file (any non-binary file format)'}, {'name': 'TopHat2.reads.pair.1', 'value': '', 'label': 'TopHat2.Reads.Pair.1', 'description': 'Unpaired reads file or first mate for paired reads. One or more files containing reads in FASTA or FASTQ format (bz2 and gz compressed files are supported).'}, {'name': 'TopHat4.reads.pair.1', 'value': '', 'label': 'TopHat4.Reads.Pair.1', 'description': 'Unpaired reads file or first mate for paired reads. One or more files containing reads in FASTA or FASTQ format (bz2 and gz compressed files are supported).'}], 'default_args': [{'name': 'task_name', 'value': 'Find.differentially.expressed.genes.in.RNASeq.data', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Find.subnetworks.of.differentially.expressed.genes.and.identify.associated.biological.functions': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'This pipeline is used in the GenomeSpace recipe, \"Find subnetworks of differentially expressed genes and identify associated biological functions\". The recipe can be found at: http://www.genomespace.org/support/guides/recipes/sections/find-subnetworks-of-differentially-expressed-genes-and-identify-associated-biological-functions.', 'required_args': [{'name': 'PreprocessDataset1.input.filename', 'value': '', 'label': 'PreprocessDataset1.Input.Filename', 'description': 'Input filename - .res, .gct, .odf'}, {'name': 'ComparativeMarkerSelection2.cls.file', 'value': '', 'label': 'ComparativeMarkerSelection2.Cls.File', 'description': 'The class file - .cls'}], 'default_args': [{'name': 'task_name', 'value': 'Find.subnetworks.of.differentially.expressed.genes.and.identify.associated.biological.functions', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'FLAMEChooseOptimalClusterNumber': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Determines the optimal number of clusters for each sample, using user-specified criterion', 'required_args': [{'name': 'mixture.model', 'value': '', 'label': 'Mixture.Model', 'description': 'A .zip file containing mixture modeling results from the FLAMEMixtureModel module'}], 'default_args': [{'name': 'task_name', 'value': 'FLAMEChooseOptimalClusterNumber', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'method', 'value': 'SWR', 'label': 'Method', 'description': 'Criterion used for determining optimal number of clusters'}, {'name': 'seed', 'value': '123456', 'label': 'Seed', 'description': 'An integer; the seed fixes random computational processes during computation. This does not need to be changed under most circumstances.'}, {'name': 'output.prefix', 'value': '<mixture.model_basename>', 'label': 'Output.Prefix', 'description': 'A prefix for output files'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'FLAMEContourDataGenerator': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Generates data for drawing FLAME contour plots', 'required_args': [{'name': 'membership.file', 'value': '', 'label': 'Membership.File', 'description': 'A membership.txt or member.txt file output from one of the FLAME modules. '}, {'name': 'parameters.file', 'value': '', 'label': 'Parameters.File', 'description': 'A parameters.txt file output from one of the FLAME modules. '}, {'name': 'channels.to.plot', 'value': '', 'label': 'Channels.To.Plot', 'description': 'A comma separated list of the index or names of three channels/antibodies to plot'}], 'default_args': [{'name': 'task_name', 'value': 'FLAMEContourDataGenerator', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.prefix', 'value': '<parameters.file_basename>', 'label': 'Output.Prefix', 'description': 'A prefix for output files'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'FLAMEContourViewer.Pipeline': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Pipeline which runs the FLAMECountorDataGenerator and the FLAMEViewer', 'required_args': [{'name': 'FLAMEContourDataGenerator1.membership.file', 'value': '', 'label': 'FLAMEContourDataGenerator1.Membership.File', 'description': 'A membership.txt or member.txt file output from one of the FLAME modules.'}, {'name': 'FLAMEContourDataGenerator1.parameters.file', 'value': '', 'label': 'FLAMEContourDataGenerator1.Parameters.File', 'description': 'A parameters.txt file output from one of the FLAME modules.'}, {'name': 'FLAMEContourDataGenerator1.channels.to.plot', 'value': '', 'label': 'FLAMEContourDataGenerator1.Channels.To.Plot', 'description': 'A comma separated list of the index or names of three channels/antibodies to plot'}], 'default_args': [{'name': 'task_name', 'value': 'FLAMEContourViewer.Pipeline', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'FLAMEContourDataGenerator1.output.prefix', 'value': 'flame_contour', 'label': 'FLAMEContourDataGenerator1.Output.Prefix', 'description': 'A prefix for output files'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'FLAMEMetacluster': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Matches populations/clusters across all samples in the dataset, assembles a table of all parameters in all samples', 'required_args': [{'name': 'optimal.g.mixture.model', 'value': '', 'label': 'Optimal.G.Mixture.Model', 'description': 'A .zip file containing the optimal mixture modeling result of each sample from the output of the FLAMEChooseOptimalClusterNumber module.'}, {'name': 'sample.class', 'value': '', 'label': 'Sample.Class', 'description': ''}], 'default_args': [{'name': 'task_name', 'value': 'FLAMEMetacluster', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'estimate.mode', 'value': 'F', 'label': 'Estimate.Mode', 'description': 'Used only for skew distributions. Whether to estimate the mode for each cluster. Must be the same value used in FLAMEMixtureModel.'}, {'name': 'estimation.increment', 'value': '1', 'label': 'Estimation.Increment', 'description': 'Used only for skew distributions and when estimate mode is set to yes. The smaller the increment, the more accurate the estimation, but the slower the estimation step. Must be the same value used in FLAMEMixtureModel.'}, {'name': 'output.intermediate.results', 'value': 'F', 'label': 'Output.Intermediate.Results', 'description': 'Choose whether to output the intermediate metaclustering results, such as within-class matching results.'}, {'name': 'output.prefix', 'value': '<optimal.g.mixture.model_basename>', 'label': 'Output.Prefix', 'description': 'A prefix for output files.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'FLAMEMixtureModel': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Clusters and estimates population parameters, using preprocessed flow cytometric data', 'required_args': [{'name': 'preprocessed.data', 'value': '', 'label': 'Preprocessed.Data', 'description': 'A .zip file containing preprocessed flow sample files in tab-delimited .txt format.'}, {'name': 'channels.to.cluster', 'value': '', 'label': 'Channels.To.Cluster', 'description': 'A comma-separated list of either channel numbers (e.g., 1, 2, 3, 7) or channel names (e.g., CD4, CD45RA, ZAP70); these denote which antibodies in the (preprocessed) data sample to be used for clustering.'}], 'default_args': [{'name': 'task_name', 'value': 'FLAMEMixtureModel', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'g.min', 'value': '3', 'label': 'G.Min', 'description': 'An integer; minimal number of clusters to be fit for each sample.'}, {'name': 'g.max', 'value': '10', 'label': 'G.Max', 'description': 'An integer; maximal number of clusters to be fit for each sample. g max should be greater or equal to g min.'}, {'name': 'density', 'value': 'skewt', 'label': 'Density', 'description': 'Density distribution to be used for clustering'}, {'name': 'estimate.mode', 'value': 'F', 'label': 'Estimate.Mode', 'description': 'Used only for skew distributions. Whether to estimate the mode for each cluster.  Must be the same value used in FLAMEMixtureModel.'}, {'name': 'estimation.increment', 'value': '1', 'label': 'Estimation.Increment', 'description': 'Used only for skew distributions and when estimate mode is set to yes. The smaller the increment, the more accurate the estimation, but the slower the estimation step. Must be the same value used in FLAMEMixtureModel.'}, {'name': 'seed', 'value': '123456', 'label': 'Seed', 'description': 'An integer; the seed fixes random computational processes during computation. This does not need to be changed under most circumstances.'}, {'name': 'output.prefix', 'value': '<preprocessed.data_basename>', 'label': 'Output.Prefix', 'description': 'A prefix for output files.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'FLAMEPreprocess': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Performs preprocessing of .fcs or .txt flow cytometric data files', 'required_args': [{'name': 'dataset', 'value': '', 'label': 'Dataset', 'description': 'A .zip file containing flow files in .txt or .fcs format. (The files should be all .txt or all .fcs, but not both.)'}, {'name': 'channels', 'value': '', 'label': 'Channels', 'description': 'A comma-separated list of channel numbers indicating which columns (antibodies) to keep (e.g., 1, 2, 3, 7).'}, {'name': 'channel.names', 'value': '', 'label': 'Channel.Names', 'description': 'A comma-separated list of channel/antibody names that corresponds to the channel numbers (e.g., FSC, SSC, CD4, CD45RA).'}], 'default_args': [{'name': 'task_name', 'value': 'FLAMEPreprocess', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'file.type', 'value': 'fcs', 'label': 'File.Type', 'description': 'File type of input sample data.'}, {'name': 'data.scale', 'value': '262144', 'label': 'Data.Scale', 'description': 'The original data scale. 18-bit (default) or 4-decade.'}, {'name': 'remove.dead', 'value': 'F', 'label': 'Remove.Dead', 'description': 'Whether to attempt removing dead cells from the samples.'}, {'name': 'scatter.channels', 'value': '1,2', 'label': 'Scatter.Channels', 'description': 'A comma-separated list of scatter channel numbers (e.g., 1, 2, 3).'}, {'name': 'transformation', 'value': 'logicle', 'label': 'Transformation', 'description': 'The transformation to apply'}, {'name': 'logicle.cofactor', 'value': '3', 'label': 'Logicle.Cofactor', 'description': 'A cofactor that tunes the logicle transformation. The cofactor is usually greater than 0 and less than 10.'}, {'name': 'arsinh.cofactor', 'value': '250', 'label': 'Arsinh.Cofactor', 'description': 'A cofactor that tunes the arsinh transformation. The cofactor is 0 usually between 0 exclusively and a few thousand.'}, {'name': 'output.prefix', 'value': '<dataset_basename>', 'label': 'Output.Prefix', 'description': 'A prefix for output files'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'FLAMEPreviewTransformation': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Evaluates data transformations based on one representative sample', 'required_args': [{'name': 'data.file', 'value': '', 'label': 'Data.File', 'description': 'A flow file in .txt or .fcs format for one sample.'}, {'name': 'channels', 'value': '', 'label': 'Channels', 'description': 'A comma-separated list of channel numbers indicating which columns (antibodies) to keep (e.g., 1, 2, 3, 7).'}, {'name': 'channel.names', 'value': '', 'label': 'Channel.Names', 'description': 'A comma-separated list of channel or antibody names that corresponds to the channel numbers (e.g., FSC, SSC, CD4, CD45RA).'}], 'default_args': [{'name': 'task_name', 'value': 'FLAMEPreviewTransformation', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'file.type', 'value': 'fcs', 'label': 'File.Type', 'description': 'File type of input sample data.'}, {'name': 'data.scale', 'value': '262144', 'label': 'Data.Scale', 'description': 'The original data scale. 18-bit (default) or 4-decade.'}, {'name': 'scatter.channels', 'value': '1,2', 'label': 'Scatter.Channels', 'description': 'A comma-separated list of scatter channel numbers (e.g., 1, 2, 3).'}, {'name': 'transformation', 'value': 'logicle', 'label': 'Transformation', 'description': 'The transformation(s) to apply'}, {'name': 'logicle.low', 'value': '0.5', 'label': 'Logicle.Low', 'description': 'The cofactor that tunes the logicle transformation is usually between 0 and 10 exclusively. Specify the lower limit of the cofactor range to examine.'}, {'name': 'logicle.high', 'value': '5', 'label': 'Logicle.High', 'description': 'The cofactor that tunes the logicle transformation is usually greater than 0 and less than 10. Specify the upper limit of the cofactor range to examine.'}, {'name': 'arsinh.low', 'value': '1', 'label': 'Arsinh.Low', 'description': 'The cofactor that tunes the arsinh transformation is usually between 0 and a few thousand. Specify the lower limit of the cofactor range to examine.'}, {'name': 'arsinh.high', 'value': '1000', 'label': 'Arsinh.High', 'description': 'The cofactor that tunes the arsinh transformation is usually between 0 and a few thousand. Specify the upper limit of the cofactor range to examine.'}, {'name': 'output.prefix', 'value': '<data.file_basename>', 'label': 'Output.Prefix', 'description': 'A prefix for output files'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'FLAMEViewer': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Visualizes results from the FLAME method', 'required_args': [{'name': 'membership.file', 'value': '', 'label': 'Membership.File', 'description': 'A membership.txt file output from the FLAMEContourDataGenerator module. This membership file must match with the contour plot file.'}, {'name': 'contour.plot.file', 'value': '', 'label': 'Contour.Plot.File', 'description': 'The contour plot output file from the FLAMEContourDataGenerator module. This contour plot file must match with the membership file.'}], 'default_args': [{'name': 'task_name', 'value': 'FLAMEViewer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'FlowClustClassifyFCS': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Classify flow cytometry data using the FlowClust algorithm.', 'required_args': [{'name': 'Input.FCS.data.file', 'value': '', 'label': 'Input.FCS.Data.File', 'description': 'The FCS file to be clustered.'}, {'name': 'K', 'value': '', 'label': 'K', 'description': 'The number of populations that the algorithm shall be looking for. If you do not know the K value, you can use the SuggestNumberOfPopulations module to estimate it.'}, {'name': 'Channels', 'value': '', 'label': 'Channels', 'description': 'A comma-separated list of parameter short names ($PnN values) that shall be used to classify the populations. All parameters will be used by default, i.e., if no value for the Channels parameter is provided. The Time parameter is excluded automatically.'}], 'default_args': [{'name': 'task_name', 'value': 'FlowClustClassifyFCS', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'CSV.file.name', 'value': '<Input.FCS.data.file_basename>', 'label': 'CSV.File.Name', 'description': 'The base file name to split the results into CSV files, each containing a single population.'}, {'name': 'CLS.file.name', 'value': '<Input.FCS.data.file_basename>.cls', 'label': 'CLS.File.Name', 'description': 'The name of the CLS output file, which will store the population number for each event.'}, {'name': 'Quantile', 'value': '0.9', 'label': 'Quantile', 'description': 'Outlier quantile. Events outside the specified quantile region will be classified as outliers and be assigned to class 0.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'FlowFingerprinting': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'takes a flow set and generates a description of the multivariate probability distribution function of its flow cytometry data in the form of a fingerprint', 'required_args': [{'name': 'Input.FCS.data.files', 'value': '', 'label': 'Input.FCS.Data.Files', 'description': 'The FCS file(s) to be fingerprinted.'}, {'name': 'Model.FCS.data.file', 'value': '', 'label': 'Model.FCS.Data.File', 'description': 'Training data (FCS file) for the underlying model. If none, the input FCS data is used for creating a default model.'}, {'name': 'Model.parameters', 'value': '', 'label': 'Model.Parameters', 'description': 'A comma-separated list of dimensions/parameters/channels that shall be used for the underlying model. Use either parameter names or indexes. All dimensions except for Time will be used if this parameter is not provided.'}, {'name': 'Model.number.of.recursions', 'value': '', 'label': 'Model.Number.Of.Recursions', 'description': 'Number of times the FCS training data will be subdivided.'}, {'name': 'Model.sample.size', 'value': '', 'label': 'Model.Sample.Size', 'description': 'Used to specify the per flowFrame sample size of the data to use in model generation. If NULL, all of the data in x is used. Setting this to a smaller number will speed up processing, at the cost of accuracy.'}, {'name': 'Workflow.parameter.file', 'value': '', 'label': 'Workflow.Parameter.File', 'description': 'A CSV file containing parameters if module is used in pipeline.'}], 'default_args': [{'name': 'task_name', 'value': 'FlowFingerprinting', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Model.dequantize', 'value': '0', 'label': 'Model.Dequantize', 'description': 'Option to make all events in the training set unique.'}, {'name': 'Output.QC.screening.fingerprint.stack', 'value': '0', 'label': 'Output.QC.Screening.Fingerprint.Stack', 'description': 'Graphical output in form of \"stack\" plot.'}, {'name': 'Output.QC.screening.gate.consistency', 'value': '0', 'label': 'Output.QC.Screening.Gate.Consistency', 'description': 'Graphical output in form of \"qc\" plot.'}, {'name': 'Output.QC.screening.plate', 'value': '1', 'label': 'Output.QC.Screening.Plate', 'description': 'Graphical output in form of \"plate\" plot.'}, {'name': 'Output.csv.file', 'value': 'fingerprints.csv', 'label': 'Output.Csv.File', 'description': 'Name for output CSV file.'}, {'name': 'Output.file.transformation', 'value': '0', 'label': 'Output.File.Transformation', 'description': 'Transformation used on data for CSV output file. \"raw\" is the number of events in each bin, \"normalized\" is the ratio of the number of events divided by the expected number (the expected number of events is calculated by dividing the total number of events in the instance used to make the fingerprint by the number of features in the fingerprint). \"log2norm\" is the log2 of the normalized values.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'FlowMeansCluster': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Cluster flow cytometry data using the FlowMeans algorithm.', 'required_args': [{'name': 'Input.FCS.data.file', 'value': '', 'label': 'Input.FCS.Data.File', 'description': 'The FCS file to be clustered.'}, {'name': 'Dimensions', 'value': '', 'label': 'Dimensions', 'description': 'A comma-separated list of dimensions (parameters/channels) that shall be used for clustering. Use either parameter names or indexes. All dimensions except for Time will be used if this parameter is not provided.'}, {'name': 'Dimensions.to.transform', 'value': '', 'label': 'Dimensions.To.Transform', 'description': 'A comma-separated list of dimensions (parameters/channels) that shall transformed as specified by previous parameter. This will be ignored if no transformation is specified above. If this parameter is not provided and transformation is specified above, the algorithm will use heuristics to identify parameters that shall be transformed. These heuristics are based on how parameters are stored in the FCS file, their resolution and their name. Again, you can use either parameter names or parameter indexes to specify dimensions to transform.'}, {'name': 'Maximum.number.of.clusters', 'value': '', 'label': 'Maximum.Number.Of.Clusters', 'description': 'Maximum number of clusters that FlowMeans will try. This value will be estimated automatically if the value of this parameter is not provided.'}, {'name': 'Number.of.clusters', 'value': '', 'label': 'Number.Of.Clusters', 'description': 'Number of clusters that FlowMeans shall search for. This parameter cannot be combined with the \"Maximum number of clusters\". Also, the number of clusters will be estimated automatically if this parameter is not provided.'}, {'name': 'Max.points.for.covariance', 'value': '', 'label': 'Max.Points.For.Covariance', 'description': 'The maximum number of data points (events) used for calculating the covariance. All data points will be used if this value is not provided.'}, {'name': 'Max.points.for.kernel.density', 'value': '', 'label': 'Max.Points.For.Kernel.Density', 'description': 'The maximum number of data points (events) used for counting the modes using kernel density estimation. All data points will be used if this value is not provided.'}, {'name': 'Seed', 'value': '', 'label': 'Seed', 'description': 'Seed value for the random generator; provide any consistent value to obtain reproducible results.'}], 'default_args': [{'name': 'task_name', 'value': 'FlowMeansCluster', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Transformation', 'value': 'ASinH', 'label': 'Transformation', 'description': 'Which transformation to apply on the data prior clustering. Fluorescence channels are usually better visualized and clustered using a transformation; the ASinH transformation produces good results on most data.'}, {'name': 'Maximum.number.of.iterations', 'value': '50', 'label': 'Maximum.Number.Of.Iterations', 'description': 'The maximum number of iterations allowed within the K-means component of the clustering.'}, {'name': 'Number.of.initialization.sets', 'value': '10', 'label': 'Number.Of.Initialization.Sets', 'description': 'The number of random sets used for initialization.'}, {'name': 'Distance.used', 'value': 'TRUE', 'label': 'Distance.Used', 'description': 'The distance being used within the FlowMeans algorithm.'}, {'name': 'Standardize.data', 'value': 'TRUE', 'label': 'Standardize.Data', 'description': 'Should each data dimension be standardized (i.e, transformed to a [0,1] interval) before clustering. This usually does not harm and is important if data in dimensions used for clustering covers significantly different ranges for different dimensions.   '}, {'name': 'Distance.matrix.update', 'value': 'Mahalanobis', 'label': 'Distance.Matrix.Update', 'description': 'How to update the distance matrix at each merging iteration. Note that updating using Mahalanobis distance is not possible if Euclidean distance is set in the \"Distance used\" above.  '}, {'name': 'Transform.residuals', 'value': 'FALSE', 'label': 'Transform.Residuals', 'description': 'Indicate whether or not to transform residuals to orthogonal distance.   '}, {'name': 'Add.noise', 'value': 'TRUE', 'label': 'Add.Noise', 'description': 'Should a little bit of uniform noise be added to the data to prevent singularity issues?'}, {'name': 'Draw.plots', 'value': 'TRUE', 'label': 'Draw.Plots', 'description': 'Should plots of clustered data be created for quick visual inspection of the results?'}, {'name': 'Print.debug.info', 'value': 'FALSE', 'label': 'Print.Debug.Info', 'description': 'Should additional debugging information be provided (e.g., this may be useful to debug cases where clustering does not work properly for certain data files).'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'FlowMergeCluster': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Cluster flow cytometry data using FlowMerge.', 'required_args': [{'name': 'Input.FCS.data.file', 'value': '', 'label': 'Input.FCS.Data.File', 'description': 'The FCS file to be clustered.'}, {'name': 'Dimensions', 'value': '', 'label': 'Dimensions', 'description': 'A comma-separated list of dimensions (parameters/channels) that shall be used for clustering. Use either parameter names or indexes. All dimensions except for Time will be used if this parameter is not provided.'}, {'name': 'Dimensions.to.transform', 'value': '', 'label': 'Dimensions.To.Transform', 'description': 'A comma-separated list of dimensions (parameters/channels) that shall transformed as specified by previous parameter. This will be ignored if no transformation is specified above. If this parameter is not provided and transformation is specified above, the algorithm will use heuristics to identify parameters that shall be transformed. These heuristics are based on how parameters are stored in the FCS file, their resolution and their name. Again, you can use either parameter names or parameter indexes to specify dimensions to transform.'}], 'default_args': [{'name': 'task_name', 'value': 'FlowMergeCluster', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Transformation', 'value': 'ASinH', 'label': 'Transformation', 'description': 'Which transformation to apply on the data prior clustering. Fluorescence channels are usually better visualized and clustered using a transformation; the ASinH transformation produces good results on most data.'}, {'name': 'Range.for.number.of.clusters', 'value': '1-10', 'label': 'Range.For.Number.Of.Clusters', 'description': 'The range for the number of subpopulations (clusters) that FlowMerge shall search for. FlowMerge will try to pick the best number of clusters from the specified range, which shall be provided in the min-max format, where both, mix and max are integers and min is smaller than max, e.g., 1-10.'}, {'name': 'Estimate.degrees.of.freedom', 'value': '0', 'label': 'Estimate.Degrees.Of.Freedom', 'description': 'An indication whether to estimate the degrees of freedom used for the t distribution when modelling data.'}, {'name': 'Degrees.of.freedom', 'value': '4', 'label': 'Degrees.Of.Freedom', 'description': 'The degrees of freedom used for the t distribution when modelling data. Provided value will be ignored if estimation is requested by the Estimate degrees of freedom parameter. Gaussian distribution will be used if Degrees of freedom are not provided and estimation is not requested.'}, {'name': 'Number.of.computing.nodes', 'value': '1', 'label': 'Number.Of.Computing.Nodes', 'description': \"How many nodes (e.g., processors, cores) to use if you wish to run the analysis in a parallel mode? Enter '1' if you wish do NOT want to use the parallel mode. Enter a number higher than 1 if your server/cluster has multiple computers/processors/cores and you want to utilize several of these for FlowMerge clustering.\"}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Fpkm_trackingToGct': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Takes a Cufflinks FPKM_tracking file and converts it to GCT format.', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'Input file in FPKM_TRACKING format'}], 'default_args': [{'name': 'task_name', 'value': 'Fpkm_trackingToGct', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'row.labels', 'value': 'Tracking ID', 'label': 'Row.Labels', 'description': 'Select row labels to be written to GCT file'}, {'name': 'row.descriptions', 'value': 'Locus', 'label': 'Row.Descriptions', 'description': 'Select row descriptions to be written to GCT file'}, {'name': 'filter.rows', 'value': 'yes', 'label': 'Filter.Rows', 'description': 'Whether to filter rows with invalid row names (ie., \"-\")'}, {'name': 'output.prefix', 'value': '<input.file_basename>', 'label': 'Output.Prefix', 'description': 'A label that will be used to name the GCT output file.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'FPmatching': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Matches Fluidigm fingerprint report to reference set of CCLE SNP6.0 array data.\\nTakes old Fluidigm software report or new PED style report (saved as .csv).\\nAlso backwards compatible with Sequenom reports (Combine 100k and AffySNP6.0 reports together into one file first).', 'required_args': [{'name': 'platereport', 'value': '', 'label': 'Platereport', 'description': 'Should be .csv format'}, {'name': 'platemapping', 'value': '', 'label': 'Platemapping', 'description': 'Should be .csv format'}], 'default_args': [{'name': 'task_name', 'value': 'FPmatching', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'format', 'value': 'new', 'label': 'Format', 'description': 'New PED style report OR Old Fluidigm software format OR Sequenom style report (expects combined 100k and SNP6.0 file)'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Gct2Arff': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Description: Convert a gene pattern .gct / .cls file pair into an .arff file', 'required_args': [{'name': 'gct.file', 'value': '', 'label': 'Gct.File', 'description': 'The dataset features - .gct'}, {'name': 'cls.file', 'value': '', 'label': 'Cls.File', 'description': 'The dataset class labels - .cls'}], 'default_args': [{'name': 'task_name', 'value': 'Gct2Arff', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<gct.file_basename>', 'label': 'Output.File', 'description': 'The output file base name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'GctToPcl': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Convert .gct to .pcl', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'GCT file to convert'}], 'default_args': [{'name': 'task_name', 'value': 'GctToPcl', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.filename', 'value': '<input.filename_basename>.pcl', 'label': 'Output.Filename', 'description': 'PCL file that will be produced'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'GENE_E': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Launches GENE-E with input file(s)', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'Dataset to be viewed in GENE-E'}, {'name': 'sample.information.or.class.file', 'value': '', 'label': 'Sample.Information.Or.Class.File', 'description': 'The cls or sample info file'}], 'default_args': [{'name': 'task_name', 'value': 'GENE_E', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'GeneListSignificanceViewer': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Views the results of marker analysis', 'required_args': [{'name': 'inputfilename', 'value': '', 'label': 'Inputfilename', 'description': 'odf format file from marker analysis (GeneNeighbors or ClassNeighbors)'}], 'default_args': [{'name': 'task_name', 'value': 'GeneListSignificanceViewer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'GeneNeighbors': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Select genes that most closely resemble a continuous profile (e.g., another gene)', 'required_args': [{'name': 'data.filename', 'value': '', 'label': 'Data.Filename', 'description': 'data file - .gct, .res, .odf'}, {'name': 'gene.accession', 'value': '', 'label': 'Gene.Accession', 'description': 'reference gene accession from data file to find neighbors for'}], 'default_args': [{'name': 'task_name', 'value': 'GeneNeighbors', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'num.neighbors', 'value': '50', 'label': 'Num.Neighbors', 'description': 'number of neighbors to find'}, {'name': 'marker.list.file', 'value': '<data.filename_basename>.markerlist', 'label': 'Marker.List.File', 'description': 'output filename for analysis results - .odf file'}, {'name': 'marker.dataset.file', 'value': '<data.filename_basename>.markerdata', 'label': 'Marker.Dataset.File', 'description': 'output filename raw data for selected markers - .gct file'}, {'name': 'distance.metric', 'value': '3', 'label': 'Distance.Metric', 'description': 'continuous metric for finding neighbors'}, {'name': 'filter.data', 'value': 'no', 'label': 'Filter.Data', 'description': 'if no, values below will be ignored'}, {'name': 'min.threshold', 'value': '20', 'label': 'Min.Threshold', 'description': 'minimum threshold for data'}, {'name': 'max.threshold', 'value': '20000', 'label': 'Max.Threshold', 'description': 'maximum threshold for data'}, {'name': 'min.fold.diff', 'value': '5', 'label': 'Min.Fold.Diff', 'description': 'minimum fold difference for filtering genes'}, {'name': 'min.abs.diff', 'value': '50', 'label': 'Min.Abs.Diff', 'description': 'minimum absolute difference for filtering genes'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'GenePatternDocumentExtractor': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Extracts GenePattern pipelines and other embedded GenePattern data from Word 2007 documents created with the Microsoft Word Add-In for the GenePattern Reproducible Research Document.', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'The file you want to extract the data from. Note that this must be a Word 2007 (.docx) file -- Files saved as Word 2003 or earlier (2003) may be compatible with the Add-In, but will not work with the extractor.'}], 'default_args': [{'name': 'task_name', 'value': 'GenePatternDocumentExtractor', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'GEOImporter': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Imports data from the Gene Expression Omnibus (GEO)', 'required_args': [{'name': 'GEO.accession', 'value': '', 'label': 'GEO.Accession', 'description': 'GEO Series record (GSExxx) or a GEO Dataset (GDSxxx). Either a GEO accession or a GEO SOFT file must be given.'}, {'name': 'GEO.SOFT.file', 'value': '', 'label': 'GEO.SOFT.File', 'description': 'A previously downloaded GEO SOFT file or its gzipped representation. Either a GEO SOFT file or a GEO accession must be given.'}, {'name': 'ftp.proxy.server', 'value': '', 'label': 'Ftp.Proxy.Server', 'description': 'FTP Proxy server'}, {'name': 'ftp.proxy.username', 'value': '', 'label': 'Ftp.Proxy.Username', 'description': 'FTP Proxy username'}, {'name': 'ftp.proxy.password', 'value': '', 'label': 'Ftp.Proxy.Password', 'description': 'FTP Proxy password'}, {'name': 'http.proxy.server', 'value': '', 'label': 'Http.Proxy.Server', 'description': 'HTTP Proxy server'}, {'name': 'http.proxy.username', 'value': '', 'label': 'Http.Proxy.Username', 'description': 'HTTP Proxy username'}, {'name': 'http.proxy.password', 'value': '', 'label': 'Http.Proxy.Password', 'description': 'HTTP Proxy password'}], 'default_args': [{'name': 'task_name', 'value': 'GEOImporter', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'data.column.name', 'value': 'VALUE', 'label': 'Data.Column.Name', 'description': 'The name of the column that contains the expression values'}, {'name': 'output.filename', 'value': '<GEO.accession><GEO.SOFT.file_basename>', 'label': 'Output.Filename', 'description': 'The name of the output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'GetDatasetInSilico': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'downloads a compressed .tgz file from the Insilico servers and extract it', 'required_args': [{'name': 'DownloadURL1.URL', 'value': '', 'label': 'DownloadURL1.URL', 'description': 'The URL to retrieve the file from'}], 'default_args': [{'name': 'task_name', 'value': 'GetDatasetInSilico', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'GISTIC_2.0': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Genomic Identification of Significant Targets in Cancer (version 2.0.22)', 'required_args': [{'name': 'refgene.file', 'value': '', 'label': 'Refgene.File', 'description': 'The reference genome file input provides cytoband and gene location information.'}, {'name': 'seg.file', 'value': '', 'label': 'Seg.File', 'description': 'The segmentation file contains the segmented data for all the samples identified by GLAD, CBS, or some other segmentation algorithm. (See GLAD file format in the Genepattern file formats documentation.) It is a six column, tab-delimited file with an optional first line identifying the columns. Positions are in base pair units.'}, {'name': 'markers.file', 'value': '', 'label': 'Markers.File', 'description': 'The markers file identifies the marker names and positions of the markers in the original dataset (before segmentation). It is a three column, tab-delimited file with an optional header. If not already, markers are sorted by genomic position.'}, {'name': 'array.list.file', 'value': '', 'label': 'Array.List.File', 'description': 'The array list file is an optional file identifying the subset of samples to be used in the analysis. It is a one column file with an optional header. The sample identifiers listed in the array list file must match the sample names given in the segmentation file.'}, {'name': 'cnv.file', 'value': '', 'label': 'Cnv.File', 'description': 'Specifies germline CNV (or other highly variant) regions to exclude from significance analysis. There are two forms for the cnv file: one allows CNVs to be identified by marker names in a single column; the other identifies excluded regions using their genomic location.'}], 'default_args': [{'name': 'task_name', 'value': 'GISTIC_2.0', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'gene.gistic', 'value': '1', 'label': 'Gene.Gistic', 'description': 'Flag indicating that the gene GISTIC algorithm should be used to calculate the significant regions of deletion using gene locations instead of a marker locations.'}, {'name': 'amplifications.threshold', 'value': '0.1', 'label': 'Amplifications.Threshold', 'description': 'Threshold for copy number amplifications. Regions with a log2 ratio above this value are considered amplified.'}, {'name': 'deletions.threshold', 'value': '0.1', 'label': 'Deletions.Threshold', 'description': 'Threshold for copy number deletions. Regions with a log2 ratio below the negative of this value are considered deletions.'}, {'name': 'join.segment.size', 'value': '4', 'label': 'Join.Segment.Size', 'description': 'Smallest number of markers to allow in segments from the segmented data. Segments that contain a number of markers less than or equal to this number are joined to the neighboring segment that is closest in copy number.'}, {'name': 'q-value.threshold', 'value': '0.25', 'label': 'Q-Value.Threshold', 'description': 'Maximum false discovery rate for a region to be considered significant.'}, {'name': 'remove.X', 'value': '1', 'label': 'Remove.X', 'description': 'Flag indicating whether to remove data from the X-chromosome before analysis.'}, {'name': 'cap.val', 'value': '1.5', 'label': 'Cap.Val', 'description': 'Minimum and maximum cap values on analyzed data. Regions with a log2 ratio greater than the cap are set to the cap value; regions with a log2 ratio less than -cap value are set to -cap.'}, {'name': 'confidence.level', 'value': '0.90', 'label': 'Confidence.Level', 'description': 'Confidence level used to calculate the region containing a driver.'}, {'name': 'run.broad.analysis', 'value': '0', 'label': 'Run.Broad.Analysis', 'description': 'Flag indicating whether an additional broad-level analysis should be performed.'}, {'name': 'focal.length.cutoff', 'value': '0.50', 'label': 'Focal.Length.Cutoff', 'description': 'Threshold used to distinguish broad from focal events, given in units of fraction of chromosome arm.'}, {'name': 'max.sample.segs', 'value': '2500', 'label': 'Max.Sample.Segs', 'description': 'Maximum number of segments allowed for a sample in the input data. Samples with more segments than this threshold are excluded from the analysis.'}, {'name': 'arm.peel', 'value': '0', 'label': 'Arm.Peel', 'description': 'Whether to perform arm level peel off. This helps separate peaks which cleans up noise.'}, {'name': 'sample.center', 'value': 'median', 'label': 'Sample.Center', 'description': 'Method for centering each sample prior to the GISTIC analysis.'}, {'name': 'gene.collapse.method', 'value': 'extreme', 'label': 'Gene.Collapse.Method', 'description': 'Method for reducing marker-level copy number data to the gene-level copy number data in the gene tables. Markers contained in the gene are used when available, otherwise the flanking marker or markers are used.'}, {'name': 'output.prefix', 'value': '<seg.file_basename>', 'label': 'Output.Prefix', 'description': 'The prefix for the output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'GISTICPreprocess': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Prepares SNP files for GISTIC', 'required_args': [{'name': 'snp.archive', 'value': '', 'label': 'Snp.Archive', 'description': 'A zipfile containing the input genotypes and signal intensities. These are .cnx files created by SNPFileCreator.'}, {'name': 'sample.info.file', 'value': '', 'label': 'Sample.Info.File', 'description': 'The sample info file is a tab-delimited text file (*.txt) containing information about the arrays.'}, {'name': 'array.list.file', 'value': '', 'label': 'Array.List.File', 'description': 'The array list file identifies the subset of samples to be used in the analysis. It is a one-column file with a header (??array??). The sample identifiers listed in the array list file must match the sample names given in the segmentation file.'}], 'default_args': [{'name': 'task_name', 'value': 'GISTICPreprocess', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'use.paired.normals', 'value': '0', 'label': 'Use.Paired.Normals', 'description': 'When set to 1, normalizes tumors against paired normals. (Default = 0; best results are usually achieved under this default setting)'}, {'name': 'quality.control.tumors', 'value': '1', 'label': 'Quality.Control.Tumors', 'description': 'When set to 1, performs histogram quality control on tumor samples and includes only those tumors with distinct copy number peaks in the output array list to be used in the GISTIC module. (Default = 1).'}, {'name': 'normal.selection.method', 'value': '1', 'label': 'Normal.Selection.Method', 'description': 'Method for selecting normals to use in sample normalization. Allowed values: 0 (use all normals), 1 (use the n normals most closely matched to the tumor??s noise profile). (Default = 1)'}, {'name': 'n.closest.n', 'value': '5', 'label': 'N.Closest.N', 'description': 'Number of normals to choose when using closest_n normalization. (Default = 5)'}, {'name': 'correct.batch.effect', 'value': '0', 'label': 'Correct.Batch.Effect', 'description': 'When set to 1, runs module to correct for marker intensity variations due to batch effect. (Default = 0)'}, {'name': 'batch.correct.minimum.size', 'value': '5', 'label': 'Batch.Correct.Minimum.Size', 'description': 'Minimum number of samples required in a batch for the batch to be eligible for batch correction. (Default = 5)'}, {'name': 'batch.correct.pvalue', 'value': '0.001', 'label': 'Batch.Correct.Pvalue', 'description': 'p-value threshold for batch effect correction. For each marker, signal intensities in each batch are compared to all others using a t-test. If the p-value for this comparison is less than the p-value threshold, the signal intensities in this batch will be brought to the mean of all other batches. (Default = 0.001)'}, {'name': 'verbosity', 'value': '0', 'label': 'Verbosity', 'description': 'Set verbosity level (stand-alone version only). At a verbosity level of 0, no text indicating program status is written to sdout. At a level of 30, the maximum amount of text is written to stdout. (Default = 0)'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'GLAD': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Gain and Loss Analysis of DNA', 'required_args': [{'name': 'copy.number.input.file', 'value': '', 'label': 'Copy.Number.Input.File', 'description': 'cn file that contains sorted SNPs and raw copy number value per probe. (note: data should not be in log2)'}], 'default_args': [{'name': 'task_name', 'value': 'GLAD', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.filename', 'value': '<copy.number.input.file_basename>.glad', 'label': 'Output.Filename', 'description': 'Name of the output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'GlobalAlignment': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'standard global sequence analysis', 'required_args': [{'name': 'subject.fasta.filename', 'value': '', 'label': 'Subject.Fasta.Filename', 'description': 'filename of subject FASTA file'}, {'name': 'match', 'value': '', 'label': 'Match', 'description': '> 0'}, {'name': 'mismatch', 'value': '', 'label': 'Mismatch', 'description': '< 0'}, {'name': 'gapopen', 'value': '', 'label': 'Gapopen', 'description': '< 0'}, {'name': 'gapextend', 'value': '', 'label': 'Gapextend', 'description': '< 0'}, {'name': 'query.filename', 'value': '', 'label': 'Query.Filename', 'description': 'FASTA file'}], 'default_args': [{'name': 'task_name', 'value': 'GlobalAlignment', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<subject.fasta.filename_basename>', 'label': 'Output.File', 'description': 'filename of output'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Golub.Slonim.1999.Nature.all.aml.pipeline': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'ALL/AML methodology, from Golub and Slonim et al., 1999', 'required_args': [], 'default_args': [{'name': 'task_name', 'value': 'Golub.Slonim.1999.Nature.all.aml.pipeline', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'GSEA': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Gene Set Enrichment Analysis', 'required_args': [{'name': 'expression.dataset', 'value': '', 'label': 'Expression.Dataset', 'description': 'Dataset file - .res, .gct'}, {'name': 'gene.sets.database', 'value': '', 'label': 'Gene.Sets.Database', 'description': 'Gene sets database from GSEA website.'}, {'name': 'gene.sets.database.file', 'value': '', 'label': 'Gene.Sets.Database.File', 'description': 'Gene sets database - .gmt, .gmx, .grp. Upload a gene set if your gene set is not listed as a choice for the gene sets database parameter.'}, {'name': 'phenotype.labels', 'value': '', 'label': 'Phenotype.Labels', 'description': 'Cls file - .cls, must be binary'}, {'name': 'target.profile', 'value': '', 'label': 'Target.Profile', 'description': 'Name of the target phenotype profile.  Only applicable if class file defines continuous labels (one or more phenotype profiles).  Leave blank if class file defines a discrete phenotype.'}, {'name': 'chip.platform.file', 'value': '', 'label': 'Chip.Platform.File', 'description': 'DNA Chip (array) annotation file from GSEA website.  Upload your own chip file if the one corresponding to your DNA Microarray platform is not listed in the drop-down menu.  A chip file is only required if collapse dataset is set to true.'}], 'default_args': [{'name': 'task_name', 'value': 'GSEA', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'number.of.permutations', 'value': '1000', 'label': 'Number.Of.Permutations', 'description': 'Number of permutations to perform'}, {'name': 'collapse.dataset', 'value': 'true', 'label': 'Collapse.Dataset', 'description': 'Select true to have GSEA collapse each probe set in the expression dataset into a single vector for the gene, which gets identified by its gene symbol.'}, {'name': 'permutation.type', 'value': 'phenotype', 'label': 'Permutation.Type', 'description': 'Type of permutations to perform'}, {'name': 'scoring.scheme', 'value': 'weighted', 'label': 'Scoring.Scheme', 'description': 'The statistic used to score hits (gene set members) and misses (non-members)'}, {'name': 'metric.for.ranking.genes', 'value': 'Signal2Noise', 'label': 'Metric.For.Ranking.Genes', 'description': 'Class separation metric - gene markers are ranked using this metric to produce the gene list'}, {'name': 'gene.list.sorting.mode', 'value': 'real', 'label': 'Gene.List.Sorting.Mode', 'description': 'Mode in which scores from the gene list should be considered'}, {'name': 'gene.list.ordering.mode', 'value': 'descending', 'label': 'Gene.List.Ordering.Mode', 'description': 'Direction in which the gene list should be ordered'}, {'name': 'max.gene.set.size', 'value': '500', 'label': 'Max.Gene.Set.Size', 'description': 'Gene sets larger than this are excluded from the analysis'}, {'name': 'min.gene.set.size', 'value': '15', 'label': 'Min.Gene.Set.Size', 'description': 'Gene sets smaller than this are excluded from the analysis'}, {'name': 'collapsing.mode.for.probe.sets.with.more.than.one.match', 'value': 'Max_probe', 'label': 'Collapsing.Mode.For.Probe.Sets.With.More.Than.One.Match', 'description': 'Collapsing mode'}, {'name': 'normalization.mode', 'value': 'meandiv', 'label': 'Normalization.Mode', 'description': 'Normalization mode'}, {'name': 'randomization.mode', 'value': 'no_balance', 'label': 'Randomization.Mode', 'description': 'Type of phenotype randomization (does not apply to gene set permutations)'}, {'name': 'omit.features.with.no.symbol.match', 'value': 'true', 'label': 'Omit.Features.With.No.Symbol.Match', 'description': 'If there is no known gene symbol match for a probe set, omit it from the collapsed dataset'}, {'name': 'make.detailed.gene.set.report', 'value': 'true', 'label': 'Make.Detailed.Gene.Set.Report', 'description': 'Create detailed gene set report (heat map, mountain plot, etc.) for every enriched gene set'}, {'name': 'median.for.class.metrics', 'value': 'false', 'label': 'Median.For.Class.Metrics', 'description': 'Use the median of each class instead of the mean for the class separation metrics'}, {'name': 'number.of.markers', 'value': '100', 'label': 'Number.Of.Markers', 'description': 'Number of markers'}, {'name': 'plot.graphs.for.the.top.sets.of.each.phenotype', 'value': '20', 'label': 'Plot.Graphs.For.The.Top.Sets.Of.Each.Phenotype', 'description': 'Plot GSEA mountain and related plots for the top sets of each phenotype'}, {'name': 'random.seed', 'value': 'timestamp', 'label': 'Random.Seed', 'description': 'Seed to use for randomization'}, {'name': 'save.random.ranked.lists', 'value': 'false', 'label': 'Save.Random.Ranked.Lists', 'description': 'Whether to save random ranked lists'}, {'name': 'output.file.name', 'value': '<expression.dataset_basename>.zip', 'label': 'Output.File.Name', 'description': 'Name of the output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'GSEALeadingEdgeViewer': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Leading Edge Viewer for GSEA results', 'required_args': [{'name': 'zip.file', 'value': '', 'label': 'Zip.File', 'description': 'zip output file from GSEA - .zip. '}], 'default_args': [{'name': 'task_name', 'value': 'GSEALeadingEdgeViewer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'GSEAPreranked': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Runs the gene set enrichment analysis against a user-supplied ranked list of genes.', 'required_args': [{'name': 'ranked.list', 'value': '', 'label': 'Ranked.List', 'description': 'Rank ordered gene list - .rnk'}, {'name': 'gene.sets.database', 'value': '', 'label': 'Gene.Sets.Database', 'description': 'Gene sets database from MSigDB website.'}, {'name': 'gene.sets.database.file', 'value': '', 'label': 'Gene.Sets.Database.File', 'description': 'Gene sets database - .gmt, .gmx, .grp. Upload a gene set if your gene set is not listed as a choice for the gene sets database parameter.'}, {'name': 'chip.platform.file', 'value': '', 'label': 'Chip.Platform.File', 'description': 'DNA Chip (array) annotation file from GSEA website.  Upload your own chip file if the one corresponding to your DNA Microarray platform is not listed in the drop-down menu.  A chip file is only required if collapse dataset is set to true.'}], 'default_args': [{'name': 'task_name', 'value': 'GSEAPreranked', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'number.of.permutations', 'value': '1000', 'label': 'Number.Of.Permutations', 'description': 'Number of permutations to perform'}, {'name': 'collapse.dataset', 'value': 'false', 'label': 'Collapse.Dataset', 'description': 'Select true to have GSEAPreranked collapse each feature set in the ranked list into a single vector for the gene, which gets identified by its gene symbol.'}, {'name': 'scoring.scheme', 'value': 'weighted', 'label': 'Scoring.Scheme', 'description': 'The statistic used to score hits (gene set members) and misses (non-members)'}, {'name': 'max.gene.set.size', 'value': '500', 'label': 'Max.Gene.Set.Size', 'description': 'Gene sets larger than this are excluded from the analysis'}, {'name': 'min.gene.set.size', 'value': '15', 'label': 'Min.Gene.Set.Size', 'description': 'Gene sets smaller than this are excluded from the analysis'}, {'name': 'normalization.mode', 'value': 'meandiv', 'label': 'Normalization.Mode', 'description': 'Normalization mode'}, {'name': 'collapsing.mode', 'value': 'Max_probe', 'label': 'Collapsing.Mode', 'description': 'Collapsing mode'}, {'name': 'omit.features.with.no.symbol.match', 'value': 'true', 'label': 'Omit.Features.With.No.Symbol.Match', 'description': 'If there is no known gene symbol match for a probe set, omit it from the collapsed dataset; only relevant when collapse dataset is set to true'}, {'name': 'make.detailed.gene.set.report', 'value': 'true', 'label': 'Make.Detailed.Gene.Set.Report', 'description': 'Create detailed gene set report (heat map, mountain plot, etc.) for every enriched gene set'}, {'name': 'num.top.sets', 'value': '20', 'label': 'Num.Top.Sets', 'description': 'Plot GSEA mountain and related plots for the top sets of each phenotype'}, {'name': 'random.seed', 'value': 'timestamp', 'label': 'Random.Seed', 'description': 'Seed to use for generating random permutations.'}, {'name': 'output.file.name', 'value': '<ranked.list_basename>.zip', 'label': 'Output.File.Name', 'description': 'Name of the output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'HAPSEG': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '[**Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] A probabilistic method to interpret bi-allelic marker data in cancer samples.', 'required_args': [{'name': 'plate.name', 'value': '', 'label': 'Plate.Name', 'description': 'Name of the sample plate'}, {'name': 'array.name', 'value': '', 'label': 'Array.Name', 'description': 'Name of the chip that was run'}, {'name': 'seg.file', 'value': '', 'label': 'Seg.File', 'description': 'Segmented copy number data file for this sample. If not provided, HAPSEG will segment the data for you.'}, {'name': 'snp.file', 'value': '', 'label': 'Snp.File', 'description': 'SNP intensity file for this sample'}, {'name': 'use.pop', 'value': '', 'label': 'Use.Pop', 'description': 'Population to run with (currently supported values: CEPH, CH, JA, YOR)'}, {'name': 'calls.file', 'value': '', 'label': 'Calls.File', 'description': 'If using a matched normal sample, a Birdseed calls file needs to be supplied'}, {'name': 'mn.sample', 'value': '', 'label': 'Mn.Sample', 'description': 'If using a matched sample, the name of that matched normal sample'}, {'name': 'clusters.file', 'value': '', 'label': 'Clusters.File', 'description': \"If calibrate.data is 'On' the user must supply a Birdseed clusters file\"}, {'name': 'prev.theta.file', 'value': '', 'label': 'Prev.Theta.File', 'description': 'An optional file storing the previous theta values'}], 'default_args': [{'name': 'task_name', 'value': 'HAPSEG', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'out.file.name', 'value': '<plate.name>_<array.name>.segdat.RData', 'label': 'Out.File.Name', 'description': 'The name of the output file.'}, {'name': 'genome.build', 'value': 'ftp://gpftp.broadinstitute.org/module_support_files/HAPSEG_1.1.1/BEAGLE/phasedBGL/hg19/', 'label': 'Genome.Build', 'description': \"Which build of the human genome to use.  Only 'hg18' and 'hg19' are currently supported.  This will select a set of phased BEAGLE files.\"}, {'name': 'platform', 'value': 'SNP_6.0', 'label': 'Platform', 'description': \"The chip type used. The supported values are currently 'SNP_250K_STY' and 'SNP_6.0'\"}, {'name': 'impute.gt', 'value': 'TRUE', 'label': 'Impute.Gt', 'description': \"If TRUE will impute genotypes via BEAGLE. We recommend this be TRUE unless there's a reason to do otherwise\"}, {'name': 'plot.segfit', 'value': 'TRUE', 'label': 'Plot.Segfit', 'description': 'If TRUE will plot JPG images of the segmentations fits'}, {'name': 'merge.small', 'value': 'TRUE', 'label': 'Merge.Small', 'description': 'If TRUE will merge small segments'}, {'name': 'merge.close', 'value': 'TRUE', 'label': 'Merge.Close', 'description': 'if TRUE will merge close segments'}, {'name': 'min.seg.size', 'value': '10', 'label': 'Min.Seg.Size', 'description': 'Minimum segment size'}, {'name': 'normal', 'value': 'FALSE', 'label': 'Normal', 'description': 'If TRUE this sample will be treated as a normal'}, {'name': 'out.p', 'value': '0.05', 'label': 'Out.P', 'description': 'Outlier probability'}, {'name': 'seg.merge.thresh', 'value': '1e-10', 'label': 'Seg.Merge.Thresh', 'description': 'Segmentation threshold'}, {'name': 'use.normal', 'value': 'FALSE', 'label': 'Use.Normal', 'description': 'If TRUE will use a matched normal sample if one is provided'}, {'name': 'drop.x', 'value': 'FALSE', 'label': 'Drop.X', 'description': 'If TRUE will remove the X chromosome from the calculation'}, {'name': 'drop.y', 'value': 'TRUE', 'label': 'Drop.Y', 'description': 'If TRUE will remove the Y chromosome from the calculation'}, {'name': 'calibrate.data', 'value': 'NA', 'label': 'Calibrate.Data', 'description': \"If 'On', will perform a calibration on the input data.  If 'Off', no calibration will be performed. If left at the default value (Inferred), the calibration status will be inferred.\"}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'HeatMapImage': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Creates a heat map from a dataset', 'required_args': [{'name': 'input.dataset', 'value': '', 'label': 'Input.Dataset', 'description': 'The input dataset - .res, .gct, .odf'}, {'name': 'rows.to.highlight', 'value': '', 'label': 'Rows.To.Highlight', 'description': 'Rows to highlight in the heat map'}, {'name': 'row.highlight.color', 'value': '', 'label': 'Row.Highlight.Color', 'description': 'The color to highlight the rows if a rows to highlight file is provided'}, {'name': 'color.palette', 'value': '', 'label': 'Color.Palette', 'description': 'Color palette consisting of one r:g:b triplet or one hex value per line'}], 'default_args': [{'name': 'task_name', 'value': 'HeatMapImage', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output', 'value': '<input.dataset_basename>', 'label': 'Output', 'description': 'The output file name -.jpeg, .png, .tiff, .bmp, .eps'}, {'name': 'output.format', 'value': 'jpeg', 'label': 'Output.Format', 'description': 'The output file format'}, {'name': 'column.size', 'value': '8', 'label': 'Column.Size', 'description': 'The size in pixels of an element along the horizontal axis'}, {'name': 'row.size', 'value': '8', 'label': 'Row.Size', 'description': 'The size in pixels of an element along the vertical axis'}, {'name': 'show.grid', 'value': 'yes', 'label': 'Show.Grid', 'description': 'Whether to show a grid between elements'}, {'name': 'grid.color', 'value': 'black', 'label': 'Grid.Color', 'description': 'The grid color'}, {'name': 'show.row.descriptions', 'value': 'no', 'label': 'Show.Row.Descriptions', 'description': 'Whether to show row descriptions'}, {'name': 'show.row.names', 'value': 'yes', 'label': 'Show.Row.Names', 'description': 'Whether to show row names'}, {'name': 'color.scheme', 'value': 'row normalized', 'label': 'Color.Scheme', 'description': 'The color scheme'}, {'name': 'use.color.gradient', 'value': 'no', 'label': 'Use.Color.Gradient', 'description': 'Whether to use a linear color gradient or discrete colors'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'HeatMapViewer': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'A configurable heat map viewer that provides users with several options for manipulating and visualizing array-based data', 'required_args': [{'name': 'dataset', 'value': '', 'label': 'Dataset', 'description': 'The dataset to display - .gct'}], 'default_args': [{'name': 'task_name', 'value': 'HeatMapViewer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'helloWorld': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Generic hello world module', 'required_args': [], 'default_args': [{'name': 'task_name', 'value': 'helloWorld', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'message', 'value': 'Hello, World!', 'label': 'Message', 'description': ''}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'helloWorld_R2.15_Rscript': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Demo R2.15_Rscript module', 'required_args': [], 'default_args': [{'name': 'task_name', 'value': 'helloWorld_R2.15_Rscript', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'HierarchicalClustering': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Hierarchical Clustering', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'input data file name - .gct, .res, .pcl'}, {'name': 'log.transform', 'value': '', 'label': 'Log.Transform', 'description': 'log-transform the data before clustering'}, {'name': 'row.center', 'value': '', 'label': 'Row.Center', 'description': 'whether to center each row (gene) in the data'}, {'name': 'row.normalize', 'value': '', 'label': 'Row.Normalize', 'description': 'whether to normalize each row (gene) in the data'}, {'name': 'column.center', 'value': '', 'label': 'Column.Center', 'description': 'whether to center each column (sample) in the data'}, {'name': 'column.normalize', 'value': '', 'label': 'Column.Normalize', 'description': 'whether to normalize each column (sample) in the data'}], 'default_args': [{'name': 'task_name', 'value': 'HierarchicalClustering', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'column.distance.measure', 'value': '2', 'label': 'Column.Distance.Measure', 'description': 'distance measure for column (sample) clustering'}, {'name': 'row.distance.measure', 'value': '0', 'label': 'Row.Distance.Measure', 'description': 'distance measure for row (gene) clustering  NOTE: Filtering beforehand is recommended since row clustering is computationally intensive.'}, {'name': 'clustering.method', 'value': 'a', 'label': 'Clustering.Method', 'description': 'hierarchical clustering method to use'}, {'name': 'output.base.name', 'value': '<input.filename_basename>', 'label': 'Output.Base.Name', 'description': 'base name for output files'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'HierarchicalClusteringImage': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Creates an image of the dendrogram generated from HierarchicalClustering', 'required_args': [{'name': 'cdt', 'value': '', 'label': 'Cdt', 'description': 'Clustered data table, contains the original data, but reordered'}, {'name': 'gtr', 'value': '', 'label': 'Gtr', 'description': 'Gene Tree file, records the order in which genes (rows) were joined (at least one of gtr and atr files are required)'}, {'name': 'atr', 'value': '', 'label': 'Atr', 'description': 'Array Tree file, records the order in which samples (columns) were joined (at least one of gtr and atr files are required)'}, {'name': 'rows.to.highlight', 'value': '', 'label': 'Rows.To.Highlight', 'description': 'Rows to highlight in the heat map'}, {'name': 'row.highlight.color', 'value': '', 'label': 'Row.Highlight.Color', 'description': 'The color to highlight the rows if a rows to highlight file is provided'}, {'name': 'color.palette', 'value': '', 'label': 'Color.Palette', 'description': 'Color palette consisting of one r:g:b triplet or one hex value per line'}], 'default_args': [{'name': 'task_name', 'value': 'HierarchicalClusteringImage', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output', 'value': '<cdt_basename>', 'label': 'Output', 'description': 'The output file name - .bmp, .jpeg, .png, .tiff'}, {'name': 'output.format', 'value': 'jpeg', 'label': 'Output.Format', 'description': 'The output file format'}, {'name': 'column.size', 'value': '8', 'label': 'Column.Size', 'description': 'The size in pixels of an element along the horizontal axis'}, {'name': 'row.size', 'value': '8', 'label': 'Row.Size', 'description': 'The size in pixels of an element along the vertical axis'}, {'name': 'show.grid', 'value': 'yes', 'label': 'Show.Grid', 'description': 'Whether to show a grid between elements'}, {'name': 'grid.color', 'value': '0:0:0', 'label': 'Grid.Color', 'description': 'The grid color'}, {'name': 'show.row.descriptions', 'value': 'no', 'label': 'Show.Row.Descriptions', 'description': 'Whether to show row descriptions'}, {'name': 'show.row.names', 'value': 'yes', 'label': 'Show.Row.Names', 'description': 'Whether to show row names'}, {'name': 'color.scheme', 'value': 'row normalized', 'label': 'Color.Scheme', 'description': 'The color scheme'}, {'name': 'use.color.gradient', 'value': 'no', 'label': 'Use.Color.Gradient', 'description': 'Whether to use a linear color gradient or discrete colors'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'HierarchicalClusteringViewer': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Views results of hierarchical clustering.', 'required_args': [{'name': 'cdt.file', 'value': '', 'label': 'Cdt.File', 'description': 'Clustered data table, contains the original data, but reordered'}, {'name': 'gtr.file', 'value': '', 'label': 'Gtr.File', 'description': 'Gene tree file, records the order in which genes (rows) were joined (at least one of gtr file and atr file is required)'}, {'name': 'atr.file', 'value': '', 'label': 'Atr.File', 'description': 'Array tree file, records the order in which samples (columns) were joined (at least one of gtr file and atr file is required)'}], 'default_args': [{'name': 'task_name', 'value': 'HierarchicalClusteringViewer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Hu68kHu35kAtoU95': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Converts a list of Affymetrix Hu6800/Hu35KsubA probes to the corresponding Affymetrix U95Av2 probes.', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'the .pol, .tag or .grp file to be converted'}], 'default_args': [{'name': 'task_name', 'value': 'Hu68kHu35kAtoU95', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output', 'value': '<input.filename_basename>.U95', 'label': 'Output', 'description': 'name of the output file (same format as input file)'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Identify.an.up.or.downregulated.pathway.from.expression.data': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'This pipeline is used in the GenomeSpace recipe, \"Identify an up- or down-regulated pathway from expression data\". The recipe can be found at: http://www.genomespace.org/support/guides/recipes/sections/identify-an-up-or-down-regulated-pathway-from-expression-data.', 'required_args': [{'name': 'ConvertLineEndings1.input.filename', 'value': '', 'label': 'ConvertLineEndings1.Input.Filename', 'description': 'The input file (any non-binary file format)'}, {'name': 'ComparativeMarkerSelection2.cls.file', 'value': '', 'label': 'ComparativeMarkerSelection2.Cls.File', 'description': 'The class file - .cls'}], 'default_args': [{'name': 'task_name', 'value': 'Identify.an.up.or.downregulated.pathway.from.expression.data', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Identify.and.Validate.a.Consensus.Signature.Using.Gene.Expression.Data': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'This pipeline is used in the GenomeSpace recipe, \"Identify and Validate a Consensus Signature Using Gene Expression Data\". The recipe can be found at: http://genomespace.org/support/guides/recipes/sections/identify-and-validate-a-consensus-signature-using-gene-expression-data.', 'required_args': [{'name': 'SelectFeaturesColumns1.input.filename', 'value': '', 'label': 'SelectFeaturesColumns1.Input.Filename', 'description': 'input file - .res, .gct, .snp, .cn, .loh, .odf, or .cls file from which the slice is extracted'}, {'name': 'SelectFeaturesColumns2.input.filename', 'value': '', 'label': 'SelectFeaturesColumns2.Input.Filename', 'description': 'input file - .res, .gct, .snp, .cn, .loh, .odf, or .cls file from which the slice is extracted'}, {'name': 'SelectFeaturesRows3.list.filename', 'value': '', 'label': 'SelectFeaturesRows3.List.Filename', 'description': 'text file containing one identifier per line that specifies the rows to select. '}], 'default_args': [{'name': 'task_name', 'value': 'Identify.and.Validate.a.Consensus.Signature.Using.Gene.Expression.Data', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Identify.and.visualize.expressed.transcripts.in.RNASeq.data': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'This pipeline is used in the GenomeSpace recipe, \"Identify and visualize expressed transcripts in RNA-Seq data\". The recipe can be found at: http://www.genomespace.org/support/guides/recipes/sections/identify-and-visualize-expressed-transcripts-in-rna-seq-data.', 'required_args': [{'name': 'ConvertLineEndings1.input.filename', 'value': '', 'label': 'ConvertLineEndings1.Input.Filename', 'description': 'The input file (any non-binary file format)'}, {'name': 'TopHat2.reads.pair.1', 'value': '', 'label': 'TopHat2.Reads.Pair.1', 'description': 'Unpaired reads file or first mate for paired reads. One or more files containing reads in FASTA or FASTQ format (bz2 and gz compressed files are supported).'}, {'name': 'TopHat2.reads.pair.2', 'value': '', 'label': 'TopHat2.Reads.Pair.2', 'description': 'Second mate for paired reads. Zero or more files in FASTA or FASTQ format (bz2 and gz compressed files are supported).'}], 'default_args': [{'name': 'task_name', 'value': 'Identify.and.visualize.expressed.transcripts.in.RNASeq.data', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'IGV': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Launches IGV with input file(s)', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'Dataset to be viewed in IGV or IGV session file'}, {'name': 'index.file', 'value': '', 'label': 'Index.File', 'description': 'The index file.'}, {'name': 'genomeId', 'value': '', 'label': 'GenomeId', 'description': 'Choose the genome for your data. IGV will launch with hg19 or the last viewed genome by default.'}, {'name': 'locus', 'value': '', 'label': 'Locus', 'description': 'Provide a locus or gene of interest for your data. IGV will launch on chr1 or the last viewed chromosome by default. (e.g. chr1:2000-3000 or egfr)'}], 'default_args': [{'name': 'task_name', 'value': 'IGV', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'IGV.js': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Launches IGV.js with input file(s)', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'Track dataset to be viewed in IGV'}, {'name': 'index.file', 'value': '', 'label': 'Index.File', 'description': 'The index file (i.e .bai, .idx, or .tbi).'}, {'name': 'genome', 'value': '', 'label': 'Genome', 'description': 'Choose the genome for your data. IGV will launch with hg19.'}, {'name': 'locus', 'value': '', 'label': 'Locus', 'description': 'Provide a locus or gene of interest for your data. IGV will launch on chr1.'}], 'default_args': [{'name': 'task_name', 'value': 'IGV.js', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'IlluminaConcatenator': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Concatenate normalized Illumina probe sets into a single GCT file', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'ZIP file of normalized GCT files output by IlluminaNormalizer'}, {'name': 'output.file.type', 'value': '', 'label': 'Output.File.Type', 'description': 'Whether to concatenate the data or detection scores'}], 'default_args': [{'name': 'task_name', 'value': 'IlluminaConcatenator', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'IlluminaDASLPipeline': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'creates a GenePattern gct file from raw Illumina scan data', 'required_args': [{'name': 'IlluminaScanExtractor1.sample.description.file', 'value': '', 'label': 'IlluminaScanExtractor1.Sample.Description.File', 'description': 'file with one sample per line, one Sentrix Position, DAP#, and Sentrix ID per DAP'}, {'name': 'IlluminaConcatenator3.output.file.type', 'value': '', 'label': 'IlluminaConcatenator3.Output.File.Type', 'description': 'Whether to concatenate the data or detection scores'}], 'default_args': [{'name': 'task_name', 'value': 'IlluminaDASLPipeline', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'IlluminaScanExtractor1.outlier.method', 'value': '> 3 Std. Deviations', 'label': 'IlluminaScanExtractor1.Outlier.Method', 'description': 'outlier detection method (only applies to extraction using bead-level data)'}, {'name': 'IlluminaNormalizer2.normalization.method', 'value': 'CubicSpline', 'label': 'IlluminaNormalizer2.Normalization.Method', 'description': 'correction method. Note that mean background subtraction does not apply when the input is a single GCT file.'}, {'name': 'IlluminaNormalizer2.discard.controls', 'value': 'no', 'label': 'IlluminaNormalizer2.Discard.Controls', 'description': 'discard top 2 and bottom 2 negative controls (applies to mean background subtraction only)'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'IlluminaExpressionFileCreator': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Creates a GCT file from a zip of Illumina IDAT files and an Illumina manifest file', 'required_args': [{'name': 'idat.zip', 'value': '', 'label': 'Idat.Zip', 'description': 'Zip of Illumina IDAT files'}, {'name': 'manifest', 'value': '', 'label': 'Manifest', 'description': 'Illumina manifest file in tab-delimited (TXT) format. A manifest file is required; either selected from this list or specified in custom manifest below.'}, {'name': 'custom.manifest', 'value': '', 'label': 'Custom.Manifest', 'description': 'Other Illumina manifest file in tab-delimited (TXT) format. A manifest file is required; either selected from the list above or specified here.'}, {'name': 'chip', 'value': '', 'label': 'Chip', 'description': 'CHIP file to map probes to genes.'}, {'name': 'custom.chip', 'value': '', 'label': 'Custom.Chip', 'description': 'Other CHIP file to map probes to genes.'}, {'name': 'clm', 'value': '', 'label': 'Clm', 'description': 'CLM file to map file names to sample names.'}], 'default_args': [{'name': 'task_name', 'value': 'IlluminaExpressionFileCreator', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<idat.zip_basename>.gct', 'label': 'Output.File', 'description': 'Output file name'}, {'name': 'background.subtraction.mode', 'value': 'false', 'label': 'Background.Subtraction.Mode', 'description': 'Perform background subtraction.'}, {'name': 'collapse.mode', 'value': 'none', 'label': 'Collapse.Mode', 'description': 'Collapse probes to genes based on the manifest or CHIP file (if provided).'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'IlluminaNormalizer': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Normalize zipped Illumina scans', 'required_args': [{'name': 'zip.or.gct.file', 'value': '', 'label': 'Zip.Or.Gct.File', 'description': 'a zip of GCT files output by IlluminaScanExtractor (6k DASL) or a GCT file (whole-genome DASL)'}], 'default_args': [{'name': 'task_name', 'value': 'IlluminaNormalizer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'normalization.method', 'value': 'CubicSpline', 'label': 'Normalization.Method', 'description': 'correction method. Note that mean background subtraction does not apply when the input is a single GCT file.'}, {'name': 'output.file', 'value': '<zip.or.gct.file_basename>_<normalization.method>', 'label': 'Output.File', 'description': 'output file name'}, {'name': 'discard.controls', 'value': 'no', 'label': 'Discard.Controls', 'description': 'discard top 2 and bottom 2 negative controls (applies to mean background subtraction only)'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'IlluminaScanExtractor': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Extract intensity values from Illumina scans', 'required_args': [{'name': 'sample.description.file', 'value': '', 'label': 'Sample.Description.File', 'description': 'file with one sample per line, one Sentrix Position, DAP#, and Sentrix ID per DAP'}], 'default_args': [{'name': 'task_name', 'value': 'IlluminaScanExtractor', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'outlier.method', 'value': '> 3 Std. Deviations', 'label': 'Outlier.Method', 'description': 'outlier detection method (only applies to extraction using bead-level data)'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ImmPort_FLOCK_CrossSample': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'ImmPort FLOCK and CrossSample pipeline', 'required_args': [{'name': 'ImmPortFCSConvLogicleTrans1.Input.FCS.file', 'value': '', 'label': 'ImmPortFCSConvLogicleTrans1.Input.FCS.File', 'description': 'An FCS file that shall be transformed with the Logicle transformation, converted and saved as tab-delimited text file.'}, {'name': 'ImmPortColSelection2.Dimension.specification', 'value': '', 'label': 'ImmPortColSelection2.Dimension.Specification', 'description': 'A list of dimensions/columns to extract, specified by their indexes (starting from 1), e.g., 1,4,6. There should not be any white space characters (e.g, spaces, tabs, new lines, etc.) in the parameter value.'}, {'name': 'ImmPortCrossSample4.Input.file', 'value': '', 'label': 'ImmPortCrossSample4.Input.File', 'description': 'A tab-delimited text file with columns corresponding to parameters and rows to events. Such a text file may be generated from an FCS file by the ImmPortFCSConvLogicleTrans module.'}], 'default_args': [{'name': 'task_name', 'value': 'ImmPort_FLOCK_CrossSample', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ImmPort_FLOCK_Individual_FCS': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'ImmPort FLOCK and Individual FCS pipeline', 'required_args': [{'name': 'ImmPortFCSConvLogicleTrans1.Input.FCS.file', 'value': '', 'label': 'ImmPortFCSConvLogicleTrans1.Input.FCS.File', 'description': 'An FCS file that shall be transformed with the Logicle transformation, converted and saved as tab-delimited text file.'}, {'name': 'ImmPortColSelection2.Dimension.specification', 'value': '', 'label': 'ImmPortColSelection2.Dimension.Specification', 'description': 'A list of dimensions/columns to extract, specified by their indexes (starting from 1), e.g., 1,4,6. There should not be any white space characters (e.g, spaces, tabs, new lines, etc.) in the parameter value.'}], 'default_args': [{'name': 'task_name', 'value': 'ImmPort_FLOCK_Individual_FCS', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ImmPortColSelection': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Select a set of columns from a txt file', 'required_args': [{'name': 'Input.file', 'value': '', 'label': 'Input.File', 'description': 'A tab-delimited text file with columns corresponding to parameters and rows to events, converted from an fcs file, typically by ImmPortFCSConvLogicleTrans.'}, {'name': 'Dimension.specification', 'value': '', 'label': 'Dimension.Specification', 'description': 'A list of dimensions/columns to extract, specified by their indexes (starting from 1), e.g., 1,4,6. There should not be any white space characters (e.g, spaces, tabs, new lines, etc.) in the parameter value.'}], 'default_args': [{'name': 'task_name', 'value': 'ImmPortColSelection', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ImmPortCrossSample': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Applying an individual FLOCK result across another FCM sample (.txt) version 1 (1-round Kmeans)', 'required_args': [{'name': 'Input.file', 'value': '', 'label': 'Input.File', 'description': 'A tab-delimited text file with columns corresponding to parameters and rows to events. Such a text file may be generated from an FCS file by the ImmPortFCSConvLogicleTrans module.'}, {'name': 'Centroid.file', 'value': '', 'label': 'Centroid.File', 'description': 'A text-file specifying the position of each of the cell populations. Such a text file is supposed to be generated by the ImmPortFLOCK module (file population_center.txt).'}], 'default_args': [{'name': 'task_name', 'value': 'ImmPortCrossSample', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Number.of.iterations', 'value': '1', 'label': 'Number.Of.Iterations', 'description': 'The number of iterations that the algorithm will use to match events with the closest population; values from the [1, 10] range are allowed.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ImmPortCSV2TXT': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Convert a .csv file to a .txt file that FLOCK can read', 'required_args': [{'name': 'Input.CSV.data.file', 'value': '', 'label': 'Input.CSV.Data.File', 'description': 'The input CSV file that will be converted to a a tab-delimited text file; the CSV shall contain a header with parameter names.'}], 'default_args': [{'name': 'task_name', 'value': 'ImmPortCSV2TXT', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ImmPortFCSConvLogicleTrans': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Logicle transformation of an fcs file to .txt file', 'required_args': [{'name': 'Input.FCS.file', 'value': '', 'label': 'Input.FCS.File', 'description': 'An FCS file that shall be transformed with the Logicle transformation, converted and saved as tab-delimited text file.'}], 'default_args': [{'name': 'task_name', 'value': 'ImmPortFCSConvLogicleTrans', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ImmPortFLOCK': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Automated Gating with ImmPort FLOCK algorithm version 1.0', 'required_args': [{'name': 'Input.file', 'value': '', 'label': 'Input.File', 'description': 'A tab-delimited text file with columns corresponding to parameters and rows to events. The first row is a heading labelling parameters. Such a text file is supposed to be generated from an FCS file by the ImmPortFCSConvLogicleTrans module optionally followed by the ImmPortColSelection module (if applicable to remove unused parameters). Alternatively, this text file may be created by the ImmPortCSV2TXT module if the original data is in the CSV format.'}], 'default_args': [{'name': 'task_name', 'value': 'ImmPortFLOCK', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Number.of.bins', 'value': '0', 'label': 'Number.Of.Bins', 'description': 'The number of bins used to partition each dimension. The default value of 0 means that the algorithm will automatically determine a suitable number of bins. Any value from the [6,29] range may be provided to manually override algorithmic estimations.'}, {'name': 'Density.threshold', 'value': '0', 'label': 'Density.Threshold', 'description': 'The minimal number of events in a bin (in the hyperspace of used markers) needed to mark the bin as a dense bin. The default value of 0 means that the algorithm will automatically determine a suitable density threshold. Any value from the [3,99] range may be provided to manually override algorithmic estimations.'}, {'name': 'Maximum.number.of.populations', 'value': '30', 'label': 'Maximum.Number.Of.Populations', 'description': 'The maximum number of populations identifiable by the clustering algorithm. The default value of 30 works well in most cases but may be overridden by any value from the [5,999] range. Larger values may increase the execution time of the module.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ImputeMissingValues.KNN': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Imputes missing values', 'required_args': [{'name': 'data.filename', 'value': '', 'label': 'Data.Filename', 'description': 'Data file (missing values are stored as NA) - .gct'}], 'default_args': [{'name': 'task_name', 'value': 'ImputeMissingValues.KNN', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'k', 'value': '10', 'label': 'K', 'description': 'Number of neighbors to be used in the imputation'}, {'name': 'rowmax', 'value': '0.5', 'label': 'Rowmax', 'description': 'The maximum percent missing data allowed in any row'}, {'name': 'colmax', 'value': '0.8', 'label': 'Colmax', 'description': 'The maximum percent missing data allowed in any column'}, {'name': 'output.file', 'value': '<data.filename_basename>.imputed.gct', 'label': 'Output.File', 'description': 'The name of the output file - .gct'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Install.Proteogenomics.R.Packages': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Installs all packages and dependencies need to run the Protegenomics GenePattern modules', 'required_args': [], 'default_args': [{'name': 'task_name', 'value': 'Install.Proteogenomics.R.Packages', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'IntersectingIdentifiers': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Given 2 tab delimited files (res, gct, tab, odf) caculate which ids in the first intersect with the second. Outputs a file of intersecting, non-intersecting, in the first but not the second and in the second but not the first', 'required_args': [{'name': 'First.file', 'value': '', 'label': 'First.File', 'description': 'Any tab delimited file'}, {'name': 'Second.file', 'value': '', 'label': 'Second.File', 'description': 'Any tab delimited file'}], 'default_args': [{'name': 'task_name', 'value': 'IntersectingIdentifiers', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'First.file.identifier.column.index', 'value': '1', 'label': 'First.File.Identifier.Column.Index', 'description': 'Index of identifiers in first file, staring at 1'}, {'name': 'Second.file.identifier.column.index', 'value': '1', 'label': 'Second.File.Identifier.Column.Index', 'description': 'Index of identifiers in second file, staring at 1'}, {'name': 'Output.file.name.prefix', 'value': '<First.file_basename>', 'label': 'Output.File.Name.Prefix', 'description': 'Name of output file'}, {'name': 'First.file.skip.rows', 'value': '2', 'label': 'First.File.Skip.Rows', 'description': 'Number of rows to skip in the first file to avoid headers'}, {'name': 'Second.file.skip.rows', 'value': '2', 'label': 'Second.File.Skip.Rows', 'description': 'Number of rows to skip in the first file to avoid headers'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'JavaTreeView': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': \"Hierarchical clustering viewer that reads in Eisen's cdt, atr, and gtr files. A detailed description of these formats is available from http://microarray.ccgb.umn.edu/smd/html/MicroArray/help/formats.shtml\", 'required_args': [{'name': 'cdt.filename', 'value': '', 'label': 'Cdt.Filename', 'description': 'Clustered data table, contains the original data, but reordered'}, {'name': 'gtr.filename', 'value': '', 'label': 'Gtr.Filename', 'description': 'Gene Tree file, records the order in which genes (rows) were joined (at least one of gtr filename and atr filename is required)'}, {'name': 'atr.filename', 'value': '', 'label': 'Atr.Filename', 'description': 'Array Tree file, records the order in which samples (columns) were joined (at least one of gtr filename and atr filename is required)'}], 'default_args': [{'name': 'task_name', 'value': 'JavaTreeView', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'KMeansClassifyFCS': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Find clusters in a flow cytometry file with k means clustering', 'required_args': [{'name': 'Input.FCS.data.file', 'value': '', 'label': 'Input.FCS.Data.File', 'description': 'The FCS file'}, {'name': 'K', 'value': '', 'label': 'K', 'description': 'The number of populations you are looking for.'}, {'name': 'Dimensions', 'value': '', 'label': 'Dimensions', 'description': 'The channels to search for populations, separated by commas. Leave blank to search in all channels. E.g. FSC Lin, SS Lin, CD4 Log'}], 'default_args': [{'name': 'task_name', 'value': 'KMeansClassifyFCS', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'CSV.file.name', 'value': '<Input.FCS.data.file_basename>', 'label': 'CSV.File.Name', 'description': 'The base filename to split the results into csv files, each containing one population'}, {'name': 'CLS.file.name', 'value': '<Input.FCS.data.file_basename>', 'label': 'CLS.File.Name', 'description': 'A .CLS filename to store the population number for each data point'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'KMeansClustering': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Module that performs the K-Means Clustering algorithm', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'File containing data to cluster - .res, .gct, .odf'}, {'name': 'number.of.clusters', 'value': '', 'label': 'Number.Of.Clusters', 'description': 'Number of centroids (clusters)'}, {'name': 'cluster.by', 'value': '', 'label': 'Cluster.By', 'description': 'Whether to cluster by rows or columns'}], 'default_args': [{'name': 'task_name', 'value': 'KMeansClustering', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.base.name', 'value': '<input.filename_basename>_KMcluster_output', 'label': 'Output.Base.Name', 'description': 'The base output file name - .gct'}, {'name': 'seed.value', 'value': '12345', 'label': 'Seed.Value', 'description': 'Seed value for random number generator'}, {'name': 'distance.metric', 'value': '0', 'label': 'Distance.Metric', 'description': 'How to compute distance between points'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'KNN': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'K-nearest neighbors classification', 'required_args': [{'name': 'train.filename', 'value': '', 'label': 'Train.Filename', 'description': 'training data file name - .gct, .res, .odf type = Dataset, ignored if a saved model (saved.model.filename) is used'}, {'name': 'train.class.filename', 'value': '', 'label': 'Train.Class.Filename', 'description': 'class file for training data - .cls, ignored if a saved model (saved.model.filename) is used'}, {'name': 'saved.model.filename', 'value': '', 'label': 'Saved.Model.Filename', 'description': 'input KNN model file - .odf type = KNN Prediction Model'}, {'name': 'test.filename', 'value': '', 'label': 'Test.Filename', 'description': 'test data file name - .gct, .res, .odf type = Dataset'}, {'name': 'test.class.filename', 'value': '', 'label': 'Test.Class.Filename', 'description': 'class file for test data - .cls'}, {'name': 'min.std', 'value': '', 'label': 'Min.Std', 'description': 'minimum standard deviation if feature selection statistic includes min std option'}, {'name': 'feature.list.filename', 'value': '', 'label': 'Feature.List.Filename', 'description': 'features to use for prediction'}], 'default_args': [{'name': 'task_name', 'value': 'KNN', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'model.file', 'value': '<train.filename_basename>.model', 'label': 'Model.File', 'description': 'name of output KNN model file  - .odf type = KNN Prediction Model'}, {'name': 'num.features', 'value': '10', 'label': 'Num.Features', 'description': 'number of selected features if feature list filename is not specified'}, {'name': 'feature.selection.statistic', 'value': '0', 'label': 'Feature.Selection.Statistic', 'description': 'statistic to use to perform feature selection'}, {'name': 'num.neighbors', 'value': '3', 'label': 'Num.Neighbors', 'description': 'number of neighbors for KNN'}, {'name': 'weighting.type', 'value': 'distance', 'label': 'Weighting.Type', 'description': 'weighting type for neighbors'}, {'name': 'distance.measure', 'value': '1', 'label': 'Distance.Measure', 'description': 'distance measure'}, {'name': 'pred.results.file', 'value': '<test.filename_basename>.pred.odf', 'label': 'Pred.Results.File', 'description': 'name of output file for prediction results - .odf'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'KNNXValidation': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'KNN classification with Leave-One-Out Cross-Validation', 'required_args': [{'name': 'data.filename', 'value': '', 'label': 'Data.Filename', 'description': 'data file - .gct, .res, .odf type = Dataset'}, {'name': 'class.filename', 'value': '', 'label': 'Class.Filename', 'description': 'class file - .cls'}, {'name': 'min.std', 'value': '', 'label': 'Min.Std', 'description': 'minimum standard deviation if feature selection statistic includes min std option'}], 'default_args': [{'name': 'task_name', 'value': 'KNNXValidation', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'num.features', 'value': '10', 'label': 'Num.Features', 'description': 'number of selected features'}, {'name': 'feature.selection.statistic', 'value': '0', 'label': 'Feature.Selection.Statistic', 'description': 'statistic to use to perform feature selection'}, {'name': 'num.neighbors', 'value': '3', 'label': 'Num.Neighbors', 'description': 'number of neighbors for KNN'}, {'name': 'weighting.type', 'value': 'distance', 'label': 'Weighting.Type', 'description': 'weighting type for neighbors'}, {'name': 'distance.measure', 'value': '1', 'label': 'Distance.Measure', 'description': 'distance measure'}, {'name': 'pred.results.file', 'value': '<data.filename_basename>.pred.odf', 'label': 'Pred.Results.File', 'description': 'prediction results output file name .odf type = Prediction Results'}, {'name': 'feature.summary.file', 'value': '<data.filename_basename>.feat.odf', 'label': 'Feature.Summary.File', 'description': 'feature summary output file name - .odf type = Prediction Features'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'KSscore': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Computes the Kolmogorov-Smirnov score for a set of genes within an ordered list.', 'required_args': [{'name': 'query.filename', 'value': '', 'label': 'Query.Filename', 'description': 'query gene set file containing one identifier per line'}, {'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'file containing an ordered list of genes (.pol format)'}], 'default_args': [{'name': 'task_name', 'value': 'KSscore', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output', 'value': '<query.filename_basename>', 'label': 'Output', 'description': 'output file name (a .pol extension will be appended)'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'LandmarkMatch': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'A proteomics method to propogate identifed peptides across multiple MS runs', 'required_args': [{'name': 'peakList.filename', 'value': '', 'label': 'PeakList.Filename', 'description': 'Zip file containing peak lists'}, {'name': 'retentionTime.filename', 'value': '', 'label': 'RetentionTime.Filename', 'description': 'Zip file containing retention time (in this case from mapquant)'}, {'name': 'sampleInfo.filename', 'value': '', 'label': 'SampleInfo.Filename', 'description': 'CSV file with header: experiment, sample, class'}, {'name': 'globalID.filename', 'value': '', 'label': 'GlobalID.Filename', 'description': 'Global list of identified peptides'}, {'name': 'accurateMass.filename', 'value': '', 'label': 'AccurateMass.Filename', 'description': 'OPTIONAL: Specify the accurate mass table'}, {'name': 'prefitCoef.filename', 'value': '', 'label': 'PrefitCoef.Filename', 'description': 'OPTIONAL: Specify the prefit coefficients in a file'}, {'name': 'Bootstrap', 'value': '', 'label': 'Bootstrap', 'description': 'Use bootstrapping to calculate landmark match statistics'}], 'default_args': [{'name': 'task_name', 'value': 'LandmarkMatch', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'running.on.LSF', 'value': 'no', 'label': 'Running.On.LSF', 'description': 'Whether you are running on an LSF cluster version of GenePattern'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Life.Cycle.Time.Point.Correlation': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Calculate correlation to life cycle time point of p.falciparum (derisi data)', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'Input data file name. Data in CSV format, with plasmodbID as first column '}], 'default_args': [{'name': 'task_name', 'value': 'Life.Cycle.Time.Point.Correlation', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<input.file_basename>-correlation.csv', 'label': 'Output.File', 'description': 'Name of output file that will contain the resulting correlation table. CSV format.'}, {'name': 'correlation', 'value': 'pearson', 'label': 'Correlation', 'description': 'Type of correlation to calculate'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ListFiles': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': \"a helper module which outputs a filelist, the list of all files in the input directory, similar to the unix 'ls' command\", 'required_args': [{'name': 'dir', 'value': '', 'label': 'Dir', 'description': 'a directory'}, {'name': 'recursive', 'value': '', 'label': 'Recursive', 'description': ''}, {'name': 'includePattern', 'value': '', 'label': 'IncludePattern', 'description': 'an optional list of filename patterns, e.g. *.cls,*.gct'}, {'name': 'pruneDir', 'value': '', 'label': 'PruneDir', 'description': 'an optional list of directory name patterns to ignore, e.g. *.svn'}, {'name': 'order.by', 'value': '', 'label': 'Order.By', 'description': 'an optional flag for how to order the files'}, {'name': 'type', 'value': '', 'label': 'Type', 'description': ''}, {'name': 'output.filename', 'value': '', 'label': 'Output.Filename', 'description': 'the name of the output file, if none specified, write to stdout'}], 'default_args': [{'name': 'task_name', 'value': 'ListFiles', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'LocatePeaks': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Locates detected peaks in a spectrum.', 'required_args': [{'name': 'peaks.output.filename', 'value': '', 'label': 'Peaks.Output.Filename', 'description': 'The output from the Peaks module'}, {'name': 'locate.peaks.as', 'value': '', 'label': 'Locate.Peaks.As', 'description': 'max-location of maximum filtered value, area-center of area of the peak'}], 'default_args': [{'name': 'task_name', 'value': 'LocatePeaks', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<peaks.output.filename_basename>.locate.odf', 'label': 'Output.File', 'description': 'output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'LogicleTransformFCS': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Logicle transformation of (selected) parameters in a list mode FCS data file.', 'required_args': [{'name': 'Input.fcs.data.file', 'value': '', 'label': 'Input.Fcs.Data.File', 'description': 'FCS data file to transform.'}, {'name': 'Parameters.to.transform', 'value': '', 'label': 'Parameters.To.Transform', 'description': 'Comma-separated list of parameters to transform, parameters are specified by their short names ($PnN values); Optional. By default, all parameters with the \"FL\" prefix in their short name will be transformed.'}, {'name': 'T', 'value': '', 'label': 'T', 'description': 'T > 0; Data value at the top of the scale range to be displayed, optional. The maximum input value will be used if no explicit value for T is provided.'}, {'name': 'W', 'value': '', 'label': 'W', 'description': 'W >= 0; Linearization width in asymptotic decades; optional. The described calculation will be used to set W for each of the transformed FCS parameters if no explicit value for W is provided.'}], 'default_args': [{'name': 'task_name', 'value': 'LogicleTransformFCS', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'M', 'value': '4.5', 'label': 'M', 'description': 'M < 0; Full width of the intended Logicle display in asymptotic decades, optional. The default value is 4.5 (decades). Note also that for the resulting Logicle transformation, Logicle (T) = M.'}, {'name': 'A', 'value': '0', 'label': 'A', 'description': 'A, usually <= 0; Additional negative display range in asymptotic decades; optional. The default value is 0.'}, {'name': 'Use.Fast.Implementation', 'value': 'no', 'label': 'Use.Fast.Implementation', 'description': 'Use a fast implementation of Logicle with less precision; optional.'}, {'name': 'Output.fcs.file.name', 'value': '<Input.fcs.data.file_basename>.logicle_transformed.fcs', 'label': 'Output.Fcs.File.Name', 'description': 'The output FCS file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'LogisticRegression': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Logistic regression using clinical data and .cls file', 'required_args': [{'name': 'input.clin.data.filename', 'value': '', 'label': 'Input.Clin.Data.Filename', 'description': 'input clinical data (.txt file)'}, {'name': 'input.cls.filename', 'value': '', 'label': 'Input.Cls.Filename', 'description': 'input class information (.cls file)'}], 'default_args': [{'name': 'task_name', 'value': 'LogisticRegression', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': 'LogisticRegression_result', 'label': 'Output.File', 'description': 'output file name'}, {'name': 'response.variable', 'value': 'NA', 'label': 'Response.Variable', 'description': ''}, {'name': 'variable.continuous', 'value': 'NA', 'label': 'Variable.Continuous', 'description': 'variable.continuous'}, {'name': 'variable.category', 'value': 'NA', 'label': 'Variable.Category', 'description': '(e.g. sex,cls,grade,...)'}, {'name': 'variable.interaction.terms', 'value': 'NA', 'label': 'Variable.Interaction.Terms', 'description': '(e.g. sex:cls,sex:grade,...)'}, {'name': 'input.subgroup', 'value': 'NA', 'label': 'Input.Subgroup', 'description': '(e.g. cls:2) *only one subgroup can be specified'}, {'name': 'variable.selection', 'value': 'none', 'label': 'Variable.Selection', 'description': ''}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'LogTransformGLAD': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Log base 2 transform last column of GLAD file (for GISTIC)', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'The file to transform the last column of (glad format)'}], 'default_args': [{'name': 'task_name', 'value': 'LogTransformGLAD', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<input.filename_basename>.LogTransformed.glad', 'label': 'Output.File', 'description': 'Output filename'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'LOHPaired': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Computes LOH for paired samples', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'A .snp file that contains genotypes'}, {'name': 'sample.info.filename', 'value': '', 'label': 'Sample.Info.Filename', 'description': 'A sample info file with the paired information'}], 'default_args': [{'name': 'task_name', 'value': 'LOHPaired', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<input.filename_basename>.loh', 'label': 'Output.File', 'description': 'Output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Lu.Getz.Miska.Nature.June.2005.mouse.lung.pipeline': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Normal/tumor classifier and kNN prediction of mouse lung samples', 'required_args': [], 'default_args': [{'name': 'task_name', 'value': 'Lu.Getz.Miska.Nature.June.2005.mouse.lung.pipeline', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'MAGEMLImportViewer': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'A visualizer to import data in MAGE-ML format into GenePattern', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'The archive file containing the MAGE-ML files - .tar, .tgz, .tar.gz, .zip, .gz'}], 'default_args': [{'name': 'task_name', 'value': 'MAGEMLImportViewer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'MAGETABImportViewer': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'A visualizer to import data in MAGE-TAB format into GenePattern', 'required_args': [{'name': 'sdrf.file', 'value': '', 'label': 'Sdrf.File', 'description': 'The sdrf file - .txt'}, {'name': 'data.files', 'value': '', 'label': 'Data.Files', 'description': 'A zip file containing MAGE-TAB data matrix files - .zip'}], 'default_args': [{'name': 'task_name', 'value': 'MAGETABImportViewer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'MapChipFeaturesGeneral': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Change (map) the features (genes) of a dataset', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'Input filename - .res'}, {'name': 'lookup.table.filename', 'value': '', 'label': 'Lookup.Table.Filename', 'description': 'Name of the look up table file with the feature mapping'}, {'name': 'table.format', 'value': '', 'label': 'Table.Format', 'description': 'Table format'}, {'name': 'direction', 'value': '', 'label': 'Direction', 'description': 'Direction of look up'}], 'default_args': [{'name': 'task_name', 'value': 'MapChipFeaturesGeneral', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.prefix', 'value': '<input.filename_basename>.fea', 'label': 'Output.Prefix', 'description': 'prefix for output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Marker.Selection': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Marker selection, ranking and classification', 'required_args': [{'name': 'gct.file', 'value': '', 'label': 'Gct.File', 'description': 'Input data file (in gct format)'}, {'name': 'cls.file', 'value': '', 'label': 'Cls.File', 'description': 'Class file (cls format) to match with input data file'}], 'default_args': [{'name': 'task_name', 'value': 'Marker.Selection', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.prefix', 'value': '<gct.file_basename>', 'label': 'Output.Prefix', 'description': 'Prefix for output file names'}, {'name': 'fdr.threshold', 'value': '0.01', 'label': 'Fdr.Threshold', 'description': 'FDR threshold for calling statistically significant markers'}, {'name': 'gct.id.col', 'value': '2', 'label': 'Gct.Id.Col', 'description': 'Column in data file containing (gene or protein) ID'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'MClustClusterLabel': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Model based labeling/matching of clusters (mate-clustering)', 'required_args': [{'name': 'Input.clustering.results', 'value': '', 'label': 'Input.Clustering.Results', 'description': 'The input clustering results (of multiple data files) in CSV files with each clustered population stored in a separate CSV file. All these CSV files shall have the same columns (dimensions/channels) or at least, they all shall include columns specified by the dimensions parameter.'}, {'name': 'No.of.labels', 'value': '', 'label': 'No.Of.Labels', 'description': 'The number of distinct labels (populations) to look for. You may run the MClustClusterLabelBIC module to get the number of labels you should look for.'}, {'name': 'Dimensions', 'value': '', 'label': 'Dimensions', 'description': 'A comma-separated list of dimensions (parameters/channels) that shall be used for meta clustering (clustering of means of clusters). All specified dimensions shall be included in all the input clustering results CSV files. All dimensions (but Time) will be used if this parameter is not provided.'}], 'default_args': [{'name': 'task_name', 'value': 'MClustClusterLabel', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Model', 'value': 'VII', 'label': 'Model', 'description': 'Specific model used for meta clustering; see documentation for details about different models.'}, {'name': 'Uncertainty.threshold', 'value': '65', 'label': 'Uncertainty.Threshold', 'description': 'The threshold (in percentage) for uncertainty when assigning labels. Input a value greater than 50 and lower than 100. Populations where no label can be assigned with probability greater or equal to this uncertainty threshold will be left unassigned in the results.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'MClustClusterLabelBIC': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Bayesian information criterion (BIC) for model based labeling/matching of clusters; helps estimating the number of labels for MClustClusterLabel.', 'required_args': [{'name': 'Input.clustering.results', 'value': '', 'label': 'Input.Clustering.Results', 'description': 'The input clustering results (of multiple data files) in CSV files with each clustered population stored in a separate CSV file. All these CSV files shall have the same columns (dimensions/channels) or at least, they all shall include columns specified by the dimensions parameter.'}, {'name': 'Dimensions', 'value': '', 'label': 'Dimensions', 'description': 'A comma-separated list of dimensions (parameters/channels) that shall be used for meta clustering (clustering of means of clusters). All specified dimensions shall be included in all the input clustering results CSV files. All dimensions (but Time) will be used if this parameter is not provided.'}], 'default_args': [{'name': 'task_name', 'value': 'MClustClusterLabelBIC', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Range', 'value': '1-10', 'label': 'Range', 'description': 'The range of distinct labels (populations, components) to calculate BIC for. Provide the range in the min-max format, where both, mix and max are integers and min is smaller than max.'}, {'name': 'EII', 'value': 'TRUE', 'label': 'Eii', 'description': 'Include the EII model (spherical, equal volume) when calculating BIC.'}, {'name': 'VII', 'value': 'TRUE', 'label': 'Vii', 'description': 'Include the VII model (spherical, unequal volume) when calculating BIC.'}, {'name': 'EEI', 'value': 'FALSE', 'label': 'Eei', 'description': 'Include the EEI model (diagonal, equal volume and shape) when calculating BIC.'}, {'name': 'VEI', 'value': 'FALSE', 'label': 'Vei', 'description': 'Include the VEI model (diagonal, varying volume, equal shape) when calculating BIC.'}, {'name': 'EVI', 'value': 'FALSE', 'label': 'Evi', 'description': 'Include the EVI model (diagonal, equal volume, varying shape) when calculating BIC.'}, {'name': 'VVI', 'value': 'FALSE', 'label': 'Vvi', 'description': 'Include the VVI model (diagonal, varying volume and shape) when calculating BIC.'}, {'name': 'EEE', 'value': 'FALSE', 'label': 'Eee', 'description': 'Include the EEE model (ellipsoidal, equal volume, shape, and orientation) when calculating BIC.'}, {'name': 'EEV', 'value': 'FALSE', 'label': 'Eev', 'description': 'Include the EEV model (ellipsoidal, equal volume and equal shape) when calculating BIC.'}, {'name': 'VEV', 'value': 'FALSE', 'label': 'Vev', 'description': 'Include the VEV model (ellipsoidal, equal shape) when calculating BIC.'}, {'name': 'VVV', 'value': 'FALSE', 'label': 'Vvv', 'description': 'Include the VVV model (ellipsoidal, varying volume, shape, and orientation) when calculating BIC.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'MergeColumns': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Merge datasets by column.', 'required_args': [{'name': 'input.filename1', 'value': '', 'label': 'Input.Filename1', 'description': 'first file to merge'}, {'name': 'input.filename2', 'value': '', 'label': 'Input.Filename2', 'description': 'second file to merge'}, {'name': 'zip.filename', 'value': '', 'label': 'Zip.Filename', 'description': 'zip file containing files to merge. Either a zip file or the input files above must be provided.'}], 'default_args': [{'name': 'task_name', 'value': 'MergeColumns', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output', 'value': '<input.filename1_basename><zip.filename_basename>.merged', 'label': 'Output', 'description': 'name of the merged file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'MergeFCSDataFiles': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Merge multiple FCS data files into a single FCS datasets; includes multimple subsampling options.', 'required_args': [{'name': 'Input.files', 'value': '', 'label': 'Input.Files', 'description': 'The input FCS files to be merged; either a zip file or a pointer to a directory or a text file with FCS file names (full path) with one filename per line.'}, {'name': 'Keep.n.events', 'value': '', 'label': 'Keep.N.Events', 'description': 'Keep approximately n events from each dataset, optional, cannot be combined with subsampling rate.'}], 'default_args': [{'name': 'task_name', 'value': 'MergeFCSDataFiles', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Subsampling.rate', 'value': '100', 'label': 'Subsampling.Rate', 'description': 'Subsampling rate as percentage of events to be included from each data set, value between 0 and 100, optional, default 100%, cannot be combined with keep n events.'}, {'name': 'Random.seed', 'value': '1234567', 'label': 'Random.Seed', 'description': 'The random seed'}, {'name': 'Output.file.name', 'value': 'MergedFCSDataFiles.fcs', 'label': 'Output.File.Name', 'description': 'The output FCS file name.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'MergeHTSeqCounts': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Merges HTSeq read count data files into one file', 'required_args': [{'name': 'input.files', 'value': '', 'label': 'Input.Files', 'description': 'A directory or one or more text files containing HTSeq read counts'}], 'default_args': [{'name': 'task_name', 'value': 'MergeHTSeqCounts', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.prefix', 'value': '<input.files_basename>', 'label': 'Output.Prefix', 'description': 'The base name of the output file. File extensions will be added automatically.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'MergeRows': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Merge datasets by row.', 'required_args': [{'name': 'input.filename1', 'value': '', 'label': 'Input.Filename1', 'description': 'first file to merge'}, {'name': 'input.filename2', 'value': '', 'label': 'Input.Filename2', 'description': 'second file to merge'}, {'name': 'zip.filename', 'value': '', 'label': 'Zip.Filename', 'description': 'zip file containing files to merge. Either a zip file or the input files above must be provided.'}], 'default_args': [{'name': 'task_name', 'value': 'MergeRows', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output', 'value': '<input.filename1_basename><zip.filename_basename>.merged', 'label': 'Output', 'description': 'name of the merged file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'MetageneProjection': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Projects one or more data sets onto the metagene representation of another data set', 'required_args': [{'name': 'model.gct.file', 'value': '', 'label': 'Model.Gct.File', 'description': 'The model .gct data file'}, {'name': 'model.cls.file', 'value': '', 'label': 'Model.Cls.File', 'description': 'The model .cls data file'}, {'name': 'model.preprocessing.file', 'value': '', 'label': 'Model.Preprocessing.File', 'description': 'The pre-processing parameters for the model dataset'}, {'name': 'test.gct.file', 'value': '', 'label': 'Test.Gct.File', 'description': 'The test .gct or zip of .gct data files'}, {'name': 'test.cls.file', 'value': '', 'label': 'Test.Cls.File', 'description': 'The test .cls or zip of .cls data files'}, {'name': 'test.preprocessing.file', 'value': '', 'label': 'Test.Preprocessing.File', 'description': 'The pre-processing parameters for the test dataset'}, {'name': 'phenotype.plotting.colors', 'value': '', 'label': 'Phenotype.Plotting.Colors', 'description': 'Plotting colors for phenotypes'}, {'name': 'phenotype.plotting.symbols', 'value': '', 'label': 'Phenotype.Plotting.Symbols', 'description': 'Plotting symbols for phenotypes'}], 'default_args': [{'name': 'task_name', 'value': 'MetageneProjection', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'num.characters', 'value': '9', 'label': 'Num.Characters', 'description': 'Number of characters to use to identify classes from the .cls files'}, {'name': 'identifier', 'value': '<model.gct.file_basename>.output', 'label': 'Identifier', 'description': 'Prefix to name output files'}, {'name': 'k.projection', 'value': '3', 'label': 'K.Projection', 'description': 'Number of metagenes in projection'}, {'name': 'algorithm', 'value': 'NMF.div', 'label': 'Algorithm', 'description': 'Algorithm for Metagene Projection'}, {'name': 'number.of.iterations', 'value': '2000', 'label': 'Number.Of.Iterations', 'description': 'Number of algorithm iterations'}, {'name': 'seed', 'value': '1234', 'label': 'Seed', 'description': 'Random seed to initialize metagene matrices (Note: results may only be reproducible on the same platform)'}, {'name': 'post.projection.normalization', 'value': 'yes', 'label': 'Post.Projection.Normalization', 'description': 'apply post-projection normalization'}, {'name': 'heatmap.row.norm', 'value': 'no', 'label': 'Heatmap.Row.Norm', 'description': 'row-normalize (standardize) the rows in the heat map'}, {'name': 'heatmap.color.scheme', 'value': 'reddish color map', 'label': 'Heatmap.Color.Scheme', 'description': ''}, {'name': 'confidence.threshold', 'value': '0.3', 'label': 'Confidence.Threshold', 'description': 'Confidence threshold (Brier score) to separate call from no-calls'}, {'name': 'symbol.scaling', 'value': '1', 'label': 'Symbol.Scaling', 'description': 'Graphical scaling for symbols in plots and plot legends'}, {'name': 'kernel', 'value': 'radial', 'label': 'Kernel', 'description': 'Kernel function for SVM'}, {'name': 'cost', 'value': '1', 'label': 'Cost', 'description': 'cost function for SVM'}, {'name': 'gamma', 'value': '0.05', 'label': 'Gamma', 'description': 'Gamma coefficient for radial base function kernel'}, {'name': 'theta', 'value': '0', 'label': 'Theta', 'description': 'Smoothing parameter only used for the NSNMF.div algorithm'}, {'name': 'lambda', 'value': '0', 'label': 'Lambda', 'description': 'Sparse parameter used only for the SNMF algorithm'}, {'name': 'model.set.refinement', 'value': 'yes', 'label': 'Model.Set.Refinement', 'description': 'perform model set refinement'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'MINDY': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Runs the MINDY algorithm for inferring genes that modulate the activity of a transcription factor at post-transcriptional levels ', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'The dataset file - .res, .gct'}, {'name': 'modulator.list', 'value': '', 'label': 'Modulator.List', 'description': 'A list of possible modulator genes to analyze - .txt'}, {'name': 'dpi.transcription.factor.list', 'value': '', 'label': 'Dpi.Transcription.Factor.List', 'description': 'List of transcription factors in dataset - .txt'}, {'name': 'transcription.factor', 'value': '', 'label': 'Transcription.Factor', 'description': 'Transcription factor'}], 'default_args': [{'name': 'task_name', 'value': 'MINDY', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'sample.percentage', 'value': '35.0', 'label': 'Sample.Percentage', 'description': 'Percentage of samples to use in analysis (must be <=50)'}, {'name': 'conditional.mi.threshold', 'value': '0.0', 'label': 'Conditional.Mi.Threshold', 'description': 'Conditional mutual information (MI) score threshold (number between 0 and 1)'}, {'name': 'conditional.mi.p.value', 'value': '1.0', 'label': 'Conditional.Mi.P.Value', 'description': 'Conditional mutual info p-value threshold (ignored if conditonal MI threshold >0)'}, {'name': 'dpi.tolerance', 'value': '1.0', 'label': 'Dpi.Tolerance', 'description': 'The data processing inequality tolerance (number between 0 and 1)'}, {'name': 'unconditional.mi.threshold', 'value': '0.0', 'label': 'Unconditional.Mi.Threshold', 'description': 'Unconditional mutual information (MI) score threshold (number between 0 and 1)'}, {'name': 'unconditional.mi.p.value', 'value': '1.0', 'label': 'Unconditional.Mi.P.Value', 'description': 'Unconditional mutual information p-value threshold (ignored if unconditonal MI threshold >0)'}, {'name': 'output.file', 'value': '<input.file_basename>.mindy', 'label': 'Output.File', 'description': 'Output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'MINDYViewer': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Visualizes the results of MINDY', 'required_args': [{'name': 'dataset.file', 'value': '', 'label': 'Dataset.File', 'description': 'The dataset file - .res, .gct'}, {'name': 'mindy.file', 'value': '', 'label': 'Mindy.File', 'description': 'Output from MINDY- .mindy'}], 'default_args': [{'name': 'task_name', 'value': 'MINDYViewer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ModEcocTrainTest': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Modified error correcting output code prediction', 'required_args': [{'name': 'train.file', 'value': '', 'label': 'Train.File', 'description': 'The training features and class label information in .arff format'}, {'name': 'test.file', 'value': '', 'label': 'Test.File', 'description': 'The testing features and possible class label information in .arff format'}, {'name': 'saved.model.file', 'value': '', 'label': 'Saved.Model.File', 'description': 'Input modEcocTrainTest model'}], 'default_args': [{'name': 'task_name', 'value': 'ModEcocTrainTest', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'model.output.file', 'value': '<train.file_basename>.model', 'label': 'Model.Output.File', 'description': 'Output modEcocTrainTest model'}, {'name': 'base.classifier', 'value': '0', 'label': 'Base.Classifier', 'description': 'The name of the base classifier to use'}, {'name': 'base.multiplier', 'value': '4', 'label': 'Base.Multiplier', 'description': 'The base classifier multiplier'}, {'name': 'pred.results.output.file', 'value': '<test.file_basename>', 'label': 'Pred.Results.Output.File', 'description': 'The name of the output file for prediction results.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ModEcocXValidation': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Modified error correcting output code cross validation prediction', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'The input features and class label information in .arff format'}], 'default_args': [{'name': 'task_name', 'value': 'ModEcocXValidation', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'num.folds', 'value': '2', 'label': 'Num.Folds', 'description': 'The number of folds should be an integer greater than or equal to two.'}, {'name': 'base.classifier', 'value': '0', 'label': 'Base.Classifier', 'description': 'The name of the base classifier to use'}, {'name': 'base.multiplier', 'value': '4', 'label': 'Base.Multiplier', 'description': 'The base classifier multiplier'}, {'name': 'pred.results.file', 'value': '<input.file_basename>', 'label': 'Pred.Results.File', 'description': 'The name of the output file for prediction results.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Multiplot': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Creates customizable plots of expression data-derived data.', 'required_args': [{'name': 'input.file.name', 'value': '', 'label': 'Input.File.Name', 'description': 'File created by MultiplotPreprocess (.zip)'}], 'default_args': [{'name': 'task_name', 'value': 'Multiplot', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'number.of.plots', 'value': '1', 'label': 'Number.Of.Plots', 'description': 'Smaller one-plot visualizer or bigger two-plot'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'MultiplotExtractor': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Creates customizable plots of expression data-derived data.', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'File created by MultiplotPreprocess (.zip)'}], 'default_args': [{'name': 'task_name', 'value': 'MultiplotExtractor', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'MultiplotPreprocess': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Creates derived data from an expression dataset for use in the Multiplot Visualizer module', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'File with data to prepare for plotting (gct)'}, {'name': 'class.filename', 'value': '', 'label': 'Class.Filename', 'description': 'Class file for this data (cls)'}], 'default_args': [{'name': 'task_name', 'value': 'MultiplotPreprocess', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<input.filename_basename>.zip', 'label': 'Output.File', 'description': 'Name of file to create for use in Multiplot visualizer'}, {'name': 'replicateFC', 'value': '0', 'label': 'ReplicateFC', 'description': 'Whether to compute fold changes between each replicate (turning this on can slow your work).'}, {'name': 'randomData', 'value': '0', 'label': 'RandomData', 'description': 'Whether to create random expression data (and fold changes, etc for this data). Turning this on can slow your work'}, {'name': 'outlierElimination', 'value': '0', 'label': 'OutlierElimination', 'description': 'Remove outlier replicate values of genes from processing (see \"Help\" for details)'}, {'name': 'doStats', 'value': '1', 'label': 'DoStats', 'description': 'Perform t-test, Hochberg p-value adjustment, and significance comparison to randomized data'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'MultiplotStudio': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Scatterplot visualization tool for analyzing differential gene expression.', 'required_args': [{'name': 'dataFile', 'value': '', 'label': 'DataFile', 'description': 'GenePattern dataset (*.gct), Multiplot Studio Workspace file (*.msdata) or CSV (see help for format requirements).'}, {'name': 'classFile', 'value': '', 'label': 'ClassFile', 'description': 'Class file (*.cls). Not required, but lack of a class file will result in fewer calculation options. Ignored if dataFile is a Workspace.'}], 'default_args': [{'name': 'task_name', 'value': 'MultiplotStudio', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'MutSigCV': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Identifies genes that are significantly mutated in cancer genomes, using a model with mutational covariates.', 'required_args': [{'name': 'mutation.table.file', 'value': '', 'label': 'Mutation.Table.File', 'description': 'Mutation list in MAF format.'}, {'name': 'coverage.table.file', 'value': '', 'label': 'Coverage.Table.File', 'description': 'File containing the coverage table.  This should be a file with per-sample coverage as produced by MuTect from the samples\\' WIG files.  If you do not have this level of coverage information, you can instead use the territory file for the reference human exome to direct MutSigCV to assume full coverage.  This file is found using \"Add Path or URL\" under \"shared_data/example_files/MutSigCV_1.3/exome_full192.coverage.txt\".'}, {'name': 'mutation.type.dictionary', 'value': '', 'label': 'Mutation.Type.Dictionary', 'description': 'The mutation type dictionary to use for automatic category and effect discovery.  This is necessary only if you are using a MAF file without the columns \"categ\" and \"effect\".  A suggested dictionary can be found using \"Add Path or URL\" under \"shared_data/example_files/MutSigCV_1.3/mutation_type_dictionary_file.txt\".'}, {'name': 'genome.build', 'value': '', 'label': 'Genome.Build', 'description': 'Genome build to use for automatic category and effect discovery.  This is necessary only if you are using a MAF file without the columns \"categ\" and \"effect\".'}], 'default_args': [{'name': 'task_name', 'value': 'MutSigCV', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'covariates.table.file', 'value': 'ftp://ftp.broadinstitute.org/pub/genepattern/example_files/MutSigCV_1.3/gene.covariates.txt', 'label': 'Covariates.Table.File', 'description': 'File containing the covariates table.  The default selection is the covariates file used in the MutSigCV paper published in Nature (see the documentation for further information and a full reference). '}, {'name': 'output.filename.base', 'value': '<mutation.table.file_basename>', 'label': 'Output.Filename.Base', 'description': 'Base name for the output files.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'mzXMLToCsv': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Converts a mzXML file to a zip of csv files', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'input filename - .mzXML'}], 'default_args': [{'name': 'task_name', 'value': 'mzXMLToCsv', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<input.filename_basename>.zip', 'label': 'Output.File', 'description': 'output file containing zip of csv files'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'NearestTemplatePrediction': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Nearest neighbor prediction based on a list of marker genes', 'required_args': [{'name': 'input.exp.filename', 'value': '', 'label': 'Input.Exp.Filename', 'description': 'Gene expression data set (.gct)'}, {'name': 'input.features.filename', 'value': '', 'label': 'Input.Features.Filename', 'description': 'List of marker genes (.txt): Probe ID, Gene name, Class (1,2,...), Weight (optional)'}], 'default_args': [{'name': 'task_name', 'value': 'NearestTemplatePrediction', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.name', 'value': 'NTP', 'label': 'Output.Name', 'description': 'Name for output files'}, {'name': 'distance.selection', 'value': 'cosine', 'label': 'Distance.Selection', 'description': 'Distance metric'}, {'name': 'weight.genes', 'value': 'T', 'label': 'Weight.Genes', 'description': 'Weight genes? (by statistic, fold change, etc. only for 2 classes)'}, {'name': 'num.resamplings', 'value': '1000', 'label': 'Num.Resamplings', 'description': '# resampling to generate null distribution for distance metric'}, {'name': 'GenePattern.output', 'value': 'F', 'label': 'GenePattern.Output', 'description': 'Create .gct and .cls files for GenePattern'}, {'name': 'random.seed', 'value': '7392854', 'label': 'Random.Seed', 'description': 'Random seed'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'NMF': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Non-negative Matrix Factorization', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'Input file in GCT format with input data (V matrix) (GCT format only). e.g. ALL_vs_AML.gct'}], 'default_args': [{'name': 'task_name', 'value': 'NMF', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<input.filename_basename>.log', 'label': 'Output.File', 'description': 'Output log file. e.g. NMF.log'}, {'name': 'nfact', 'value': '4', 'label': 'Nfact', 'description': 'Number of NMF factors. e.g. 2, 4, 10'}, {'name': 'niter', 'value': '100', 'label': 'Niter', 'description': 'Number of NMF iterations. e.g. 100, 250, 500'}, {'name': 'print.dump', 'value': '0', 'label': 'Print.Dump', 'description': 'Print matrices in output for debug'}, {'name': 'cnorm.flag', 'value': '0', 'label': 'Cnorm.Flag', 'description': 'Column normalization flag'}, {'name': 'rnorm.flag', 'value': '0', 'label': 'Rnorm.Flag', 'description': 'Row normalization flag'}, {'name': 'gshift', 'value': '0', 'label': 'Gshift', 'description': 'Global additive shift after normalization. e.g. 1, 4'}, {'name': 'norm.type', 'value': '0', 'label': 'Norm.Type', 'description': 'Type of normalization'}, {'name': 'seed', 'value': '12345', 'label': 'Seed', 'description': 'Random seed used to initialize W and H matrices by the randon number generator. e.g. 4585, 4567, 5980'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'NMFConsensus': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Non-negative Matrix Factorization (NMF) Consensus Clustering', 'required_args': [{'name': 'dataset.filename', 'value': '', 'label': 'Dataset.Filename', 'description': 'Input dataset (gct or res)'}], 'default_args': [{'name': 'task_name', 'value': 'NMFConsensus', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'k.initial', 'value': '2', 'label': 'K.Initial', 'description': 'Initial value of K'}, {'name': 'k.final', 'value': '5', 'label': 'K.Final', 'description': 'Final value of K'}, {'name': 'num.clusterings', 'value': '20', 'label': 'Num.Clusterings', 'description': 'Number of clusterings per value of K'}, {'name': 'max.num.iterations', 'value': '2000', 'label': 'Max.Num.Iterations', 'description': 'Maximum number of NMF iterations per clustering'}, {'name': 'error.function', 'value': 'divergence', 'label': 'Error.Function', 'description': 'Error function'}, {'name': 'random.seed', 'value': '123456789', 'label': 'Random.Seed', 'description': 'Seed for random number generator'}, {'name': 'output.file.prefix', 'value': '<dataset.filename_basename>', 'label': 'Output.File.Prefix', 'description': 'Prefix to append to all output files'}, {'name': 'stop.convergence', 'value': '40', 'label': 'Stop.Convergence', 'description': 'How many \"no change\" checks are needed to stop NMF iterations before max iterations is reached(convergence)'}, {'name': 'stop.frequency', 'value': '10', 'label': 'Stop.Frequency', 'description': 'Frequency (NMF iterations) of \"no change\" checks'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'NMFConsensus_combine_lewu': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Non-negative Matrix Factorization (NMF) Consensus Clustering (combine single k result)', 'required_args': [{'name': 'param.filename', 'value': '', 'label': 'Param.Filename', 'description': 'Input param file '}], 'default_args': [{'name': 'task_name', 'value': 'NMFConsensus_combine_lewu', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'NMFConsensus_sk_lewu': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Non-negative Matrix Factorization (NMF) Consensus Clustering', 'required_args': [{'name': 'param.filename', 'value': '', 'label': 'Param.Filename', 'description': 'Input param file '}], 'default_args': [{'name': 'task_name', 'value': 'NMFConsensus_sk_lewu', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'NMFPartition': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'The module will output a list of parameter files', 'required_args': [{'name': 'dataset.filename', 'value': '', 'label': 'Dataset.Filename', 'description': 'Input dataset (gct or res)'}], 'default_args': [{'name': 'task_name', 'value': 'NMFPartition', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'k.initial', 'value': '2', 'label': 'K.Initial', 'description': 'Initial value of K'}, {'name': 'k.final', 'value': '5', 'label': 'K.Final', 'description': 'Final value of K'}, {'name': 'num.clusterings', 'value': '20', 'label': 'Num.Clusterings', 'description': 'Number of clusterings per value of K'}, {'name': 'max.num.iterations', 'value': '2000', 'label': 'Max.Num.Iterations', 'description': 'Maximum number of NMF iterations per clustering'}, {'name': 'error.function', 'value': 'divergence', 'label': 'Error.Function', 'description': 'Error function'}, {'name': 'random.seed', 'value': '123456789', 'label': 'Random.Seed', 'description': 'Seed for random number generator'}, {'name': 'output.file.prefix', 'value': '<dataset.filename_basename>', 'label': 'Output.File.Prefix', 'description': 'Prefix to append to all output files'}, {'name': 'stop.convergence', 'value': '40', 'label': 'Stop.Convergence', 'description': 'How many \"no change\" checks are needed to stop NMF iterations before max iterations is reached(convergence)'}, {'name': 'stop.frequency', 'value': '10', 'label': 'Stop.Frequency', 'description': 'Frequency (NMF iterations) of \"no change\" checks'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'NMFSingleKPipe': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '', 'required_args': [{'name': 'NMFConsensus_sk_lewu1.param.filename', 'value': '', 'label': 'NMFConsensus Sk Lewu1.Param.Filename', 'description': 'Input param file '}], 'default_args': [{'name': 'task_name', 'value': 'NMFSingleKPipe', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'NormLines': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Normalizes RNAi hairpin scores across cell lines using ZMAD, PMAD, rank or quantile- ignores NaNs', 'required_args': [{'name': 'gct.file', 'value': '', 'label': 'Gct.File', 'description': 'Input gct file to normalize by samples'}], 'default_args': [{'name': 'task_name', 'value': 'NormLines', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'method', 'value': 'quantile', 'label': 'Method', 'description': 'Options are ZMAD, PMAD, quantile'}, {'name': 'plotfile', 'value': 'NormPDF.pdf', 'label': 'Plotfile', 'description': ''}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ParallelCBS': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Runs CBS algorithm on multiple samples in parallel', 'required_args': [{'name': 'SplitColumns1.input.file', 'value': '', 'label': 'SplitColumns1.Input.File', 'description': 'A tab delimited file'}], 'default_args': [{'name': 'task_name', 'value': 'ParallelCBS', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'ConcatenateFilelist4.output.filename', 'value': '<filelist_basename>.catted.seg.txt', 'label': 'ConcatenateFilelist4.Output.Filename', 'description': 'The name of the output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ParallelNMFConsensusPipe': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'A parallelized implementation of the NMFConsensus module, implemented as a GenePattern scatter-gather pipeline.', 'required_args': [{'name': 'NMFPartition1.dataset.filename', 'value': '', 'label': 'NMFPartition1.Dataset.Filename', 'description': 'Input dataset (gct or res)'}], 'default_args': [{'name': 'task_name', 'value': 'ParallelNMFConsensusPipe', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'NMFPartition1.k.initial', 'value': '2', 'label': 'NMFPartition1.K.Initial', 'description': 'Initial value of K'}, {'name': 'NMFPartition1.k.final', 'value': '5', 'label': 'NMFPartition1.K.Final', 'description': 'Final value of K'}, {'name': 'NMFPartition1.num.clusterings', 'value': '20', 'label': 'NMFPartition1.Num.Clusterings', 'description': 'Number of clusterings per value of K'}, {'name': 'NMFPartition1.max.num.iterations', 'value': '2000', 'label': 'NMFPartition1.Max.Num.Iterations', 'description': 'Maximum number of NMF iterations per clustering'}, {'name': 'NMFPartition1.error.function', 'value': 'divergence', 'label': 'NMFPartition1.Error.Function', 'description': 'Error function'}, {'name': 'NMFPartition1.random.seed', 'value': '123456789', 'label': 'NMFPartition1.Random.Seed', 'description': 'Seed for random number generator'}, {'name': 'NMFPartition1.output.file.prefix', 'value': '<dataset.filename_basename>', 'label': 'NMFPartition1.Output.File.Prefix', 'description': 'Prefix to append to all output files'}, {'name': 'NMFPartition1.stop.convergence', 'value': '40', 'label': 'NMFPartition1.Stop.Convergence', 'description': 'How many \"no change\" checks are needed to stop NMF iterations before max iterations is reached(convergence)'}, {'name': 'NMFPartition1.stop.frequency', 'value': '10', 'label': 'NMFPartition1.Stop.Frequency', 'description': 'Frequency (NMF iterations) of \"no change\" checks'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'PARIS': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Feature selection by class or against a continuous vector (Probablility Analysis by Ranked Information Score)', 'required_args': [{'name': 'target.profile', 'value': '', 'label': 'Target.Profile', 'description': 'Target profile file (either .cls or .gct).  Used to classify cell line samples into groups, or assign a continuous target vector to the cell line samples.'}, {'name': 'target', 'value': '', 'label': 'Target', 'description': \"Target gene or ID from target.profile file.  Should match a row in the 'Name' or 'Description' column of a .gct file, or match one label in the 2nd row of the .cls file\"}, {'name': 'data', 'value': '', 'label': 'Data', 'description': 'Data (feature) file to rank based on target profile'}, {'name': 'prefix', 'value': '', 'label': 'Prefix', 'description': 'prefix for labeling output files'}, {'name': 'advanced', 'value': '', 'label': 'Advanced', 'description': 'Leave blank for default.  This parameter allows the user to change the underlying default metric.  Options will be listed in the documentation.'}], 'default_args': [{'name': 'task_name', 'value': 'PARIS', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'permtype', 'value': 'standard', 'label': 'Permtype', 'description': 'standard; subclass.stratified (lineage-corrected)'}, {'name': 'type.flag', 'value': 'gene', 'label': 'Type.Flag', 'description': 'perturbagen level data (shRNA) or gene level scores (gene)'}, {'name': 'quality', 'value': 'normal', 'label': 'Quality', 'description': 'controls how many permutations are run; draft will take less time'}, {'name': 'direction', 'value': 'positive', 'label': 'Direction', 'description': 'Direction of feature matching; positive means high values to high values, low to low; negative means high to low values'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'PCA': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Principal Component Analysis (adapted from MeV, http://www.tm4.org/m)', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'input file - .gct, .res, .odf'}], 'default_args': [{'name': 'task_name', 'value': 'PCA', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'cluster.by', 'value': '3', 'label': 'Cluster.By', 'description': 'Cluster by rows or columns'}, {'name': 'output.file', 'value': '<input.filename_basename>', 'label': 'Output.File', 'description': 'stub name of the output files'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'PCAViewer': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Principal Component Analysis Visualizer (adapted from MeV, http://www.tigr.org/software/tm4/mev.html)', 'required_args': [{'name': 'dataset.file', 'value': '', 'label': 'Dataset.File', 'description': 'The dataset - .res, .gct'}, {'name': 's.matrix.file', 'value': '', 'label': 'S.Matrix.File', 'description': 'The s matrix - .odf'}, {'name': 't.matrix.file', 'value': '', 'label': 'T.Matrix.File', 'description': 'The t matrix - .odf'}, {'name': 'u.matrix.file', 'value': '', 'label': 'U.Matrix.File', 'description': 'The u matrix - .odf'}, {'name': 'cls.or.sample.info.file', 'value': '', 'label': 'Cls.Or.Sample.Info.File', 'description': 'A cls or sample info file - .cls, .txt'}, {'name': 'sample.info.class', 'value': '', 'label': 'Sample.Info.Class', 'description': 'The name of the class column to use in the sample info file (required if sample info file provided)'}, {'name': 'featureset.file', 'value': '', 'label': 'Featureset.File', 'description': 'The feature set file - .gmt, .gmx, .grp'}], 'default_args': [{'name': 'task_name', 'value': 'PCAViewer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'PclToGct': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Convert Stanford preclustering file (.pcl) to .gct', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'pcl file to convert'}], 'default_args': [{'name': 'task_name', 'value': 'PclToGct', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.filename', 'value': '<input.filename_basename>.gct', 'label': 'Output.Filename', 'description': 'output gct file'}, {'name': 'weight.handling', 'value': '3', 'label': 'Weight.Handling', 'description': 'How to handle manually edited weights (weights not equal to 1) present in PCL file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'PeakMatch': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Perform peak matching on LC-MS data', 'required_args': [{'name': 'peakList.filename', 'value': '', 'label': 'PeakList.Filename', 'description': 'Zip file containing peak lists'}, {'name': 'sampleInfo.filename', 'value': '', 'label': 'SampleInfo.Filename', 'description': 'CSV file with header: experiment, sample, class'}, {'name': 'LandmarkMatchOutput.filename', 'value': '', 'label': 'LandmarkMatchOutput.Filename', 'description': 'Zip file containing the output from landmark matching.'}, {'name': 'MZ.tolerance', 'value': '', 'label': 'MZ.Tolerance', 'description': 'm/z tolerance (ppm). Used if landmark match output is not supplied. Defaults to 10 ppm.'}, {'name': 'RT.tolerance', 'value': '', 'label': 'RT.Tolerance', 'description': 'retention time tolerance (min). Used if landmark match is not supplied. Defaults to 2 min.'}, {'name': 'numberProcesses', 'value': '', 'label': 'NumberProcesses', 'description': 'Number of processess when running in parallel. Defaults to 1 (sequential).'}], 'default_args': [{'name': 'task_name', 'value': 'PeakMatch', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'outputName', 'value': 'PeakMatchOutput', 'label': 'OutputName', 'description': 'File name prefix for the output files'}, {'name': 'running.on.LSF', 'value': 'no', 'label': 'Running.On.LSF', 'description': 'Whether you are running on an LSF cluster version of GenePattern'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Peaks': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Determine peaks in the spectrum using a series of digital filters.', 'required_args': [{'name': 'spectrum.filename', 'value': '', 'label': 'Spectrum.Filename', 'description': 'spectrum data file - .csv'}, {'name': 'low.Da', 'value': '', 'label': 'Low.Da', 'description': 'Minimum M/Z to include'}, {'name': 'high.Da', 'value': '', 'label': 'High.Da', 'description': 'Maximum M/Z to include'}], 'default_args': [{'name': 'task_name', 'value': 'Peaks', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'percentile', 'value': '0.65', 'label': 'Percentile', 'description': 'After applying filters, threshold above which peaks are located'}, {'name': 'smoothing.size', 'value': '21', 'label': 'Smoothing.Size', 'description': 'Size of the smoothing filter'}, {'name': 'adaptive.background.correction.strength', 'value': '0.75', 'label': 'Adaptive.Background.Correction.Strength', 'description': 'Strength of the adaptive background correction filter'}, {'name': 'adaptive.background.correction.size', 'value': '21', 'label': 'Adaptive.Background.Correction.Size', 'description': 'Size of the adaptive background correction filter'}, {'name': 'high.pass.filter.strength', 'value': '10', 'label': 'High.Pass.Filter.Strength', 'description': 'Strength of the high pass filter'}, {'name': 'high.pass.filter.factor', 'value': '5', 'label': 'High.Pass.Filter.Factor', 'description': 'Filter factor for the high pass filter'}, {'name': 'output.file', 'value': '<spectrum.filename_basename>.peaks.odf', 'label': 'Output.File', 'description': 'output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Picard.AddOrReplaceReadGroups': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Replaces all read groups in the input file with a new read group and assigns all reads to this read group in the output', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'Input file (BAM or SAM).'}, {'name': 'read.group.id', 'value': '', 'label': 'Read.Group.Id', 'description': 'Read Group ID'}, {'name': 'read.group.library', 'value': '', 'label': 'Read.Group.Library', 'description': 'DNA preparation library. Note that this field is required by the module Picard.MarkDuplicates. It is used to determine which read groups may contain molecular duplicates, in the case where the same DNA library was sequenced on multiple lanes.'}, {'name': 'read.group.platform', 'value': '', 'label': 'Read.Group.Platform', 'description': 'Read Group platform (e.g., Illumina, SOLID).'}, {'name': 'read.group.platform.unit', 'value': '', 'label': 'Read.Group.Platform.Unit', 'description': 'Read Group platform unit (e.g., run barcode).'}, {'name': 'read.group.sample.name', 'value': '', 'label': 'Read.Group.Sample.Name', 'description': 'Read Group sample name.'}, {'name': 'read.group.sequencing.center.name', 'value': '', 'label': 'Read.Group.Sequencing.Center.Name', 'description': 'Read Group sequencing center name.'}, {'name': 'read.group.description', 'value': '', 'label': 'Read.Group.Description', 'description': 'Read Group description.'}], 'default_args': [{'name': 'task_name', 'value': 'Picard.AddOrReplaceReadGroups', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.prefix', 'value': '<input.file_basename>', 'label': 'Output.Prefix', 'description': 'A label used to name the output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Picard.BamToSam': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Converts a BAM file to a SAM file', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'A BAM file'}], 'default_args': [{'name': 'task_name', 'value': 'Picard.BamToSam', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.prefix', 'value': '<input.file_basename>', 'label': 'Output.Prefix', 'description': 'A label used to name the output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Picard.CreateSequenceDictionary': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Reads FASTA or FASTA.GZ files containing reference sequences, and writes them as a SAM file containing a sequence dictionary', 'required_args': [{'name': 'reference.sequence.file', 'value': '', 'label': 'Reference.Sequence.File', 'description': 'Input reference fasta or fasta.gz '}, {'name': 'genome.assembly', 'value': '', 'label': 'Genome.Assembly', 'description': 'The genome assembly to put into AS field of sequence dictionary entry'}, {'name': 'uri', 'value': '', 'label': 'Uri', 'description': 'The uri to put into UR field of sequence dictionary entry'}, {'name': 'num.sequences', 'value': '', 'label': 'Num.Sequences', 'description': 'Stop after writing this many sequences'}], 'default_args': [{'name': 'task_name', 'value': 'Picard.CreateSequenceDictionary', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'truncate.names.at.white.space', 'value': 'no', 'label': 'Truncate.Names.At.White.Space', 'description': 'Make sequence name the first word from the > line in the reference file'}, {'name': 'output.file', 'value': '<reference.sequence.file_basename>.dict', 'label': 'Output.File', 'description': 'The name of the output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Picard.FastqToSam': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Extracts read sequences and qualities from the input fastq file and writes them into the output file in unaligned SAM or BAM format', 'required_args': [{'name': 'fastq.file', 'value': '', 'label': 'Fastq.File', 'description': 'Input fastq file (optionally gzipped) for single end data, or first read in paired end data'}, {'name': 'fastq.file.2', 'value': '', 'label': 'Fastq.File.2', 'description': 'The fastq file (optionally gzipped) for second read in paired end data'}, {'name': 'read.group.name', 'value': '', 'label': 'Read.Group.Name', 'description': 'Read group name'}, {'name': 'sample.name', 'value': '', 'label': 'Sample.Name', 'description': 'Sample name to insert into the read group header'}, {'name': 'library.name', 'value': '', 'label': 'Library.Name', 'description': 'The library name to insert into the LB attribute in the read group header'}, {'name': 'platform.unit', 'value': '', 'label': 'Platform.Unit', 'description': 'The platform unit (often run_barcode.lane) to insert into the read group header'}, {'name': 'platform', 'value': '', 'label': 'Platform', 'description': 'The platform type (e.g., Illumina, SOLiD) to insert into the read group header'}], 'default_args': [{'name': 'task_name', 'value': 'Picard.FastqToSam', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'quality.format', 'value': 'Standard', 'label': 'Quality.Format', 'description': 'A value describing how the quality values are encoded in the fastq'}, {'name': 'sort.order', 'value': 'coordinate', 'label': 'Sort.Order', 'description': ''}, {'name': 'output.format', 'value': 'bam', 'label': 'Output.Format', 'description': 'Whether the file output format should be SAM or BAM'}, {'name': 'output.prefix', 'value': '<fastq.file_basename>', 'label': 'Output.Prefix', 'description': 'The prefix to use for the output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Picard.MarkDuplicates': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Examines aligned records in the supplied SAM or BAM file to locate duplicate reads.', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'A SAM or BAM file'}, {'name': 'read.name.regex', 'value': '', 'label': 'Read.Name.Regex', 'description': 'Regular expression that can be used to parse read names in the input file'}], 'default_args': [{'name': 'task_name', 'value': 'Picard.MarkDuplicates', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'remove.duplicates', 'value': 'false', 'label': 'Remove.Duplicates', 'description': 'Whether to remove duplicates'}, {'name': 'max.file.handles', 'value': '8000', 'label': 'Max.File.Handles', 'description': 'Maximum number of file handles to keep open when spilling read ends to disk'}, {'name': 'optical.duplicate.distance', 'value': '100', 'label': 'Optical.Duplicate.Distance', 'description': 'The maximum offset between two duplicate clusters in order to consider them optical duplicates. This should usually be set to some fairly small number (e.g. 5-10 pixels) unless using later versions of the Illumina pipeline that multiply pixel values by 10, in which case 50-100 is more normal'}, {'name': 'sorting.collection.size.ratio', 'value': '0.25', 'label': 'Sorting.Collection.Size.Ratio', 'description': 'This number, plus the maximum RAM available to the JVM, determine the memory footprint used by some of the sorting collections.  If you are running out of memory, try reducing this number'}, {'name': 'output.prefix', 'value': '<input.file_basename>', 'label': 'Output.Prefix', 'description': 'The prefix to use for the output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Picard.ReorderSam': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Reorders a SAM file or a BAM file to match contig ordering in a provided reference file', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'A SAM or BAM file.'}, {'name': 'reference.file', 'value': '', 'label': 'Reference.File', 'description': 'Reference sequence to which to reorder reads in FASTA format.'}, {'name': 'reference.sequence.dictionary', 'value': '', 'label': 'Reference.Sequence.Dictionary', 'description': 'The sequence dictionary for the reference sequence file'}], 'default_args': [{'name': 'task_name', 'value': 'Picard.ReorderSam', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'allow.partial.overlap', 'value': 'false', 'label': 'Allow.Partial.Overlap', 'description': 'If yes, then allows only a partial overlap of the BAM contigs with the new reference sequence contigs. This requires a corresponding contig in the new reference for each read contig.'}, {'name': 'allow.contig.length.discordance', 'value': 'false', 'label': 'Allow.Contig.Length.Discordance', 'description': 'If yes, then permits mapping from a read contig to a new reference contig with the same name but a different length. Use with extreme caution and a full knowledge of the possible consequences.'}, {'name': 'output.prefix', 'value': '<input.file_basename>', 'label': 'Output.Prefix', 'description': 'The prefix of the output SAM or BAM file.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Picard.SamToBam': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Converts a SAM file to a BAM file', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'A SAM file'}], 'default_args': [{'name': 'task_name', 'value': 'Picard.SamToBam', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.prefix', 'value': '<input.file_basename>', 'label': 'Output.Prefix', 'description': 'A label used to name the output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Picard.SamToFastq': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Convert SAM or BAM files to FASTQ format', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'The SAM or BAM file'}, {'name': 'clipping.attribute', 'value': '', 'label': 'Clipping.Attribute', 'description': 'Attribute that stores the position at which the SAM record should be clipped'}, {'name': 'clipping.action.quality.value', 'value': '', 'label': 'Clipping.Action.Quality.Value', 'description': 'Value to set base qualities to when clipping action is \"change base quality values\"'}, {'name': 'fastq.output.prefix.1', 'value': '', 'label': 'Fastq.Output.Prefix.1', 'description': 'The fastq output file name for single-end fastq or first end of a paired fastq. Only specify when \"per read group\" is \"no\".'}, {'name': 'fastq.output.prefix.2', 'value': '', 'label': 'Fastq.Output.Prefix.2', 'description': 'The second end of a paired fastq. Only specify when \"per read group\" is \"no\".'}], 'default_args': [{'name': 'task_name', 'value': 'Picard.SamToFastq', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'per.read.group', 'value': 'no', 'label': 'Per.Read.Group', 'description': 'Whether to output a fastq file per read group'}, {'name': 're-reverse.bases', 'value': 'no', 'label': 'Re-Reverse.Bases', 'description': 'Whether to re-reverse bases and qualities of reads with negative strand flag set'}, {'name': 'include.non.pf.reads', 'value': 'no', 'label': 'Include.Non.Pf.Reads', 'description': 'Whether to include non-PF reads'}, {'name': 'clipping.action', 'value': 'none', 'label': 'Clipping.Action', 'description': 'Action to take with clipped reads when a clipping attribute is specified'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Picard.SortSam': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Sorts a SAM or BAM file in a specified order', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'Input file in SAM or BAM format'}, {'name': 'output.format', 'value': '', 'label': 'Output.Format', 'description': 'Whether the file output format should be SAM or BAM'}], 'default_args': [{'name': 'task_name', 'value': 'Picard.SortSam', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'sort.order', 'value': 'coordinate', 'label': 'Sort.Order', 'description': 'Sort order'}, {'name': 'output.prefix', 'value': '<input.file_basename>', 'label': 'Output.Prefix', 'description': 'prefix for the name of output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'PlateQAFCS': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Look for consistency between wells of a single plate by computing summary statistics.', 'required_args': [{'name': 'Input.FCS.files', 'value': '', 'label': 'Input.FCS.Files', 'description': 'FCS files from a single plate.'}], 'default_args': [{'name': 'task_name', 'value': 'PlateQAFCS', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Check.cell.counts', 'value': '1', 'label': 'Check.Cell.Counts', 'description': 'Count cells in each file?'}, {'name': 'Check.medians', 'value': '1', 'label': 'Check.Medians', 'description': 'Report the median measured values?'}, {'name': 'Check.PDF', 'value': '1', 'label': 'Check.PDF', 'description': 'Check the Probability Density Functions (PDF)?'}, {'name': 'Check.ECDF', 'value': '1', 'label': 'Check.ECDF', 'description': 'Check the Empirical Cumulative Distribution Functions (ECDF)?'}, {'name': 'Check.boxplots', 'value': '1', 'label': 'Check.Boxplots', 'description': 'Detect unexpected data distributions (unusual upper/lower quantiles)?'}, {'name': 'Check.timeline', 'value': '1', 'label': 'Check.Timeline', 'description': 'Check for unusual patterns in the acquisition of fluorescence and light scatter measurements over time?'}, {'name': 'Check.timeflow', 'value': '1', 'label': 'Check.Timeflow', 'description': 'Check for disturbances in the flow of cells over time?'}, {'name': 'FSC.channel', 'value': 'FSC-H', 'label': 'FSC.Channel', 'description': 'The name of the forward scatter parameter (short name, value of the $PnN keyword).'}, {'name': 'SSC.channel', 'value': 'SSC-H', 'label': 'SSC.Channel', 'description': 'The name of the side scatter parameter (short name, value of the $PnN keyword).'}, {'name': 'Zip.output', 'value': 'TRUE', 'label': 'Zip.Output', 'description': 'Zip the resulting report together?'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'PlotPeaks': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Plot peaks', 'required_args': [{'name': 'peaks.output.filename', 'value': '', 'label': 'Peaks.Output.Filename', 'description': 'The output from the Peaks module'}], 'default_args': [{'name': 'task_name', 'value': 'PlotPeaks', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<peaks.output.filename_basename>.pdf', 'label': 'Output.File', 'description': 'output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'PredictionResultsViewer': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Visualizes prediction results', 'required_args': [{'name': 'prediction.results.file', 'value': '', 'label': 'Prediction.Results.File', 'description': 'Prediction results - .odf'}], 'default_args': [{'name': 'task_name', 'value': 'PredictionResultsViewer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Preprocess.and.quality.check.RNASeq.data': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'This pipeline is used in the GenomeSpace recipe, \"Preprocess and quality check RNA-Seq data\". The recipe can be found at: http://www.genomespace.org/support/guides/recipes/sections/preprocess-and-qc-rna-seq-data.', 'required_args': [{'name': 'ConvertLineEndings1.input.filename', 'value': '', 'label': 'ConvertLineEndings1.Input.Filename', 'description': 'The input file (any non-binary file format)'}, {'name': 'TopHat4.GTF.file', 'value': '', 'label': 'TopHat4.GTF.File', 'description': 'A GTF file (v. 2.2 or higher) or GFF3 file containing a list of gene model annotations. TopHat will first extract the transcript sequences and align them to this virtual transcriptome first. Only the reads that do not fully map to the transcriptome will then be mapped on the genome. The reads that did map on the transcriptome will be converted to genomic mappings (spliced as needed) and merged with the novel mappings.'}, {'name': 'TopHat4.reads.pair.1', 'value': '', 'label': 'TopHat4.Reads.Pair.1', 'description': 'Unpaired reads file or first mate for paired reads. One or more files containing reads in FASTA or FASTQ format (bz2 and gz compressed files are supported).'}], 'default_args': [{'name': 'task_name', 'value': 'Preprocess.and.quality.check.RNASeq.data', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'PreprocessDataset': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Performs several preprocessing steps on a res, gct, or odf input file', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'Input filename - .res, .gct, .odf'}, {'name': 'threshold.for.removing.rows', 'value': '', 'label': 'Threshold.For.Removing.Rows', 'description': 'Threshold for removing rows'}, {'name': 'number.of.columns.above.threshold', 'value': '', 'label': 'Number.Of.Columns.Above.Threshold', 'description': 'Remove row if this number of columns not >= given threshold'}], 'default_args': [{'name': 'task_name', 'value': 'PreprocessDataset', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'threshold.and.filter', 'value': '1', 'label': 'Threshold.And.Filter', 'description': 'Apply floor, ceiling and variation filter'}, {'name': 'floor', 'value': '20', 'label': 'Floor', 'description': 'Value for floor threshold'}, {'name': 'ceiling', 'value': '20000', 'label': 'Ceiling', 'description': 'Value for ceiling threshold'}, {'name': 'min.fold.change', 'value': '3', 'label': 'Min.Fold.Change', 'description': 'Minimum fold change for variation filter'}, {'name': 'min.delta', 'value': '100', 'label': 'Min.Delta', 'description': 'Minimum delta for variation filter'}, {'name': 'num.outliers.to.exclude', 'value': '0', 'label': 'Num.Outliers.To.Exclude', 'description': 'Number of outliers per row to ignore when calculating row min and max for variation filter'}, {'name': 'row.normalization', 'value': '0', 'label': 'Row.Normalization', 'description': 'Perform row normalization'}, {'name': 'row.sampling.rate', 'value': '1', 'label': 'Row.Sampling.Rate', 'description': 'Sample rows without replacement to obtain this fraction of the total number of rows'}, {'name': 'log2.transform', 'value': '0', 'label': 'Log2.Transform', 'description': 'Apply log2 transform after all other preprocessing steps'}, {'name': 'output.file.format', 'value': '3', 'label': 'Output.File.Format', 'description': 'Output file format'}, {'name': 'output.file', 'value': '<input.filename_basename>.preprocessed', 'label': 'Output.File', 'description': 'Output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'PreprocessReadCounts': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Preprocess RNA-Seq count data in a GCT file so that it is suitable for use in GenePattern analyses.', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'A GCT file containing raw RNA-Seq counts, such as is produced by MergeHTSeqCounts'}, {'name': 'cls.file', 'value': '', 'label': 'Cls.File', 'description': 'A categorical CLS file specifying the phenotype classes for the samples in the GCT file.  '}], 'default_args': [{'name': 'task_name', 'value': 'PreprocessReadCounts', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<input.file_basename>.preprocessed.gct', 'label': 'Output.File', 'description': 'Output file name'}, {'name': 'expression.value.filter.threshold', 'value': '1', 'label': 'Expression.Value.Filter.Threshold', 'description': 'Threshold to use when filtering CPM expression values; rows are kept only if the values (in CPM) for all columns are greater than this threshold'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'PreviewFCS': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Allows viewing of structural metadata, parameters, and descriptive statistics from a Flow Cytometry Standard (FCS) data file', 'required_args': [{'name': 'Input.FCS.data.file', 'value': '', 'label': 'Input.FCS.Data.File', 'description': 'An input FCS data file to be described.'}], 'default_args': [{'name': 'task_name', 'value': 'PreviewFCS', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Output.file.format', 'value': 'HTML', 'label': 'Output.File.Format', 'description': 'Output file format, either HTML or XML, default HTML.'}, {'name': 'Output.file.name', 'value': '<Input.FCS.data.file_basename>', 'label': 'Output.File.Name', 'description': 'The output file name (optional).'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Profile.Plots': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Create profile plots for samples', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'Input data file (in gct format)'}, {'name': 'cls.file', 'value': '', 'label': 'Cls.File', 'description': 'File indicating type of samples (sample group) in input.file. Profiles for each group will be plotted in a different color'}], 'default_args': [{'name': 'task_name', 'value': 'Profile.Plots', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<input.file_basename>-profile-plots.pdf', 'label': 'Output.File', 'description': 'Output file name'}, {'name': 'data.type', 'value': 'proteome', 'label': 'Data.Type', 'description': 'Indicate type of input data (proteome, phosphoprotome, etc.) for use in plots'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Protein.mRNA.Corrlation': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Tabulate and plot correlation between proteome and mRNA', 'required_args': [{'name': 'proteome.file', 'value': '', 'label': 'Proteome.File', 'description': 'Proteome data file in gct format'}, {'name': 'mRNA.file', 'value': '', 'label': 'MRNA.File', 'description': 'mRNA (RNA-seq) data file in gct.format'}], 'default_args': [{'name': 'task_name', 'value': 'Protein.mRNA.Corrlation', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.prefix', 'value': '<proteome.file_basename>', 'label': 'Output.Prefix', 'description': 'Filename prefix for results files'}, {'name': 'fdr.threshold', 'value': '0.01', 'label': 'Fdr.Threshold', 'description': 'FDR threshold for marking statistically significant correlations'}, {'name': 'correlation.type', 'value': 'pearson', 'label': 'Correlation.Type', 'description': 'Correlation type to use for calculating protein-mRNA correlation'}, {'name': 'data.type', 'value': 'proteome', 'label': 'Data.Type', 'description': 'Indicate type of input data (proteome, phosphoprotome, etc.) for use in plots'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ProteinDatasetCreation': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Extract features from protein .FASTA files for use with standard prediction algorithms', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'The sequence information in .FASTA format'}], 'default_args': [{'name': 'task_name', 'value': 'ProteinDatasetCreation', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'feature.set.1', 'value': '1', 'label': 'Feature.Set.1', 'description': 'Use amino acid sequence length (1 feature)'}, {'name': 'feature.set.2', 'value': '0', 'label': 'Feature.Set.2', 'description': 'Use amino acid composition (20 features)'}, {'name': 'feature.set.3', 'value': '0', 'label': 'Feature.Set.3', 'description': 'Use reduced alphabet composition of the n terminal region (19 features)'}, {'name': 'feature.set.4', 'value': '0', 'label': 'Feature.Set.4', 'description': 'Use reduced alphabet composition of the c terminal region (19 features)'}, {'name': 'feature.set.5', 'value': '0', 'label': 'Feature.Set.5', 'description': 'Use the reduced alphabet for the gap pairs of amino acids (165 features)'}, {'name': 'output.file', 'value': '<input.file_basename>', 'label': 'Output.File', 'description': 'The output files base name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ProteoArray': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'LC-MS proteomic data processing module', 'required_args': [{'name': 'input.zip.filename', 'value': '', 'label': 'Input.Zip.Filename', 'description': 'zip file of LC-MS runs in mzXML format.'}], 'default_args': [{'name': 'task_name', 'value': 'ProteoArray', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<input.zip.filename_basename>.prot', 'label': 'Output.File', 'description': 'output filename'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ProteomicsAnalysis': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Runs the proteomics analysis on the set of input spectra', 'required_args': [{'name': 'zip.filename', 'value': '', 'label': 'Zip.Filename', 'description': 'Zip file containing spectra data. Spectra data are stored as M/Z,Intensity'}, {'name': 'peak.list.filename', 'value': '', 'label': 'Peak.List.Filename', 'description': 'file containing one M/Z per line when peaks=input'}, {'name': 'random.seed', 'value': '', 'label': 'Random.Seed', 'description': 'random seed when peaks=random'}, {'name': 'random.n.peaks', 'value': '', 'label': 'Random.N.Peaks', 'description': \"number of random M/Z's selected when peaks=random\"}, {'name': 'low.Da', 'value': '', 'label': 'Low.Da', 'description': 'Minimum M/Z to include'}, {'name': 'high.Da', 'value': '', 'label': 'High.Da', 'description': 'Maximum M/Z to include'}], 'default_args': [{'name': 'task_name', 'value': 'ProteomicsAnalysis', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<zip.filename_basename>', 'label': 'Output.File', 'description': 'The base name for output files'}, {'name': 'quality.threshold', 'value': '0.45', 'label': 'Quality.Threshold', 'description': 'discard spectra with quality < quality threshold'}, {'name': 'fill.na', 'value': 'yes', 'label': 'Fill.Na', 'description': 'whether to fill missing peaks'}, {'name': 'normalize', 'value': 'tan', 'label': 'Normalize', 'description': 'normalization strategy'}, {'name': 'peak.detection.method', 'value': 'detect', 'label': 'Peak.Detection.Method', 'description': 'peak detection method'}, {'name': 'filter.peaks', 'value': 'yes', 'label': 'Filter.Peaks', 'description': 'filter peaks based on peak intensity / noise '}, {'name': 'filter.peaks.factor', 'value': '10', 'label': 'Filter.Peaks.Factor', 'description': 'retains peaks if peak intensity >= filter.peaks.factor * stdev(noise)'}, {'name': 'percentile', 'value': '0.65', 'label': 'Percentile', 'description': 'After applying filters, threshold above which peaks are located'}, {'name': 'smoothing.size', 'value': '21', 'label': 'Smoothing.Size', 'description': 'Size of the smoothing filter'}, {'name': 'adaptive.background.correction.strength', 'value': '0.75', 'label': 'Adaptive.Background.Correction.Strength', 'description': 'Strength of the adaptive background correction filter'}, {'name': 'adaptive.background.correction.size', 'value': '21', 'label': 'Adaptive.Background.Correction.Size', 'description': 'Size of the adaptive background correction filter'}, {'name': 'high.pass.filter.strength', 'value': '10', 'label': 'High.Pass.Filter.Strength', 'description': 'Strength of the high pass filter'}, {'name': 'high.pass.filter.factor', 'value': '5', 'label': 'High.Pass.Filter.Factor', 'description': 'Filter factor for the high pass filter'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'QuaSAR': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'A suite of tools for QC, analysis and visualization of data from MRM-MS experiments including assay configuration, calibration curves and new sample analysis.', 'required_args': [{'name': 'skyline.file', 'value': '', 'label': 'Skyline.File', 'description': 'This is a Skyline export in csv file format. The file has to include the following 10 columns with the exact column names given here (in no particular order): SampleName, PeptideSequence, ProteinName, ReplicateName, PrecursorCharge, ProductCharge, FragmentIon,  AverageMeasuredRetentionTime, light Area, heavy Area. The file may contain many more columns, but it has to include at least these 10 columns'}, {'name': 'concentration.file', 'value': '', 'label': 'Concentration.File', 'description': 'Concentration values given in csv file format. The file has to include the following 5 columns with the exact column names given here (in no particular order): SampleName, SampleID, Analyte Concentration, IS Conc, Concentration Ratio. Notice the single space between Analyte Concentration, IS Conc and Concentration Ratio. The space has to be maintained.'}, {'name': 'title', 'value': '', 'label': 'Title', 'description': 'The title is displayed on each calibration plot'}], 'default_args': [{'name': 'task_name', 'value': 'QuaSAR', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'analyte', 'value': 'light Area', 'label': 'Analyte', 'description': 'User must specify the exact analyte column header name as it is appears in the input CSV file. The default is \"light Area\"'}, {'name': 'standard', 'value': 'heavy Area', 'label': 'Standard', 'description': 'User must specify the exact internal standard column header name as it is appears in the input CSV file. The default is \"heavy Area\"'}, {'name': 'generate.cv.table', 'value': '1', 'label': 'Generate.Cv.Table', 'description': 'If \"yes\" then generate CV (coefficient of variation) table'}, {'name': 'generate.calcurves', 'value': '1', 'label': 'Generate.Calcurves', 'description': 'If \"yes\" then generate calibration curves'}, {'name': 'generate.lodloq.table', 'value': '1', 'label': 'Generate.Lodloq.Table', 'description': 'If \"yes\" then generate LOD/LOQ table'}, {'name': 'generate.peak.area.plots', 'value': '0', 'label': 'Generate.Peak.Area.Plots', 'description': 'If \"yes\" then generate peak area plots'}, {'name': 'use.par', 'value': '0', 'label': 'Use.Par', 'description': 'If \"yes\" then use peak area ratio (PAR) for analysis (instead of concentration)'}, {'name': 'max.calcurve.linear.scale', 'value': '150', 'label': 'Max.Calcurve.Linear.Scale', 'description': 'The maximum value for linear scale in fmols/ul'}, {'name': 'max.calcurve.log.scale', 'value': '150', 'label': 'Max.Calcurve.Log.Scale', 'description': 'The maximum value for log scale in fmols/ul'}, {'name': 'perform.audit', 'value': '1', 'label': 'Perform.Audit', 'description': 'If \"yes\" then perform AuDIT for interference detection.'}, {'name': 'audit.cv.threshold', 'value': '0.2', 'label': 'Audit.Cv.Threshold', 'description': 'For AuDIT the threshold for coefficient of variation below which transition is quantification-worthy. Default is 0.2.'}, {'name': 'output.prefix', 'value': '<skyline.file_basename>', 'label': 'Output.Prefix', 'description': 'Prefix used for all output file names'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'RankNormalize': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Normalize the rows in a data set by rank', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'The dataset to be normalized'}, {'name': 'scale.to.value', 'value': '', 'label': 'Scale.To.Value', 'description': 'Result values will be scaled to this value by multiplication after normalization.  Leave this blank to scale to the number of rows in the data set.'}, {'name': 'threshold', 'value': '', 'label': 'Threshold', 'description': 'Minimum threshold for values in the data set.  Any data set value below this will be increased to the threshold value before normalization.  If more than one of these parameters is specified, the order of precedence is threshold, then ceiling, then shift.'}, {'name': 'ceiling', 'value': '', 'label': 'Ceiling', 'description': 'Maximum ceiling for values in the data set.  Any data set value above this will be decreased to the ceiling value before normalization.  If more than one of these parameters is specified, the order of precedence is threshold, then ceiling, then shift.'}, {'name': 'shift', 'value': '', 'label': 'Shift', 'description': 'Shift all values in the data set.  The shift value will be added to all data set values before normalization.  If more than one of these parameters is specified, the order of precedence is threshold, then ceiling, then shift.'}], 'default_args': [{'name': 'task_name', 'value': 'RankNormalize', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file.name', 'value': '<input.file_basename>.NORM.<input.file_extension>', 'label': 'Output.File.Name', 'description': 'The name to be given to the output file.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Read_group_trackingToGct': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '[**Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] Converts a Cufflinks read_group_tracking file into GCT format', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'The read_group_tracking file to be converted into GCT format.'}, {'name': 'gtf.file', 'value': '', 'label': 'Gtf.File', 'description': 'An optional GTF file.  If provided, this should be the same file that you provided to Cuffdiff. You can use this to map tracking IDs in the read_group_tracking file to gene symbols for output in the GCT.  Make sure that you set the feature level to match your read_group_tracking file.'}, {'name': 'override.attribute.name.for.lookup', 'value': '', 'label': 'Override.Attribute.Name.For.Lookup', 'description': 'Use this to override the name of the attribute to search against to look up IDs when mapping with a GTF (gene_id, for example). Type in the attribute name exactly as it appears in the GTF.  The retrieval override must also be provided if this parameter is set.'}, {'name': 'override.attribute.name.for.retrieval', 'value': '', 'label': 'Override.Attribute.Name.For.Retrieval', 'description': 'Use this to override the name of the attribute to retrieve on match when mapping with a GTF (gene_name, for example). Type in the attribute name exactly as it appears in the GTF.  The lookup override must also be provided if this parameter is set.'}], 'default_args': [{'name': 'task_name', 'value': 'Read_group_trackingToGct', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file.name', 'value': '<input.file_basename>.gct', 'label': 'Output.File.Name', 'description': 'The name to be given to the GCT output file.'}, {'name': 'expression.value.column', 'value': '3', 'label': 'Expression.Value.Column', 'description': 'The column to use for extracting expression values.'}, {'name': 'screen.out.errors', 'value': 'false', 'label': 'Screen.Out.Errors', 'description': \"Set to 'yes' to exclude from the GCT file any feature having at least one non-OK quantification status across all of the sequenced samples.\"}, {'name': 'output.value.for.error', 'value': 'NA', 'label': 'Output.Value.For.Error', 'description': \"This parameter controls what expression value to write to a GCT file in those cases where a feature's expression estimate carries a non-OK quantification status.  The parameter is only relevant to your final GCT when 'screen out errors' is set to 'no'.  We strongly recommend treating these as missing values (select 'NA' or 'blank').\"}, {'name': 'feature.level.for.symbol.lookup', 'value': 'gene', 'label': 'Feature.Level.For.Symbol.Lookup', 'description': 'Select the feature level of the read_group_tracking file for mapping gene symbols from a GTF. This is ignored if no GTF is provided.'}, {'name': 'choose.gct.column.for.mapping', 'value': 'auto', 'label': 'Choose.Gct.Column.For.Mapping', 'description': 'Use this to explicitly set the GCT column to use when writing symbols retrieved from the GCT. This is automatic by default, meaning that retrieved symbols go into the Name column for gene-level files and into the Description column for other feature levels.'}, {'name': 'report.naming.conflict.details', 'value': 'false', 'label': 'Report.Naming.Conflict.Details', 'description': 'Include the naming conflict details in the summary report, rather than just giving counts of the issues. '}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ReannotateGCT': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Reannotate a GCT file\\'s \"Description\" column with data from a CHIP file', 'required_args': [{'name': 'gct.file', 'value': '', 'label': 'Gct.File', 'description': 'A GCT file to be annotated'}, {'name': 'chip.file', 'value': '', 'label': 'Chip.File', 'description': 'CHIP file to use for annotations'}], 'default_args': [{'name': 'task_name', 'value': 'ReannotateGCT', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<gct.file_basename>_reannot.gct', 'label': 'Output.File', 'description': 'The name of your re-annotated GCT file. '}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'RemoveCopyNumberOutliers': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Reads rows of a CN file and removes outliers on a sample-by-sample basis using improvised method.', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'A CN file (.cn, .RData, or .txt)'}], 'default_args': [{'name': 'task_name', 'value': 'RemoveCopyNumberOutliers', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'num.probes', 'value': '5', 'label': 'Num.Probes', 'description': 'The number of trailing and leading probes to use to compute the median'}, {'name': 'outlier.probe.replacement', 'value': 'NA', 'label': 'Outlier.Probe.Replacement', 'description': 'The replacement value for the outlier probe'}, {'name': 'multiplicative.threshold', 'value': '4', 'label': 'Multiplicative.Threshold', 'description': 'Multiplicative threshold'}, {'name': 'additive.threshold', 'value': '0.3', 'label': 'Additive.Threshold', 'description': 'Additive threshold'}, {'name': 'output.prefix', 'value': '<input.file_basename>', 'label': 'Output.Prefix', 'description': 'A prefix for the output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'RemoveMissingValues': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Removes features and/or samples containing missing (NA) values', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'Input file (gct, res)'}], 'default_args': [{'name': 'task_name', 'value': 'RemoveMissingValues', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'remove.opt', 'value': '1', 'label': 'Remove.Opt', 'description': 'Remove rows, columns, or both containing missing values'}, {'name': 'output.filename', 'value': '<input.file_basename>.na-rm.gct', 'label': 'Output.Filename', 'description': 'output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'removeSamples': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Remove samples in dataset (.gct), using a list of samples provided', 'required_args': [{'name': 'gct.file', 'value': '', 'label': 'Gct.File', 'description': 'data file to filter'}, {'name': 'removelist', 'value': '', 'label': 'Removelist', 'description': 'list of samples to remove'}, {'name': 'extension', 'value': '', 'label': 'Extension', 'description': 'filename prefix'}], 'default_args': [{'name': 'task_name', 'value': 'removeSamples', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'RemoveSaturatedFCSEvents': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Remove saturated events from an FCS data file', 'required_args': [{'name': 'Input.FCS.data.file', 'value': '', 'label': 'Input.FCS.Data.File', 'description': 'An FCS data file to remove saturated events from.'}], 'default_args': [{'name': 'task_name', 'value': 'RemoveSaturatedFCSEvents', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Output.FCS.file.name', 'value': '<Input.FCS.data.file_basename>.without_saturated.fcs', 'label': 'Output.FCS.File.Name', 'description': 'The output FCS file name.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'RenameFile': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Creates a result file with the contents of the input but with a new name.  Where possible, this is done without copying the file but instead by simply making a link.  While it is possible to run this with a submitted file, the common use will be to change the name of a result file or a previously uploaded file.  Note that the original file will still be present in its original location. ', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'Input file to be renamed'}], 'default_args': [{'name': 'task_name', 'value': 'RenameFile', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.filename', 'value': '<input.file_basename>.cp.<input.file_extension>', 'label': 'Output.Filename', 'description': 'The name to give the output file'}, {'name': 'screen.filename', 'value': 'yes', 'label': 'Screen.Filename', 'description': 'Automatically screen out problematic characters in the filename'}, {'name': 'force.copy', 'value': 'no', 'label': 'Force.Copy', 'description': 'Force a copy to be made rather than a link. '}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ReorderByClass': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Reorder the samples in an expression dataset and class file by class', 'required_args': [{'name': 'expression.data.filename', 'value': '', 'label': 'Expression.Data.Filename', 'description': 'Expression data - .res, .gct, .odf'}, {'name': 'class.filename', 'value': '', 'label': 'Class.Filename', 'description': 'Class file - .cls'}], 'default_args': [{'name': 'task_name', 'value': 'ReorderByClass', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'expression.data.file.output.name', 'value': '<expression.data.filename_basename>_reordered', 'label': 'Expression.Data.File.Output.Name', 'description': 'Name of expression data output file'}, {'name': 'class.file.output.name', 'value': '<class.filename_basename>_reordered', 'label': 'Class.File.Output.Name', 'description': 'Name of class output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ReplicatesQC': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Removes cell line replicate samples that fail quality control check of cell line replicate correlations and distribution scores.', 'required_args': [{'name': 'GCT.files', 'value': '', 'label': 'GCT.Files', 'description': 'Please enter at least one and up to three GCT files.'}, {'name': 'extension', 'value': '', 'label': 'Extension', 'description': ''}], 'default_args': [{'name': 'task_name', 'value': 'ReplicatesQC', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'REVEALER': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'REVEALER (Repeated Evaluation of VariablEs conditionAL Entropy and Redundancy) is an analysis method specifically suited to find groups of genomic alterations that match in a complementary way, a predefined functional activation, dependency of drug response target profile. The method starts by considering, if available, already known genomic alterations (seed) that are the known causes or are known or assumed associated with the target. REVEALER starts by finding the genomic alteration that best matches the target profile conditional to the known seed profile using the conditional mutual information. The newly discovered alteration is then merged with the seed to form a new summary feature, and then the process repeats itself finding additional complementary alterations that explain more and more of the target profile.', 'required_args': [], 'default_args': [{'name': 'task_name', 'value': 'REVEALER', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'ds1', 'value': 'https://personal.broadinstitute.org/tamayo/REVEALER/datasets/CTNBB1_transcriptional_reporter.gct', 'label': 'Ds1', 'description': 'Target profile dataset (e.g. https://personal.broadinstitute.org/tamayo/REVEALER/datasets/CTNBB1_transcriptional_reporter.gct)'}, {'name': 'target.name', 'value': 'BETA-CATENIN-REPORTER', 'label': 'Target.Name', 'description': 'Target profile name (e.g. BETA-CATENIN-REPORTER)'}, {'name': 'target.match', 'value': 'positive', 'label': 'Target.Match', 'description': 'Direction of match (e.g. positive)'}, {'name': 'ds2', 'value': 'https://personal.broadinstitute.org/tamayo/REVEALER/datasets/CCLE_MUT_CNA_AMP_DEL_0.70_2fold.MC.gct', 'label': 'Ds2', 'description': 'Features Dataset (e.g. https://personal.broadinstitute.org/tamayo/REVEALER/datasets/CCLE_MUT_CNA_AMP_DEL_0.70_2fold.MC.gct)'}, {'name': 'seed.names', 'value': 'CTNNB1.MC_MUT', 'label': 'Seed.Names', 'description': 'seed(s) names (e.g. CTNNB1.MC_MUT)'}, {'name': 'max.n.iter', 'value': '2', 'label': 'Max.N.Iter', 'description': 'Maximum number of iterations (e.g. 2)'}, {'name': 'n.markers', 'value': '30', 'label': 'N.Markers', 'description': 'Number of top hits to show in heatmap for every iteration (e.g. 30)'}, {'name': 'locs.table.file', 'value': 'https://personal.broadinstitute.org/tamayo/REVEALER/datasets/hgnc_complete_set.Feb_20_2014.v2.txt', 'label': 'Locs.Table.File', 'description': 'Table with chromosomal location for each gene symbol (e.g. https://personal.broadinstitute.org/tamayo/REVEALER/datasets/hgnc_complete_set.Feb_20_2014.v2.txt)'}, {'name': 'count.thres.low', 'value': '3', 'label': 'Count.Thres.Low', 'description': \"Filter out features with less than count.thres.low 1's (e.g. 3)\"}, {'name': 'count.thres.high', 'value': '50', 'label': 'Count.Thres.High', 'description': \"Filter out features with more than count.thres.low 1's (e.g. 50)\"}, {'name': 'pdf.output.file', 'value': 'BCAT_vs_MUT_CNA.pdf', 'label': 'Pdf.Output.File', 'description': 'Output PDF file with results (e.g. BCAT_vs_MUT_CNA.pdf)'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'RLibDemo': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '', 'required_args': [{'name': 'message', 'value': '', 'label': 'Message', 'description': ''}], 'default_args': [{'name': 'task_name', 'value': 'RLibDemo', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'RNASeQC': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '[**Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] Calculates metrics on aligned RNA-seq data.', 'required_args': [{'name': 'bam.files', 'value': '', 'label': 'Bam.Files', 'description': 'An indexed BAM file or zipped set of indexed BAM files to be analyzed. If you are supplying a single BAM file, it should be located in the same folder as its associated index file (BAI). (If SortSam is used to index the BAM file, then the BAM and BAI files are located in the same folder.)'}, {'name': 'sample.info.file', 'value': '', 'label': 'Sample.Info.File', 'description': 'A TXT format sample info file containing a sample ID, sample file name, and notes column in tab-delimited format. - .txt'}, {'name': 'annotation.gtf', 'value': '', 'label': 'Annotation.Gtf', 'description': \"A file containing annotations in GTF format. Either an annotation gtf or annotation gtf file must be provided.Note that the provided hg19 reference sequence requires contigs using in the '1' format.\"}, {'name': 'annotation.gtf.file', 'value': '', 'label': 'Annotation.Gtf.File', 'description': 'A file containing annotations in GTF format. Either an annotation gtf file or annotation gtf must be provided.'}, {'name': 'reference.sequence', 'value': '', 'label': 'Reference.Sequence', 'description': \"A FASTA file containing a reference sequence. The reference sequence provided here has an index (.fai) and a sequence dictionary (.dict). Either a reference sequence or reference sequence file (with index and dictionary) must be provided, but not both. Note that the provided hg19 reference sequence requires contigs using in the '1' format.\"}, {'name': 'reference.sequence.file', 'value': '', 'label': 'Reference.Sequence.File', 'description': 'A FASTA file containing a reference sequence. The reference sequence must have an index (.fai) and a sequence dictionary (.dict) either in the same location or specified in the reference sequence index and reference sequence dictionary parameters. Reference sequence human_assembly19 is available in the documentation. Click on the \"documentation\" link in the upper right hand corner of this page.'}, {'name': 'reference.sequence.index', 'value': '', 'label': 'Reference.Sequence.Index', 'description': 'A file containing the index for the reference sequence - .fai'}, {'name': 'reference.sequence.dictionary', 'value': '', 'label': 'Reference.Sequence.Dictionary', 'description': 'A file containing the dictionary for the reference sequence. - .dict'}, {'name': 'transcript.type.field', 'value': '', 'label': 'Transcript.Type.Field', 'description': 'Specifies the column of the GTF file in which the transcript type is specified. By default, the module looks for an attribute called transcript_type in the annotation gtf.'}, {'name': 'rRNA.interval.file', 'value': '', 'label': 'RRNA.Interval.File', 'description': 'A file containing the genomic coordinates of rRNA. - .list'}, {'name': 'rRNA.reference.file.index', 'value': '', 'label': 'RRNA.Reference.File.Index', 'description': 'zip of index files created by BWA for the supplied rRNA reference file. Required if rRNA reference file is supplied. -.zip'}, {'name': 'transcript.level.metrics', 'value': '', 'label': 'Transcript.Level.Metrics', 'description': 'Whether to calculate transcript-level metrics in addition to sample-level metrics.'}, {'name': 'gc.content.file', 'value': '', 'label': 'Gc.Content.File', 'description': 'A file containing GC content for each of the transcripts. The file must be tab-delimited with 2 columns containing transcript name and GC content. The transcript name must appear in the GTF file.'}, {'name': 'num.downsampling.reads', 'value': '', 'label': 'Num.Downsampling.Reads', 'description': 'Perform downsampling on the given number of reads. It randomly samples the specified number of reads in all experimental samples when calculating metrics.'}, {'name': 'correlation.comparison.file', 'value': '', 'label': 'Correlation.Comparison.File', 'description': 'A GCT file to use to calculate correlation between expression values (only uses first sample if gct file contains more than one sample. Note also that expression values must be log normalized, and the identifiers must match those of the annotation GTF file)'}, {'name': 'gap.length', 'value': '', 'label': 'Gap.Length', 'description': 'Whether to plot the distribution of gap lengths'}, {'name': 'GATK.flags', 'value': '', 'label': 'GATK.Flags', 'description': 'Pass a string of flags directly to the GATK (e.g. \"-DBQ 0\" to set missing base qualities to zero).'}], 'default_args': [{'name': 'task_name', 'value': 'RNASeQC', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'single.end', 'value': 'yes', 'label': 'Single.End', 'description': 'Whether the bam file contains single end reads. Select no for paired-end reads.'}, {'name': 'num.genes', 'value': '1000', 'label': 'Num.Genes', 'description': 'The number of top-expressed genes for which to calculate metrics.'}, {'name': 'num.rRNA.downsampling.reads', 'value': '1000000', 'label': 'Num.RRNA.Downsampling.Reads', 'description': 'Downsamples to calculate rRNA rate more efficiently. Default is 1 million. Set to 0 to disable.'}, {'name': 'transcript.end.length', 'value': '200', 'label': 'Transcript.End.Length', 'description': \"The length of the 3' or 5' end of a transcript.\"}, {'name': 'gc.margin', 'value': '0.375', 'label': 'Gc.Margin', 'description': 'Used in conjunction with the gc content file to specify the percent gc content to use as boundaries. E.g. 0.25 would set a lower cutoff of 25% and an upper cutoff of 75% (default is 0.375).'}, {'name': 'output.prefix', 'value': '<bam.files_basename>', 'label': 'Output.Prefix', 'description': 'The prefix to use for the output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'RNASeQC_CEGS': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '', 'required_args': [{'name': 'Picard.CreateSequenceDictionary1.reference.sequence.file', 'value': '', 'label': 'Picard.CreateSequenceDictionary1.Reference.Sequence.File', 'description': 'Input reference fasta or fasta.gz '}, {'name': 'TopHat2.reads.pair.1', 'value': '', 'label': 'TopHat2.Reads.Pair.1', 'description': 'Unpaired reads file or first mate for paired reads. A file or zip of files containing reads in FASTA or FASTQ format (bz2 and gz compressed files are supported).'}, {'name': 'TopHat2.reads.pair.2', 'value': '', 'label': 'TopHat2.Reads.Pair.2', 'description': 'Second mate for paired reads. A file or zip of files in FASTA or FASTQ format (bz2 and gz compressed files are supported).'}, {'name': 'Picard.ReorderSam3.reference.file', 'value': '', 'label': 'Picard.ReorderSam3.Reference.File', 'description': 'Reference sequence to which to reorder reads in FASTA format.'}, {'name': 'RNASeQC7.gc.content.file', 'value': '', 'label': 'RNASeQC7.Gc.Content.File', 'description': 'A file containing GC content for each of the transcripts. The file must be tab-delimited with 2 columns containing transcript name and GC content. The transcript name must appear in the GTF file.'}], 'default_args': [{'name': 'task_name', 'value': 'RNASeQC_CEGS', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'SamSPECTRALClusterFCS': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Cluster flow cytometry data using the SamSPECTRAL algorithm.', 'required_args': [{'name': 'Input.FCS.data.file', 'value': '', 'label': 'Input.FCS.Data.File', 'description': 'The FCS file to be clustered.'}, {'name': 'Dimensions', 'value': '', 'label': 'Dimensions', 'description': 'A comma-separated list of dimensions (parameters/channels) that shall be used for clustering. Use either parameter names or indexes. All dimensions except for Time will be used if this parameter is not provided.'}, {'name': 'Dimensions.to.transform', 'value': '', 'label': 'Dimensions.To.Transform', 'description': 'A comma-separated list of dimensions (parameters/channels) that shall transformed as specified by previous parameter. This will be ignored if no transformation is specified above. If this parameter is not provided and transformation is specified above, the algorithm will use heuristics to identify parameters that shall be transformed. These heuristics are based on how parameters are stored in the FCS file, their resolution and their name. Again, you can use either parameter names or parameter indexes to specify dimensions to transform.'}, {'name': 'Sigma', 'value': '', 'label': 'Sigma', 'description': 'A scaling parameter that determines the resolution in the spectral clustering stage. By increasing it, more spectral clusters are identified. This can be usefull when small population are aimed. See documentation on more details how Sigma can be adjusted.'}, {'name': 'Separation.factor', 'value': '', 'label': 'Separation.Factor', 'description': 'Threshold that controls to what extend clusters should be combined or kept separate. See documentation on more details how the separation factor can be adjusted.'}, {'name': 'Seed', 'value': '', 'label': 'Seed', 'description': 'Seed value for the random generator; provide any consistent value to obtain reproducible results.'}], 'default_args': [{'name': 'task_name', 'value': 'SamSPECTRALClusterFCS', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Transformation', 'value': 'ASinH', 'label': 'Transformation', 'description': 'Which transformation to apply on the data prior clustering. Fluorescence channels are usually better visualized and clustered using a transformation; the ASinH transformation produces good results on most data.'}, {'name': 'Beta', 'value': '4', 'label': 'Beta', 'description': 'The size of neighborhood that is being searched during the conductance calculation step. The default value of 4 should not be changed unless the input data set contains measurements of more that 100,000. Setting beta to zero or a low value will reduce computational time by applying an approximation to the conductance calculation step. This will allow you to cluster large data sets in reasonable time; however, the quality of the result may be affected. Setting beta to 5 is not recommended since it will increase computational time while not significantly improving clustering results in most cases.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'SAMtools.FastaIndex': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Indexes a reference sequence in FASTA format', 'required_args': [{'name': 'fasta.file', 'value': '', 'label': 'Fasta.File', 'description': 'A sequence in FASTA format.'}], 'default_args': [{'name': 'task_name', 'value': 'SAMtools.FastaIndex', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.prefix', 'value': '<fasta.file_basename>', 'label': 'Output.Prefix', 'description': 'The prefix to use for the output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Scripture': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Segmentation-based transcript reconstruction algorithm', 'required_args': [{'name': 'alignment.file', 'value': '', 'label': 'Alignment.File', 'description': 'Alignment file in SAM or BAM format, sorted by start position'}, {'name': 'alignment.index.file', 'value': '', 'label': 'Alignment.Index.File', 'description': 'Index file for alignment file, in SAI or BAI format. The index file must have the same root name as the alignment file. (Example: foo.sam and foo.sam.sai OR foo.sai) '}, {'name': 'chromosome.size.file', 'value': '', 'label': 'Chromosome.Size.File', 'description': 'File containing sizes of chromosomes'}, {'name': 'chromosome', 'value': '', 'label': 'Chromosome', 'description': 'Selected chromosome, e.g., chr19'}, {'name': 'chromosome.sequence.file', 'value': '', 'label': 'Chromosome.Sequence.File', 'description': 'Chromosome sequence in FASTA format'}], 'default_args': [{'name': 'task_name', 'value': 'Scripture', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.prefix', 'value': '<alignment.file_basename>', 'label': 'Output.Prefix', 'description': 'A label that will be used to name output files'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ScripturePipeline': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Segmentation-based transcript reconstruction using single- or paired-end reads', 'required_args': [{'name': 'TopHat1.prebuilt.bowtie.index', 'value': '', 'label': 'TopHat1.Prebuilt.Bowtie.Index', 'description': 'A prebuilt bowtie index. Either a prebuilt or a custom Bowtie index must be specified.'}, {'name': 'TopHat1.custom.bowtie.index', 'value': '', 'label': 'TopHat1.Custom.Bowtie.Index', 'description': 'A ZIP archive containing Bowtie index files. Either a prebuilt or a custom Bowtie index must be specified.'}, {'name': 'TopHat1.reads.pair.1', 'value': '', 'label': 'TopHat1.Reads.Pair.1', 'description': 'Unpaired reads file or first mate for paired reads. A file or zip of files containing reads in FASTA or FASTQ format.'}, {'name': 'TopHat1.reads.pair.2', 'value': '', 'label': 'TopHat1.Reads.Pair.2', 'description': 'Second mate for paired reads. A file or zip of files in FASTA or FASTQ format.'}, {'name': 'TopHat1.mate.inner.dist', 'value': '', 'label': 'TopHat1.Mate.Inner.Dist', 'description': 'The expected mean inner distance between mate pairs.'}, {'name': 'TopHat1.mate.std.dev', 'value': '', 'label': 'TopHat1.Mate.Std.Dev', 'description': 'The standard deviation for the distribution on inner distances between mate pairs.'}, {'name': 'TopHat1.library.type', 'value': '', 'label': 'TopHat1.Library.Type', 'description': 'Library type for strand specific reads.'}, {'name': 'Scripture3.chromosome.size.file', 'value': '', 'label': 'Scripture3.Chromosome.Size.File', 'description': 'File containing sizes of chromosomes'}, {'name': 'Scripture3.chromosome', 'value': '', 'label': 'Scripture3.Chromosome', 'description': 'Selected chromosome, e.g., chr19'}, {'name': 'Scripture3.chromosome.sequence.file', 'value': '', 'label': 'Scripture3.Chromosome.Sequence.File', 'description': 'Chromosome sequence in FASTA format'}], 'default_args': [{'name': 'task_name', 'value': 'ScripturePipeline', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'TopHat1.integer.quals', 'value': 'no', 'label': 'TopHat1.Integer.Quals', 'description': 'Quality values are space-delimited integer values, this becomes default when you specify color option.'}, {'name': 'Scripture3.output.prefix', 'value': '<alignment.file_basename>', 'label': 'Scripture3.Output.Prefix', 'description': 'A label that will be used to name output files'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ScripturePrealigned': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Segmentation-based transcript reconstruction using pre-aligned reads', 'required_args': [{'name': 'SortSam1.input.file', 'value': '', 'label': 'SortSam1.Input.File', 'description': 'Input file in SAM or BAM format'}, {'name': 'Scripture2.chromosome.size.file', 'value': '', 'label': 'Scripture2.Chromosome.Size.File', 'description': 'File containing sizes of chromosomes'}, {'name': 'Scripture2.chromosome', 'value': '', 'label': 'Scripture2.Chromosome', 'description': 'Selected chromosome, e.g., chr19'}, {'name': 'Scripture2.chromosome.sequence.file', 'value': '', 'label': 'Scripture2.Chromosome.Sequence.File', 'description': 'Chromosome sequence in FASTA format'}], 'default_args': [{'name': 'task_name', 'value': 'ScripturePrealigned', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Scripture2.output.prefix', 'value': '<alignment.file_basename>', 'label': 'Scripture2.Output.Prefix', 'description': 'A label that will be used to name output files'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'SegmentationCount': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Counts the number of segments in a SEG file.', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'A segmentation file - .seg, .txt'}], 'default_args': [{'name': 'task_name', 'value': 'SegmentationCount', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<input.file_basename>.segcount.txt', 'label': 'Output.File', 'description': 'The name of the output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'SelectFeaturesColumns': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': \"Takes a 'column slice' from a .res, .gct, .snp, .cn, .loh, .odf, or .cls file.\", 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'input file - .res, .gct, .snp, .cn, .loh, .odf, or .cls file from which the slice is extracted'}, {'name': 'columns', 'value': '', 'label': 'Columns', 'description': 'columns to select (e.g. 0-3, 5, 9). Either columns or a list file must be provided.'}, {'name': 'list.filename', 'value': '', 'label': 'List.Filename', 'description': 'text file containing one identifier per line that specifies the columns to select. Either columns or a list file must be provided.'}], 'default_args': [{'name': 'task_name', 'value': 'SelectFeaturesColumns', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output', 'value': '<input.filename_basename>.slice', 'label': 'Output', 'description': 'output filename - .res, .gct, .snp, .cn, .loh, .odf, .cls'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'SelectFeaturesRows': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Generate a .res, .gct, .snp, .cn, .loh, or .odf file containing only specified features (i.e. rows).', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'input file - .res, .gct, .snp, .cn, .loh, or .odf file from which the slice is extracted'}, {'name': 'list.filename', 'value': '', 'label': 'List.Filename', 'description': 'text file containing one identifier per line that specifies the rows to select. '}], 'default_args': [{'name': 'task_name', 'value': 'SelectFeaturesRows', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output', 'value': '<list.filename_basename>.slice', 'label': 'Output', 'description': 'output filename - .res, .gct, .snp, .cn, .loh, .odf'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'SelectFileMatrix': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'A helper module which selects a matrix out of a delimited text file.', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'Name of the input file from which to pull the matrix.  Must be a delimited text file.'}, {'name': 'start.row', 'value': '', 'label': 'Start.Row', 'description': 'Starting row of the matrix (line number within the file).  Defaults to the first row.'}, {'name': 'end.row', 'value': '', 'label': 'End.Row', 'description': 'Ending row of the matrix (line number within the file).  Defaults to the last row.'}, {'name': 'start.column', 'value': '', 'label': 'Start.Column', 'description': 'Starting column of the matrix.  Defaults to the first column.'}, {'name': 'end.column', 'value': '', 'label': 'End.Column', 'description': 'Ending column of the matrix. Defaults to the last column.'}, {'name': 'row.select', 'value': '', 'label': 'Row.Select', 'description': 'Select specific rows to include in the matrix (e.g. \"3,5,7-12\").  Note that start/end row parameters may limit this selection.'}, {'name': 'column.select', 'value': '', 'label': 'Column.Select', 'description': 'Select specific columns to include in the matrix (e.g. \"1,2,4-6\").  Note that start/end column parameters may limit this selection'}], 'default_args': [{'name': 'task_name', 'value': 'SelectFileMatrix', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file.base.name', 'value': '<input.file_basename>.select', 'label': 'Output.File.Base.Name', 'description': 'Base name for the output file.  The module will add an appropriate extension, either .csv for a comma-delimted file or .txt for a space- or tab-delimited file).'}, {'name': 'delimiter', 'value': 'TAB', 'label': 'Delimiter', 'description': 'Token naming the character for the output column delimiter. Choices are TAB, COMMA, and SPACE. TAB is the default.'}, {'name': 'output.rejects', 'value': 'false', 'label': 'Output.Rejects', 'description': 'Produce an output file containing the items not meeting the selection criteria.  The file will be named <output.file.base.name>.REJECT.<ext>'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'SetFCSKeywords': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Sets keyword/value(s) in a Flow Cytometry Standard (FCS) file.', 'required_args': [{'name': 'Input.FCS.data.file', 'value': '', 'label': 'Input.FCS.Data.File', 'description': 'The input FCS data file in which to set keyword/value pairs.'}, {'name': 'Keywords.input.file', 'value': '', 'label': 'Keywords.Input.File', 'description': 'The file with keyword/value pairs to add (or change values of existing keywords). The keyword/value pairs can be submitted either as an CSV file with two columns: \"Keyword\" and \"Value\" (heading is optional) or as a text file with a chunk of FCS TEXT segment, e.g., /$P3S/MUC 1 APC/$P4S/Thy1 PE/BD$CUSTOM_VALUE23/12/.'}, {'name': 'Output.fcs.file.name', 'value': '', 'label': 'Output.Fcs.File.Name', 'description': 'The file name for the output FCS data file.'}], 'default_args': [{'name': 'task_name', 'value': 'SetFCSKeywords', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'shRNAcollapseReps': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Collapses replicate samples in a gct data file, based on a sample information file', 'required_args': [{'name': 'gct.file', 'value': '', 'label': 'Gct.File', 'description': 'The input gct data file.'}, {'name': 'sample.info.file', 'value': '', 'label': 'Sample.Info.File', 'description': 'The cell line (sample) information file.'}, {'name': 'prefix', 'value': '', 'label': 'Prefix', 'description': 'The file name prefix.'}], 'default_args': [{'name': 'task_name', 'value': 'shRNAcollapseReps', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'shRNAfoldChange': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Calculate the fold change of each sample in a gct file, based on the appropriate DNA reference value (present in a gct file of the same dimension).', 'required_args': [{'name': 'gct.file', 'value': '', 'label': 'Gct.File', 'description': 'The input gct data file.'}, {'name': 'DNA.reference.file', 'value': '', 'label': 'DNA.Reference.File', 'description': 'The values of the DNA pool reference per sample/shRNA in a gct file (same dimension as input gct file).'}, {'name': 'prefix', 'value': '', 'label': 'Prefix', 'description': 'The file name prefix.'}], 'default_args': [{'name': 'task_name', 'value': 'shRNAfoldChange', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'shRNAmapGenes': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Maps shRNAs in a gct data file to genes based on a mapping file.', 'required_args': [{'name': 'gct.file', 'value': '', 'label': 'Gct.File', 'description': 'The input gct data file.'}, {'name': 'mapping', 'value': '', 'label': 'Mapping', 'description': 'The shRNA to gene mapping file.'}, {'name': 'prefix', 'value': '', 'label': 'Prefix', 'description': 'The file name prefix.'}], 'default_args': [{'name': 'task_name', 'value': 'shRNAmapGenes', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'shRNAremoveOverlap': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Remove shRNAs overlapping in sequence > 3 basepairs from a gct data file', 'required_args': [{'name': 'list', 'value': '', 'label': 'List', 'description': 'The list of shRNAs that are not overlapping.'}, {'name': 'gct.file', 'value': '', 'label': 'Gct.File', 'description': 'The input gct data file.'}, {'name': 'mapping', 'value': '', 'label': 'Mapping', 'description': 'The shRNA to gene mapping file.'}, {'name': 'prefix', 'value': '', 'label': 'Prefix', 'description': 'The file name prefix.'}], 'default_args': [{'name': 'task_name', 'value': 'shRNAremoveOverlap', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'SNPFileCreator': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Process Affymetrix SNP probe-level data into an expression value', 'required_args': [{'name': 'chip.type', 'value': '', 'label': 'Chip.Type', 'description': 'Specify the chip type.'}, {'name': 'network.directory.name', 'value': '', 'label': 'Network.Directory.Name', 'description': 'Networked directory name where the CEL ( and optionally TXT) files reside.'}, {'name': 'zip.filename', 'value': '', 'label': 'Zip.Filename', 'description': 'A zip file containing CEL and optionally TXT files.'}, {'name': 'reference.filename', 'value': '', 'label': 'Reference.Filename', 'description': 'Reference file to use. Leave blank if reference method is median.'}, {'name': 'output.file', 'value': '', 'label': 'Output.File', 'description': 'The name of the output file.'}], 'default_args': [{'name': 'task_name', 'value': 'SNPFileCreator', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'normalization.method', 'value': '2', 'label': 'Normalization.Method', 'description': 'The normalization method to apply.'}, {'name': 'reference.method', 'value': 'median', 'label': 'Reference.Method', 'description': 'Reference method to use in normalization'}, {'name': 'model.method', 'value': '2', 'label': 'Model.Method', 'description': 'The summarization method to apply.'}, {'name': 'allele.specific', 'value': 'no', 'label': 'Allele.Specific', 'description': 'Create Allele Specific probe intensities'}, {'name': 'sort.snp.file', 'value': 'yes', 'label': 'Sort.Snp.File', 'description': 'Sorts output file by Chromosome and location. Filters out probesets without location information'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'SNPFileSorter': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Sorts a .snp file by Chromosome and location', 'required_args': [{'name': 'snp.filename', 'value': '', 'label': 'Snp.Filename', 'description': 'snp file to sort'}], 'default_args': [{'name': 'task_name', 'value': 'SNPFileSorter', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<snp.filename_basename>.sorted.snp', 'label': 'Output.File', 'description': 'output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'SNPMultipleSampleAnalysis': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Determine Regions of Concordant Copy Number Aberrations', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'The input file to analyze - .cn, .xcn, .snp, .txt'}, {'name': 'run.name', 'value': '', 'label': 'Run.Name', 'description': 'The name of the job'}], 'default_args': [{'name': 'task_name', 'value': 'SNPMultipleSampleAnalysis', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'java.max.memory', 'value': '512m', 'label': 'Java.Max.Memory', 'description': ''}, {'name': 'save.dir', 'value': './', 'label': 'Save.Dir', 'description': 'Directory to save to'}, {'name': 'permutations', 'value': '500', 'label': 'Permutations', 'description': 'Number of permutations to use'}, {'name': 'num.tests', 'value': '9', 'label': 'Num.Tests', 'description': 'Number of MSA tests to use'}, {'name': 'num.positions.bin', 'value': '6', 'label': 'Num.Positions.Bin', 'description': 'Number of positions per bin'}, {'name': 'resolution', 'value': '250000', 'label': 'Resolution', 'description': 'The size of the plotted chromosome diagram'}, {'name': 'chromosomes', 'value': 'chr22', 'label': 'Chromosomes', 'description': 'Comma delimited list of chromosomes to analyze'}, {'name': 'save.images', 'value': './', 'label': 'Save.Images', 'description': 'Saves images to the specified directory'}, {'name': 'significance.level', 'value': '.05', 'label': 'Significance.Level', 'description': 'The value at which a region is called significant'}, {'name': 'analyze.by', 'value': 'false', 'label': 'Analyze.By', 'description': ''}, {'name': 'cytoband.filename', 'value': '<libdir>CytoBands.txt', 'label': 'Cytoband.Filename', 'description': 'Cytoband file to use'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'SnpViewer': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Displays a heat map of snp data', 'required_args': [{'name': 'dataset.filename', 'value': '', 'label': 'Dataset.Filename', 'description': 'The dataset to display - .snp, .cn, .xcn'}, {'name': 'sample.info.filename', 'value': '', 'label': 'Sample.Info.Filename', 'description': 'Sample information file'}, {'name': 'sample.list.filename', 'value': '', 'label': 'Sample.List.Filename', 'description': 'Sample list file'}], 'default_args': [{'name': 'task_name', 'value': 'SnpViewer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'max.memory', 'value': '700m', 'label': 'Max.Memory', 'description': 'Maximum memory'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'SOMClustering': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Self-Organizing Maps algorithm', 'required_args': [{'name': 'dataset.filename', 'value': '', 'label': 'Dataset.Filename', 'description': 'Dataset (res, gct, or odf dataset)'}], 'default_args': [{'name': 'task_name', 'value': 'SOMClustering', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.stub', 'value': '<dataset.filename_basename>', 'label': 'Output.Stub', 'description': 'The base output file name - .odf'}, {'name': 'cluster.range', 'value': '2-3', 'label': 'Cluster.Range', 'description': 'Range of values can be entered and the program will automatically run the algorithm for each number of clusters in the range.'}, {'name': 'seed.range', 'value': '42', 'label': 'Seed.Range', 'description': 'The seed for the random number generator, is exposed to allow the user to recreate a given session at a later time, (as opposed to allowing the program to generate a random initial seed which could result in different outcomes even if all the other parameters are identical).'}, {'name': 'iterations', 'value': '50000', 'label': 'Iterations', 'description': 'How many times the algorithm should try to refine the clusters. Initially, this value can be set low for faster exploration, but should be set high (e.g., 50,000) for good convergence.'}, {'name': 'cluster.by', 'value': 'rows', 'label': 'Cluster.By', 'description': 'Whether to cluster by rows or columns'}, {'name': 'som.rows', 'value': '0', 'label': 'Som.Rows', 'description': 'Setting this and som.cols to a non zero value will override cluster.range and the computation will be for the specified geometry'}, {'name': 'som.cols', 'value': '0', 'label': 'Som.Cols', 'description': 'Setting this and som.rows to a non zero value will override cluster.range and the computation will be for the specified geometry'}, {'name': 'initialization', 'value': 'Random_Vectors', 'label': 'Initialization', 'description': 'How to select inital random centroids'}, {'name': 'neighborhood', 'value': 'Bubble', 'label': 'Neighborhood', 'description': 'Neighborhood function determining how centroids near the target centroid are updated'}, {'name': 'alpha.initial', 'value': '0.1', 'label': 'Alpha.Initial', 'description': 'Initial learning weight for centroid updates'}, {'name': 'alpha.final', 'value': '0.005', 'label': 'Alpha.Final', 'description': 'Final learning weight for centroid updates'}, {'name': 'sigma.initial', 'value': '5.0', 'label': 'Sigma.Initial', 'description': 'Initial sigma to determine update neighborhood size around the target centroid'}, {'name': 'sigma.final', 'value': '0.5', 'label': 'Sigma.Final', 'description': 'Final sigma to determine update neighborhood size around the target centroid'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'SOMClusterViewer': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Visualize clusters created with the SOM algorithm', 'required_args': [{'name': 'som.cluster.filename', 'value': '', 'label': 'Som.Cluster.Filename', 'description': 'SOM Properties ODF file'}], 'default_args': [{'name': 'task_name', 'value': 'SOMClusterViewer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'SparseHierarchicalClustering': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Cluster the samples in a data matrix using an adaptively-chosen subset of the features.', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'A GCT file containing data to cluster'}], 'default_args': [{'name': 'task_name', 'value': 'SparseHierarchicalClustering', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'method', 'value': 'average', 'label': 'Method', 'description': 'The type of linkage to use in the hierarchical clustering - \"single\", \"complete\", \"centroid\", or \"average.\"'}, {'name': 'wbound', 'value': '-1', 'label': 'Wbound', 'description': 'The sum of gene weights; this is the tuning parameter for sparse hierarchical clustering. If a non-negative number is given, then this tuning parameter is used. Otherwise, a tuning parameter is selected via a permutation approach. Note: The program runs much more quickly if wbound is specified rather than left at -1.'}, {'name': 'maxnumgenes', 'value': '5000', 'label': 'Maxnumgenes', 'description': 'By default, only the 5000 genes with highest variance will be used (for computational reasons). To use all the genes, set to -1. Warning: using all genes will be slow if there are many genes in the data set.'}, {'name': 'cluster.features', 'value': 'false', 'label': 'Cluster.Features', 'description': 'If a clustering of the genes with non-zero weights is desired, set to TRUE.'}, {'name': 'method.features', 'value': 'average', 'label': 'Method.Features', 'description': 'The type of linkage used to cluster the features (if desired) - \"single\", \"complete\", \"centroid\", or \"average\". Used only if cluster features is set to \"TRUE\"'}, {'name': 'standardize.arrays', 'value': 'true', 'label': 'Standardize.Arrays', 'description': 'Whether or not the arrays should be standardized to have mean zero and standard deviation one before clustering is performed.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'SplitColumns': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Creates a separate file for each column specified from the input file', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'A tab delimited file'}, {'name': 'split.columns', 'value': '', 'label': 'Split.Columns', 'description': 'Specify columns to extract (i.e. 2,4, 5-, 11-17,). Leave blank to split all columns.'}, {'name': 'additional.columns', 'value': '', 'label': 'Additional.Columns', 'description': 'Specify subset of additional columns to include in each (i.e. 2,4, 5-, 11-17) output file.'}], 'default_args': [{'name': 'task_name', 'value': 'SplitColumns', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.prefix', 'value': '<input.file_basename>', 'label': 'Output.Prefix', 'description': 'The prefix to use for the output file name.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'SplitDatasetTrainTest': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Splits a dataset (and cls file) into a number of train and test subsets', 'required_args': [{'name': 'input.dataset.file', 'value': '', 'label': 'Input.Dataset.File', 'description': 'input dataset - .gct, .res'}, {'name': 'cls.file', 'value': '', 'label': 'Cls.File', 'description': 'class template - .cls'}], 'default_args': [{'name': 'task_name', 'value': 'SplitDatasetTrainTest', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'split.method', 'value': 'random', 'label': 'Split.Method', 'description': 'whether to split the data using a percentage split or cross validation approach'}, {'name': 'stratified', 'value': 'T', 'label': 'Stratified', 'description': 'whether to create splits stratified with respect to the class template'}, {'name': 'folds', 'value': '1', 'label': 'Folds', 'description': 'number of train/test splits to generate'}, {'name': 'percentage.split.proportion', 'value': '0.5', 'label': 'Percentage.Split.Proportion', 'description': 'proportion of data to be allocated to train file when split method is percentage split (ignored when split method is cross-validation)'}, {'name': 'seed', 'value': '123456', 'label': 'Seed', 'description': 'random number generator seed'}, {'name': 'output.prefix', 'value': '<input.dataset.file_basename>', 'label': 'Output.Prefix', 'description': 'output saved to output.prefix.{train,test}.n.{gct|res,cls}, where n is the fold index'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'SraToFastQ': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Converts data in SRA format to FastQ using the SRAToolkit version 2.3.5-2.', 'required_args': [{'name': 'accession', 'value': '', 'label': 'Accession', 'description': 'A Sequence Read Archive accession.  Either an accession or an sra.file is required.'}, {'name': 'sra.file', 'value': '', 'label': 'Sra.File', 'description': 'A Sequence Read Archive (.sra) file.  Either an accession or an sra.file is required.'}, {'name': 'paired.end', 'value': '', 'label': 'Paired.End', 'description': 'Whether the reads are paired end'}, {'name': 'quality.score.offset', 'value': '', 'label': 'Quality.Score.Offset', 'description': 'Offset to use for ASCII quality scores. Default is 33 (\"!\").'}, {'name': 'biological.reads.only', 'value': '', 'label': 'Biological.Reads.Only', 'description': 'Whether to include only biological reads and no technical reads in output'}], 'default_args': [{'name': 'task_name', 'value': 'SraToFastQ', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'alignment.status', 'value': 'any', 'label': 'Alignment.Status', 'description': 'The alignment status of the reads to include in output'}, {'name': 'output.compression', 'value': 'none', 'label': 'Output.Compression', 'description': 'Compression to apply to the output files'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ssGSEAProjection': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Performs single sample GSEA Projection', 'required_args': [{'name': 'input.gct.file', 'value': '', 'label': 'Input.Gct.File', 'description': 'Input gct file.  This module does not allow missing values.'}, {'name': 'output.file.prefix', 'value': '', 'label': 'Output.File.Prefix', 'description': 'Output filename prefix.  If unspecified, is set to <prefix of input.gct.file>.PROJ.'}, {'name': 'gene.sets.database.files', 'value': '', 'label': 'Gene.Sets.Database.Files', 'description': 'Gene sets description file(s) - .gmt, .gmx.  Upload a gene set if your gene set is not listed as a choice for the gene sets database parameter.'}], 'default_args': [{'name': 'task_name', 'value': 'ssGSEAProjection', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'gene.sets.database', 'value': 'c1.all.v5.2.symbols.gmt', 'label': 'Gene.Sets.Database', 'description': 'Gene sets database from GSEA website.'}, {'name': 'gene.symbol.column', 'value': 'Name', 'label': 'Gene.Symbol.Column', 'description': 'Specifies the gct data column containing gene symbols.  In most cases this will be \"Column 1\".'}, {'name': 'gene.set.selection', 'value': 'ALL', 'label': 'Gene.Set.Selection', 'description': 'Comma-separated list of gene set names on which to project expression data. Set to \"ALL\" if projecting to all defined gene sets.'}, {'name': 'sample.normalization.method', 'value': 'rank', 'label': 'Sample.Normalization.Method', 'description': 'Method used to normalize gene expression data'}, {'name': 'weighting.exponent', 'value': '0.75', 'label': 'Weighting.Exponent', 'description': 'Module authors strongly recommend against changing from default.'}, {'name': 'min.gene.set.size', 'value': '10', 'label': 'Min.Gene.Set.Size', 'description': 'Gene sets with overlap smaller than this are excluded from the analysis.'}, {'name': 'combine.mode', 'value': 'combine.add', 'label': 'Combine.Mode', 'description': 'Options for combining enrichment scores for paired *_UP and *_DN gene sets'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'StrandSpecRNAseqEval': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Evaluates strand-specific yeast (S. cerevisiae) RNA-seq libraries', 'required_args': [{'name': 'sam.file', 'value': '', 'label': 'Sam.File', 'description': 'The input SAM file to read and evaluate - .sam'}], 'default_args': [{'name': 'task_name', 'value': 'StrandSpecRNAseqEval', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'average.gene.stats', 'value': '0', 'label': 'Average.Gene.Stats', 'description': 'A flag to indicate whether to run the average gene statistics. This is a long process.'}, {'name': 'flip.strands', 'value': '0', 'label': 'Flip.Strands', 'description': 'A flag to indicate whether the data is flipped, so a forward gene would appear as a reverse one'}, {'name': 'output.prefix', 'value': '<sam.file_basename>', 'label': 'Output.Prefix', 'description': 'A prefix for the output files'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'SubMap': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Maps subclasses between two data sets', 'required_args': [{'name': 'datasetA.file', 'value': '', 'label': 'DatasetA.File', 'description': 'Input dataset A - .gct, should have common gene ID with dataset B'}, {'name': 'datasetB.file', 'value': '', 'label': 'DatasetB.File', 'description': 'Input dataset B - .gct,  should have common gene ID with dataset A'}, {'name': 'classA.file', 'value': '', 'label': 'ClassA.File', 'description': 'Input class label A - .cls, 3rd line should be numeric, +1 if starts from \"0\"'}, {'name': 'classB.file', 'value': '', 'label': 'ClassB.File', 'description': 'Input class label B - .cls, 3rd line should be numeric, +1 if starts from \"0\"'}], 'default_args': [{'name': 'task_name', 'value': 'SubMap', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'num.marker.genes', 'value': '100', 'label': 'Num.Marker.Genes', 'description': 'Number of marker genes to be mapped'}, {'name': 'num.perm', 'value': '100', 'label': 'Num.Perm', 'description': 'Number of random permutations for ES in each cell of SA matrix'}, {'name': 'num.perm.fisher', 'value': '1000', 'label': 'Num.Perm.Fisher', 'description': \"Number of random permutations for Fisher's statistics\"}, {'name': 'weighted.score.type', 'value': '1', 'label': 'Weighted.Score.Type', 'description': 'Weight enrichment by correlation vector (SNR)'}, {'name': 'null.distribution', 'value': 'pool', 'label': 'Null.Distribution', 'description': 'Choice of null distribution'}, {'name': 'p.value.correction', 'value': 'Bonferroni', 'label': 'P.Value.Correction', 'description': 'p-value correction method'}, {'name': 'cluster.rows', 'value': '1', 'label': 'Cluster.Rows', 'description': \"Cluster dataset A's subclass in heatmap of SA matrix\"}, {'name': 'cluster.columns', 'value': '1', 'label': 'Cluster.Columns', 'description': \"Cluster dataset B's subclass in heatmap of SA matrix\"}, {'name': 'nominal.p.value.matrix', 'value': 'T', 'label': 'Nominal.P.Value.Matrix', 'description': 'Create heatmap of each nominal-p matrix'}, {'name': 'create.legend', 'value': 'T', 'label': 'Create.Legend', 'description': 'Create legend for heatmap'}, {'name': 'random.seed', 'value': '47365321', 'label': 'Random.Seed', 'description': 'Random seed'}, {'name': 'output.filename', 'value': 'SubMap', 'label': 'Output.Filename', 'description': 'Name of output files containing heatmap of SA matrices, summary of ES matrix, nominal p-values, and corrected p-values'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'SuggestNumberOfPopulationsFCS': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'The FlowClust module requires that you specify how many populations you are looking for.  This module takes a data driven approach and attempts to discover how many populations are present in an FCS file', 'required_args': [{'name': 'Input.FCS.data.file', 'value': '', 'label': 'Input.FCS.Data.File', 'description': 'The FCS file'}, {'name': 'Dimensions', 'value': '', 'label': 'Dimensions', 'description': 'The channels to search for populations, separated by commas. Leave blank to search in all channels. E.g. FSC Lin, SS Lin, CD4 Log'}, {'name': 'Range', 'value': '', 'label': 'Range', 'description': 'By default, we try between 1 and 30 populations and report the most likely number in the file. As this is a time consuming procedure, you can specify a more limited range. e.g. 1,12'}], 'default_args': [{'name': 'task_name', 'value': 'SuggestNumberOfPopulationsFCS', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'SurvivalCurve': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Draws a survival curve based on a cls file', 'required_args': [{'name': 'input.surv.data.file', 'value': '', 'label': 'Input.Surv.Data.File', 'description': 'input clinical data including survival time & censor status -.txt'}, {'name': 'input.cls.file', 'value': '', 'label': 'Input.Cls.File', 'description': 'Class information file -.cls'}, {'name': 'cls.field', 'value': '', 'label': 'Cls.Field', 'description': 'Name of class field in survival data file (optional)'}, {'name': 'manual.line.type', 'value': '', 'label': 'Manual.Line.Type', 'description': 'e.g. 1,2,3,... (1: solid, 2: dashed, 3: dotted, 4: dot-dashed)'}, {'name': 'manual.line.color', 'value': '', 'label': 'Manual.Line.Color', 'description': 'e.g. black, blue, red, ...'}, {'name': 'max.time', 'value': '', 'label': 'Max.Time', 'description': 'Maximum survival time (in original scale)'}], 'default_args': [{'name': 'task_name', 'value': 'SurvivalCurve', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'time.field', 'value': 'time', 'label': 'Time.Field', 'description': 'Name of survival time field'}, {'name': 'censor.field', 'value': 'status', 'label': 'Censor.Field', 'description': 'Name of survival time field (value should be \"event:1, censor:0\")'}, {'name': 'print.fit.results', 'value': 'F', 'label': 'Print.Fit.Results', 'description': 'Output file name for survival fit results'}, {'name': 'line.type.color.assign', 'value': 'automatic', 'label': 'Line.Type.Color.Assign', 'description': 'Assign line type/color automatically/manually?'}, {'name': 'line.width', 'value': '1', 'label': 'Line.Width', 'description': 'Line width'}, {'name': 'time.conversion', 'value': '1', 'label': 'Time.Conversion', 'description': 'Survival time conversion'}, {'name': 'surv.function.lower', 'value': '0', 'label': 'Surv.Function.Lower', 'description': 'Lower limit of y axis range'}, {'name': 'surv.function.higher', 'value': '1', 'label': 'Surv.Function.Higher', 'description': 'Upper limit of y axis range'}, {'name': 'curve.type', 'value': 'log', 'label': 'Curve.Type', 'description': 'Type of survival curve'}, {'name': 'show.conf.interval', 'value': '0', 'label': 'Show.Conf.Interval', 'description': 'Show confidence inteval?'}, {'name': 'add.legend', 'value': 'T', 'label': 'Add.Legend', 'description': 'Show legend?'}, {'name': 'legend.position', 'value': 'left-bottom', 'label': 'Legend.Position', 'description': 'Position of legend'}, {'name': 'output.filename', 'value': '<input.surv.data.file_basename>', 'label': 'Output.Filename', 'description': 'Name for output files'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'SurvivalDifference': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Tests for survival difference based on cls file', 'required_args': [{'name': 'input.surv.data.file', 'value': '', 'label': 'Input.Surv.Data.File', 'description': 'input clinical data including survival time & censor status  -.txt'}, {'name': 'input.cls.file', 'value': '', 'label': 'Input.Cls.File', 'description': 'class information file (.cls format)'}, {'name': 'cls.field', 'value': '', 'label': 'Cls.Field', 'description': 'Name of class field in survival data file (optional)'}, {'name': 'subgroup', 'value': '', 'label': 'Subgroup', 'description': 'Suggroups (classes) to be compared (e.g., A, C as in 3rd line of cls file)'}], 'default_args': [{'name': 'task_name', 'value': 'SurvivalDifference', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'time.field', 'value': 'time', 'label': 'Time.Field', 'description': 'Name of survival time field'}, {'name': 'censor.field', 'value': 'status', 'label': 'Censor.Field', 'description': 'Name of censor field (value should be \"event:1, no event: 0\")'}, {'name': 'test.type', 'value': '0', 'label': 'Test.Type', 'description': 'Type of test'}, {'name': 'output.filename', 'value': '<input.surv.data.file_basename>', 'label': 'Output.Filename', 'description': 'Output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'SVM': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Classify samples using SVM', 'required_args': [{'name': 'train.data.filename', 'value': '', 'label': 'Train.Data.Filename', 'description': 'The training data file - .gct, .res'}, {'name': 'train.cls.filename', 'value': '', 'label': 'Train.Cls.Filename', 'description': 'The training class file - .cls'}, {'name': 'saved.model.filename', 'value': '', 'label': 'Saved.Model.Filename', 'description': 'A saved model from a previous run of SVM'}, {'name': 'test.data.filename', 'value': '', 'label': 'Test.Data.Filename', 'description': 'The test data file - .gct, .res'}, {'name': 'test.cls.filename', 'value': '', 'label': 'Test.Cls.Filename', 'description': 'The test class file - .cls'}], 'default_args': [{'name': 'task_name', 'value': 'SVM', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'pred.results.output.file', 'value': '<test.data.filename_basename>.pred.odf', 'label': 'Pred.Results.Output.File', 'description': 'The name of the output file for prediction results'}, {'name': 'model.output.file', 'value': '<train.data.filename_basename>.model', 'label': 'Model.Output.File', 'description': 'The name of the file containing the saved model'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'TmapToGct': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Converts a TMAP file to GCT format', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'Input file in TMAP format'}], 'default_args': [{'name': 'task_name', 'value': 'TmapToGct', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'row.labels', 'value': 'Cufflinks gene ID', 'label': 'Row.Labels', 'description': 'Select row labels to be written to GCT file'}, {'name': 'row.descriptions', 'value': 'Cufflinks transcript ID', 'label': 'Row.Descriptions', 'description': 'Select row descriptions to be written to GCT file'}, {'name': 'filter.rows', 'value': 'no', 'label': 'Filter.Rows', 'description': 'Filter rows with invalid row names (ie., \"-\") '}, {'name': 'output.prefix', 'value': '<input.file_basename>', 'label': 'Output.Prefix', 'description': 'A label that will be used to name the GCT output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'TopHat': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '[**Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] TopHat 2.0.11 is a fast splice junction mapper for RNA-Seq reads. It aligns RNA-Seq reads to mammalian-sized genomes using the ultra high-throughput short read aligner Bowtie, and then analyzes the mapping results to identify splice junctions between exons.', 'required_args': [{'name': 'GTF.file', 'value': '', 'label': 'GTF.File', 'description': 'A GTF file (v. 2.2 or higher) or GFF3 file containing a list of gene model annotations. TopHat will first extract the transcript sequences and align them to this virtual transcriptome first. Only the reads that do not fully map to the transcriptome will then be mapped on the genome. The reads that did map on the transcriptome will be converted to genomic mappings (spliced as needed) and merged with the novel mappings.'}, {'name': 'transcriptome.index', 'value': '', 'label': 'Transcriptome.Index', 'description': 'A directory containing a transcriptome index generated from a previous run of TopHat'}, {'name': 'bowtie.index', 'value': '', 'label': 'Bowtie.Index', 'description': 'A zip file or directory containing a Bowtie 2 index.'}, {'name': 'reads.pair.1', 'value': '', 'label': 'Reads.Pair.1', 'description': 'Unpaired reads file or first mate for paired reads. One or more files containing reads in FASTA or FASTQ format (bz2 and gz compressed files are supported).'}, {'name': 'reads.pair.2', 'value': '', 'label': 'Reads.Pair.2', 'description': 'Second mate for paired reads. Zero or more files in FASTA or FASTQ format (bz2 and gz compressed files are supported).'}, {'name': 'mate.std.dev', 'value': '', 'label': 'Mate.Std.Dev', 'description': 'The standard deviation for the distribution of inner distances between mate pairs.'}, {'name': 'library.type', 'value': '', 'label': 'Library.Type', 'description': 'Library type for strand specific reads.'}, {'name': 'Bowtie.preset.options', 'value': '', 'label': 'Bowtie.Preset.Options', 'description': 'A combination of pre-packaged options for Bowtie 2 based on speed and sensitivity/accuracy.'}, {'name': 'max.transcriptome.hits', 'value': '', 'label': 'Max.Transcriptome.Hits', 'description': 'The maximum number of mappings allowed for a read when it is aligned to the virtual transcriptome (provided in the GTF file parameter). Any reads found with more than this number of mappings will be discarded.'}, {'name': 'prefilter.multihits', 'value': '', 'label': 'Prefilter.Multihits', 'description': 'When mapping reads on the virtual transcriptome (provided in the GTF file parameter), some repetitive or low complexity reads that would be discarded in the context of the genome may appear to align to the transcript sequences and thus may end up reported as mapped to those genes only. This option directs TopHat to first align the reads to the whole genome, then exclude such multi-mapped reads.'}, {'name': 'raw.junctions.file', 'value': '', 'label': 'Raw.Junctions.File', 'description': 'A file containing raw junctions. Junctions are specified one per line in a tab-delimited format.'}, {'name': 'find.novel.junctions', 'value': '', 'label': 'Find.Novel.Junctions', 'description': 'If you select no, then the module will only look for junctions indicated in the GTF file supplied in the GTF file parameter. (This parameter is ignored when no GTF file is specified.)'}, {'name': 'min.intron.length', 'value': '', 'label': 'Min.Intron.Length', 'description': 'The minimum intron length. TopHat will ignore donor/acceptor pairs closer than this many bases apart.'}, {'name': 'quality.value.scale', 'value': '', 'label': 'Quality.Value.Scale', 'description': 'Whether to use the Solexa, Phred 33, or  Solexa v. 1.3 (Phred 64) quality value scale.'}, {'name': 'microexon.search', 'value': '', 'label': 'Microexon.Search', 'description': 'Attempts to find alignments incident to microexons. Works only for reads 50bp or longer.'}, {'name': 'fusion.mapping', 'value': '', 'label': 'Fusion.Mapping', 'description': 'Whether to enable fusion mapping'}, {'name': 'read.edit.dist', 'value': '', 'label': 'Read.Edit.Dist', 'description': 'Final read alignments having more than these many edit distance are discarded.'}, {'name': 'read.gap.length', 'value': '', 'label': 'Read.Gap.Length', 'description': 'Final read alignments having more than these many total length of gaps are discarded.'}, {'name': 'additional.tophat.options', 'value': '', 'label': 'Additional.Tophat.Options', 'description': 'Additional options to be passed along to the TopHat program at the command line. This parameter gives you a means to specify otherwise unavailable TopHat options and switches not supported by the module; check the TopHat manual for details. Recommended for experts only; use this at your own discretion.'}], 'default_args': [{'name': 'task_name', 'value': 'TopHat', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'mate.inner.dist', 'value': '50', 'label': 'Mate.Inner.Dist', 'description': 'The expected mean inner distance between mate pairs.'}, {'name': 'transcriptome.only', 'value': 'no', 'label': 'Transcriptome.Only', 'description': 'Whether to align the reads to the virtual transcriptome (provided in the GTF file parameter) and report only those mappings as genomic mappings.'}, {'name': 'min.anchor.length', 'value': '8', 'label': 'Min.Anchor.Length', 'description': 'The anchor length. This value must be at least 3.'}, {'name': 'max.splice.mismatches', 'value': '0', 'label': 'Max.Splice.Mismatches', 'description': 'The maximum number of mismatches that may appear in the \"anchor\" region of a spliced alignment.'}, {'name': 'max.intron.length', 'value': '500000', 'label': 'Max.Intron.Length', 'description': 'The maximum intron length. When searching for junctions ab initio, TopHat will ignore donor/acceptor pairs farther than this many bases apart, except when such a pair is supported by a split segment alignment of a long read.'}, {'name': 'max.insertion.length', 'value': '3', 'label': 'Max.Insertion.Length', 'description': 'The maximum insertion length'}, {'name': 'max.deletion.length', 'value': '3', 'label': 'Max.Deletion.Length', 'description': 'The maximum deletion length'}, {'name': 'max.multihits', 'value': '20', 'label': 'Max.Multihits', 'description': 'The maximum number of times a read can be aligned to the reference genome. If a read is aligned more than this number of times, then TopHat will choose the alignments based on their alignment scores, reporting the alignments with the best alignment scores. If there are more than this number of alignments with the same score for a read, TopHat will randomly report only this many alignments.'}, {'name': 'read.mismatches', 'value': '2', 'label': 'Read.Mismatches', 'description': 'Final read alignments having more than these many mismatches should be discarded.'}, {'name': 'coverage.search', 'value': '--no-coverage-search', 'label': 'Coverage.Search', 'description': 'Enables or disables the coverage based search for junctions. Use when coverage search is disabled by default (such as for reads 75bp or longer), for maximum sensitivity.'}, {'name': 'fusion.anchor.length', 'value': '20', 'label': 'Fusion.Anchor.Length', 'description': 'A \"supporting\" read must map to both sides of a fusion by at least these many bases. Only applies when fusion mapping is set to yes.'}, {'name': 'fusion.read.mismatches', 'value': '2', 'label': 'Fusion.Read.Mismatches', 'description': 'Reads support fusions if they map across fusion with at most these many mismatches. Only applies when fusion mapping is set to yes.'}, {'name': 'output.prefix', 'value': '<job_id>_<reads.pair.1_basename>', 'label': 'Output.Prefix', 'description': 'The prefix to use for the output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'TransposeDataset': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Transpose a Dataset - .res .gct, .odf', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'Dataset - .res .gct, .odf'}], 'default_args': [{'name': 'task_name', 'value': 'TransposeDataset', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file.format', 'value': 'gct', 'label': 'Output.File.Format', 'description': 'The output file format'}, {'name': 'output.file.name', 'value': '<input.filename_basename>.transposed', 'label': 'Output.File.Name', 'description': 'The name of the output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Trimmomatic': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Provides a variety of options for trimming Illumina FASTQ files of adapter sequences and low-quality reads.', 'required_args': [{'name': 'input.file.1', 'value': '', 'label': 'Input.File.1', 'description': 'The input FASTQ to be trimmed.  For paired-end data, this should be the forward (\"*_1\" or \"left\") input file.'}, {'name': 'input.file.2', 'value': '', 'label': 'Input.File.2', 'description': 'The reverse (\"*_2\" or \"right\") input FASTQ of paired-end data to be trimmed.'}, {'name': 'adapter.clip.sequence.file', 'value': '', 'label': 'Adapter.Clip.Sequence.File', 'description': 'A FASTA file containing the adapter sequences, PCR sequences, etc. to be clipped. This parameter is required to enable adapter clipping. Files are provided for several Illumina pipelines but you can also provide your own; see the documentation for details. Be sure to choose a PE file for paired-end data and an SE file for single-end data.'}, {'name': 'adapter.clip.seed.mismatches', 'value': '', 'label': 'Adapter.Clip.Seed.Mismatches', 'description': 'Specifies the maximum mismatch count which will still allow a full match to be performed. A value of 2 is recommended. This parameter is required to enable adapter clipping.'}, {'name': 'adapter.clip.palindrome.clip.threshold', 'value': '', 'label': 'Adapter.Clip.Palindrome.Clip.Threshold', 'description': \"Specifies how accurate the match between the two 'adapter ligated' reads must be for PE palindrome read alignment. This is the log10 probability against getting a match by random chance; values around 30 or more are recommended. This parameter is required to enable adapter clipping.\"}, {'name': 'adapter.clip.simple.clip.threshold', 'value': '', 'label': 'Adapter.Clip.Simple.Clip.Threshold', 'description': 'Specifies how accurate the match between any adapter etc. sequence must be against a read as a log10 probability against getting a match by random chance; values between 7-15 are recommended. This parameter is required to enable adapter clipping.'}, {'name': 'adapter.clip.min.length', 'value': '', 'label': 'Adapter.Clip.Min.Length', 'description': 'In addition to the alignment score, palindrome mode can verify that a minimum length of adapter has been detected. If unspecified, this defaults to 8 bases, for historical reasons. However, since palindrome mode has a very low false positive rate, this can be safely reduced, even down to 1, to allow shorter adapter fragments to be removed.'}, {'name': 'trim.leading.quality.threshold', 'value': '', 'label': 'Trim.Leading.Quality.Threshold', 'description': 'Remove low quality bases from the beginning. As long as a base has a value below this threshold the base is removed and the next base will be investigated.'}, {'name': 'trim.trailing.quality.threshold', 'value': '', 'label': 'Trim.Trailing.Quality.Threshold', 'description': 'Remove low quality bases from the end. As long as a base has a value below this threshold the base is removed and the next trailing base will be investigated.'}, {'name': 'max.info.target.length', 'value': '', 'label': 'Max.Info.Target.Length', 'description': 'Performs a Max Info adaptive quality trim, balancing the benefits of retaining longer reads against the costs of retaining bases with errors. This parameter specifies the read length which is likely to allow the location of the read within the target sequence to be determined.'}, {'name': 'max.info.strictness', 'value': '', 'label': 'Max.Info.Strictness', 'description': 'This value, which should be set between 0 and 1, specifies the balance between preserving as much read length as possible vs. removal of incorrect bases. A low value of this parameter (<0.2) favors longer reads, while a high value (>0.8) favors read correctness.  Both max.info.target.length and max.info.strictness are required for the Max Info quality trim.'}, {'name': 'sliding.window.size', 'value': '', 'label': 'Sliding.Window.Size', 'description': 'Perform a sliding window trimming, cutting once the average quality within the window falls below a threshold. By considering multiple bases, a single poor quality base will not cause the removal of high quality data later in the read.  This parameter specifies the number of bases to average across.'}, {'name': 'sliding.window.quality.threshold', 'value': '', 'label': 'Sliding.Window.Quality.Threshold', 'description': 'Specifies the average quality required for the sliding window trimming. Both sliding.window.size and sliding.window.quality.threshold are required to enable the sliding window trimming.'}, {'name': 'min.read.length', 'value': '', 'label': 'Min.Read.Length', 'description': 'Remove reads that fall below the specified minimal length.'}, {'name': 'extra.steps', 'value': '', 'label': 'Extra.Steps', 'description': 'Extra steps to be performed after any other processing. These must be specified in exactly the format described in the Trimmomatic manual; see the documentation for details.  This is recommended for advanced users only.'}, {'name': 'convert.phred.scores', 'value': '', 'label': 'Convert.Phred.Scores', 'description': 'Convert phred scores into a particular encoding.  Leave this blank for no conversion.'}], 'default_args': [{'name': 'task_name', 'value': 'Trimmomatic', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.filename.base', 'value': '<input.file.1_basename>', 'label': 'Output.Filename.Base', 'description': 'A base name to be used for the output files.'}, {'name': 'adapter.clip.keep.both.reads', 'value': 'true', 'label': 'Adapter.Clip.Keep.Both.Reads', 'description': 'Controls whether to keep both forward and reverse reads when trimming in palindrome mode. The reverse read is the same as the forward but in reverse complement and so carries no additional information. The default is \"yes\" (retain the reverse read) which is useful when downstream tools cannot handle a combination of paired and unpaired reads.'}, {'name': 'phred.encoding', 'value': 'phred33', 'label': 'Phred.Encoding', 'description': 'Allows you to specify the phred quality encoding.  The default is phred33, which matches modern Illumina pipelines.'}, {'name': 'create.trimlog', 'value': 'false', 'label': 'Create.Trimlog', 'description': 'Create a log of the trimming process.  This gives details on what operations were performed, etc. but can be quite lengthy.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Two.Component.Normalization': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Normalization for unimodal and bimodal samples using mixture models', 'required_args': [{'name': 'gct.file', 'value': '', 'label': 'Gct.File', 'description': 'Input data file (gct format)'}, {'name': 'cls.file', 'value': '', 'label': 'Cls.File', 'description': 'File indicating type of samples (unimodal or bimodal) in gct.file'}], 'default_args': [{'name': 'task_name', 'value': 'Two.Component.Normalization', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.prefix', 'value': '<gct.file_basename>', 'label': 'Output.Prefix', 'description': 'Filename prefix for results files'}, {'name': 'data.type', 'value': 'proteome', 'label': 'Data.Type', 'description': 'Indicate type of input data (proteome, phosphoprotome, etc.) for use in plots'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'UniquifyLabels': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Makes row and column labels unique', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'input filename - .res, .gct, .odf type=Dataset'}], 'default_args': [{'name': 'task_name', 'value': 'UniquifyLabels', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<input.filename_basename>.unique', 'label': 'Output.File', 'description': 'output file'}, {'name': 'make.row.labels.unique', 'value': 'yes', 'label': 'Make.Row.Labels.Unique', 'description': 'whether to make the row labels unique'}, {'name': 'make.column.labels.unique', 'value': 'yes', 'label': 'Make.Column.Labels.Unique', 'description': 'whether to make the column labels unique'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'untar': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'untars .tgz files', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': ''}], 'default_args': [{'name': 'task_name', 'value': 'untar', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'VennDiagram': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Displays a Venn diagram', 'required_args': [{'name': 'feature.list.one', 'value': '', 'label': 'Feature.List.One', 'description': 'Feature list one - .grp, .txt'}, {'name': 'feature.list.two', 'value': '', 'label': 'Feature.List.Two', 'description': 'Feature list two - .grp, .txt'}, {'name': 'feature.list.three', 'value': '', 'label': 'Feature.List.Three', 'description': 'Feature list three - .grp, .txt'}], 'default_args': [{'name': 'task_name', 'value': 'VennDiagram', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'WeightedVoting': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Weighted Voting classification', 'required_args': [{'name': 'train.filename', 'value': '', 'label': 'Train.Filename', 'description': 'training data file name - .gct, .res, .odf type = Dataset, ignored if a saved model (saved.model.filename) is used'}, {'name': 'train.class.filename', 'value': '', 'label': 'Train.Class.Filename', 'description': 'class file for training data - .cls, ignored if a saved model (saved.model.filename) is used'}, {'name': 'saved.model.filename', 'value': '', 'label': 'Saved.Model.Filename', 'description': 'input Weighted Voting model file - .odf type = Weighted Voting Prediction Model'}, {'name': 'test.filename', 'value': '', 'label': 'Test.Filename', 'description': 'test data file name - .gct, .res, .odf type = Dataset'}, {'name': 'test.class.filename', 'value': '', 'label': 'Test.Class.Filename', 'description': 'class file for test data - .cls'}, {'name': 'min.std', 'value': '', 'label': 'Min.Std', 'description': 'minimum standard deviation if feature selection statistic includes min std option'}, {'name': 'feature.list.filename', 'value': '', 'label': 'Feature.List.Filename', 'description': 'features to use for prediction'}], 'default_args': [{'name': 'task_name', 'value': 'WeightedVoting', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'model.file', 'value': '<train.filename_basename>.model.odf', 'label': 'Model.File', 'description': 'name of output Weighted Voting model file  - .odf type = Weighted Voting Prediction Model'}, {'name': 'num.features', 'value': '10', 'label': 'Num.Features', 'description': 'number of selected features if feature list filename is not specified'}, {'name': 'feature.selection.statistic', 'value': '0', 'label': 'Feature.Selection.Statistic', 'description': 'statistic to use to perform feature selection'}, {'name': 'pred.results.file', 'value': '<test.filename_basename>.pred.odf', 'label': 'Pred.Results.File', 'description': 'name of prediction results output file - .odf type = Prediction Results'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'WeightedVotingXValidation': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Weighted Voting classification with Leave-One-Out Cross-Validation', 'required_args': [{'name': 'data.filename', 'value': '', 'label': 'Data.Filename', 'description': 'Dataset input data file - res, gct, odf'}, {'name': 'class.filename', 'value': '', 'label': 'Class.Filename', 'description': 'Class input file - cls'}, {'name': 'min.std', 'value': '', 'label': 'Min.Std', 'description': 'minimum standard deviation if feature selection statistic includes min std option'}], 'default_args': [{'name': 'task_name', 'value': 'WeightedVotingXValidation', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'num.features', 'value': '10', 'label': 'Num.Features', 'description': 'number of selected features'}, {'name': 'feature.selection.statistic', 'value': '0', 'label': 'Feature.Selection.Statistic', 'description': 'statistic to use to perform feature selection'}, {'name': 'pred.results.file', 'value': '<data.filename_basename>.pred.odf', 'label': 'Pred.Results.File', 'description': 'Prediction results output file name - odf'}, {'name': 'feature.summary.file', 'value': '<data.filename_basename>.feat.odf', 'label': 'Feature.Summary.File', 'description': 'Feature summary output file name - odf'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'XChromosomeCorrect': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': \"Corrects X Chromosome SNP's for male samples.\", 'required_args': [{'name': 'snp.filename', 'value': '', 'label': 'Snp.Filename', 'description': 'Summarized probe intensity data in .snp format'}, {'name': 'sample.info.filename', 'value': '', 'label': 'Sample.Info.Filename', 'description': 'A Sample information file that contains gender information'}], 'default_args': [{'name': 'task_name', 'value': 'XChromosomeCorrect', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<snp.filename_basename>.xcorrected.snp', 'label': 'Output.File', 'description': 'Output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}} ...\n",
      "(Getting tasks ...)\n",
      "(Setting tasks ...)\n",
      "Loading task-specifying JSON /Users/kateme/.Simpli/json/nbpackage_tasks.json ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task Make Data Filepath: {'library_path': '', 'library_name': 'simpli.nbpackage_tasks', 'function_name': 'make_data_filepath', 'description': 'No description.', 'required_args': [{'name': 'filename', 'value': '', 'label': 'File Name', 'description': 'No description.'}], 'default_args': [], 'optional_args': [], 'returns': [{'label': 'Variable for Data File Path', 'description': 'No description.'}]}.\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing args ...\n",
      "Processing returns ...\n",
      "\t\tLoaded task Make Result Filepath: {'library_path': '', 'library_name': 'simpli.nbpackage_tasks', 'function_name': 'make_result_filepath', 'description': 'No description.', 'required_args': [{'name': 'filename', 'value': '', 'label': 'File Name', 'description': 'No description.'}], 'default_args': [], 'optional_args': [], 'returns': [{'label': 'Variable for Result File Path', 'description': 'No description.'}]}.\n",
      "(Getting tasks ...)\n",
      "Setting/updating task {'Load JSON': {'library_path': '', 'library_name': 'simpli.default_tasks', 'function_name': 'link_json', 'description': 'No description.', 'required_args': [{'name': 'filepath', 'value': '', 'label': 'Simpli JSON to be Loaded', 'description': 'No description.'}], 'default_args': [], 'optional_args': [], 'returns': []}, 'Reset JSONs': {'library_path': '', 'library_name': 'simpli.default_tasks', 'function_name': 'reset_jsons', 'description': 'No description.', 'required_args': [], 'default_args': [], 'optional_args': [], 'returns': []}, 'Assign': {'library_path': '', 'library_name': 'simpli.default_tasks', 'function_name': 'just_return', 'description': 'No description.', 'required_args': [{'name': 'value', 'value': '', 'label': 'Value to Assign', 'description': 'No description.'}], 'default_args': [], 'optional_args': [], 'returns': [{'label': 'Variable Name to Assign to', 'description': 'No description.'}]}, 'Slice DataFrame by Row': {'library_path': '', 'library_name': 'simpli.default_tasks', 'function_name': 'slice_dataframe', 'description': 'Return a subset of rows of a given DataFrame object.', 'required_args': [{'name': 'dataframe', 'value': '', 'label': 'DataFrame', 'description': 'Target dataframe object.'}, {'name': 'indices', 'value': '', 'label': 'Row Index(ices)', 'description': 'An integer or list of integers separated by commas of the rows to slice. Note that indices are 0 based.'}], 'default_args': [{'name': 'ax', 'value': '0', 'label': 'Ax', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Sliced DataFrame', 'description': 'A slice of the DataFrame including the specified rows.'}]}, 'Slice DataFrame by Column': {'library_path': '', 'library_name': 'simpli.default_tasks', 'function_name': 'slice_dataframe', 'description': 'No description.', 'required_args': [{'name': 'dataframe', 'value': '', 'label': 'DataFrame', 'description': 'No description.'}, {'name': 'indices', 'value': '', 'label': 'Column Index(ices)', 'description': 'No description.'}], 'default_args': [{'name': 'ax', 'value': '1', 'label': 'Ax', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Sliced DataFrame', 'description': 'No description.'}]}, 'Set Notebook Theme': {'library_path': '', 'library_name': 'simpli.default_tasks', 'function_name': 'set_notebook_theme', 'description': 'No description.', 'required_args': [{'name': 'filepath', 'value': '', 'label': 'CSS File', 'description': 'No description.'}], 'default_args': [], 'optional_args': [], 'returns': []}, 'Center Align Outputs': {'library_path': '', 'library_name': 'simpli.default_tasks', 'function_name': 'center_align_output_cells', 'description': 'No description.', 'required_args': [], 'default_args': [], 'optional_args': [], 'returns': []}, 'Display Start Banner and Logos': {'library_path': '', 'library_name': 'simpli.default_tasks', 'function_name': 'display_start_banner_and_logos', 'description': 'No description.', 'required_args': [], 'default_args': [], 'optional_args': [], 'returns': []}, 'Display End Banner': {'library_path': '', 'library_name': 'simpli.default_tasks', 'function_name': 'display_end_banner', 'description': 'No description.', 'required_args': [], 'default_args': [], 'optional_args': [], 'returns': []}, 'Link YouTube Video': {'library_path': '', 'library_name': 'simpli.default_tasks', 'function_name': 'youtube', 'description': 'No description.', 'required_args': [{'name': 'url', 'value': '', 'label': 'YouTube Video URL', 'description': 'No description.'}], 'default_args': [], 'optional_args': [], 'returns': []}, 'Toggle Input Cells': {'library_path': '', 'library_name': 'simpli.default_tasks', 'function_name': 'toggle_input_cells', 'description': 'No description.', 'required_args': [], 'default_args': [], 'optional_args': [], 'returns': []}, 'Execute HTML': {'library_path': '', 'library_name': 'simpli.default_tasks', 'function_name': 'display_raw_html', 'description': 'No description.', 'required_args': [], 'default_args': [], 'optional_args': [], 'returns': []}, 'Login GenePattern (Broad Institute)': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'login', 'description': 'Login into the GenePattern server at the Broad Institure.', 'required_args': [{'name': 'user_name', 'value': '', 'label': 'User name', 'description': 'GenePattern user name.'}, {'name': 'passcode', 'value': '', 'label': 'Passcode', 'description': 'GenePattern passcode.'}], 'default_args': [{'name': 'server_name', 'value': 'https://genepattern.broadinstitute.org/gp', 'label': 'Server Name', 'description': 'No description.'}], 'optional_args': [], 'returns': []}, 'ABSOLUTE': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '[**Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] Extracts absolute copy numbers per cancer cell from a mixed DNA population. Use this module for the per-sample processing step in the workflow (usually after HAPSEG).', 'required_args': [{'name': 'seg.dat.file', 'value': '', 'label': 'Seg.Dat.File', 'description': 'A HAPSEG output file'}, {'name': 'primary.disease', 'value': '', 'label': 'Primary.Disease', 'description': 'Primary disease of the sample'}, {'name': 'sample.name', 'value': '', 'label': 'Sample.Name', 'description': 'The name of the sample, included in output plots'}, {'name': 'maf.file', 'value': '', 'label': 'Maf.File', 'description': 'If available, a minor allele frequency file. This specifies the data for somatic point mutations to be used by ABSOLUTE.'}, {'name': 'min.mut.af', 'value': '', 'label': 'Min.Mut.Af', 'description': 'If specified, a minimum mutation allelic fraction. Mutations with lower allelic fractions will be filtered out before analysis. Note that if maf.fn is specified, min.mut.af must also be specified.'}], 'default_args': [{'name': 'task_name', 'value': 'ABSOLUTE', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file.name.base', 'value': '<sample.name>', 'label': 'Output.File.Name.Base', 'description': 'If specified, provides a base filename for all output files. The default value is the sample.name.'}, {'name': 'sigma.p', 'value': '0', 'label': 'Sigma.P', 'description': 'Provisional value of excess sample level variance used for mode search'}, {'name': 'max.sigma.h', 'value': '0.015', 'label': 'Max.Sigma.H', 'description': 'Maximum value of excess sample level variance (Eq. 6)'}, {'name': 'min.ploidy', 'value': '0.95', 'label': 'Min.Ploidy', 'description': 'Minimum ploidy value to consider. Solutions implying lower ploidy values will be discarded'}, {'name': 'max.ploidy', 'value': '10', 'label': 'Max.Ploidy', 'description': 'Maximum ploidy value to consider. Solutions implying greater ploidy values will be discarded'}, {'name': 'platform', 'value': 'SNP_6.0', 'label': 'Platform', 'description': \"The chip type used, supported values are currently 'SNP_250K_STY', 'SNP_6.0' and 'Illumina_WES'\"}, {'name': 'max.as.seg.count', 'value': '1500', 'label': 'Max.As.Seg.Count', 'description': \"Maximum number of allelic segments. Samples with a higher segment count will be flagged as 'failed'\"}, {'name': 'max.neg.genome', 'value': '0.005', 'label': 'Max.Neg.Genome', 'description': 'Maximum genome fraction that may be modeled as non-clonal with copy-ratio below that of clonal homozygous deletion. Solutions implying greater values will be discarded.'}, {'name': 'max.non.clonal', 'value': '0.05', 'label': 'Max.Non.Clonal', 'description': 'Maximum genome fraction that may be modeled as non-clonal (subclonal SCNA). Solutions implying greater values will be discarded.'}, {'name': 'copy.number.type', 'value': 'allelic', 'label': 'Copy.Number.Type', 'description': \"The copy number type to assess, can be one of 'allelic' or 'total'\"}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ABSOLUTE.review': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '[**Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] Extracts the absolute copy number per cancer cell from a mixed DNA population.  Use this module for the Review and Extract step in the workflow (after ABSOLUTE).', 'required_args': [{'name': 'reviewed.pp.calls.file', 'value': '', 'label': 'Reviewed.Pp.Calls.File', 'description': 'The calls file from a previous run of the ABSOLUTE.summarize module that has been reviewed and annotated by an analyst.'}, {'name': 'analyst.id', 'value': '', 'label': 'Analyst.Id', 'description': 'The user id of the analyst who called the solutions.  This is used for display/reporting purposes only.'}, {'name': 'modes.file', 'value': '', 'label': 'Modes.File', 'description': 'The obj_summary.PP-modes.data.RData file produced by the previous run of the ABSOLUTE.summarize module.'}, {'name': 'collection.name', 'value': '', 'label': 'Collection.Name', 'description': 'A descriptive name for this collection of samples.  Use the same name as in the previous run of the ABSOLUTE.summarize module.  This is used for display/reporting purposes only.'}], 'default_args': [{'name': 'task_name', 'value': 'ABSOLUTE.review', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'copy.number.type', 'value': 'allelic', 'label': 'Copy.Number.Type', 'description': \"The copy number type to assess, can be one of 'allelic' or 'total'\"}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ABSOLUTE.summarize': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '[**Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] Summarizes the results from multiple ABSOLUTE runs so that an analyst can manually review the solutions.', 'required_args': [{'name': 'collection.name', 'value': '', 'label': 'Collection.Name', 'description': 'A descriptive name for this collection of samples.'}, {'name': 'absolute.files', 'value': '', 'label': 'Absolute.Files', 'description': 'A group of ABSOLUTE RData output files from individual sample runs to be summarized together.'}], 'default_args': [{'name': 'task_name', 'value': 'ABSOLUTE.summarize', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'copy.number.type', 'value': 'allelic', 'label': 'Copy.Number.Type', 'description': \"The copy number type to assess, can be one of 'allelic' or 'total'.\"}, {'name': 'plot.modes', 'value': 'TRUE', 'label': 'Plot.Modes', 'description': 'Set this to FALSE to disable plotting of the purity/ploidy modes.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Acgh2Tab': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Converts acgh files to a tab-delimited format usable by Genomica ', 'required_args': [{'name': 'acgh.input.file', 'value': '', 'label': 'Acgh.Input.File', 'description': 'Data in aCHG format'}, {'name': 'presentlist.file', 'value': '', 'label': 'Presentlist.File', 'description': 'List of samples with a certain gene signature (from Genomica).'}, {'name': 'absentlist.file', 'value': '', 'label': 'Absentlist.File', 'description': 'List of samples without a certain gene signature (from Genomica).'}], 'default_args': [{'name': 'task_name', 'value': 'Acgh2Tab', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'genelocs.file', 'value': '<libdir>hg18_refseq_genes.txt', 'label': 'Genelocs.File', 'description': 'File giving gene locations (from GenomeBrowser).'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'AddFCSEventIndex': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Adds indexes to events in a Flow Cytometry Standard (FCS) data file.', 'required_args': [{'name': 'Input.data.file', 'value': '', 'label': 'Input.Data.File', 'description': 'An FCS data file to which you want to add indexes.'}, {'name': 'Index.parameter.long.name', 'value': '', 'label': 'Index.Parameter.Long.Name', 'description': 'The long name ($PnS keyword) of the parameter to get the index; the default is blank.'}], 'default_args': [{'name': 'task_name', 'value': 'AddFCSEventIndex', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Start.index', 'value': '1', 'label': 'Start.Index', 'description': 'The index number for the first event in the data set; the default is 1. The index will be incremented by 1 for each of the events in the data set. '}, {'name': 'Index.parameter.short.name', 'value': 'Index', 'label': 'Index.Parameter.Short.Name', 'description': 'The short name ($PnN keyword) of the parameter to get the index; the default is \"Index.\" '}, {'name': 'Output.fcs.file.name', 'value': '<Input.data.file_basename>', 'label': 'Output.Fcs.File.Name', 'description': 'The output file name.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'AddFCSParameter': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Add parameters and their values to a FCS data file', 'required_args': [{'name': 'Input.fcs.data.file', 'value': '', 'label': 'Input.Fcs.Data.File', 'description': 'A FCS data file to add parameters to.'}, {'name': 'Input.csv.file', 'value': '', 'label': 'Input.Csv.File', 'description': 'A CSV file with parameters and values to add.'}], 'default_args': [{'name': 'task_name', 'value': 'AddFCSParameter', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Output.file.name', 'value': '<Input.fcs.data.file_basename>_with_extra_params.fcs', 'label': 'Output.File.Name', 'description': 'The output FCS file name.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'AddNoiseToFCS': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Add noise to specified parameters in an FCS data file.', 'required_args': [{'name': 'Input.FCS.data.file', 'value': '', 'label': 'Input.FCS.Data.File', 'description': 'An FCS data file to add noise to.'}, {'name': 'Parameters.to.add.noise.to', 'value': '', 'label': 'Parameters.To.Add.Noise.To', 'description': 'List of parameters to which the module will add noise. The list may be provided as a list of indexes (e.g., \"1-3,5,7\") or as a list of parameter names (i.e., values of the $PnN keyword, such as \"FSC-H,FL1-H,FL2-A\"). By default, the noise will be added to all parameters but the Time parameter.'}], 'default_args': [{'name': 'task_name', 'value': 'AddNoiseToFCS', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Output.FCS.file.name', 'value': '<Input.FCS.data.file_basename>.with_noise.fcs', 'label': 'Output.FCS.File.Name', 'description': 'The output FCS file name.'}, {'name': 'Noise.level', 'value': '0.5', 'label': 'Noise.Level', 'description': 'Noise level in percent. This is the maximum noise that will be added to an event based on the current event parameter value or on the maximum range of the appropriate parameter. The latter option will be effective if the Uniform noise option is selected.'}, {'name': 'Noise', 'value': 'Proportional', 'label': 'Noise', 'description': 'How to add the noise: Proportional (The maximum amount of noise added to each value will be based on the value itself, e.g., low values will get lower amount of noise.) or Uniform (The maximum amount of noise added to each value is only dependent on the range of the particular parameter).'}, {'name': 'Saturated.events', 'value': 'RemoveSaturated', 'label': 'Saturated.Events', 'description': 'Whether to remove saturated events. An event is considered saturated if one or more of its parameter values are closer than 0.1 percent from the maximum range for that parameter (i.e., value of the appropriate $PnR keyword).'}, {'name': 'Random.Seed', 'value': '1234567', 'label': 'Random.Seed', 'description': 'The seed to use for the random generator. Leave this blank if want to use a randomly selected seed.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'AffySTExpressionFileCreator': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '[Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] Creates a GCT file from a set of CEL files from Affymetrix ST arrays.', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'One or more Affymetrix ST CEL files either uploaded directly, packaged into a ZIP or TAR bundle, or supplied through a directory input.  The CEL files can be in GZ format and the TAR can be in GZ, XZ, or BZ2 format.  The parameter will accept multiple inputs in any of these forms. '}, {'name': 'clm.file', 'value': '', 'label': 'Clm.File', 'description': 'A tab-delimited text file containing one scan, sample, and class per line'}], 'default_args': [{'name': 'task_name', 'value': 'AffySTExpressionFileCreator', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'normalize', 'value': 'yes', 'label': 'Normalize', 'description': 'Whether to normalize data using quantile normalization'}, {'name': 'background.correct', 'value': 'yes', 'label': 'Background.Correct', 'description': 'Whether to perform background correction'}, {'name': 'qc.plot.format', 'value': 'skip', 'label': 'Qc.Plot.Format', 'description': 'Choose a file format for the QC plots (or skip them completely)'}, {'name': 'annotate.rows', 'value': 'no', 'label': 'Annotate.Rows', 'description': 'Whether to annotate rows with the gene symbol and description'}, {'name': 'output.file.base', 'value': '<job_id>.expression.gct', 'label': 'Output.File.Base', 'description': 'The base name of the output file(s). File extensions will be added automatically.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ApplyGatingML': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Apply a Gating-ML file on an FCS data file (gate and/or transform list mode data) ', 'required_args': [{'name': 'Input.FCS.data.file', 'value': '', 'label': 'Input.FCS.Data.File', 'description': 'An FCS data file to be gated.'}, {'name': 'Input.Gating-ML.file', 'value': '', 'label': 'Input.Gating-ML.File', 'description': 'A Gating-ML file specifying the gates.'}], 'default_args': [{'name': 'task_name', 'value': 'ApplyGatingML', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Output.base.name', 'value': '<Input.FCS.data.file_basename>', 'label': 'Output.Base.Name', 'description': 'Base for output file names'}, {'name': 'Output.figures', 'value': 'none', 'label': 'Output.Figures', 'description': 'Also output figures of the gates'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ARACNE': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Runs the ARACNE algorithm for reverse engineering cellular networks', 'required_args': [{'name': 'dataset.file', 'value': '', 'label': 'Dataset.File', 'description': 'Dataset file - .res, .gct'}, {'name': 'hub.gene', 'value': '', 'label': 'Hub.Gene', 'description': 'The name of a gene whose network interactions you want to reconstruct'}, {'name': 'hub.genes.file', 'value': '', 'label': 'Hub.Genes.File', 'description': 'A file containing a list of hub genes with one gene per line - .txt'}, {'name': 'transcription.factor.file', 'value': '', 'label': 'Transcription.Factor.File', 'description': 'A file containing a list of genes with one gene per line that encode transcription factors - .txt'}, {'name': 'kernel.width', 'value': '', 'label': 'Kernel.Width', 'description': 'The kernel width (or window width) of the Gaussian Kernel Estimator'}], 'default_args': [{'name': 'task_name', 'value': 'ARACNE', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'mi.threshold', 'value': '0', 'label': 'Mi.Threshold', 'description': 'Threshold for the mutual information (MI) estimate to be considered statistically significant'}, {'name': 'p.value', 'value': '1', 'label': 'P.Value', 'description': 'Threshold for the p-value of a MI estimate to be considered statistically significant'}, {'name': 'dpi.tolerance', 'value': '1', 'label': 'Dpi.Tolerance', 'description': 'The percentage of MI estimation considered as sampling error'}, {'name': 'mean.filter', 'value': '0', 'label': 'Mean.Filter', 'description': 'Filter out non-informative genes whose mean expression value is smaller than mean filter'}, {'name': 'cv.filter', 'value': '0', 'label': 'Cv.Filter', 'description': 'Filter out non-informative genes whose coefficient of variance is smaller than cv filter'}, {'name': 'output.file', 'value': '<dataset.file_basename>.output.adj', 'label': 'Output.File', 'description': 'Output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'AreaChange': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Calculates fraction of area under the spectrum that is attributable to signal (area after noise removal / original area)', 'required_args': [{'name': 'spectrum.filename', 'value': '', 'label': 'Spectrum.Filename', 'description': 'spectrum data file - .csv'}, {'name': 'low.Da', 'value': '', 'label': 'Low.Da', 'description': 'minimum M/Z to include'}, {'name': 'high.Da', 'value': '', 'label': 'High.Da', 'description': 'maximum M/Z to include'}], 'default_args': [{'name': 'task_name', 'value': 'AreaChange', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'quality.threshold', 'value': '0.45', 'label': 'Quality.Threshold', 'description': 'discard spectra with quality < quality threshold'}, {'name': 'window.filter', 'value': '5', 'label': 'Window.Filter', 'description': 'window filter size'}, {'name': 'factor', 'value': '3', 'label': 'Factor', 'description': 'factor to multiply standard deviation by'}, {'name': 'smooth', 'value': 'median', 'label': 'Smooth', 'description': 'averaging function'}, {'name': 'output.file', 'value': '<spectrum.filename_basename>.area.odf', 'label': 'Output.File', 'description': 'output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Arff2Gct': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Convert an .arff file into a gene pattern .gct / .cls file pair', 'required_args': [{'name': 'arff.file', 'value': '', 'label': 'Arff.File', 'description': 'The dataset in ARFF format.  Multi-label class names are separated by colons (:). - .arff'}], 'default_args': [{'name': 'task_name', 'value': 'Arff2Gct', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<arff.file_basename>', 'label': 'Output.File', 'description': 'Name to give the .gct containing dataset features and .cls containing dataset class labels output files.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ATARiS': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Runs ATARiS on RNAi reagent-level data.', 'required_args': [{'name': 'gct.file', 'value': '', 'label': 'Gct.File', 'description': 'input GCT file. Unique reagent identifier in first column, gene symbol in second.'}], 'default_args': [{'name': 'task_name', 'value': 'ATARiS', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'identifier', 'value': '<gct.file_basename>', 'label': 'Identifier', 'description': 'text string to identify the current run'}, {'name': 'random.seed', 'value': '12345', 'label': 'Random.Seed', 'description': 'seed for random number generator'}, {'name': 'null.significance', 'value': '0.15', 'label': 'Null.Significance', 'description': 'gene solution significance level (1 - null percentile)'}, {'name': 'min.A.value', 'value': '0.3', 'label': 'Min.A.Value', 'description': 'minimum effect magnitude for RNAi reagent (relative to reagent with maximal effect)'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'AuDIT': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Automated Detection of Inaccurate and Imprecise Transitions in MRM Mass Spectrometry', 'required_args': [{'name': 'data.file', 'value': '', 'label': 'Data.File', 'description': 'input file with pre-processed MRM data in csv format'}], 'default_args': [{'name': 'task_name', 'value': 'AuDIT', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'skyline.export', 'value': 'TRUE', 'label': 'Skyline.Export', 'description': 'if \"yes\", data file is a Skyline csv export including following columns: Sample (usually derived from SampleName, having identical value for all its replicates), PeptideSequence, ReplicateName, FragmentIon, PrecursorCharge, ProductCharge, light Area, heavy Area. When \"no\", input must contain sample, replicate, peptide, transition.id, area, IS.area in that order.'}, {'name': 'pvalue.threshold', 'value': '0.00001', 'label': 'Pvalue.Threshold', 'description': 'p-value threshold above which transition is quantification-worthy. Must be a value between 0 and 1 inclusive.'}, {'name': 'cv.threshold', 'value': '0.2', 'label': 'Cv.Threshold', 'description': 'threshold for coefficient of variation below which transition is quantification-worthy'}, {'name': 'all.pairs', 'value': 'FALSE', 'label': 'All.Pairs', 'description': 'calculate all possible pairs of relative ratios for a transition'}, {'name': 'output.intermediate.results', 'value': 'FALSE', 'label': 'Output.Intermediate.Results', 'description': 'create files with intermediate results'}, {'name': 'output.prefix', 'value': '<data.file_basename>-results', 'label': 'Output.Prefix', 'description': 'file name prefix used for output file(s)'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'BedToGtf': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Converts BED files to GFF or GTF format', 'required_args': [{'name': 'bed.file', 'value': '', 'label': 'Bed.File', 'description': 'A bed file'}], 'default_args': [{'name': 'task_name', 'value': 'BedToGtf', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.format', 'value': 'gtf', 'label': 'Output.Format', 'description': 'Whether to output a gff file or exon only gtf file. The bed file must contain at least 12 columns in order to generate a gtf file.'}, {'name': 'output.prefix', 'value': '<bed.file_basename>', 'label': 'Output.Prefix', 'description': 'The prefix for the output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Beroukhim.Getz.2007.PNAS.Glioma.GISTIC': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '', 'required_args': [], 'default_args': [{'name': 'task_name', 'value': 'Beroukhim.Getz.2007.PNAS.Glioma.GISTIC', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'BlastTrainTest': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Sequence similarity classification using BLAST', 'required_args': [{'name': 'train.file', 'value': '', 'label': 'Train.File', 'description': 'The training file in .FASTA format. Multi-label class names are separated by colons (:).'}, {'name': 'test.file', 'value': '', 'label': 'Test.File', 'description': 'The testing file in .FASTA format. Multi-label class names are separated by colons (:).'}], 'default_args': [{'name': 'task_name', 'value': 'BlastTrainTest', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'sub.matrix', 'value': 'PAM30', 'label': 'Sub.Matrix', 'description': 'The name of the substitution matrix used by Blast.'}, {'name': 'pred.results.file', 'value': '<test.file_basename>', 'label': 'Pred.Results.File', 'description': 'The name of the output file for prediction results.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'BlastXValidation': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Sequence similarity cross validation prediction using BLAST', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'The input sequence and annotation information in .FASTA format.'}], 'default_args': [{'name': 'task_name', 'value': 'BlastXValidation', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'num.folds', 'value': '2', 'label': 'Num.Folds', 'description': 'The number of folds should be an integer greater than or equal to two.'}, {'name': 'sub.matrix', 'value': 'PAM30', 'label': 'Sub.Matrix', 'description': 'The name of the substitution matrix used by Blast.'}, {'name': 'pred.results.file', 'value': '<input.file_basename>', 'label': 'Pred.Results.File', 'description': 'The name of the output file for prediction results.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Bowtie.aligner': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Bowtie2 (v. 2.1.0) is an ultrafast and memory-efficient short read aligner.', 'required_args': [{'name': 'prebuilt.bowtie.index', 'value': '', 'label': 'Prebuilt.Bowtie.Index', 'description': 'A prebuilt Bowtie 2 index. Either a prebuilt or a custom Bowtie 2 index must be specified.'}, {'name': 'custom.bowtie.index', 'value': '', 'label': 'Custom.Bowtie.Index', 'description': 'A ZIP archive containing Bowtie 2 index files. Either a prebuilt or a custom Bowtie 2 index must be specified.'}, {'name': 'input.format', 'value': '', 'label': 'Input.Format', 'description': 'The format of the reads input files.'}, {'name': 'reads.pair.1', 'value': '', 'label': 'Reads.Pair.1', 'description': 'Unpaired reads file or first mate for paired reads. A file or zip of files containing reads in FASTA or FASTQ format (can be compressed - ie .gz).'}, {'name': 'reads.pair.2', 'value': '', 'label': 'Reads.Pair.2', 'description': 'Second mate for paired reads. A file or zip of files in FASTA or FASTQ format (can be compressed - ie .gz).'}, {'name': 'quality.value.scale', 'value': '', 'label': 'Quality.Value.Scale', 'description': 'The scale to use for quality values.'}, {'name': 'integer.quality.values', 'value': '', 'label': 'Integer.Quality.Values', 'description': 'Whether the quality values are represented as space-separated ASCII integers (i.e 40 40 30 40 ..).'}, {'name': 'max.reads.to.align', 'value': '', 'label': 'Max.Reads.To.Align', 'description': 'The maximum number of reads to align.'}, {'name': 'mate.orientations', 'value': '', 'label': 'Mate.Orientations', 'description': 'The upstream/downstream mate orientations for a valid paired-end alignment against the forward reference strand.'}], 'default_args': [{'name': 'task_name', 'value': 'Bowtie.aligner', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'alignment.mode', 'value': '--end-to-end', 'label': 'Alignment.Mode', 'description': 'The alignment mode to use.'}, {'name': 'preset.options', 'value': 'sensitive', 'label': 'Preset.Options', 'description': 'A combination of pre-packaged options based on speed and sensitivity/accuracy.'}, {'name': 'min.fragment.length', 'value': '0', 'label': 'Min.Fragment.Length', 'description': 'The minimum fragment length for valid paired-end alignments.'}, {'name': 'max.fragment.length', 'value': '500', 'label': 'Max.Fragment.Length', 'description': 'The maximum fragment length for valid paired-end alignments.'}, {'name': 'random.seed', 'value': '123456', 'label': 'Random.Seed', 'description': 'The seed to use for the pseudo-random number generator.'}, {'name': 'output.prefix', 'value': '<reads.pair.1_basename>_<reads.pair.2_basename>', 'label': 'Output.Prefix', 'description': 'The prefix to use for the output file name.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Bowtie.indexer': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Builds a Bowtie2 (v. 2.1.0) index from a set of DNA sequences', 'required_args': [{'name': 'fasta.file', 'value': '', 'label': 'Fasta.File', 'description': 'A file or zip of files containing sequences in FASTA format'}], 'default_args': [{'name': 'task_name', 'value': 'Bowtie.indexer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'random.seed', 'value': '123456', 'label': 'Random.Seed', 'description': 'The seed for the random number generator'}, {'name': 'index.name', 'value': '<fasta.file_basename>', 'label': 'Index.Name', 'description': 'The base name of the index files'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'BWA.aln': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '[**Beta Release** Contact gp-help with any issues.  Check stdout.txt and stderr.txt for errors] A fast and accurate short-read alignment tool that allows for mismatches and gaps. Alignments are output in a SAM format file, which provides Phred-scale quality scores for each alignment.  Uses BWA 0.7.4 aln and samse/sampe.', 'required_args': [{'name': 'BWA.index', 'value': '', 'label': 'BWA.Index', 'description': 'A zip file or directory containing BWA index files.'}, {'name': 'reads.pair.1', 'value': '', 'label': 'Reads.Pair.1', 'description': 'Single-end or first paired-end reads file in FASTA, FASTQ, or BAM format.  For paired-end data, this should be the forward (\"*_1\" or \"left\") input file.  Note: the FASTA or FASTQ can be gzipped.'}, {'name': 'reads.pair.2', 'value': '', 'label': 'Reads.Pair.2', 'description': 'The reverse (\"*_2\" or \"right\") input file for paired-end reads in FASTA, FASTQ, or BAM format. Note: the FASTA or FASTQ can be gzipped.'}, {'name': 'bam.mapping', 'value': '', 'label': 'Bam.Mapping', 'description': 'Specifies how to map BAM input. This is only required if input file is in BAM format.'}, {'name': 'max.edit.distance', 'value': '', 'label': 'Max.Edit.Distance', 'description': 'The max edit distance. This specifies a threshold of the maximum number of deletions, insertions, and substitutions needed to transform the reference sequence into the read sequence.'}, {'name': 'max.num.gap', 'value': '', 'label': 'Max.Num.Gap', 'description': 'Maximum number of gap opens. This specifies a threshold of the maximum number of gaps that can be initiated to match a given read to the reference.'}, {'name': 'max.gap.extension', 'value': '', 'label': 'Max.Gap.Extension', 'description': 'Maximum number of gap extensions. This specifies a threshold of the maximum number of bases by which gaps in a read can be extended.'}, {'name': 'max.deletion.length', 'value': '', 'label': 'Max.Deletion.Length', 'description': \"Disallow a long deletion within this many bp of the 3' end.\"}, {'name': 'max.indel.length', 'value': '', 'label': 'Max.Indel.Length', 'description': 'Disallow an indel within this many bp of the ends'}, {'name': 'seed.length', 'value': '', 'label': 'Seed.Length', 'description': 'The set of bases determined by this option in the high-quality (left) end of the read is the seed.'}, {'name': 'max.seed.edit.distance', 'value': '', 'label': 'Max.Seed.Edit.Distance', 'description': 'Maximum edit distance in the seed; that is, the maximum number of changes required to transform the reference sequence of the seed into the read sequence of the seed.'}, {'name': 'mismatch.penalty', 'value': '', 'label': 'Mismatch.Penalty', 'description': 'Mismatch penalty'}, {'name': 'gap.open.penalty', 'value': '', 'label': 'Gap.Open.Penalty', 'description': 'Gap open penalty.  The gap open penalty is the score taken away for the initiation of the gap in sequence. To make the match more significant you can try to make the gap penalty larger.'}, {'name': 'gap.extension.penalty', 'value': '', 'label': 'Gap.Extension.Penalty', 'description': 'Gap extension penalty.  The gap extension penalty is added to the standard gap penalty for each base or residue in the gap.  To reduce long gaps, increase the extension gap penalty.'}, {'name': 'max.best.hits', 'value': '', 'label': 'Max.Best.Hits', 'description': 'Proceed with suboptimal alignments if there are no more than this many equally best hits. This option only affects paired-end mapping.'}, {'name': 'trim.reads', 'value': '', 'label': 'Trim.Reads', 'description': 'Specifies a quality threshold for read trimming'}, {'name': 'barcode.length', 'value': '', 'label': 'Barcode.Length', 'description': \"Length of barcode starting from the 5' end\"}, {'name': 'max.insert.size', 'value': '', 'label': 'Max.Insert.Size', 'description': 'Specifies the maximum insert size for a read pair to be considered to be mapped properly'}, {'name': 'max.occurrences', 'value': '', 'label': 'Max.Occurrences', 'description': 'Specifies the maximum occurrences of a read for pairing'}, {'name': 'max.alignments', 'value': '', 'label': 'Max.Alignments', 'description': 'Maximum number of alignments to output in the XA tag for reads paired properly'}, {'name': 'max.dc.alignments', 'value': '', 'label': 'Max.Dc.Alignments', 'description': 'Maximum number of alignments to output in the XA tag for disconcordant read pairs (excluding singletons)'}], 'default_args': [{'name': 'task_name', 'value': 'BWA.aln', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'iterative.search', 'value': 'no', 'label': 'Iterative.Search', 'description': 'Whether to disable iterative search'}, {'name': 'Illumina.1.3.format', 'value': 'no', 'label': 'Illumina.1.3.Format', 'description': 'The input is in the Illumina 1.3+ read format'}, {'name': 'output.prefix', 'value': '<reads.pair.1_basename>', 'label': 'Output.Prefix', 'description': 'Prefix to use for output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'BWA.bwasw': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '[**Beta Release** Contact gp-help with any issues.  Check stdout.txt and stderr.txt for errors] Aligns long sequences (>200 bp) to a sequence database using BWA 0.7.4 bwasw.', 'required_args': [{'name': 'BWA.index', 'value': '', 'label': 'BWA.Index', 'description': 'A zip file or directory containing BWA index files.'}, {'name': 'read.file', 'value': '', 'label': 'Read.File', 'description': 'Single-end or first paired-end reads file in FASTA or FASTQ format.  For paired-end data, this should be the forward (\"*_1\" or \"left\") input file.'}, {'name': 'mate.read.file', 'value': '', 'label': 'Mate.Read.File', 'description': 'The reverse (\"*_2\" or \"right\") reads file for paired-end reads in FASTA or FASTQ format.'}, {'name': 'match.score', 'value': '', 'label': 'Match.Score', 'description': 'The score of the match'}, {'name': 'mismatch.penalty', 'value': '', 'label': 'Mismatch.Penalty', 'description': 'Specifies the mismatch penalty.'}, {'name': 'gap.open.penalty', 'value': '', 'label': 'Gap.Open.Penalty', 'description': 'Gap open penalty.  The gap open penalty is the score taken away for the initiation of the gap in sequence. To make the match more significant you can try to make the gap penalty larger.'}, {'name': 'gap.extension.penalty', 'value': '', 'label': 'Gap.Extension.Penalty', 'description': 'Gap extension penalty.  The gap extension penalty is added to the standard gap penalty for each base or residue in the gap.  To reduce long gaps, increase the extension gap penalty. '}, {'name': 'band.width', 'value': '', 'label': 'Band.Width', 'description': 'band width in the banded alignment'}, {'name': 'min.score.threshold', 'value': '', 'label': 'Min.Score.Threshold', 'description': 'Minimum score threshold divided by the match score'}, {'name': 'threshold.coefficient', 'value': '', 'label': 'Threshold.Coefficient', 'description': 'Coefficient for threshold adjustment according to query length'}, {'name': 'z.best.heuristics', 'value': '', 'label': 'Z.Best.Heuristics', 'description': 'Z-best heuristics. Specifying a higher number increases accuracy at the cost of speed'}, {'name': 'max.sa.interval.size', 'value': '', 'label': 'Max.Sa.Interval.Size', 'description': 'Maximum SA interval size for initiating a seed. Specifying a higher number increases accuracy at the cost of speed.'}, {'name': 'min.num.seeds', 'value': '', 'label': 'Min.Num.Seeds', 'description': 'The minimum number of seeds contained in the best alignment from the forward-forward alignment process that allows the algorithm to skip performing the reverse alignment. BWA.bwasw tends to be faster and more accurate if the alignment is supported by more seeds.'}], 'default_args': [{'name': 'task_name', 'value': 'BWA.bwasw', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'num.threads', 'value': '1', 'label': 'Num.Threads', 'description': 'Number of threads'}, {'name': 'output.prefix', 'value': '<read.file_basename>', 'label': 'Output.Prefix', 'description': 'Prefix to use for the output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'BWA.indexer': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '[**Beta Release** Contact gp-help with any issues.  Check stdout.txt and stderr.txt for errors] Creates a BWT index from a sequence FASTA file for use with the BWA 0.7.4 aligners.', 'required_args': [{'name': 'fasta.file', 'value': '', 'label': 'Fasta.File', 'description': 'A sequence fasta file'}, {'name': 'algorithm', 'value': '', 'label': 'Algorithm', 'description': 'The algorithm to use for constructing the BWT index'}], 'default_args': [{'name': 'task_name', 'value': 'BWA.indexer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.prefix', 'value': '<fasta.file_basename>', 'label': 'Output.Prefix', 'description': 'A prefix for the output file name.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'CaArray2ImportViewer': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Imports data files from CaArray 2.4.1 and creates gct or cls files ', 'required_args': [{'name': 'experiment', 'value': '', 'label': 'Experiment', 'description': 'Title or identifier of the caArray experiment to import data from.  https://array.nci.nih.gov/caarray/home.action'}], 'default_args': [{'name': 'task_name', 'value': 'CaArray2ImportViewer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'url', 'value': 'http://array.nci.nih.gov:8080', 'label': 'Url', 'description': 'URL for caArray server.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'CART': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Classification and Regression Tree', 'required_args': [{'name': 'train.data.filename', 'value': '', 'label': 'Train.Data.Filename', 'description': 'The training data file - .gct, .res, ignored if a saved model (saved.model.filename) is used'}, {'name': 'train.cls.filename', 'value': '', 'label': 'Train.Cls.Filename', 'description': 'The training class file - .cls, ignored if a saved model (saved.model.filename) is used'}, {'name': 'saved.model.filename', 'value': '', 'label': 'Saved.Model.Filename', 'description': 'input CART model'}, {'name': 'test.data.filename', 'value': '', 'label': 'Test.Data.Filename', 'description': 'The test data file - .gct, .res'}, {'name': 'test.cls.filename', 'value': '', 'label': 'Test.Cls.Filename', 'description': 'The test class file - .cls'}], 'default_args': [{'name': 'task_name', 'value': 'CART', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'model.output.file', 'value': '<train.data.filename_basename>.cart.model', 'label': 'Model.Output.File', 'description': 'name of output CART model file  '}, {'name': 'pred.results.file', 'value': '<test.data.filename_basename>.pred.odf', 'label': 'Pred.Results.File', 'description': 'The name of the output file for prediction results'}, {'name': 'tree.output.file', 'value': '<test.data.filename_basename>.tree.pdf', 'label': 'Tree.Output.File', 'description': 'The name of the file containing a plot of the classification tree'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'CARTXValidation': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Classification and Regression Tree Cross-Validation', 'required_args': [{'name': 'data.filename', 'value': '', 'label': 'Data.Filename', 'description': 'The data file - .gct, .res'}, {'name': 'cls.filename', 'value': '', 'label': 'Cls.Filename', 'description': 'The class file - .cls '}], 'default_args': [{'name': 'task_name', 'value': 'CARTXValidation', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'prediction.results.file', 'value': '<data.filename_basename>.pred.odf', 'label': 'Prediction.Results.File', 'description': 'The name of the output file for prediction results'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'CBS': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Segments DNA copy number data into regions of estimated equal copy number using circular binary segmentation (CBS). ', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'Copy number file - .cn'}, {'name': 'sample.name', 'value': '', 'label': 'Sample.Name', 'description': 'The name of the sample to use. Leave blank to use the first sample in the input file.'}], 'default_args': [{'name': 'task_name', 'value': 'CBS', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'permutations', 'value': '10000', 'label': 'Permutations', 'description': 'Number of permutations used for p-value computation'}, {'name': 'significance.level', 'value': '0.01', 'label': 'Significance.Level', 'description': 'Significance levels for the test to accept change-points'}, {'name': 'output.prefix', 'value': '<input.file_basename>', 'label': 'Output.Prefix', 'description': 'The prefix to use for the output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ChIPSeq.CreateHeatmap': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Generates a heatmap based on the ChIP-Seq signal extracted from a BAM file, according to the chromosomal locations specified in the BED file used in ExtractSignal.  This is an experimental module built for the Broad CEGS workshop of July 29-30, 2013.', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'Input file in TAB format, generated by ChipSeq.ExtractSignal'}], 'default_args': [{'name': 'task_name', 'value': 'ChIPSeq.CreateHeatmap', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'normalization', 'value': 'z-score', 'label': 'Normalization', 'description': 'Choose a normalization method'}, {'name': 'percent.start', 'value': '0.45', 'label': 'Percent.Start', 'description': 'percent of the bed region to start clustering at'}, {'name': 'percent.end', 'value': '0.55', 'label': 'Percent.End', 'description': 'percent of the bed region to end clustering at '}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ChIPSeq.CreatePlots': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Generates a scatter plot and correlation matrix based on the ChIP-Seq signal extracted from a BAM file, according to the chromosomal locations specified in the BED file used in ExtractSignal.  This is an experimental module built for the Broad CEGS workshop of July 29-30, 2013.', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'Input file in TAB format, generated by ChipSeq.ExtractSignal'}], 'default_args': [{'name': 'task_name', 'value': 'ChIPSeq.CreatePlots', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ChIPSeq.ExtractSignal': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Segment the read from a BAM file into bins by chromosome and extract the signal values to a tab delimited file for subsequent visualization.  This is an experimental module built for the Broad CEGS workshop of July 29-30, 2013.', 'required_args': [{'name': 'bam.file', 'value': '', 'label': 'Bam.File', 'description': 'Input file in BAM format.'}, {'name': 'chromosome.sizes.file', 'value': '', 'label': 'Chromosome.Sizes.File', 'description': \"Specify a chromosome sizes file.  Chromosome names must start with 'chr'.\"}, {'name': 'bin.size', 'value': '', 'label': 'Bin.Size', 'description': 'The desired size of the bins.  Defaults to 25 if no value is supplied.'}, {'name': 'whole.cell.extract.control.file', 'value': '', 'label': 'Whole.Cell.Extract.Control.File', 'description': 'A bam file containing Whole Cell Extract control data to use in normalizing the bin values.'}, {'name': 'region.bed.file', 'value': '', 'label': 'Region.Bed.File', 'description': 'The BED file containing regions of interest.'}, {'name': 'region.bin.size', 'value': '', 'label': 'Region.Bin.Size', 'description': 'Bin the bed regions into sub-regions of this size (in bps).  Must be a multiple of the bin size such that an integer number of bin file bins greater than zero fits into a single region bin. (E.g. One could have bin files of 25 bp, extend bed regions by 100 bp, then have region bins of 50 bp for a total of four region bins.). Defaults to bin.size if no value is supplied.'}, {'name': 'extend.by', 'value': '', 'label': 'Extend.By', 'description': 'The number of basepairs to extend a region from its center.  Total extended region size will be twice this value.  This option is most useful if one wants to analyze  regions of the same length and number of bins.  To simply use the bin boundaries nearest the BED coordinates (i.e. analyze variable numbers of bins per BED line), do not specify this parameter.'}], 'default_args': [{'name': 'task_name', 'value': 'ChIPSeq.ExtractSignal', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'assembly.name', 'value': 'hg19', 'label': 'Assembly.Name', 'description': 'Specify an assembly (i.e. hg19, hg18, etc.).'}, {'name': 'normalize', 'value': 'yes', 'label': 'Normalize', 'description': 'Normalize each bin value by scaling it to 20000000 reads.'}, {'name': 'aggregation.function', 'value': 'avg', 'label': 'Aggregation.Function', 'description': 'Function to run on the bin values over the regions of interest.'}, {'name': 'output.filename', 'value': '<bam.file_basename>.<region.bed.file_basename>.tab', 'label': 'Output.Filename', 'description': 'The name of the output file.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ClassNeighbors': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Select genes that most closely resemble a profile', 'required_args': [{'name': 'data.filename', 'value': '', 'label': 'Data.Filename', 'description': 'Dataset file - .res, .gct, .odf'}, {'name': 'class.filename', 'value': '', 'label': 'Class.Filename', 'description': 'class file - .cls, .odf'}], 'default_args': [{'name': 'task_name', 'value': 'ClassNeighbors', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'marker.gene.list.file', 'value': '<data.filename_basename>.genelist', 'label': 'Marker.Gene.List.File', 'description': 'output filename for marker gene list results - .odf file'}, {'name': 'marker.data.set.file', 'value': '<data.filename_basename>.neighbors', 'label': 'Marker.Data.Set.File', 'description': 'output filename for marker data - .gct file'}, {'name': 'num.neighbors', 'value': '50', 'label': 'Num.Neighbors', 'description': 'number of neighbors to find'}, {'name': 'num.permutations', 'value': '100', 'label': 'Num.Permutations', 'description': 'number of permutations in permutation test'}, {'name': 'user.pval', 'value': '0.5', 'label': 'User.Pval', 'description': 'user-set significance value for permutation test'}, {'name': 'mean.or.median', 'value': 'mean', 'label': 'Mean.Or.Median', 'description': 'use mean or median for feature selection'}, {'name': 'ttest.or.snr', 'value': 't-test', 'label': 'Ttest.Or.Snr', 'description': 'use signal-to-noise or t-test to select neighbors'}, {'name': 'filter.data', 'value': 'no', 'label': 'Filter.Data', 'description': 'if no, values below will be ignored'}, {'name': 'min.threshold', 'value': '10', 'label': 'Min.Threshold', 'description': 'minimum threshold for data'}, {'name': 'max.threshold', 'value': '16000', 'label': 'Max.Threshold', 'description': 'maximum threshold for data'}, {'name': 'min.fold.diff', 'value': '5', 'label': 'Min.Fold.Diff', 'description': 'minimum fold difference for filtering genes'}, {'name': 'min.abs.diff', 'value': '50', 'label': 'Min.Abs.Diff', 'description': 'minimum absolute difference for filtering genes'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ClsFileCreator': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'A tool to create a class label (CLS) file.', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'A dataset - .gct'}], 'default_args': [{'name': 'task_name', 'value': 'ClsFileCreator', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'CNToXCN': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '', 'required_args': [{'name': 'cn.file', 'value': '', 'label': 'Cn.File', 'description': 'Input CN file (a tab-delimited file format that contains SNP copy numbers)'}], 'default_args': [{'name': 'task_name', 'value': 'CNToXCN', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.prefix', 'value': '<cn.file_basename>', 'label': 'Output.Prefix', 'description': 'Prefix for the output XCN file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'CoClustering': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Co-clustering of RNA and global proteome data', 'required_args': [], 'default_args': [{'name': 'task_name', 'value': 'CoClustering', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'min.correlation', 'value': '0.4', 'label': 'Min.Correlation', 'description': 'Minimum correlation between RNA and proteome for gene to be included in clustering'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'CollapseDataset': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Collapses all probe set values for a gene into a single vector of values', 'required_args': [{'name': 'dataset.file', 'value': '', 'label': 'Dataset.File', 'description': 'Dataset file - .gct, .res'}, {'name': 'chip.platform', 'value': '', 'label': 'Chip.Platform', 'description': 'The chip platform'}], 'default_args': [{'name': 'task_name', 'value': 'CollapseDataset', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'collapse.mode', 'value': 'Maximum', 'label': 'Collapse.Mode', 'description': 'Collapse mode for probe sets => 1 gene'}, {'name': 'output.file.name', 'value': '<dataset.file_basename>.collapsed', 'label': 'Output.File.Name', 'description': 'The output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ComBat': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Batch Correction', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'input file - .res, .gct'}, {'name': 'sample.info.file', 'value': '', 'label': 'Sample.Info.File', 'description': 'A file containing information about the samples. The sample info file must contain the three columns: Array, Sample, and Batch. Any additional columns are used as covariates in the analysis.(.txt file)'}, {'name': 'absent.calls.filter', 'value': '', 'label': 'Absent.Calls.Filter', 'description': 'The filter to apply to genes with absent calls (a value between 0 and 1). Filters genes with absent calls in 1-(absent calls filter) of the samples. Ignored if input file is not a .res file.\\t'}], 'default_args': [{'name': 'task_name', 'value': 'ComBat', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'covariate.columns', 'value': 'all', 'label': 'Covariate.Columns', 'description': \"The covariate columns to use in analysis. This is either set to all, none, or a list specifying one or more covariate columns from sample info file to use in the analysis (i.e, 4, 5, 7, ...). If this is set to 'all' then all the covariate columns are used. If this is set to 'none' then none of the covariate columns are used.\"}, {'name': 'create.prior.plots', 'value': 'T', 'label': 'Create.Prior.Plots', 'description': 'Whether to generate prior probability distribution plots. Ignored if non-parametric method is used.'}, {'name': 'prior.method', 'value': 'T', 'label': 'Prior.Method', 'description': 'Empirical Bayes priors distribution estimation method to use. Either a parametric or non-parametric method can be applied. If unsure of which method to use try the parametric method first since it is faster.'}, {'name': 'output.file', 'value': '<input.file_basename>.combat.<input.file_extension>', 'label': 'Output.File', 'description': 'Name of batch adjusted output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'CombineOdf': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Combine two prediction results files into a single, weighted, multi-label prediction results file', 'required_args': [{'name': 'pred1.file', 'value': '', 'label': 'Pred1.File', 'description': 'The first prediction file in .odf format.'}, {'name': 'pred2.file', 'value': '', 'label': 'Pred2.File', 'description': 'The second prediction file in .odf format.'}], 'default_args': [{'name': 'task_name', 'value': 'CombineOdf', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'pred1.file.weight', 'value': '0.5', 'label': 'Pred1.File.Weight', 'description': 'Weight assigned to classifier that produced first prediction file.'}, {'name': 'pred.file', 'value': '<pred1.file_basename>_<pred2.file_basename>', 'label': 'Pred.File', 'description': 'The name of the prediction file in modified .odf format.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ComparativeMarkerSelection': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Identify differentially expressed genes that can discriminate between distinct classes of samples.', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'The input file - .res, .gct. Note that if your data is log transformed, you will need to set the \"log transformed data\" parameter below to \"yes\".'}, {'name': 'cls.file', 'value': '', 'label': 'Cls.File', 'description': 'The class file - .cls'}, {'name': 'confounding.variable.cls.file', 'value': '', 'label': 'Confounding.Variable.Cls.File', 'description': 'The class file containing the confounding variable.'}, {'name': 'min.std', 'value': '', 'label': 'Min.Std', 'description': 'The minimum standard deviation if test statistic includes min std option'}], 'default_args': [{'name': 'task_name', 'value': 'ComparativeMarkerSelection', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'test.direction', 'value': '2', 'label': 'Test.Direction', 'description': 'The test to perform'}, {'name': 'test.statistic', 'value': '0', 'label': 'Test.Statistic', 'description': 'The statistic to use'}, {'name': 'number.of.permutations', 'value': '10000', 'label': 'Number.Of.Permutations', 'description': 'The number of permutations to perform (use 0 to calculate asymptotic p-values using the standard independent two-sample t-test)'}, {'name': 'log.transformed.data', 'value': 'false', 'label': 'Log.Transformed.Data', 'description': 'Whether the input data has been log transformed'}, {'name': 'complete', 'value': 'false', 'label': 'Complete', 'description': 'Whether to perform all possible permutations'}, {'name': 'balanced', 'value': 'false', 'label': 'Balanced', 'description': 'Whether to perform balanced permutations'}, {'name': 'random.seed', 'value': '779948241', 'label': 'Random.Seed', 'description': 'The seed of the random number generator used to produce permutations'}, {'name': 'smooth.p.values', 'value': 'true', 'label': 'Smooth.P.Values', 'description': 'Whether to smooth p-values'}, {'name': 'phenotype.test', 'value': 'one versus all', 'label': 'Phenotype.Test', 'description': 'Tests to perform when cls file has more than two classes (Note: not fully corrected for multiple hypothesis testing)'}, {'name': 'output.filename', 'value': '<input.file_basename>.comp.marker.odf', 'label': 'Output.Filename', 'description': 'The name of the output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ComparativeMarkerSelectionViewer': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Views the results from ComparativeMarkerSelection', 'required_args': [{'name': 'comparative.marker.selection.filename', 'value': '', 'label': 'Comparative.Marker.Selection.Filename', 'description': 'The output from ComparativeMarkerSelection'}, {'name': 'dataset.filename', 'value': '', 'label': 'Dataset.Filename', 'description': 'The dataset file used as input to ComparativeMarkerSelection'}], 'default_args': [{'name': 'task_name', 'value': 'ComparativeMarkerSelectionViewer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'CompareSpectra': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Compares two spectra to determine similarity', 'required_args': [{'name': 'spectrum1.filename', 'value': '', 'label': 'Spectrum1.Filename', 'description': 'Spectrum file -.csv'}, {'name': 'spectrum2.filename', 'value': '', 'label': 'Spectrum2.Filename', 'description': 'Spectrum file - .csv'}, {'name': 'low.Da', 'value': '', 'label': 'Low.Da', 'description': 'Minimum M/Z to include'}, {'name': 'high.Da', 'value': '', 'label': 'High.Da', 'description': 'Maximum M/Z to include'}], 'default_args': [{'name': 'task_name', 'value': 'CompareSpectra', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<spectrum1.filename_basename>_<spectrum2.filename_basename>_compare.odf', 'label': 'Output.File', 'description': 'Output file name - .odf'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'CompensateFCS': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Compensates an FCS data file', 'required_args': [{'name': 'FCS.data.file', 'value': '', 'label': 'FCS.Data.File', 'description': 'A FCS data file to be compensated'}, {'name': 'Spillover.matrix.file', 'value': '', 'label': 'Spillover.Matrix.File', 'description': 'A text file with spillover matrix. Use this parameter if there is no spillover matrix in the input FCS file or to use a spillover matrix different from the matrix in the input FCS file. The value from the FCS data file will be used if this parameter is not specified.'}], 'default_args': [{'name': 'task_name', 'value': 'CompensateFCS', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Output.file.name', 'value': '<FCS.data.file_basename>.compensated', 'label': 'Output.File.Name', 'description': 'The name of the output file. The extension will match the file format specified in Output file format.'}, {'name': 'Output.file.format', 'value': 'FCS', 'label': 'Output.File.Format', 'description': 'The file format in which to save the compensated data.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ConcatenateFilelist': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Concatenate all of the items in the input filelist into a single large file.', 'required_args': [{'name': 'filelist', 'value': '', 'label': 'Filelist', 'description': 'a text file containing a list of file paths'}, {'name': 'num.header.lines', 'value': '', 'label': 'Num.Header.Lines', 'description': 'specify number of header lines so that they are output only once'}, {'name': 'output.filename', 'value': '', 'label': 'Output.Filename', 'description': 'The name of the output file'}], 'default_args': [{'name': 'task_name', 'value': 'ConcatenateFilelist', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ConsensusClustering': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Resampling-based clustering method ', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'The data to be clustered (.gct, .res, .odf)'}, {'name': 'cluster.by', 'value': '', 'label': 'Cluster.By', 'description': 'Whether to cluster by rows/genes or columns/experiments'}, {'name': 'create.heat.map', 'value': '', 'label': 'Create.Heat.Map', 'description': 'Whether to create heatmaps (one for each cluster number)'}], 'default_args': [{'name': 'task_name', 'value': 'ConsensusClustering', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'kmax', 'value': '5', 'label': 'Kmax', 'description': 'Try K=2,3,...,kmax clusters (must be > 1)'}, {'name': 'resampling.iterations', 'value': '20', 'label': 'Resampling.Iterations', 'description': 'Number of resampling iterations'}, {'name': 'seed.value', 'value': '12345', 'label': 'Seed.Value', 'description': 'Random number generator seed'}, {'name': 'clustering.algorithm', 'value': 'HIERARCHICAL', 'label': 'Clustering.Algorithm', 'description': 'Type of clustering algorithm'}, {'name': 'distance.measure', 'value': 'EUCLIDEAN', 'label': 'Distance.Measure', 'description': 'Distance measure'}, {'name': 'resample', 'value': 'subsample', 'label': 'Resample', 'description': \"resampling scheme (one of 'subsample[ratio]', 'features[nfeat]', 'nosampling'). The default is subsampling with a proportion of 0.80. NOTE: The square brackets should not be included (i.e subsample0.7, features210).\"}, {'name': 'merge.type', 'value': 'average', 'label': 'Merge.Type', 'description': 'Ignored when algorithm other than hierarchical selected'}, {'name': 'descent.iterations', 'value': '2000', 'label': 'Descent.Iterations', 'description': 'Number of SOM/NMF iterations'}, {'name': 'output.stub', 'value': '<input.filename_basename>', 'label': 'Output.Stub', 'description': 'Stub pre-pended to all the output file names'}, {'name': 'normalize.type', 'value': '-n1', 'label': 'Normalize.Type', 'description': 'row-wise, column-wise, both'}, {'name': 'normalization.iterations', 'value': '0', 'label': 'Normalization.Iterations', 'description': 'number of row/column normalization iterations (supercedes normalize.type)'}, {'name': 'heat.map.size', 'value': '2', 'label': 'Heat.Map.Size', 'description': \"point size of a consensus matrix's heat map (between 1 and 20)\"}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ConstellationMap': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'A downstream visualization and analysis tool for gene set enrichment results with interactive web visualizer.', 'required_args': [{'name': 'input.gct.file', 'value': '', 'label': 'Input.Gct.File', 'description': 'Input gene set enrichment file. Expected to be output from ssGSEAProjection module.'}, {'name': 'input.cls.file', 'value': '', 'label': 'Input.Cls.File', 'description': 'Phenotype labels. Input cls file describing phenotypes of gct file.'}, {'name': 'gene.sets.file', 'value': '', 'label': 'Gene.Sets.File', 'description': 'Gene set file. Must match gene sets listed in input gct file. '}, {'name': 'top.n', 'value': '', 'label': 'Top.N', 'description': 'The number of gene sets to be plotted.'}, {'name': 'target.class', 'value': '', 'label': 'Target.Class', 'description': 'The phenotypic category of interest.'}], 'default_args': [{'name': 'task_name', 'value': 'ConstellationMap', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'gene.sets.database', 'value': 'c1.all.v5.1.symbols.gmt', 'label': 'Gene.Sets.Database', 'description': 'Gene sets database from GSEA website.'}, {'name': 'direction', 'value': 'positive', 'label': 'Direction', 'description': 'The direction of correlation, as measured by mutual information, to be investigated.'}, {'name': 'image.format', 'value': 'PNG', 'label': 'Image.Format', 'description': 'The image file type of outputted plots, either bitmap (PNG) or vector graphics (PDF).'}, {'name': 'jaccard.threshold', 'value': '0.1', 'label': 'Jaccard.Threshold', 'description': 'The Jaccard index threshold above which connective edges between gene sets will be plotted.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ConvertLineEndings': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': \"Converts line endings to the host operating system's format.\", 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'The input file (any non-binary file format)'}], 'default_args': [{'name': 'task_name', 'value': 'ConvertLineEndings', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<input.filename_basename>.cvt.<input.filename_extension>', 'label': 'Output.File', 'description': 'The output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ConvertToMAGEML': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Converts a gct, res, or odf dataset file to a MAGE-ML file', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'The gct, res, or odf file to convert'}], 'default_args': [{'name': 'task_name', 'value': 'ConvertToMAGEML', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file.base.name', 'value': '<input.filename_basename>', 'label': 'Output.File.Base.Name', 'description': 'The base name of the zip output file or the MAGE-ML file'}, {'name': 'create.zip.file', 'value': 'yes', 'label': 'Create.Zip.File', 'description': 'Whether to zip the MAGE-ML files'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ConvertToMAGETAB': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'A module to export data from GenePattern in MAGE-TAB format', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'A dataset file  - .gct, .res'}, {'name': 'cls.file', 'value': '', 'label': 'Cls.File', 'description': 'A class file  - .cls'}, {'name': 'intensity.type.label', 'value': '', 'label': 'Intensity.Type.Label', 'description': 'label that specifies the type of intensity values in data matrix file '}], 'default_args': [{'name': 'task_name', 'value': 'ConvertToMAGETAB', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'sample.reference.label', 'value': 'Hybridization REF', 'label': 'Sample.Reference.Label', 'description': 'reference label for samples in the data matrix file'}, {'name': 'feature.reference.label', 'value': 'CompositeElement REF', 'label': 'Feature.Reference.Label', 'description': 'reference label for features in the data matrix file'}, {'name': 'output.prefix', 'value': '<input.file_basename>', 'label': 'Output.Prefix', 'description': 'prefix to use for output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'COPA': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Performs Cancer Outlier Profile Analysis (COPA)', 'required_args': [{'name': 'dataset.file', 'value': '', 'label': 'Dataset.File', 'description': 'The dataset file - .res, .gct'}, {'name': 'sample.info.file', 'value': '', 'label': 'Sample.Info.File', 'description': 'sample info file - .txt'}], 'default_args': [{'name': 'task_name', 'value': 'COPA', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'outlier.cutoff', 'value': '5', 'label': 'Outlier.Cutoff', 'description': \"The cutoff to determine 'outlier' status.\"}, {'name': 'pre-filter.percentile', 'value': '0.95', 'label': 'Pre-Filter.Percentile', 'description': 'The percentile to use for pre-filtering the data. All genes with a number of outlier samples less than the top (default 95th) percentile will be removed from further consideration. This value must be between 0 and 1 exclusive.'}, {'name': 'compute.large', 'value': 'no', 'label': 'Compute.Large', 'description': 'Whether to continue analysis when the number of genes remaining after pre-filtering is greater than 1000'}, {'name': 'max.overlap', 'value': '0', 'label': 'Max.Overlap', 'description': \"The maximum number of samples that can be considered 'outliers' when comparing two genes. The default is 0, indicating that there can be no overlap.\"}, {'name': 'normal.outliers', 'value': '0', 'label': 'Normal.Outliers', 'description': \"The number of normal samples that can be considered 'outliers'. The default is 0, meaning that no normals may be outliers.\"}, {'name': 'gene.pairs.to.plot', 'value': '1-5', 'label': 'Gene.Pairs.To.Plot', 'description': 'A range of gene pairs to plot. (e.g., 2-5) based on the gene pair rank. The ranks are determined by the number of outliers. The more outliers a gene pair has the greater the ranking. Specify none if no gene pair plots should be generated.'}, {'name': 'output.file', 'value': '<dataset.file_basename>', 'label': 'Output.File', 'description': 'The name of the output files'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'CopyNumberDivideByNormals': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Determines the copy number of a target SNP', 'required_args': [{'name': 'snp.file', 'value': '', 'label': 'Snp.File', 'description': 'SNP file containing both normal and target samples.'}, {'name': 'sample.info.file', 'value': '', 'label': 'Sample.Info.File', 'description': 'dChip format sample info file containing Ploidy(numeric) and Sample columns'}], 'default_args': [{'name': 'task_name', 'value': 'CopyNumberDivideByNormals', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'include.genotype.calls', 'value': 'cn', 'label': 'Include.Genotype.Calls', 'description': 'Whether to include genotype calls from the input snp file if they exist.'}, {'name': 'output.file.name', 'value': '<snp.file_basename>', 'label': 'Output.File.Name', 'description': 'Output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'CopyNumberInferencePipeline': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '[**Available on public server only**] Generates segmented copy number calls from raw tumor and normal SNP6 CEL files.', 'required_args': [{'name': 'CopyNumberInferencePipeline.Part1of21.SNPFileCreator_SNP62.cel.files', 'value': '', 'label': 'CopyNumberInferencePipeline.Part1Of21.SNPFileCreator SNP62.Cel.Files', 'description': 'Zip of SNP 6.0 CEL files'}, {'name': 'CopyNumberInferencePipeline.Part1of21.ConvertLineEndings1.input.filename', 'value': '', 'label': 'CopyNumberInferencePipeline.Part1Of21.ConvertLineEndings1.Input.Filename', 'description': 'The input file (any non-binary file format)'}], 'default_args': [{'name': 'task_name', 'value': 'CopyNumberInferencePipeline', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'CopyNumberInferencePipeline.Part1of21.TangentNormalization9.output.files', 'value': 'nocnv', 'label': 'CopyNumberInferencePipeline.Part1Of21.TangentNormalization9.Output.Files', 'description': 'The types of files to output. Whether to output files with CNVs included, removed, or both.'}, {'name': 'CopyNumberInferencePipeline.Part2of2.Wrapper4.CopyNumberInferencePipeline.Part2of21.InsertGenomicLocations3.genome.annotation', 'value': 'hg19', 'label': 'CopyNumberInferencePipeline.Part2Of2.Wrapper4.CopyNumberInferencePipeline.Part2Of21.InsertGenomicLocations3.Genome.Annotation', 'description': 'The internally provided genome annotation to insert into the input file.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'CopyNumberNoise': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Measures noise, calculates QC metrics on copynumber data.', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'The quality scores computed by the CopyNumberInference module. '}], 'default_args': [{'name': 'task_name', 'value': 'CopyNumberNoise', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<input.file_basename>.noise.txt', 'label': 'Output.File', 'description': 'The name of the output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'copyOfReorderGCT': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Reorder a GCT to be more human readable by sorting on a property such as the Gene Symbol, mean expression, sum and variance', 'required_args': [{'name': 'gct.file', 'value': '', 'label': 'Gct.File', 'description': 'A GCT file to be reordered'}], 'default_args': [{'name': 'task_name', 'value': 'copyOfReorderGCT', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'sort.feature', 'value': 'geneSymbol', 'label': 'Sort.Feature', 'description': 'The feature to sort on'}, {'name': 'order.as', 'value': 'a', 'label': 'Order.As', 'description': 'sort in Ascending or Descending order'}, {'name': 'output.file', 'value': '<gct.file_basename>_reord.gct', 'label': 'Output.File', 'description': 'Name of the reordered GCT file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'CoxRegression': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Cox regression using clinical data and .cls file', 'required_args': [{'name': 'input.surv.data.filename', 'value': '', 'label': 'Input.Surv.Data.Filename', 'description': 'input clinical data. Case ID in 1st column (.txt file)'}, {'name': 'input.cls.filename', 'value': '', 'label': 'Input.Cls.Filename', 'description': 'input class information (.cls file)'}], 'default_args': [{'name': 'task_name', 'value': 'CoxRegression', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': 'CoxRegression_result', 'label': 'Output.File', 'description': ''}, {'name': 'time', 'value': 'time', 'label': 'Time', 'description': 'Column name for survival time'}, {'name': 'status', 'value': 'status', 'label': 'Status', 'description': 'Column name for censor (0: censored/alive, 1: event/death)'}, {'name': 'variable.continuous', 'value': 'NA', 'label': 'Variable.Continuous', 'description': '(e.g. age,labdata1,...)'}, {'name': 'variable.category', 'value': 'NA', 'label': 'Variable.Category', 'description': '(e.g. sex,cls,grade,...)'}, {'name': 'variable.interaction.terms', 'value': 'NA', 'label': 'Variable.Interaction.Terms', 'description': '(e.g. sex:cls,sex:grade,...)'}, {'name': 'strata', 'value': 'NA', 'label': 'Strata', 'description': 'Fit is done for each strata'}, {'name': 'input.subgroup', 'value': 'NA', 'label': 'Input.Subgroup', 'description': '(e.g. cls:2) *only one subgroup can be specified'}, {'name': 'variable.selection', 'value': 'none', 'label': 'Variable.Selection', 'description': ''}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'CreateSymlinks': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Creates symlinks to the input files in the job results directory. This is intended to be a helper module for creating scatter-gather pipelines.  Note that this module is not supported on Windows', 'required_args': [{'name': 'input.files', 'value': '', 'label': 'Input.Files', 'description': 'One or more input files'}], 'default_args': [{'name': 'task_name', 'value': 'CreateSymlinks', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'CsvToFcs': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Converts a Comma separated values (CSV) file to a Flow Cytometry Standard (FCS) file.', 'required_args': [{'name': 'Input.CSV.data.file', 'value': '', 'label': 'Input.CSV.Data.File', 'description': 'A CSV file to be converted to FCS.'}], 'default_args': [{'name': 'task_name', 'value': 'CsvToFcs', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Output.FCS.file.name', 'value': '<Input.CSV.data.file_basename>.fcs', 'label': 'Output.FCS.File.Name', 'description': 'The name of the output FCS file.'}, {'name': 'Range', 'value': 'auto', 'label': 'Range', 'description': 'Value of the $PnR keyword for all parameters. By default, this tool will try to set the best settings based on the data in the file; use \"auto\", \"auto-exact\", \"BD\", or an integer value.'}, {'name': 'Data.type', 'value': 'auto', 'label': 'Data.Type', 'description': 'Value of the $DATATYPE keyword. By default, this tool will try to set the best settings based on the data in the input CSV file.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Cuffcompare': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Cufflinks 2.0.2 - Cuffcompare - Analyzes the transcribed fragments in an assembly', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'One or more GTF file outputs from Cufflinks.'}, {'name': 'reference.GTF', 'value': '', 'label': 'Reference.GTF', 'description': 'A reference annotation GTF'}, {'name': 'reference.genome.file', 'value': '', 'label': 'Reference.Genome.File', 'description': 'Fasta file or zip of fasta files against which your reads were aligned'}, {'name': 'additional.cuffcompare.options', 'value': '', 'label': 'Additional.Cuffcompare.Options', 'description': 'Additional options to be passed along to the Cuffcompare program at the command line.  This parameter gives you a means to specify otherwise unavailable Cuffcompare options and switches not supported by the module; check the Cufflinks manual for details.  Recommended for experts only; use this at your own discretion.'}], 'default_args': [{'name': 'task_name', 'value': 'Cuffcompare', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.prefix', 'value': '<input.file_basename>', 'label': 'Output.Prefix', 'description': 'A prefix for the module output'}, {'name': 'exclude.transcripts', 'value': 'true', 'label': 'Exclude.Transcripts', 'description': 'Whether to ignore reference transcripts that are not overlapped by any transcript in the input files.  Does not apply if no reference GTF is provided.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Cuffdiff': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Cufflinks 2.0.2 - Finds significant changes in transcript expression, splicing, and promoter use', 'required_args': [{'name': 'aligned.files', 'value': '', 'label': 'Aligned.Files', 'description': 'A set of aligned files grouped by condition'}, {'name': 'GTF.file', 'value': '', 'label': 'GTF.File', 'description': 'A transcript annotation file in GTF/GFF format produced by cufflinks, cuffcompare, cuffmerge, or other source (such as a reference annotation GTF)'}, {'name': 'frag.bias.correct', 'value': '', 'label': 'Frag.Bias.Correct', 'description': 'Reference (FASTA/FA) for bias detection and correction algorithm'}, {'name': 'mask.file', 'value': '', 'label': 'Mask.File', 'description': 'This file tells Cuffdiff to ignore all reads that could have come from transcripts in this GTF/GFF file. It is recommended that annotated rRNA, mitochondrial transcripts, and other abundant transcripts you want to ignore in your analysis be included in this file.'}, {'name': 'additional.cuffdiff.options', 'value': '', 'label': 'Additional.Cuffdiff.Options', 'description': 'Additional options to be passed along to the Cuffdiff program at the command line.  This parameter gives you a means to specify otherwise unavailable Cuffdiff options and switches not supported by the module; check the Cufflinks manual for details.  Recommended for experts only; use this at your own discretion.'}], 'default_args': [{'name': 'task_name', 'value': 'Cuffdiff', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'time.series', 'value': 'false', 'label': 'Time.Series', 'description': 'Analyze the provided samples as a time series, rather than testing for differences between all pairs of samples.'}, {'name': 'normalization.method', 'value': 'geometric', 'label': 'Normalization.Method', 'description': 'The normalization method to be used. Choices are geometric mean (the default), upper quartile, or standard FPKM normalization. See the documentation for a discussion of these methods.'}, {'name': 'FPKM.scaling', 'value': 'compatible-hits', 'label': 'FPKM.Scaling', 'description': 'Controls how Cuffdiff includes mapped fragments toward the number used in the FPKM denominator.  Use compatible-hits (the default) to count only those fragments compatible with some reference transcript.  Use total-hits to count all fragments, even those not compatible with a reference transcript.'}, {'name': 'multi.read.correct', 'value': 'true', 'label': 'Multi.Read.Correct', 'description': \"Instructs Cuffdiff to do use its two-stage read weighting algorithm to more accurately distribute a multi-mapped read's count contribution across the multiple loci the read mapped to.\"}, {'name': 'min.alignment.count', 'value': '10', 'label': 'Min.Alignment.Count', 'description': \"The minimum number of alignments in a locus needed to conduct significance testing on changes in that locus observed between samples.  If no testing is performed, changes in the locus are deemed not significant, and the locus's observed changes do not contribute to correction for multiple testing.\"}, {'name': 'FDR', 'value': '.05', 'label': 'Fdr', 'description': 'The allowed false discovery rate.'}, {'name': 'library.type', 'value': 'inferred', 'label': 'Library.Type', 'description': 'The library type used to generate reads. The default is inferred, meaning that no library type information is passed.'}, {'name': 'skip.diff.exp', 'value': 'false', 'label': 'Skip.Diff.Exp', 'description': 'Tells Cuffdiff to perform quantification and normalization only and to skip its differential expression calculations which are computationally expensive.  The default is to compute differential expression.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Cufflinks': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Cufflinks 2.0.2 - Assembles transcripts, estimates abundances, and tests for differential expression and regulation in RNA-seq samples', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'Input file in SAM or BAM format'}, {'name': 'transfrag.label', 'value': '', 'label': 'Transfrag.Label', 'description': 'A label for the transfrags in the output files'}, {'name': 'GTF', 'value': '', 'label': 'Gtf', 'description': 'Reference annotation (GFF/GTF file) for isoform expression estimates'}, {'name': 'GTF.guide', 'value': '', 'label': 'GTF.Guide', 'description': 'Reference annotation (GFF/GTF file) to guide RABT assembly'}, {'name': 'mask.file', 'value': '', 'label': 'Mask.File', 'description': 'A GTF file specifying transcripts to be ignored'}, {'name': 'frag.bias.correct', 'value': '', 'label': 'Frag.Bias.Correct', 'description': 'Reference (FASTA/FA) for bias detection and correction algorithm'}, {'name': 'additional.cufflinks.options', 'value': '', 'label': 'Additional.Cufflinks.Options', 'description': 'Additional options to be passed along to the Cufflinks program at the command line.  This parameter gives you a means to specify otherwise unavailable Cufflinks options and switches not supported by the module; check the Cufflinks manual for details.  Recommended for experts only; use this at your own discretion.'}], 'default_args': [{'name': 'task_name', 'value': 'Cufflinks', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'multi.read.correct', 'value': 'false', 'label': 'Multi.Read.Correct', 'description': 'Whether to do an initial estimation procedure to more accurately weight reads mapping to multiple locations in the genome'}, {'name': 'library.type', 'value': 'inferred', 'label': 'Library.Type', 'description': 'The library type used to generate reads.'}, {'name': 'min.frags.per.transfrag', 'value': '10', 'label': 'Min.Frags.Per.Transfrag', 'description': 'Assembled transfrags supported by fewer than this many aligned RNA-Seq fragments are not reported.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'CufflinksCuffmergePipeline': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Creates multiple transcript assemblies using Cufflinks and then merges these assemblies into one', 'required_args': [{'name': 'CreateSymlinks1.input.files', 'value': '', 'label': 'CreateSymlinks1.Input.Files', 'description': 'One or more input files'}, {'name': 'CufflinksWrapperPipeline2.Cufflinks1.GTF.guide', 'value': '', 'label': 'CufflinksWrapperPipeline2.Cufflinks1.GTF.Guide', 'description': 'Reference annotation (GFF/GTF file) to guide RABT assembly'}, {'name': 'CufflinksWrapperPipeline2.Cufflinks1.mask.file', 'value': '', 'label': 'CufflinksWrapperPipeline2.Cufflinks1.Mask.File', 'description': 'A GTF file specifying transcripts to be ignored'}, {'name': 'CufflinksWrapperPipeline2.Cufflinks1.additional.cufflinks.options', 'value': '', 'label': 'CufflinksWrapperPipeline2.Cufflinks1.Additional.Cufflinks.Options', 'description': 'Additional options to be passed along to the Cufflinks program at the command line.  This parameter gives you a means to specify otherwise unavailable Cufflinks options and switches not supported by the module; check the Cufflinks manual for details.  Recommended for experts only; use this at your own discretion.'}, {'name': 'Cufflinks.cuffmerge4.reference.GTF', 'value': '', 'label': 'Cufflinks.Cuffmerge4.Reference.GTF', 'description': 'An optional reference annotation GTF. The input assemblies are merged together with the reference GTF and included in the final output. Cuffmerge will use this to attach gene names and other metadata to the merged catalog.'}, {'name': 'Cufflinks.cuffmerge4.genome.file', 'value': '', 'label': 'Cufflinks.Cuffmerge4.Genome.File', 'description': 'A file containing the genomic DNA sequences for the reference.  This should be a multi-FASTA file with all contigs present.'}], 'default_args': [{'name': 'task_name', 'value': 'CufflinksCuffmergePipeline', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'CufflinksWrapperPipeline2.Cufflinks1.library.type', 'value': 'inferred', 'label': 'CufflinksWrapperPipeline2.Cufflinks1.Library.Type', 'description': 'The library type used to generate reads.'}, {'name': 'CufflinksWrapperPipeline2.Cufflinks1.min.frags.per.transfrag', 'value': '10', 'label': 'CufflinksWrapperPipeline2.Cufflinks1.Min.Frags.Per.Transfrag', 'description': 'Assembled transfrags supported by fewer than this many aligned RNA-Seq fragments are not reported.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Cuffmerge': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Cufflinks 2.0.2: Cuffmerge - Merge several Cufflinks assemblies', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'GTF files to be merged'}, {'name': 'reference.GTF', 'value': '', 'label': 'Reference.GTF', 'description': 'An optional reference annotation GTF. The input assemblies are merged together with the reference GTF and included in the final output. Cuffmerge will use this to attach gene names and other metadata to the merged catalog.'}, {'name': 'genome.file', 'value': '', 'label': 'Genome.File', 'description': 'A file containing the genomic DNA sequences for the reference.  This should be a multi-FASTA file with all contigs present.'}], 'default_args': [{'name': 'task_name', 'value': 'Cuffmerge', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'CummeRbund.GeneSetReport': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '[Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] Cuffdiff visualization package providing plots based on a user-provided list of genes.', 'required_args': [{'name': 'cuffdiff.input', 'value': '', 'label': 'Cuffdiff.Input', 'description': 'A Cuffdiff job, a previous CummeRbund job, or a cuffData.db file from a previous CummeRbund job.'}, {'name': 'geneset.file', 'value': '', 'label': 'Geneset.File', 'description': 'A text file containing the list of genes for this report, one per line.  Each of these can be represented by a gene symbol (short name), gene ID, isoform_id, tss_group_id, or cds_id.'}, {'name': 'selected.conditions', 'value': '', 'label': 'Selected.Conditions', 'description': 'Specifies the conditions (samples) to be used in the plots.  This should be a comma-separated list of conditions, using the same names as in the upstream Cuffdiff job; leave this blank to use all conditions.  If blank, no pairwise comparison plots will be generated.'}], 'default_args': [{'name': 'task_name', 'value': 'CummeRbund.GeneSetReport', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.format', 'value': 'pdf', 'label': 'Output.Format', 'description': 'The output file format.'}, {'name': 'feature.level', 'value': 'genes', 'label': 'Feature.Level', 'description': 'Feature level for the report.'}, {'name': 'report.as.aggregate', 'value': 'yes', 'label': 'Report.As.Aggregate', 'description': 'Controls whether reporting should be done for individual replicates or aggregate condition/sample values.  The default is to use aggregate sample values.  Note that the Volcano plot always shows aggregate samples.'}, {'name': 'log.transform', 'value': 'yes', 'label': 'Log.Transform', 'description': 'Whether or not to log transform the FPKM values.  Note that the FPKM values are always log transformed for the Volcano plots and that the Scatter plot will use a log2 rather than log10 transformation.  For the Expression Plot and Expression Barplot, this directs that the y-axis will be drawn on a log10 scale.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'CummeRbund.QcReport': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '[Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] Cuffdiff visualization package providing plots and tables related to QC and Global Stats.', 'required_args': [{'name': 'cuffdiff.input', 'value': '', 'label': 'Cuffdiff.Input', 'description': 'A Cuffdiff job, a previous CummeRbund job, or a cuffData.db file from a previous CummeRbund job.'}], 'default_args': [{'name': 'task_name', 'value': 'CummeRbund.QcReport', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.format', 'value': 'pdf', 'label': 'Output.Format', 'description': 'The output file format.'}, {'name': 'feature.level', 'value': 'genes', 'label': 'Feature.Level', 'description': 'Feature level for the report'}, {'name': 'report.as.aggregate', 'value': 'yes', 'label': 'Report.As.Aggregate', 'description': 'Controls whether reporting should be done for individual replicates or aggregate condition/sample values.  The default is to use aggregate sample values.  Controls whether reporting should be done for individual replicates or aggregate condition/sample values.  The default is to use aggregate sample values.  Note that the Dispersion, FPKM.SCV, and MA plots always show aggregate samples.'}, {'name': 'log.transform', 'value': 'yes', 'label': 'Log.Transform', 'description': 'Whether or not to log transform FPKM values.  Note that the FPKM values are always log transformed for the FPKM.SCV plot.  Log transformation does not apply to the Dispersion, MDS, and PCA plots.'}, {'name': 'pca.x', 'value': 'PC2', 'label': 'Pca.X', 'description': 'Indicates which principal component is to be presented on the x-axis of the PCA plot.  Must differ from pca.y.'}, {'name': 'pca.y', 'value': 'PC3', 'label': 'Pca.Y', 'description': 'Indicates which principal component is to be presented on the y-axis of the PCA plot.  Must differ from pca.x.'}, {'name': 'attempt.to.merge.names', 'value': 'no', 'label': 'Attempt.To.Merge.Names', 'description': 'Should the module attempt to merge gene names into the text reports?  Depending on the GTF file used in the Cuffdiff run, the text reports may instead contain tracking ID references, particularly if a Cuffmerge merged.gtf file was used.  Note that this setting does not affect the plots.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'CummeRbund.SelectedConditionsReport': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '[**Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] Cuffdiff visualization package providing plots based on a user-provided list of conditions.', 'required_args': [{'name': 'cuffdiff.input', 'value': '', 'label': 'Cuffdiff.Input', 'description': 'A Cuffdiff job, a previous CummeRbund job, or a cuffData.db file from a previous CummeRbund job.'}, {'name': 'selected.conditions', 'value': '', 'label': 'Selected.Conditions', 'description': 'Specifies the conditions (samples) to be used in the plots.  This should be a comma-separated list of conditions, using the same names as in the upstream Cuffdiff job; leave this blank to use all conditions.  If blank, no pairwise comparison plots will be generated.'}], 'default_args': [{'name': 'task_name', 'value': 'CummeRbund.SelectedConditionsReport', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.format', 'value': 'pdf', 'label': 'Output.Format', 'description': 'The output file format.'}, {'name': 'feature.level', 'value': 'genes', 'label': 'Feature.Level', 'description': 'Feature level for the report.'}, {'name': 'report.as.aggregate', 'value': 'yes', 'label': 'Report.As.Aggregate', 'description': 'Controls whether reporting should be done for individual replicates or aggregate condition/sample values.  The default is to use aggregate sample values.  Note that the Volcano plot always shows aggregate samples.'}, {'name': 'log.transform', 'value': 'yes', 'label': 'Log.Transform', 'description': 'Whether or not to log transform the FPKM values.  Note that the FPKM values are always log transformed for the Volcano plots and that the Scatter plot will use a log2 rather than log10 transformation.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'CummeRbund.SelectedGeneReport': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '[**Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] Cuffdiff visualization package providing plots based on a single user-specified gene.', 'required_args': [{'name': 'cuffdiff.input', 'value': '', 'label': 'Cuffdiff.Input', 'description': 'A Cuffdiff job, a previous CummeRbund job, or a cuffData.db file from a previous CummeRbund job.'}, {'name': 'feature.id', 'value': '', 'label': 'Feature.Id', 'description': 'The gene or feature of interest.  This can be a gene symbol (short name), gene ID, isoform_id, tss_group_id, or cds_id.'}, {'name': 'selected.conditions', 'value': '', 'label': 'Selected.Conditions', 'description': 'Specifies the conditions (samples) to be used in the plots.  This should be a comma-separated list of conditions, using the same names as in the upstream Cuffdiff job; leave this blank to use all conditions.  The Similarity plots will always operate across all conditions.'}], 'default_args': [{'name': 'task_name', 'value': 'CummeRbund.SelectedGeneReport', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'find.similar', 'value': '10', 'label': 'Find.Similar', 'description': 'Optionally, find and plot the top genes (up to this count) with an expression profile most similar to the given gene of interest.  If blank, this will be skipped.    '}, {'name': 'output.format', 'value': 'pdf', 'label': 'Output.Format', 'description': 'The output file format.'}, {'name': 'feature.level', 'value': 'genes', 'label': 'Feature.Level', 'description': 'Feature level for the report.  Note that the Similarity plots will always be generated at the \"genes\" level.'}, {'name': 'report.as.aggregate', 'value': 'yes', 'label': 'Report.As.Aggregate', 'description': 'Controls whether reporting should be done for individual replicates or aggregate condition/sample values.  The default is to use aggregate sample values.  Note that the Similarity plots always show aggregate samples.'}, {'name': 'log.transform', 'value': 'yes', 'label': 'Log.Transform', 'description': 'Whether or not to log transform the FPKM values.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'CytoscapeViewer': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Visualizes a gene network', 'required_args': [{'name': 'dataset.file', 'value': '', 'label': 'Dataset.File', 'description': 'Dataset file - .res, .gct'}, {'name': 'adjacency.matrix.file', 'value': '', 'label': 'Adjacency.Matrix.File', 'description': 'Adjacency matrix - .adj'}], 'default_args': [{'name': 'task_name', 'value': 'CytoscapeViewer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'DAPPLE': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'DAPPLE', 'required_args': [{'name': 'Input.File', 'value': '', 'label': 'Input.File', 'description': 'File containing a list of SNPs, genes or regions. Each input should be on a new line.'}, {'name': 'Keyword', 'value': '', 'label': 'Keyword', 'description': ''}], 'default_args': [{'name': 'task_name', 'value': 'DAPPLE', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Permute.Length', 'value': '0', 'label': 'Permute.Length', 'description': 'To increase speed, this is adaptive permutation (see README)'}, {'name': 'Plot', 'value': 'true', 'label': 'Plot', 'description': ''}, {'name': 'nearest.gene', 'value': 'true', 'label': 'Nearest.Gene', 'description': 'Do you want to use the gene nearest to the SNP (rather than grabbing all genes in the LD wingspan)? Only relevant for SNP inputs.'}, {'name': 'genome.build', 'value': '18', 'label': 'Genome.Build', 'description': 'Which set of SNPs should we use? hg18, hg19, 1KG?'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'DeIdentifyFCS': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'DeIdentifyFCS an FCS data file; remove keywords from a list or matching a regular expression; useful to anonymize FCS data files and/or to remove specific (e.g., clinical) information ', 'required_args': [{'name': 'Input.FCS.data.file', 'value': '', 'label': 'Input.FCS.Data.File', 'description': 'An FCS data file to be de-identified.'}, {'name': 'Remove.keywords', 'value': '', 'label': 'Remove.Keywords', 'description': \"Comma-separated list of keyword names (escape comma by '\\\\' if part of a keyword name) or regular expression to match keywords to be removed.\"}], 'default_args': [{'name': 'task_name', 'value': 'DeIdentifyFCS', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Is.regular.expression.used', 'value': 'No', 'label': 'Is.Regular.Expression.Used', 'description': \"Is a regular expression used in the list of keywords to remove? If not (default) a simple comma-separated list of keyword names is expected. In this list, comma may be escaped by '\\\\' if part of a keyword name, e.g., this \\\\, is in the keyword.\"}, {'name': 'Output.FCS.file.name', 'value': '<Input.FCS.data.file_basename>.deidentified.<Input.FCS.data.file_extension>', 'label': 'Output.FCS.File.Name', 'description': 'Output FCS file name.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'DependencyBiomarkerDotplot': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Creates a dotplot of Achilles gene dependency scores. The dots are colored based on the biomarker status of the cell line (i.e. mutation in gene or no mutation in gene). This module was created for use in the Education Outreach high school computational curriculum. ', 'required_args': [{'name': 'Achilles.Gene.Dependency.Data.File', 'value': '', 'label': 'Achilles.Gene.Dependency.Data.File', 'description': 'Format is a csv file, with cell lines in rows and genes in columns. '}, {'name': 'Biomarker.Data.File', 'value': '', 'label': 'Biomarker.Data.File', 'description': 'Format is a csv file, with cell lines in rows and genes in columns. '}, {'name': 'Gene.1', 'value': '', 'label': 'Gene.1', 'description': 'The gene name should match exactly to the column names in the data files and are case-sensitive. \\n'}, {'name': 'Gene.#2', 'value': '', 'label': 'Gene.#2', 'description': 'The gene name should match exactly to the column names in the data files and are case-sensitive.'}, {'name': 'Gene.#3', 'value': '', 'label': 'Gene.#3', 'description': 'The gene name should match exactly to the column names in the data files and are case-sensitive.'}, {'name': 'Gene.#4', 'value': '', 'label': 'Gene.#4', 'description': 'The gene name should match exactly to the column names in the data files and are case-sensitive.'}, {'name': 'Gene.#5', 'value': '', 'label': 'Gene.#5', 'description': 'The gene name should match exactly to the column names in the data files and are case-sensitive.'}], 'default_args': [{'name': 'task_name', 'value': 'DependencyBiomarkerDotplot', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'DESeq2': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'A Differential Expression workflow for RNA-Seq data based on the DESeq2 package from Bioconductor ', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'A GCT file containing raw RNA-Seq counts, such as is produced by MergeHTSeqCounts'}, {'name': 'cls.file', 'value': '', 'label': 'Cls.File', 'description': 'A categorical CLS file specifying the phenotype classes for the samples in the GCT file.  This should contain exactly two classes with the control specified first.'}, {'name': 'confounding.variable.cls.file', 'value': '', 'label': 'Confounding.Variable.Cls.File', 'description': 'A categorical CLS file specifying an additional confounding variable, mapped to the input file samples.  Use this for a two-factor comparison.'}], 'default_args': [{'name': 'task_name', 'value': 'DESeq2', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file.base', 'value': '<input.file_basename>', 'label': 'Output.File.Base', 'description': 'The base name of the output file(s). File extensions will be added automatically.'}, {'name': 'qc.plot.format', 'value': 'skip', 'label': 'Qc.Plot.Format', 'description': 'Choose a file format for the QC plots (or skip them completely)'}, {'name': 'fdr.threshold', 'value': '0.1', 'label': 'Fdr.Threshold', 'description': 'FDR threshold to use in filtering before creating the reports of top up-regulated and down-regulated genes '}, {'name': 'top.N.count', 'value': '20', 'label': 'Top.N.Count', 'description': 'Max gene count to use when creating the reports of top up-regulated and down-regulated genes '}, {'name': 'random.seed', 'value': '779948241', 'label': 'Random.Seed', 'description': 'Seed to use for randomization'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'DownloadURL': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Downloads a file from a URL', 'required_args': [{'name': 'URL', 'value': '', 'label': 'Url', 'description': 'The URL to retrieve the file from'}, {'name': 'output.file', 'value': '', 'label': 'Output.File', 'description': 'The name of the output file'}], 'default_args': [{'name': 'task_name', 'value': 'DownloadURL', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ESPPredictor': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Determine which tryptic peptides will generate the highest signal in ESI-MS', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'List of peptide sequences (one per line, no column header)'}], 'default_args': [{'name': 'task_name', 'value': 'ESPPredictor', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ExpressionFileCreator': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Creates a RES or GCT file from a set of Affymetrix CEL files.  For IVT arrays only; use AffySTExpressionFileCreator for ST arrays.', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'A zip file of CEL files'}, {'name': 'quantile.normalization', 'value': '', 'label': 'Quantile.Normalization', 'description': '(GCRMA and RMA only) Whether to normalize data using quantile normalization'}, {'name': 'background.correct', 'value': '', 'label': 'Background.Correct', 'description': '(RMA only) Whether to background correct using RMA background correction'}, {'name': 'value.to.scale.to', 'value': '', 'label': 'Value.To.Scale.To', 'description': '(median/mean scaling only) The value to scale to.'}, {'name': 'clm.file', 'value': '', 'label': 'Clm.File', 'description': 'A tab-delimited text file containing one scan, sample, and class per line'}, {'name': 'cdf.file', 'value': '', 'label': 'Cdf.File', 'description': 'Custom CDF file. Leave blank to use default internally provided CDF file (custom cdf file is not implemented for GCRMA).'}], 'default_args': [{'name': 'task_name', 'value': 'ExpressionFileCreator', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'method', 'value': 'RMA', 'label': 'Method', 'description': 'The method to use. Note that dchip and MAS5 will not work with ST arrays.'}, {'name': 'compute.present.absent.calls', 'value': 'no', 'label': 'Compute.Present.Absent.Calls', 'description': 'Whether to compute Present/Absent calls'}, {'name': 'normalization.method', 'value': 'median scaling', 'label': 'Normalization.Method', 'description': '(MAS5 only) The normalization method to apply after expression values are computed. The column having the median of the means is used as the reference unless the parameter value to scale to is given.'}, {'name': 'annotate.probes', 'value': 'yes', 'label': 'Annotate.Probes', 'description': 'Whether to annotate probes with the gene symbol and description'}, {'name': 'output.file', 'value': '<input.file_basename>', 'label': 'Output.File', 'description': 'The base name of the output file(s)'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ExprToGct': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Converts an EXPR file to GCT format', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'Input file in EXPR format'}], 'default_args': [{'name': 'task_name', 'value': 'ExprToGct', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.prefix', 'value': '<input.file_basename>', 'label': 'Output.Prefix', 'description': 'A label that will be used to name the GCT output file.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ExtractColumnNames': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Lists the sample descriptors from a .res file.', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'Input filename - .res'}], 'default_args': [{'name': 'task_name', 'value': 'ExtractColumnNames', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output', 'value': '<input.filename_basename>', 'label': 'Output', 'description': 'Name of the output file (a .txt extension will be appended)'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ExtractComparativeMarkerResults': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Creates a derived dataset and feature list file from the results of ComparativeMarkerSelection', 'required_args': [{'name': 'comparative.marker.selection.filename', 'value': '', 'label': 'Comparative.Marker.Selection.Filename', 'description': 'The results from ComparativeMarkerSelection - .odf'}, {'name': 'dataset.filename', 'value': '', 'label': 'Dataset.Filename', 'description': 'The dataset file used to select markers - .gct, .res, Dataset'}, {'name': 'statistic', 'value': '', 'label': 'Statistic', 'description': 'The statistic to filter features on'}, {'name': 'min', 'value': '', 'label': 'Min', 'description': 'Select features with statistic >= min'}, {'name': 'max', 'value': '', 'label': 'Max', 'description': 'Select features with statistic <= max'}, {'name': 'number.of.neighbors', 'value': '', 'label': 'Number.Of.Neighbors', 'description': 'Number of neighbors to select by score in each direction'}], 'default_args': [{'name': 'task_name', 'value': 'ExtractComparativeMarkerResults', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'base.output.name', 'value': '<comparative.marker.selection.filename_basename>.filt', 'label': 'Base.Output.Name', 'description': 'The base name for the output files'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ExtractFCSDataset': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Extract one or more FCS datasets from an FCS data file', 'required_args': [{'name': 'Input.data.file', 'value': '', 'label': 'Input.Data.File', 'description': 'The input FCS data file; one or more data sets will be extracted from this file.'}, {'name': 'Output.file.name', 'value': '', 'label': 'Output.File.Name', 'description': 'The name for the output file. Uses the input data file name (for example, name.fcs) as the default, and adds _datasetIndex to the name (e.g., name_datasetIndex.fcs).'}, {'name': 'Dataset.indexes', 'value': '', 'label': 'Dataset.Indexes', 'description': 'A list of data set indexes to be extracted from the data file (e.g., 1-3, 5, 8). The first data set has the index 1, the second the index 2, etc. All data sets will be extracted if no indexes are provided.'}], 'default_args': [{'name': 'task_name', 'value': 'ExtractFCSDataset', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ExtractFCSKeywords': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Extracts keyword(s) value(s) from a Flow Cytometry Standard (FCS) file.', 'required_args': [{'name': 'Input.FCS.data.file', 'value': '', 'label': 'Input.FCS.Data.File', 'description': 'The input FCS data file from which the module will extract keyword/value pairs.'}], 'default_args': [{'name': 'task_name', 'value': 'ExtractFCSKeywords', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Keywords.output.mode', 'value': 'CSV', 'label': 'Keywords.Output.Mode', 'description': 'The file format to be used to store the extracted keywords/value pairs.'}, {'name': 'Keywords.output.file.name', 'value': '<Input.FCS.data.file_basename>', 'label': 'Keywords.Output.File.Name', 'description': 'The name for the file in which the keyword/value pairs are stored.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ExtractFCSParameters': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Extract specified parameters (dimensions) from an FCS data file.', 'required_args': [{'name': 'Input.FCS.data.file', 'value': '', 'label': 'Input.FCS.Data.File', 'description': 'The input FCS data file from which to extract parameters.'}, {'name': 'Parameters.to.extract', 'value': '', 'label': 'Parameters.To.Extract', 'description': 'A list of parameters to extract; parameters may be specified by their indexes (e.g., 1-3, 5, 8) or by their short names (e.g., FSC-H, SSC-H, FL1-A, FL3-H).'}, {'name': 'Output.file.name', 'value': '', 'label': 'Output.File.Name', 'description': 'The output file name. The default uses the input file name as the base.'}], 'default_args': [{'name': 'task_name', 'value': 'ExtractFCSParameters', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ExtractRowNames': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Extracts the row names from a .res, .gct, or .odf file.', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'input file - .res, .gct, .odf, file from which the row names are extracted'}], 'default_args': [{'name': 'task_name', 'value': 'ExtractRowNames', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output', 'value': '<input.filename_basename>.row.names.txt', 'label': 'Output', 'description': 'output filename containing the row names - .txt'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'FastQC': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '[**Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] Provides quality control metrics on raw sequence data', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'RNA-seq reads file in FASTQ (bz2 and gz compressed files are supported), SAM, or BAM format.'}, {'name': 'input.format', 'value': '', 'label': 'Input.Format', 'description': 'Bypasses the normal sequence file format detection and forces the program to use the specified format.'}, {'name': 'contaminant.file', 'value': '', 'label': 'Contaminant.File', 'description': 'FastQC screens overrepresented sequences (> 0.1% of total) against a set of possible contaminants consisting of all the primers and adapters used by the various sequencing platforms.  The \"Overrepresented sequences\" report generated by FastQC will indicate when an overrepresented sequence corresponds to a known contaminant.  This optional contaminant file provides a means of adding more sequences to FastQC\\'s built-in list of known contaminants. See module documentation for a description and example of this file\\'s format.'}], 'default_args': [{'name': 'task_name', 'value': 'FastQC', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'kmer.size', 'value': '5', 'label': 'Kmer.Size', 'description': 'The length of Kmer to look for in the Kmer content module. The specified Kmer length must be between 2 and 10. Default length is 5.'}, {'name': 'extract.output', 'value': 'true', 'label': 'Extract.Output', 'description': 'Whether to output an uncompressed version of the report. Set this to yes to view the report directly from within GenePattern.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'FCMFeatureExtraction': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Calculation of features from clustered flow cytometry data.', 'required_args': [{'name': 'Input.CSV.files', 'value': '', 'label': 'Input.CSV.Files', 'description': 'The set of CSV files with clustering results (of a single data file). May be a .zip file, a directory, or a text file listing all input file paths (one file per line).'}], 'default_args': [{'name': 'task_name', 'value': 'FCMFeatureExtraction', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Calculate.number.of.cells', 'value': 'TRUE', 'label': 'Calculate.Number.Of.Cells', 'description': 'Calculate and output the number of cells in each sub-population?'}, {'name': 'Calculate.percentage.of.cells', 'value': 'TRUE', 'label': 'Calculate.Percentage.Of.Cells', 'description': 'Calculate and output the percentage (proportion) of cells in each sub-population?'}, {'name': 'Calculate.means', 'value': 'TRUE', 'label': 'Calculate.Means', 'description': 'Calculate mean parameter values (e.g. Mean Fluorescence Intensities) for all parameters of each sub-population?'}, {'name': 'Output.CSV.file.name', 'value': 'features.csv', 'label': 'Output.CSV.File.Name', 'description': 'The name of the output CSV file.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'FCMSinglePanelQC': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Single panel flow cytometry data quality control', 'required_args': [{'name': 'Input.FCS.data.files', 'value': '', 'label': 'Input.FCS.Data.Files', 'description': 'The set of input FCS files to be tested.'}], 'default_args': [{'name': 'task_name', 'value': 'FCMSinglePanelQC', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Perform.cell.number.test', 'value': 'TRUE', 'label': 'Perform.Cell.Number.Test', 'description': 'Run the cell number control test.'}, {'name': 'Perform.boundary.events.test', 'value': 'TRUE', 'label': 'Perform.Boundary.Events.Test', 'description': 'Run the boundary events test.'}, {'name': 'Perform.timeflow.test', 'value': 'TRUE', 'label': 'Perform.Timeflow.Test', 'description': 'Run the timeflow test, i.e., check whether the flow rate was steady and uninterrupted.'}, {'name': 'Perform.timeline.test', 'value': 'TRUE', 'label': 'Perform.Timeline.Test', 'description': 'Run the timeline test, i.e., detect unexpected non-random acquisition of fluorescence intensities.'}, {'name': 'Zip.output', 'value': 'TRUE', 'label': 'Zip.Output', 'description': 'Zip the resulting report together?'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'FCSClean': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Clean up an FCS file by marking events affected by problems during data acquisition', 'required_args': [{'name': 'Input.FCS', 'value': '', 'label': 'Input.FCS', 'description': 'The FCS file to be cleaned up, i.e., bad events/cells will be marked by a high value of the added Good_vs_Bad channel.'}, {'name': 'Markers', 'value': '', 'label': 'Markers', 'description': 'A comma-separated list of dimensions (flow cytometry parameters/channels) to be checked. The module accepts both, a list of marker names as well as a list of parameter indexes. All dimensions but scatters and time will be used if the Markers parameter is not provided.'}], 'default_args': [{'name': 'task_name', 'value': 'FCSClean', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Output.FCS', 'value': '<Input.FCS_basename>.cleaned.fcs', 'label': 'Output.FCS', 'description': 'The name of the output FCS file.'}, {'name': 'Bin.Size', 'value': '0.01', 'label': 'Bin.Size', 'description': 'The size of a bin as proportion of cells in the FCS file, typically between 0.01 and 0.2.'}, {'name': 'Cell.Cutoff', 'value': '500', 'label': 'Cell.Cutoff', 'description': 'The number of cells to be used for a cut off, typically between 300 and 800, see documentation for more details.'}, {'name': 'Cutoff', 'value': 'median', 'label': 'Cutoff', 'description': 'The method of how cutoff is performed by the algorithm. Typically, a median works well. Alternatively, you may specify a value from the [0,1] interval as a cutoff percentile, or a value greater than 1, which will be assumed as a direct specification of the cutoff threshold.'}, {'name': 'Max.Fc', 'value': '1.3', 'label': 'Max.Fc', 'description': 'Relative increase from mean norm of presumed good data below which data is still presumed to be good.'}, {'name': 'Announce', 'value': 'TRUE', 'label': 'Announce', 'description': 'Shall we print information about identifying vs. not identifying any bad events to the standard output?'}, {'name': 'Diagnostic', 'value': 'FALSE', 'label': 'Diagnostic', 'description': 'Shall we include a diagnostics plot?'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'FCSNormalization': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Per-channel basis normalization of flow cytometry data.', 'required_args': [{'name': 'Input.FCS.files', 'value': '', 'label': 'Input.FCS.Files', 'description': 'The set of input FCS files to be normalized.'}, {'name': 'Channels.to.normalize', 'value': '', 'label': 'Channels.To.Normalize', 'description': 'Comma-separated list of parameter short names to normalize.'}], 'default_args': [{'name': 'task_name', 'value': 'FCSNormalization', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Apply.ASinH', 'value': 'TRUE', 'label': 'Apply.ASinH', 'description': 'Apply the asinh transformation prior normalization?'}, {'name': 'Create.report', 'value': 'TRUE', 'label': 'Create.Report', 'description': 'Create additional histogram channel report?'}, {'name': 'Landmarks.number', 'value': '1', 'label': 'Landmarks.Number', 'description': 'Number of landmarks to be used for normalization.'}, {'name': 'Zip.output', 'value': 'TRUE', 'label': 'Zip.Output', 'description': 'Zip the resulting normalized FCS data files together?'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'FcsToCsv': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Converts a Flow Cytometry Standard (FCS) file to a Comma Separated Values (CSV) file.', 'required_args': [{'name': 'Input.FCS.data.file', 'value': '', 'label': 'Input.FCS.Data.File', 'description': 'An FCS data file to be converted to CSV.'}], 'default_args': [{'name': 'task_name', 'value': 'FcsToCsv', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Output.CSV.file.name', 'value': '<Input.FCS.data.file_basename>.csv', 'label': 'Output.CSV.File.Name', 'description': 'The name of the output CSV file.'}, {'name': 'Use.full.names', 'value': 'TRUE', 'label': 'Use.Full.Names', 'description': 'Whether to try to extract full parameter names from the FCS file.'}, {'name': 'Output.keywords.file.name', 'value': '<Input.FCS.data.file_basename>', 'label': 'Output.Keywords.File.Name', 'description': 'The file name to store keyword value pairs extracted from the FCS file.'}, {'name': 'Output.keywords.mode', 'value': 'CSV', 'label': 'Output.Keywords.Mode', 'description': 'How to store keywords.'}, {'name': 'Channel.to.scale.conversion', 'value': 'TRUE', 'label': 'Channel.To.Scale.Conversion', 'description': 'Whether to convert log channel values to linear scale.'}, {'name': 'Precision', 'value': 'FALSE', 'label': 'Precision', 'description': 'Whether to use regular (32bit, recommended) or double precision.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'FeatureSummaryViewer': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Feature Summary Viewer', 'required_args': [{'name': 'feature.filename', 'value': '', 'label': 'Feature.Filename', 'description': 'odf file with or without features to summarize'}], 'default_args': [{'name': 'task_name', 'value': 'FeatureSummaryViewer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'FileSplitter': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Splits a file into equal-sized chunks, given the number of lines per output file', 'required_args': [{'name': 'inputFile', 'value': '', 'label': 'InputFile', 'description': 'The input file, to be split into chunks'}, {'name': 'numFiles', 'value': '', 'label': 'NumFiles', 'description': \"[Integer, >=1], The maximum number of output files to split the file into. When both 'numFiles' and 'numLines' are set, treat this as the maximum number of output files to create, even if the files have more than the requested number of lines\"}, {'name': 'numLines', 'value': '', 'label': 'NumLines', 'description': '[Integer, >=1], The number of lines in each output file.'}], 'default_args': [{'name': 'task_name', 'value': 'FileSplitter', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'FilterFilelist': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'a helper module which generates a filtered filelist from the input filelist and some filter parameters.', 'required_args': [{'name': 'input.filelist', 'value': '', 'label': 'Input.Filelist', 'description': 'the initial input filelist, a file containing a list of filepaths'}, {'name': 'includePattern', 'value': '', 'label': 'IncludePattern', 'description': 'an optional list of filename patterns to match, e.g. *.cls,*.gct'}, {'name': 'errorOnEmptySet', 'value': '', 'label': 'ErrorOnEmptySet', 'description': 'Whether the module should output stderr if the output result is empty'}], 'default_args': [{'name': 'task_name', 'value': 'FilterFilelist', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'type', 'value': 'f', 'label': 'Type', 'description': 'Include files, directories, or both files and directories in the output filelist'}, {'name': 'output.filename', 'value': '<input.filelist_basename>.filtered.<input.filelist_extension>', 'label': 'Output.Filename', 'description': 'the name of the output file, if none specified, write to stdout'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'FilterLowshRNAs': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Removes shRNAs with low read counts in original DNA plasmid pool. Outputs a GCT file containing filtered shRNA data and a GCT file containing the mean read count for each shRNA in the original DNA plasmid pool. ', 'required_args': [{'name': 'GCT.Files', 'value': '', 'label': 'GCT.Files', 'description': 'Please enter one or more GCT files.'}, {'name': 'File.Prefix', 'value': '', 'label': 'File.Prefix', 'description': ''}], 'default_args': [{'name': 'task_name', 'value': 'FilterLowshRNAs', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Find.differentially.expressed.genes.in.RNASeq.data': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'This pipeline is used in the GenomeSpace recipe, \"Find differentially expressed genes in RNA-Seq data\". The recipe can be found at: http://www.genomespace.org/support/guides/recipes/sections/find-differentially-expressed-genes-in-rna-seq-data.', 'required_args': [{'name': 'ConvertLineEndings1.input.filename', 'value': '', 'label': 'ConvertLineEndings1.Input.Filename', 'description': 'The input file (any non-binary file format)'}, {'name': 'TopHat2.reads.pair.1', 'value': '', 'label': 'TopHat2.Reads.Pair.1', 'description': 'Unpaired reads file or first mate for paired reads. One or more files containing reads in FASTA or FASTQ format (bz2 and gz compressed files are supported).'}, {'name': 'TopHat4.reads.pair.1', 'value': '', 'label': 'TopHat4.Reads.Pair.1', 'description': 'Unpaired reads file or first mate for paired reads. One or more files containing reads in FASTA or FASTQ format (bz2 and gz compressed files are supported).'}], 'default_args': [{'name': 'task_name', 'value': 'Find.differentially.expressed.genes.in.RNASeq.data', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Find.subnetworks.of.differentially.expressed.genes.and.identify.associated.biological.functions': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'This pipeline is used in the GenomeSpace recipe, \"Find subnetworks of differentially expressed genes and identify associated biological functions\". The recipe can be found at: http://www.genomespace.org/support/guides/recipes/sections/find-subnetworks-of-differentially-expressed-genes-and-identify-associated-biological-functions.', 'required_args': [{'name': 'PreprocessDataset1.input.filename', 'value': '', 'label': 'PreprocessDataset1.Input.Filename', 'description': 'Input filename - .res, .gct, .odf'}, {'name': 'ComparativeMarkerSelection2.cls.file', 'value': '', 'label': 'ComparativeMarkerSelection2.Cls.File', 'description': 'The class file - .cls'}], 'default_args': [{'name': 'task_name', 'value': 'Find.subnetworks.of.differentially.expressed.genes.and.identify.associated.biological.functions', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'FLAMEChooseOptimalClusterNumber': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Determines the optimal number of clusters for each sample, using user-specified criterion', 'required_args': [{'name': 'mixture.model', 'value': '', 'label': 'Mixture.Model', 'description': 'A .zip file containing mixture modeling results from the FLAMEMixtureModel module'}], 'default_args': [{'name': 'task_name', 'value': 'FLAMEChooseOptimalClusterNumber', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'method', 'value': 'SWR', 'label': 'Method', 'description': 'Criterion used for determining optimal number of clusters'}, {'name': 'seed', 'value': '123456', 'label': 'Seed', 'description': 'An integer; the seed fixes random computational processes during computation. This does not need to be changed under most circumstances.'}, {'name': 'output.prefix', 'value': '<mixture.model_basename>', 'label': 'Output.Prefix', 'description': 'A prefix for output files'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'FLAMEContourDataGenerator': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Generates data for drawing FLAME contour plots', 'required_args': [{'name': 'membership.file', 'value': '', 'label': 'Membership.File', 'description': 'A membership.txt or member.txt file output from one of the FLAME modules. '}, {'name': 'parameters.file', 'value': '', 'label': 'Parameters.File', 'description': 'A parameters.txt file output from one of the FLAME modules. '}, {'name': 'channels.to.plot', 'value': '', 'label': 'Channels.To.Plot', 'description': 'A comma separated list of the index or names of three channels/antibodies to plot'}], 'default_args': [{'name': 'task_name', 'value': 'FLAMEContourDataGenerator', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.prefix', 'value': '<parameters.file_basename>', 'label': 'Output.Prefix', 'description': 'A prefix for output files'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'FLAMEContourViewer.Pipeline': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Pipeline which runs the FLAMECountorDataGenerator and the FLAMEViewer', 'required_args': [{'name': 'FLAMEContourDataGenerator1.membership.file', 'value': '', 'label': 'FLAMEContourDataGenerator1.Membership.File', 'description': 'A membership.txt or member.txt file output from one of the FLAME modules.'}, {'name': 'FLAMEContourDataGenerator1.parameters.file', 'value': '', 'label': 'FLAMEContourDataGenerator1.Parameters.File', 'description': 'A parameters.txt file output from one of the FLAME modules.'}, {'name': 'FLAMEContourDataGenerator1.channels.to.plot', 'value': '', 'label': 'FLAMEContourDataGenerator1.Channels.To.Plot', 'description': 'A comma separated list of the index or names of three channels/antibodies to plot'}], 'default_args': [{'name': 'task_name', 'value': 'FLAMEContourViewer.Pipeline', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'FLAMEContourDataGenerator1.output.prefix', 'value': 'flame_contour', 'label': 'FLAMEContourDataGenerator1.Output.Prefix', 'description': 'A prefix for output files'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'FLAMEMetacluster': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Matches populations/clusters across all samples in the dataset, assembles a table of all parameters in all samples', 'required_args': [{'name': 'optimal.g.mixture.model', 'value': '', 'label': 'Optimal.G.Mixture.Model', 'description': 'A .zip file containing the optimal mixture modeling result of each sample from the output of the FLAMEChooseOptimalClusterNumber module.'}, {'name': 'sample.class', 'value': '', 'label': 'Sample.Class', 'description': ''}], 'default_args': [{'name': 'task_name', 'value': 'FLAMEMetacluster', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'estimate.mode', 'value': 'F', 'label': 'Estimate.Mode', 'description': 'Used only for skew distributions. Whether to estimate the mode for each cluster. Must be the same value used in FLAMEMixtureModel.'}, {'name': 'estimation.increment', 'value': '1', 'label': 'Estimation.Increment', 'description': 'Used only for skew distributions and when estimate mode is set to yes. The smaller the increment, the more accurate the estimation, but the slower the estimation step. Must be the same value used in FLAMEMixtureModel.'}, {'name': 'output.intermediate.results', 'value': 'F', 'label': 'Output.Intermediate.Results', 'description': 'Choose whether to output the intermediate metaclustering results, such as within-class matching results.'}, {'name': 'output.prefix', 'value': '<optimal.g.mixture.model_basename>', 'label': 'Output.Prefix', 'description': 'A prefix for output files.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'FLAMEMixtureModel': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Clusters and estimates population parameters, using preprocessed flow cytometric data', 'required_args': [{'name': 'preprocessed.data', 'value': '', 'label': 'Preprocessed.Data', 'description': 'A .zip file containing preprocessed flow sample files in tab-delimited .txt format.'}, {'name': 'channels.to.cluster', 'value': '', 'label': 'Channels.To.Cluster', 'description': 'A comma-separated list of either channel numbers (e.g., 1, 2, 3, 7) or channel names (e.g., CD4, CD45RA, ZAP70); these denote which antibodies in the (preprocessed) data sample to be used for clustering.'}], 'default_args': [{'name': 'task_name', 'value': 'FLAMEMixtureModel', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'g.min', 'value': '3', 'label': 'G.Min', 'description': 'An integer; minimal number of clusters to be fit for each sample.'}, {'name': 'g.max', 'value': '10', 'label': 'G.Max', 'description': 'An integer; maximal number of clusters to be fit for each sample. g max should be greater or equal to g min.'}, {'name': 'density', 'value': 'skewt', 'label': 'Density', 'description': 'Density distribution to be used for clustering'}, {'name': 'estimate.mode', 'value': 'F', 'label': 'Estimate.Mode', 'description': 'Used only for skew distributions. Whether to estimate the mode for each cluster.  Must be the same value used in FLAMEMixtureModel.'}, {'name': 'estimation.increment', 'value': '1', 'label': 'Estimation.Increment', 'description': 'Used only for skew distributions and when estimate mode is set to yes. The smaller the increment, the more accurate the estimation, but the slower the estimation step. Must be the same value used in FLAMEMixtureModel.'}, {'name': 'seed', 'value': '123456', 'label': 'Seed', 'description': 'An integer; the seed fixes random computational processes during computation. This does not need to be changed under most circumstances.'}, {'name': 'output.prefix', 'value': '<preprocessed.data_basename>', 'label': 'Output.Prefix', 'description': 'A prefix for output files.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'FLAMEPreprocess': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Performs preprocessing of .fcs or .txt flow cytometric data files', 'required_args': [{'name': 'dataset', 'value': '', 'label': 'Dataset', 'description': 'A .zip file containing flow files in .txt or .fcs format. (The files should be all .txt or all .fcs, but not both.)'}, {'name': 'channels', 'value': '', 'label': 'Channels', 'description': 'A comma-separated list of channel numbers indicating which columns (antibodies) to keep (e.g., 1, 2, 3, 7).'}, {'name': 'channel.names', 'value': '', 'label': 'Channel.Names', 'description': 'A comma-separated list of channel/antibody names that corresponds to the channel numbers (e.g., FSC, SSC, CD4, CD45RA).'}], 'default_args': [{'name': 'task_name', 'value': 'FLAMEPreprocess', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'file.type', 'value': 'fcs', 'label': 'File.Type', 'description': 'File type of input sample data.'}, {'name': 'data.scale', 'value': '262144', 'label': 'Data.Scale', 'description': 'The original data scale. 18-bit (default) or 4-decade.'}, {'name': 'remove.dead', 'value': 'F', 'label': 'Remove.Dead', 'description': 'Whether to attempt removing dead cells from the samples.'}, {'name': 'scatter.channels', 'value': '1,2', 'label': 'Scatter.Channels', 'description': 'A comma-separated list of scatter channel numbers (e.g., 1, 2, 3).'}, {'name': 'transformation', 'value': 'logicle', 'label': 'Transformation', 'description': 'The transformation to apply'}, {'name': 'logicle.cofactor', 'value': '3', 'label': 'Logicle.Cofactor', 'description': 'A cofactor that tunes the logicle transformation. The cofactor is usually greater than 0 and less than 10.'}, {'name': 'arsinh.cofactor', 'value': '250', 'label': 'Arsinh.Cofactor', 'description': 'A cofactor that tunes the arsinh transformation. The cofactor is 0 usually between 0 exclusively and a few thousand.'}, {'name': 'output.prefix', 'value': '<dataset_basename>', 'label': 'Output.Prefix', 'description': 'A prefix for output files'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'FLAMEPreviewTransformation': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Evaluates data transformations based on one representative sample', 'required_args': [{'name': 'data.file', 'value': '', 'label': 'Data.File', 'description': 'A flow file in .txt or .fcs format for one sample.'}, {'name': 'channels', 'value': '', 'label': 'Channels', 'description': 'A comma-separated list of channel numbers indicating which columns (antibodies) to keep (e.g., 1, 2, 3, 7).'}, {'name': 'channel.names', 'value': '', 'label': 'Channel.Names', 'description': 'A comma-separated list of channel or antibody names that corresponds to the channel numbers (e.g., FSC, SSC, CD4, CD45RA).'}], 'default_args': [{'name': 'task_name', 'value': 'FLAMEPreviewTransformation', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'file.type', 'value': 'fcs', 'label': 'File.Type', 'description': 'File type of input sample data.'}, {'name': 'data.scale', 'value': '262144', 'label': 'Data.Scale', 'description': 'The original data scale. 18-bit (default) or 4-decade.'}, {'name': 'scatter.channels', 'value': '1,2', 'label': 'Scatter.Channels', 'description': 'A comma-separated list of scatter channel numbers (e.g., 1, 2, 3).'}, {'name': 'transformation', 'value': 'logicle', 'label': 'Transformation', 'description': 'The transformation(s) to apply'}, {'name': 'logicle.low', 'value': '0.5', 'label': 'Logicle.Low', 'description': 'The cofactor that tunes the logicle transformation is usually between 0 and 10 exclusively. Specify the lower limit of the cofactor range to examine.'}, {'name': 'logicle.high', 'value': '5', 'label': 'Logicle.High', 'description': 'The cofactor that tunes the logicle transformation is usually greater than 0 and less than 10. Specify the upper limit of the cofactor range to examine.'}, {'name': 'arsinh.low', 'value': '1', 'label': 'Arsinh.Low', 'description': 'The cofactor that tunes the arsinh transformation is usually between 0 and a few thousand. Specify the lower limit of the cofactor range to examine.'}, {'name': 'arsinh.high', 'value': '1000', 'label': 'Arsinh.High', 'description': 'The cofactor that tunes the arsinh transformation is usually between 0 and a few thousand. Specify the upper limit of the cofactor range to examine.'}, {'name': 'output.prefix', 'value': '<data.file_basename>', 'label': 'Output.Prefix', 'description': 'A prefix for output files'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'FLAMEViewer': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Visualizes results from the FLAME method', 'required_args': [{'name': 'membership.file', 'value': '', 'label': 'Membership.File', 'description': 'A membership.txt file output from the FLAMEContourDataGenerator module. This membership file must match with the contour plot file.'}, {'name': 'contour.plot.file', 'value': '', 'label': 'Contour.Plot.File', 'description': 'The contour plot output file from the FLAMEContourDataGenerator module. This contour plot file must match with the membership file.'}], 'default_args': [{'name': 'task_name', 'value': 'FLAMEViewer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'FlowClustClassifyFCS': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Classify flow cytometry data using the FlowClust algorithm.', 'required_args': [{'name': 'Input.FCS.data.file', 'value': '', 'label': 'Input.FCS.Data.File', 'description': 'The FCS file to be clustered.'}, {'name': 'K', 'value': '', 'label': 'K', 'description': 'The number of populations that the algorithm shall be looking for. If you do not know the K value, you can use the SuggestNumberOfPopulations module to estimate it.'}, {'name': 'Channels', 'value': '', 'label': 'Channels', 'description': 'A comma-separated list of parameter short names ($PnN values) that shall be used to classify the populations. All parameters will be used by default, i.e., if no value for the Channels parameter is provided. The Time parameter is excluded automatically.'}], 'default_args': [{'name': 'task_name', 'value': 'FlowClustClassifyFCS', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'CSV.file.name', 'value': '<Input.FCS.data.file_basename>', 'label': 'CSV.File.Name', 'description': 'The base file name to split the results into CSV files, each containing a single population.'}, {'name': 'CLS.file.name', 'value': '<Input.FCS.data.file_basename>.cls', 'label': 'CLS.File.Name', 'description': 'The name of the CLS output file, which will store the population number for each event.'}, {'name': 'Quantile', 'value': '0.9', 'label': 'Quantile', 'description': 'Outlier quantile. Events outside the specified quantile region will be classified as outliers and be assigned to class 0.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'FlowFingerprinting': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'takes a flow set and generates a description of the multivariate probability distribution function of its flow cytometry data in the form of a fingerprint', 'required_args': [{'name': 'Input.FCS.data.files', 'value': '', 'label': 'Input.FCS.Data.Files', 'description': 'The FCS file(s) to be fingerprinted.'}, {'name': 'Model.FCS.data.file', 'value': '', 'label': 'Model.FCS.Data.File', 'description': 'Training data (FCS file) for the underlying model. If none, the input FCS data is used for creating a default model.'}, {'name': 'Model.parameters', 'value': '', 'label': 'Model.Parameters', 'description': 'A comma-separated list of dimensions/parameters/channels that shall be used for the underlying model. Use either parameter names or indexes. All dimensions except for Time will be used if this parameter is not provided.'}, {'name': 'Model.number.of.recursions', 'value': '', 'label': 'Model.Number.Of.Recursions', 'description': 'Number of times the FCS training data will be subdivided.'}, {'name': 'Model.sample.size', 'value': '', 'label': 'Model.Sample.Size', 'description': 'Used to specify the per flowFrame sample size of the data to use in model generation. If NULL, all of the data in x is used. Setting this to a smaller number will speed up processing, at the cost of accuracy.'}, {'name': 'Workflow.parameter.file', 'value': '', 'label': 'Workflow.Parameter.File', 'description': 'A CSV file containing parameters if module is used in pipeline.'}], 'default_args': [{'name': 'task_name', 'value': 'FlowFingerprinting', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Model.dequantize', 'value': '0', 'label': 'Model.Dequantize', 'description': 'Option to make all events in the training set unique.'}, {'name': 'Output.QC.screening.fingerprint.stack', 'value': '0', 'label': 'Output.QC.Screening.Fingerprint.Stack', 'description': 'Graphical output in form of \"stack\" plot.'}, {'name': 'Output.QC.screening.gate.consistency', 'value': '0', 'label': 'Output.QC.Screening.Gate.Consistency', 'description': 'Graphical output in form of \"qc\" plot.'}, {'name': 'Output.QC.screening.plate', 'value': '1', 'label': 'Output.QC.Screening.Plate', 'description': 'Graphical output in form of \"plate\" plot.'}, {'name': 'Output.csv.file', 'value': 'fingerprints.csv', 'label': 'Output.Csv.File', 'description': 'Name for output CSV file.'}, {'name': 'Output.file.transformation', 'value': '0', 'label': 'Output.File.Transformation', 'description': 'Transformation used on data for CSV output file. \"raw\" is the number of events in each bin, \"normalized\" is the ratio of the number of events divided by the expected number (the expected number of events is calculated by dividing the total number of events in the instance used to make the fingerprint by the number of features in the fingerprint). \"log2norm\" is the log2 of the normalized values.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'FlowMeansCluster': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Cluster flow cytometry data using the FlowMeans algorithm.', 'required_args': [{'name': 'Input.FCS.data.file', 'value': '', 'label': 'Input.FCS.Data.File', 'description': 'The FCS file to be clustered.'}, {'name': 'Dimensions', 'value': '', 'label': 'Dimensions', 'description': 'A comma-separated list of dimensions (parameters/channels) that shall be used for clustering. Use either parameter names or indexes. All dimensions except for Time will be used if this parameter is not provided.'}, {'name': 'Dimensions.to.transform', 'value': '', 'label': 'Dimensions.To.Transform', 'description': 'A comma-separated list of dimensions (parameters/channels) that shall transformed as specified by previous parameter. This will be ignored if no transformation is specified above. If this parameter is not provided and transformation is specified above, the algorithm will use heuristics to identify parameters that shall be transformed. These heuristics are based on how parameters are stored in the FCS file, their resolution and their name. Again, you can use either parameter names or parameter indexes to specify dimensions to transform.'}, {'name': 'Maximum.number.of.clusters', 'value': '', 'label': 'Maximum.Number.Of.Clusters', 'description': 'Maximum number of clusters that FlowMeans will try. This value will be estimated automatically if the value of this parameter is not provided.'}, {'name': 'Number.of.clusters', 'value': '', 'label': 'Number.Of.Clusters', 'description': 'Number of clusters that FlowMeans shall search for. This parameter cannot be combined with the \"Maximum number of clusters\". Also, the number of clusters will be estimated automatically if this parameter is not provided.'}, {'name': 'Max.points.for.covariance', 'value': '', 'label': 'Max.Points.For.Covariance', 'description': 'The maximum number of data points (events) used for calculating the covariance. All data points will be used if this value is not provided.'}, {'name': 'Max.points.for.kernel.density', 'value': '', 'label': 'Max.Points.For.Kernel.Density', 'description': 'The maximum number of data points (events) used for counting the modes using kernel density estimation. All data points will be used if this value is not provided.'}, {'name': 'Seed', 'value': '', 'label': 'Seed', 'description': 'Seed value for the random generator; provide any consistent value to obtain reproducible results.'}], 'default_args': [{'name': 'task_name', 'value': 'FlowMeansCluster', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Transformation', 'value': 'ASinH', 'label': 'Transformation', 'description': 'Which transformation to apply on the data prior clustering. Fluorescence channels are usually better visualized and clustered using a transformation; the ASinH transformation produces good results on most data.'}, {'name': 'Maximum.number.of.iterations', 'value': '50', 'label': 'Maximum.Number.Of.Iterations', 'description': 'The maximum number of iterations allowed within the K-means component of the clustering.'}, {'name': 'Number.of.initialization.sets', 'value': '10', 'label': 'Number.Of.Initialization.Sets', 'description': 'The number of random sets used for initialization.'}, {'name': 'Distance.used', 'value': 'TRUE', 'label': 'Distance.Used', 'description': 'The distance being used within the FlowMeans algorithm.'}, {'name': 'Standardize.data', 'value': 'TRUE', 'label': 'Standardize.Data', 'description': 'Should each data dimension be standardized (i.e, transformed to a [0,1] interval) before clustering. This usually does not harm and is important if data in dimensions used for clustering covers significantly different ranges for different dimensions.   '}, {'name': 'Distance.matrix.update', 'value': 'Mahalanobis', 'label': 'Distance.Matrix.Update', 'description': 'How to update the distance matrix at each merging iteration. Note that updating using Mahalanobis distance is not possible if Euclidean distance is set in the \"Distance used\" above.  '}, {'name': 'Transform.residuals', 'value': 'FALSE', 'label': 'Transform.Residuals', 'description': 'Indicate whether or not to transform residuals to orthogonal distance.   '}, {'name': 'Add.noise', 'value': 'TRUE', 'label': 'Add.Noise', 'description': 'Should a little bit of uniform noise be added to the data to prevent singularity issues?'}, {'name': 'Draw.plots', 'value': 'TRUE', 'label': 'Draw.Plots', 'description': 'Should plots of clustered data be created for quick visual inspection of the results?'}, {'name': 'Print.debug.info', 'value': 'FALSE', 'label': 'Print.Debug.Info', 'description': 'Should additional debugging information be provided (e.g., this may be useful to debug cases where clustering does not work properly for certain data files).'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'FlowMergeCluster': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Cluster flow cytometry data using FlowMerge.', 'required_args': [{'name': 'Input.FCS.data.file', 'value': '', 'label': 'Input.FCS.Data.File', 'description': 'The FCS file to be clustered.'}, {'name': 'Dimensions', 'value': '', 'label': 'Dimensions', 'description': 'A comma-separated list of dimensions (parameters/channels) that shall be used for clustering. Use either parameter names or indexes. All dimensions except for Time will be used if this parameter is not provided.'}, {'name': 'Dimensions.to.transform', 'value': '', 'label': 'Dimensions.To.Transform', 'description': 'A comma-separated list of dimensions (parameters/channels) that shall transformed as specified by previous parameter. This will be ignored if no transformation is specified above. If this parameter is not provided and transformation is specified above, the algorithm will use heuristics to identify parameters that shall be transformed. These heuristics are based on how parameters are stored in the FCS file, their resolution and their name. Again, you can use either parameter names or parameter indexes to specify dimensions to transform.'}], 'default_args': [{'name': 'task_name', 'value': 'FlowMergeCluster', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Transformation', 'value': 'ASinH', 'label': 'Transformation', 'description': 'Which transformation to apply on the data prior clustering. Fluorescence channels are usually better visualized and clustered using a transformation; the ASinH transformation produces good results on most data.'}, {'name': 'Range.for.number.of.clusters', 'value': '1-10', 'label': 'Range.For.Number.Of.Clusters', 'description': 'The range for the number of subpopulations (clusters) that FlowMerge shall search for. FlowMerge will try to pick the best number of clusters from the specified range, which shall be provided in the min-max format, where both, mix and max are integers and min is smaller than max, e.g., 1-10.'}, {'name': 'Estimate.degrees.of.freedom', 'value': '0', 'label': 'Estimate.Degrees.Of.Freedom', 'description': 'An indication whether to estimate the degrees of freedom used for the t distribution when modelling data.'}, {'name': 'Degrees.of.freedom', 'value': '4', 'label': 'Degrees.Of.Freedom', 'description': 'The degrees of freedom used for the t distribution when modelling data. Provided value will be ignored if estimation is requested by the Estimate degrees of freedom parameter. Gaussian distribution will be used if Degrees of freedom are not provided and estimation is not requested.'}, {'name': 'Number.of.computing.nodes', 'value': '1', 'label': 'Number.Of.Computing.Nodes', 'description': \"How many nodes (e.g., processors, cores) to use if you wish to run the analysis in a parallel mode? Enter '1' if you wish do NOT want to use the parallel mode. Enter a number higher than 1 if your server/cluster has multiple computers/processors/cores and you want to utilize several of these for FlowMerge clustering.\"}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Fpkm_trackingToGct': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Takes a Cufflinks FPKM_tracking file and converts it to GCT format.', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'Input file in FPKM_TRACKING format'}], 'default_args': [{'name': 'task_name', 'value': 'Fpkm_trackingToGct', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'row.labels', 'value': 'Tracking ID', 'label': 'Row.Labels', 'description': 'Select row labels to be written to GCT file'}, {'name': 'row.descriptions', 'value': 'Locus', 'label': 'Row.Descriptions', 'description': 'Select row descriptions to be written to GCT file'}, {'name': 'filter.rows', 'value': 'yes', 'label': 'Filter.Rows', 'description': 'Whether to filter rows with invalid row names (ie., \"-\")'}, {'name': 'output.prefix', 'value': '<input.file_basename>', 'label': 'Output.Prefix', 'description': 'A label that will be used to name the GCT output file.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'FPmatching': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Matches Fluidigm fingerprint report to reference set of CCLE SNP6.0 array data.\\nTakes old Fluidigm software report or new PED style report (saved as .csv).\\nAlso backwards compatible with Sequenom reports (Combine 100k and AffySNP6.0 reports together into one file first).', 'required_args': [{'name': 'platereport', 'value': '', 'label': 'Platereport', 'description': 'Should be .csv format'}, {'name': 'platemapping', 'value': '', 'label': 'Platemapping', 'description': 'Should be .csv format'}], 'default_args': [{'name': 'task_name', 'value': 'FPmatching', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'format', 'value': 'new', 'label': 'Format', 'description': 'New PED style report OR Old Fluidigm software format OR Sequenom style report (expects combined 100k and SNP6.0 file)'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Gct2Arff': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Description: Convert a gene pattern .gct / .cls file pair into an .arff file', 'required_args': [{'name': 'gct.file', 'value': '', 'label': 'Gct.File', 'description': 'The dataset features - .gct'}, {'name': 'cls.file', 'value': '', 'label': 'Cls.File', 'description': 'The dataset class labels - .cls'}], 'default_args': [{'name': 'task_name', 'value': 'Gct2Arff', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<gct.file_basename>', 'label': 'Output.File', 'description': 'The output file base name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'GctToPcl': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Convert .gct to .pcl', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'GCT file to convert'}], 'default_args': [{'name': 'task_name', 'value': 'GctToPcl', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.filename', 'value': '<input.filename_basename>.pcl', 'label': 'Output.Filename', 'description': 'PCL file that will be produced'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'GENE_E': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Launches GENE-E with input file(s)', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'Dataset to be viewed in GENE-E'}, {'name': 'sample.information.or.class.file', 'value': '', 'label': 'Sample.Information.Or.Class.File', 'description': 'The cls or sample info file'}], 'default_args': [{'name': 'task_name', 'value': 'GENE_E', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'GeneListSignificanceViewer': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Views the results of marker analysis', 'required_args': [{'name': 'inputfilename', 'value': '', 'label': 'Inputfilename', 'description': 'odf format file from marker analysis (GeneNeighbors or ClassNeighbors)'}], 'default_args': [{'name': 'task_name', 'value': 'GeneListSignificanceViewer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'GeneNeighbors': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Select genes that most closely resemble a continuous profile (e.g., another gene)', 'required_args': [{'name': 'data.filename', 'value': '', 'label': 'Data.Filename', 'description': 'data file - .gct, .res, .odf'}, {'name': 'gene.accession', 'value': '', 'label': 'Gene.Accession', 'description': 'reference gene accession from data file to find neighbors for'}], 'default_args': [{'name': 'task_name', 'value': 'GeneNeighbors', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'num.neighbors', 'value': '50', 'label': 'Num.Neighbors', 'description': 'number of neighbors to find'}, {'name': 'marker.list.file', 'value': '<data.filename_basename>.markerlist', 'label': 'Marker.List.File', 'description': 'output filename for analysis results - .odf file'}, {'name': 'marker.dataset.file', 'value': '<data.filename_basename>.markerdata', 'label': 'Marker.Dataset.File', 'description': 'output filename raw data for selected markers - .gct file'}, {'name': 'distance.metric', 'value': '3', 'label': 'Distance.Metric', 'description': 'continuous metric for finding neighbors'}, {'name': 'filter.data', 'value': 'no', 'label': 'Filter.Data', 'description': 'if no, values below will be ignored'}, {'name': 'min.threshold', 'value': '20', 'label': 'Min.Threshold', 'description': 'minimum threshold for data'}, {'name': 'max.threshold', 'value': '20000', 'label': 'Max.Threshold', 'description': 'maximum threshold for data'}, {'name': 'min.fold.diff', 'value': '5', 'label': 'Min.Fold.Diff', 'description': 'minimum fold difference for filtering genes'}, {'name': 'min.abs.diff', 'value': '50', 'label': 'Min.Abs.Diff', 'description': 'minimum absolute difference for filtering genes'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'GenePatternDocumentExtractor': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Extracts GenePattern pipelines and other embedded GenePattern data from Word 2007 documents created with the Microsoft Word Add-In for the GenePattern Reproducible Research Document.', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'The file you want to extract the data from. Note that this must be a Word 2007 (.docx) file -- Files saved as Word 2003 or earlier (2003) may be compatible with the Add-In, but will not work with the extractor.'}], 'default_args': [{'name': 'task_name', 'value': 'GenePatternDocumentExtractor', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'GEOImporter': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Imports data from the Gene Expression Omnibus (GEO)', 'required_args': [{'name': 'GEO.accession', 'value': '', 'label': 'GEO.Accession', 'description': 'GEO Series record (GSExxx) or a GEO Dataset (GDSxxx). Either a GEO accession or a GEO SOFT file must be given.'}, {'name': 'GEO.SOFT.file', 'value': '', 'label': 'GEO.SOFT.File', 'description': 'A previously downloaded GEO SOFT file or its gzipped representation. Either a GEO SOFT file or a GEO accession must be given.'}, {'name': 'ftp.proxy.server', 'value': '', 'label': 'Ftp.Proxy.Server', 'description': 'FTP Proxy server'}, {'name': 'ftp.proxy.username', 'value': '', 'label': 'Ftp.Proxy.Username', 'description': 'FTP Proxy username'}, {'name': 'ftp.proxy.password', 'value': '', 'label': 'Ftp.Proxy.Password', 'description': 'FTP Proxy password'}, {'name': 'http.proxy.server', 'value': '', 'label': 'Http.Proxy.Server', 'description': 'HTTP Proxy server'}, {'name': 'http.proxy.username', 'value': '', 'label': 'Http.Proxy.Username', 'description': 'HTTP Proxy username'}, {'name': 'http.proxy.password', 'value': '', 'label': 'Http.Proxy.Password', 'description': 'HTTP Proxy password'}], 'default_args': [{'name': 'task_name', 'value': 'GEOImporter', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'data.column.name', 'value': 'VALUE', 'label': 'Data.Column.Name', 'description': 'The name of the column that contains the expression values'}, {'name': 'output.filename', 'value': '<GEO.accession><GEO.SOFT.file_basename>', 'label': 'Output.Filename', 'description': 'The name of the output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'GetDatasetInSilico': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'downloads a compressed .tgz file from the Insilico servers and extract it', 'required_args': [{'name': 'DownloadURL1.URL', 'value': '', 'label': 'DownloadURL1.URL', 'description': 'The URL to retrieve the file from'}], 'default_args': [{'name': 'task_name', 'value': 'GetDatasetInSilico', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'GISTIC_2.0': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Genomic Identification of Significant Targets in Cancer (version 2.0.22)', 'required_args': [{'name': 'refgene.file', 'value': '', 'label': 'Refgene.File', 'description': 'The reference genome file input provides cytoband and gene location information.'}, {'name': 'seg.file', 'value': '', 'label': 'Seg.File', 'description': 'The segmentation file contains the segmented data for all the samples identified by GLAD, CBS, or some other segmentation algorithm. (See GLAD file format in the Genepattern file formats documentation.) It is a six column, tab-delimited file with an optional first line identifying the columns. Positions are in base pair units.'}, {'name': 'markers.file', 'value': '', 'label': 'Markers.File', 'description': 'The markers file identifies the marker names and positions of the markers in the original dataset (before segmentation). It is a three column, tab-delimited file with an optional header. If not already, markers are sorted by genomic position.'}, {'name': 'array.list.file', 'value': '', 'label': 'Array.List.File', 'description': 'The array list file is an optional file identifying the subset of samples to be used in the analysis. It is a one column file with an optional header. The sample identifiers listed in the array list file must match the sample names given in the segmentation file.'}, {'name': 'cnv.file', 'value': '', 'label': 'Cnv.File', 'description': 'Specifies germline CNV (or other highly variant) regions to exclude from significance analysis. There are two forms for the cnv file: one allows CNVs to be identified by marker names in a single column; the other identifies excluded regions using their genomic location.'}], 'default_args': [{'name': 'task_name', 'value': 'GISTIC_2.0', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'gene.gistic', 'value': '1', 'label': 'Gene.Gistic', 'description': 'Flag indicating that the gene GISTIC algorithm should be used to calculate the significant regions of deletion using gene locations instead of a marker locations.'}, {'name': 'amplifications.threshold', 'value': '0.1', 'label': 'Amplifications.Threshold', 'description': 'Threshold for copy number amplifications. Regions with a log2 ratio above this value are considered amplified.'}, {'name': 'deletions.threshold', 'value': '0.1', 'label': 'Deletions.Threshold', 'description': 'Threshold for copy number deletions. Regions with a log2 ratio below the negative of this value are considered deletions.'}, {'name': 'join.segment.size', 'value': '4', 'label': 'Join.Segment.Size', 'description': 'Smallest number of markers to allow in segments from the segmented data. Segments that contain a number of markers less than or equal to this number are joined to the neighboring segment that is closest in copy number.'}, {'name': 'q-value.threshold', 'value': '0.25', 'label': 'Q-Value.Threshold', 'description': 'Maximum false discovery rate for a region to be considered significant.'}, {'name': 'remove.X', 'value': '1', 'label': 'Remove.X', 'description': 'Flag indicating whether to remove data from the X-chromosome before analysis.'}, {'name': 'cap.val', 'value': '1.5', 'label': 'Cap.Val', 'description': 'Minimum and maximum cap values on analyzed data. Regions with a log2 ratio greater than the cap are set to the cap value; regions with a log2 ratio less than -cap value are set to -cap.'}, {'name': 'confidence.level', 'value': '0.90', 'label': 'Confidence.Level', 'description': 'Confidence level used to calculate the region containing a driver.'}, {'name': 'run.broad.analysis', 'value': '0', 'label': 'Run.Broad.Analysis', 'description': 'Flag indicating whether an additional broad-level analysis should be performed.'}, {'name': 'focal.length.cutoff', 'value': '0.50', 'label': 'Focal.Length.Cutoff', 'description': 'Threshold used to distinguish broad from focal events, given in units of fraction of chromosome arm.'}, {'name': 'max.sample.segs', 'value': '2500', 'label': 'Max.Sample.Segs', 'description': 'Maximum number of segments allowed for a sample in the input data. Samples with more segments than this threshold are excluded from the analysis.'}, {'name': 'arm.peel', 'value': '0', 'label': 'Arm.Peel', 'description': 'Whether to perform arm level peel off. This helps separate peaks which cleans up noise.'}, {'name': 'sample.center', 'value': 'median', 'label': 'Sample.Center', 'description': 'Method for centering each sample prior to the GISTIC analysis.'}, {'name': 'gene.collapse.method', 'value': 'extreme', 'label': 'Gene.Collapse.Method', 'description': 'Method for reducing marker-level copy number data to the gene-level copy number data in the gene tables. Markers contained in the gene are used when available, otherwise the flanking marker or markers are used.'}, {'name': 'output.prefix', 'value': '<seg.file_basename>', 'label': 'Output.Prefix', 'description': 'The prefix for the output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'GISTICPreprocess': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Prepares SNP files for GISTIC', 'required_args': [{'name': 'snp.archive', 'value': '', 'label': 'Snp.Archive', 'description': 'A zipfile containing the input genotypes and signal intensities. These are .cnx files created by SNPFileCreator.'}, {'name': 'sample.info.file', 'value': '', 'label': 'Sample.Info.File', 'description': 'The sample info file is a tab-delimited text file (*.txt) containing information about the arrays.'}, {'name': 'array.list.file', 'value': '', 'label': 'Array.List.File', 'description': 'The array list file identifies the subset of samples to be used in the analysis. It is a one-column file with a header (??array??). The sample identifiers listed in the array list file must match the sample names given in the segmentation file.'}], 'default_args': [{'name': 'task_name', 'value': 'GISTICPreprocess', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'use.paired.normals', 'value': '0', 'label': 'Use.Paired.Normals', 'description': 'When set to 1, normalizes tumors against paired normals. (Default = 0; best results are usually achieved under this default setting)'}, {'name': 'quality.control.tumors', 'value': '1', 'label': 'Quality.Control.Tumors', 'description': 'When set to 1, performs histogram quality control on tumor samples and includes only those tumors with distinct copy number peaks in the output array list to be used in the GISTIC module. (Default = 1).'}, {'name': 'normal.selection.method', 'value': '1', 'label': 'Normal.Selection.Method', 'description': 'Method for selecting normals to use in sample normalization. Allowed values: 0 (use all normals), 1 (use the n normals most closely matched to the tumor??s noise profile). (Default = 1)'}, {'name': 'n.closest.n', 'value': '5', 'label': 'N.Closest.N', 'description': 'Number of normals to choose when using closest_n normalization. (Default = 5)'}, {'name': 'correct.batch.effect', 'value': '0', 'label': 'Correct.Batch.Effect', 'description': 'When set to 1, runs module to correct for marker intensity variations due to batch effect. (Default = 0)'}, {'name': 'batch.correct.minimum.size', 'value': '5', 'label': 'Batch.Correct.Minimum.Size', 'description': 'Minimum number of samples required in a batch for the batch to be eligible for batch correction. (Default = 5)'}, {'name': 'batch.correct.pvalue', 'value': '0.001', 'label': 'Batch.Correct.Pvalue', 'description': 'p-value threshold for batch effect correction. For each marker, signal intensities in each batch are compared to all others using a t-test. If the p-value for this comparison is less than the p-value threshold, the signal intensities in this batch will be brought to the mean of all other batches. (Default = 0.001)'}, {'name': 'verbosity', 'value': '0', 'label': 'Verbosity', 'description': 'Set verbosity level (stand-alone version only). At a verbosity level of 0, no text indicating program status is written to sdout. At a level of 30, the maximum amount of text is written to stdout. (Default = 0)'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'GLAD': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Gain and Loss Analysis of DNA', 'required_args': [{'name': 'copy.number.input.file', 'value': '', 'label': 'Copy.Number.Input.File', 'description': 'cn file that contains sorted SNPs and raw copy number value per probe. (note: data should not be in log2)'}], 'default_args': [{'name': 'task_name', 'value': 'GLAD', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.filename', 'value': '<copy.number.input.file_basename>.glad', 'label': 'Output.Filename', 'description': 'Name of the output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'GlobalAlignment': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'standard global sequence analysis', 'required_args': [{'name': 'subject.fasta.filename', 'value': '', 'label': 'Subject.Fasta.Filename', 'description': 'filename of subject FASTA file'}, {'name': 'match', 'value': '', 'label': 'Match', 'description': '> 0'}, {'name': 'mismatch', 'value': '', 'label': 'Mismatch', 'description': '< 0'}, {'name': 'gapopen', 'value': '', 'label': 'Gapopen', 'description': '< 0'}, {'name': 'gapextend', 'value': '', 'label': 'Gapextend', 'description': '< 0'}, {'name': 'query.filename', 'value': '', 'label': 'Query.Filename', 'description': 'FASTA file'}], 'default_args': [{'name': 'task_name', 'value': 'GlobalAlignment', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<subject.fasta.filename_basename>', 'label': 'Output.File', 'description': 'filename of output'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Golub.Slonim.1999.Nature.all.aml.pipeline': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'ALL/AML methodology, from Golub and Slonim et al., 1999', 'required_args': [], 'default_args': [{'name': 'task_name', 'value': 'Golub.Slonim.1999.Nature.all.aml.pipeline', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'GSEA': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Gene Set Enrichment Analysis', 'required_args': [{'name': 'expression.dataset', 'value': '', 'label': 'Expression.Dataset', 'description': 'Dataset file - .res, .gct'}, {'name': 'gene.sets.database', 'value': '', 'label': 'Gene.Sets.Database', 'description': 'Gene sets database from GSEA website.'}, {'name': 'gene.sets.database.file', 'value': '', 'label': 'Gene.Sets.Database.File', 'description': 'Gene sets database - .gmt, .gmx, .grp. Upload a gene set if your gene set is not listed as a choice for the gene sets database parameter.'}, {'name': 'phenotype.labels', 'value': '', 'label': 'Phenotype.Labels', 'description': 'Cls file - .cls, must be binary'}, {'name': 'target.profile', 'value': '', 'label': 'Target.Profile', 'description': 'Name of the target phenotype profile.  Only applicable if class file defines continuous labels (one or more phenotype profiles).  Leave blank if class file defines a discrete phenotype.'}, {'name': 'chip.platform.file', 'value': '', 'label': 'Chip.Platform.File', 'description': 'DNA Chip (array) annotation file from GSEA website.  Upload your own chip file if the one corresponding to your DNA Microarray platform is not listed in the drop-down menu.  A chip file is only required if collapse dataset is set to true.'}], 'default_args': [{'name': 'task_name', 'value': 'GSEA', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'number.of.permutations', 'value': '1000', 'label': 'Number.Of.Permutations', 'description': 'Number of permutations to perform'}, {'name': 'collapse.dataset', 'value': 'true', 'label': 'Collapse.Dataset', 'description': 'Select true to have GSEA collapse each probe set in the expression dataset into a single vector for the gene, which gets identified by its gene symbol.'}, {'name': 'permutation.type', 'value': 'phenotype', 'label': 'Permutation.Type', 'description': 'Type of permutations to perform'}, {'name': 'scoring.scheme', 'value': 'weighted', 'label': 'Scoring.Scheme', 'description': 'The statistic used to score hits (gene set members) and misses (non-members)'}, {'name': 'metric.for.ranking.genes', 'value': 'Signal2Noise', 'label': 'Metric.For.Ranking.Genes', 'description': 'Class separation metric - gene markers are ranked using this metric to produce the gene list'}, {'name': 'gene.list.sorting.mode', 'value': 'real', 'label': 'Gene.List.Sorting.Mode', 'description': 'Mode in which scores from the gene list should be considered'}, {'name': 'gene.list.ordering.mode', 'value': 'descending', 'label': 'Gene.List.Ordering.Mode', 'description': 'Direction in which the gene list should be ordered'}, {'name': 'max.gene.set.size', 'value': '500', 'label': 'Max.Gene.Set.Size', 'description': 'Gene sets larger than this are excluded from the analysis'}, {'name': 'min.gene.set.size', 'value': '15', 'label': 'Min.Gene.Set.Size', 'description': 'Gene sets smaller than this are excluded from the analysis'}, {'name': 'collapsing.mode.for.probe.sets.with.more.than.one.match', 'value': 'Max_probe', 'label': 'Collapsing.Mode.For.Probe.Sets.With.More.Than.One.Match', 'description': 'Collapsing mode'}, {'name': 'normalization.mode', 'value': 'meandiv', 'label': 'Normalization.Mode', 'description': 'Normalization mode'}, {'name': 'randomization.mode', 'value': 'no_balance', 'label': 'Randomization.Mode', 'description': 'Type of phenotype randomization (does not apply to gene set permutations)'}, {'name': 'omit.features.with.no.symbol.match', 'value': 'true', 'label': 'Omit.Features.With.No.Symbol.Match', 'description': 'If there is no known gene symbol match for a probe set, omit it from the collapsed dataset'}, {'name': 'make.detailed.gene.set.report', 'value': 'true', 'label': 'Make.Detailed.Gene.Set.Report', 'description': 'Create detailed gene set report (heat map, mountain plot, etc.) for every enriched gene set'}, {'name': 'median.for.class.metrics', 'value': 'false', 'label': 'Median.For.Class.Metrics', 'description': 'Use the median of each class instead of the mean for the class separation metrics'}, {'name': 'number.of.markers', 'value': '100', 'label': 'Number.Of.Markers', 'description': 'Number of markers'}, {'name': 'plot.graphs.for.the.top.sets.of.each.phenotype', 'value': '20', 'label': 'Plot.Graphs.For.The.Top.Sets.Of.Each.Phenotype', 'description': 'Plot GSEA mountain and related plots for the top sets of each phenotype'}, {'name': 'random.seed', 'value': 'timestamp', 'label': 'Random.Seed', 'description': 'Seed to use for randomization'}, {'name': 'save.random.ranked.lists', 'value': 'false', 'label': 'Save.Random.Ranked.Lists', 'description': 'Whether to save random ranked lists'}, {'name': 'output.file.name', 'value': '<expression.dataset_basename>.zip', 'label': 'Output.File.Name', 'description': 'Name of the output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'GSEALeadingEdgeViewer': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Leading Edge Viewer for GSEA results', 'required_args': [{'name': 'zip.file', 'value': '', 'label': 'Zip.File', 'description': 'zip output file from GSEA - .zip. '}], 'default_args': [{'name': 'task_name', 'value': 'GSEALeadingEdgeViewer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'GSEAPreranked': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Runs the gene set enrichment analysis against a user-supplied ranked list of genes.', 'required_args': [{'name': 'ranked.list', 'value': '', 'label': 'Ranked.List', 'description': 'Rank ordered gene list - .rnk'}, {'name': 'gene.sets.database', 'value': '', 'label': 'Gene.Sets.Database', 'description': 'Gene sets database from MSigDB website.'}, {'name': 'gene.sets.database.file', 'value': '', 'label': 'Gene.Sets.Database.File', 'description': 'Gene sets database - .gmt, .gmx, .grp. Upload a gene set if your gene set is not listed as a choice for the gene sets database parameter.'}, {'name': 'chip.platform.file', 'value': '', 'label': 'Chip.Platform.File', 'description': 'DNA Chip (array) annotation file from GSEA website.  Upload your own chip file if the one corresponding to your DNA Microarray platform is not listed in the drop-down menu.  A chip file is only required if collapse dataset is set to true.'}], 'default_args': [{'name': 'task_name', 'value': 'GSEAPreranked', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'number.of.permutations', 'value': '1000', 'label': 'Number.Of.Permutations', 'description': 'Number of permutations to perform'}, {'name': 'collapse.dataset', 'value': 'false', 'label': 'Collapse.Dataset', 'description': 'Select true to have GSEAPreranked collapse each feature set in the ranked list into a single vector for the gene, which gets identified by its gene symbol.'}, {'name': 'scoring.scheme', 'value': 'weighted', 'label': 'Scoring.Scheme', 'description': 'The statistic used to score hits (gene set members) and misses (non-members)'}, {'name': 'max.gene.set.size', 'value': '500', 'label': 'Max.Gene.Set.Size', 'description': 'Gene sets larger than this are excluded from the analysis'}, {'name': 'min.gene.set.size', 'value': '15', 'label': 'Min.Gene.Set.Size', 'description': 'Gene sets smaller than this are excluded from the analysis'}, {'name': 'normalization.mode', 'value': 'meandiv', 'label': 'Normalization.Mode', 'description': 'Normalization mode'}, {'name': 'collapsing.mode', 'value': 'Max_probe', 'label': 'Collapsing.Mode', 'description': 'Collapsing mode'}, {'name': 'omit.features.with.no.symbol.match', 'value': 'true', 'label': 'Omit.Features.With.No.Symbol.Match', 'description': 'If there is no known gene symbol match for a probe set, omit it from the collapsed dataset; only relevant when collapse dataset is set to true'}, {'name': 'make.detailed.gene.set.report', 'value': 'true', 'label': 'Make.Detailed.Gene.Set.Report', 'description': 'Create detailed gene set report (heat map, mountain plot, etc.) for every enriched gene set'}, {'name': 'num.top.sets', 'value': '20', 'label': 'Num.Top.Sets', 'description': 'Plot GSEA mountain and related plots for the top sets of each phenotype'}, {'name': 'random.seed', 'value': 'timestamp', 'label': 'Random.Seed', 'description': 'Seed to use for generating random permutations.'}, {'name': 'output.file.name', 'value': '<ranked.list_basename>.zip', 'label': 'Output.File.Name', 'description': 'Name of the output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'HAPSEG': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '[**Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] A probabilistic method to interpret bi-allelic marker data in cancer samples.', 'required_args': [{'name': 'plate.name', 'value': '', 'label': 'Plate.Name', 'description': 'Name of the sample plate'}, {'name': 'array.name', 'value': '', 'label': 'Array.Name', 'description': 'Name of the chip that was run'}, {'name': 'seg.file', 'value': '', 'label': 'Seg.File', 'description': 'Segmented copy number data file for this sample. If not provided, HAPSEG will segment the data for you.'}, {'name': 'snp.file', 'value': '', 'label': 'Snp.File', 'description': 'SNP intensity file for this sample'}, {'name': 'use.pop', 'value': '', 'label': 'Use.Pop', 'description': 'Population to run with (currently supported values: CEPH, CH, JA, YOR)'}, {'name': 'calls.file', 'value': '', 'label': 'Calls.File', 'description': 'If using a matched normal sample, a Birdseed calls file needs to be supplied'}, {'name': 'mn.sample', 'value': '', 'label': 'Mn.Sample', 'description': 'If using a matched sample, the name of that matched normal sample'}, {'name': 'clusters.file', 'value': '', 'label': 'Clusters.File', 'description': \"If calibrate.data is 'On' the user must supply a Birdseed clusters file\"}, {'name': 'prev.theta.file', 'value': '', 'label': 'Prev.Theta.File', 'description': 'An optional file storing the previous theta values'}], 'default_args': [{'name': 'task_name', 'value': 'HAPSEG', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'out.file.name', 'value': '<plate.name>_<array.name>.segdat.RData', 'label': 'Out.File.Name', 'description': 'The name of the output file.'}, {'name': 'genome.build', 'value': 'ftp://gpftp.broadinstitute.org/module_support_files/HAPSEG_1.1.1/BEAGLE/phasedBGL/hg19/', 'label': 'Genome.Build', 'description': \"Which build of the human genome to use.  Only 'hg18' and 'hg19' are currently supported.  This will select a set of phased BEAGLE files.\"}, {'name': 'platform', 'value': 'SNP_6.0', 'label': 'Platform', 'description': \"The chip type used. The supported values are currently 'SNP_250K_STY' and 'SNP_6.0'\"}, {'name': 'impute.gt', 'value': 'TRUE', 'label': 'Impute.Gt', 'description': \"If TRUE will impute genotypes via BEAGLE. We recommend this be TRUE unless there's a reason to do otherwise\"}, {'name': 'plot.segfit', 'value': 'TRUE', 'label': 'Plot.Segfit', 'description': 'If TRUE will plot JPG images of the segmentations fits'}, {'name': 'merge.small', 'value': 'TRUE', 'label': 'Merge.Small', 'description': 'If TRUE will merge small segments'}, {'name': 'merge.close', 'value': 'TRUE', 'label': 'Merge.Close', 'description': 'if TRUE will merge close segments'}, {'name': 'min.seg.size', 'value': '10', 'label': 'Min.Seg.Size', 'description': 'Minimum segment size'}, {'name': 'normal', 'value': 'FALSE', 'label': 'Normal', 'description': 'If TRUE this sample will be treated as a normal'}, {'name': 'out.p', 'value': '0.05', 'label': 'Out.P', 'description': 'Outlier probability'}, {'name': 'seg.merge.thresh', 'value': '1e-10', 'label': 'Seg.Merge.Thresh', 'description': 'Segmentation threshold'}, {'name': 'use.normal', 'value': 'FALSE', 'label': 'Use.Normal', 'description': 'If TRUE will use a matched normal sample if one is provided'}, {'name': 'drop.x', 'value': 'FALSE', 'label': 'Drop.X', 'description': 'If TRUE will remove the X chromosome from the calculation'}, {'name': 'drop.y', 'value': 'TRUE', 'label': 'Drop.Y', 'description': 'If TRUE will remove the Y chromosome from the calculation'}, {'name': 'calibrate.data', 'value': 'NA', 'label': 'Calibrate.Data', 'description': \"If 'On', will perform a calibration on the input data.  If 'Off', no calibration will be performed. If left at the default value (Inferred), the calibration status will be inferred.\"}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'HeatMapImage': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Creates a heat map from a dataset', 'required_args': [{'name': 'input.dataset', 'value': '', 'label': 'Input.Dataset', 'description': 'The input dataset - .res, .gct, .odf'}, {'name': 'rows.to.highlight', 'value': '', 'label': 'Rows.To.Highlight', 'description': 'Rows to highlight in the heat map'}, {'name': 'row.highlight.color', 'value': '', 'label': 'Row.Highlight.Color', 'description': 'The color to highlight the rows if a rows to highlight file is provided'}, {'name': 'color.palette', 'value': '', 'label': 'Color.Palette', 'description': 'Color palette consisting of one r:g:b triplet or one hex value per line'}], 'default_args': [{'name': 'task_name', 'value': 'HeatMapImage', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output', 'value': '<input.dataset_basename>', 'label': 'Output', 'description': 'The output file name -.jpeg, .png, .tiff, .bmp, .eps'}, {'name': 'output.format', 'value': 'jpeg', 'label': 'Output.Format', 'description': 'The output file format'}, {'name': 'column.size', 'value': '8', 'label': 'Column.Size', 'description': 'The size in pixels of an element along the horizontal axis'}, {'name': 'row.size', 'value': '8', 'label': 'Row.Size', 'description': 'The size in pixels of an element along the vertical axis'}, {'name': 'show.grid', 'value': 'yes', 'label': 'Show.Grid', 'description': 'Whether to show a grid between elements'}, {'name': 'grid.color', 'value': 'black', 'label': 'Grid.Color', 'description': 'The grid color'}, {'name': 'show.row.descriptions', 'value': 'no', 'label': 'Show.Row.Descriptions', 'description': 'Whether to show row descriptions'}, {'name': 'show.row.names', 'value': 'yes', 'label': 'Show.Row.Names', 'description': 'Whether to show row names'}, {'name': 'color.scheme', 'value': 'row normalized', 'label': 'Color.Scheme', 'description': 'The color scheme'}, {'name': 'use.color.gradient', 'value': 'no', 'label': 'Use.Color.Gradient', 'description': 'Whether to use a linear color gradient or discrete colors'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'HeatMapViewer': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'A configurable heat map viewer that provides users with several options for manipulating and visualizing array-based data', 'required_args': [{'name': 'dataset', 'value': '', 'label': 'Dataset', 'description': 'The dataset to display - .gct'}], 'default_args': [{'name': 'task_name', 'value': 'HeatMapViewer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'helloWorld': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Generic hello world module', 'required_args': [], 'default_args': [{'name': 'task_name', 'value': 'helloWorld', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'message', 'value': 'Hello, World!', 'label': 'Message', 'description': ''}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'helloWorld_R2.15_Rscript': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Demo R2.15_Rscript module', 'required_args': [], 'default_args': [{'name': 'task_name', 'value': 'helloWorld_R2.15_Rscript', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'HierarchicalClustering': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Hierarchical Clustering', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'input data file name - .gct, .res, .pcl'}, {'name': 'log.transform', 'value': '', 'label': 'Log.Transform', 'description': 'log-transform the data before clustering'}, {'name': 'row.center', 'value': '', 'label': 'Row.Center', 'description': 'whether to center each row (gene) in the data'}, {'name': 'row.normalize', 'value': '', 'label': 'Row.Normalize', 'description': 'whether to normalize each row (gene) in the data'}, {'name': 'column.center', 'value': '', 'label': 'Column.Center', 'description': 'whether to center each column (sample) in the data'}, {'name': 'column.normalize', 'value': '', 'label': 'Column.Normalize', 'description': 'whether to normalize each column (sample) in the data'}], 'default_args': [{'name': 'task_name', 'value': 'HierarchicalClustering', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'column.distance.measure', 'value': '2', 'label': 'Column.Distance.Measure', 'description': 'distance measure for column (sample) clustering'}, {'name': 'row.distance.measure', 'value': '0', 'label': 'Row.Distance.Measure', 'description': 'distance measure for row (gene) clustering  NOTE: Filtering beforehand is recommended since row clustering is computationally intensive.'}, {'name': 'clustering.method', 'value': 'a', 'label': 'Clustering.Method', 'description': 'hierarchical clustering method to use'}, {'name': 'output.base.name', 'value': '<input.filename_basename>', 'label': 'Output.Base.Name', 'description': 'base name for output files'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'HierarchicalClusteringImage': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Creates an image of the dendrogram generated from HierarchicalClustering', 'required_args': [{'name': 'cdt', 'value': '', 'label': 'Cdt', 'description': 'Clustered data table, contains the original data, but reordered'}, {'name': 'gtr', 'value': '', 'label': 'Gtr', 'description': 'Gene Tree file, records the order in which genes (rows) were joined (at least one of gtr and atr files are required)'}, {'name': 'atr', 'value': '', 'label': 'Atr', 'description': 'Array Tree file, records the order in which samples (columns) were joined (at least one of gtr and atr files are required)'}, {'name': 'rows.to.highlight', 'value': '', 'label': 'Rows.To.Highlight', 'description': 'Rows to highlight in the heat map'}, {'name': 'row.highlight.color', 'value': '', 'label': 'Row.Highlight.Color', 'description': 'The color to highlight the rows if a rows to highlight file is provided'}, {'name': 'color.palette', 'value': '', 'label': 'Color.Palette', 'description': 'Color palette consisting of one r:g:b triplet or one hex value per line'}], 'default_args': [{'name': 'task_name', 'value': 'HierarchicalClusteringImage', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output', 'value': '<cdt_basename>', 'label': 'Output', 'description': 'The output file name - .bmp, .jpeg, .png, .tiff'}, {'name': 'output.format', 'value': 'jpeg', 'label': 'Output.Format', 'description': 'The output file format'}, {'name': 'column.size', 'value': '8', 'label': 'Column.Size', 'description': 'The size in pixels of an element along the horizontal axis'}, {'name': 'row.size', 'value': '8', 'label': 'Row.Size', 'description': 'The size in pixels of an element along the vertical axis'}, {'name': 'show.grid', 'value': 'yes', 'label': 'Show.Grid', 'description': 'Whether to show a grid between elements'}, {'name': 'grid.color', 'value': '0:0:0', 'label': 'Grid.Color', 'description': 'The grid color'}, {'name': 'show.row.descriptions', 'value': 'no', 'label': 'Show.Row.Descriptions', 'description': 'Whether to show row descriptions'}, {'name': 'show.row.names', 'value': 'yes', 'label': 'Show.Row.Names', 'description': 'Whether to show row names'}, {'name': 'color.scheme', 'value': 'row normalized', 'label': 'Color.Scheme', 'description': 'The color scheme'}, {'name': 'use.color.gradient', 'value': 'no', 'label': 'Use.Color.Gradient', 'description': 'Whether to use a linear color gradient or discrete colors'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'HierarchicalClusteringViewer': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Views results of hierarchical clustering.', 'required_args': [{'name': 'cdt.file', 'value': '', 'label': 'Cdt.File', 'description': 'Clustered data table, contains the original data, but reordered'}, {'name': 'gtr.file', 'value': '', 'label': 'Gtr.File', 'description': 'Gene tree file, records the order in which genes (rows) were joined (at least one of gtr file and atr file is required)'}, {'name': 'atr.file', 'value': '', 'label': 'Atr.File', 'description': 'Array tree file, records the order in which samples (columns) were joined (at least one of gtr file and atr file is required)'}], 'default_args': [{'name': 'task_name', 'value': 'HierarchicalClusteringViewer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Hu68kHu35kAtoU95': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Converts a list of Affymetrix Hu6800/Hu35KsubA probes to the corresponding Affymetrix U95Av2 probes.', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'the .pol, .tag or .grp file to be converted'}], 'default_args': [{'name': 'task_name', 'value': 'Hu68kHu35kAtoU95', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output', 'value': '<input.filename_basename>.U95', 'label': 'Output', 'description': 'name of the output file (same format as input file)'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Identify.an.up.or.downregulated.pathway.from.expression.data': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'This pipeline is used in the GenomeSpace recipe, \"Identify an up- or down-regulated pathway from expression data\". The recipe can be found at: http://www.genomespace.org/support/guides/recipes/sections/identify-an-up-or-down-regulated-pathway-from-expression-data.', 'required_args': [{'name': 'ConvertLineEndings1.input.filename', 'value': '', 'label': 'ConvertLineEndings1.Input.Filename', 'description': 'The input file (any non-binary file format)'}, {'name': 'ComparativeMarkerSelection2.cls.file', 'value': '', 'label': 'ComparativeMarkerSelection2.Cls.File', 'description': 'The class file - .cls'}], 'default_args': [{'name': 'task_name', 'value': 'Identify.an.up.or.downregulated.pathway.from.expression.data', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Identify.and.Validate.a.Consensus.Signature.Using.Gene.Expression.Data': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'This pipeline is used in the GenomeSpace recipe, \"Identify and Validate a Consensus Signature Using Gene Expression Data\". The recipe can be found at: http://genomespace.org/support/guides/recipes/sections/identify-and-validate-a-consensus-signature-using-gene-expression-data.', 'required_args': [{'name': 'SelectFeaturesColumns1.input.filename', 'value': '', 'label': 'SelectFeaturesColumns1.Input.Filename', 'description': 'input file - .res, .gct, .snp, .cn, .loh, .odf, or .cls file from which the slice is extracted'}, {'name': 'SelectFeaturesColumns2.input.filename', 'value': '', 'label': 'SelectFeaturesColumns2.Input.Filename', 'description': 'input file - .res, .gct, .snp, .cn, .loh, .odf, or .cls file from which the slice is extracted'}, {'name': 'SelectFeaturesRows3.list.filename', 'value': '', 'label': 'SelectFeaturesRows3.List.Filename', 'description': 'text file containing one identifier per line that specifies the rows to select. '}], 'default_args': [{'name': 'task_name', 'value': 'Identify.and.Validate.a.Consensus.Signature.Using.Gene.Expression.Data', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Identify.and.visualize.expressed.transcripts.in.RNASeq.data': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'This pipeline is used in the GenomeSpace recipe, \"Identify and visualize expressed transcripts in RNA-Seq data\". The recipe can be found at: http://www.genomespace.org/support/guides/recipes/sections/identify-and-visualize-expressed-transcripts-in-rna-seq-data.', 'required_args': [{'name': 'ConvertLineEndings1.input.filename', 'value': '', 'label': 'ConvertLineEndings1.Input.Filename', 'description': 'The input file (any non-binary file format)'}, {'name': 'TopHat2.reads.pair.1', 'value': '', 'label': 'TopHat2.Reads.Pair.1', 'description': 'Unpaired reads file or first mate for paired reads. One or more files containing reads in FASTA or FASTQ format (bz2 and gz compressed files are supported).'}, {'name': 'TopHat2.reads.pair.2', 'value': '', 'label': 'TopHat2.Reads.Pair.2', 'description': 'Second mate for paired reads. Zero or more files in FASTA or FASTQ format (bz2 and gz compressed files are supported).'}], 'default_args': [{'name': 'task_name', 'value': 'Identify.and.visualize.expressed.transcripts.in.RNASeq.data', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'IGV': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Launches IGV with input file(s)', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'Dataset to be viewed in IGV or IGV session file'}, {'name': 'index.file', 'value': '', 'label': 'Index.File', 'description': 'The index file.'}, {'name': 'genomeId', 'value': '', 'label': 'GenomeId', 'description': 'Choose the genome for your data. IGV will launch with hg19 or the last viewed genome by default.'}, {'name': 'locus', 'value': '', 'label': 'Locus', 'description': 'Provide a locus or gene of interest for your data. IGV will launch on chr1 or the last viewed chromosome by default. (e.g. chr1:2000-3000 or egfr)'}], 'default_args': [{'name': 'task_name', 'value': 'IGV', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'IGV.js': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Launches IGV.js with input file(s)', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'Track dataset to be viewed in IGV'}, {'name': 'index.file', 'value': '', 'label': 'Index.File', 'description': 'The index file (i.e .bai, .idx, or .tbi).'}, {'name': 'genome', 'value': '', 'label': 'Genome', 'description': 'Choose the genome for your data. IGV will launch with hg19.'}, {'name': 'locus', 'value': '', 'label': 'Locus', 'description': 'Provide a locus or gene of interest for your data. IGV will launch on chr1.'}], 'default_args': [{'name': 'task_name', 'value': 'IGV.js', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'IlluminaConcatenator': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Concatenate normalized Illumina probe sets into a single GCT file', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'ZIP file of normalized GCT files output by IlluminaNormalizer'}, {'name': 'output.file.type', 'value': '', 'label': 'Output.File.Type', 'description': 'Whether to concatenate the data or detection scores'}], 'default_args': [{'name': 'task_name', 'value': 'IlluminaConcatenator', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'IlluminaDASLPipeline': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'creates a GenePattern gct file from raw Illumina scan data', 'required_args': [{'name': 'IlluminaScanExtractor1.sample.description.file', 'value': '', 'label': 'IlluminaScanExtractor1.Sample.Description.File', 'description': 'file with one sample per line, one Sentrix Position, DAP#, and Sentrix ID per DAP'}, {'name': 'IlluminaConcatenator3.output.file.type', 'value': '', 'label': 'IlluminaConcatenator3.Output.File.Type', 'description': 'Whether to concatenate the data or detection scores'}], 'default_args': [{'name': 'task_name', 'value': 'IlluminaDASLPipeline', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'IlluminaScanExtractor1.outlier.method', 'value': '> 3 Std. Deviations', 'label': 'IlluminaScanExtractor1.Outlier.Method', 'description': 'outlier detection method (only applies to extraction using bead-level data)'}, {'name': 'IlluminaNormalizer2.normalization.method', 'value': 'CubicSpline', 'label': 'IlluminaNormalizer2.Normalization.Method', 'description': 'correction method. Note that mean background subtraction does not apply when the input is a single GCT file.'}, {'name': 'IlluminaNormalizer2.discard.controls', 'value': 'no', 'label': 'IlluminaNormalizer2.Discard.Controls', 'description': 'discard top 2 and bottom 2 negative controls (applies to mean background subtraction only)'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'IlluminaExpressionFileCreator': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Creates a GCT file from a zip of Illumina IDAT files and an Illumina manifest file', 'required_args': [{'name': 'idat.zip', 'value': '', 'label': 'Idat.Zip', 'description': 'Zip of Illumina IDAT files'}, {'name': 'manifest', 'value': '', 'label': 'Manifest', 'description': 'Illumina manifest file in tab-delimited (TXT) format. A manifest file is required; either selected from this list or specified in custom manifest below.'}, {'name': 'custom.manifest', 'value': '', 'label': 'Custom.Manifest', 'description': 'Other Illumina manifest file in tab-delimited (TXT) format. A manifest file is required; either selected from the list above or specified here.'}, {'name': 'chip', 'value': '', 'label': 'Chip', 'description': 'CHIP file to map probes to genes.'}, {'name': 'custom.chip', 'value': '', 'label': 'Custom.Chip', 'description': 'Other CHIP file to map probes to genes.'}, {'name': 'clm', 'value': '', 'label': 'Clm', 'description': 'CLM file to map file names to sample names.'}], 'default_args': [{'name': 'task_name', 'value': 'IlluminaExpressionFileCreator', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<idat.zip_basename>.gct', 'label': 'Output.File', 'description': 'Output file name'}, {'name': 'background.subtraction.mode', 'value': 'false', 'label': 'Background.Subtraction.Mode', 'description': 'Perform background subtraction.'}, {'name': 'collapse.mode', 'value': 'none', 'label': 'Collapse.Mode', 'description': 'Collapse probes to genes based on the manifest or CHIP file (if provided).'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'IlluminaNormalizer': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Normalize zipped Illumina scans', 'required_args': [{'name': 'zip.or.gct.file', 'value': '', 'label': 'Zip.Or.Gct.File', 'description': 'a zip of GCT files output by IlluminaScanExtractor (6k DASL) or a GCT file (whole-genome DASL)'}], 'default_args': [{'name': 'task_name', 'value': 'IlluminaNormalizer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'normalization.method', 'value': 'CubicSpline', 'label': 'Normalization.Method', 'description': 'correction method. Note that mean background subtraction does not apply when the input is a single GCT file.'}, {'name': 'output.file', 'value': '<zip.or.gct.file_basename>_<normalization.method>', 'label': 'Output.File', 'description': 'output file name'}, {'name': 'discard.controls', 'value': 'no', 'label': 'Discard.Controls', 'description': 'discard top 2 and bottom 2 negative controls (applies to mean background subtraction only)'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'IlluminaScanExtractor': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Extract intensity values from Illumina scans', 'required_args': [{'name': 'sample.description.file', 'value': '', 'label': 'Sample.Description.File', 'description': 'file with one sample per line, one Sentrix Position, DAP#, and Sentrix ID per DAP'}], 'default_args': [{'name': 'task_name', 'value': 'IlluminaScanExtractor', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'outlier.method', 'value': '> 3 Std. Deviations', 'label': 'Outlier.Method', 'description': 'outlier detection method (only applies to extraction using bead-level data)'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ImmPort_FLOCK_CrossSample': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'ImmPort FLOCK and CrossSample pipeline', 'required_args': [{'name': 'ImmPortFCSConvLogicleTrans1.Input.FCS.file', 'value': '', 'label': 'ImmPortFCSConvLogicleTrans1.Input.FCS.File', 'description': 'An FCS file that shall be transformed with the Logicle transformation, converted and saved as tab-delimited text file.'}, {'name': 'ImmPortColSelection2.Dimension.specification', 'value': '', 'label': 'ImmPortColSelection2.Dimension.Specification', 'description': 'A list of dimensions/columns to extract, specified by their indexes (starting from 1), e.g., 1,4,6. There should not be any white space characters (e.g, spaces, tabs, new lines, etc.) in the parameter value.'}, {'name': 'ImmPortCrossSample4.Input.file', 'value': '', 'label': 'ImmPortCrossSample4.Input.File', 'description': 'A tab-delimited text file with columns corresponding to parameters and rows to events. Such a text file may be generated from an FCS file by the ImmPortFCSConvLogicleTrans module.'}], 'default_args': [{'name': 'task_name', 'value': 'ImmPort_FLOCK_CrossSample', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ImmPort_FLOCK_Individual_FCS': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'ImmPort FLOCK and Individual FCS pipeline', 'required_args': [{'name': 'ImmPortFCSConvLogicleTrans1.Input.FCS.file', 'value': '', 'label': 'ImmPortFCSConvLogicleTrans1.Input.FCS.File', 'description': 'An FCS file that shall be transformed with the Logicle transformation, converted and saved as tab-delimited text file.'}, {'name': 'ImmPortColSelection2.Dimension.specification', 'value': '', 'label': 'ImmPortColSelection2.Dimension.Specification', 'description': 'A list of dimensions/columns to extract, specified by their indexes (starting from 1), e.g., 1,4,6. There should not be any white space characters (e.g, spaces, tabs, new lines, etc.) in the parameter value.'}], 'default_args': [{'name': 'task_name', 'value': 'ImmPort_FLOCK_Individual_FCS', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ImmPortColSelection': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Select a set of columns from a txt file', 'required_args': [{'name': 'Input.file', 'value': '', 'label': 'Input.File', 'description': 'A tab-delimited text file with columns corresponding to parameters and rows to events, converted from an fcs file, typically by ImmPortFCSConvLogicleTrans.'}, {'name': 'Dimension.specification', 'value': '', 'label': 'Dimension.Specification', 'description': 'A list of dimensions/columns to extract, specified by their indexes (starting from 1), e.g., 1,4,6. There should not be any white space characters (e.g, spaces, tabs, new lines, etc.) in the parameter value.'}], 'default_args': [{'name': 'task_name', 'value': 'ImmPortColSelection', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ImmPortCrossSample': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Applying an individual FLOCK result across another FCM sample (.txt) version 1 (1-round Kmeans)', 'required_args': [{'name': 'Input.file', 'value': '', 'label': 'Input.File', 'description': 'A tab-delimited text file with columns corresponding to parameters and rows to events. Such a text file may be generated from an FCS file by the ImmPortFCSConvLogicleTrans module.'}, {'name': 'Centroid.file', 'value': '', 'label': 'Centroid.File', 'description': 'A text-file specifying the position of each of the cell populations. Such a text file is supposed to be generated by the ImmPortFLOCK module (file population_center.txt).'}], 'default_args': [{'name': 'task_name', 'value': 'ImmPortCrossSample', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Number.of.iterations', 'value': '1', 'label': 'Number.Of.Iterations', 'description': 'The number of iterations that the algorithm will use to match events with the closest population; values from the [1, 10] range are allowed.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ImmPortCSV2TXT': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Convert a .csv file to a .txt file that FLOCK can read', 'required_args': [{'name': 'Input.CSV.data.file', 'value': '', 'label': 'Input.CSV.Data.File', 'description': 'The input CSV file that will be converted to a a tab-delimited text file; the CSV shall contain a header with parameter names.'}], 'default_args': [{'name': 'task_name', 'value': 'ImmPortCSV2TXT', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ImmPortFCSConvLogicleTrans': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Logicle transformation of an fcs file to .txt file', 'required_args': [{'name': 'Input.FCS.file', 'value': '', 'label': 'Input.FCS.File', 'description': 'An FCS file that shall be transformed with the Logicle transformation, converted and saved as tab-delimited text file.'}], 'default_args': [{'name': 'task_name', 'value': 'ImmPortFCSConvLogicleTrans', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ImmPortFLOCK': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Automated Gating with ImmPort FLOCK algorithm version 1.0', 'required_args': [{'name': 'Input.file', 'value': '', 'label': 'Input.File', 'description': 'A tab-delimited text file with columns corresponding to parameters and rows to events. The first row is a heading labelling parameters. Such a text file is supposed to be generated from an FCS file by the ImmPortFCSConvLogicleTrans module optionally followed by the ImmPortColSelection module (if applicable to remove unused parameters). Alternatively, this text file may be created by the ImmPortCSV2TXT module if the original data is in the CSV format.'}], 'default_args': [{'name': 'task_name', 'value': 'ImmPortFLOCK', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Number.of.bins', 'value': '0', 'label': 'Number.Of.Bins', 'description': 'The number of bins used to partition each dimension. The default value of 0 means that the algorithm will automatically determine a suitable number of bins. Any value from the [6,29] range may be provided to manually override algorithmic estimations.'}, {'name': 'Density.threshold', 'value': '0', 'label': 'Density.Threshold', 'description': 'The minimal number of events in a bin (in the hyperspace of used markers) needed to mark the bin as a dense bin. The default value of 0 means that the algorithm will automatically determine a suitable density threshold. Any value from the [3,99] range may be provided to manually override algorithmic estimations.'}, {'name': 'Maximum.number.of.populations', 'value': '30', 'label': 'Maximum.Number.Of.Populations', 'description': 'The maximum number of populations identifiable by the clustering algorithm. The default value of 30 works well in most cases but may be overridden by any value from the [5,999] range. Larger values may increase the execution time of the module.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ImputeMissingValues.KNN': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Imputes missing values', 'required_args': [{'name': 'data.filename', 'value': '', 'label': 'Data.Filename', 'description': 'Data file (missing values are stored as NA) - .gct'}], 'default_args': [{'name': 'task_name', 'value': 'ImputeMissingValues.KNN', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'k', 'value': '10', 'label': 'K', 'description': 'Number of neighbors to be used in the imputation'}, {'name': 'rowmax', 'value': '0.5', 'label': 'Rowmax', 'description': 'The maximum percent missing data allowed in any row'}, {'name': 'colmax', 'value': '0.8', 'label': 'Colmax', 'description': 'The maximum percent missing data allowed in any column'}, {'name': 'output.file', 'value': '<data.filename_basename>.imputed.gct', 'label': 'Output.File', 'description': 'The name of the output file - .gct'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Install.Proteogenomics.R.Packages': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Installs all packages and dependencies need to run the Protegenomics GenePattern modules', 'required_args': [], 'default_args': [{'name': 'task_name', 'value': 'Install.Proteogenomics.R.Packages', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'IntersectingIdentifiers': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Given 2 tab delimited files (res, gct, tab, odf) caculate which ids in the first intersect with the second. Outputs a file of intersecting, non-intersecting, in the first but not the second and in the second but not the first', 'required_args': [{'name': 'First.file', 'value': '', 'label': 'First.File', 'description': 'Any tab delimited file'}, {'name': 'Second.file', 'value': '', 'label': 'Second.File', 'description': 'Any tab delimited file'}], 'default_args': [{'name': 'task_name', 'value': 'IntersectingIdentifiers', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'First.file.identifier.column.index', 'value': '1', 'label': 'First.File.Identifier.Column.Index', 'description': 'Index of identifiers in first file, staring at 1'}, {'name': 'Second.file.identifier.column.index', 'value': '1', 'label': 'Second.File.Identifier.Column.Index', 'description': 'Index of identifiers in second file, staring at 1'}, {'name': 'Output.file.name.prefix', 'value': '<First.file_basename>', 'label': 'Output.File.Name.Prefix', 'description': 'Name of output file'}, {'name': 'First.file.skip.rows', 'value': '2', 'label': 'First.File.Skip.Rows', 'description': 'Number of rows to skip in the first file to avoid headers'}, {'name': 'Second.file.skip.rows', 'value': '2', 'label': 'Second.File.Skip.Rows', 'description': 'Number of rows to skip in the first file to avoid headers'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'JavaTreeView': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': \"Hierarchical clustering viewer that reads in Eisen's cdt, atr, and gtr files. A detailed description of these formats is available from http://microarray.ccgb.umn.edu/smd/html/MicroArray/help/formats.shtml\", 'required_args': [{'name': 'cdt.filename', 'value': '', 'label': 'Cdt.Filename', 'description': 'Clustered data table, contains the original data, but reordered'}, {'name': 'gtr.filename', 'value': '', 'label': 'Gtr.Filename', 'description': 'Gene Tree file, records the order in which genes (rows) were joined (at least one of gtr filename and atr filename is required)'}, {'name': 'atr.filename', 'value': '', 'label': 'Atr.Filename', 'description': 'Array Tree file, records the order in which samples (columns) were joined (at least one of gtr filename and atr filename is required)'}], 'default_args': [{'name': 'task_name', 'value': 'JavaTreeView', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'KMeansClassifyFCS': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Find clusters in a flow cytometry file with k means clustering', 'required_args': [{'name': 'Input.FCS.data.file', 'value': '', 'label': 'Input.FCS.Data.File', 'description': 'The FCS file'}, {'name': 'K', 'value': '', 'label': 'K', 'description': 'The number of populations you are looking for.'}, {'name': 'Dimensions', 'value': '', 'label': 'Dimensions', 'description': 'The channels to search for populations, separated by commas. Leave blank to search in all channels. E.g. FSC Lin, SS Lin, CD4 Log'}], 'default_args': [{'name': 'task_name', 'value': 'KMeansClassifyFCS', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'CSV.file.name', 'value': '<Input.FCS.data.file_basename>', 'label': 'CSV.File.Name', 'description': 'The base filename to split the results into csv files, each containing one population'}, {'name': 'CLS.file.name', 'value': '<Input.FCS.data.file_basename>', 'label': 'CLS.File.Name', 'description': 'A .CLS filename to store the population number for each data point'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'KMeansClustering': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Module that performs the K-Means Clustering algorithm', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'File containing data to cluster - .res, .gct, .odf'}, {'name': 'number.of.clusters', 'value': '', 'label': 'Number.Of.Clusters', 'description': 'Number of centroids (clusters)'}, {'name': 'cluster.by', 'value': '', 'label': 'Cluster.By', 'description': 'Whether to cluster by rows or columns'}], 'default_args': [{'name': 'task_name', 'value': 'KMeansClustering', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.base.name', 'value': '<input.filename_basename>_KMcluster_output', 'label': 'Output.Base.Name', 'description': 'The base output file name - .gct'}, {'name': 'seed.value', 'value': '12345', 'label': 'Seed.Value', 'description': 'Seed value for random number generator'}, {'name': 'distance.metric', 'value': '0', 'label': 'Distance.Metric', 'description': 'How to compute distance between points'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'KNN': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'K-nearest neighbors classification', 'required_args': [{'name': 'train.filename', 'value': '', 'label': 'Train.Filename', 'description': 'training data file name - .gct, .res, .odf type = Dataset, ignored if a saved model (saved.model.filename) is used'}, {'name': 'train.class.filename', 'value': '', 'label': 'Train.Class.Filename', 'description': 'class file for training data - .cls, ignored if a saved model (saved.model.filename) is used'}, {'name': 'saved.model.filename', 'value': '', 'label': 'Saved.Model.Filename', 'description': 'input KNN model file - .odf type = KNN Prediction Model'}, {'name': 'test.filename', 'value': '', 'label': 'Test.Filename', 'description': 'test data file name - .gct, .res, .odf type = Dataset'}, {'name': 'test.class.filename', 'value': '', 'label': 'Test.Class.Filename', 'description': 'class file for test data - .cls'}, {'name': 'min.std', 'value': '', 'label': 'Min.Std', 'description': 'minimum standard deviation if feature selection statistic includes min std option'}, {'name': 'feature.list.filename', 'value': '', 'label': 'Feature.List.Filename', 'description': 'features to use for prediction'}], 'default_args': [{'name': 'task_name', 'value': 'KNN', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'model.file', 'value': '<train.filename_basename>.model', 'label': 'Model.File', 'description': 'name of output KNN model file  - .odf type = KNN Prediction Model'}, {'name': 'num.features', 'value': '10', 'label': 'Num.Features', 'description': 'number of selected features if feature list filename is not specified'}, {'name': 'feature.selection.statistic', 'value': '0', 'label': 'Feature.Selection.Statistic', 'description': 'statistic to use to perform feature selection'}, {'name': 'num.neighbors', 'value': '3', 'label': 'Num.Neighbors', 'description': 'number of neighbors for KNN'}, {'name': 'weighting.type', 'value': 'distance', 'label': 'Weighting.Type', 'description': 'weighting type for neighbors'}, {'name': 'distance.measure', 'value': '1', 'label': 'Distance.Measure', 'description': 'distance measure'}, {'name': 'pred.results.file', 'value': '<test.filename_basename>.pred.odf', 'label': 'Pred.Results.File', 'description': 'name of output file for prediction results - .odf'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'KNNXValidation': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'KNN classification with Leave-One-Out Cross-Validation', 'required_args': [{'name': 'data.filename', 'value': '', 'label': 'Data.Filename', 'description': 'data file - .gct, .res, .odf type = Dataset'}, {'name': 'class.filename', 'value': '', 'label': 'Class.Filename', 'description': 'class file - .cls'}, {'name': 'min.std', 'value': '', 'label': 'Min.Std', 'description': 'minimum standard deviation if feature selection statistic includes min std option'}], 'default_args': [{'name': 'task_name', 'value': 'KNNXValidation', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'num.features', 'value': '10', 'label': 'Num.Features', 'description': 'number of selected features'}, {'name': 'feature.selection.statistic', 'value': '0', 'label': 'Feature.Selection.Statistic', 'description': 'statistic to use to perform feature selection'}, {'name': 'num.neighbors', 'value': '3', 'label': 'Num.Neighbors', 'description': 'number of neighbors for KNN'}, {'name': 'weighting.type', 'value': 'distance', 'label': 'Weighting.Type', 'description': 'weighting type for neighbors'}, {'name': 'distance.measure', 'value': '1', 'label': 'Distance.Measure', 'description': 'distance measure'}, {'name': 'pred.results.file', 'value': '<data.filename_basename>.pred.odf', 'label': 'Pred.Results.File', 'description': 'prediction results output file name .odf type = Prediction Results'}, {'name': 'feature.summary.file', 'value': '<data.filename_basename>.feat.odf', 'label': 'Feature.Summary.File', 'description': 'feature summary output file name - .odf type = Prediction Features'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'KSscore': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Computes the Kolmogorov-Smirnov score for a set of genes within an ordered list.', 'required_args': [{'name': 'query.filename', 'value': '', 'label': 'Query.Filename', 'description': 'query gene set file containing one identifier per line'}, {'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'file containing an ordered list of genes (.pol format)'}], 'default_args': [{'name': 'task_name', 'value': 'KSscore', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output', 'value': '<query.filename_basename>', 'label': 'Output', 'description': 'output file name (a .pol extension will be appended)'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'LandmarkMatch': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'A proteomics method to propogate identifed peptides across multiple MS runs', 'required_args': [{'name': 'peakList.filename', 'value': '', 'label': 'PeakList.Filename', 'description': 'Zip file containing peak lists'}, {'name': 'retentionTime.filename', 'value': '', 'label': 'RetentionTime.Filename', 'description': 'Zip file containing retention time (in this case from mapquant)'}, {'name': 'sampleInfo.filename', 'value': '', 'label': 'SampleInfo.Filename', 'description': 'CSV file with header: experiment, sample, class'}, {'name': 'globalID.filename', 'value': '', 'label': 'GlobalID.Filename', 'description': 'Global list of identified peptides'}, {'name': 'accurateMass.filename', 'value': '', 'label': 'AccurateMass.Filename', 'description': 'OPTIONAL: Specify the accurate mass table'}, {'name': 'prefitCoef.filename', 'value': '', 'label': 'PrefitCoef.Filename', 'description': 'OPTIONAL: Specify the prefit coefficients in a file'}, {'name': 'Bootstrap', 'value': '', 'label': 'Bootstrap', 'description': 'Use bootstrapping to calculate landmark match statistics'}], 'default_args': [{'name': 'task_name', 'value': 'LandmarkMatch', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'running.on.LSF', 'value': 'no', 'label': 'Running.On.LSF', 'description': 'Whether you are running on an LSF cluster version of GenePattern'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Life.Cycle.Time.Point.Correlation': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Calculate correlation to life cycle time point of p.falciparum (derisi data)', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'Input data file name. Data in CSV format, with plasmodbID as first column '}], 'default_args': [{'name': 'task_name', 'value': 'Life.Cycle.Time.Point.Correlation', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<input.file_basename>-correlation.csv', 'label': 'Output.File', 'description': 'Name of output file that will contain the resulting correlation table. CSV format.'}, {'name': 'correlation', 'value': 'pearson', 'label': 'Correlation', 'description': 'Type of correlation to calculate'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ListFiles': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': \"a helper module which outputs a filelist, the list of all files in the input directory, similar to the unix 'ls' command\", 'required_args': [{'name': 'dir', 'value': '', 'label': 'Dir', 'description': 'a directory'}, {'name': 'recursive', 'value': '', 'label': 'Recursive', 'description': ''}, {'name': 'includePattern', 'value': '', 'label': 'IncludePattern', 'description': 'an optional list of filename patterns, e.g. *.cls,*.gct'}, {'name': 'pruneDir', 'value': '', 'label': 'PruneDir', 'description': 'an optional list of directory name patterns to ignore, e.g. *.svn'}, {'name': 'order.by', 'value': '', 'label': 'Order.By', 'description': 'an optional flag for how to order the files'}, {'name': 'type', 'value': '', 'label': 'Type', 'description': ''}, {'name': 'output.filename', 'value': '', 'label': 'Output.Filename', 'description': 'the name of the output file, if none specified, write to stdout'}], 'default_args': [{'name': 'task_name', 'value': 'ListFiles', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'LocatePeaks': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Locates detected peaks in a spectrum.', 'required_args': [{'name': 'peaks.output.filename', 'value': '', 'label': 'Peaks.Output.Filename', 'description': 'The output from the Peaks module'}, {'name': 'locate.peaks.as', 'value': '', 'label': 'Locate.Peaks.As', 'description': 'max-location of maximum filtered value, area-center of area of the peak'}], 'default_args': [{'name': 'task_name', 'value': 'LocatePeaks', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<peaks.output.filename_basename>.locate.odf', 'label': 'Output.File', 'description': 'output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'LogicleTransformFCS': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Logicle transformation of (selected) parameters in a list mode FCS data file.', 'required_args': [{'name': 'Input.fcs.data.file', 'value': '', 'label': 'Input.Fcs.Data.File', 'description': 'FCS data file to transform.'}, {'name': 'Parameters.to.transform', 'value': '', 'label': 'Parameters.To.Transform', 'description': 'Comma-separated list of parameters to transform, parameters are specified by their short names ($PnN values); Optional. By default, all parameters with the \"FL\" prefix in their short name will be transformed.'}, {'name': 'T', 'value': '', 'label': 'T', 'description': 'T > 0; Data value at the top of the scale range to be displayed, optional. The maximum input value will be used if no explicit value for T is provided.'}, {'name': 'W', 'value': '', 'label': 'W', 'description': 'W >= 0; Linearization width in asymptotic decades; optional. The described calculation will be used to set W for each of the transformed FCS parameters if no explicit value for W is provided.'}], 'default_args': [{'name': 'task_name', 'value': 'LogicleTransformFCS', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'M', 'value': '4.5', 'label': 'M', 'description': 'M < 0; Full width of the intended Logicle display in asymptotic decades, optional. The default value is 4.5 (decades). Note also that for the resulting Logicle transformation, Logicle (T) = M.'}, {'name': 'A', 'value': '0', 'label': 'A', 'description': 'A, usually <= 0; Additional negative display range in asymptotic decades; optional. The default value is 0.'}, {'name': 'Use.Fast.Implementation', 'value': 'no', 'label': 'Use.Fast.Implementation', 'description': 'Use a fast implementation of Logicle with less precision; optional.'}, {'name': 'Output.fcs.file.name', 'value': '<Input.fcs.data.file_basename>.logicle_transformed.fcs', 'label': 'Output.Fcs.File.Name', 'description': 'The output FCS file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'LogisticRegression': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Logistic regression using clinical data and .cls file', 'required_args': [{'name': 'input.clin.data.filename', 'value': '', 'label': 'Input.Clin.Data.Filename', 'description': 'input clinical data (.txt file)'}, {'name': 'input.cls.filename', 'value': '', 'label': 'Input.Cls.Filename', 'description': 'input class information (.cls file)'}], 'default_args': [{'name': 'task_name', 'value': 'LogisticRegression', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': 'LogisticRegression_result', 'label': 'Output.File', 'description': 'output file name'}, {'name': 'response.variable', 'value': 'NA', 'label': 'Response.Variable', 'description': ''}, {'name': 'variable.continuous', 'value': 'NA', 'label': 'Variable.Continuous', 'description': 'variable.continuous'}, {'name': 'variable.category', 'value': 'NA', 'label': 'Variable.Category', 'description': '(e.g. sex,cls,grade,...)'}, {'name': 'variable.interaction.terms', 'value': 'NA', 'label': 'Variable.Interaction.Terms', 'description': '(e.g. sex:cls,sex:grade,...)'}, {'name': 'input.subgroup', 'value': 'NA', 'label': 'Input.Subgroup', 'description': '(e.g. cls:2) *only one subgroup can be specified'}, {'name': 'variable.selection', 'value': 'none', 'label': 'Variable.Selection', 'description': ''}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'LogTransformGLAD': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Log base 2 transform last column of GLAD file (for GISTIC)', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'The file to transform the last column of (glad format)'}], 'default_args': [{'name': 'task_name', 'value': 'LogTransformGLAD', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<input.filename_basename>.LogTransformed.glad', 'label': 'Output.File', 'description': 'Output filename'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'LOHPaired': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Computes LOH for paired samples', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'A .snp file that contains genotypes'}, {'name': 'sample.info.filename', 'value': '', 'label': 'Sample.Info.Filename', 'description': 'A sample info file with the paired information'}], 'default_args': [{'name': 'task_name', 'value': 'LOHPaired', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<input.filename_basename>.loh', 'label': 'Output.File', 'description': 'Output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Lu.Getz.Miska.Nature.June.2005.mouse.lung.pipeline': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Normal/tumor classifier and kNN prediction of mouse lung samples', 'required_args': [], 'default_args': [{'name': 'task_name', 'value': 'Lu.Getz.Miska.Nature.June.2005.mouse.lung.pipeline', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'MAGEMLImportViewer': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'A visualizer to import data in MAGE-ML format into GenePattern', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'The archive file containing the MAGE-ML files - .tar, .tgz, .tar.gz, .zip, .gz'}], 'default_args': [{'name': 'task_name', 'value': 'MAGEMLImportViewer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'MAGETABImportViewer': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'A visualizer to import data in MAGE-TAB format into GenePattern', 'required_args': [{'name': 'sdrf.file', 'value': '', 'label': 'Sdrf.File', 'description': 'The sdrf file - .txt'}, {'name': 'data.files', 'value': '', 'label': 'Data.Files', 'description': 'A zip file containing MAGE-TAB data matrix files - .zip'}], 'default_args': [{'name': 'task_name', 'value': 'MAGETABImportViewer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'MapChipFeaturesGeneral': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Change (map) the features (genes) of a dataset', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'Input filename - .res'}, {'name': 'lookup.table.filename', 'value': '', 'label': 'Lookup.Table.Filename', 'description': 'Name of the look up table file with the feature mapping'}, {'name': 'table.format', 'value': '', 'label': 'Table.Format', 'description': 'Table format'}, {'name': 'direction', 'value': '', 'label': 'Direction', 'description': 'Direction of look up'}], 'default_args': [{'name': 'task_name', 'value': 'MapChipFeaturesGeneral', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.prefix', 'value': '<input.filename_basename>.fea', 'label': 'Output.Prefix', 'description': 'prefix for output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Marker.Selection': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Marker selection, ranking and classification', 'required_args': [{'name': 'gct.file', 'value': '', 'label': 'Gct.File', 'description': 'Input data file (in gct format)'}, {'name': 'cls.file', 'value': '', 'label': 'Cls.File', 'description': 'Class file (cls format) to match with input data file'}], 'default_args': [{'name': 'task_name', 'value': 'Marker.Selection', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.prefix', 'value': '<gct.file_basename>', 'label': 'Output.Prefix', 'description': 'Prefix for output file names'}, {'name': 'fdr.threshold', 'value': '0.01', 'label': 'Fdr.Threshold', 'description': 'FDR threshold for calling statistically significant markers'}, {'name': 'gct.id.col', 'value': '2', 'label': 'Gct.Id.Col', 'description': 'Column in data file containing (gene or protein) ID'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'MClustClusterLabel': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Model based labeling/matching of clusters (mate-clustering)', 'required_args': [{'name': 'Input.clustering.results', 'value': '', 'label': 'Input.Clustering.Results', 'description': 'The input clustering results (of multiple data files) in CSV files with each clustered population stored in a separate CSV file. All these CSV files shall have the same columns (dimensions/channels) or at least, they all shall include columns specified by the dimensions parameter.'}, {'name': 'No.of.labels', 'value': '', 'label': 'No.Of.Labels', 'description': 'The number of distinct labels (populations) to look for. You may run the MClustClusterLabelBIC module to get the number of labels you should look for.'}, {'name': 'Dimensions', 'value': '', 'label': 'Dimensions', 'description': 'A comma-separated list of dimensions (parameters/channels) that shall be used for meta clustering (clustering of means of clusters). All specified dimensions shall be included in all the input clustering results CSV files. All dimensions (but Time) will be used if this parameter is not provided.'}], 'default_args': [{'name': 'task_name', 'value': 'MClustClusterLabel', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Model', 'value': 'VII', 'label': 'Model', 'description': 'Specific model used for meta clustering; see documentation for details about different models.'}, {'name': 'Uncertainty.threshold', 'value': '65', 'label': 'Uncertainty.Threshold', 'description': 'The threshold (in percentage) for uncertainty when assigning labels. Input a value greater than 50 and lower than 100. Populations where no label can be assigned with probability greater or equal to this uncertainty threshold will be left unassigned in the results.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'MClustClusterLabelBIC': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Bayesian information criterion (BIC) for model based labeling/matching of clusters; helps estimating the number of labels for MClustClusterLabel.', 'required_args': [{'name': 'Input.clustering.results', 'value': '', 'label': 'Input.Clustering.Results', 'description': 'The input clustering results (of multiple data files) in CSV files with each clustered population stored in a separate CSV file. All these CSV files shall have the same columns (dimensions/channels) or at least, they all shall include columns specified by the dimensions parameter.'}, {'name': 'Dimensions', 'value': '', 'label': 'Dimensions', 'description': 'A comma-separated list of dimensions (parameters/channels) that shall be used for meta clustering (clustering of means of clusters). All specified dimensions shall be included in all the input clustering results CSV files. All dimensions (but Time) will be used if this parameter is not provided.'}], 'default_args': [{'name': 'task_name', 'value': 'MClustClusterLabelBIC', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Range', 'value': '1-10', 'label': 'Range', 'description': 'The range of distinct labels (populations, components) to calculate BIC for. Provide the range in the min-max format, where both, mix and max are integers and min is smaller than max.'}, {'name': 'EII', 'value': 'TRUE', 'label': 'Eii', 'description': 'Include the EII model (spherical, equal volume) when calculating BIC.'}, {'name': 'VII', 'value': 'TRUE', 'label': 'Vii', 'description': 'Include the VII model (spherical, unequal volume) when calculating BIC.'}, {'name': 'EEI', 'value': 'FALSE', 'label': 'Eei', 'description': 'Include the EEI model (diagonal, equal volume and shape) when calculating BIC.'}, {'name': 'VEI', 'value': 'FALSE', 'label': 'Vei', 'description': 'Include the VEI model (diagonal, varying volume, equal shape) when calculating BIC.'}, {'name': 'EVI', 'value': 'FALSE', 'label': 'Evi', 'description': 'Include the EVI model (diagonal, equal volume, varying shape) when calculating BIC.'}, {'name': 'VVI', 'value': 'FALSE', 'label': 'Vvi', 'description': 'Include the VVI model (diagonal, varying volume and shape) when calculating BIC.'}, {'name': 'EEE', 'value': 'FALSE', 'label': 'Eee', 'description': 'Include the EEE model (ellipsoidal, equal volume, shape, and orientation) when calculating BIC.'}, {'name': 'EEV', 'value': 'FALSE', 'label': 'Eev', 'description': 'Include the EEV model (ellipsoidal, equal volume and equal shape) when calculating BIC.'}, {'name': 'VEV', 'value': 'FALSE', 'label': 'Vev', 'description': 'Include the VEV model (ellipsoidal, equal shape) when calculating BIC.'}, {'name': 'VVV', 'value': 'FALSE', 'label': 'Vvv', 'description': 'Include the VVV model (ellipsoidal, varying volume, shape, and orientation) when calculating BIC.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'MergeColumns': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Merge datasets by column.', 'required_args': [{'name': 'input.filename1', 'value': '', 'label': 'Input.Filename1', 'description': 'first file to merge'}, {'name': 'input.filename2', 'value': '', 'label': 'Input.Filename2', 'description': 'second file to merge'}, {'name': 'zip.filename', 'value': '', 'label': 'Zip.Filename', 'description': 'zip file containing files to merge. Either a zip file or the input files above must be provided.'}], 'default_args': [{'name': 'task_name', 'value': 'MergeColumns', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output', 'value': '<input.filename1_basename><zip.filename_basename>.merged', 'label': 'Output', 'description': 'name of the merged file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'MergeFCSDataFiles': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Merge multiple FCS data files into a single FCS datasets; includes multimple subsampling options.', 'required_args': [{'name': 'Input.files', 'value': '', 'label': 'Input.Files', 'description': 'The input FCS files to be merged; either a zip file or a pointer to a directory or a text file with FCS file names (full path) with one filename per line.'}, {'name': 'Keep.n.events', 'value': '', 'label': 'Keep.N.Events', 'description': 'Keep approximately n events from each dataset, optional, cannot be combined with subsampling rate.'}], 'default_args': [{'name': 'task_name', 'value': 'MergeFCSDataFiles', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Subsampling.rate', 'value': '100', 'label': 'Subsampling.Rate', 'description': 'Subsampling rate as percentage of events to be included from each data set, value between 0 and 100, optional, default 100%, cannot be combined with keep n events.'}, {'name': 'Random.seed', 'value': '1234567', 'label': 'Random.Seed', 'description': 'The random seed'}, {'name': 'Output.file.name', 'value': 'MergedFCSDataFiles.fcs', 'label': 'Output.File.Name', 'description': 'The output FCS file name.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'MergeHTSeqCounts': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Merges HTSeq read count data files into one file', 'required_args': [{'name': 'input.files', 'value': '', 'label': 'Input.Files', 'description': 'A directory or one or more text files containing HTSeq read counts'}], 'default_args': [{'name': 'task_name', 'value': 'MergeHTSeqCounts', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.prefix', 'value': '<input.files_basename>', 'label': 'Output.Prefix', 'description': 'The base name of the output file. File extensions will be added automatically.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'MergeRows': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Merge datasets by row.', 'required_args': [{'name': 'input.filename1', 'value': '', 'label': 'Input.Filename1', 'description': 'first file to merge'}, {'name': 'input.filename2', 'value': '', 'label': 'Input.Filename2', 'description': 'second file to merge'}, {'name': 'zip.filename', 'value': '', 'label': 'Zip.Filename', 'description': 'zip file containing files to merge. Either a zip file or the input files above must be provided.'}], 'default_args': [{'name': 'task_name', 'value': 'MergeRows', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output', 'value': '<input.filename1_basename><zip.filename_basename>.merged', 'label': 'Output', 'description': 'name of the merged file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'MetageneProjection': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Projects one or more data sets onto the metagene representation of another data set', 'required_args': [{'name': 'model.gct.file', 'value': '', 'label': 'Model.Gct.File', 'description': 'The model .gct data file'}, {'name': 'model.cls.file', 'value': '', 'label': 'Model.Cls.File', 'description': 'The model .cls data file'}, {'name': 'model.preprocessing.file', 'value': '', 'label': 'Model.Preprocessing.File', 'description': 'The pre-processing parameters for the model dataset'}, {'name': 'test.gct.file', 'value': '', 'label': 'Test.Gct.File', 'description': 'The test .gct or zip of .gct data files'}, {'name': 'test.cls.file', 'value': '', 'label': 'Test.Cls.File', 'description': 'The test .cls or zip of .cls data files'}, {'name': 'test.preprocessing.file', 'value': '', 'label': 'Test.Preprocessing.File', 'description': 'The pre-processing parameters for the test dataset'}, {'name': 'phenotype.plotting.colors', 'value': '', 'label': 'Phenotype.Plotting.Colors', 'description': 'Plotting colors for phenotypes'}, {'name': 'phenotype.plotting.symbols', 'value': '', 'label': 'Phenotype.Plotting.Symbols', 'description': 'Plotting symbols for phenotypes'}], 'default_args': [{'name': 'task_name', 'value': 'MetageneProjection', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'num.characters', 'value': '9', 'label': 'Num.Characters', 'description': 'Number of characters to use to identify classes from the .cls files'}, {'name': 'identifier', 'value': '<model.gct.file_basename>.output', 'label': 'Identifier', 'description': 'Prefix to name output files'}, {'name': 'k.projection', 'value': '3', 'label': 'K.Projection', 'description': 'Number of metagenes in projection'}, {'name': 'algorithm', 'value': 'NMF.div', 'label': 'Algorithm', 'description': 'Algorithm for Metagene Projection'}, {'name': 'number.of.iterations', 'value': '2000', 'label': 'Number.Of.Iterations', 'description': 'Number of algorithm iterations'}, {'name': 'seed', 'value': '1234', 'label': 'Seed', 'description': 'Random seed to initialize metagene matrices (Note: results may only be reproducible on the same platform)'}, {'name': 'post.projection.normalization', 'value': 'yes', 'label': 'Post.Projection.Normalization', 'description': 'apply post-projection normalization'}, {'name': 'heatmap.row.norm', 'value': 'no', 'label': 'Heatmap.Row.Norm', 'description': 'row-normalize (standardize) the rows in the heat map'}, {'name': 'heatmap.color.scheme', 'value': 'reddish color map', 'label': 'Heatmap.Color.Scheme', 'description': ''}, {'name': 'confidence.threshold', 'value': '0.3', 'label': 'Confidence.Threshold', 'description': 'Confidence threshold (Brier score) to separate call from no-calls'}, {'name': 'symbol.scaling', 'value': '1', 'label': 'Symbol.Scaling', 'description': 'Graphical scaling for symbols in plots and plot legends'}, {'name': 'kernel', 'value': 'radial', 'label': 'Kernel', 'description': 'Kernel function for SVM'}, {'name': 'cost', 'value': '1', 'label': 'Cost', 'description': 'cost function for SVM'}, {'name': 'gamma', 'value': '0.05', 'label': 'Gamma', 'description': 'Gamma coefficient for radial base function kernel'}, {'name': 'theta', 'value': '0', 'label': 'Theta', 'description': 'Smoothing parameter only used for the NSNMF.div algorithm'}, {'name': 'lambda', 'value': '0', 'label': 'Lambda', 'description': 'Sparse parameter used only for the SNMF algorithm'}, {'name': 'model.set.refinement', 'value': 'yes', 'label': 'Model.Set.Refinement', 'description': 'perform model set refinement'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'MINDY': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Runs the MINDY algorithm for inferring genes that modulate the activity of a transcription factor at post-transcriptional levels ', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'The dataset file - .res, .gct'}, {'name': 'modulator.list', 'value': '', 'label': 'Modulator.List', 'description': 'A list of possible modulator genes to analyze - .txt'}, {'name': 'dpi.transcription.factor.list', 'value': '', 'label': 'Dpi.Transcription.Factor.List', 'description': 'List of transcription factors in dataset - .txt'}, {'name': 'transcription.factor', 'value': '', 'label': 'Transcription.Factor', 'description': 'Transcription factor'}], 'default_args': [{'name': 'task_name', 'value': 'MINDY', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'sample.percentage', 'value': '35.0', 'label': 'Sample.Percentage', 'description': 'Percentage of samples to use in analysis (must be <=50)'}, {'name': 'conditional.mi.threshold', 'value': '0.0', 'label': 'Conditional.Mi.Threshold', 'description': 'Conditional mutual information (MI) score threshold (number between 0 and 1)'}, {'name': 'conditional.mi.p.value', 'value': '1.0', 'label': 'Conditional.Mi.P.Value', 'description': 'Conditional mutual info p-value threshold (ignored if conditonal MI threshold >0)'}, {'name': 'dpi.tolerance', 'value': '1.0', 'label': 'Dpi.Tolerance', 'description': 'The data processing inequality tolerance (number between 0 and 1)'}, {'name': 'unconditional.mi.threshold', 'value': '0.0', 'label': 'Unconditional.Mi.Threshold', 'description': 'Unconditional mutual information (MI) score threshold (number between 0 and 1)'}, {'name': 'unconditional.mi.p.value', 'value': '1.0', 'label': 'Unconditional.Mi.P.Value', 'description': 'Unconditional mutual information p-value threshold (ignored if unconditonal MI threshold >0)'}, {'name': 'output.file', 'value': '<input.file_basename>.mindy', 'label': 'Output.File', 'description': 'Output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'MINDYViewer': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Visualizes the results of MINDY', 'required_args': [{'name': 'dataset.file', 'value': '', 'label': 'Dataset.File', 'description': 'The dataset file - .res, .gct'}, {'name': 'mindy.file', 'value': '', 'label': 'Mindy.File', 'description': 'Output from MINDY- .mindy'}], 'default_args': [{'name': 'task_name', 'value': 'MINDYViewer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ModEcocTrainTest': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Modified error correcting output code prediction', 'required_args': [{'name': 'train.file', 'value': '', 'label': 'Train.File', 'description': 'The training features and class label information in .arff format'}, {'name': 'test.file', 'value': '', 'label': 'Test.File', 'description': 'The testing features and possible class label information in .arff format'}, {'name': 'saved.model.file', 'value': '', 'label': 'Saved.Model.File', 'description': 'Input modEcocTrainTest model'}], 'default_args': [{'name': 'task_name', 'value': 'ModEcocTrainTest', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'model.output.file', 'value': '<train.file_basename>.model', 'label': 'Model.Output.File', 'description': 'Output modEcocTrainTest model'}, {'name': 'base.classifier', 'value': '0', 'label': 'Base.Classifier', 'description': 'The name of the base classifier to use'}, {'name': 'base.multiplier', 'value': '4', 'label': 'Base.Multiplier', 'description': 'The base classifier multiplier'}, {'name': 'pred.results.output.file', 'value': '<test.file_basename>', 'label': 'Pred.Results.Output.File', 'description': 'The name of the output file for prediction results.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ModEcocXValidation': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Modified error correcting output code cross validation prediction', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'The input features and class label information in .arff format'}], 'default_args': [{'name': 'task_name', 'value': 'ModEcocXValidation', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'num.folds', 'value': '2', 'label': 'Num.Folds', 'description': 'The number of folds should be an integer greater than or equal to two.'}, {'name': 'base.classifier', 'value': '0', 'label': 'Base.Classifier', 'description': 'The name of the base classifier to use'}, {'name': 'base.multiplier', 'value': '4', 'label': 'Base.Multiplier', 'description': 'The base classifier multiplier'}, {'name': 'pred.results.file', 'value': '<input.file_basename>', 'label': 'Pred.Results.File', 'description': 'The name of the output file for prediction results.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Multiplot': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Creates customizable plots of expression data-derived data.', 'required_args': [{'name': 'input.file.name', 'value': '', 'label': 'Input.File.Name', 'description': 'File created by MultiplotPreprocess (.zip)'}], 'default_args': [{'name': 'task_name', 'value': 'Multiplot', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'number.of.plots', 'value': '1', 'label': 'Number.Of.Plots', 'description': 'Smaller one-plot visualizer or bigger two-plot'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'MultiplotExtractor': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Creates customizable plots of expression data-derived data.', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'File created by MultiplotPreprocess (.zip)'}], 'default_args': [{'name': 'task_name', 'value': 'MultiplotExtractor', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'MultiplotPreprocess': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Creates derived data from an expression dataset for use in the Multiplot Visualizer module', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'File with data to prepare for plotting (gct)'}, {'name': 'class.filename', 'value': '', 'label': 'Class.Filename', 'description': 'Class file for this data (cls)'}], 'default_args': [{'name': 'task_name', 'value': 'MultiplotPreprocess', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<input.filename_basename>.zip', 'label': 'Output.File', 'description': 'Name of file to create for use in Multiplot visualizer'}, {'name': 'replicateFC', 'value': '0', 'label': 'ReplicateFC', 'description': 'Whether to compute fold changes between each replicate (turning this on can slow your work).'}, {'name': 'randomData', 'value': '0', 'label': 'RandomData', 'description': 'Whether to create random expression data (and fold changes, etc for this data). Turning this on can slow your work'}, {'name': 'outlierElimination', 'value': '0', 'label': 'OutlierElimination', 'description': 'Remove outlier replicate values of genes from processing (see \"Help\" for details)'}, {'name': 'doStats', 'value': '1', 'label': 'DoStats', 'description': 'Perform t-test, Hochberg p-value adjustment, and significance comparison to randomized data'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'MultiplotStudio': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Scatterplot visualization tool for analyzing differential gene expression.', 'required_args': [{'name': 'dataFile', 'value': '', 'label': 'DataFile', 'description': 'GenePattern dataset (*.gct), Multiplot Studio Workspace file (*.msdata) or CSV (see help for format requirements).'}, {'name': 'classFile', 'value': '', 'label': 'ClassFile', 'description': 'Class file (*.cls). Not required, but lack of a class file will result in fewer calculation options. Ignored if dataFile is a Workspace.'}], 'default_args': [{'name': 'task_name', 'value': 'MultiplotStudio', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'MutSigCV': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Identifies genes that are significantly mutated in cancer genomes, using a model with mutational covariates.', 'required_args': [{'name': 'mutation.table.file', 'value': '', 'label': 'Mutation.Table.File', 'description': 'Mutation list in MAF format.'}, {'name': 'coverage.table.file', 'value': '', 'label': 'Coverage.Table.File', 'description': 'File containing the coverage table.  This should be a file with per-sample coverage as produced by MuTect from the samples\\' WIG files.  If you do not have this level of coverage information, you can instead use the territory file for the reference human exome to direct MutSigCV to assume full coverage.  This file is found using \"Add Path or URL\" under \"shared_data/example_files/MutSigCV_1.3/exome_full192.coverage.txt\".'}, {'name': 'mutation.type.dictionary', 'value': '', 'label': 'Mutation.Type.Dictionary', 'description': 'The mutation type dictionary to use for automatic category and effect discovery.  This is necessary only if you are using a MAF file without the columns \"categ\" and \"effect\".  A suggested dictionary can be found using \"Add Path or URL\" under \"shared_data/example_files/MutSigCV_1.3/mutation_type_dictionary_file.txt\".'}, {'name': 'genome.build', 'value': '', 'label': 'Genome.Build', 'description': 'Genome build to use for automatic category and effect discovery.  This is necessary only if you are using a MAF file without the columns \"categ\" and \"effect\".'}], 'default_args': [{'name': 'task_name', 'value': 'MutSigCV', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'covariates.table.file', 'value': 'ftp://ftp.broadinstitute.org/pub/genepattern/example_files/MutSigCV_1.3/gene.covariates.txt', 'label': 'Covariates.Table.File', 'description': 'File containing the covariates table.  The default selection is the covariates file used in the MutSigCV paper published in Nature (see the documentation for further information and a full reference). '}, {'name': 'output.filename.base', 'value': '<mutation.table.file_basename>', 'label': 'Output.Filename.Base', 'description': 'Base name for the output files.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'mzXMLToCsv': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Converts a mzXML file to a zip of csv files', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'input filename - .mzXML'}], 'default_args': [{'name': 'task_name', 'value': 'mzXMLToCsv', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<input.filename_basename>.zip', 'label': 'Output.File', 'description': 'output file containing zip of csv files'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'NearestTemplatePrediction': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Nearest neighbor prediction based on a list of marker genes', 'required_args': [{'name': 'input.exp.filename', 'value': '', 'label': 'Input.Exp.Filename', 'description': 'Gene expression data set (.gct)'}, {'name': 'input.features.filename', 'value': '', 'label': 'Input.Features.Filename', 'description': 'List of marker genes (.txt): Probe ID, Gene name, Class (1,2,...), Weight (optional)'}], 'default_args': [{'name': 'task_name', 'value': 'NearestTemplatePrediction', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.name', 'value': 'NTP', 'label': 'Output.Name', 'description': 'Name for output files'}, {'name': 'distance.selection', 'value': 'cosine', 'label': 'Distance.Selection', 'description': 'Distance metric'}, {'name': 'weight.genes', 'value': 'T', 'label': 'Weight.Genes', 'description': 'Weight genes? (by statistic, fold change, etc. only for 2 classes)'}, {'name': 'num.resamplings', 'value': '1000', 'label': 'Num.Resamplings', 'description': '# resampling to generate null distribution for distance metric'}, {'name': 'GenePattern.output', 'value': 'F', 'label': 'GenePattern.Output', 'description': 'Create .gct and .cls files for GenePattern'}, {'name': 'random.seed', 'value': '7392854', 'label': 'Random.Seed', 'description': 'Random seed'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'NMF': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Non-negative Matrix Factorization', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'Input file in GCT format with input data (V matrix) (GCT format only). e.g. ALL_vs_AML.gct'}], 'default_args': [{'name': 'task_name', 'value': 'NMF', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<input.filename_basename>.log', 'label': 'Output.File', 'description': 'Output log file. e.g. NMF.log'}, {'name': 'nfact', 'value': '4', 'label': 'Nfact', 'description': 'Number of NMF factors. e.g. 2, 4, 10'}, {'name': 'niter', 'value': '100', 'label': 'Niter', 'description': 'Number of NMF iterations. e.g. 100, 250, 500'}, {'name': 'print.dump', 'value': '0', 'label': 'Print.Dump', 'description': 'Print matrices in output for debug'}, {'name': 'cnorm.flag', 'value': '0', 'label': 'Cnorm.Flag', 'description': 'Column normalization flag'}, {'name': 'rnorm.flag', 'value': '0', 'label': 'Rnorm.Flag', 'description': 'Row normalization flag'}, {'name': 'gshift', 'value': '0', 'label': 'Gshift', 'description': 'Global additive shift after normalization. e.g. 1, 4'}, {'name': 'norm.type', 'value': '0', 'label': 'Norm.Type', 'description': 'Type of normalization'}, {'name': 'seed', 'value': '12345', 'label': 'Seed', 'description': 'Random seed used to initialize W and H matrices by the randon number generator. e.g. 4585, 4567, 5980'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'NMFConsensus': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Non-negative Matrix Factorization (NMF) Consensus Clustering', 'required_args': [{'name': 'dataset.filename', 'value': '', 'label': 'Dataset.Filename', 'description': 'Input dataset (gct or res)'}], 'default_args': [{'name': 'task_name', 'value': 'NMFConsensus', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'k.initial', 'value': '2', 'label': 'K.Initial', 'description': 'Initial value of K'}, {'name': 'k.final', 'value': '5', 'label': 'K.Final', 'description': 'Final value of K'}, {'name': 'num.clusterings', 'value': '20', 'label': 'Num.Clusterings', 'description': 'Number of clusterings per value of K'}, {'name': 'max.num.iterations', 'value': '2000', 'label': 'Max.Num.Iterations', 'description': 'Maximum number of NMF iterations per clustering'}, {'name': 'error.function', 'value': 'divergence', 'label': 'Error.Function', 'description': 'Error function'}, {'name': 'random.seed', 'value': '123456789', 'label': 'Random.Seed', 'description': 'Seed for random number generator'}, {'name': 'output.file.prefix', 'value': '<dataset.filename_basename>', 'label': 'Output.File.Prefix', 'description': 'Prefix to append to all output files'}, {'name': 'stop.convergence', 'value': '40', 'label': 'Stop.Convergence', 'description': 'How many \"no change\" checks are needed to stop NMF iterations before max iterations is reached(convergence)'}, {'name': 'stop.frequency', 'value': '10', 'label': 'Stop.Frequency', 'description': 'Frequency (NMF iterations) of \"no change\" checks'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'NMFConsensus_combine_lewu': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Non-negative Matrix Factorization (NMF) Consensus Clustering (combine single k result)', 'required_args': [{'name': 'param.filename', 'value': '', 'label': 'Param.Filename', 'description': 'Input param file '}], 'default_args': [{'name': 'task_name', 'value': 'NMFConsensus_combine_lewu', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'NMFConsensus_sk_lewu': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Non-negative Matrix Factorization (NMF) Consensus Clustering', 'required_args': [{'name': 'param.filename', 'value': '', 'label': 'Param.Filename', 'description': 'Input param file '}], 'default_args': [{'name': 'task_name', 'value': 'NMFConsensus_sk_lewu', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'NMFPartition': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'The module will output a list of parameter files', 'required_args': [{'name': 'dataset.filename', 'value': '', 'label': 'Dataset.Filename', 'description': 'Input dataset (gct or res)'}], 'default_args': [{'name': 'task_name', 'value': 'NMFPartition', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'k.initial', 'value': '2', 'label': 'K.Initial', 'description': 'Initial value of K'}, {'name': 'k.final', 'value': '5', 'label': 'K.Final', 'description': 'Final value of K'}, {'name': 'num.clusterings', 'value': '20', 'label': 'Num.Clusterings', 'description': 'Number of clusterings per value of K'}, {'name': 'max.num.iterations', 'value': '2000', 'label': 'Max.Num.Iterations', 'description': 'Maximum number of NMF iterations per clustering'}, {'name': 'error.function', 'value': 'divergence', 'label': 'Error.Function', 'description': 'Error function'}, {'name': 'random.seed', 'value': '123456789', 'label': 'Random.Seed', 'description': 'Seed for random number generator'}, {'name': 'output.file.prefix', 'value': '<dataset.filename_basename>', 'label': 'Output.File.Prefix', 'description': 'Prefix to append to all output files'}, {'name': 'stop.convergence', 'value': '40', 'label': 'Stop.Convergence', 'description': 'How many \"no change\" checks are needed to stop NMF iterations before max iterations is reached(convergence)'}, {'name': 'stop.frequency', 'value': '10', 'label': 'Stop.Frequency', 'description': 'Frequency (NMF iterations) of \"no change\" checks'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'NMFSingleKPipe': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '', 'required_args': [{'name': 'NMFConsensus_sk_lewu1.param.filename', 'value': '', 'label': 'NMFConsensus Sk Lewu1.Param.Filename', 'description': 'Input param file '}], 'default_args': [{'name': 'task_name', 'value': 'NMFSingleKPipe', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'NormLines': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Normalizes RNAi hairpin scores across cell lines using ZMAD, PMAD, rank or quantile- ignores NaNs', 'required_args': [{'name': 'gct.file', 'value': '', 'label': 'Gct.File', 'description': 'Input gct file to normalize by samples'}], 'default_args': [{'name': 'task_name', 'value': 'NormLines', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'method', 'value': 'quantile', 'label': 'Method', 'description': 'Options are ZMAD, PMAD, quantile'}, {'name': 'plotfile', 'value': 'NormPDF.pdf', 'label': 'Plotfile', 'description': ''}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ParallelCBS': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Runs CBS algorithm on multiple samples in parallel', 'required_args': [{'name': 'SplitColumns1.input.file', 'value': '', 'label': 'SplitColumns1.Input.File', 'description': 'A tab delimited file'}], 'default_args': [{'name': 'task_name', 'value': 'ParallelCBS', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'ConcatenateFilelist4.output.filename', 'value': '<filelist_basename>.catted.seg.txt', 'label': 'ConcatenateFilelist4.Output.Filename', 'description': 'The name of the output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ParallelNMFConsensusPipe': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'A parallelized implementation of the NMFConsensus module, implemented as a GenePattern scatter-gather pipeline.', 'required_args': [{'name': 'NMFPartition1.dataset.filename', 'value': '', 'label': 'NMFPartition1.Dataset.Filename', 'description': 'Input dataset (gct or res)'}], 'default_args': [{'name': 'task_name', 'value': 'ParallelNMFConsensusPipe', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'NMFPartition1.k.initial', 'value': '2', 'label': 'NMFPartition1.K.Initial', 'description': 'Initial value of K'}, {'name': 'NMFPartition1.k.final', 'value': '5', 'label': 'NMFPartition1.K.Final', 'description': 'Final value of K'}, {'name': 'NMFPartition1.num.clusterings', 'value': '20', 'label': 'NMFPartition1.Num.Clusterings', 'description': 'Number of clusterings per value of K'}, {'name': 'NMFPartition1.max.num.iterations', 'value': '2000', 'label': 'NMFPartition1.Max.Num.Iterations', 'description': 'Maximum number of NMF iterations per clustering'}, {'name': 'NMFPartition1.error.function', 'value': 'divergence', 'label': 'NMFPartition1.Error.Function', 'description': 'Error function'}, {'name': 'NMFPartition1.random.seed', 'value': '123456789', 'label': 'NMFPartition1.Random.Seed', 'description': 'Seed for random number generator'}, {'name': 'NMFPartition1.output.file.prefix', 'value': '<dataset.filename_basename>', 'label': 'NMFPartition1.Output.File.Prefix', 'description': 'Prefix to append to all output files'}, {'name': 'NMFPartition1.stop.convergence', 'value': '40', 'label': 'NMFPartition1.Stop.Convergence', 'description': 'How many \"no change\" checks are needed to stop NMF iterations before max iterations is reached(convergence)'}, {'name': 'NMFPartition1.stop.frequency', 'value': '10', 'label': 'NMFPartition1.Stop.Frequency', 'description': 'Frequency (NMF iterations) of \"no change\" checks'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'PARIS': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Feature selection by class or against a continuous vector (Probablility Analysis by Ranked Information Score)', 'required_args': [{'name': 'target.profile', 'value': '', 'label': 'Target.Profile', 'description': 'Target profile file (either .cls or .gct).  Used to classify cell line samples into groups, or assign a continuous target vector to the cell line samples.'}, {'name': 'target', 'value': '', 'label': 'Target', 'description': \"Target gene or ID from target.profile file.  Should match a row in the 'Name' or 'Description' column of a .gct file, or match one label in the 2nd row of the .cls file\"}, {'name': 'data', 'value': '', 'label': 'Data', 'description': 'Data (feature) file to rank based on target profile'}, {'name': 'prefix', 'value': '', 'label': 'Prefix', 'description': 'prefix for labeling output files'}, {'name': 'advanced', 'value': '', 'label': 'Advanced', 'description': 'Leave blank for default.  This parameter allows the user to change the underlying default metric.  Options will be listed in the documentation.'}], 'default_args': [{'name': 'task_name', 'value': 'PARIS', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'permtype', 'value': 'standard', 'label': 'Permtype', 'description': 'standard; subclass.stratified (lineage-corrected)'}, {'name': 'type.flag', 'value': 'gene', 'label': 'Type.Flag', 'description': 'perturbagen level data (shRNA) or gene level scores (gene)'}, {'name': 'quality', 'value': 'normal', 'label': 'Quality', 'description': 'controls how many permutations are run; draft will take less time'}, {'name': 'direction', 'value': 'positive', 'label': 'Direction', 'description': 'Direction of feature matching; positive means high values to high values, low to low; negative means high to low values'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'PCA': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Principal Component Analysis (adapted from MeV, http://www.tm4.org/m)', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'input file - .gct, .res, .odf'}], 'default_args': [{'name': 'task_name', 'value': 'PCA', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'cluster.by', 'value': '3', 'label': 'Cluster.By', 'description': 'Cluster by rows or columns'}, {'name': 'output.file', 'value': '<input.filename_basename>', 'label': 'Output.File', 'description': 'stub name of the output files'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'PCAViewer': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Principal Component Analysis Visualizer (adapted from MeV, http://www.tigr.org/software/tm4/mev.html)', 'required_args': [{'name': 'dataset.file', 'value': '', 'label': 'Dataset.File', 'description': 'The dataset - .res, .gct'}, {'name': 's.matrix.file', 'value': '', 'label': 'S.Matrix.File', 'description': 'The s matrix - .odf'}, {'name': 't.matrix.file', 'value': '', 'label': 'T.Matrix.File', 'description': 'The t matrix - .odf'}, {'name': 'u.matrix.file', 'value': '', 'label': 'U.Matrix.File', 'description': 'The u matrix - .odf'}, {'name': 'cls.or.sample.info.file', 'value': '', 'label': 'Cls.Or.Sample.Info.File', 'description': 'A cls or sample info file - .cls, .txt'}, {'name': 'sample.info.class', 'value': '', 'label': 'Sample.Info.Class', 'description': 'The name of the class column to use in the sample info file (required if sample info file provided)'}, {'name': 'featureset.file', 'value': '', 'label': 'Featureset.File', 'description': 'The feature set file - .gmt, .gmx, .grp'}], 'default_args': [{'name': 'task_name', 'value': 'PCAViewer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'PclToGct': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Convert Stanford preclustering file (.pcl) to .gct', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'pcl file to convert'}], 'default_args': [{'name': 'task_name', 'value': 'PclToGct', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.filename', 'value': '<input.filename_basename>.gct', 'label': 'Output.Filename', 'description': 'output gct file'}, {'name': 'weight.handling', 'value': '3', 'label': 'Weight.Handling', 'description': 'How to handle manually edited weights (weights not equal to 1) present in PCL file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'PeakMatch': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Perform peak matching on LC-MS data', 'required_args': [{'name': 'peakList.filename', 'value': '', 'label': 'PeakList.Filename', 'description': 'Zip file containing peak lists'}, {'name': 'sampleInfo.filename', 'value': '', 'label': 'SampleInfo.Filename', 'description': 'CSV file with header: experiment, sample, class'}, {'name': 'LandmarkMatchOutput.filename', 'value': '', 'label': 'LandmarkMatchOutput.Filename', 'description': 'Zip file containing the output from landmark matching.'}, {'name': 'MZ.tolerance', 'value': '', 'label': 'MZ.Tolerance', 'description': 'm/z tolerance (ppm). Used if landmark match output is not supplied. Defaults to 10 ppm.'}, {'name': 'RT.tolerance', 'value': '', 'label': 'RT.Tolerance', 'description': 'retention time tolerance (min). Used if landmark match is not supplied. Defaults to 2 min.'}, {'name': 'numberProcesses', 'value': '', 'label': 'NumberProcesses', 'description': 'Number of processess when running in parallel. Defaults to 1 (sequential).'}], 'default_args': [{'name': 'task_name', 'value': 'PeakMatch', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'outputName', 'value': 'PeakMatchOutput', 'label': 'OutputName', 'description': 'File name prefix for the output files'}, {'name': 'running.on.LSF', 'value': 'no', 'label': 'Running.On.LSF', 'description': 'Whether you are running on an LSF cluster version of GenePattern'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Peaks': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Determine peaks in the spectrum using a series of digital filters.', 'required_args': [{'name': 'spectrum.filename', 'value': '', 'label': 'Spectrum.Filename', 'description': 'spectrum data file - .csv'}, {'name': 'low.Da', 'value': '', 'label': 'Low.Da', 'description': 'Minimum M/Z to include'}, {'name': 'high.Da', 'value': '', 'label': 'High.Da', 'description': 'Maximum M/Z to include'}], 'default_args': [{'name': 'task_name', 'value': 'Peaks', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'percentile', 'value': '0.65', 'label': 'Percentile', 'description': 'After applying filters, threshold above which peaks are located'}, {'name': 'smoothing.size', 'value': '21', 'label': 'Smoothing.Size', 'description': 'Size of the smoothing filter'}, {'name': 'adaptive.background.correction.strength', 'value': '0.75', 'label': 'Adaptive.Background.Correction.Strength', 'description': 'Strength of the adaptive background correction filter'}, {'name': 'adaptive.background.correction.size', 'value': '21', 'label': 'Adaptive.Background.Correction.Size', 'description': 'Size of the adaptive background correction filter'}, {'name': 'high.pass.filter.strength', 'value': '10', 'label': 'High.Pass.Filter.Strength', 'description': 'Strength of the high pass filter'}, {'name': 'high.pass.filter.factor', 'value': '5', 'label': 'High.Pass.Filter.Factor', 'description': 'Filter factor for the high pass filter'}, {'name': 'output.file', 'value': '<spectrum.filename_basename>.peaks.odf', 'label': 'Output.File', 'description': 'output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Picard.AddOrReplaceReadGroups': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Replaces all read groups in the input file with a new read group and assigns all reads to this read group in the output', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'Input file (BAM or SAM).'}, {'name': 'read.group.id', 'value': '', 'label': 'Read.Group.Id', 'description': 'Read Group ID'}, {'name': 'read.group.library', 'value': '', 'label': 'Read.Group.Library', 'description': 'DNA preparation library. Note that this field is required by the module Picard.MarkDuplicates. It is used to determine which read groups may contain molecular duplicates, in the case where the same DNA library was sequenced on multiple lanes.'}, {'name': 'read.group.platform', 'value': '', 'label': 'Read.Group.Platform', 'description': 'Read Group platform (e.g., Illumina, SOLID).'}, {'name': 'read.group.platform.unit', 'value': '', 'label': 'Read.Group.Platform.Unit', 'description': 'Read Group platform unit (e.g., run barcode).'}, {'name': 'read.group.sample.name', 'value': '', 'label': 'Read.Group.Sample.Name', 'description': 'Read Group sample name.'}, {'name': 'read.group.sequencing.center.name', 'value': '', 'label': 'Read.Group.Sequencing.Center.Name', 'description': 'Read Group sequencing center name.'}, {'name': 'read.group.description', 'value': '', 'label': 'Read.Group.Description', 'description': 'Read Group description.'}], 'default_args': [{'name': 'task_name', 'value': 'Picard.AddOrReplaceReadGroups', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.prefix', 'value': '<input.file_basename>', 'label': 'Output.Prefix', 'description': 'A label used to name the output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Picard.BamToSam': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Converts a BAM file to a SAM file', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'A BAM file'}], 'default_args': [{'name': 'task_name', 'value': 'Picard.BamToSam', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.prefix', 'value': '<input.file_basename>', 'label': 'Output.Prefix', 'description': 'A label used to name the output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Picard.CreateSequenceDictionary': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Reads FASTA or FASTA.GZ files containing reference sequences, and writes them as a SAM file containing a sequence dictionary', 'required_args': [{'name': 'reference.sequence.file', 'value': '', 'label': 'Reference.Sequence.File', 'description': 'Input reference fasta or fasta.gz '}, {'name': 'genome.assembly', 'value': '', 'label': 'Genome.Assembly', 'description': 'The genome assembly to put into AS field of sequence dictionary entry'}, {'name': 'uri', 'value': '', 'label': 'Uri', 'description': 'The uri to put into UR field of sequence dictionary entry'}, {'name': 'num.sequences', 'value': '', 'label': 'Num.Sequences', 'description': 'Stop after writing this many sequences'}], 'default_args': [{'name': 'task_name', 'value': 'Picard.CreateSequenceDictionary', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'truncate.names.at.white.space', 'value': 'no', 'label': 'Truncate.Names.At.White.Space', 'description': 'Make sequence name the first word from the > line in the reference file'}, {'name': 'output.file', 'value': '<reference.sequence.file_basename>.dict', 'label': 'Output.File', 'description': 'The name of the output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Picard.FastqToSam': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Extracts read sequences and qualities from the input fastq file and writes them into the output file in unaligned SAM or BAM format', 'required_args': [{'name': 'fastq.file', 'value': '', 'label': 'Fastq.File', 'description': 'Input fastq file (optionally gzipped) for single end data, or first read in paired end data'}, {'name': 'fastq.file.2', 'value': '', 'label': 'Fastq.File.2', 'description': 'The fastq file (optionally gzipped) for second read in paired end data'}, {'name': 'read.group.name', 'value': '', 'label': 'Read.Group.Name', 'description': 'Read group name'}, {'name': 'sample.name', 'value': '', 'label': 'Sample.Name', 'description': 'Sample name to insert into the read group header'}, {'name': 'library.name', 'value': '', 'label': 'Library.Name', 'description': 'The library name to insert into the LB attribute in the read group header'}, {'name': 'platform.unit', 'value': '', 'label': 'Platform.Unit', 'description': 'The platform unit (often run_barcode.lane) to insert into the read group header'}, {'name': 'platform', 'value': '', 'label': 'Platform', 'description': 'The platform type (e.g., Illumina, SOLiD) to insert into the read group header'}], 'default_args': [{'name': 'task_name', 'value': 'Picard.FastqToSam', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'quality.format', 'value': 'Standard', 'label': 'Quality.Format', 'description': 'A value describing how the quality values are encoded in the fastq'}, {'name': 'sort.order', 'value': 'coordinate', 'label': 'Sort.Order', 'description': ''}, {'name': 'output.format', 'value': 'bam', 'label': 'Output.Format', 'description': 'Whether the file output format should be SAM or BAM'}, {'name': 'output.prefix', 'value': '<fastq.file_basename>', 'label': 'Output.Prefix', 'description': 'The prefix to use for the output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Picard.MarkDuplicates': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Examines aligned records in the supplied SAM or BAM file to locate duplicate reads.', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'A SAM or BAM file'}, {'name': 'read.name.regex', 'value': '', 'label': 'Read.Name.Regex', 'description': 'Regular expression that can be used to parse read names in the input file'}], 'default_args': [{'name': 'task_name', 'value': 'Picard.MarkDuplicates', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'remove.duplicates', 'value': 'false', 'label': 'Remove.Duplicates', 'description': 'Whether to remove duplicates'}, {'name': 'max.file.handles', 'value': '8000', 'label': 'Max.File.Handles', 'description': 'Maximum number of file handles to keep open when spilling read ends to disk'}, {'name': 'optical.duplicate.distance', 'value': '100', 'label': 'Optical.Duplicate.Distance', 'description': 'The maximum offset between two duplicate clusters in order to consider them optical duplicates. This should usually be set to some fairly small number (e.g. 5-10 pixels) unless using later versions of the Illumina pipeline that multiply pixel values by 10, in which case 50-100 is more normal'}, {'name': 'sorting.collection.size.ratio', 'value': '0.25', 'label': 'Sorting.Collection.Size.Ratio', 'description': 'This number, plus the maximum RAM available to the JVM, determine the memory footprint used by some of the sorting collections.  If you are running out of memory, try reducing this number'}, {'name': 'output.prefix', 'value': '<input.file_basename>', 'label': 'Output.Prefix', 'description': 'The prefix to use for the output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Picard.ReorderSam': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Reorders a SAM file or a BAM file to match contig ordering in a provided reference file', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'A SAM or BAM file.'}, {'name': 'reference.file', 'value': '', 'label': 'Reference.File', 'description': 'Reference sequence to which to reorder reads in FASTA format.'}, {'name': 'reference.sequence.dictionary', 'value': '', 'label': 'Reference.Sequence.Dictionary', 'description': 'The sequence dictionary for the reference sequence file'}], 'default_args': [{'name': 'task_name', 'value': 'Picard.ReorderSam', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'allow.partial.overlap', 'value': 'false', 'label': 'Allow.Partial.Overlap', 'description': 'If yes, then allows only a partial overlap of the BAM contigs with the new reference sequence contigs. This requires a corresponding contig in the new reference for each read contig.'}, {'name': 'allow.contig.length.discordance', 'value': 'false', 'label': 'Allow.Contig.Length.Discordance', 'description': 'If yes, then permits mapping from a read contig to a new reference contig with the same name but a different length. Use with extreme caution and a full knowledge of the possible consequences.'}, {'name': 'output.prefix', 'value': '<input.file_basename>', 'label': 'Output.Prefix', 'description': 'The prefix of the output SAM or BAM file.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Picard.SamToBam': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Converts a SAM file to a BAM file', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'A SAM file'}], 'default_args': [{'name': 'task_name', 'value': 'Picard.SamToBam', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.prefix', 'value': '<input.file_basename>', 'label': 'Output.Prefix', 'description': 'A label used to name the output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Picard.SamToFastq': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Convert SAM or BAM files to FASTQ format', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'The SAM or BAM file'}, {'name': 'clipping.attribute', 'value': '', 'label': 'Clipping.Attribute', 'description': 'Attribute that stores the position at which the SAM record should be clipped'}, {'name': 'clipping.action.quality.value', 'value': '', 'label': 'Clipping.Action.Quality.Value', 'description': 'Value to set base qualities to when clipping action is \"change base quality values\"'}, {'name': 'fastq.output.prefix.1', 'value': '', 'label': 'Fastq.Output.Prefix.1', 'description': 'The fastq output file name for single-end fastq or first end of a paired fastq. Only specify when \"per read group\" is \"no\".'}, {'name': 'fastq.output.prefix.2', 'value': '', 'label': 'Fastq.Output.Prefix.2', 'description': 'The second end of a paired fastq. Only specify when \"per read group\" is \"no\".'}], 'default_args': [{'name': 'task_name', 'value': 'Picard.SamToFastq', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'per.read.group', 'value': 'no', 'label': 'Per.Read.Group', 'description': 'Whether to output a fastq file per read group'}, {'name': 're-reverse.bases', 'value': 'no', 'label': 'Re-Reverse.Bases', 'description': 'Whether to re-reverse bases and qualities of reads with negative strand flag set'}, {'name': 'include.non.pf.reads', 'value': 'no', 'label': 'Include.Non.Pf.Reads', 'description': 'Whether to include non-PF reads'}, {'name': 'clipping.action', 'value': 'none', 'label': 'Clipping.Action', 'description': 'Action to take with clipped reads when a clipping attribute is specified'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Picard.SortSam': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Sorts a SAM or BAM file in a specified order', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'Input file in SAM or BAM format'}, {'name': 'output.format', 'value': '', 'label': 'Output.Format', 'description': 'Whether the file output format should be SAM or BAM'}], 'default_args': [{'name': 'task_name', 'value': 'Picard.SortSam', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'sort.order', 'value': 'coordinate', 'label': 'Sort.Order', 'description': 'Sort order'}, {'name': 'output.prefix', 'value': '<input.file_basename>', 'label': 'Output.Prefix', 'description': 'prefix for the name of output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'PlateQAFCS': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Look for consistency between wells of a single plate by computing summary statistics.', 'required_args': [{'name': 'Input.FCS.files', 'value': '', 'label': 'Input.FCS.Files', 'description': 'FCS files from a single plate.'}], 'default_args': [{'name': 'task_name', 'value': 'PlateQAFCS', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Check.cell.counts', 'value': '1', 'label': 'Check.Cell.Counts', 'description': 'Count cells in each file?'}, {'name': 'Check.medians', 'value': '1', 'label': 'Check.Medians', 'description': 'Report the median measured values?'}, {'name': 'Check.PDF', 'value': '1', 'label': 'Check.PDF', 'description': 'Check the Probability Density Functions (PDF)?'}, {'name': 'Check.ECDF', 'value': '1', 'label': 'Check.ECDF', 'description': 'Check the Empirical Cumulative Distribution Functions (ECDF)?'}, {'name': 'Check.boxplots', 'value': '1', 'label': 'Check.Boxplots', 'description': 'Detect unexpected data distributions (unusual upper/lower quantiles)?'}, {'name': 'Check.timeline', 'value': '1', 'label': 'Check.Timeline', 'description': 'Check for unusual patterns in the acquisition of fluorescence and light scatter measurements over time?'}, {'name': 'Check.timeflow', 'value': '1', 'label': 'Check.Timeflow', 'description': 'Check for disturbances in the flow of cells over time?'}, {'name': 'FSC.channel', 'value': 'FSC-H', 'label': 'FSC.Channel', 'description': 'The name of the forward scatter parameter (short name, value of the $PnN keyword).'}, {'name': 'SSC.channel', 'value': 'SSC-H', 'label': 'SSC.Channel', 'description': 'The name of the side scatter parameter (short name, value of the $PnN keyword).'}, {'name': 'Zip.output', 'value': 'TRUE', 'label': 'Zip.Output', 'description': 'Zip the resulting report together?'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'PlotPeaks': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Plot peaks', 'required_args': [{'name': 'peaks.output.filename', 'value': '', 'label': 'Peaks.Output.Filename', 'description': 'The output from the Peaks module'}], 'default_args': [{'name': 'task_name', 'value': 'PlotPeaks', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<peaks.output.filename_basename>.pdf', 'label': 'Output.File', 'description': 'output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'PredictionResultsViewer': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Visualizes prediction results', 'required_args': [{'name': 'prediction.results.file', 'value': '', 'label': 'Prediction.Results.File', 'description': 'Prediction results - .odf'}], 'default_args': [{'name': 'task_name', 'value': 'PredictionResultsViewer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Preprocess.and.quality.check.RNASeq.data': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'This pipeline is used in the GenomeSpace recipe, \"Preprocess and quality check RNA-Seq data\". The recipe can be found at: http://www.genomespace.org/support/guides/recipes/sections/preprocess-and-qc-rna-seq-data.', 'required_args': [{'name': 'ConvertLineEndings1.input.filename', 'value': '', 'label': 'ConvertLineEndings1.Input.Filename', 'description': 'The input file (any non-binary file format)'}, {'name': 'TopHat4.GTF.file', 'value': '', 'label': 'TopHat4.GTF.File', 'description': 'A GTF file (v. 2.2 or higher) or GFF3 file containing a list of gene model annotations. TopHat will first extract the transcript sequences and align them to this virtual transcriptome first. Only the reads that do not fully map to the transcriptome will then be mapped on the genome. The reads that did map on the transcriptome will be converted to genomic mappings (spliced as needed) and merged with the novel mappings.'}, {'name': 'TopHat4.reads.pair.1', 'value': '', 'label': 'TopHat4.Reads.Pair.1', 'description': 'Unpaired reads file or first mate for paired reads. One or more files containing reads in FASTA or FASTQ format (bz2 and gz compressed files are supported).'}], 'default_args': [{'name': 'task_name', 'value': 'Preprocess.and.quality.check.RNASeq.data', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'PreprocessDataset': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Performs several preprocessing steps on a res, gct, or odf input file', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'Input filename - .res, .gct, .odf'}, {'name': 'threshold.for.removing.rows', 'value': '', 'label': 'Threshold.For.Removing.Rows', 'description': 'Threshold for removing rows'}, {'name': 'number.of.columns.above.threshold', 'value': '', 'label': 'Number.Of.Columns.Above.Threshold', 'description': 'Remove row if this number of columns not >= given threshold'}], 'default_args': [{'name': 'task_name', 'value': 'PreprocessDataset', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'threshold.and.filter', 'value': '1', 'label': 'Threshold.And.Filter', 'description': 'Apply floor, ceiling and variation filter'}, {'name': 'floor', 'value': '20', 'label': 'Floor', 'description': 'Value for floor threshold'}, {'name': 'ceiling', 'value': '20000', 'label': 'Ceiling', 'description': 'Value for ceiling threshold'}, {'name': 'min.fold.change', 'value': '3', 'label': 'Min.Fold.Change', 'description': 'Minimum fold change for variation filter'}, {'name': 'min.delta', 'value': '100', 'label': 'Min.Delta', 'description': 'Minimum delta for variation filter'}, {'name': 'num.outliers.to.exclude', 'value': '0', 'label': 'Num.Outliers.To.Exclude', 'description': 'Number of outliers per row to ignore when calculating row min and max for variation filter'}, {'name': 'row.normalization', 'value': '0', 'label': 'Row.Normalization', 'description': 'Perform row normalization'}, {'name': 'row.sampling.rate', 'value': '1', 'label': 'Row.Sampling.Rate', 'description': 'Sample rows without replacement to obtain this fraction of the total number of rows'}, {'name': 'log2.transform', 'value': '0', 'label': 'Log2.Transform', 'description': 'Apply log2 transform after all other preprocessing steps'}, {'name': 'output.file.format', 'value': '3', 'label': 'Output.File.Format', 'description': 'Output file format'}, {'name': 'output.file', 'value': '<input.filename_basename>.preprocessed', 'label': 'Output.File', 'description': 'Output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'PreprocessReadCounts': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Preprocess RNA-Seq count data in a GCT file so that it is suitable for use in GenePattern analyses.', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'A GCT file containing raw RNA-Seq counts, such as is produced by MergeHTSeqCounts'}, {'name': 'cls.file', 'value': '', 'label': 'Cls.File', 'description': 'A categorical CLS file specifying the phenotype classes for the samples in the GCT file.  '}], 'default_args': [{'name': 'task_name', 'value': 'PreprocessReadCounts', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<input.file_basename>.preprocessed.gct', 'label': 'Output.File', 'description': 'Output file name'}, {'name': 'expression.value.filter.threshold', 'value': '1', 'label': 'Expression.Value.Filter.Threshold', 'description': 'Threshold to use when filtering CPM expression values; rows are kept only if the values (in CPM) for all columns are greater than this threshold'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'PreviewFCS': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Allows viewing of structural metadata, parameters, and descriptive statistics from a Flow Cytometry Standard (FCS) data file', 'required_args': [{'name': 'Input.FCS.data.file', 'value': '', 'label': 'Input.FCS.Data.File', 'description': 'An input FCS data file to be described.'}], 'default_args': [{'name': 'task_name', 'value': 'PreviewFCS', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Output.file.format', 'value': 'HTML', 'label': 'Output.File.Format', 'description': 'Output file format, either HTML or XML, default HTML.'}, {'name': 'Output.file.name', 'value': '<Input.FCS.data.file_basename>', 'label': 'Output.File.Name', 'description': 'The output file name (optional).'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Profile.Plots': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Create profile plots for samples', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'Input data file (in gct format)'}, {'name': 'cls.file', 'value': '', 'label': 'Cls.File', 'description': 'File indicating type of samples (sample group) in input.file. Profiles for each group will be plotted in a different color'}], 'default_args': [{'name': 'task_name', 'value': 'Profile.Plots', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<input.file_basename>-profile-plots.pdf', 'label': 'Output.File', 'description': 'Output file name'}, {'name': 'data.type', 'value': 'proteome', 'label': 'Data.Type', 'description': 'Indicate type of input data (proteome, phosphoprotome, etc.) for use in plots'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Protein.mRNA.Corrlation': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Tabulate and plot correlation between proteome and mRNA', 'required_args': [{'name': 'proteome.file', 'value': '', 'label': 'Proteome.File', 'description': 'Proteome data file in gct format'}, {'name': 'mRNA.file', 'value': '', 'label': 'MRNA.File', 'description': 'mRNA (RNA-seq) data file in gct.format'}], 'default_args': [{'name': 'task_name', 'value': 'Protein.mRNA.Corrlation', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.prefix', 'value': '<proteome.file_basename>', 'label': 'Output.Prefix', 'description': 'Filename prefix for results files'}, {'name': 'fdr.threshold', 'value': '0.01', 'label': 'Fdr.Threshold', 'description': 'FDR threshold for marking statistically significant correlations'}, {'name': 'correlation.type', 'value': 'pearson', 'label': 'Correlation.Type', 'description': 'Correlation type to use for calculating protein-mRNA correlation'}, {'name': 'data.type', 'value': 'proteome', 'label': 'Data.Type', 'description': 'Indicate type of input data (proteome, phosphoprotome, etc.) for use in plots'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ProteinDatasetCreation': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Extract features from protein .FASTA files for use with standard prediction algorithms', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'The sequence information in .FASTA format'}], 'default_args': [{'name': 'task_name', 'value': 'ProteinDatasetCreation', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'feature.set.1', 'value': '1', 'label': 'Feature.Set.1', 'description': 'Use amino acid sequence length (1 feature)'}, {'name': 'feature.set.2', 'value': '0', 'label': 'Feature.Set.2', 'description': 'Use amino acid composition (20 features)'}, {'name': 'feature.set.3', 'value': '0', 'label': 'Feature.Set.3', 'description': 'Use reduced alphabet composition of the n terminal region (19 features)'}, {'name': 'feature.set.4', 'value': '0', 'label': 'Feature.Set.4', 'description': 'Use reduced alphabet composition of the c terminal region (19 features)'}, {'name': 'feature.set.5', 'value': '0', 'label': 'Feature.Set.5', 'description': 'Use the reduced alphabet for the gap pairs of amino acids (165 features)'}, {'name': 'output.file', 'value': '<input.file_basename>', 'label': 'Output.File', 'description': 'The output files base name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ProteoArray': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'LC-MS proteomic data processing module', 'required_args': [{'name': 'input.zip.filename', 'value': '', 'label': 'Input.Zip.Filename', 'description': 'zip file of LC-MS runs in mzXML format.'}], 'default_args': [{'name': 'task_name', 'value': 'ProteoArray', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<input.zip.filename_basename>.prot', 'label': 'Output.File', 'description': 'output filename'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ProteomicsAnalysis': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Runs the proteomics analysis on the set of input spectra', 'required_args': [{'name': 'zip.filename', 'value': '', 'label': 'Zip.Filename', 'description': 'Zip file containing spectra data. Spectra data are stored as M/Z,Intensity'}, {'name': 'peak.list.filename', 'value': '', 'label': 'Peak.List.Filename', 'description': 'file containing one M/Z per line when peaks=input'}, {'name': 'random.seed', 'value': '', 'label': 'Random.Seed', 'description': 'random seed when peaks=random'}, {'name': 'random.n.peaks', 'value': '', 'label': 'Random.N.Peaks', 'description': \"number of random M/Z's selected when peaks=random\"}, {'name': 'low.Da', 'value': '', 'label': 'Low.Da', 'description': 'Minimum M/Z to include'}, {'name': 'high.Da', 'value': '', 'label': 'High.Da', 'description': 'Maximum M/Z to include'}], 'default_args': [{'name': 'task_name', 'value': 'ProteomicsAnalysis', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<zip.filename_basename>', 'label': 'Output.File', 'description': 'The base name for output files'}, {'name': 'quality.threshold', 'value': '0.45', 'label': 'Quality.Threshold', 'description': 'discard spectra with quality < quality threshold'}, {'name': 'fill.na', 'value': 'yes', 'label': 'Fill.Na', 'description': 'whether to fill missing peaks'}, {'name': 'normalize', 'value': 'tan', 'label': 'Normalize', 'description': 'normalization strategy'}, {'name': 'peak.detection.method', 'value': 'detect', 'label': 'Peak.Detection.Method', 'description': 'peak detection method'}, {'name': 'filter.peaks', 'value': 'yes', 'label': 'Filter.Peaks', 'description': 'filter peaks based on peak intensity / noise '}, {'name': 'filter.peaks.factor', 'value': '10', 'label': 'Filter.Peaks.Factor', 'description': 'retains peaks if peak intensity >= filter.peaks.factor * stdev(noise)'}, {'name': 'percentile', 'value': '0.65', 'label': 'Percentile', 'description': 'After applying filters, threshold above which peaks are located'}, {'name': 'smoothing.size', 'value': '21', 'label': 'Smoothing.Size', 'description': 'Size of the smoothing filter'}, {'name': 'adaptive.background.correction.strength', 'value': '0.75', 'label': 'Adaptive.Background.Correction.Strength', 'description': 'Strength of the adaptive background correction filter'}, {'name': 'adaptive.background.correction.size', 'value': '21', 'label': 'Adaptive.Background.Correction.Size', 'description': 'Size of the adaptive background correction filter'}, {'name': 'high.pass.filter.strength', 'value': '10', 'label': 'High.Pass.Filter.Strength', 'description': 'Strength of the high pass filter'}, {'name': 'high.pass.filter.factor', 'value': '5', 'label': 'High.Pass.Filter.Factor', 'description': 'Filter factor for the high pass filter'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'QuaSAR': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'A suite of tools for QC, analysis and visualization of data from MRM-MS experiments including assay configuration, calibration curves and new sample analysis.', 'required_args': [{'name': 'skyline.file', 'value': '', 'label': 'Skyline.File', 'description': 'This is a Skyline export in csv file format. The file has to include the following 10 columns with the exact column names given here (in no particular order): SampleName, PeptideSequence, ProteinName, ReplicateName, PrecursorCharge, ProductCharge, FragmentIon,  AverageMeasuredRetentionTime, light Area, heavy Area. The file may contain many more columns, but it has to include at least these 10 columns'}, {'name': 'concentration.file', 'value': '', 'label': 'Concentration.File', 'description': 'Concentration values given in csv file format. The file has to include the following 5 columns with the exact column names given here (in no particular order): SampleName, SampleID, Analyte Concentration, IS Conc, Concentration Ratio. Notice the single space between Analyte Concentration, IS Conc and Concentration Ratio. The space has to be maintained.'}, {'name': 'title', 'value': '', 'label': 'Title', 'description': 'The title is displayed on each calibration plot'}], 'default_args': [{'name': 'task_name', 'value': 'QuaSAR', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'analyte', 'value': 'light Area', 'label': 'Analyte', 'description': 'User must specify the exact analyte column header name as it is appears in the input CSV file. The default is \"light Area\"'}, {'name': 'standard', 'value': 'heavy Area', 'label': 'Standard', 'description': 'User must specify the exact internal standard column header name as it is appears in the input CSV file. The default is \"heavy Area\"'}, {'name': 'generate.cv.table', 'value': '1', 'label': 'Generate.Cv.Table', 'description': 'If \"yes\" then generate CV (coefficient of variation) table'}, {'name': 'generate.calcurves', 'value': '1', 'label': 'Generate.Calcurves', 'description': 'If \"yes\" then generate calibration curves'}, {'name': 'generate.lodloq.table', 'value': '1', 'label': 'Generate.Lodloq.Table', 'description': 'If \"yes\" then generate LOD/LOQ table'}, {'name': 'generate.peak.area.plots', 'value': '0', 'label': 'Generate.Peak.Area.Plots', 'description': 'If \"yes\" then generate peak area plots'}, {'name': 'use.par', 'value': '0', 'label': 'Use.Par', 'description': 'If \"yes\" then use peak area ratio (PAR) for analysis (instead of concentration)'}, {'name': 'max.calcurve.linear.scale', 'value': '150', 'label': 'Max.Calcurve.Linear.Scale', 'description': 'The maximum value for linear scale in fmols/ul'}, {'name': 'max.calcurve.log.scale', 'value': '150', 'label': 'Max.Calcurve.Log.Scale', 'description': 'The maximum value for log scale in fmols/ul'}, {'name': 'perform.audit', 'value': '1', 'label': 'Perform.Audit', 'description': 'If \"yes\" then perform AuDIT for interference detection.'}, {'name': 'audit.cv.threshold', 'value': '0.2', 'label': 'Audit.Cv.Threshold', 'description': 'For AuDIT the threshold for coefficient of variation below which transition is quantification-worthy. Default is 0.2.'}, {'name': 'output.prefix', 'value': '<skyline.file_basename>', 'label': 'Output.Prefix', 'description': 'Prefix used for all output file names'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'RankNormalize': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Normalize the rows in a data set by rank', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'The dataset to be normalized'}, {'name': 'scale.to.value', 'value': '', 'label': 'Scale.To.Value', 'description': 'Result values will be scaled to this value by multiplication after normalization.  Leave this blank to scale to the number of rows in the data set.'}, {'name': 'threshold', 'value': '', 'label': 'Threshold', 'description': 'Minimum threshold for values in the data set.  Any data set value below this will be increased to the threshold value before normalization.  If more than one of these parameters is specified, the order of precedence is threshold, then ceiling, then shift.'}, {'name': 'ceiling', 'value': '', 'label': 'Ceiling', 'description': 'Maximum ceiling for values in the data set.  Any data set value above this will be decreased to the ceiling value before normalization.  If more than one of these parameters is specified, the order of precedence is threshold, then ceiling, then shift.'}, {'name': 'shift', 'value': '', 'label': 'Shift', 'description': 'Shift all values in the data set.  The shift value will be added to all data set values before normalization.  If more than one of these parameters is specified, the order of precedence is threshold, then ceiling, then shift.'}], 'default_args': [{'name': 'task_name', 'value': 'RankNormalize', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file.name', 'value': '<input.file_basename>.NORM.<input.file_extension>', 'label': 'Output.File.Name', 'description': 'The name to be given to the output file.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Read_group_trackingToGct': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '[**Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] Converts a Cufflinks read_group_tracking file into GCT format', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'The read_group_tracking file to be converted into GCT format.'}, {'name': 'gtf.file', 'value': '', 'label': 'Gtf.File', 'description': 'An optional GTF file.  If provided, this should be the same file that you provided to Cuffdiff. You can use this to map tracking IDs in the read_group_tracking file to gene symbols for output in the GCT.  Make sure that you set the feature level to match your read_group_tracking file.'}, {'name': 'override.attribute.name.for.lookup', 'value': '', 'label': 'Override.Attribute.Name.For.Lookup', 'description': 'Use this to override the name of the attribute to search against to look up IDs when mapping with a GTF (gene_id, for example). Type in the attribute name exactly as it appears in the GTF.  The retrieval override must also be provided if this parameter is set.'}, {'name': 'override.attribute.name.for.retrieval', 'value': '', 'label': 'Override.Attribute.Name.For.Retrieval', 'description': 'Use this to override the name of the attribute to retrieve on match when mapping with a GTF (gene_name, for example). Type in the attribute name exactly as it appears in the GTF.  The lookup override must also be provided if this parameter is set.'}], 'default_args': [{'name': 'task_name', 'value': 'Read_group_trackingToGct', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file.name', 'value': '<input.file_basename>.gct', 'label': 'Output.File.Name', 'description': 'The name to be given to the GCT output file.'}, {'name': 'expression.value.column', 'value': '3', 'label': 'Expression.Value.Column', 'description': 'The column to use for extracting expression values.'}, {'name': 'screen.out.errors', 'value': 'false', 'label': 'Screen.Out.Errors', 'description': \"Set to 'yes' to exclude from the GCT file any feature having at least one non-OK quantification status across all of the sequenced samples.\"}, {'name': 'output.value.for.error', 'value': 'NA', 'label': 'Output.Value.For.Error', 'description': \"This parameter controls what expression value to write to a GCT file in those cases where a feature's expression estimate carries a non-OK quantification status.  The parameter is only relevant to your final GCT when 'screen out errors' is set to 'no'.  We strongly recommend treating these as missing values (select 'NA' or 'blank').\"}, {'name': 'feature.level.for.symbol.lookup', 'value': 'gene', 'label': 'Feature.Level.For.Symbol.Lookup', 'description': 'Select the feature level of the read_group_tracking file for mapping gene symbols from a GTF. This is ignored if no GTF is provided.'}, {'name': 'choose.gct.column.for.mapping', 'value': 'auto', 'label': 'Choose.Gct.Column.For.Mapping', 'description': 'Use this to explicitly set the GCT column to use when writing symbols retrieved from the GCT. This is automatic by default, meaning that retrieved symbols go into the Name column for gene-level files and into the Description column for other feature levels.'}, {'name': 'report.naming.conflict.details', 'value': 'false', 'label': 'Report.Naming.Conflict.Details', 'description': 'Include the naming conflict details in the summary report, rather than just giving counts of the issues. '}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ReannotateGCT': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Reannotate a GCT file\\'s \"Description\" column with data from a CHIP file', 'required_args': [{'name': 'gct.file', 'value': '', 'label': 'Gct.File', 'description': 'A GCT file to be annotated'}, {'name': 'chip.file', 'value': '', 'label': 'Chip.File', 'description': 'CHIP file to use for annotations'}], 'default_args': [{'name': 'task_name', 'value': 'ReannotateGCT', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<gct.file_basename>_reannot.gct', 'label': 'Output.File', 'description': 'The name of your re-annotated GCT file. '}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'RemoveCopyNumberOutliers': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Reads rows of a CN file and removes outliers on a sample-by-sample basis using improvised method.', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'A CN file (.cn, .RData, or .txt)'}], 'default_args': [{'name': 'task_name', 'value': 'RemoveCopyNumberOutliers', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'num.probes', 'value': '5', 'label': 'Num.Probes', 'description': 'The number of trailing and leading probes to use to compute the median'}, {'name': 'outlier.probe.replacement', 'value': 'NA', 'label': 'Outlier.Probe.Replacement', 'description': 'The replacement value for the outlier probe'}, {'name': 'multiplicative.threshold', 'value': '4', 'label': 'Multiplicative.Threshold', 'description': 'Multiplicative threshold'}, {'name': 'additive.threshold', 'value': '0.3', 'label': 'Additive.Threshold', 'description': 'Additive threshold'}, {'name': 'output.prefix', 'value': '<input.file_basename>', 'label': 'Output.Prefix', 'description': 'A prefix for the output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'RemoveMissingValues': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Removes features and/or samples containing missing (NA) values', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'Input file (gct, res)'}], 'default_args': [{'name': 'task_name', 'value': 'RemoveMissingValues', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'remove.opt', 'value': '1', 'label': 'Remove.Opt', 'description': 'Remove rows, columns, or both containing missing values'}, {'name': 'output.filename', 'value': '<input.file_basename>.na-rm.gct', 'label': 'Output.Filename', 'description': 'output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'removeSamples': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Remove samples in dataset (.gct), using a list of samples provided', 'required_args': [{'name': 'gct.file', 'value': '', 'label': 'Gct.File', 'description': 'data file to filter'}, {'name': 'removelist', 'value': '', 'label': 'Removelist', 'description': 'list of samples to remove'}, {'name': 'extension', 'value': '', 'label': 'Extension', 'description': 'filename prefix'}], 'default_args': [{'name': 'task_name', 'value': 'removeSamples', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'RemoveSaturatedFCSEvents': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Remove saturated events from an FCS data file', 'required_args': [{'name': 'Input.FCS.data.file', 'value': '', 'label': 'Input.FCS.Data.File', 'description': 'An FCS data file to remove saturated events from.'}], 'default_args': [{'name': 'task_name', 'value': 'RemoveSaturatedFCSEvents', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Output.FCS.file.name', 'value': '<Input.FCS.data.file_basename>.without_saturated.fcs', 'label': 'Output.FCS.File.Name', 'description': 'The output FCS file name.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'RenameFile': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Creates a result file with the contents of the input but with a new name.  Where possible, this is done without copying the file but instead by simply making a link.  While it is possible to run this with a submitted file, the common use will be to change the name of a result file or a previously uploaded file.  Note that the original file will still be present in its original location. ', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'Input file to be renamed'}], 'default_args': [{'name': 'task_name', 'value': 'RenameFile', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.filename', 'value': '<input.file_basename>.cp.<input.file_extension>', 'label': 'Output.Filename', 'description': 'The name to give the output file'}, {'name': 'screen.filename', 'value': 'yes', 'label': 'Screen.Filename', 'description': 'Automatically screen out problematic characters in the filename'}, {'name': 'force.copy', 'value': 'no', 'label': 'Force.Copy', 'description': 'Force a copy to be made rather than a link. '}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ReorderByClass': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Reorder the samples in an expression dataset and class file by class', 'required_args': [{'name': 'expression.data.filename', 'value': '', 'label': 'Expression.Data.Filename', 'description': 'Expression data - .res, .gct, .odf'}, {'name': 'class.filename', 'value': '', 'label': 'Class.Filename', 'description': 'Class file - .cls'}], 'default_args': [{'name': 'task_name', 'value': 'ReorderByClass', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'expression.data.file.output.name', 'value': '<expression.data.filename_basename>_reordered', 'label': 'Expression.Data.File.Output.Name', 'description': 'Name of expression data output file'}, {'name': 'class.file.output.name', 'value': '<class.filename_basename>_reordered', 'label': 'Class.File.Output.Name', 'description': 'Name of class output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ReplicatesQC': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Removes cell line replicate samples that fail quality control check of cell line replicate correlations and distribution scores.', 'required_args': [{'name': 'GCT.files', 'value': '', 'label': 'GCT.Files', 'description': 'Please enter at least one and up to three GCT files.'}, {'name': 'extension', 'value': '', 'label': 'Extension', 'description': ''}], 'default_args': [{'name': 'task_name', 'value': 'ReplicatesQC', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'REVEALER': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'REVEALER (Repeated Evaluation of VariablEs conditionAL Entropy and Redundancy) is an analysis method specifically suited to find groups of genomic alterations that match in a complementary way, a predefined functional activation, dependency of drug response target profile. The method starts by considering, if available, already known genomic alterations (seed) that are the known causes or are known or assumed associated with the target. REVEALER starts by finding the genomic alteration that best matches the target profile conditional to the known seed profile using the conditional mutual information. The newly discovered alteration is then merged with the seed to form a new summary feature, and then the process repeats itself finding additional complementary alterations that explain more and more of the target profile.', 'required_args': [], 'default_args': [{'name': 'task_name', 'value': 'REVEALER', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'ds1', 'value': 'https://personal.broadinstitute.org/tamayo/REVEALER/datasets/CTNBB1_transcriptional_reporter.gct', 'label': 'Ds1', 'description': 'Target profile dataset (e.g. https://personal.broadinstitute.org/tamayo/REVEALER/datasets/CTNBB1_transcriptional_reporter.gct)'}, {'name': 'target.name', 'value': 'BETA-CATENIN-REPORTER', 'label': 'Target.Name', 'description': 'Target profile name (e.g. BETA-CATENIN-REPORTER)'}, {'name': 'target.match', 'value': 'positive', 'label': 'Target.Match', 'description': 'Direction of match (e.g. positive)'}, {'name': 'ds2', 'value': 'https://personal.broadinstitute.org/tamayo/REVEALER/datasets/CCLE_MUT_CNA_AMP_DEL_0.70_2fold.MC.gct', 'label': 'Ds2', 'description': 'Features Dataset (e.g. https://personal.broadinstitute.org/tamayo/REVEALER/datasets/CCLE_MUT_CNA_AMP_DEL_0.70_2fold.MC.gct)'}, {'name': 'seed.names', 'value': 'CTNNB1.MC_MUT', 'label': 'Seed.Names', 'description': 'seed(s) names (e.g. CTNNB1.MC_MUT)'}, {'name': 'max.n.iter', 'value': '2', 'label': 'Max.N.Iter', 'description': 'Maximum number of iterations (e.g. 2)'}, {'name': 'n.markers', 'value': '30', 'label': 'N.Markers', 'description': 'Number of top hits to show in heatmap for every iteration (e.g. 30)'}, {'name': 'locs.table.file', 'value': 'https://personal.broadinstitute.org/tamayo/REVEALER/datasets/hgnc_complete_set.Feb_20_2014.v2.txt', 'label': 'Locs.Table.File', 'description': 'Table with chromosomal location for each gene symbol (e.g. https://personal.broadinstitute.org/tamayo/REVEALER/datasets/hgnc_complete_set.Feb_20_2014.v2.txt)'}, {'name': 'count.thres.low', 'value': '3', 'label': 'Count.Thres.Low', 'description': \"Filter out features with less than count.thres.low 1's (e.g. 3)\"}, {'name': 'count.thres.high', 'value': '50', 'label': 'Count.Thres.High', 'description': \"Filter out features with more than count.thres.low 1's (e.g. 50)\"}, {'name': 'pdf.output.file', 'value': 'BCAT_vs_MUT_CNA.pdf', 'label': 'Pdf.Output.File', 'description': 'Output PDF file with results (e.g. BCAT_vs_MUT_CNA.pdf)'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'RLibDemo': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '', 'required_args': [{'name': 'message', 'value': '', 'label': 'Message', 'description': ''}], 'default_args': [{'name': 'task_name', 'value': 'RLibDemo', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'RNASeQC': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '[**Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] Calculates metrics on aligned RNA-seq data.', 'required_args': [{'name': 'bam.files', 'value': '', 'label': 'Bam.Files', 'description': 'An indexed BAM file or zipped set of indexed BAM files to be analyzed. If you are supplying a single BAM file, it should be located in the same folder as its associated index file (BAI). (If SortSam is used to index the BAM file, then the BAM and BAI files are located in the same folder.)'}, {'name': 'sample.info.file', 'value': '', 'label': 'Sample.Info.File', 'description': 'A TXT format sample info file containing a sample ID, sample file name, and notes column in tab-delimited format. - .txt'}, {'name': 'annotation.gtf', 'value': '', 'label': 'Annotation.Gtf', 'description': \"A file containing annotations in GTF format. Either an annotation gtf or annotation gtf file must be provided.Note that the provided hg19 reference sequence requires contigs using in the '1' format.\"}, {'name': 'annotation.gtf.file', 'value': '', 'label': 'Annotation.Gtf.File', 'description': 'A file containing annotations in GTF format. Either an annotation gtf file or annotation gtf must be provided.'}, {'name': 'reference.sequence', 'value': '', 'label': 'Reference.Sequence', 'description': \"A FASTA file containing a reference sequence. The reference sequence provided here has an index (.fai) and a sequence dictionary (.dict). Either a reference sequence or reference sequence file (with index and dictionary) must be provided, but not both. Note that the provided hg19 reference sequence requires contigs using in the '1' format.\"}, {'name': 'reference.sequence.file', 'value': '', 'label': 'Reference.Sequence.File', 'description': 'A FASTA file containing a reference sequence. The reference sequence must have an index (.fai) and a sequence dictionary (.dict) either in the same location or specified in the reference sequence index and reference sequence dictionary parameters. Reference sequence human_assembly19 is available in the documentation. Click on the \"documentation\" link in the upper right hand corner of this page.'}, {'name': 'reference.sequence.index', 'value': '', 'label': 'Reference.Sequence.Index', 'description': 'A file containing the index for the reference sequence - .fai'}, {'name': 'reference.sequence.dictionary', 'value': '', 'label': 'Reference.Sequence.Dictionary', 'description': 'A file containing the dictionary for the reference sequence. - .dict'}, {'name': 'transcript.type.field', 'value': '', 'label': 'Transcript.Type.Field', 'description': 'Specifies the column of the GTF file in which the transcript type is specified. By default, the module looks for an attribute called transcript_type in the annotation gtf.'}, {'name': 'rRNA.interval.file', 'value': '', 'label': 'RRNA.Interval.File', 'description': 'A file containing the genomic coordinates of rRNA. - .list'}, {'name': 'rRNA.reference.file.index', 'value': '', 'label': 'RRNA.Reference.File.Index', 'description': 'zip of index files created by BWA for the supplied rRNA reference file. Required if rRNA reference file is supplied. -.zip'}, {'name': 'transcript.level.metrics', 'value': '', 'label': 'Transcript.Level.Metrics', 'description': 'Whether to calculate transcript-level metrics in addition to sample-level metrics.'}, {'name': 'gc.content.file', 'value': '', 'label': 'Gc.Content.File', 'description': 'A file containing GC content for each of the transcripts. The file must be tab-delimited with 2 columns containing transcript name and GC content. The transcript name must appear in the GTF file.'}, {'name': 'num.downsampling.reads', 'value': '', 'label': 'Num.Downsampling.Reads', 'description': 'Perform downsampling on the given number of reads. It randomly samples the specified number of reads in all experimental samples when calculating metrics.'}, {'name': 'correlation.comparison.file', 'value': '', 'label': 'Correlation.Comparison.File', 'description': 'A GCT file to use to calculate correlation between expression values (only uses first sample if gct file contains more than one sample. Note also that expression values must be log normalized, and the identifiers must match those of the annotation GTF file)'}, {'name': 'gap.length', 'value': '', 'label': 'Gap.Length', 'description': 'Whether to plot the distribution of gap lengths'}, {'name': 'GATK.flags', 'value': '', 'label': 'GATK.Flags', 'description': 'Pass a string of flags directly to the GATK (e.g. \"-DBQ 0\" to set missing base qualities to zero).'}], 'default_args': [{'name': 'task_name', 'value': 'RNASeQC', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'single.end', 'value': 'yes', 'label': 'Single.End', 'description': 'Whether the bam file contains single end reads. Select no for paired-end reads.'}, {'name': 'num.genes', 'value': '1000', 'label': 'Num.Genes', 'description': 'The number of top-expressed genes for which to calculate metrics.'}, {'name': 'num.rRNA.downsampling.reads', 'value': '1000000', 'label': 'Num.RRNA.Downsampling.Reads', 'description': 'Downsamples to calculate rRNA rate more efficiently. Default is 1 million. Set to 0 to disable.'}, {'name': 'transcript.end.length', 'value': '200', 'label': 'Transcript.End.Length', 'description': \"The length of the 3' or 5' end of a transcript.\"}, {'name': 'gc.margin', 'value': '0.375', 'label': 'Gc.Margin', 'description': 'Used in conjunction with the gc content file to specify the percent gc content to use as boundaries. E.g. 0.25 would set a lower cutoff of 25% and an upper cutoff of 75% (default is 0.375).'}, {'name': 'output.prefix', 'value': '<bam.files_basename>', 'label': 'Output.Prefix', 'description': 'The prefix to use for the output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'RNASeQC_CEGS': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '', 'required_args': [{'name': 'Picard.CreateSequenceDictionary1.reference.sequence.file', 'value': '', 'label': 'Picard.CreateSequenceDictionary1.Reference.Sequence.File', 'description': 'Input reference fasta or fasta.gz '}, {'name': 'TopHat2.reads.pair.1', 'value': '', 'label': 'TopHat2.Reads.Pair.1', 'description': 'Unpaired reads file or first mate for paired reads. A file or zip of files containing reads in FASTA or FASTQ format (bz2 and gz compressed files are supported).'}, {'name': 'TopHat2.reads.pair.2', 'value': '', 'label': 'TopHat2.Reads.Pair.2', 'description': 'Second mate for paired reads. A file or zip of files in FASTA or FASTQ format (bz2 and gz compressed files are supported).'}, {'name': 'Picard.ReorderSam3.reference.file', 'value': '', 'label': 'Picard.ReorderSam3.Reference.File', 'description': 'Reference sequence to which to reorder reads in FASTA format.'}, {'name': 'RNASeQC7.gc.content.file', 'value': '', 'label': 'RNASeQC7.Gc.Content.File', 'description': 'A file containing GC content for each of the transcripts. The file must be tab-delimited with 2 columns containing transcript name and GC content. The transcript name must appear in the GTF file.'}], 'default_args': [{'name': 'task_name', 'value': 'RNASeQC_CEGS', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'SamSPECTRALClusterFCS': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Cluster flow cytometry data using the SamSPECTRAL algorithm.', 'required_args': [{'name': 'Input.FCS.data.file', 'value': '', 'label': 'Input.FCS.Data.File', 'description': 'The FCS file to be clustered.'}, {'name': 'Dimensions', 'value': '', 'label': 'Dimensions', 'description': 'A comma-separated list of dimensions (parameters/channels) that shall be used for clustering. Use either parameter names or indexes. All dimensions except for Time will be used if this parameter is not provided.'}, {'name': 'Dimensions.to.transform', 'value': '', 'label': 'Dimensions.To.Transform', 'description': 'A comma-separated list of dimensions (parameters/channels) that shall transformed as specified by previous parameter. This will be ignored if no transformation is specified above. If this parameter is not provided and transformation is specified above, the algorithm will use heuristics to identify parameters that shall be transformed. These heuristics are based on how parameters are stored in the FCS file, their resolution and their name. Again, you can use either parameter names or parameter indexes to specify dimensions to transform.'}, {'name': 'Sigma', 'value': '', 'label': 'Sigma', 'description': 'A scaling parameter that determines the resolution in the spectral clustering stage. By increasing it, more spectral clusters are identified. This can be usefull when small population are aimed. See documentation on more details how Sigma can be adjusted.'}, {'name': 'Separation.factor', 'value': '', 'label': 'Separation.Factor', 'description': 'Threshold that controls to what extend clusters should be combined or kept separate. See documentation on more details how the separation factor can be adjusted.'}, {'name': 'Seed', 'value': '', 'label': 'Seed', 'description': 'Seed value for the random generator; provide any consistent value to obtain reproducible results.'}], 'default_args': [{'name': 'task_name', 'value': 'SamSPECTRALClusterFCS', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Transformation', 'value': 'ASinH', 'label': 'Transformation', 'description': 'Which transformation to apply on the data prior clustering. Fluorescence channels are usually better visualized and clustered using a transformation; the ASinH transformation produces good results on most data.'}, {'name': 'Beta', 'value': '4', 'label': 'Beta', 'description': 'The size of neighborhood that is being searched during the conductance calculation step. The default value of 4 should not be changed unless the input data set contains measurements of more that 100,000. Setting beta to zero or a low value will reduce computational time by applying an approximation to the conductance calculation step. This will allow you to cluster large data sets in reasonable time; however, the quality of the result may be affected. Setting beta to 5 is not recommended since it will increase computational time while not significantly improving clustering results in most cases.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'SAMtools.FastaIndex': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Indexes a reference sequence in FASTA format', 'required_args': [{'name': 'fasta.file', 'value': '', 'label': 'Fasta.File', 'description': 'A sequence in FASTA format.'}], 'default_args': [{'name': 'task_name', 'value': 'SAMtools.FastaIndex', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.prefix', 'value': '<fasta.file_basename>', 'label': 'Output.Prefix', 'description': 'The prefix to use for the output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Scripture': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Segmentation-based transcript reconstruction algorithm', 'required_args': [{'name': 'alignment.file', 'value': '', 'label': 'Alignment.File', 'description': 'Alignment file in SAM or BAM format, sorted by start position'}, {'name': 'alignment.index.file', 'value': '', 'label': 'Alignment.Index.File', 'description': 'Index file for alignment file, in SAI or BAI format. The index file must have the same root name as the alignment file. (Example: foo.sam and foo.sam.sai OR foo.sai) '}, {'name': 'chromosome.size.file', 'value': '', 'label': 'Chromosome.Size.File', 'description': 'File containing sizes of chromosomes'}, {'name': 'chromosome', 'value': '', 'label': 'Chromosome', 'description': 'Selected chromosome, e.g., chr19'}, {'name': 'chromosome.sequence.file', 'value': '', 'label': 'Chromosome.Sequence.File', 'description': 'Chromosome sequence in FASTA format'}], 'default_args': [{'name': 'task_name', 'value': 'Scripture', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.prefix', 'value': '<alignment.file_basename>', 'label': 'Output.Prefix', 'description': 'A label that will be used to name output files'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ScripturePipeline': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Segmentation-based transcript reconstruction using single- or paired-end reads', 'required_args': [{'name': 'TopHat1.prebuilt.bowtie.index', 'value': '', 'label': 'TopHat1.Prebuilt.Bowtie.Index', 'description': 'A prebuilt bowtie index. Either a prebuilt or a custom Bowtie index must be specified.'}, {'name': 'TopHat1.custom.bowtie.index', 'value': '', 'label': 'TopHat1.Custom.Bowtie.Index', 'description': 'A ZIP archive containing Bowtie index files. Either a prebuilt or a custom Bowtie index must be specified.'}, {'name': 'TopHat1.reads.pair.1', 'value': '', 'label': 'TopHat1.Reads.Pair.1', 'description': 'Unpaired reads file or first mate for paired reads. A file or zip of files containing reads in FASTA or FASTQ format.'}, {'name': 'TopHat1.reads.pair.2', 'value': '', 'label': 'TopHat1.Reads.Pair.2', 'description': 'Second mate for paired reads. A file or zip of files in FASTA or FASTQ format.'}, {'name': 'TopHat1.mate.inner.dist', 'value': '', 'label': 'TopHat1.Mate.Inner.Dist', 'description': 'The expected mean inner distance between mate pairs.'}, {'name': 'TopHat1.mate.std.dev', 'value': '', 'label': 'TopHat1.Mate.Std.Dev', 'description': 'The standard deviation for the distribution on inner distances between mate pairs.'}, {'name': 'TopHat1.library.type', 'value': '', 'label': 'TopHat1.Library.Type', 'description': 'Library type for strand specific reads.'}, {'name': 'Scripture3.chromosome.size.file', 'value': '', 'label': 'Scripture3.Chromosome.Size.File', 'description': 'File containing sizes of chromosomes'}, {'name': 'Scripture3.chromosome', 'value': '', 'label': 'Scripture3.Chromosome', 'description': 'Selected chromosome, e.g., chr19'}, {'name': 'Scripture3.chromosome.sequence.file', 'value': '', 'label': 'Scripture3.Chromosome.Sequence.File', 'description': 'Chromosome sequence in FASTA format'}], 'default_args': [{'name': 'task_name', 'value': 'ScripturePipeline', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'TopHat1.integer.quals', 'value': 'no', 'label': 'TopHat1.Integer.Quals', 'description': 'Quality values are space-delimited integer values, this becomes default when you specify color option.'}, {'name': 'Scripture3.output.prefix', 'value': '<alignment.file_basename>', 'label': 'Scripture3.Output.Prefix', 'description': 'A label that will be used to name output files'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ScripturePrealigned': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Segmentation-based transcript reconstruction using pre-aligned reads', 'required_args': [{'name': 'SortSam1.input.file', 'value': '', 'label': 'SortSam1.Input.File', 'description': 'Input file in SAM or BAM format'}, {'name': 'Scripture2.chromosome.size.file', 'value': '', 'label': 'Scripture2.Chromosome.Size.File', 'description': 'File containing sizes of chromosomes'}, {'name': 'Scripture2.chromosome', 'value': '', 'label': 'Scripture2.Chromosome', 'description': 'Selected chromosome, e.g., chr19'}, {'name': 'Scripture2.chromosome.sequence.file', 'value': '', 'label': 'Scripture2.Chromosome.Sequence.File', 'description': 'Chromosome sequence in FASTA format'}], 'default_args': [{'name': 'task_name', 'value': 'ScripturePrealigned', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'Scripture2.output.prefix', 'value': '<alignment.file_basename>', 'label': 'Scripture2.Output.Prefix', 'description': 'A label that will be used to name output files'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'SegmentationCount': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Counts the number of segments in a SEG file.', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'A segmentation file - .seg, .txt'}], 'default_args': [{'name': 'task_name', 'value': 'SegmentationCount', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<input.file_basename>.segcount.txt', 'label': 'Output.File', 'description': 'The name of the output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'SelectFeaturesColumns': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': \"Takes a 'column slice' from a .res, .gct, .snp, .cn, .loh, .odf, or .cls file.\", 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'input file - .res, .gct, .snp, .cn, .loh, .odf, or .cls file from which the slice is extracted'}, {'name': 'columns', 'value': '', 'label': 'Columns', 'description': 'columns to select (e.g. 0-3, 5, 9). Either columns or a list file must be provided.'}, {'name': 'list.filename', 'value': '', 'label': 'List.Filename', 'description': 'text file containing one identifier per line that specifies the columns to select. Either columns or a list file must be provided.'}], 'default_args': [{'name': 'task_name', 'value': 'SelectFeaturesColumns', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output', 'value': '<input.filename_basename>.slice', 'label': 'Output', 'description': 'output filename - .res, .gct, .snp, .cn, .loh, .odf, .cls'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'SelectFeaturesRows': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Generate a .res, .gct, .snp, .cn, .loh, or .odf file containing only specified features (i.e. rows).', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'input file - .res, .gct, .snp, .cn, .loh, or .odf file from which the slice is extracted'}, {'name': 'list.filename', 'value': '', 'label': 'List.Filename', 'description': 'text file containing one identifier per line that specifies the rows to select. '}], 'default_args': [{'name': 'task_name', 'value': 'SelectFeaturesRows', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output', 'value': '<list.filename_basename>.slice', 'label': 'Output', 'description': 'output filename - .res, .gct, .snp, .cn, .loh, .odf'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'SelectFileMatrix': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'A helper module which selects a matrix out of a delimited text file.', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'Name of the input file from which to pull the matrix.  Must be a delimited text file.'}, {'name': 'start.row', 'value': '', 'label': 'Start.Row', 'description': 'Starting row of the matrix (line number within the file).  Defaults to the first row.'}, {'name': 'end.row', 'value': '', 'label': 'End.Row', 'description': 'Ending row of the matrix (line number within the file).  Defaults to the last row.'}, {'name': 'start.column', 'value': '', 'label': 'Start.Column', 'description': 'Starting column of the matrix.  Defaults to the first column.'}, {'name': 'end.column', 'value': '', 'label': 'End.Column', 'description': 'Ending column of the matrix. Defaults to the last column.'}, {'name': 'row.select', 'value': '', 'label': 'Row.Select', 'description': 'Select specific rows to include in the matrix (e.g. \"3,5,7-12\").  Note that start/end row parameters may limit this selection.'}, {'name': 'column.select', 'value': '', 'label': 'Column.Select', 'description': 'Select specific columns to include in the matrix (e.g. \"1,2,4-6\").  Note that start/end column parameters may limit this selection'}], 'default_args': [{'name': 'task_name', 'value': 'SelectFileMatrix', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file.base.name', 'value': '<input.file_basename>.select', 'label': 'Output.File.Base.Name', 'description': 'Base name for the output file.  The module will add an appropriate extension, either .csv for a comma-delimted file or .txt for a space- or tab-delimited file).'}, {'name': 'delimiter', 'value': 'TAB', 'label': 'Delimiter', 'description': 'Token naming the character for the output column delimiter. Choices are TAB, COMMA, and SPACE. TAB is the default.'}, {'name': 'output.rejects', 'value': 'false', 'label': 'Output.Rejects', 'description': 'Produce an output file containing the items not meeting the selection criteria.  The file will be named <output.file.base.name>.REJECT.<ext>'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'SetFCSKeywords': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Sets keyword/value(s) in a Flow Cytometry Standard (FCS) file.', 'required_args': [{'name': 'Input.FCS.data.file', 'value': '', 'label': 'Input.FCS.Data.File', 'description': 'The input FCS data file in which to set keyword/value pairs.'}, {'name': 'Keywords.input.file', 'value': '', 'label': 'Keywords.Input.File', 'description': 'The file with keyword/value pairs to add (or change values of existing keywords). The keyword/value pairs can be submitted either as an CSV file with two columns: \"Keyword\" and \"Value\" (heading is optional) or as a text file with a chunk of FCS TEXT segment, e.g., /$P3S/MUC 1 APC/$P4S/Thy1 PE/BD$CUSTOM_VALUE23/12/.'}, {'name': 'Output.fcs.file.name', 'value': '', 'label': 'Output.Fcs.File.Name', 'description': 'The file name for the output FCS data file.'}], 'default_args': [{'name': 'task_name', 'value': 'SetFCSKeywords', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'shRNAcollapseReps': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Collapses replicate samples in a gct data file, based on a sample information file', 'required_args': [{'name': 'gct.file', 'value': '', 'label': 'Gct.File', 'description': 'The input gct data file.'}, {'name': 'sample.info.file', 'value': '', 'label': 'Sample.Info.File', 'description': 'The cell line (sample) information file.'}, {'name': 'prefix', 'value': '', 'label': 'Prefix', 'description': 'The file name prefix.'}], 'default_args': [{'name': 'task_name', 'value': 'shRNAcollapseReps', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'shRNAfoldChange': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Calculate the fold change of each sample in a gct file, based on the appropriate DNA reference value (present in a gct file of the same dimension).', 'required_args': [{'name': 'gct.file', 'value': '', 'label': 'Gct.File', 'description': 'The input gct data file.'}, {'name': 'DNA.reference.file', 'value': '', 'label': 'DNA.Reference.File', 'description': 'The values of the DNA pool reference per sample/shRNA in a gct file (same dimension as input gct file).'}, {'name': 'prefix', 'value': '', 'label': 'Prefix', 'description': 'The file name prefix.'}], 'default_args': [{'name': 'task_name', 'value': 'shRNAfoldChange', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'shRNAmapGenes': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Maps shRNAs in a gct data file to genes based on a mapping file.', 'required_args': [{'name': 'gct.file', 'value': '', 'label': 'Gct.File', 'description': 'The input gct data file.'}, {'name': 'mapping', 'value': '', 'label': 'Mapping', 'description': 'The shRNA to gene mapping file.'}, {'name': 'prefix', 'value': '', 'label': 'Prefix', 'description': 'The file name prefix.'}], 'default_args': [{'name': 'task_name', 'value': 'shRNAmapGenes', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'shRNAremoveOverlap': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Remove shRNAs overlapping in sequence > 3 basepairs from a gct data file', 'required_args': [{'name': 'list', 'value': '', 'label': 'List', 'description': 'The list of shRNAs that are not overlapping.'}, {'name': 'gct.file', 'value': '', 'label': 'Gct.File', 'description': 'The input gct data file.'}, {'name': 'mapping', 'value': '', 'label': 'Mapping', 'description': 'The shRNA to gene mapping file.'}, {'name': 'prefix', 'value': '', 'label': 'Prefix', 'description': 'The file name prefix.'}], 'default_args': [{'name': 'task_name', 'value': 'shRNAremoveOverlap', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'SNPFileCreator': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Process Affymetrix SNP probe-level data into an expression value', 'required_args': [{'name': 'chip.type', 'value': '', 'label': 'Chip.Type', 'description': 'Specify the chip type.'}, {'name': 'network.directory.name', 'value': '', 'label': 'Network.Directory.Name', 'description': 'Networked directory name where the CEL ( and optionally TXT) files reside.'}, {'name': 'zip.filename', 'value': '', 'label': 'Zip.Filename', 'description': 'A zip file containing CEL and optionally TXT files.'}, {'name': 'reference.filename', 'value': '', 'label': 'Reference.Filename', 'description': 'Reference file to use. Leave blank if reference method is median.'}, {'name': 'output.file', 'value': '', 'label': 'Output.File', 'description': 'The name of the output file.'}], 'default_args': [{'name': 'task_name', 'value': 'SNPFileCreator', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'normalization.method', 'value': '2', 'label': 'Normalization.Method', 'description': 'The normalization method to apply.'}, {'name': 'reference.method', 'value': 'median', 'label': 'Reference.Method', 'description': 'Reference method to use in normalization'}, {'name': 'model.method', 'value': '2', 'label': 'Model.Method', 'description': 'The summarization method to apply.'}, {'name': 'allele.specific', 'value': 'no', 'label': 'Allele.Specific', 'description': 'Create Allele Specific probe intensities'}, {'name': 'sort.snp.file', 'value': 'yes', 'label': 'Sort.Snp.File', 'description': 'Sorts output file by Chromosome and location. Filters out probesets without location information'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'SNPFileSorter': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Sorts a .snp file by Chromosome and location', 'required_args': [{'name': 'snp.filename', 'value': '', 'label': 'Snp.Filename', 'description': 'snp file to sort'}], 'default_args': [{'name': 'task_name', 'value': 'SNPFileSorter', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<snp.filename_basename>.sorted.snp', 'label': 'Output.File', 'description': 'output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'SNPMultipleSampleAnalysis': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Determine Regions of Concordant Copy Number Aberrations', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'The input file to analyze - .cn, .xcn, .snp, .txt'}, {'name': 'run.name', 'value': '', 'label': 'Run.Name', 'description': 'The name of the job'}], 'default_args': [{'name': 'task_name', 'value': 'SNPMultipleSampleAnalysis', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'java.max.memory', 'value': '512m', 'label': 'Java.Max.Memory', 'description': ''}, {'name': 'save.dir', 'value': './', 'label': 'Save.Dir', 'description': 'Directory to save to'}, {'name': 'permutations', 'value': '500', 'label': 'Permutations', 'description': 'Number of permutations to use'}, {'name': 'num.tests', 'value': '9', 'label': 'Num.Tests', 'description': 'Number of MSA tests to use'}, {'name': 'num.positions.bin', 'value': '6', 'label': 'Num.Positions.Bin', 'description': 'Number of positions per bin'}, {'name': 'resolution', 'value': '250000', 'label': 'Resolution', 'description': 'The size of the plotted chromosome diagram'}, {'name': 'chromosomes', 'value': 'chr22', 'label': 'Chromosomes', 'description': 'Comma delimited list of chromosomes to analyze'}, {'name': 'save.images', 'value': './', 'label': 'Save.Images', 'description': 'Saves images to the specified directory'}, {'name': 'significance.level', 'value': '.05', 'label': 'Significance.Level', 'description': 'The value at which a region is called significant'}, {'name': 'analyze.by', 'value': 'false', 'label': 'Analyze.By', 'description': ''}, {'name': 'cytoband.filename', 'value': '<libdir>CytoBands.txt', 'label': 'Cytoband.Filename', 'description': 'Cytoband file to use'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'SnpViewer': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Displays a heat map of snp data', 'required_args': [{'name': 'dataset.filename', 'value': '', 'label': 'Dataset.Filename', 'description': 'The dataset to display - .snp, .cn, .xcn'}, {'name': 'sample.info.filename', 'value': '', 'label': 'Sample.Info.Filename', 'description': 'Sample information file'}, {'name': 'sample.list.filename', 'value': '', 'label': 'Sample.List.Filename', 'description': 'Sample list file'}], 'default_args': [{'name': 'task_name', 'value': 'SnpViewer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'max.memory', 'value': '700m', 'label': 'Max.Memory', 'description': 'Maximum memory'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'SOMClustering': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Self-Organizing Maps algorithm', 'required_args': [{'name': 'dataset.filename', 'value': '', 'label': 'Dataset.Filename', 'description': 'Dataset (res, gct, or odf dataset)'}], 'default_args': [{'name': 'task_name', 'value': 'SOMClustering', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.stub', 'value': '<dataset.filename_basename>', 'label': 'Output.Stub', 'description': 'The base output file name - .odf'}, {'name': 'cluster.range', 'value': '2-3', 'label': 'Cluster.Range', 'description': 'Range of values can be entered and the program will automatically run the algorithm for each number of clusters in the range.'}, {'name': 'seed.range', 'value': '42', 'label': 'Seed.Range', 'description': 'The seed for the random number generator, is exposed to allow the user to recreate a given session at a later time, (as opposed to allowing the program to generate a random initial seed which could result in different outcomes even if all the other parameters are identical).'}, {'name': 'iterations', 'value': '50000', 'label': 'Iterations', 'description': 'How many times the algorithm should try to refine the clusters. Initially, this value can be set low for faster exploration, but should be set high (e.g., 50,000) for good convergence.'}, {'name': 'cluster.by', 'value': 'rows', 'label': 'Cluster.By', 'description': 'Whether to cluster by rows or columns'}, {'name': 'som.rows', 'value': '0', 'label': 'Som.Rows', 'description': 'Setting this and som.cols to a non zero value will override cluster.range and the computation will be for the specified geometry'}, {'name': 'som.cols', 'value': '0', 'label': 'Som.Cols', 'description': 'Setting this and som.rows to a non zero value will override cluster.range and the computation will be for the specified geometry'}, {'name': 'initialization', 'value': 'Random_Vectors', 'label': 'Initialization', 'description': 'How to select inital random centroids'}, {'name': 'neighborhood', 'value': 'Bubble', 'label': 'Neighborhood', 'description': 'Neighborhood function determining how centroids near the target centroid are updated'}, {'name': 'alpha.initial', 'value': '0.1', 'label': 'Alpha.Initial', 'description': 'Initial learning weight for centroid updates'}, {'name': 'alpha.final', 'value': '0.005', 'label': 'Alpha.Final', 'description': 'Final learning weight for centroid updates'}, {'name': 'sigma.initial', 'value': '5.0', 'label': 'Sigma.Initial', 'description': 'Initial sigma to determine update neighborhood size around the target centroid'}, {'name': 'sigma.final', 'value': '0.5', 'label': 'Sigma.Final', 'description': 'Final sigma to determine update neighborhood size around the target centroid'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'SOMClusterViewer': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Visualize clusters created with the SOM algorithm', 'required_args': [{'name': 'som.cluster.filename', 'value': '', 'label': 'Som.Cluster.Filename', 'description': 'SOM Properties ODF file'}], 'default_args': [{'name': 'task_name', 'value': 'SOMClusterViewer', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'SparseHierarchicalClustering': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Cluster the samples in a data matrix using an adaptively-chosen subset of the features.', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'A GCT file containing data to cluster'}], 'default_args': [{'name': 'task_name', 'value': 'SparseHierarchicalClustering', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'method', 'value': 'average', 'label': 'Method', 'description': 'The type of linkage to use in the hierarchical clustering - \"single\", \"complete\", \"centroid\", or \"average.\"'}, {'name': 'wbound', 'value': '-1', 'label': 'Wbound', 'description': 'The sum of gene weights; this is the tuning parameter for sparse hierarchical clustering. If a non-negative number is given, then this tuning parameter is used. Otherwise, a tuning parameter is selected via a permutation approach. Note: The program runs much more quickly if wbound is specified rather than left at -1.'}, {'name': 'maxnumgenes', 'value': '5000', 'label': 'Maxnumgenes', 'description': 'By default, only the 5000 genes with highest variance will be used (for computational reasons). To use all the genes, set to -1. Warning: using all genes will be slow if there are many genes in the data set.'}, {'name': 'cluster.features', 'value': 'false', 'label': 'Cluster.Features', 'description': 'If a clustering of the genes with non-zero weights is desired, set to TRUE.'}, {'name': 'method.features', 'value': 'average', 'label': 'Method.Features', 'description': 'The type of linkage used to cluster the features (if desired) - \"single\", \"complete\", \"centroid\", or \"average\". Used only if cluster features is set to \"TRUE\"'}, {'name': 'standardize.arrays', 'value': 'true', 'label': 'Standardize.Arrays', 'description': 'Whether or not the arrays should be standardized to have mean zero and standard deviation one before clustering is performed.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'SplitColumns': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Creates a separate file for each column specified from the input file', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'A tab delimited file'}, {'name': 'split.columns', 'value': '', 'label': 'Split.Columns', 'description': 'Specify columns to extract (i.e. 2,4, 5-, 11-17,). Leave blank to split all columns.'}, {'name': 'additional.columns', 'value': '', 'label': 'Additional.Columns', 'description': 'Specify subset of additional columns to include in each (i.e. 2,4, 5-, 11-17) output file.'}], 'default_args': [{'name': 'task_name', 'value': 'SplitColumns', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.prefix', 'value': '<input.file_basename>', 'label': 'Output.Prefix', 'description': 'The prefix to use for the output file name.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'SplitDatasetTrainTest': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Splits a dataset (and cls file) into a number of train and test subsets', 'required_args': [{'name': 'input.dataset.file', 'value': '', 'label': 'Input.Dataset.File', 'description': 'input dataset - .gct, .res'}, {'name': 'cls.file', 'value': '', 'label': 'Cls.File', 'description': 'class template - .cls'}], 'default_args': [{'name': 'task_name', 'value': 'SplitDatasetTrainTest', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'split.method', 'value': 'random', 'label': 'Split.Method', 'description': 'whether to split the data using a percentage split or cross validation approach'}, {'name': 'stratified', 'value': 'T', 'label': 'Stratified', 'description': 'whether to create splits stratified with respect to the class template'}, {'name': 'folds', 'value': '1', 'label': 'Folds', 'description': 'number of train/test splits to generate'}, {'name': 'percentage.split.proportion', 'value': '0.5', 'label': 'Percentage.Split.Proportion', 'description': 'proportion of data to be allocated to train file when split method is percentage split (ignored when split method is cross-validation)'}, {'name': 'seed', 'value': '123456', 'label': 'Seed', 'description': 'random number generator seed'}, {'name': 'output.prefix', 'value': '<input.dataset.file_basename>', 'label': 'Output.Prefix', 'description': 'output saved to output.prefix.{train,test}.n.{gct|res,cls}, where n is the fold index'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'SraToFastQ': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Converts data in SRA format to FastQ using the SRAToolkit version 2.3.5-2.', 'required_args': [{'name': 'accession', 'value': '', 'label': 'Accession', 'description': 'A Sequence Read Archive accession.  Either an accession or an sra.file is required.'}, {'name': 'sra.file', 'value': '', 'label': 'Sra.File', 'description': 'A Sequence Read Archive (.sra) file.  Either an accession or an sra.file is required.'}, {'name': 'paired.end', 'value': '', 'label': 'Paired.End', 'description': 'Whether the reads are paired end'}, {'name': 'quality.score.offset', 'value': '', 'label': 'Quality.Score.Offset', 'description': 'Offset to use for ASCII quality scores. Default is 33 (\"!\").'}, {'name': 'biological.reads.only', 'value': '', 'label': 'Biological.Reads.Only', 'description': 'Whether to include only biological reads and no technical reads in output'}], 'default_args': [{'name': 'task_name', 'value': 'SraToFastQ', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'alignment.status', 'value': 'any', 'label': 'Alignment.Status', 'description': 'The alignment status of the reads to include in output'}, {'name': 'output.compression', 'value': 'none', 'label': 'Output.Compression', 'description': 'Compression to apply to the output files'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'ssGSEAProjection': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Performs single sample GSEA Projection', 'required_args': [{'name': 'input.gct.file', 'value': '', 'label': 'Input.Gct.File', 'description': 'Input gct file.  This module does not allow missing values.'}, {'name': 'output.file.prefix', 'value': '', 'label': 'Output.File.Prefix', 'description': 'Output filename prefix.  If unspecified, is set to <prefix of input.gct.file>.PROJ.'}, {'name': 'gene.sets.database.files', 'value': '', 'label': 'Gene.Sets.Database.Files', 'description': 'Gene sets description file(s) - .gmt, .gmx.  Upload a gene set if your gene set is not listed as a choice for the gene sets database parameter.'}], 'default_args': [{'name': 'task_name', 'value': 'ssGSEAProjection', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'gene.sets.database', 'value': 'c1.all.v5.2.symbols.gmt', 'label': 'Gene.Sets.Database', 'description': 'Gene sets database from GSEA website.'}, {'name': 'gene.symbol.column', 'value': 'Name', 'label': 'Gene.Symbol.Column', 'description': 'Specifies the gct data column containing gene symbols.  In most cases this will be \"Column 1\".'}, {'name': 'gene.set.selection', 'value': 'ALL', 'label': 'Gene.Set.Selection', 'description': 'Comma-separated list of gene set names on which to project expression data. Set to \"ALL\" if projecting to all defined gene sets.'}, {'name': 'sample.normalization.method', 'value': 'rank', 'label': 'Sample.Normalization.Method', 'description': 'Method used to normalize gene expression data'}, {'name': 'weighting.exponent', 'value': '0.75', 'label': 'Weighting.Exponent', 'description': 'Module authors strongly recommend against changing from default.'}, {'name': 'min.gene.set.size', 'value': '10', 'label': 'Min.Gene.Set.Size', 'description': 'Gene sets with overlap smaller than this are excluded from the analysis.'}, {'name': 'combine.mode', 'value': 'combine.add', 'label': 'Combine.Mode', 'description': 'Options for combining enrichment scores for paired *_UP and *_DN gene sets'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'StrandSpecRNAseqEval': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Evaluates strand-specific yeast (S. cerevisiae) RNA-seq libraries', 'required_args': [{'name': 'sam.file', 'value': '', 'label': 'Sam.File', 'description': 'The input SAM file to read and evaluate - .sam'}], 'default_args': [{'name': 'task_name', 'value': 'StrandSpecRNAseqEval', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'average.gene.stats', 'value': '0', 'label': 'Average.Gene.Stats', 'description': 'A flag to indicate whether to run the average gene statistics. This is a long process.'}, {'name': 'flip.strands', 'value': '0', 'label': 'Flip.Strands', 'description': 'A flag to indicate whether the data is flipped, so a forward gene would appear as a reverse one'}, {'name': 'output.prefix', 'value': '<sam.file_basename>', 'label': 'Output.Prefix', 'description': 'A prefix for the output files'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'SubMap': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Maps subclasses between two data sets', 'required_args': [{'name': 'datasetA.file', 'value': '', 'label': 'DatasetA.File', 'description': 'Input dataset A - .gct, should have common gene ID with dataset B'}, {'name': 'datasetB.file', 'value': '', 'label': 'DatasetB.File', 'description': 'Input dataset B - .gct,  should have common gene ID with dataset A'}, {'name': 'classA.file', 'value': '', 'label': 'ClassA.File', 'description': 'Input class label A - .cls, 3rd line should be numeric, +1 if starts from \"0\"'}, {'name': 'classB.file', 'value': '', 'label': 'ClassB.File', 'description': 'Input class label B - .cls, 3rd line should be numeric, +1 if starts from \"0\"'}], 'default_args': [{'name': 'task_name', 'value': 'SubMap', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'num.marker.genes', 'value': '100', 'label': 'Num.Marker.Genes', 'description': 'Number of marker genes to be mapped'}, {'name': 'num.perm', 'value': '100', 'label': 'Num.Perm', 'description': 'Number of random permutations for ES in each cell of SA matrix'}, {'name': 'num.perm.fisher', 'value': '1000', 'label': 'Num.Perm.Fisher', 'description': \"Number of random permutations for Fisher's statistics\"}, {'name': 'weighted.score.type', 'value': '1', 'label': 'Weighted.Score.Type', 'description': 'Weight enrichment by correlation vector (SNR)'}, {'name': 'null.distribution', 'value': 'pool', 'label': 'Null.Distribution', 'description': 'Choice of null distribution'}, {'name': 'p.value.correction', 'value': 'Bonferroni', 'label': 'P.Value.Correction', 'description': 'p-value correction method'}, {'name': 'cluster.rows', 'value': '1', 'label': 'Cluster.Rows', 'description': \"Cluster dataset A's subclass in heatmap of SA matrix\"}, {'name': 'cluster.columns', 'value': '1', 'label': 'Cluster.Columns', 'description': \"Cluster dataset B's subclass in heatmap of SA matrix\"}, {'name': 'nominal.p.value.matrix', 'value': 'T', 'label': 'Nominal.P.Value.Matrix', 'description': 'Create heatmap of each nominal-p matrix'}, {'name': 'create.legend', 'value': 'T', 'label': 'Create.Legend', 'description': 'Create legend for heatmap'}, {'name': 'random.seed', 'value': '47365321', 'label': 'Random.Seed', 'description': 'Random seed'}, {'name': 'output.filename', 'value': 'SubMap', 'label': 'Output.Filename', 'description': 'Name of output files containing heatmap of SA matrices, summary of ES matrix, nominal p-values, and corrected p-values'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'SuggestNumberOfPopulationsFCS': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'The FlowClust module requires that you specify how many populations you are looking for.  This module takes a data driven approach and attempts to discover how many populations are present in an FCS file', 'required_args': [{'name': 'Input.FCS.data.file', 'value': '', 'label': 'Input.FCS.Data.File', 'description': 'The FCS file'}, {'name': 'Dimensions', 'value': '', 'label': 'Dimensions', 'description': 'The channels to search for populations, separated by commas. Leave blank to search in all channels. E.g. FSC Lin, SS Lin, CD4 Log'}, {'name': 'Range', 'value': '', 'label': 'Range', 'description': 'By default, we try between 1 and 30 populations and report the most likely number in the file. As this is a time consuming procedure, you can specify a more limited range. e.g. 1,12'}], 'default_args': [{'name': 'task_name', 'value': 'SuggestNumberOfPopulationsFCS', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'SurvivalCurve': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Draws a survival curve based on a cls file', 'required_args': [{'name': 'input.surv.data.file', 'value': '', 'label': 'Input.Surv.Data.File', 'description': 'input clinical data including survival time & censor status -.txt'}, {'name': 'input.cls.file', 'value': '', 'label': 'Input.Cls.File', 'description': 'Class information file -.cls'}, {'name': 'cls.field', 'value': '', 'label': 'Cls.Field', 'description': 'Name of class field in survival data file (optional)'}, {'name': 'manual.line.type', 'value': '', 'label': 'Manual.Line.Type', 'description': 'e.g. 1,2,3,... (1: solid, 2: dashed, 3: dotted, 4: dot-dashed)'}, {'name': 'manual.line.color', 'value': '', 'label': 'Manual.Line.Color', 'description': 'e.g. black, blue, red, ...'}, {'name': 'max.time', 'value': '', 'label': 'Max.Time', 'description': 'Maximum survival time (in original scale)'}], 'default_args': [{'name': 'task_name', 'value': 'SurvivalCurve', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'time.field', 'value': 'time', 'label': 'Time.Field', 'description': 'Name of survival time field'}, {'name': 'censor.field', 'value': 'status', 'label': 'Censor.Field', 'description': 'Name of survival time field (value should be \"event:1, censor:0\")'}, {'name': 'print.fit.results', 'value': 'F', 'label': 'Print.Fit.Results', 'description': 'Output file name for survival fit results'}, {'name': 'line.type.color.assign', 'value': 'automatic', 'label': 'Line.Type.Color.Assign', 'description': 'Assign line type/color automatically/manually?'}, {'name': 'line.width', 'value': '1', 'label': 'Line.Width', 'description': 'Line width'}, {'name': 'time.conversion', 'value': '1', 'label': 'Time.Conversion', 'description': 'Survival time conversion'}, {'name': 'surv.function.lower', 'value': '0', 'label': 'Surv.Function.Lower', 'description': 'Lower limit of y axis range'}, {'name': 'surv.function.higher', 'value': '1', 'label': 'Surv.Function.Higher', 'description': 'Upper limit of y axis range'}, {'name': 'curve.type', 'value': 'log', 'label': 'Curve.Type', 'description': 'Type of survival curve'}, {'name': 'show.conf.interval', 'value': '0', 'label': 'Show.Conf.Interval', 'description': 'Show confidence inteval?'}, {'name': 'add.legend', 'value': 'T', 'label': 'Add.Legend', 'description': 'Show legend?'}, {'name': 'legend.position', 'value': 'left-bottom', 'label': 'Legend.Position', 'description': 'Position of legend'}, {'name': 'output.filename', 'value': '<input.surv.data.file_basename>', 'label': 'Output.Filename', 'description': 'Name for output files'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'SurvivalDifference': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Tests for survival difference based on cls file', 'required_args': [{'name': 'input.surv.data.file', 'value': '', 'label': 'Input.Surv.Data.File', 'description': 'input clinical data including survival time & censor status  -.txt'}, {'name': 'input.cls.file', 'value': '', 'label': 'Input.Cls.File', 'description': 'class information file (.cls format)'}, {'name': 'cls.field', 'value': '', 'label': 'Cls.Field', 'description': 'Name of class field in survival data file (optional)'}, {'name': 'subgroup', 'value': '', 'label': 'Subgroup', 'description': 'Suggroups (classes) to be compared (e.g., A, C as in 3rd line of cls file)'}], 'default_args': [{'name': 'task_name', 'value': 'SurvivalDifference', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'time.field', 'value': 'time', 'label': 'Time.Field', 'description': 'Name of survival time field'}, {'name': 'censor.field', 'value': 'status', 'label': 'Censor.Field', 'description': 'Name of censor field (value should be \"event:1, no event: 0\")'}, {'name': 'test.type', 'value': '0', 'label': 'Test.Type', 'description': 'Type of test'}, {'name': 'output.filename', 'value': '<input.surv.data.file_basename>', 'label': 'Output.Filename', 'description': 'Output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'SVM': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Classify samples using SVM', 'required_args': [{'name': 'train.data.filename', 'value': '', 'label': 'Train.Data.Filename', 'description': 'The training data file - .gct, .res'}, {'name': 'train.cls.filename', 'value': '', 'label': 'Train.Cls.Filename', 'description': 'The training class file - .cls'}, {'name': 'saved.model.filename', 'value': '', 'label': 'Saved.Model.Filename', 'description': 'A saved model from a previous run of SVM'}, {'name': 'test.data.filename', 'value': '', 'label': 'Test.Data.Filename', 'description': 'The test data file - .gct, .res'}, {'name': 'test.cls.filename', 'value': '', 'label': 'Test.Cls.Filename', 'description': 'The test class file - .cls'}], 'default_args': [{'name': 'task_name', 'value': 'SVM', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'pred.results.output.file', 'value': '<test.data.filename_basename>.pred.odf', 'label': 'Pred.Results.Output.File', 'description': 'The name of the output file for prediction results'}, {'name': 'model.output.file', 'value': '<train.data.filename_basename>.model', 'label': 'Model.Output.File', 'description': 'The name of the file containing the saved model'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'TmapToGct': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Converts a TMAP file to GCT format', 'required_args': [{'name': 'input.file', 'value': '', 'label': 'Input.File', 'description': 'Input file in TMAP format'}], 'default_args': [{'name': 'task_name', 'value': 'TmapToGct', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'row.labels', 'value': 'Cufflinks gene ID', 'label': 'Row.Labels', 'description': 'Select row labels to be written to GCT file'}, {'name': 'row.descriptions', 'value': 'Cufflinks transcript ID', 'label': 'Row.Descriptions', 'description': 'Select row descriptions to be written to GCT file'}, {'name': 'filter.rows', 'value': 'no', 'label': 'Filter.Rows', 'description': 'Filter rows with invalid row names (ie., \"-\") '}, {'name': 'output.prefix', 'value': '<input.file_basename>', 'label': 'Output.Prefix', 'description': 'A label that will be used to name the GCT output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'TopHat': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': '[**Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] TopHat 2.0.11 is a fast splice junction mapper for RNA-Seq reads. It aligns RNA-Seq reads to mammalian-sized genomes using the ultra high-throughput short read aligner Bowtie, and then analyzes the mapping results to identify splice junctions between exons.', 'required_args': [{'name': 'GTF.file', 'value': '', 'label': 'GTF.File', 'description': 'A GTF file (v. 2.2 or higher) or GFF3 file containing a list of gene model annotations. TopHat will first extract the transcript sequences and align them to this virtual transcriptome first. Only the reads that do not fully map to the transcriptome will then be mapped on the genome. The reads that did map on the transcriptome will be converted to genomic mappings (spliced as needed) and merged with the novel mappings.'}, {'name': 'transcriptome.index', 'value': '', 'label': 'Transcriptome.Index', 'description': 'A directory containing a transcriptome index generated from a previous run of TopHat'}, {'name': 'bowtie.index', 'value': '', 'label': 'Bowtie.Index', 'description': 'A zip file or directory containing a Bowtie 2 index.'}, {'name': 'reads.pair.1', 'value': '', 'label': 'Reads.Pair.1', 'description': 'Unpaired reads file or first mate for paired reads. One or more files containing reads in FASTA or FASTQ format (bz2 and gz compressed files are supported).'}, {'name': 'reads.pair.2', 'value': '', 'label': 'Reads.Pair.2', 'description': 'Second mate for paired reads. Zero or more files in FASTA or FASTQ format (bz2 and gz compressed files are supported).'}, {'name': 'mate.std.dev', 'value': '', 'label': 'Mate.Std.Dev', 'description': 'The standard deviation for the distribution of inner distances between mate pairs.'}, {'name': 'library.type', 'value': '', 'label': 'Library.Type', 'description': 'Library type for strand specific reads.'}, {'name': 'Bowtie.preset.options', 'value': '', 'label': 'Bowtie.Preset.Options', 'description': 'A combination of pre-packaged options for Bowtie 2 based on speed and sensitivity/accuracy.'}, {'name': 'max.transcriptome.hits', 'value': '', 'label': 'Max.Transcriptome.Hits', 'description': 'The maximum number of mappings allowed for a read when it is aligned to the virtual transcriptome (provided in the GTF file parameter). Any reads found with more than this number of mappings will be discarded.'}, {'name': 'prefilter.multihits', 'value': '', 'label': 'Prefilter.Multihits', 'description': 'When mapping reads on the virtual transcriptome (provided in the GTF file parameter), some repetitive or low complexity reads that would be discarded in the context of the genome may appear to align to the transcript sequences and thus may end up reported as mapped to those genes only. This option directs TopHat to first align the reads to the whole genome, then exclude such multi-mapped reads.'}, {'name': 'raw.junctions.file', 'value': '', 'label': 'Raw.Junctions.File', 'description': 'A file containing raw junctions. Junctions are specified one per line in a tab-delimited format.'}, {'name': 'find.novel.junctions', 'value': '', 'label': 'Find.Novel.Junctions', 'description': 'If you select no, then the module will only look for junctions indicated in the GTF file supplied in the GTF file parameter. (This parameter is ignored when no GTF file is specified.)'}, {'name': 'min.intron.length', 'value': '', 'label': 'Min.Intron.Length', 'description': 'The minimum intron length. TopHat will ignore donor/acceptor pairs closer than this many bases apart.'}, {'name': 'quality.value.scale', 'value': '', 'label': 'Quality.Value.Scale', 'description': 'Whether to use the Solexa, Phred 33, or  Solexa v. 1.3 (Phred 64) quality value scale.'}, {'name': 'microexon.search', 'value': '', 'label': 'Microexon.Search', 'description': 'Attempts to find alignments incident to microexons. Works only for reads 50bp or longer.'}, {'name': 'fusion.mapping', 'value': '', 'label': 'Fusion.Mapping', 'description': 'Whether to enable fusion mapping'}, {'name': 'read.edit.dist', 'value': '', 'label': 'Read.Edit.Dist', 'description': 'Final read alignments having more than these many edit distance are discarded.'}, {'name': 'read.gap.length', 'value': '', 'label': 'Read.Gap.Length', 'description': 'Final read alignments having more than these many total length of gaps are discarded.'}, {'name': 'additional.tophat.options', 'value': '', 'label': 'Additional.Tophat.Options', 'description': 'Additional options to be passed along to the TopHat program at the command line. This parameter gives you a means to specify otherwise unavailable TopHat options and switches not supported by the module; check the TopHat manual for details. Recommended for experts only; use this at your own discretion.'}], 'default_args': [{'name': 'task_name', 'value': 'TopHat', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'mate.inner.dist', 'value': '50', 'label': 'Mate.Inner.Dist', 'description': 'The expected mean inner distance between mate pairs.'}, {'name': 'transcriptome.only', 'value': 'no', 'label': 'Transcriptome.Only', 'description': 'Whether to align the reads to the virtual transcriptome (provided in the GTF file parameter) and report only those mappings as genomic mappings.'}, {'name': 'min.anchor.length', 'value': '8', 'label': 'Min.Anchor.Length', 'description': 'The anchor length. This value must be at least 3.'}, {'name': 'max.splice.mismatches', 'value': '0', 'label': 'Max.Splice.Mismatches', 'description': 'The maximum number of mismatches that may appear in the \"anchor\" region of a spliced alignment.'}, {'name': 'max.intron.length', 'value': '500000', 'label': 'Max.Intron.Length', 'description': 'The maximum intron length. When searching for junctions ab initio, TopHat will ignore donor/acceptor pairs farther than this many bases apart, except when such a pair is supported by a split segment alignment of a long read.'}, {'name': 'max.insertion.length', 'value': '3', 'label': 'Max.Insertion.Length', 'description': 'The maximum insertion length'}, {'name': 'max.deletion.length', 'value': '3', 'label': 'Max.Deletion.Length', 'description': 'The maximum deletion length'}, {'name': 'max.multihits', 'value': '20', 'label': 'Max.Multihits', 'description': 'The maximum number of times a read can be aligned to the reference genome. If a read is aligned more than this number of times, then TopHat will choose the alignments based on their alignment scores, reporting the alignments with the best alignment scores. If there are more than this number of alignments with the same score for a read, TopHat will randomly report only this many alignments.'}, {'name': 'read.mismatches', 'value': '2', 'label': 'Read.Mismatches', 'description': 'Final read alignments having more than these many mismatches should be discarded.'}, {'name': 'coverage.search', 'value': '--no-coverage-search', 'label': 'Coverage.Search', 'description': 'Enables or disables the coverage based search for junctions. Use when coverage search is disabled by default (such as for reads 75bp or longer), for maximum sensitivity.'}, {'name': 'fusion.anchor.length', 'value': '20', 'label': 'Fusion.Anchor.Length', 'description': 'A \"supporting\" read must map to both sides of a fusion by at least these many bases. Only applies when fusion mapping is set to yes.'}, {'name': 'fusion.read.mismatches', 'value': '2', 'label': 'Fusion.Read.Mismatches', 'description': 'Reads support fusions if they map across fusion with at most these many mismatches. Only applies when fusion mapping is set to yes.'}, {'name': 'output.prefix', 'value': '<job_id>_<reads.pair.1_basename>', 'label': 'Output.Prefix', 'description': 'The prefix to use for the output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'TransposeDataset': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Transpose a Dataset - .res .gct, .odf', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'Dataset - .res .gct, .odf'}], 'default_args': [{'name': 'task_name', 'value': 'TransposeDataset', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file.format', 'value': 'gct', 'label': 'Output.File.Format', 'description': 'The output file format'}, {'name': 'output.file.name', 'value': '<input.filename_basename>.transposed', 'label': 'Output.File.Name', 'description': 'The name of the output file'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Trimmomatic': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Provides a variety of options for trimming Illumina FASTQ files of adapter sequences and low-quality reads.', 'required_args': [{'name': 'input.file.1', 'value': '', 'label': 'Input.File.1', 'description': 'The input FASTQ to be trimmed.  For paired-end data, this should be the forward (\"*_1\" or \"left\") input file.'}, {'name': 'input.file.2', 'value': '', 'label': 'Input.File.2', 'description': 'The reverse (\"*_2\" or \"right\") input FASTQ of paired-end data to be trimmed.'}, {'name': 'adapter.clip.sequence.file', 'value': '', 'label': 'Adapter.Clip.Sequence.File', 'description': 'A FASTA file containing the adapter sequences, PCR sequences, etc. to be clipped. This parameter is required to enable adapter clipping. Files are provided for several Illumina pipelines but you can also provide your own; see the documentation for details. Be sure to choose a PE file for paired-end data and an SE file for single-end data.'}, {'name': 'adapter.clip.seed.mismatches', 'value': '', 'label': 'Adapter.Clip.Seed.Mismatches', 'description': 'Specifies the maximum mismatch count which will still allow a full match to be performed. A value of 2 is recommended. This parameter is required to enable adapter clipping.'}, {'name': 'adapter.clip.palindrome.clip.threshold', 'value': '', 'label': 'Adapter.Clip.Palindrome.Clip.Threshold', 'description': \"Specifies how accurate the match between the two 'adapter ligated' reads must be for PE palindrome read alignment. This is the log10 probability against getting a match by random chance; values around 30 or more are recommended. This parameter is required to enable adapter clipping.\"}, {'name': 'adapter.clip.simple.clip.threshold', 'value': '', 'label': 'Adapter.Clip.Simple.Clip.Threshold', 'description': 'Specifies how accurate the match between any adapter etc. sequence must be against a read as a log10 probability against getting a match by random chance; values between 7-15 are recommended. This parameter is required to enable adapter clipping.'}, {'name': 'adapter.clip.min.length', 'value': '', 'label': 'Adapter.Clip.Min.Length', 'description': 'In addition to the alignment score, palindrome mode can verify that a minimum length of adapter has been detected. If unspecified, this defaults to 8 bases, for historical reasons. However, since palindrome mode has a very low false positive rate, this can be safely reduced, even down to 1, to allow shorter adapter fragments to be removed.'}, {'name': 'trim.leading.quality.threshold', 'value': '', 'label': 'Trim.Leading.Quality.Threshold', 'description': 'Remove low quality bases from the beginning. As long as a base has a value below this threshold the base is removed and the next base will be investigated.'}, {'name': 'trim.trailing.quality.threshold', 'value': '', 'label': 'Trim.Trailing.Quality.Threshold', 'description': 'Remove low quality bases from the end. As long as a base has a value below this threshold the base is removed and the next trailing base will be investigated.'}, {'name': 'max.info.target.length', 'value': '', 'label': 'Max.Info.Target.Length', 'description': 'Performs a Max Info adaptive quality trim, balancing the benefits of retaining longer reads against the costs of retaining bases with errors. This parameter specifies the read length which is likely to allow the location of the read within the target sequence to be determined.'}, {'name': 'max.info.strictness', 'value': '', 'label': 'Max.Info.Strictness', 'description': 'This value, which should be set between 0 and 1, specifies the balance between preserving as much read length as possible vs. removal of incorrect bases. A low value of this parameter (<0.2) favors longer reads, while a high value (>0.8) favors read correctness.  Both max.info.target.length and max.info.strictness are required for the Max Info quality trim.'}, {'name': 'sliding.window.size', 'value': '', 'label': 'Sliding.Window.Size', 'description': 'Perform a sliding window trimming, cutting once the average quality within the window falls below a threshold. By considering multiple bases, a single poor quality base will not cause the removal of high quality data later in the read.  This parameter specifies the number of bases to average across.'}, {'name': 'sliding.window.quality.threshold', 'value': '', 'label': 'Sliding.Window.Quality.Threshold', 'description': 'Specifies the average quality required for the sliding window trimming. Both sliding.window.size and sliding.window.quality.threshold are required to enable the sliding window trimming.'}, {'name': 'min.read.length', 'value': '', 'label': 'Min.Read.Length', 'description': 'Remove reads that fall below the specified minimal length.'}, {'name': 'extra.steps', 'value': '', 'label': 'Extra.Steps', 'description': 'Extra steps to be performed after any other processing. These must be specified in exactly the format described in the Trimmomatic manual; see the documentation for details.  This is recommended for advanced users only.'}, {'name': 'convert.phred.scores', 'value': '', 'label': 'Convert.Phred.Scores', 'description': 'Convert phred scores into a particular encoding.  Leave this blank for no conversion.'}], 'default_args': [{'name': 'task_name', 'value': 'Trimmomatic', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.filename.base', 'value': '<input.file.1_basename>', 'label': 'Output.Filename.Base', 'description': 'A base name to be used for the output files.'}, {'name': 'adapter.clip.keep.both.reads', 'value': 'true', 'label': 'Adapter.Clip.Keep.Both.Reads', 'description': 'Controls whether to keep both forward and reverse reads when trimming in palindrome mode. The reverse read is the same as the forward but in reverse complement and so carries no additional information. The default is \"yes\" (retain the reverse read) which is useful when downstream tools cannot handle a combination of paired and unpaired reads.'}, {'name': 'phred.encoding', 'value': 'phred33', 'label': 'Phred.Encoding', 'description': 'Allows you to specify the phred quality encoding.  The default is phred33, which matches modern Illumina pipelines.'}, {'name': 'create.trimlog', 'value': 'false', 'label': 'Create.Trimlog', 'description': 'Create a log of the trimming process.  This gives details on what operations were performed, etc. but can be quite lengthy.'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'Two.Component.Normalization': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Normalization for unimodal and bimodal samples using mixture models', 'required_args': [{'name': 'gct.file', 'value': '', 'label': 'Gct.File', 'description': 'Input data file (gct format)'}, {'name': 'cls.file', 'value': '', 'label': 'Cls.File', 'description': 'File indicating type of samples (unimodal or bimodal) in gct.file'}], 'default_args': [{'name': 'task_name', 'value': 'Two.Component.Normalization', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.prefix', 'value': '<gct.file_basename>', 'label': 'Output.Prefix', 'description': 'Filename prefix for results files'}, {'name': 'data.type', 'value': 'proteome', 'label': 'Data.Type', 'description': 'Indicate type of input data (proteome, phosphoprotome, etc.) for use in plots'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'UniquifyLabels': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Makes row and column labels unique', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': 'input filename - .res, .gct, .odf type=Dataset'}], 'default_args': [{'name': 'task_name', 'value': 'UniquifyLabels', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<input.filename_basename>.unique', 'label': 'Output.File', 'description': 'output file'}, {'name': 'make.row.labels.unique', 'value': 'yes', 'label': 'Make.Row.Labels.Unique', 'description': 'whether to make the row labels unique'}, {'name': 'make.column.labels.unique', 'value': 'yes', 'label': 'Make.Column.Labels.Unique', 'description': 'whether to make the column labels unique'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'untar': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'untars .tgz files', 'required_args': [{'name': 'input.filename', 'value': '', 'label': 'Input.Filename', 'description': ''}], 'default_args': [{'name': 'task_name', 'value': 'untar', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'VennDiagram': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Displays a Venn diagram', 'required_args': [{'name': 'feature.list.one', 'value': '', 'label': 'Feature.List.One', 'description': 'Feature list one - .grp, .txt'}, {'name': 'feature.list.two', 'value': '', 'label': 'Feature.List.Two', 'description': 'Feature list two - .grp, .txt'}, {'name': 'feature.list.three', 'value': '', 'label': 'Feature.List.Three', 'description': 'Feature list three - .grp, .txt'}], 'default_args': [{'name': 'task_name', 'value': 'VennDiagram', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'WeightedVoting': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Weighted Voting classification', 'required_args': [{'name': 'train.filename', 'value': '', 'label': 'Train.Filename', 'description': 'training data file name - .gct, .res, .odf type = Dataset, ignored if a saved model (saved.model.filename) is used'}, {'name': 'train.class.filename', 'value': '', 'label': 'Train.Class.Filename', 'description': 'class file for training data - .cls, ignored if a saved model (saved.model.filename) is used'}, {'name': 'saved.model.filename', 'value': '', 'label': 'Saved.Model.Filename', 'description': 'input Weighted Voting model file - .odf type = Weighted Voting Prediction Model'}, {'name': 'test.filename', 'value': '', 'label': 'Test.Filename', 'description': 'test data file name - .gct, .res, .odf type = Dataset'}, {'name': 'test.class.filename', 'value': '', 'label': 'Test.Class.Filename', 'description': 'class file for test data - .cls'}, {'name': 'min.std', 'value': '', 'label': 'Min.Std', 'description': 'minimum standard deviation if feature selection statistic includes min std option'}, {'name': 'feature.list.filename', 'value': '', 'label': 'Feature.List.Filename', 'description': 'features to use for prediction'}], 'default_args': [{'name': 'task_name', 'value': 'WeightedVoting', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'model.file', 'value': '<train.filename_basename>.model.odf', 'label': 'Model.File', 'description': 'name of output Weighted Voting model file  - .odf type = Weighted Voting Prediction Model'}, {'name': 'num.features', 'value': '10', 'label': 'Num.Features', 'description': 'number of selected features if feature list filename is not specified'}, {'name': 'feature.selection.statistic', 'value': '0', 'label': 'Feature.Selection.Statistic', 'description': 'statistic to use to perform feature selection'}, {'name': 'pred.results.file', 'value': '<test.filename_basename>.pred.odf', 'label': 'Pred.Results.File', 'description': 'name of prediction results output file - .odf type = Prediction Results'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'WeightedVotingXValidation': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': 'Weighted Voting classification with Leave-One-Out Cross-Validation', 'required_args': [{'name': 'data.filename', 'value': '', 'label': 'Data.Filename', 'description': 'Dataset input data file - res, gct, odf'}, {'name': 'class.filename', 'value': '', 'label': 'Class.Filename', 'description': 'Class input file - cls'}, {'name': 'min.std', 'value': '', 'label': 'Min.Std', 'description': 'minimum standard deviation if feature selection statistic includes min std option'}], 'default_args': [{'name': 'task_name', 'value': 'WeightedVotingXValidation', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'num.features', 'value': '10', 'label': 'Num.Features', 'description': 'number of selected features'}, {'name': 'feature.selection.statistic', 'value': '0', 'label': 'Feature.Selection.Statistic', 'description': 'statistic to use to perform feature selection'}, {'name': 'pred.results.file', 'value': '<data.filename_basename>.pred.odf', 'label': 'Pred.Results.File', 'description': 'Prediction results output file name - odf'}, {'name': 'feature.summary.file', 'value': '<data.filename_basename>.feat.odf', 'label': 'Feature.Summary.File', 'description': 'Feature summary output file name - odf'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}, 'XChromosomeCorrect': {'library_path': '/Users/kateme/Jumis/tools/genepattern-python', 'library_name': 'gp.functions', 'function_name': 'execute', 'description': \"Corrects X Chromosome SNP's for male samples.\", 'required_args': [{'name': 'snp.filename', 'value': '', 'label': 'Snp.Filename', 'description': 'Summarized probe intensity data in .snp format'}, {'name': 'sample.info.filename', 'value': '', 'label': 'Sample.Info.Filename', 'description': 'A Sample information file that contains gender information'}], 'default_args': [{'name': 'task_name', 'value': 'XChromosomeCorrect', 'label': 'Task Name', 'description': 'No description.'}], 'optional_args': [{'name': 'output.file', 'value': '<snp.filename_basename>.xcorrected.snp', 'label': 'Output.File', 'description': 'Output file name'}], 'returns': [{'label': 'Job', 'description': 'ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED'}]}} to be {'Make Data Filepath': {'library_path': '', 'library_name': 'simpli.nbpackage_tasks', 'function_name': 'make_data_filepath', 'description': 'No description.', 'required_args': [{'name': 'filename', 'value': '', 'label': 'File Name', 'description': 'No description.'}], 'default_args': [], 'optional_args': [], 'returns': [{'label': 'Variable for Data File Path', 'description': 'No description.'}]}, 'Make Result Filepath': {'library_path': '', 'library_name': 'simpli.nbpackage_tasks', 'function_name': 'make_result_filepath', 'description': 'No description.', 'required_args': [{'name': 'filename', 'value': '', 'label': 'File Name', 'description': 'No description.'}], 'default_args': [], 'optional_args': [], 'returns': [{'label': 'Variable for Result File Path', 'description': 'No description.'}]}} ...\n",
      "(Getting tasks ...)\n",
      "(Setting tasks ...)\n",
      "{\"Load JSON\": {\"library_path\": \"\", \"library_name\": \"simpli.default_tasks\", \"function_name\": \"link_json\", \"description\": \"No description.\", \"required_args\": [{\"name\": \"filepath\", \"value\": \"\", \"label\": \"Simpli JSON to be Loaded\", \"description\": \"No description.\"}], \"default_args\": [], \"optional_args\": [], \"returns\": []}, \"Reset JSONs\": {\"library_path\": \"\", \"library_name\": \"simpli.default_tasks\", \"function_name\": \"reset_jsons\", \"description\": \"No description.\", \"required_args\": [], \"default_args\": [], \"optional_args\": [], \"returns\": []}, \"Assign\": {\"library_path\": \"\", \"library_name\": \"simpli.default_tasks\", \"function_name\": \"just_return\", \"description\": \"No description.\", \"required_args\": [{\"name\": \"value\", \"value\": \"\", \"label\": \"Value to Assign\", \"description\": \"No description.\"}], \"default_args\": [], \"optional_args\": [], \"returns\": [{\"label\": \"Variable Name to Assign to\", \"description\": \"No description.\"}]}, \"Slice DataFrame by Row\": {\"library_path\": \"\", \"library_name\": \"simpli.default_tasks\", \"function_name\": \"slice_dataframe\", \"description\": \"Return a subset of rows of a given DataFrame object.\", \"required_args\": [{\"name\": \"dataframe\", \"value\": \"\", \"label\": \"DataFrame\", \"description\": \"Target dataframe object.\"}, {\"name\": \"indices\", \"value\": \"\", \"label\": \"Row Index(ices)\", \"description\": \"An integer or list of integers separated by commas of the rows to slice. Note that indices are 0 based.\"}], \"default_args\": [{\"name\": \"ax\", \"value\": \"0\", \"label\": \"Ax\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Sliced DataFrame\", \"description\": \"A slice of the DataFrame including the specified rows.\"}]}, \"Slice DataFrame by Column\": {\"library_path\": \"\", \"library_name\": \"simpli.default_tasks\", \"function_name\": \"slice_dataframe\", \"description\": \"No description.\", \"required_args\": [{\"name\": \"dataframe\", \"value\": \"\", \"label\": \"DataFrame\", \"description\": \"No description.\"}, {\"name\": \"indices\", \"value\": \"\", \"label\": \"Column Index(ices)\", \"description\": \"No description.\"}], \"default_args\": [{\"name\": \"ax\", \"value\": \"1\", \"label\": \"Ax\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Sliced DataFrame\", \"description\": \"No description.\"}]}, \"Set Notebook Theme\": {\"library_path\": \"\", \"library_name\": \"simpli.default_tasks\", \"function_name\": \"set_notebook_theme\", \"description\": \"No description.\", \"required_args\": [{\"name\": \"filepath\", \"value\": \"\", \"label\": \"CSS File\", \"description\": \"No description.\"}], \"default_args\": [], \"optional_args\": [], \"returns\": []}, \"Center Align Outputs\": {\"library_path\": \"\", \"library_name\": \"simpli.default_tasks\", \"function_name\": \"center_align_output_cells\", \"description\": \"No description.\", \"required_args\": [], \"default_args\": [], \"optional_args\": [], \"returns\": []}, \"Display Start Banner and Logos\": {\"library_path\": \"\", \"library_name\": \"simpli.default_tasks\", \"function_name\": \"display_start_banner_and_logos\", \"description\": \"No description.\", \"required_args\": [], \"default_args\": [], \"optional_args\": [], \"returns\": []}, \"Display End Banner\": {\"library_path\": \"\", \"library_name\": \"simpli.default_tasks\", \"function_name\": \"display_end_banner\", \"description\": \"No description.\", \"required_args\": [], \"default_args\": [], \"optional_args\": [], \"returns\": []}, \"Link YouTube Video\": {\"library_path\": \"\", \"library_name\": \"simpli.default_tasks\", \"function_name\": \"youtube\", \"description\": \"No description.\", \"required_args\": [{\"name\": \"url\", \"value\": \"\", \"label\": \"YouTube Video URL\", \"description\": \"No description.\"}], \"default_args\": [], \"optional_args\": [], \"returns\": []}, \"Toggle Input Cells\": {\"library_path\": \"\", \"library_name\": \"simpli.default_tasks\", \"function_name\": \"toggle_input_cells\", \"description\": \"No description.\", \"required_args\": [], \"default_args\": [], \"optional_args\": [], \"returns\": []}, \"Execute HTML\": {\"library_path\": \"\", \"library_name\": \"simpli.default_tasks\", \"function_name\": \"display_raw_html\", \"description\": \"No description.\", \"required_args\": [], \"default_args\": [], \"optional_args\": [], \"returns\": []}, \"Login GenePattern (Broad Institute)\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"login\", \"description\": \"Login into the GenePattern server at the Broad Institure.\", \"required_args\": [{\"name\": \"user_name\", \"value\": \"\", \"label\": \"User name\", \"description\": \"GenePattern user name.\"}, {\"name\": \"passcode\", \"value\": \"\", \"label\": \"Passcode\", \"description\": \"GenePattern passcode.\"}], \"default_args\": [{\"name\": \"server_name\", \"value\": \"https://genepattern.broadinstitute.org/gp\", \"label\": \"Server Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": []}, \"ABSOLUTE\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"[**Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] Extracts absolute copy numbers per cancer cell from a mixed DNA population. Use this module for the per-sample processing step in the workflow (usually after HAPSEG).\", \"required_args\": [{\"name\": \"seg.dat.file\", \"value\": \"\", \"label\": \"Seg.Dat.File\", \"description\": \"A HAPSEG output file\"}, {\"name\": \"primary.disease\", \"value\": \"\", \"label\": \"Primary.Disease\", \"description\": \"Primary disease of the sample\"}, {\"name\": \"sample.name\", \"value\": \"\", \"label\": \"Sample.Name\", \"description\": \"The name of the sample, included in output plots\"}, {\"name\": \"maf.file\", \"value\": \"\", \"label\": \"Maf.File\", \"description\": \"If available, a minor allele frequency file. This specifies the data for somatic point mutations to be used by ABSOLUTE.\"}, {\"name\": \"min.mut.af\", \"value\": \"\", \"label\": \"Min.Mut.Af\", \"description\": \"If specified, a minimum mutation allelic fraction. Mutations with lower allelic fractions will be filtered out before analysis. Note that if maf.fn is specified, min.mut.af must also be specified.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"ABSOLUTE\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.file.name.base\", \"value\": \"<sample.name>\", \"label\": \"Output.File.Name.Base\", \"description\": \"If specified, provides a base filename for all output files. The default value is the sample.name.\"}, {\"name\": \"sigma.p\", \"value\": \"0\", \"label\": \"Sigma.P\", \"description\": \"Provisional value of excess sample level variance used for mode search\"}, {\"name\": \"max.sigma.h\", \"value\": \"0.015\", \"label\": \"Max.Sigma.H\", \"description\": \"Maximum value of excess sample level variance (Eq. 6)\"}, {\"name\": \"min.ploidy\", \"value\": \"0.95\", \"label\": \"Min.Ploidy\", \"description\": \"Minimum ploidy value to consider. Solutions implying lower ploidy values will be discarded\"}, {\"name\": \"max.ploidy\", \"value\": \"10\", \"label\": \"Max.Ploidy\", \"description\": \"Maximum ploidy value to consider. Solutions implying greater ploidy values will be discarded\"}, {\"name\": \"platform\", \"value\": \"SNP_6.0\", \"label\": \"Platform\", \"description\": \"The chip type used, supported values are currently 'SNP_250K_STY', 'SNP_6.0' and 'Illumina_WES'\"}, {\"name\": \"max.as.seg.count\", \"value\": \"1500\", \"label\": \"Max.As.Seg.Count\", \"description\": \"Maximum number of allelic segments. Samples with a higher segment count will be flagged as 'failed'\"}, {\"name\": \"max.neg.genome\", \"value\": \"0.005\", \"label\": \"Max.Neg.Genome\", \"description\": \"Maximum genome fraction that may be modeled as non-clonal with copy-ratio below that of clonal homozygous deletion. Solutions implying greater values will be discarded.\"}, {\"name\": \"max.non.clonal\", \"value\": \"0.05\", \"label\": \"Max.Non.Clonal\", \"description\": \"Maximum genome fraction that may be modeled as non-clonal (subclonal SCNA). Solutions implying greater values will be discarded.\"}, {\"name\": \"copy.number.type\", \"value\": \"allelic\", \"label\": \"Copy.Number.Type\", \"description\": \"The copy number type to assess, can be one of 'allelic' or 'total'\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"ABSOLUTE.review\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"[**Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] Extracts the absolute copy number per cancer cell from a mixed DNA population.  Use this module for the Review and Extract step in the workflow (after ABSOLUTE).\", \"required_args\": [{\"name\": \"reviewed.pp.calls.file\", \"value\": \"\", \"label\": \"Reviewed.Pp.Calls.File\", \"description\": \"The calls file from a previous run of the ABSOLUTE.summarize module that has been reviewed and annotated by an analyst.\"}, {\"name\": \"analyst.id\", \"value\": \"\", \"label\": \"Analyst.Id\", \"description\": \"The user id of the analyst who called the solutions.  This is used for display/reporting purposes only.\"}, {\"name\": \"modes.file\", \"value\": \"\", \"label\": \"Modes.File\", \"description\": \"The obj_summary.PP-modes.data.RData file produced by the previous run of the ABSOLUTE.summarize module.\"}, {\"name\": \"collection.name\", \"value\": \"\", \"label\": \"Collection.Name\", \"description\": \"A descriptive name for this collection of samples.  Use the same name as in the previous run of the ABSOLUTE.summarize module.  This is used for display/reporting purposes only.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"ABSOLUTE.review\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"copy.number.type\", \"value\": \"allelic\", \"label\": \"Copy.Number.Type\", \"description\": \"The copy number type to assess, can be one of 'allelic' or 'total'\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"ABSOLUTE.summarize\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"[**Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] Summarizes the results from multiple ABSOLUTE runs so that an analyst can manually review the solutions.\", \"required_args\": [{\"name\": \"collection.name\", \"value\": \"\", \"label\": \"Collection.Name\", \"description\": \"A descriptive name for this collection of samples.\"}, {\"name\": \"absolute.files\", \"value\": \"\", \"label\": \"Absolute.Files\", \"description\": \"A group of ABSOLUTE RData output files from individual sample runs to be summarized together.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"ABSOLUTE.summarize\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"copy.number.type\", \"value\": \"allelic\", \"label\": \"Copy.Number.Type\", \"description\": \"The copy number type to assess, can be one of 'allelic' or 'total'.\"}, {\"name\": \"plot.modes\", \"value\": \"TRUE\", \"label\": \"Plot.Modes\", \"description\": \"Set this to FALSE to disable plotting of the purity/ploidy modes.\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"Acgh2Tab\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Converts acgh files to a tab-delimited format usable by Genomica \", \"required_args\": [{\"name\": \"acgh.input.file\", \"value\": \"\", \"label\": \"Acgh.Input.File\", \"description\": \"Data in aCHG format\"}, {\"name\": \"presentlist.file\", \"value\": \"\", \"label\": \"Presentlist.File\", \"description\": \"List of samples with a certain gene signature (from Genomica).\"}, {\"name\": \"absentlist.file\", \"value\": \"\", \"label\": \"Absentlist.File\", \"description\": \"List of samples without a certain gene signature (from Genomica).\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"Acgh2Tab\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"genelocs.file\", \"value\": \"<libdir>hg18_refseq_genes.txt\", \"label\": \"Genelocs.File\", \"description\": \"File giving gene locations (from GenomeBrowser).\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"AddFCSEventIndex\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Adds indexes to events in a Flow Cytometry Standard (FCS) data file.\", \"required_args\": [{\"name\": \"Input.data.file\", \"value\": \"\", \"label\": \"Input.Data.File\", \"description\": \"An FCS data file to which you want to add indexes.\"}, {\"name\": \"Index.parameter.long.name\", \"value\": \"\", \"label\": \"Index.Parameter.Long.Name\", \"description\": \"The long name ($PnS keyword) of the parameter to get the index; the default is blank.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"AddFCSEventIndex\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"Start.index\", \"value\": \"1\", \"label\": \"Start.Index\", \"description\": \"The index number for the first event in the data set; the default is 1. The index will be incremented by 1 for each of the events in the data set. \"}, {\"name\": \"Index.parameter.short.name\", \"value\": \"Index\", \"label\": \"Index.Parameter.Short.Name\", \"description\": \"The short name ($PnN keyword) of the parameter to get the index; the default is \\\"Index.\\\" \"}, {\"name\": \"Output.fcs.file.name\", \"value\": \"<Input.data.file_basename>\", \"label\": \"Output.Fcs.File.Name\", \"description\": \"The output file name.\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"AddFCSParameter\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Add parameters and their values to a FCS data file\", \"required_args\": [{\"name\": \"Input.fcs.data.file\", \"value\": \"\", \"label\": \"Input.Fcs.Data.File\", \"description\": \"A FCS data file to add parameters to.\"}, {\"name\": \"Input.csv.file\", \"value\": \"\", \"label\": \"Input.Csv.File\", \"description\": \"A CSV file with parameters and values to add.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"AddFCSParameter\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"Output.file.name\", \"value\": \"<Input.fcs.data.file_basename>_with_extra_params.fcs\", \"label\": \"Output.File.Name\", \"description\": \"The output FCS file name.\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"AddNoiseToFCS\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Add noise to specified parameters in an FCS data file.\", \"required_args\": [{\"name\": \"Input.FCS.data.file\", \"value\": \"\", \"label\": \"Input.FCS.Data.File\", \"description\": \"An FCS data file to add noise to.\"}, {\"name\": \"Parameters.to.add.noise.to\", \"value\": \"\", \"label\": \"Parameters.To.Add.Noise.To\", \"description\": \"List of parameters to which the module will add noise. The list may be provided as a list of indexes (e.g., \\\"1-3,5,7\\\") or as a list of parameter names (i.e., values of the $PnN keyword, such as \\\"FSC-H,FL1-H,FL2-A\\\"). By default, the noise will be added to all parameters but the Time parameter.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"AddNoiseToFCS\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"Output.FCS.file.name\", \"value\": \"<Input.FCS.data.file_basename>.with_noise.fcs\", \"label\": \"Output.FCS.File.Name\", \"description\": \"The output FCS file name.\"}, {\"name\": \"Noise.level\", \"value\": \"0.5\", \"label\": \"Noise.Level\", \"description\": \"Noise level in percent. This is the maximum noise that will be added to an event based on the current event parameter value or on the maximum range of the appropriate parameter. The latter option will be effective if the Uniform noise option is selected.\"}, {\"name\": \"Noise\", \"value\": \"Proportional\", \"label\": \"Noise\", \"description\": \"How to add the noise: Proportional (The maximum amount of noise added to each value will be based on the value itself, e.g., low values will get lower amount of noise.) or Uniform (The maximum amount of noise added to each value is only dependent on the range of the particular parameter).\"}, {\"name\": \"Saturated.events\", \"value\": \"RemoveSaturated\", \"label\": \"Saturated.Events\", \"description\": \"Whether to remove saturated events. An event is considered saturated if one or more of its parameter values are closer than 0.1 percent from the maximum range for that parameter (i.e., value of the appropriate $PnR keyword).\"}, {\"name\": \"Random.Seed\", \"value\": \"1234567\", \"label\": \"Random.Seed\", \"description\": \"The seed to use for the random generator. Leave this blank if want to use a randomly selected seed.\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"AffySTExpressionFileCreator\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"[Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] Creates a GCT file from a set of CEL files from Affymetrix ST arrays.\", \"required_args\": [{\"name\": \"input.file\", \"value\": \"\", \"label\": \"Input.File\", \"description\": \"One or more Affymetrix ST CEL files either uploaded directly, packaged into a ZIP or TAR bundle, or supplied through a directory input.  The CEL files can be in GZ format and the TAR can be in GZ, XZ, or BZ2 format.  The parameter will accept multiple inputs in any of these forms. \"}, {\"name\": \"clm.file\", \"value\": \"\", \"label\": \"Clm.File\", \"description\": \"A tab-delimited text file containing one scan, sample, and class per line\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"AffySTExpressionFileCreator\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"normalize\", \"value\": \"yes\", \"label\": \"Normalize\", \"description\": \"Whether to normalize data using quantile normalization\"}, {\"name\": \"background.correct\", \"value\": \"yes\", \"label\": \"Background.Correct\", \"description\": \"Whether to perform background correction\"}, {\"name\": \"qc.plot.format\", \"value\": \"skip\", \"label\": \"Qc.Plot.Format\", \"description\": \"Choose a file format for the QC plots (or skip them completely)\"}, {\"name\": \"annotate.rows\", \"value\": \"no\", \"label\": \"Annotate.Rows\", \"description\": \"Whether to annotate rows with the gene symbol and description\"}, {\"name\": \"output.file.base\", \"value\": \"<job_id>.expression.gct\", \"label\": \"Output.File.Base\", \"description\": \"The base name of the output file(s). File extensions will be added automatically.\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"ApplyGatingML\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Apply a Gating-ML file on an FCS data file (gate and/or transform list mode data) \", \"required_args\": [{\"name\": \"Input.FCS.data.file\", \"value\": \"\", \"label\": \"Input.FCS.Data.File\", \"description\": \"An FCS data file to be gated.\"}, {\"name\": \"Input.Gating-ML.file\", \"value\": \"\", \"label\": \"Input.Gating-ML.File\", \"description\": \"A Gating-ML file specifying the gates.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"ApplyGatingML\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"Output.base.name\", \"value\": \"<Input.FCS.data.file_basename>\", \"label\": \"Output.Base.Name\", \"description\": \"Base for output file names\"}, {\"name\": \"Output.figures\", \"value\": \"none\", \"label\": \"Output.Figures\", \"description\": \"Also output figures of the gates\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"ARACNE\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Runs the ARACNE algorithm for reverse engineering cellular networks\", \"required_args\": [{\"name\": \"dataset.file\", \"value\": \"\", \"label\": \"Dataset.File\", \"description\": \"Dataset file - .res, .gct\"}, {\"name\": \"hub.gene\", \"value\": \"\", \"label\": \"Hub.Gene\", \"description\": \"The name of a gene whose network interactions you want to reconstruct\"}, {\"name\": \"hub.genes.file\", \"value\": \"\", \"label\": \"Hub.Genes.File\", \"description\": \"A file containing a list of hub genes with one gene per line - .txt\"}, {\"name\": \"transcription.factor.file\", \"value\": \"\", \"label\": \"Transcription.Factor.File\", \"description\": \"A file containing a list of genes with one gene per line that encode transcription factors - .txt\"}, {\"name\": \"kernel.width\", \"value\": \"\", \"label\": \"Kernel.Width\", \"description\": \"The kernel width (or window width) of the Gaussian Kernel Estimator\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"ARACNE\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"mi.threshold\", \"value\": \"0\", \"label\": \"Mi.Threshold\", \"description\": \"Threshold for the mutual information (MI) estimate to be considered statistically significant\"}, {\"name\": \"p.value\", \"value\": \"1\", \"label\": \"P.Value\", \"description\": \"Threshold for the p-value of a MI estimate to be considered statistically significant\"}, {\"name\": \"dpi.tolerance\", \"value\": \"1\", \"label\": \"Dpi.Tolerance\", \"description\": \"The percentage of MI estimation considered as sampling error\"}, {\"name\": \"mean.filter\", \"value\": \"0\", \"label\": \"Mean.Filter\", \"description\": \"Filter out non-informative genes whose mean expression value is smaller than mean filter\"}, {\"name\": \"cv.filter\", \"value\": \"0\", \"label\": \"Cv.Filter\", \"description\": \"Filter out non-informative genes whose coefficient of variance is smaller than cv filter\"}, {\"name\": \"output.file\", \"value\": \"<dataset.file_basename>.output.adj\", \"label\": \"Output.File\", \"description\": \"Output file name\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"AreaChange\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Calculates fraction of area under the spectrum that is attributable to signal (area after noise removal / original area)\", \"required_args\": [{\"name\": \"spectrum.filename\", \"value\": \"\", \"label\": \"Spectrum.Filename\", \"description\": \"spectrum data file - .csv\"}, {\"name\": \"low.Da\", \"value\": \"\", \"label\": \"Low.Da\", \"description\": \"minimum M/Z to include\"}, {\"name\": \"high.Da\", \"value\": \"\", \"label\": \"High.Da\", \"description\": \"maximum M/Z to include\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"AreaChange\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"quality.threshold\", \"value\": \"0.45\", \"label\": \"Quality.Threshold\", \"description\": \"discard spectra with quality < quality threshold\"}, {\"name\": \"window.filter\", \"value\": \"5\", \"label\": \"Window.Filter\", \"description\": \"window filter size\"}, {\"name\": \"factor\", \"value\": \"3\", \"label\": \"Factor\", \"description\": \"factor to multiply standard deviation by\"}, {\"name\": \"smooth\", \"value\": \"median\", \"label\": \"Smooth\", \"description\": \"averaging function\"}, {\"name\": \"output.file\", \"value\": \"<spectrum.filename_basename>.area.odf\", \"label\": \"Output.File\", \"description\": \"output file\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"Arff2Gct\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Convert an .arff file into a gene pattern .gct / .cls file pair\", \"required_args\": [{\"name\": \"arff.file\", \"value\": \"\", \"label\": \"Arff.File\", \"description\": \"The dataset in ARFF format.  Multi-label class names are separated by colons (:). - .arff\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"Arff2Gct\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.file\", \"value\": \"<arff.file_basename>\", \"label\": \"Output.File\", \"description\": \"Name to give the .gct containing dataset features and .cls containing dataset class labels output files.\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"ATARiS\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Runs ATARiS on RNAi reagent-level data.\", \"required_args\": [{\"name\": \"gct.file\", \"value\": \"\", \"label\": \"Gct.File\", \"description\": \"input GCT file. Unique reagent identifier in first column, gene symbol in second.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"ATARiS\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"identifier\", \"value\": \"<gct.file_basename>\", \"label\": \"Identifier\", \"description\": \"text string to identify the current run\"}, {\"name\": \"random.seed\", \"value\": \"12345\", \"label\": \"Random.Seed\", \"description\": \"seed for random number generator\"}, {\"name\": \"null.significance\", \"value\": \"0.15\", \"label\": \"Null.Significance\", \"description\": \"gene solution significance level (1 - null percentile)\"}, {\"name\": \"min.A.value\", \"value\": \"0.3\", \"label\": \"Min.A.Value\", \"description\": \"minimum effect magnitude for RNAi reagent (relative to reagent with maximal effect)\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"AuDIT\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Automated Detection of Inaccurate and Imprecise Transitions in MRM Mass Spectrometry\", \"required_args\": [{\"name\": \"data.file\", \"value\": \"\", \"label\": \"Data.File\", \"description\": \"input file with pre-processed MRM data in csv format\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"AuDIT\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"skyline.export\", \"value\": \"TRUE\", \"label\": \"Skyline.Export\", \"description\": \"if \\\"yes\\\", data file is a Skyline csv export including following columns: Sample (usually derived from SampleName, having identical value for all its replicates), PeptideSequence, ReplicateName, FragmentIon, PrecursorCharge, ProductCharge, light Area, heavy Area. When \\\"no\\\", input must contain sample, replicate, peptide, transition.id, area, IS.area in that order.\"}, {\"name\": \"pvalue.threshold\", \"value\": \"0.00001\", \"label\": \"Pvalue.Threshold\", \"description\": \"p-value threshold above which transition is quantification-worthy. Must be a value between 0 and 1 inclusive.\"}, {\"name\": \"cv.threshold\", \"value\": \"0.2\", \"label\": \"Cv.Threshold\", \"description\": \"threshold for coefficient of variation below which transition is quantification-worthy\"}, {\"name\": \"all.pairs\", \"value\": \"FALSE\", \"label\": \"All.Pairs\", \"description\": \"calculate all possible pairs of relative ratios for a transition\"}, {\"name\": \"output.intermediate.results\", \"value\": \"FALSE\", \"label\": \"Output.Intermediate.Results\", \"description\": \"create files with intermediate results\"}, {\"name\": \"output.prefix\", \"value\": \"<data.file_basename>-results\", \"label\": \"Output.Prefix\", \"description\": \"file name prefix used for output file(s)\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"BedToGtf\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Converts BED files to GFF or GTF format\", \"required_args\": [{\"name\": \"bed.file\", \"value\": \"\", \"label\": \"Bed.File\", \"description\": \"A bed file\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"BedToGtf\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.format\", \"value\": \"gtf\", \"label\": \"Output.Format\", \"description\": \"Whether to output a gff file or exon only gtf file. The bed file must contain at least 12 columns in order to generate a gtf file.\"}, {\"name\": \"output.prefix\", \"value\": \"<bed.file_basename>\", \"label\": \"Output.Prefix\", \"description\": \"The prefix for the output file\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"Beroukhim.Getz.2007.PNAS.Glioma.GISTIC\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"\", \"required_args\": [], \"default_args\": [{\"name\": \"task_name\", \"value\": \"Beroukhim.Getz.2007.PNAS.Glioma.GISTIC\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"BlastTrainTest\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Sequence similarity classification using BLAST\", \"required_args\": [{\"name\": \"train.file\", \"value\": \"\", \"label\": \"Train.File\", \"description\": \"The training file in .FASTA format. Multi-label class names are separated by colons (:).\"}, {\"name\": \"test.file\", \"value\": \"\", \"label\": \"Test.File\", \"description\": \"The testing file in .FASTA format. Multi-label class names are separated by colons (:).\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"BlastTrainTest\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"sub.matrix\", \"value\": \"PAM30\", \"label\": \"Sub.Matrix\", \"description\": \"The name of the substitution matrix used by Blast.\"}, {\"name\": \"pred.results.file\", \"value\": \"<test.file_basename>\", \"label\": \"Pred.Results.File\", \"description\": \"The name of the output file for prediction results.\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"BlastXValidation\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Sequence similarity cross validation prediction using BLAST\", \"required_args\": [{\"name\": \"input.file\", \"value\": \"\", \"label\": \"Input.File\", \"description\": \"The input sequence and annotation information in .FASTA format.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"BlastXValidation\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"num.folds\", \"value\": \"2\", \"label\": \"Num.Folds\", \"description\": \"The number of folds should be an integer greater than or equal to two.\"}, {\"name\": \"sub.matrix\", \"value\": \"PAM30\", \"label\": \"Sub.Matrix\", \"description\": \"The name of the substitution matrix used by Blast.\"}, {\"name\": \"pred.results.file\", \"value\": \"<input.file_basename>\", \"label\": \"Pred.Results.File\", \"description\": \"The name of the output file for prediction results.\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"Bowtie.aligner\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Bowtie2 (v. 2.1.0) is an ultrafast and memory-efficient short read aligner.\", \"required_args\": [{\"name\": \"prebuilt.bowtie.index\", \"value\": \"\", \"label\": \"Prebuilt.Bowtie.Index\", \"description\": \"A prebuilt Bowtie 2 index. Either a prebuilt or a custom Bowtie 2 index must be specified.\"}, {\"name\": \"custom.bowtie.index\", \"value\": \"\", \"label\": \"Custom.Bowtie.Index\", \"description\": \"A ZIP archive containing Bowtie 2 index files. Either a prebuilt or a custom Bowtie 2 index must be specified.\"}, {\"name\": \"input.format\", \"value\": \"\", \"label\": \"Input.Format\", \"description\": \"The format of the reads input files.\"}, {\"name\": \"reads.pair.1\", \"value\": \"\", \"label\": \"Reads.Pair.1\", \"description\": \"Unpaired reads file or first mate for paired reads. A file or zip of files containing reads in FASTA or FASTQ format (can be compressed - ie .gz).\"}, {\"name\": \"reads.pair.2\", \"value\": \"\", \"label\": \"Reads.Pair.2\", \"description\": \"Second mate for paired reads. A file or zip of files in FASTA or FASTQ format (can be compressed - ie .gz).\"}, {\"name\": \"quality.value.scale\", \"value\": \"\", \"label\": \"Quality.Value.Scale\", \"description\": \"The scale to use for quality values.\"}, {\"name\": \"integer.quality.values\", \"value\": \"\", \"label\": \"Integer.Quality.Values\", \"description\": \"Whether the quality values are represented as space-separated ASCII integers (i.e 40 40 30 40 ..).\"}, {\"name\": \"max.reads.to.align\", \"value\": \"\", \"label\": \"Max.Reads.To.Align\", \"description\": \"The maximum number of reads to align.\"}, {\"name\": \"mate.orientations\", \"value\": \"\", \"label\": \"Mate.Orientations\", \"description\": \"The upstream/downstream mate orientations for a valid paired-end alignment against the forward reference strand.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"Bowtie.aligner\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"alignment.mode\", \"value\": \"--end-to-end\", \"label\": \"Alignment.Mode\", \"description\": \"The alignment mode to use.\"}, {\"name\": \"preset.options\", \"value\": \"sensitive\", \"label\": \"Preset.Options\", \"description\": \"A combination of pre-packaged options based on speed and sensitivity/accuracy.\"}, {\"name\": \"min.fragment.length\", \"value\": \"0\", \"label\": \"Min.Fragment.Length\", \"description\": \"The minimum fragment length for valid paired-end alignments.\"}, {\"name\": \"max.fragment.length\", \"value\": \"500\", \"label\": \"Max.Fragment.Length\", \"description\": \"The maximum fragment length for valid paired-end alignments.\"}, {\"name\": \"random.seed\", \"value\": \"123456\", \"label\": \"Random.Seed\", \"description\": \"The seed to use for the pseudo-random number generator.\"}, {\"name\": \"output.prefix\", \"value\": \"<reads.pair.1_basename>_<reads.pair.2_basename>\", \"label\": \"Output.Prefix\", \"description\": \"The prefix to use for the output file name.\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"Bowtie.indexer\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Builds a Bowtie2 (v. 2.1.0) index from a set of DNA sequences\", \"required_args\": [{\"name\": \"fasta.file\", \"value\": \"\", \"label\": \"Fasta.File\", \"description\": \"A file or zip of files containing sequences in FASTA format\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"Bowtie.indexer\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"random.seed\", \"value\": \"123456\", \"label\": \"Random.Seed\", \"description\": \"The seed for the random number generator\"}, {\"name\": \"index.name\", \"value\": \"<fasta.file_basename>\", \"label\": \"Index.Name\", \"description\": \"The base name of the index files\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"BWA.aln\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"[**Beta Release** Contact gp-help with any issues.  Check stdout.txt and stderr.txt for errors] A fast and accurate short-read alignment tool that allows for mismatches and gaps. Alignments are output in a SAM format file, which provides Phred-scale quality scores for each alignment.  Uses BWA 0.7.4 aln and samse/sampe.\", \"required_args\": [{\"name\": \"BWA.index\", \"value\": \"\", \"label\": \"BWA.Index\", \"description\": \"A zip file or directory containing BWA index files.\"}, {\"name\": \"reads.pair.1\", \"value\": \"\", \"label\": \"Reads.Pair.1\", \"description\": \"Single-end or first paired-end reads file in FASTA, FASTQ, or BAM format.  For paired-end data, this should be the forward (\\\"*_1\\\" or \\\"left\\\") input file.  Note: the FASTA or FASTQ can be gzipped.\"}, {\"name\": \"reads.pair.2\", \"value\": \"\", \"label\": \"Reads.Pair.2\", \"description\": \"The reverse (\\\"*_2\\\" or \\\"right\\\") input file for paired-end reads in FASTA, FASTQ, or BAM format. Note: the FASTA or FASTQ can be gzipped.\"}, {\"name\": \"bam.mapping\", \"value\": \"\", \"label\": \"Bam.Mapping\", \"description\": \"Specifies how to map BAM input. This is only required if input file is in BAM format.\"}, {\"name\": \"max.edit.distance\", \"value\": \"\", \"label\": \"Max.Edit.Distance\", \"description\": \"The max edit distance. This specifies a threshold of the maximum number of deletions, insertions, and substitutions needed to transform the reference sequence into the read sequence.\"}, {\"name\": \"max.num.gap\", \"value\": \"\", \"label\": \"Max.Num.Gap\", \"description\": \"Maximum number of gap opens. This specifies a threshold of the maximum number of gaps that can be initiated to match a given read to the reference.\"}, {\"name\": \"max.gap.extension\", \"value\": \"\", \"label\": \"Max.Gap.Extension\", \"description\": \"Maximum number of gap extensions. This specifies a threshold of the maximum number of bases by which gaps in a read can be extended.\"}, {\"name\": \"max.deletion.length\", \"value\": \"\", \"label\": \"Max.Deletion.Length\", \"description\": \"Disallow a long deletion within this many bp of the 3' end.\"}, {\"name\": \"max.indel.length\", \"value\": \"\", \"label\": \"Max.Indel.Length\", \"description\": \"Disallow an indel within this many bp of the ends\"}, {\"name\": \"seed.length\", \"value\": \"\", \"label\": \"Seed.Length\", \"description\": \"The set of bases determined by this option in the high-quality (left) end of the read is the seed.\"}, {\"name\": \"max.seed.edit.distance\", \"value\": \"\", \"label\": \"Max.Seed.Edit.Distance\", \"description\": \"Maximum edit distance in the seed; that is, the maximum number of changes required to transform the reference sequence of the seed into the read sequence of the seed.\"}, {\"name\": \"mismatch.penalty\", \"value\": \"\", \"label\": \"Mismatch.Penalty\", \"description\": \"Mismatch penalty\"}, {\"name\": \"gap.open.penalty\", \"value\": \"\", \"label\": \"Gap.Open.Penalty\", \"description\": \"Gap open penalty.  The gap open penalty is the score taken away for the initiation of the gap in sequence. To make the match more significant you can try to make the gap penalty larger.\"}, {\"name\": \"gap.extension.penalty\", \"value\": \"\", \"label\": \"Gap.Extension.Penalty\", \"description\": \"Gap extension penalty.  The gap extension penalty is added to the standard gap penalty for each base or residue in the gap.  To reduce long gaps, increase the extension gap penalty.\"}, {\"name\": \"max.best.hits\", \"value\": \"\", \"label\": \"Max.Best.Hits\", \"description\": \"Proceed with suboptimal alignments if there are no more than this many equally best hits. This option only affects paired-end mapping.\"}, {\"name\": \"trim.reads\", \"value\": \"\", \"label\": \"Trim.Reads\", \"description\": \"Specifies a quality threshold for read trimming\"}, {\"name\": \"barcode.length\", \"value\": \"\", \"label\": \"Barcode.Length\", \"description\": \"Length of barcode starting from the 5' end\"}, {\"name\": \"max.insert.size\", \"value\": \"\", \"label\": \"Max.Insert.Size\", \"description\": \"Specifies the maximum insert size for a read pair to be considered to be mapped properly\"}, {\"name\": \"max.occurrences\", \"value\": \"\", \"label\": \"Max.Occurrences\", \"description\": \"Specifies the maximum occurrences of a read for pairing\"}, {\"name\": \"max.alignments\", \"value\": \"\", \"label\": \"Max.Alignments\", \"description\": \"Maximum number of alignments to output in the XA tag for reads paired properly\"}, {\"name\": \"max.dc.alignments\", \"value\": \"\", \"label\": \"Max.Dc.Alignments\", \"description\": \"Maximum number of alignments to output in the XA tag for disconcordant read pairs (excluding singletons)\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"BWA.aln\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"iterative.search\", \"value\": \"no\", \"label\": \"Iterative.Search\", \"description\": \"Whether to disable iterative search\"}, {\"name\": \"Illumina.1.3.format\", \"value\": \"no\", \"label\": \"Illumina.1.3.Format\", \"description\": \"The input is in the Illumina 1.3+ read format\"}, {\"name\": \"output.prefix\", \"value\": \"<reads.pair.1_basename>\", \"label\": \"Output.Prefix\", \"description\": \"Prefix to use for output file name\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"BWA.bwasw\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"[**Beta Release** Contact gp-help with any issues.  Check stdout.txt and stderr.txt for errors] Aligns long sequences (>200 bp) to a sequence database using BWA 0.7.4 bwasw.\", \"required_args\": [{\"name\": \"BWA.index\", \"value\": \"\", \"label\": \"BWA.Index\", \"description\": \"A zip file or directory containing BWA index files.\"}, {\"name\": \"read.file\", \"value\": \"\", \"label\": \"Read.File\", \"description\": \"Single-end or first paired-end reads file in FASTA or FASTQ format.  For paired-end data, this should be the forward (\\\"*_1\\\" or \\\"left\\\") input file.\"}, {\"name\": \"mate.read.file\", \"value\": \"\", \"label\": \"Mate.Read.File\", \"description\": \"The reverse (\\\"*_2\\\" or \\\"right\\\") reads file for paired-end reads in FASTA or FASTQ format.\"}, {\"name\": \"match.score\", \"value\": \"\", \"label\": \"Match.Score\", \"description\": \"The score of the match\"}, {\"name\": \"mismatch.penalty\", \"value\": \"\", \"label\": \"Mismatch.Penalty\", \"description\": \"Specifies the mismatch penalty.\"}, {\"name\": \"gap.open.penalty\", \"value\": \"\", \"label\": \"Gap.Open.Penalty\", \"description\": \"Gap open penalty.  The gap open penalty is the score taken away for the initiation of the gap in sequence. To make the match more significant you can try to make the gap penalty larger.\"}, {\"name\": \"gap.extension.penalty\", \"value\": \"\", \"label\": \"Gap.Extension.Penalty\", \"description\": \"Gap extension penalty.  The gap extension penalty is added to the standard gap penalty for each base or residue in the gap.  To reduce long gaps, increase the extension gap penalty. \"}, {\"name\": \"band.width\", \"value\": \"\", \"label\": \"Band.Width\", \"description\": \"band width in the banded alignment\"}, {\"name\": \"min.score.threshold\", \"value\": \"\", \"label\": \"Min.Score.Threshold\", \"description\": \"Minimum score threshold divided by the match score\"}, {\"name\": \"threshold.coefficient\", \"value\": \"\", \"label\": \"Threshold.Coefficient\", \"description\": \"Coefficient for threshold adjustment according to query length\"}, {\"name\": \"z.best.heuristics\", \"value\": \"\", \"label\": \"Z.Best.Heuristics\", \"description\": \"Z-best heuristics. Specifying a higher number increases accuracy at the cost of speed\"}, {\"name\": \"max.sa.interval.size\", \"value\": \"\", \"label\": \"Max.Sa.Interval.Size\", \"description\": \"Maximum SA interval size for initiating a seed. Specifying a higher number increases accuracy at the cost of speed.\"}, {\"name\": \"min.num.seeds\", \"value\": \"\", \"label\": \"Min.Num.Seeds\", \"description\": \"The minimum number of seeds contained in the best alignment from the forward-forward alignment process that allows the algorithm to skip performing the reverse alignment. BWA.bwasw tends to be faster and more accurate if the alignment is supported by more seeds.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"BWA.bwasw\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"num.threads\", \"value\": \"1\", \"label\": \"Num.Threads\", \"description\": \"Number of threads\"}, {\"name\": \"output.prefix\", \"value\": \"<read.file_basename>\", \"label\": \"Output.Prefix\", \"description\": \"Prefix to use for the output file name\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"BWA.indexer\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"[**Beta Release** Contact gp-help with any issues.  Check stdout.txt and stderr.txt for errors] Creates a BWT index from a sequence FASTA file for use with the BWA 0.7.4 aligners.\", \"required_args\": [{\"name\": \"fasta.file\", \"value\": \"\", \"label\": \"Fasta.File\", \"description\": \"A sequence fasta file\"}, {\"name\": \"algorithm\", \"value\": \"\", \"label\": \"Algorithm\", \"description\": \"The algorithm to use for constructing the BWT index\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"BWA.indexer\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.prefix\", \"value\": \"<fasta.file_basename>\", \"label\": \"Output.Prefix\", \"description\": \"A prefix for the output file name.\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"CaArray2ImportViewer\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Imports data files from CaArray 2.4.1 and creates gct or cls files \", \"required_args\": [{\"name\": \"experiment\", \"value\": \"\", \"label\": \"Experiment\", \"description\": \"Title or identifier of the caArray experiment to import data from.  https://array.nci.nih.gov/caarray/home.action\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"CaArray2ImportViewer\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"url\", \"value\": \"http://array.nci.nih.gov:8080\", \"label\": \"Url\", \"description\": \"URL for caArray server.\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"CART\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Classification and Regression Tree\", \"required_args\": [{\"name\": \"train.data.filename\", \"value\": \"\", \"label\": \"Train.Data.Filename\", \"description\": \"The training data file - .gct, .res, ignored if a saved model (saved.model.filename) is used\"}, {\"name\": \"train.cls.filename\", \"value\": \"\", \"label\": \"Train.Cls.Filename\", \"description\": \"The training class file - .cls, ignored if a saved model (saved.model.filename) is used\"}, {\"name\": \"saved.model.filename\", \"value\": \"\", \"label\": \"Saved.Model.Filename\", \"description\": \"input CART model\"}, {\"name\": \"test.data.filename\", \"value\": \"\", \"label\": \"Test.Data.Filename\", \"description\": \"The test data file - .gct, .res\"}, {\"name\": \"test.cls.filename\", \"value\": \"\", \"label\": \"Test.Cls.Filename\", \"description\": \"The test class file - .cls\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"CART\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"model.output.file\", \"value\": \"<train.data.filename_basename>.cart.model\", \"label\": \"Model.Output.File\", \"description\": \"name of output CART model file  \"}, {\"name\": \"pred.results.file\", \"value\": \"<test.data.filename_basename>.pred.odf\", \"label\": \"Pred.Results.File\", \"description\": \"The name of the output file for prediction results\"}, {\"name\": \"tree.output.file\", \"value\": \"<test.data.filename_basename>.tree.pdf\", \"label\": \"Tree.Output.File\", \"description\": \"The name of the file containing a plot of the classification tree\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"CARTXValidation\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Classification and Regression Tree Cross-Validation\", \"required_args\": [{\"name\": \"data.filename\", \"value\": \"\", \"label\": \"Data.Filename\", \"description\": \"The data file - .gct, .res\"}, {\"name\": \"cls.filename\", \"value\": \"\", \"label\": \"Cls.Filename\", \"description\": \"The class file - .cls \"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"CARTXValidation\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"prediction.results.file\", \"value\": \"<data.filename_basename>.pred.odf\", \"label\": \"Prediction.Results.File\", \"description\": \"The name of the output file for prediction results\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"CBS\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Segments DNA copy number data into regions of estimated equal copy number using circular binary segmentation (CBS). \", \"required_args\": [{\"name\": \"input.file\", \"value\": \"\", \"label\": \"Input.File\", \"description\": \"Copy number file - .cn\"}, {\"name\": \"sample.name\", \"value\": \"\", \"label\": \"Sample.Name\", \"description\": \"The name of the sample to use. Leave blank to use the first sample in the input file.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"CBS\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"permutations\", \"value\": \"10000\", \"label\": \"Permutations\", \"description\": \"Number of permutations used for p-value computation\"}, {\"name\": \"significance.level\", \"value\": \"0.01\", \"label\": \"Significance.Level\", \"description\": \"Significance levels for the test to accept change-points\"}, {\"name\": \"output.prefix\", \"value\": \"<input.file_basename>\", \"label\": \"Output.Prefix\", \"description\": \"The prefix to use for the output file name\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"ChIPSeq.CreateHeatmap\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Generates a heatmap based on the ChIP-Seq signal extracted from a BAM file, according to the chromosomal locations specified in the BED file used in ExtractSignal.  This is an experimental module built for the Broad CEGS workshop of July 29-30, 2013.\", \"required_args\": [{\"name\": \"input.file\", \"value\": \"\", \"label\": \"Input.File\", \"description\": \"Input file in TAB format, generated by ChipSeq.ExtractSignal\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"ChIPSeq.CreateHeatmap\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"normalization\", \"value\": \"z-score\", \"label\": \"Normalization\", \"description\": \"Choose a normalization method\"}, {\"name\": \"percent.start\", \"value\": \"0.45\", \"label\": \"Percent.Start\", \"description\": \"percent of the bed region to start clustering at\"}, {\"name\": \"percent.end\", \"value\": \"0.55\", \"label\": \"Percent.End\", \"description\": \"percent of the bed region to end clustering at \"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"ChIPSeq.CreatePlots\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Generates a scatter plot and correlation matrix based on the ChIP-Seq signal extracted from a BAM file, according to the chromosomal locations specified in the BED file used in ExtractSignal.  This is an experimental module built for the Broad CEGS workshop of July 29-30, 2013.\", \"required_args\": [{\"name\": \"input.file\", \"value\": \"\", \"label\": \"Input.File\", \"description\": \"Input file in TAB format, generated by ChipSeq.ExtractSignal\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"ChIPSeq.CreatePlots\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"ChIPSeq.ExtractSignal\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Segment the read from a BAM file into bins by chromosome and extract the signal values to a tab delimited file for subsequent visualization.  This is an experimental module built for the Broad CEGS workshop of July 29-30, 2013.\", \"required_args\": [{\"name\": \"bam.file\", \"value\": \"\", \"label\": \"Bam.File\", \"description\": \"Input file in BAM format.\"}, {\"name\": \"chromosome.sizes.file\", \"value\": \"\", \"label\": \"Chromosome.Sizes.File\", \"description\": \"Specify a chromosome sizes file.  Chromosome names must start with 'chr'.\"}, {\"name\": \"bin.size\", \"value\": \"\", \"label\": \"Bin.Size\", \"description\": \"The desired size of the bins.  Defaults to 25 if no value is supplied.\"}, {\"name\": \"whole.cell.extract.control.file\", \"value\": \"\", \"label\": \"Whole.Cell.Extract.Control.File\", \"description\": \"A bam file containing Whole Cell Extract control data to use in normalizing the bin values.\"}, {\"name\": \"region.bed.file\", \"value\": \"\", \"label\": \"Region.Bed.File\", \"description\": \"The BED file containing regions of interest.\"}, {\"name\": \"region.bin.size\", \"value\": \"\", \"label\": \"Region.Bin.Size\", \"description\": \"Bin the bed regions into sub-regions of this size (in bps).  Must be a multiple of the bin size such that an integer number of bin file bins greater than zero fits into a single region bin. (E.g. One could have bin files of 25 bp, extend bed regions by 100 bp, then have region bins of 50 bp for a total of four region bins.). Defaults to bin.size if no value is supplied.\"}, {\"name\": \"extend.by\", \"value\": \"\", \"label\": \"Extend.By\", \"description\": \"The number of basepairs to extend a region from its center.  Total extended region size will be twice this value.  This option is most useful if one wants to analyze  regions of the same length and number of bins.  To simply use the bin boundaries nearest the BED coordinates (i.e. analyze variable numbers of bins per BED line), do not specify this parameter.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"ChIPSeq.ExtractSignal\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"assembly.name\", \"value\": \"hg19\", \"label\": \"Assembly.Name\", \"description\": \"Specify an assembly (i.e. hg19, hg18, etc.).\"}, {\"name\": \"normalize\", \"value\": \"yes\", \"label\": \"Normalize\", \"description\": \"Normalize each bin value by scaling it to 20000000 reads.\"}, {\"name\": \"aggregation.function\", \"value\": \"avg\", \"label\": \"Aggregation.Function\", \"description\": \"Function to run on the bin values over the regions of interest.\"}, {\"name\": \"output.filename\", \"value\": \"<bam.file_basename>.<region.bed.file_basename>.tab\", \"label\": \"Output.Filename\", \"description\": \"The name of the output file.\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"ClassNeighbors\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Select genes that most closely resemble a profile\", \"required_args\": [{\"name\": \"data.filename\", \"value\": \"\", \"label\": \"Data.Filename\", \"description\": \"Dataset file - .res, .gct, .odf\"}, {\"name\": \"class.filename\", \"value\": \"\", \"label\": \"Class.Filename\", \"description\": \"class file - .cls, .odf\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"ClassNeighbors\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"marker.gene.list.file\", \"value\": \"<data.filename_basename>.genelist\", \"label\": \"Marker.Gene.List.File\", \"description\": \"output filename for marker gene list results - .odf file\"}, {\"name\": \"marker.data.set.file\", \"value\": \"<data.filename_basename>.neighbors\", \"label\": \"Marker.Data.Set.File\", \"description\": \"output filename for marker data - .gct file\"}, {\"name\": \"num.neighbors\", \"value\": \"50\", \"label\": \"Num.Neighbors\", \"description\": \"number of neighbors to find\"}, {\"name\": \"num.permutations\", \"value\": \"100\", \"label\": \"Num.Permutations\", \"description\": \"number of permutations in permutation test\"}, {\"name\": \"user.pval\", \"value\": \"0.5\", \"label\": \"User.Pval\", \"description\": \"user-set significance value for permutation test\"}, {\"name\": \"mean.or.median\", \"value\": \"mean\", \"label\": \"Mean.Or.Median\", \"description\": \"use mean or median for feature selection\"}, {\"name\": \"ttest.or.snr\", \"value\": \"t-test\", \"label\": \"Ttest.Or.Snr\", \"description\": \"use signal-to-noise or t-test to select neighbors\"}, {\"name\": \"filter.data\", \"value\": \"no\", \"label\": \"Filter.Data\", \"description\": \"if no, values below will be ignored\"}, {\"name\": \"min.threshold\", \"value\": \"10\", \"label\": \"Min.Threshold\", \"description\": \"minimum threshold for data\"}, {\"name\": \"max.threshold\", \"value\": \"16000\", \"label\": \"Max.Threshold\", \"description\": \"maximum threshold for data\"}, {\"name\": \"min.fold.diff\", \"value\": \"5\", \"label\": \"Min.Fold.Diff\", \"description\": \"minimum fold difference for filtering genes\"}, {\"name\": \"min.abs.diff\", \"value\": \"50\", \"label\": \"Min.Abs.Diff\", \"description\": \"minimum absolute difference for filtering genes\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"ClsFileCreator\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"A tool to create a class label (CLS) file.\", \"required_args\": [{\"name\": \"input.file\", \"value\": \"\", \"label\": \"Input.File\", \"description\": \"A dataset - .gct\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"ClsFileCreator\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"CNToXCN\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"\", \"required_args\": [{\"name\": \"cn.file\", \"value\": \"\", \"label\": \"Cn.File\", \"description\": \"Input CN file (a tab-delimited file format that contains SNP copy numbers)\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"CNToXCN\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.prefix\", \"value\": \"<cn.file_basename>\", \"label\": \"Output.Prefix\", \"description\": \"Prefix for the output XCN file name\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"CoClustering\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Co-clustering of RNA and global proteome data\", \"required_args\": [], \"default_args\": [{\"name\": \"task_name\", \"value\": \"CoClustering\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"min.correlation\", \"value\": \"0.4\", \"label\": \"Min.Correlation\", \"description\": \"Minimum correlation between RNA and proteome for gene to be included in clustering\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"CollapseDataset\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Collapses all probe set values for a gene into a single vector of values\", \"required_args\": [{\"name\": \"dataset.file\", \"value\": \"\", \"label\": \"Dataset.File\", \"description\": \"Dataset file - .gct, .res\"}, {\"name\": \"chip.platform\", \"value\": \"\", \"label\": \"Chip.Platform\", \"description\": \"The chip platform\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"CollapseDataset\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"collapse.mode\", \"value\": \"Maximum\", \"label\": \"Collapse.Mode\", \"description\": \"Collapse mode for probe sets => 1 gene\"}, {\"name\": \"output.file.name\", \"value\": \"<dataset.file_basename>.collapsed\", \"label\": \"Output.File.Name\", \"description\": \"The output file name\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"ComBat\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Batch Correction\", \"required_args\": [{\"name\": \"input.file\", \"value\": \"\", \"label\": \"Input.File\", \"description\": \"input file - .res, .gct\"}, {\"name\": \"sample.info.file\", \"value\": \"\", \"label\": \"Sample.Info.File\", \"description\": \"A file containing information about the samples. The sample info file must contain the three columns: Array, Sample, and Batch. Any additional columns are used as covariates in the analysis.(.txt file)\"}, {\"name\": \"absent.calls.filter\", \"value\": \"\", \"label\": \"Absent.Calls.Filter\", \"description\": \"The filter to apply to genes with absent calls (a value between 0 and 1). Filters genes with absent calls in 1-(absent calls filter) of the samples. Ignored if input file is not a .res file.\\t\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"ComBat\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"covariate.columns\", \"value\": \"all\", \"label\": \"Covariate.Columns\", \"description\": \"The covariate columns to use in analysis. This is either set to all, none, or a list specifying one or more covariate columns from sample info file to use in the analysis (i.e, 4, 5, 7, ...). If this is set to 'all' then all the covariate columns are used. If this is set to 'none' then none of the covariate columns are used.\"}, {\"name\": \"create.prior.plots\", \"value\": \"T\", \"label\": \"Create.Prior.Plots\", \"description\": \"Whether to generate prior probability distribution plots. Ignored if non-parametric method is used.\"}, {\"name\": \"prior.method\", \"value\": \"T\", \"label\": \"Prior.Method\", \"description\": \"Empirical Bayes priors distribution estimation method to use. Either a parametric or non-parametric method can be applied. If unsure of which method to use try the parametric method first since it is faster.\"}, {\"name\": \"output.file\", \"value\": \"<input.file_basename>.combat.<input.file_extension>\", \"label\": \"Output.File\", \"description\": \"Name of batch adjusted output file\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"CombineOdf\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Combine two prediction results files into a single, weighted, multi-label prediction results file\", \"required_args\": [{\"name\": \"pred1.file\", \"value\": \"\", \"label\": \"Pred1.File\", \"description\": \"The first prediction file in .odf format.\"}, {\"name\": \"pred2.file\", \"value\": \"\", \"label\": \"Pred2.File\", \"description\": \"The second prediction file in .odf format.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"CombineOdf\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"pred1.file.weight\", \"value\": \"0.5\", \"label\": \"Pred1.File.Weight\", \"description\": \"Weight assigned to classifier that produced first prediction file.\"}, {\"name\": \"pred.file\", \"value\": \"<pred1.file_basename>_<pred2.file_basename>\", \"label\": \"Pred.File\", \"description\": \"The name of the prediction file in modified .odf format.\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"ComparativeMarkerSelection\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Identify differentially expressed genes that can discriminate between distinct classes of samples.\", \"required_args\": [{\"name\": \"input.file\", \"value\": \"\", \"label\": \"Input.File\", \"description\": \"The input file - .res, .gct. Note that if your data is log transformed, you will need to set the \\\"log transformed data\\\" parameter below to \\\"yes\\\".\"}, {\"name\": \"cls.file\", \"value\": \"\", \"label\": \"Cls.File\", \"description\": \"The class file - .cls\"}, {\"name\": \"confounding.variable.cls.file\", \"value\": \"\", \"label\": \"Confounding.Variable.Cls.File\", \"description\": \"The class file containing the confounding variable.\"}, {\"name\": \"min.std\", \"value\": \"\", \"label\": \"Min.Std\", \"description\": \"The minimum standard deviation if test statistic includes min std option\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"ComparativeMarkerSelection\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"test.direction\", \"value\": \"2\", \"label\": \"Test.Direction\", \"description\": \"The test to perform\"}, {\"name\": \"test.statistic\", \"value\": \"0\", \"label\": \"Test.Statistic\", \"description\": \"The statistic to use\"}, {\"name\": \"number.of.permutations\", \"value\": \"10000\", \"label\": \"Number.Of.Permutations\", \"description\": \"The number of permutations to perform (use 0 to calculate asymptotic p-values using the standard independent two-sample t-test)\"}, {\"name\": \"log.transformed.data\", \"value\": \"false\", \"label\": \"Log.Transformed.Data\", \"description\": \"Whether the input data has been log transformed\"}, {\"name\": \"complete\", \"value\": \"false\", \"label\": \"Complete\", \"description\": \"Whether to perform all possible permutations\"}, {\"name\": \"balanced\", \"value\": \"false\", \"label\": \"Balanced\", \"description\": \"Whether to perform balanced permutations\"}, {\"name\": \"random.seed\", \"value\": \"779948241\", \"label\": \"Random.Seed\", \"description\": \"The seed of the random number generator used to produce permutations\"}, {\"name\": \"smooth.p.values\", \"value\": \"true\", \"label\": \"Smooth.P.Values\", \"description\": \"Whether to smooth p-values\"}, {\"name\": \"phenotype.test\", \"value\": \"one versus all\", \"label\": \"Phenotype.Test\", \"description\": \"Tests to perform when cls file has more than two classes (Note: not fully corrected for multiple hypothesis testing)\"}, {\"name\": \"output.filename\", \"value\": \"<input.file_basename>.comp.marker.odf\", \"label\": \"Output.Filename\", \"description\": \"The name of the output file\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"ComparativeMarkerSelectionViewer\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Views the results from ComparativeMarkerSelection\", \"required_args\": [{\"name\": \"comparative.marker.selection.filename\", \"value\": \"\", \"label\": \"Comparative.Marker.Selection.Filename\", \"description\": \"The output from ComparativeMarkerSelection\"}, {\"name\": \"dataset.filename\", \"value\": \"\", \"label\": \"Dataset.Filename\", \"description\": \"The dataset file used as input to ComparativeMarkerSelection\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"ComparativeMarkerSelectionViewer\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"CompareSpectra\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Compares two spectra to determine similarity\", \"required_args\": [{\"name\": \"spectrum1.filename\", \"value\": \"\", \"label\": \"Spectrum1.Filename\", \"description\": \"Spectrum file -.csv\"}, {\"name\": \"spectrum2.filename\", \"value\": \"\", \"label\": \"Spectrum2.Filename\", \"description\": \"Spectrum file - .csv\"}, {\"name\": \"low.Da\", \"value\": \"\", \"label\": \"Low.Da\", \"description\": \"Minimum M/Z to include\"}, {\"name\": \"high.Da\", \"value\": \"\", \"label\": \"High.Da\", \"description\": \"Maximum M/Z to include\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"CompareSpectra\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.file\", \"value\": \"<spectrum1.filename_basename>_<spectrum2.filename_basename>_compare.odf\", \"label\": \"Output.File\", \"description\": \"Output file name - .odf\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"CompensateFCS\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Compensates an FCS data file\", \"required_args\": [{\"name\": \"FCS.data.file\", \"value\": \"\", \"label\": \"FCS.Data.File\", \"description\": \"A FCS data file to be compensated\"}, {\"name\": \"Spillover.matrix.file\", \"value\": \"\", \"label\": \"Spillover.Matrix.File\", \"description\": \"A text file with spillover matrix. Use this parameter if there is no spillover matrix in the input FCS file or to use a spillover matrix different from the matrix in the input FCS file. The value from the FCS data file will be used if this parameter is not specified.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"CompensateFCS\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"Output.file.name\", \"value\": \"<FCS.data.file_basename>.compensated\", \"label\": \"Output.File.Name\", \"description\": \"The name of the output file. The extension will match the file format specified in Output file format.\"}, {\"name\": \"Output.file.format\", \"value\": \"FCS\", \"label\": \"Output.File.Format\", \"description\": \"The file format in which to save the compensated data.\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"ConcatenateFilelist\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Concatenate all of the items in the input filelist into a single large file.\", \"required_args\": [{\"name\": \"filelist\", \"value\": \"\", \"label\": \"Filelist\", \"description\": \"a text file containing a list of file paths\"}, {\"name\": \"num.header.lines\", \"value\": \"\", \"label\": \"Num.Header.Lines\", \"description\": \"specify number of header lines so that they are output only once\"}, {\"name\": \"output.filename\", \"value\": \"\", \"label\": \"Output.Filename\", \"description\": \"The name of the output file\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"ConcatenateFilelist\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"ConsensusClustering\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Resampling-based clustering method \", \"required_args\": [{\"name\": \"input.filename\", \"value\": \"\", \"label\": \"Input.Filename\", \"description\": \"The data to be clustered (.gct, .res, .odf)\"}, {\"name\": \"cluster.by\", \"value\": \"\", \"label\": \"Cluster.By\", \"description\": \"Whether to cluster by rows/genes or columns/experiments\"}, {\"name\": \"create.heat.map\", \"value\": \"\", \"label\": \"Create.Heat.Map\", \"description\": \"Whether to create heatmaps (one for each cluster number)\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"ConsensusClustering\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"kmax\", \"value\": \"5\", \"label\": \"Kmax\", \"description\": \"Try K=2,3,...,kmax clusters (must be > 1)\"}, {\"name\": \"resampling.iterations\", \"value\": \"20\", \"label\": \"Resampling.Iterations\", \"description\": \"Number of resampling iterations\"}, {\"name\": \"seed.value\", \"value\": \"12345\", \"label\": \"Seed.Value\", \"description\": \"Random number generator seed\"}, {\"name\": \"clustering.algorithm\", \"value\": \"HIERARCHICAL\", \"label\": \"Clustering.Algorithm\", \"description\": \"Type of clustering algorithm\"}, {\"name\": \"distance.measure\", \"value\": \"EUCLIDEAN\", \"label\": \"Distance.Measure\", \"description\": \"Distance measure\"}, {\"name\": \"resample\", \"value\": \"subsample\", \"label\": \"Resample\", \"description\": \"resampling scheme (one of 'subsample[ratio]', 'features[nfeat]', 'nosampling'). The default is subsampling with a proportion of 0.80. NOTE: The square brackets should not be included (i.e subsample0.7, features210).\"}, {\"name\": \"merge.type\", \"value\": \"average\", \"label\": \"Merge.Type\", \"description\": \"Ignored when algorithm other than hierarchical selected\"}, {\"name\": \"descent.iterations\", \"value\": \"2000\", \"label\": \"Descent.Iterations\", \"description\": \"Number of SOM/NMF iterations\"}, {\"name\": \"output.stub\", \"value\": \"<input.filename_basename>\", \"label\": \"Output.Stub\", \"description\": \"Stub pre-pended to all the output file names\"}, {\"name\": \"normalize.type\", \"value\": \"-n1\", \"label\": \"Normalize.Type\", \"description\": \"row-wise, column-wise, both\"}, {\"name\": \"normalization.iterations\", \"value\": \"0\", \"label\": \"Normalization.Iterations\", \"description\": \"number of row/column normalization iterations (supercedes normalize.type)\"}, {\"name\": \"heat.map.size\", \"value\": \"2\", \"label\": \"Heat.Map.Size\", \"description\": \"point size of a consensus matrix's heat map (between 1 and 20)\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"ConstellationMap\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"A downstream visualization and analysis tool for gene set enrichment results with interactive web visualizer.\", \"required_args\": [{\"name\": \"input.gct.file\", \"value\": \"\", \"label\": \"Input.Gct.File\", \"description\": \"Input gene set enrichment file. Expected to be output from ssGSEAProjection module.\"}, {\"name\": \"input.cls.file\", \"value\": \"\", \"label\": \"Input.Cls.File\", \"description\": \"Phenotype labels. Input cls file describing phenotypes of gct file.\"}, {\"name\": \"gene.sets.file\", \"value\": \"\", \"label\": \"Gene.Sets.File\", \"description\": \"Gene set file. Must match gene sets listed in input gct file. \"}, {\"name\": \"top.n\", \"value\": \"\", \"label\": \"Top.N\", \"description\": \"The number of gene sets to be plotted.\"}, {\"name\": \"target.class\", \"value\": \"\", \"label\": \"Target.Class\", \"description\": \"The phenotypic category of interest.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"ConstellationMap\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"gene.sets.database\", \"value\": \"c1.all.v5.1.symbols.gmt\", \"label\": \"Gene.Sets.Database\", \"description\": \"Gene sets database from GSEA website.\"}, {\"name\": \"direction\", \"value\": \"positive\", \"label\": \"Direction\", \"description\": \"The direction of correlation, as measured by mutual information, to be investigated.\"}, {\"name\": \"image.format\", \"value\": \"PNG\", \"label\": \"Image.Format\", \"description\": \"The image file type of outputted plots, either bitmap (PNG) or vector graphics (PDF).\"}, {\"name\": \"jaccard.threshold\", \"value\": \"0.1\", \"label\": \"Jaccard.Threshold\", \"description\": \"The Jaccard index threshold above which connective edges between gene sets will be plotted.\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"ConvertLineEndings\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Converts line endings to the host operating system's format.\", \"required_args\": [{\"name\": \"input.filename\", \"value\": \"\", \"label\": \"Input.Filename\", \"description\": \"The input file (any non-binary file format)\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"ConvertLineEndings\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.file\", \"value\": \"<input.filename_basename>.cvt.<input.filename_extension>\", \"label\": \"Output.File\", \"description\": \"The output file\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"ConvertToMAGEML\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Converts a gct, res, or odf dataset file to a MAGE-ML file\", \"required_args\": [{\"name\": \"input.filename\", \"value\": \"\", \"label\": \"Input.Filename\", \"description\": \"The gct, res, or odf file to convert\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"ConvertToMAGEML\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.file.base.name\", \"value\": \"<input.filename_basename>\", \"label\": \"Output.File.Base.Name\", \"description\": \"The base name of the zip output file or the MAGE-ML file\"}, {\"name\": \"create.zip.file\", \"value\": \"yes\", \"label\": \"Create.Zip.File\", \"description\": \"Whether to zip the MAGE-ML files\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"ConvertToMAGETAB\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"A module to export data from GenePattern in MAGE-TAB format\", \"required_args\": [{\"name\": \"input.file\", \"value\": \"\", \"label\": \"Input.File\", \"description\": \"A dataset file  - .gct, .res\"}, {\"name\": \"cls.file\", \"value\": \"\", \"label\": \"Cls.File\", \"description\": \"A class file  - .cls\"}, {\"name\": \"intensity.type.label\", \"value\": \"\", \"label\": \"Intensity.Type.Label\", \"description\": \"label that specifies the type of intensity values in data matrix file \"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"ConvertToMAGETAB\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"sample.reference.label\", \"value\": \"Hybridization REF\", \"label\": \"Sample.Reference.Label\", \"description\": \"reference label for samples in the data matrix file\"}, {\"name\": \"feature.reference.label\", \"value\": \"CompositeElement REF\", \"label\": \"Feature.Reference.Label\", \"description\": \"reference label for features in the data matrix file\"}, {\"name\": \"output.prefix\", \"value\": \"<input.file_basename>\", \"label\": \"Output.Prefix\", \"description\": \"prefix to use for output file name\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"COPA\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Performs Cancer Outlier Profile Analysis (COPA)\", \"required_args\": [{\"name\": \"dataset.file\", \"value\": \"\", \"label\": \"Dataset.File\", \"description\": \"The dataset file - .res, .gct\"}, {\"name\": \"sample.info.file\", \"value\": \"\", \"label\": \"Sample.Info.File\", \"description\": \"sample info file - .txt\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"COPA\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"outlier.cutoff\", \"value\": \"5\", \"label\": \"Outlier.Cutoff\", \"description\": \"The cutoff to determine 'outlier' status.\"}, {\"name\": \"pre-filter.percentile\", \"value\": \"0.95\", \"label\": \"Pre-Filter.Percentile\", \"description\": \"The percentile to use for pre-filtering the data. All genes with a number of outlier samples less than the top (default 95th) percentile will be removed from further consideration. This value must be between 0 and 1 exclusive.\"}, {\"name\": \"compute.large\", \"value\": \"no\", \"label\": \"Compute.Large\", \"description\": \"Whether to continue analysis when the number of genes remaining after pre-filtering is greater than 1000\"}, {\"name\": \"max.overlap\", \"value\": \"0\", \"label\": \"Max.Overlap\", \"description\": \"The maximum number of samples that can be considered 'outliers' when comparing two genes. The default is 0, indicating that there can be no overlap.\"}, {\"name\": \"normal.outliers\", \"value\": \"0\", \"label\": \"Normal.Outliers\", \"description\": \"The number of normal samples that can be considered 'outliers'. The default is 0, meaning that no normals may be outliers.\"}, {\"name\": \"gene.pairs.to.plot\", \"value\": \"1-5\", \"label\": \"Gene.Pairs.To.Plot\", \"description\": \"A range of gene pairs to plot. (e.g., 2-5) based on the gene pair rank. The ranks are determined by the number of outliers. The more outliers a gene pair has the greater the ranking. Specify none if no gene pair plots should be generated.\"}, {\"name\": \"output.file\", \"value\": \"<dataset.file_basename>\", \"label\": \"Output.File\", \"description\": \"The name of the output files\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"CopyNumberDivideByNormals\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Determines the copy number of a target SNP\", \"required_args\": [{\"name\": \"snp.file\", \"value\": \"\", \"label\": \"Snp.File\", \"description\": \"SNP file containing both normal and target samples.\"}, {\"name\": \"sample.info.file\", \"value\": \"\", \"label\": \"Sample.Info.File\", \"description\": \"dChip format sample info file containing Ploidy(numeric) and Sample columns\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"CopyNumberDivideByNormals\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"include.genotype.calls\", \"value\": \"cn\", \"label\": \"Include.Genotype.Calls\", \"description\": \"Whether to include genotype calls from the input snp file if they exist.\"}, {\"name\": \"output.file.name\", \"value\": \"<snp.file_basename>\", \"label\": \"Output.File.Name\", \"description\": \"Output file name\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"CopyNumberInferencePipeline\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"[**Available on public server only**] Generates segmented copy number calls from raw tumor and normal SNP6 CEL files.\", \"required_args\": [{\"name\": \"CopyNumberInferencePipeline.Part1of21.SNPFileCreator_SNP62.cel.files\", \"value\": \"\", \"label\": \"CopyNumberInferencePipeline.Part1Of21.SNPFileCreator SNP62.Cel.Files\", \"description\": \"Zip of SNP 6.0 CEL files\"}, {\"name\": \"CopyNumberInferencePipeline.Part1of21.ConvertLineEndings1.input.filename\", \"value\": \"\", \"label\": \"CopyNumberInferencePipeline.Part1Of21.ConvertLineEndings1.Input.Filename\", \"description\": \"The input file (any non-binary file format)\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"CopyNumberInferencePipeline\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"CopyNumberInferencePipeline.Part1of21.TangentNormalization9.output.files\", \"value\": \"nocnv\", \"label\": \"CopyNumberInferencePipeline.Part1Of21.TangentNormalization9.Output.Files\", \"description\": \"The types of files to output. Whether to output files with CNVs included, removed, or both.\"}, {\"name\": \"CopyNumberInferencePipeline.Part2of2.Wrapper4.CopyNumberInferencePipeline.Part2of21.InsertGenomicLocations3.genome.annotation\", \"value\": \"hg19\", \"label\": \"CopyNumberInferencePipeline.Part2Of2.Wrapper4.CopyNumberInferencePipeline.Part2Of21.InsertGenomicLocations3.Genome.Annotation\", \"description\": \"The internally provided genome annotation to insert into the input file.\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"CopyNumberNoise\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Measures noise, calculates QC metrics on copynumber data.\", \"required_args\": [{\"name\": \"input.file\", \"value\": \"\", \"label\": \"Input.File\", \"description\": \"The quality scores computed by the CopyNumberInference module. \"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"CopyNumberNoise\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.file\", \"value\": \"<input.file_basename>.noise.txt\", \"label\": \"Output.File\", \"description\": \"The name of the output file\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"copyOfReorderGCT\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Reorder a GCT to be more human readable by sorting on a property such as the Gene Symbol, mean expression, sum and variance\", \"required_args\": [{\"name\": \"gct.file\", \"value\": \"\", \"label\": \"Gct.File\", \"description\": \"A GCT file to be reordered\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"copyOfReorderGCT\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"sort.feature\", \"value\": \"geneSymbol\", \"label\": \"Sort.Feature\", \"description\": \"The feature to sort on\"}, {\"name\": \"order.as\", \"value\": \"a\", \"label\": \"Order.As\", \"description\": \"sort in Ascending or Descending order\"}, {\"name\": \"output.file\", \"value\": \"<gct.file_basename>_reord.gct\", \"label\": \"Output.File\", \"description\": \"Name of the reordered GCT file\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"CoxRegression\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Cox regression using clinical data and .cls file\", \"required_args\": [{\"name\": \"input.surv.data.filename\", \"value\": \"\", \"label\": \"Input.Surv.Data.Filename\", \"description\": \"input clinical data. Case ID in 1st column (.txt file)\"}, {\"name\": \"input.cls.filename\", \"value\": \"\", \"label\": \"Input.Cls.Filename\", \"description\": \"input class information (.cls file)\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"CoxRegression\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.file\", \"value\": \"CoxRegression_result\", \"label\": \"Output.File\", \"description\": \"\"}, {\"name\": \"time\", \"value\": \"time\", \"label\": \"Time\", \"description\": \"Column name for survival time\"}, {\"name\": \"status\", \"value\": \"status\", \"label\": \"Status\", \"description\": \"Column name for censor (0: censored/alive, 1: event/death)\"}, {\"name\": \"variable.continuous\", \"value\": \"NA\", \"label\": \"Variable.Continuous\", \"description\": \"(e.g. age,labdata1,...)\"}, {\"name\": \"variable.category\", \"value\": \"NA\", \"label\": \"Variable.Category\", \"description\": \"(e.g. sex,cls,grade,...)\"}, {\"name\": \"variable.interaction.terms\", \"value\": \"NA\", \"label\": \"Variable.Interaction.Terms\", \"description\": \"(e.g. sex:cls,sex:grade,...)\"}, {\"name\": \"strata\", \"value\": \"NA\", \"label\": \"Strata\", \"description\": \"Fit is done for each strata\"}, {\"name\": \"input.subgroup\", \"value\": \"NA\", \"label\": \"Input.Subgroup\", \"description\": \"(e.g. cls:2) *only one subgroup can be specified\"}, {\"name\": \"variable.selection\", \"value\": \"none\", \"label\": \"Variable.Selection\", \"description\": \"\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"CreateSymlinks\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Creates symlinks to the input files in the job results directory. This is intended to be a helper module for creating scatter-gather pipelines.  Note that this module is not supported on Windows\", \"required_args\": [{\"name\": \"input.files\", \"value\": \"\", \"label\": \"Input.Files\", \"description\": \"One or more input files\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"CreateSymlinks\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"CsvToFcs\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Converts a Comma separated values (CSV) file to a Flow Cytometry Standard (FCS) file.\", \"required_args\": [{\"name\": \"Input.CSV.data.file\", \"value\": \"\", \"label\": \"Input.CSV.Data.File\", \"description\": \"A CSV file to be converted to FCS.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"CsvToFcs\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"Output.FCS.file.name\", \"value\": \"<Input.CSV.data.file_basename>.fcs\", \"label\": \"Output.FCS.File.Name\", \"description\": \"The name of the output FCS file.\"}, {\"name\": \"Range\", \"value\": \"auto\", \"label\": \"Range\", \"description\": \"Value of the $PnR keyword for all parameters. By default, this tool will try to set the best settings based on the data in the file; use \\\"auto\\\", \\\"auto-exact\\\", \\\"BD\\\", or an integer value.\"}, {\"name\": \"Data.type\", \"value\": \"auto\", \"label\": \"Data.Type\", \"description\": \"Value of the $DATATYPE keyword. By default, this tool will try to set the best settings based on the data in the input CSV file.\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"Cuffcompare\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Cufflinks 2.0.2 - Cuffcompare - Analyzes the transcribed fragments in an assembly\", \"required_args\": [{\"name\": \"input.file\", \"value\": \"\", \"label\": \"Input.File\", \"description\": \"One or more GTF file outputs from Cufflinks.\"}, {\"name\": \"reference.GTF\", \"value\": \"\", \"label\": \"Reference.GTF\", \"description\": \"A reference annotation GTF\"}, {\"name\": \"reference.genome.file\", \"value\": \"\", \"label\": \"Reference.Genome.File\", \"description\": \"Fasta file or zip of fasta files against which your reads were aligned\"}, {\"name\": \"additional.cuffcompare.options\", \"value\": \"\", \"label\": \"Additional.Cuffcompare.Options\", \"description\": \"Additional options to be passed along to the Cuffcompare program at the command line.  This parameter gives you a means to specify otherwise unavailable Cuffcompare options and switches not supported by the module; check the Cufflinks manual for details.  Recommended for experts only; use this at your own discretion.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"Cuffcompare\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.prefix\", \"value\": \"<input.file_basename>\", \"label\": \"Output.Prefix\", \"description\": \"A prefix for the module output\"}, {\"name\": \"exclude.transcripts\", \"value\": \"true\", \"label\": \"Exclude.Transcripts\", \"description\": \"Whether to ignore reference transcripts that are not overlapped by any transcript in the input files.  Does not apply if no reference GTF is provided.\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"Cuffdiff\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Cufflinks 2.0.2 - Finds significant changes in transcript expression, splicing, and promoter use\", \"required_args\": [{\"name\": \"aligned.files\", \"value\": \"\", \"label\": \"Aligned.Files\", \"description\": \"A set of aligned files grouped by condition\"}, {\"name\": \"GTF.file\", \"value\": \"\", \"label\": \"GTF.File\", \"description\": \"A transcript annotation file in GTF/GFF format produced by cufflinks, cuffcompare, cuffmerge, or other source (such as a reference annotation GTF)\"}, {\"name\": \"frag.bias.correct\", \"value\": \"\", \"label\": \"Frag.Bias.Correct\", \"description\": \"Reference (FASTA/FA) for bias detection and correction algorithm\"}, {\"name\": \"mask.file\", \"value\": \"\", \"label\": \"Mask.File\", \"description\": \"This file tells Cuffdiff to ignore all reads that could have come from transcripts in this GTF/GFF file. It is recommended that annotated rRNA, mitochondrial transcripts, and other abundant transcripts you want to ignore in your analysis be included in this file.\"}, {\"name\": \"additional.cuffdiff.options\", \"value\": \"\", \"label\": \"Additional.Cuffdiff.Options\", \"description\": \"Additional options to be passed along to the Cuffdiff program at the command line.  This parameter gives you a means to specify otherwise unavailable Cuffdiff options and switches not supported by the module; check the Cufflinks manual for details.  Recommended for experts only; use this at your own discretion.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"Cuffdiff\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"time.series\", \"value\": \"false\", \"label\": \"Time.Series\", \"description\": \"Analyze the provided samples as a time series, rather than testing for differences between all pairs of samples.\"}, {\"name\": \"normalization.method\", \"value\": \"geometric\", \"label\": \"Normalization.Method\", \"description\": \"The normalization method to be used. Choices are geometric mean (the default), upper quartile, or standard FPKM normalization. See the documentation for a discussion of these methods.\"}, {\"name\": \"FPKM.scaling\", \"value\": \"compatible-hits\", \"label\": \"FPKM.Scaling\", \"description\": \"Controls how Cuffdiff includes mapped fragments toward the number used in the FPKM denominator.  Use compatible-hits (the default) to count only those fragments compatible with some reference transcript.  Use total-hits to count all fragments, even those not compatible with a reference transcript.\"}, {\"name\": \"multi.read.correct\", \"value\": \"true\", \"label\": \"Multi.Read.Correct\", \"description\": \"Instructs Cuffdiff to do use its two-stage read weighting algorithm to more accurately distribute a multi-mapped read's count contribution across the multiple loci the read mapped to.\"}, {\"name\": \"min.alignment.count\", \"value\": \"10\", \"label\": \"Min.Alignment.Count\", \"description\": \"The minimum number of alignments in a locus needed to conduct significance testing on changes in that locus observed between samples.  If no testing is performed, changes in the locus are deemed not significant, and the locus's observed changes do not contribute to correction for multiple testing.\"}, {\"name\": \"FDR\", \"value\": \".05\", \"label\": \"Fdr\", \"description\": \"The allowed false discovery rate.\"}, {\"name\": \"library.type\", \"value\": \"inferred\", \"label\": \"Library.Type\", \"description\": \"The library type used to generate reads. The default is inferred, meaning that no library type information is passed.\"}, {\"name\": \"skip.diff.exp\", \"value\": \"false\", \"label\": \"Skip.Diff.Exp\", \"description\": \"Tells Cuffdiff to perform quantification and normalization only and to skip its differential expression calculations which are computationally expensive.  The default is to compute differential expression.\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"Cufflinks\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Cufflinks 2.0.2 - Assembles transcripts, estimates abundances, and tests for differential expression and regulation in RNA-seq samples\", \"required_args\": [{\"name\": \"input.file\", \"value\": \"\", \"label\": \"Input.File\", \"description\": \"Input file in SAM or BAM format\"}, {\"name\": \"transfrag.label\", \"value\": \"\", \"label\": \"Transfrag.Label\", \"description\": \"A label for the transfrags in the output files\"}, {\"name\": \"GTF\", \"value\": \"\", \"label\": \"Gtf\", \"description\": \"Reference annotation (GFF/GTF file) for isoform expression estimates\"}, {\"name\": \"GTF.guide\", \"value\": \"\", \"label\": \"GTF.Guide\", \"description\": \"Reference annotation (GFF/GTF file) to guide RABT assembly\"}, {\"name\": \"mask.file\", \"value\": \"\", \"label\": \"Mask.File\", \"description\": \"A GTF file specifying transcripts to be ignored\"}, {\"name\": \"frag.bias.correct\", \"value\": \"\", \"label\": \"Frag.Bias.Correct\", \"description\": \"Reference (FASTA/FA) for bias detection and correction algorithm\"}, {\"name\": \"additional.cufflinks.options\", \"value\": \"\", \"label\": \"Additional.Cufflinks.Options\", \"description\": \"Additional options to be passed along to the Cufflinks program at the command line.  This parameter gives you a means to specify otherwise unavailable Cufflinks options and switches not supported by the module; check the Cufflinks manual for details.  Recommended for experts only; use this at your own discretion.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"Cufflinks\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"multi.read.correct\", \"value\": \"false\", \"label\": \"Multi.Read.Correct\", \"description\": \"Whether to do an initial estimation procedure to more accurately weight reads mapping to multiple locations in the genome\"}, {\"name\": \"library.type\", \"value\": \"inferred\", \"label\": \"Library.Type\", \"description\": \"The library type used to generate reads.\"}, {\"name\": \"min.frags.per.transfrag\", \"value\": \"10\", \"label\": \"Min.Frags.Per.Transfrag\", \"description\": \"Assembled transfrags supported by fewer than this many aligned RNA-Seq fragments are not reported.\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"CufflinksCuffmergePipeline\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Creates multiple transcript assemblies using Cufflinks and then merges these assemblies into one\", \"required_args\": [{\"name\": \"CreateSymlinks1.input.files\", \"value\": \"\", \"label\": \"CreateSymlinks1.Input.Files\", \"description\": \"One or more input files\"}, {\"name\": \"CufflinksWrapperPipeline2.Cufflinks1.GTF.guide\", \"value\": \"\", \"label\": \"CufflinksWrapperPipeline2.Cufflinks1.GTF.Guide\", \"description\": \"Reference annotation (GFF/GTF file) to guide RABT assembly\"}, {\"name\": \"CufflinksWrapperPipeline2.Cufflinks1.mask.file\", \"value\": \"\", \"label\": \"CufflinksWrapperPipeline2.Cufflinks1.Mask.File\", \"description\": \"A GTF file specifying transcripts to be ignored\"}, {\"name\": \"CufflinksWrapperPipeline2.Cufflinks1.additional.cufflinks.options\", \"value\": \"\", \"label\": \"CufflinksWrapperPipeline2.Cufflinks1.Additional.Cufflinks.Options\", \"description\": \"Additional options to be passed along to the Cufflinks program at the command line.  This parameter gives you a means to specify otherwise unavailable Cufflinks options and switches not supported by the module; check the Cufflinks manual for details.  Recommended for experts only; use this at your own discretion.\"}, {\"name\": \"Cufflinks.cuffmerge4.reference.GTF\", \"value\": \"\", \"label\": \"Cufflinks.Cuffmerge4.Reference.GTF\", \"description\": \"An optional reference annotation GTF. The input assemblies are merged together with the reference GTF and included in the final output. Cuffmerge will use this to attach gene names and other metadata to the merged catalog.\"}, {\"name\": \"Cufflinks.cuffmerge4.genome.file\", \"value\": \"\", \"label\": \"Cufflinks.Cuffmerge4.Genome.File\", \"description\": \"A file containing the genomic DNA sequences for the reference.  This should be a multi-FASTA file with all contigs present.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"CufflinksCuffmergePipeline\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"CufflinksWrapperPipeline2.Cufflinks1.library.type\", \"value\": \"inferred\", \"label\": \"CufflinksWrapperPipeline2.Cufflinks1.Library.Type\", \"description\": \"The library type used to generate reads.\"}, {\"name\": \"CufflinksWrapperPipeline2.Cufflinks1.min.frags.per.transfrag\", \"value\": \"10\", \"label\": \"CufflinksWrapperPipeline2.Cufflinks1.Min.Frags.Per.Transfrag\", \"description\": \"Assembled transfrags supported by fewer than this many aligned RNA-Seq fragments are not reported.\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"Cuffmerge\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Cufflinks 2.0.2: Cuffmerge - Merge several Cufflinks assemblies\", \"required_args\": [{\"name\": \"input.file\", \"value\": \"\", \"label\": \"Input.File\", \"description\": \"GTF files to be merged\"}, {\"name\": \"reference.GTF\", \"value\": \"\", \"label\": \"Reference.GTF\", \"description\": \"An optional reference annotation GTF. The input assemblies are merged together with the reference GTF and included in the final output. Cuffmerge will use this to attach gene names and other metadata to the merged catalog.\"}, {\"name\": \"genome.file\", \"value\": \"\", \"label\": \"Genome.File\", \"description\": \"A file containing the genomic DNA sequences for the reference.  This should be a multi-FASTA file with all contigs present.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"Cuffmerge\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"CummeRbund.GeneSetReport\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"[Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] Cuffdiff visualization package providing plots based on a user-provided list of genes.\", \"required_args\": [{\"name\": \"cuffdiff.input\", \"value\": \"\", \"label\": \"Cuffdiff.Input\", \"description\": \"A Cuffdiff job, a previous CummeRbund job, or a cuffData.db file from a previous CummeRbund job.\"}, {\"name\": \"geneset.file\", \"value\": \"\", \"label\": \"Geneset.File\", \"description\": \"A text file containing the list of genes for this report, one per line.  Each of these can be represented by a gene symbol (short name), gene ID, isoform_id, tss_group_id, or cds_id.\"}, {\"name\": \"selected.conditions\", \"value\": \"\", \"label\": \"Selected.Conditions\", \"description\": \"Specifies the conditions (samples) to be used in the plots.  This should be a comma-separated list of conditions, using the same names as in the upstream Cuffdiff job; leave this blank to use all conditions.  If blank, no pairwise comparison plots will be generated.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"CummeRbund.GeneSetReport\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.format\", \"value\": \"pdf\", \"label\": \"Output.Format\", \"description\": \"The output file format.\"}, {\"name\": \"feature.level\", \"value\": \"genes\", \"label\": \"Feature.Level\", \"description\": \"Feature level for the report.\"}, {\"name\": \"report.as.aggregate\", \"value\": \"yes\", \"label\": \"Report.As.Aggregate\", \"description\": \"Controls whether reporting should be done for individual replicates or aggregate condition/sample values.  The default is to use aggregate sample values.  Note that the Volcano plot always shows aggregate samples.\"}, {\"name\": \"log.transform\", \"value\": \"yes\", \"label\": \"Log.Transform\", \"description\": \"Whether or not to log transform the FPKM values.  Note that the FPKM values are always log transformed for the Volcano plots and that the Scatter plot will use a log2 rather than log10 transformation.  For the Expression Plot and Expression Barplot, this directs that the y-axis will be drawn on a log10 scale.\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"CummeRbund.QcReport\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"[Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] Cuffdiff visualization package providing plots and tables related to QC and Global Stats.\", \"required_args\": [{\"name\": \"cuffdiff.input\", \"value\": \"\", \"label\": \"Cuffdiff.Input\", \"description\": \"A Cuffdiff job, a previous CummeRbund job, or a cuffData.db file from a previous CummeRbund job.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"CummeRbund.QcReport\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.format\", \"value\": \"pdf\", \"label\": \"Output.Format\", \"description\": \"The output file format.\"}, {\"name\": \"feature.level\", \"value\": \"genes\", \"label\": \"Feature.Level\", \"description\": \"Feature level for the report\"}, {\"name\": \"report.as.aggregate\", \"value\": \"yes\", \"label\": \"Report.As.Aggregate\", \"description\": \"Controls whether reporting should be done for individual replicates or aggregate condition/sample values.  The default is to use aggregate sample values.  Controls whether reporting should be done for individual replicates or aggregate condition/sample values.  The default is to use aggregate sample values.  Note that the Dispersion, FPKM.SCV, and MA plots always show aggregate samples.\"}, {\"name\": \"log.transform\", \"value\": \"yes\", \"label\": \"Log.Transform\", \"description\": \"Whether or not to log transform FPKM values.  Note that the FPKM values are always log transformed for the FPKM.SCV plot.  Log transformation does not apply to the Dispersion, MDS, and PCA plots.\"}, {\"name\": \"pca.x\", \"value\": \"PC2\", \"label\": \"Pca.X\", \"description\": \"Indicates which principal component is to be presented on the x-axis of the PCA plot.  Must differ from pca.y.\"}, {\"name\": \"pca.y\", \"value\": \"PC3\", \"label\": \"Pca.Y\", \"description\": \"Indicates which principal component is to be presented on the y-axis of the PCA plot.  Must differ from pca.x.\"}, {\"name\": \"attempt.to.merge.names\", \"value\": \"no\", \"label\": \"Attempt.To.Merge.Names\", \"description\": \"Should the module attempt to merge gene names into the text reports?  Depending on the GTF file used in the Cuffdiff run, the text reports may instead contain tracking ID references, particularly if a Cuffmerge merged.gtf file was used.  Note that this setting does not affect the plots.\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"CummeRbund.SelectedConditionsReport\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"[**Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] Cuffdiff visualization package providing plots based on a user-provided list of conditions.\", \"required_args\": [{\"name\": \"cuffdiff.input\", \"value\": \"\", \"label\": \"Cuffdiff.Input\", \"description\": \"A Cuffdiff job, a previous CummeRbund job, or a cuffData.db file from a previous CummeRbund job.\"}, {\"name\": \"selected.conditions\", \"value\": \"\", \"label\": \"Selected.Conditions\", \"description\": \"Specifies the conditions (samples) to be used in the plots.  This should be a comma-separated list of conditions, using the same names as in the upstream Cuffdiff job; leave this blank to use all conditions.  If blank, no pairwise comparison plots will be generated.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"CummeRbund.SelectedConditionsReport\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.format\", \"value\": \"pdf\", \"label\": \"Output.Format\", \"description\": \"The output file format.\"}, {\"name\": \"feature.level\", \"value\": \"genes\", \"label\": \"Feature.Level\", \"description\": \"Feature level for the report.\"}, {\"name\": \"report.as.aggregate\", \"value\": \"yes\", \"label\": \"Report.As.Aggregate\", \"description\": \"Controls whether reporting should be done for individual replicates or aggregate condition/sample values.  The default is to use aggregate sample values.  Note that the Volcano plot always shows aggregate samples.\"}, {\"name\": \"log.transform\", \"value\": \"yes\", \"label\": \"Log.Transform\", \"description\": \"Whether or not to log transform the FPKM values.  Note that the FPKM values are always log transformed for the Volcano plots and that the Scatter plot will use a log2 rather than log10 transformation.\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"CummeRbund.SelectedGeneReport\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"[**Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] Cuffdiff visualization package providing plots based on a single user-specified gene.\", \"required_args\": [{\"name\": \"cuffdiff.input\", \"value\": \"\", \"label\": \"Cuffdiff.Input\", \"description\": \"A Cuffdiff job, a previous CummeRbund job, or a cuffData.db file from a previous CummeRbund job.\"}, {\"name\": \"feature.id\", \"value\": \"\", \"label\": \"Feature.Id\", \"description\": \"The gene or feature of interest.  This can be a gene symbol (short name), gene ID, isoform_id, tss_group_id, or cds_id.\"}, {\"name\": \"selected.conditions\", \"value\": \"\", \"label\": \"Selected.Conditions\", \"description\": \"Specifies the conditions (samples) to be used in the plots.  This should be a comma-separated list of conditions, using the same names as in the upstream Cuffdiff job; leave this blank to use all conditions.  The Similarity plots will always operate across all conditions.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"CummeRbund.SelectedGeneReport\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"find.similar\", \"value\": \"10\", \"label\": \"Find.Similar\", \"description\": \"Optionally, find and plot the top genes (up to this count) with an expression profile most similar to the given gene of interest.  If blank, this will be skipped.    \"}, {\"name\": \"output.format\", \"value\": \"pdf\", \"label\": \"Output.Format\", \"description\": \"The output file format.\"}, {\"name\": \"feature.level\", \"value\": \"genes\", \"label\": \"Feature.Level\", \"description\": \"Feature level for the report.  Note that the Similarity plots will always be generated at the \\\"genes\\\" level.\"}, {\"name\": \"report.as.aggregate\", \"value\": \"yes\", \"label\": \"Report.As.Aggregate\", \"description\": \"Controls whether reporting should be done for individual replicates or aggregate condition/sample values.  The default is to use aggregate sample values.  Note that the Similarity plots always show aggregate samples.\"}, {\"name\": \"log.transform\", \"value\": \"yes\", \"label\": \"Log.Transform\", \"description\": \"Whether or not to log transform the FPKM values.\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"CytoscapeViewer\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Visualizes a gene network\", \"required_args\": [{\"name\": \"dataset.file\", \"value\": \"\", \"label\": \"Dataset.File\", \"description\": \"Dataset file - .res, .gct\"}, {\"name\": \"adjacency.matrix.file\", \"value\": \"\", \"label\": \"Adjacency.Matrix.File\", \"description\": \"Adjacency matrix - .adj\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"CytoscapeViewer\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"DAPPLE\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"DAPPLE\", \"required_args\": [{\"name\": \"Input.File\", \"value\": \"\", \"label\": \"Input.File\", \"description\": \"File containing a list of SNPs, genes or regions. Each input should be on a new line.\"}, {\"name\": \"Keyword\", \"value\": \"\", \"label\": \"Keyword\", \"description\": \"\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"DAPPLE\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"Permute.Length\", \"value\": \"0\", \"label\": \"Permute.Length\", \"description\": \"To increase speed, this is adaptive permutation (see README)\"}, {\"name\": \"Plot\", \"value\": \"true\", \"label\": \"Plot\", \"description\": \"\"}, {\"name\": \"nearest.gene\", \"value\": \"true\", \"label\": \"Nearest.Gene\", \"description\": \"Do you want to use the gene nearest to the SNP (rather than grabbing all genes in the LD wingspan)? Only relevant for SNP inputs.\"}, {\"name\": \"genome.build\", \"value\": \"18\", \"label\": \"Genome.Build\", \"description\": \"Which set of SNPs should we use? hg18, hg19, 1KG?\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"DeIdentifyFCS\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"DeIdentifyFCS an FCS data file; remove keywords from a list or matching a regular expression; useful to anonymize FCS data files and/or to remove specific (e.g., clinical) information \", \"required_args\": [{\"name\": \"Input.FCS.data.file\", \"value\": \"\", \"label\": \"Input.FCS.Data.File\", \"description\": \"An FCS data file to be de-identified.\"}, {\"name\": \"Remove.keywords\", \"value\": \"\", \"label\": \"Remove.Keywords\", \"description\": \"Comma-separated list of keyword names (escape comma by '\\\\' if part of a keyword name) or regular expression to match keywords to be removed.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"DeIdentifyFCS\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"Is.regular.expression.used\", \"value\": \"No\", \"label\": \"Is.Regular.Expression.Used\", \"description\": \"Is a regular expression used in the list of keywords to remove? If not (default) a simple comma-separated list of keyword names is expected. In this list, comma may be escaped by '\\\\' if part of a keyword name, e.g., this \\\\, is in the keyword.\"}, {\"name\": \"Output.FCS.file.name\", \"value\": \"<Input.FCS.data.file_basename>.deidentified.<Input.FCS.data.file_extension>\", \"label\": \"Output.FCS.File.Name\", \"description\": \"Output FCS file name.\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"DependencyBiomarkerDotplot\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Creates a dotplot of Achilles gene dependency scores. The dots are colored based on the biomarker status of the cell line (i.e. mutation in gene or no mutation in gene). This module was created for use in the Education Outreach high school computational curriculum. \", \"required_args\": [{\"name\": \"Achilles.Gene.Dependency.Data.File\", \"value\": \"\", \"label\": \"Achilles.Gene.Dependency.Data.File\", \"description\": \"Format is a csv file, with cell lines in rows and genes in columns. \"}, {\"name\": \"Biomarker.Data.File\", \"value\": \"\", \"label\": \"Biomarker.Data.File\", \"description\": \"Format is a csv file, with cell lines in rows and genes in columns. \"}, {\"name\": \"Gene.1\", \"value\": \"\", \"label\": \"Gene.1\", \"description\": \"The gene name should match exactly to the column names in the data files and are case-sensitive. \\n\"}, {\"name\": \"Gene.#2\", \"value\": \"\", \"label\": \"Gene.#2\", \"description\": \"The gene name should match exactly to the column names in the data files and are case-sensitive.\"}, {\"name\": \"Gene.#3\", \"value\": \"\", \"label\": \"Gene.#3\", \"description\": \"The gene name should match exactly to the column names in the data files and are case-sensitive.\"}, {\"name\": \"Gene.#4\", \"value\": \"\", \"label\": \"Gene.#4\", \"description\": \"The gene name should match exactly to the column names in the data files and are case-sensitive.\"}, {\"name\": \"Gene.#5\", \"value\": \"\", \"label\": \"Gene.#5\", \"description\": \"The gene name should match exactly to the column names in the data files and are case-sensitive.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"DependencyBiomarkerDotplot\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"DESeq2\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"A Differential Expression workflow for RNA-Seq data based on the DESeq2 package from Bioconductor \", \"required_args\": [{\"name\": \"input.file\", \"value\": \"\", \"label\": \"Input.File\", \"description\": \"A GCT file containing raw RNA-Seq counts, such as is produced by MergeHTSeqCounts\"}, {\"name\": \"cls.file\", \"value\": \"\", \"label\": \"Cls.File\", \"description\": \"A categorical CLS file specifying the phenotype classes for the samples in the GCT file.  This should contain exactly two classes with the control specified first.\"}, {\"name\": \"confounding.variable.cls.file\", \"value\": \"\", \"label\": \"Confounding.Variable.Cls.File\", \"description\": \"A categorical CLS file specifying an additional confounding variable, mapped to the input file samples.  Use this for a two-factor comparison.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"DESeq2\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.file.base\", \"value\": \"<input.file_basename>\", \"label\": \"Output.File.Base\", \"description\": \"The base name of the output file(s). File extensions will be added automatically.\"}, {\"name\": \"qc.plot.format\", \"value\": \"skip\", \"label\": \"Qc.Plot.Format\", \"description\": \"Choose a file format for the QC plots (or skip them completely)\"}, {\"name\": \"fdr.threshold\", \"value\": \"0.1\", \"label\": \"Fdr.Threshold\", \"description\": \"FDR threshold to use in filtering before creating the reports of top up-regulated and down-regulated genes \"}, {\"name\": \"top.N.count\", \"value\": \"20\", \"label\": \"Top.N.Count\", \"description\": \"Max gene count to use when creating the reports of top up-regulated and down-regulated genes \"}, {\"name\": \"random.seed\", \"value\": \"779948241\", \"label\": \"Random.Seed\", \"description\": \"Seed to use for randomization\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"DownloadURL\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Downloads a file from a URL\", \"required_args\": [{\"name\": \"URL\", \"value\": \"\", \"label\": \"Url\", \"description\": \"The URL to retrieve the file from\"}, {\"name\": \"output.file\", \"value\": \"\", \"label\": \"Output.File\", \"description\": \"The name of the output file\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"DownloadURL\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"ESPPredictor\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Determine which tryptic peptides will generate the highest signal in ESI-MS\", \"required_args\": [{\"name\": \"input.file\", \"value\": \"\", \"label\": \"Input.File\", \"description\": \"List of peptide sequences (one per line, no column header)\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"ESPPredictor\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"ExpressionFileCreator\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Creates a RES or GCT file from a set of Affymetrix CEL files.  For IVT arrays only; use AffySTExpressionFileCreator for ST arrays.\", \"required_args\": [{\"name\": \"input.file\", \"value\": \"\", \"label\": \"Input.File\", \"description\": \"A zip file of CEL files\"}, {\"name\": \"quantile.normalization\", \"value\": \"\", \"label\": \"Quantile.Normalization\", \"description\": \"(GCRMA and RMA only) Whether to normalize data using quantile normalization\"}, {\"name\": \"background.correct\", \"value\": \"\", \"label\": \"Background.Correct\", \"description\": \"(RMA only) Whether to background correct using RMA background correction\"}, {\"name\": \"value.to.scale.to\", \"value\": \"\", \"label\": \"Value.To.Scale.To\", \"description\": \"(median/mean scaling only) The value to scale to.\"}, {\"name\": \"clm.file\", \"value\": \"\", \"label\": \"Clm.File\", \"description\": \"A tab-delimited text file containing one scan, sample, and class per line\"}, {\"name\": \"cdf.file\", \"value\": \"\", \"label\": \"Cdf.File\", \"description\": \"Custom CDF file. Leave blank to use default internally provided CDF file (custom cdf file is not implemented for GCRMA).\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"ExpressionFileCreator\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"method\", \"value\": \"RMA\", \"label\": \"Method\", \"description\": \"The method to use. Note that dchip and MAS5 will not work with ST arrays.\"}, {\"name\": \"compute.present.absent.calls\", \"value\": \"no\", \"label\": \"Compute.Present.Absent.Calls\", \"description\": \"Whether to compute Present/Absent calls\"}, {\"name\": \"normalization.method\", \"value\": \"median scaling\", \"label\": \"Normalization.Method\", \"description\": \"(MAS5 only) The normalization method to apply after expression values are computed. The column having the median of the means is used as the reference unless the parameter value to scale to is given.\"}, {\"name\": \"annotate.probes\", \"value\": \"yes\", \"label\": \"Annotate.Probes\", \"description\": \"Whether to annotate probes with the gene symbol and description\"}, {\"name\": \"output.file\", \"value\": \"<input.file_basename>\", \"label\": \"Output.File\", \"description\": \"The base name of the output file(s)\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"ExprToGct\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Converts an EXPR file to GCT format\", \"required_args\": [{\"name\": \"input.file\", \"value\": \"\", \"label\": \"Input.File\", \"description\": \"Input file in EXPR format\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"ExprToGct\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.prefix\", \"value\": \"<input.file_basename>\", \"label\": \"Output.Prefix\", \"description\": \"A label that will be used to name the GCT output file.\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"ExtractColumnNames\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Lists the sample descriptors from a .res file.\", \"required_args\": [{\"name\": \"input.filename\", \"value\": \"\", \"label\": \"Input.Filename\", \"description\": \"Input filename - .res\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"ExtractColumnNames\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output\", \"value\": \"<input.filename_basename>\", \"label\": \"Output\", \"description\": \"Name of the output file (a .txt extension will be appended)\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"ExtractComparativeMarkerResults\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Creates a derived dataset and feature list file from the results of ComparativeMarkerSelection\", \"required_args\": [{\"name\": \"comparative.marker.selection.filename\", \"value\": \"\", \"label\": \"Comparative.Marker.Selection.Filename\", \"description\": \"The results from ComparativeMarkerSelection - .odf\"}, {\"name\": \"dataset.filename\", \"value\": \"\", \"label\": \"Dataset.Filename\", \"description\": \"The dataset file used to select markers - .gct, .res, Dataset\"}, {\"name\": \"statistic\", \"value\": \"\", \"label\": \"Statistic\", \"description\": \"The statistic to filter features on\"}, {\"name\": \"min\", \"value\": \"\", \"label\": \"Min\", \"description\": \"Select features with statistic >= min\"}, {\"name\": \"max\", \"value\": \"\", \"label\": \"Max\", \"description\": \"Select features with statistic <= max\"}, {\"name\": \"number.of.neighbors\", \"value\": \"\", \"label\": \"Number.Of.Neighbors\", \"description\": \"Number of neighbors to select by score in each direction\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"ExtractComparativeMarkerResults\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"base.output.name\", \"value\": \"<comparative.marker.selection.filename_basename>.filt\", \"label\": \"Base.Output.Name\", \"description\": \"The base name for the output files\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"ExtractFCSDataset\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Extract one or more FCS datasets from an FCS data file\", \"required_args\": [{\"name\": \"Input.data.file\", \"value\": \"\", \"label\": \"Input.Data.File\", \"description\": \"The input FCS data file; one or more data sets will be extracted from this file.\"}, {\"name\": \"Output.file.name\", \"value\": \"\", \"label\": \"Output.File.Name\", \"description\": \"The name for the output file. Uses the input data file name (for example, name.fcs) as the default, and adds _datasetIndex to the name (e.g., name_datasetIndex.fcs).\"}, {\"name\": \"Dataset.indexes\", \"value\": \"\", \"label\": \"Dataset.Indexes\", \"description\": \"A list of data set indexes to be extracted from the data file (e.g., 1-3, 5, 8). The first data set has the index 1, the second the index 2, etc. All data sets will be extracted if no indexes are provided.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"ExtractFCSDataset\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"ExtractFCSKeywords\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Extracts keyword(s) value(s) from a Flow Cytometry Standard (FCS) file.\", \"required_args\": [{\"name\": \"Input.FCS.data.file\", \"value\": \"\", \"label\": \"Input.FCS.Data.File\", \"description\": \"The input FCS data file from which the module will extract keyword/value pairs.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"ExtractFCSKeywords\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"Keywords.output.mode\", \"value\": \"CSV\", \"label\": \"Keywords.Output.Mode\", \"description\": \"The file format to be used to store the extracted keywords/value pairs.\"}, {\"name\": \"Keywords.output.file.name\", \"value\": \"<Input.FCS.data.file_basename>\", \"label\": \"Keywords.Output.File.Name\", \"description\": \"The name for the file in which the keyword/value pairs are stored.\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"ExtractFCSParameters\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Extract specified parameters (dimensions) from an FCS data file.\", \"required_args\": [{\"name\": \"Input.FCS.data.file\", \"value\": \"\", \"label\": \"Input.FCS.Data.File\", \"description\": \"The input FCS data file from which to extract parameters.\"}, {\"name\": \"Parameters.to.extract\", \"value\": \"\", \"label\": \"Parameters.To.Extract\", \"description\": \"A list of parameters to extract; parameters may be specified by their indexes (e.g., 1-3, 5, 8) or by their short names (e.g., FSC-H, SSC-H, FL1-A, FL3-H).\"}, {\"name\": \"Output.file.name\", \"value\": \"\", \"label\": \"Output.File.Name\", \"description\": \"The output file name. The default uses the input file name as the base.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"ExtractFCSParameters\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"ExtractRowNames\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Extracts the row names from a .res, .gct, or .odf file.\", \"required_args\": [{\"name\": \"input.filename\", \"value\": \"\", \"label\": \"Input.Filename\", \"description\": \"input file - .res, .gct, .odf, file from which the row names are extracted\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"ExtractRowNames\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output\", \"value\": \"<input.filename_basename>.row.names.txt\", \"label\": \"Output\", \"description\": \"output filename containing the row names - .txt\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"FastQC\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"[**Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] Provides quality control metrics on raw sequence data\", \"required_args\": [{\"name\": \"input.file\", \"value\": \"\", \"label\": \"Input.File\", \"description\": \"RNA-seq reads file in FASTQ (bz2 and gz compressed files are supported), SAM, or BAM format.\"}, {\"name\": \"input.format\", \"value\": \"\", \"label\": \"Input.Format\", \"description\": \"Bypasses the normal sequence file format detection and forces the program to use the specified format.\"}, {\"name\": \"contaminant.file\", \"value\": \"\", \"label\": \"Contaminant.File\", \"description\": \"FastQC screens overrepresented sequences (> 0.1% of total) against a set of possible contaminants consisting of all the primers and adapters used by the various sequencing platforms.  The \\\"Overrepresented sequences\\\" report generated by FastQC will indicate when an overrepresented sequence corresponds to a known contaminant.  This optional contaminant file provides a means of adding more sequences to FastQC's built-in list of known contaminants. See module documentation for a description and example of this file's format.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"FastQC\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"kmer.size\", \"value\": \"5\", \"label\": \"Kmer.Size\", \"description\": \"The length of Kmer to look for in the Kmer content module. The specified Kmer length must be between 2 and 10. Default length is 5.\"}, {\"name\": \"extract.output\", \"value\": \"true\", \"label\": \"Extract.Output\", \"description\": \"Whether to output an uncompressed version of the report. Set this to yes to view the report directly from within GenePattern.\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"FCMFeatureExtraction\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Calculation of features from clustered flow cytometry data.\", \"required_args\": [{\"name\": \"Input.CSV.files\", \"value\": \"\", \"label\": \"Input.CSV.Files\", \"description\": \"The set of CSV files with clustering results (of a single data file). May be a .zip file, a directory, or a text file listing all input file paths (one file per line).\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"FCMFeatureExtraction\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"Calculate.number.of.cells\", \"value\": \"TRUE\", \"label\": \"Calculate.Number.Of.Cells\", \"description\": \"Calculate and output the number of cells in each sub-population?\"}, {\"name\": \"Calculate.percentage.of.cells\", \"value\": \"TRUE\", \"label\": \"Calculate.Percentage.Of.Cells\", \"description\": \"Calculate and output the percentage (proportion) of cells in each sub-population?\"}, {\"name\": \"Calculate.means\", \"value\": \"TRUE\", \"label\": \"Calculate.Means\", \"description\": \"Calculate mean parameter values (e.g. Mean Fluorescence Intensities) for all parameters of each sub-population?\"}, {\"name\": \"Output.CSV.file.name\", \"value\": \"features.csv\", \"label\": \"Output.CSV.File.Name\", \"description\": \"The name of the output CSV file.\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"FCMSinglePanelQC\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Single panel flow cytometry data quality control\", \"required_args\": [{\"name\": \"Input.FCS.data.files\", \"value\": \"\", \"label\": \"Input.FCS.Data.Files\", \"description\": \"The set of input FCS files to be tested.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"FCMSinglePanelQC\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"Perform.cell.number.test\", \"value\": \"TRUE\", \"label\": \"Perform.Cell.Number.Test\", \"description\": \"Run the cell number control test.\"}, {\"name\": \"Perform.boundary.events.test\", \"value\": \"TRUE\", \"label\": \"Perform.Boundary.Events.Test\", \"description\": \"Run the boundary events test.\"}, {\"name\": \"Perform.timeflow.test\", \"value\": \"TRUE\", \"label\": \"Perform.Timeflow.Test\", \"description\": \"Run the timeflow test, i.e., check whether the flow rate was steady and uninterrupted.\"}, {\"name\": \"Perform.timeline.test\", \"value\": \"TRUE\", \"label\": \"Perform.Timeline.Test\", \"description\": \"Run the timeline test, i.e., detect unexpected non-random acquisition of fluorescence intensities.\"}, {\"name\": \"Zip.output\", \"value\": \"TRUE\", \"label\": \"Zip.Output\", \"description\": \"Zip the resulting report together?\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"FCSClean\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Clean up an FCS file by marking events affected by problems during data acquisition\", \"required_args\": [{\"name\": \"Input.FCS\", \"value\": \"\", \"label\": \"Input.FCS\", \"description\": \"The FCS file to be cleaned up, i.e., bad events/cells will be marked by a high value of the added Good_vs_Bad channel.\"}, {\"name\": \"Markers\", \"value\": \"\", \"label\": \"Markers\", \"description\": \"A comma-separated list of dimensions (flow cytometry parameters/channels) to be checked. The module accepts both, a list of marker names as well as a list of parameter indexes. All dimensions but scatters and time will be used if the Markers parameter is not provided.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"FCSClean\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"Output.FCS\", \"value\": \"<Input.FCS_basename>.cleaned.fcs\", \"label\": \"Output.FCS\", \"description\": \"The name of the output FCS file.\"}, {\"name\": \"Bin.Size\", \"value\": \"0.01\", \"label\": \"Bin.Size\", \"description\": \"The size of a bin as proportion of cells in the FCS file, typically between 0.01 and 0.2.\"}, {\"name\": \"Cell.Cutoff\", \"value\": \"500\", \"label\": \"Cell.Cutoff\", \"description\": \"The number of cells to be used for a cut off, typically between 300 and 800, see documentation for more details.\"}, {\"name\": \"Cutoff\", \"value\": \"median\", \"label\": \"Cutoff\", \"description\": \"The method of how cutoff is performed by the algorithm. Typically, a median works well. Alternatively, you may specify a value from the [0,1] interval as a cutoff percentile, or a value greater than 1, which will be assumed as a direct specification of the cutoff threshold.\"}, {\"name\": \"Max.Fc\", \"value\": \"1.3\", \"label\": \"Max.Fc\", \"description\": \"Relative increase from mean norm of presumed good data below which data is still presumed to be good.\"}, {\"name\": \"Announce\", \"value\": \"TRUE\", \"label\": \"Announce\", \"description\": \"Shall we print information about identifying vs. not identifying any bad events to the standard output?\"}, {\"name\": \"Diagnostic\", \"value\": \"FALSE\", \"label\": \"Diagnostic\", \"description\": \"Shall we include a diagnostics plot?\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"FCSNormalization\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Per-channel basis normalization of flow cytometry data.\", \"required_args\": [{\"name\": \"Input.FCS.files\", \"value\": \"\", \"label\": \"Input.FCS.Files\", \"description\": \"The set of input FCS files to be normalized.\"}, {\"name\": \"Channels.to.normalize\", \"value\": \"\", \"label\": \"Channels.To.Normalize\", \"description\": \"Comma-separated list of parameter short names to normalize.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"FCSNormalization\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"Apply.ASinH\", \"value\": \"TRUE\", \"label\": \"Apply.ASinH\", \"description\": \"Apply the asinh transformation prior normalization?\"}, {\"name\": \"Create.report\", \"value\": \"TRUE\", \"label\": \"Create.Report\", \"description\": \"Create additional histogram channel report?\"}, {\"name\": \"Landmarks.number\", \"value\": \"1\", \"label\": \"Landmarks.Number\", \"description\": \"Number of landmarks to be used for normalization.\"}, {\"name\": \"Zip.output\", \"value\": \"TRUE\", \"label\": \"Zip.Output\", \"description\": \"Zip the resulting normalized FCS data files together?\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"FcsToCsv\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Converts a Flow Cytometry Standard (FCS) file to a Comma Separated Values (CSV) file.\", \"required_args\": [{\"name\": \"Input.FCS.data.file\", \"value\": \"\", \"label\": \"Input.FCS.Data.File\", \"description\": \"An FCS data file to be converted to CSV.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"FcsToCsv\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"Output.CSV.file.name\", \"value\": \"<Input.FCS.data.file_basename>.csv\", \"label\": \"Output.CSV.File.Name\", \"description\": \"The name of the output CSV file.\"}, {\"name\": \"Use.full.names\", \"value\": \"TRUE\", \"label\": \"Use.Full.Names\", \"description\": \"Whether to try to extract full parameter names from the FCS file.\"}, {\"name\": \"Output.keywords.file.name\", \"value\": \"<Input.FCS.data.file_basename>\", \"label\": \"Output.Keywords.File.Name\", \"description\": \"The file name to store keyword value pairs extracted from the FCS file.\"}, {\"name\": \"Output.keywords.mode\", \"value\": \"CSV\", \"label\": \"Output.Keywords.Mode\", \"description\": \"How to store keywords.\"}, {\"name\": \"Channel.to.scale.conversion\", \"value\": \"TRUE\", \"label\": \"Channel.To.Scale.Conversion\", \"description\": \"Whether to convert log channel values to linear scale.\"}, {\"name\": \"Precision\", \"value\": \"FALSE\", \"label\": \"Precision\", \"description\": \"Whether to use regular (32bit, recommended) or double precision.\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"FeatureSummaryViewer\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Feature Summary Viewer\", \"required_args\": [{\"name\": \"feature.filename\", \"value\": \"\", \"label\": \"Feature.Filename\", \"description\": \"odf file with or without features to summarize\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"FeatureSummaryViewer\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"FileSplitter\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Splits a file into equal-sized chunks, given the number of lines per output file\", \"required_args\": [{\"name\": \"inputFile\", \"value\": \"\", \"label\": \"InputFile\", \"description\": \"The input file, to be split into chunks\"}, {\"name\": \"numFiles\", \"value\": \"\", \"label\": \"NumFiles\", \"description\": \"[Integer, >=1], The maximum number of output files to split the file into. When both 'numFiles' and 'numLines' are set, treat this as the maximum number of output files to create, even if the files have more than the requested number of lines\"}, {\"name\": \"numLines\", \"value\": \"\", \"label\": \"NumLines\", \"description\": \"[Integer, >=1], The number of lines in each output file.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"FileSplitter\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"FilterFilelist\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"a helper module which generates a filtered filelist from the input filelist and some filter parameters.\", \"required_args\": [{\"name\": \"input.filelist\", \"value\": \"\", \"label\": \"Input.Filelist\", \"description\": \"the initial input filelist, a file containing a list of filepaths\"}, {\"name\": \"includePattern\", \"value\": \"\", \"label\": \"IncludePattern\", \"description\": \"an optional list of filename patterns to match, e.g. *.cls,*.gct\"}, {\"name\": \"errorOnEmptySet\", \"value\": \"\", \"label\": \"ErrorOnEmptySet\", \"description\": \"Whether the module should output stderr if the output result is empty\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"FilterFilelist\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"type\", \"value\": \"f\", \"label\": \"Type\", \"description\": \"Include files, directories, or both files and directories in the output filelist\"}, {\"name\": \"output.filename\", \"value\": \"<input.filelist_basename>.filtered.<input.filelist_extension>\", \"label\": \"Output.Filename\", \"description\": \"the name of the output file, if none specified, write to stdout\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"FilterLowshRNAs\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Removes shRNAs with low read counts in original DNA plasmid pool. Outputs a GCT file containing filtered shRNA data and a GCT file containing the mean read count for each shRNA in the original DNA plasmid pool. \", \"required_args\": [{\"name\": \"GCT.Files\", \"value\": \"\", \"label\": \"GCT.Files\", \"description\": \"Please enter one or more GCT files.\"}, {\"name\": \"File.Prefix\", \"value\": \"\", \"label\": \"File.Prefix\", \"description\": \"\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"FilterLowshRNAs\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"Find.differentially.expressed.genes.in.RNASeq.data\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"This pipeline is used in the GenomeSpace recipe, \\\"Find differentially expressed genes in RNA-Seq data\\\". The recipe can be found at: http://www.genomespace.org/support/guides/recipes/sections/find-differentially-expressed-genes-in-rna-seq-data.\", \"required_args\": [{\"name\": \"ConvertLineEndings1.input.filename\", \"value\": \"\", \"label\": \"ConvertLineEndings1.Input.Filename\", \"description\": \"The input file (any non-binary file format)\"}, {\"name\": \"TopHat2.reads.pair.1\", \"value\": \"\", \"label\": \"TopHat2.Reads.Pair.1\", \"description\": \"Unpaired reads file or first mate for paired reads. One or more files containing reads in FASTA or FASTQ format (bz2 and gz compressed files are supported).\"}, {\"name\": \"TopHat4.reads.pair.1\", \"value\": \"\", \"label\": \"TopHat4.Reads.Pair.1\", \"description\": \"Unpaired reads file or first mate for paired reads. One or more files containing reads in FASTA or FASTQ format (bz2 and gz compressed files are supported).\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"Find.differentially.expressed.genes.in.RNASeq.data\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"Find.subnetworks.of.differentially.expressed.genes.and.identify.associated.biological.functions\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"This pipeline is used in the GenomeSpace recipe, \\\"Find subnetworks of differentially expressed genes and identify associated biological functions\\\". The recipe can be found at: http://www.genomespace.org/support/guides/recipes/sections/find-subnetworks-of-differentially-expressed-genes-and-identify-associated-biological-functions.\", \"required_args\": [{\"name\": \"PreprocessDataset1.input.filename\", \"value\": \"\", \"label\": \"PreprocessDataset1.Input.Filename\", \"description\": \"Input filename - .res, .gct, .odf\"}, {\"name\": \"ComparativeMarkerSelection2.cls.file\", \"value\": \"\", \"label\": \"ComparativeMarkerSelection2.Cls.File\", \"description\": \"The class file - .cls\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"Find.subnetworks.of.differentially.expressed.genes.and.identify.associated.biological.functions\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"FLAMEChooseOptimalClusterNumber\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Determines the optimal number of clusters for each sample, using user-specified criterion\", \"required_args\": [{\"name\": \"mixture.model\", \"value\": \"\", \"label\": \"Mixture.Model\", \"description\": \"A .zip file containing mixture modeling results from the FLAMEMixtureModel module\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"FLAMEChooseOptimalClusterNumber\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"method\", \"value\": \"SWR\", \"label\": \"Method\", \"description\": \"Criterion used for determining optimal number of clusters\"}, {\"name\": \"seed\", \"value\": \"123456\", \"label\": \"Seed\", \"description\": \"An integer; the seed fixes random computational processes during computation. This does not need to be changed under most circumstances.\"}, {\"name\": \"output.prefix\", \"value\": \"<mixture.model_basename>\", \"label\": \"Output.Prefix\", \"description\": \"A prefix for output files\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"FLAMEContourDataGenerator\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Generates data for drawing FLAME contour plots\", \"required_args\": [{\"name\": \"membership.file\", \"value\": \"\", \"label\": \"Membership.File\", \"description\": \"A membership.txt or member.txt file output from one of the FLAME modules. \"}, {\"name\": \"parameters.file\", \"value\": \"\", \"label\": \"Parameters.File\", \"description\": \"A parameters.txt file output from one of the FLAME modules. \"}, {\"name\": \"channels.to.plot\", \"value\": \"\", \"label\": \"Channels.To.Plot\", \"description\": \"A comma separated list of the index or names of three channels/antibodies to plot\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"FLAMEContourDataGenerator\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.prefix\", \"value\": \"<parameters.file_basename>\", \"label\": \"Output.Prefix\", \"description\": \"A prefix for output files\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"FLAMEContourViewer.Pipeline\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Pipeline which runs the FLAMECountorDataGenerator and the FLAMEViewer\", \"required_args\": [{\"name\": \"FLAMEContourDataGenerator1.membership.file\", \"value\": \"\", \"label\": \"FLAMEContourDataGenerator1.Membership.File\", \"description\": \"A membership.txt or member.txt file output from one of the FLAME modules.\"}, {\"name\": \"FLAMEContourDataGenerator1.parameters.file\", \"value\": \"\", \"label\": \"FLAMEContourDataGenerator1.Parameters.File\", \"description\": \"A parameters.txt file output from one of the FLAME modules.\"}, {\"name\": \"FLAMEContourDataGenerator1.channels.to.plot\", \"value\": \"\", \"label\": \"FLAMEContourDataGenerator1.Channels.To.Plot\", \"description\": \"A comma separated list of the index or names of three channels/antibodies to plot\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"FLAMEContourViewer.Pipeline\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"FLAMEContourDataGenerator1.output.prefix\", \"value\": \"flame_contour\", \"label\": \"FLAMEContourDataGenerator1.Output.Prefix\", \"description\": \"A prefix for output files\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"FLAMEMetacluster\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Matches populations/clusters across all samples in the dataset, assembles a table of all parameters in all samples\", \"required_args\": [{\"name\": \"optimal.g.mixture.model\", \"value\": \"\", \"label\": \"Optimal.G.Mixture.Model\", \"description\": \"A .zip file containing the optimal mixture modeling result of each sample from the output of the FLAMEChooseOptimalClusterNumber module.\"}, {\"name\": \"sample.class\", \"value\": \"\", \"label\": \"Sample.Class\", \"description\": \"\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"FLAMEMetacluster\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"estimate.mode\", \"value\": \"F\", \"label\": \"Estimate.Mode\", \"description\": \"Used only for skew distributions. Whether to estimate the mode for each cluster. Must be the same value used in FLAMEMixtureModel.\"}, {\"name\": \"estimation.increment\", \"value\": \"1\", \"label\": \"Estimation.Increment\", \"description\": \"Used only for skew distributions and when estimate mode is set to yes. The smaller the increment, the more accurate the estimation, but the slower the estimation step. Must be the same value used in FLAMEMixtureModel.\"}, {\"name\": \"output.intermediate.results\", \"value\": \"F\", \"label\": \"Output.Intermediate.Results\", \"description\": \"Choose whether to output the intermediate metaclustering results, such as within-class matching results.\"}, {\"name\": \"output.prefix\", \"value\": \"<optimal.g.mixture.model_basename>\", \"label\": \"Output.Prefix\", \"description\": \"A prefix for output files.\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"FLAMEMixtureModel\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Clusters and estimates population parameters, using preprocessed flow cytometric data\", \"required_args\": [{\"name\": \"preprocessed.data\", \"value\": \"\", \"label\": \"Preprocessed.Data\", \"description\": \"A .zip file containing preprocessed flow sample files in tab-delimited .txt format.\"}, {\"name\": \"channels.to.cluster\", \"value\": \"\", \"label\": \"Channels.To.Cluster\", \"description\": \"A comma-separated list of either channel numbers (e.g., 1, 2, 3, 7) or channel names (e.g., CD4, CD45RA, ZAP70); these denote which antibodies in the (preprocessed) data sample to be used for clustering.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"FLAMEMixtureModel\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"g.min\", \"value\": \"3\", \"label\": \"G.Min\", \"description\": \"An integer; minimal number of clusters to be fit for each sample.\"}, {\"name\": \"g.max\", \"value\": \"10\", \"label\": \"G.Max\", \"description\": \"An integer; maximal number of clusters to be fit for each sample. g max should be greater or equal to g min.\"}, {\"name\": \"density\", \"value\": \"skewt\", \"label\": \"Density\", \"description\": \"Density distribution to be used for clustering\"}, {\"name\": \"estimate.mode\", \"value\": \"F\", \"label\": \"Estimate.Mode\", \"description\": \"Used only for skew distributions. Whether to estimate the mode for each cluster.  Must be the same value used in FLAMEMixtureModel.\"}, {\"name\": \"estimation.increment\", \"value\": \"1\", \"label\": \"Estimation.Increment\", \"description\": \"Used only for skew distributions and when estimate mode is set to yes. The smaller the increment, the more accurate the estimation, but the slower the estimation step. Must be the same value used in FLAMEMixtureModel.\"}, {\"name\": \"seed\", \"value\": \"123456\", \"label\": \"Seed\", \"description\": \"An integer; the seed fixes random computational processes during computation. This does not need to be changed under most circumstances.\"}, {\"name\": \"output.prefix\", \"value\": \"<preprocessed.data_basename>\", \"label\": \"Output.Prefix\", \"description\": \"A prefix for output files.\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"FLAMEPreprocess\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Performs preprocessing of .fcs or .txt flow cytometric data files\", \"required_args\": [{\"name\": \"dataset\", \"value\": \"\", \"label\": \"Dataset\", \"description\": \"A .zip file containing flow files in .txt or .fcs format. (The files should be all .txt or all .fcs, but not both.)\"}, {\"name\": \"channels\", \"value\": \"\", \"label\": \"Channels\", \"description\": \"A comma-separated list of channel numbers indicating which columns (antibodies) to keep (e.g., 1, 2, 3, 7).\"}, {\"name\": \"channel.names\", \"value\": \"\", \"label\": \"Channel.Names\", \"description\": \"A comma-separated list of channel/antibody names that corresponds to the channel numbers (e.g., FSC, SSC, CD4, CD45RA).\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"FLAMEPreprocess\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"file.type\", \"value\": \"fcs\", \"label\": \"File.Type\", \"description\": \"File type of input sample data.\"}, {\"name\": \"data.scale\", \"value\": \"262144\", \"label\": \"Data.Scale\", \"description\": \"The original data scale. 18-bit (default) or 4-decade.\"}, {\"name\": \"remove.dead\", \"value\": \"F\", \"label\": \"Remove.Dead\", \"description\": \"Whether to attempt removing dead cells from the samples.\"}, {\"name\": \"scatter.channels\", \"value\": \"1,2\", \"label\": \"Scatter.Channels\", \"description\": \"A comma-separated list of scatter channel numbers (e.g., 1, 2, 3).\"}, {\"name\": \"transformation\", \"value\": \"logicle\", \"label\": \"Transformation\", \"description\": \"The transformation to apply\"}, {\"name\": \"logicle.cofactor\", \"value\": \"3\", \"label\": \"Logicle.Cofactor\", \"description\": \"A cofactor that tunes the logicle transformation. The cofactor is usually greater than 0 and less than 10.\"}, {\"name\": \"arsinh.cofactor\", \"value\": \"250\", \"label\": \"Arsinh.Cofactor\", \"description\": \"A cofactor that tunes the arsinh transformation. The cofactor is 0 usually between 0 exclusively and a few thousand.\"}, {\"name\": \"output.prefix\", \"value\": \"<dataset_basename>\", \"label\": \"Output.Prefix\", \"description\": \"A prefix for output files\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"FLAMEPreviewTransformation\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Evaluates data transformations based on one representative sample\", \"required_args\": [{\"name\": \"data.file\", \"value\": \"\", \"label\": \"Data.File\", \"description\": \"A flow file in .txt or .fcs format for one sample.\"}, {\"name\": \"channels\", \"value\": \"\", \"label\": \"Channels\", \"description\": \"A comma-separated list of channel numbers indicating which columns (antibodies) to keep (e.g., 1, 2, 3, 7).\"}, {\"name\": \"channel.names\", \"value\": \"\", \"label\": \"Channel.Names\", \"description\": \"A comma-separated list of channel or antibody names that corresponds to the channel numbers (e.g., FSC, SSC, CD4, CD45RA).\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"FLAMEPreviewTransformation\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"file.type\", \"value\": \"fcs\", \"label\": \"File.Type\", \"description\": \"File type of input sample data.\"}, {\"name\": \"data.scale\", \"value\": \"262144\", \"label\": \"Data.Scale\", \"description\": \"The original data scale. 18-bit (default) or 4-decade.\"}, {\"name\": \"scatter.channels\", \"value\": \"1,2\", \"label\": \"Scatter.Channels\", \"description\": \"A comma-separated list of scatter channel numbers (e.g., 1, 2, 3).\"}, {\"name\": \"transformation\", \"value\": \"logicle\", \"label\": \"Transformation\", \"description\": \"The transformation(s) to apply\"}, {\"name\": \"logicle.low\", \"value\": \"0.5\", \"label\": \"Logicle.Low\", \"description\": \"The cofactor that tunes the logicle transformation is usually between 0 and 10 exclusively. Specify the lower limit of the cofactor range to examine.\"}, {\"name\": \"logicle.high\", \"value\": \"5\", \"label\": \"Logicle.High\", \"description\": \"The cofactor that tunes the logicle transformation is usually greater than 0 and less than 10. Specify the upper limit of the cofactor range to examine.\"}, {\"name\": \"arsinh.low\", \"value\": \"1\", \"label\": \"Arsinh.Low\", \"description\": \"The cofactor that tunes the arsinh transformation is usually between 0 and a few thousand. Specify the lower limit of the cofactor range to examine.\"}, {\"name\": \"arsinh.high\", \"value\": \"1000\", \"label\": \"Arsinh.High\", \"description\": \"The cofactor that tunes the arsinh transformation is usually between 0 and a few thousand. Specify the upper limit of the cofactor range to examine.\"}, {\"name\": \"output.prefix\", \"value\": \"<data.file_basename>\", \"label\": \"Output.Prefix\", \"description\": \"A prefix for output files\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"FLAMEViewer\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Visualizes results from the FLAME method\", \"required_args\": [{\"name\": \"membership.file\", \"value\": \"\", \"label\": \"Membership.File\", \"description\": \"A membership.txt file output from the FLAMEContourDataGenerator module. This membership file must match with the contour plot file.\"}, {\"name\": \"contour.plot.file\", \"value\": \"\", \"label\": \"Contour.Plot.File\", \"description\": \"The contour plot output file from the FLAMEContourDataGenerator module. This contour plot file must match with the membership file.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"FLAMEViewer\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"FlowClustClassifyFCS\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Classify flow cytometry data using the FlowClust algorithm.\", \"required_args\": [{\"name\": \"Input.FCS.data.file\", \"value\": \"\", \"label\": \"Input.FCS.Data.File\", \"description\": \"The FCS file to be clustered.\"}, {\"name\": \"K\", \"value\": \"\", \"label\": \"K\", \"description\": \"The number of populations that the algorithm shall be looking for. If you do not know the K value, you can use the SuggestNumberOfPopulations module to estimate it.\"}, {\"name\": \"Channels\", \"value\": \"\", \"label\": \"Channels\", \"description\": \"A comma-separated list of parameter short names ($PnN values) that shall be used to classify the populations. All parameters will be used by default, i.e., if no value for the Channels parameter is provided. The Time parameter is excluded automatically.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"FlowClustClassifyFCS\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"CSV.file.name\", \"value\": \"<Input.FCS.data.file_basename>\", \"label\": \"CSV.File.Name\", \"description\": \"The base file name to split the results into CSV files, each containing a single population.\"}, {\"name\": \"CLS.file.name\", \"value\": \"<Input.FCS.data.file_basename>.cls\", \"label\": \"CLS.File.Name\", \"description\": \"The name of the CLS output file, which will store the population number for each event.\"}, {\"name\": \"Quantile\", \"value\": \"0.9\", \"label\": \"Quantile\", \"description\": \"Outlier quantile. Events outside the specified quantile region will be classified as outliers and be assigned to class 0.\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"FlowFingerprinting\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"takes a flow set and generates a description of the multivariate probability distribution function of its flow cytometry data in the form of a fingerprint\", \"required_args\": [{\"name\": \"Input.FCS.data.files\", \"value\": \"\", \"label\": \"Input.FCS.Data.Files\", \"description\": \"The FCS file(s) to be fingerprinted.\"}, {\"name\": \"Model.FCS.data.file\", \"value\": \"\", \"label\": \"Model.FCS.Data.File\", \"description\": \"Training data (FCS file) for the underlying model. If none, the input FCS data is used for creating a default model.\"}, {\"name\": \"Model.parameters\", \"value\": \"\", \"label\": \"Model.Parameters\", \"description\": \"A comma-separated list of dimensions/parameters/channels that shall be used for the underlying model. Use either parameter names or indexes. All dimensions except for Time will be used if this parameter is not provided.\"}, {\"name\": \"Model.number.of.recursions\", \"value\": \"\", \"label\": \"Model.Number.Of.Recursions\", \"description\": \"Number of times the FCS training data will be subdivided.\"}, {\"name\": \"Model.sample.size\", \"value\": \"\", \"label\": \"Model.Sample.Size\", \"description\": \"Used to specify the per flowFrame sample size of the data to use in model generation. If NULL, all of the data in x is used. Setting this to a smaller number will speed up processing, at the cost of accuracy.\"}, {\"name\": \"Workflow.parameter.file\", \"value\": \"\", \"label\": \"Workflow.Parameter.File\", \"description\": \"A CSV file containing parameters if module is used in pipeline.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"FlowFingerprinting\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"Model.dequantize\", \"value\": \"0\", \"label\": \"Model.Dequantize\", \"description\": \"Option to make all events in the training set unique.\"}, {\"name\": \"Output.QC.screening.fingerprint.stack\", \"value\": \"0\", \"label\": \"Output.QC.Screening.Fingerprint.Stack\", \"description\": \"Graphical output in form of \\\"stack\\\" plot.\"}, {\"name\": \"Output.QC.screening.gate.consistency\", \"value\": \"0\", \"label\": \"Output.QC.Screening.Gate.Consistency\", \"description\": \"Graphical output in form of \\\"qc\\\" plot.\"}, {\"name\": \"Output.QC.screening.plate\", \"value\": \"1\", \"label\": \"Output.QC.Screening.Plate\", \"description\": \"Graphical output in form of \\\"plate\\\" plot.\"}, {\"name\": \"Output.csv.file\", \"value\": \"fingerprints.csv\", \"label\": \"Output.Csv.File\", \"description\": \"Name for output CSV file.\"}, {\"name\": \"Output.file.transformation\", \"value\": \"0\", \"label\": \"Output.File.Transformation\", \"description\": \"Transformation used on data for CSV output file. \\\"raw\\\" is the number of events in each bin, \\\"normalized\\\" is the ratio of the number of events divided by the expected number (the expected number of events is calculated by dividing the total number of events in the instance used to make the fingerprint by the number of features in the fingerprint). \\\"log2norm\\\" is the log2 of the normalized values.\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"FlowMeansCluster\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Cluster flow cytometry data using the FlowMeans algorithm.\", \"required_args\": [{\"name\": \"Input.FCS.data.file\", \"value\": \"\", \"label\": \"Input.FCS.Data.File\", \"description\": \"The FCS file to be clustered.\"}, {\"name\": \"Dimensions\", \"value\": \"\", \"label\": \"Dimensions\", \"description\": \"A comma-separated list of dimensions (parameters/channels) that shall be used for clustering. Use either parameter names or indexes. All dimensions except for Time will be used if this parameter is not provided.\"}, {\"name\": \"Dimensions.to.transform\", \"value\": \"\", \"label\": \"Dimensions.To.Transform\", \"description\": \"A comma-separated list of dimensions (parameters/channels) that shall transformed as specified by previous parameter. This will be ignored if no transformation is specified above. If this parameter is not provided and transformation is specified above, the algorithm will use heuristics to identify parameters that shall be transformed. These heuristics are based on how parameters are stored in the FCS file, their resolution and their name. Again, you can use either parameter names or parameter indexes to specify dimensions to transform.\"}, {\"name\": \"Maximum.number.of.clusters\", \"value\": \"\", \"label\": \"Maximum.Number.Of.Clusters\", \"description\": \"Maximum number of clusters that FlowMeans will try. This value will be estimated automatically if the value of this parameter is not provided.\"}, {\"name\": \"Number.of.clusters\", \"value\": \"\", \"label\": \"Number.Of.Clusters\", \"description\": \"Number of clusters that FlowMeans shall search for. This parameter cannot be combined with the \\\"Maximum number of clusters\\\". Also, the number of clusters will be estimated automatically if this parameter is not provided.\"}, {\"name\": \"Max.points.for.covariance\", \"value\": \"\", \"label\": \"Max.Points.For.Covariance\", \"description\": \"The maximum number of data points (events) used for calculating the covariance. All data points will be used if this value is not provided.\"}, {\"name\": \"Max.points.for.kernel.density\", \"value\": \"\", \"label\": \"Max.Points.For.Kernel.Density\", \"description\": \"The maximum number of data points (events) used for counting the modes using kernel density estimation. All data points will be used if this value is not provided.\"}, {\"name\": \"Seed\", \"value\": \"\", \"label\": \"Seed\", \"description\": \"Seed value for the random generator; provide any consistent value to obtain reproducible results.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"FlowMeansCluster\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"Transformation\", \"value\": \"ASinH\", \"label\": \"Transformation\", \"description\": \"Which transformation to apply on the data prior clustering. Fluorescence channels are usually better visualized and clustered using a transformation; the ASinH transformation produces good results on most data.\"}, {\"name\": \"Maximum.number.of.iterations\", \"value\": \"50\", \"label\": \"Maximum.Number.Of.Iterations\", \"description\": \"The maximum number of iterations allowed within the K-means component of the clustering.\"}, {\"name\": \"Number.of.initialization.sets\", \"value\": \"10\", \"label\": \"Number.Of.Initialization.Sets\", \"description\": \"The number of random sets used for initialization.\"}, {\"name\": \"Distance.used\", \"value\": \"TRUE\", \"label\": \"Distance.Used\", \"description\": \"The distance being used within the FlowMeans algorithm.\"}, {\"name\": \"Standardize.data\", \"value\": \"TRUE\", \"label\": \"Standardize.Data\", \"description\": \"Should each data dimension be standardized (i.e, transformed to a [0,1] interval) before clustering. This usually does not harm and is important if data in dimensions used for clustering covers significantly different ranges for different dimensions.   \"}, {\"name\": \"Distance.matrix.update\", \"value\": \"Mahalanobis\", \"label\": \"Distance.Matrix.Update\", \"description\": \"How to update the distance matrix at each merging iteration. Note that updating using Mahalanobis distance is not possible if Euclidean distance is set in the \\\"Distance used\\\" above.  \"}, {\"name\": \"Transform.residuals\", \"value\": \"FALSE\", \"label\": \"Transform.Residuals\", \"description\": \"Indicate whether or not to transform residuals to orthogonal distance.   \"}, {\"name\": \"Add.noise\", \"value\": \"TRUE\", \"label\": \"Add.Noise\", \"description\": \"Should a little bit of uniform noise be added to the data to prevent singularity issues?\"}, {\"name\": \"Draw.plots\", \"value\": \"TRUE\", \"label\": \"Draw.Plots\", \"description\": \"Should plots of clustered data be created for quick visual inspection of the results?\"}, {\"name\": \"Print.debug.info\", \"value\": \"FALSE\", \"label\": \"Print.Debug.Info\", \"description\": \"Should additional debugging information be provided (e.g., this may be useful to debug cases where clustering does not work properly for certain data files).\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"FlowMergeCluster\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Cluster flow cytometry data using FlowMerge.\", \"required_args\": [{\"name\": \"Input.FCS.data.file\", \"value\": \"\", \"label\": \"Input.FCS.Data.File\", \"description\": \"The FCS file to be clustered.\"}, {\"name\": \"Dimensions\", \"value\": \"\", \"label\": \"Dimensions\", \"description\": \"A comma-separated list of dimensions (parameters/channels) that shall be used for clustering. Use either parameter names or indexes. All dimensions except for Time will be used if this parameter is not provided.\"}, {\"name\": \"Dimensions.to.transform\", \"value\": \"\", \"label\": \"Dimensions.To.Transform\", \"description\": \"A comma-separated list of dimensions (parameters/channels) that shall transformed as specified by previous parameter. This will be ignored if no transformation is specified above. If this parameter is not provided and transformation is specified above, the algorithm will use heuristics to identify parameters that shall be transformed. These heuristics are based on how parameters are stored in the FCS file, their resolution and their name. Again, you can use either parameter names or parameter indexes to specify dimensions to transform.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"FlowMergeCluster\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"Transformation\", \"value\": \"ASinH\", \"label\": \"Transformation\", \"description\": \"Which transformation to apply on the data prior clustering. Fluorescence channels are usually better visualized and clustered using a transformation; the ASinH transformation produces good results on most data.\"}, {\"name\": \"Range.for.number.of.clusters\", \"value\": \"1-10\", \"label\": \"Range.For.Number.Of.Clusters\", \"description\": \"The range for the number of subpopulations (clusters) that FlowMerge shall search for. FlowMerge will try to pick the best number of clusters from the specified range, which shall be provided in the min-max format, where both, mix and max are integers and min is smaller than max, e.g., 1-10.\"}, {\"name\": \"Estimate.degrees.of.freedom\", \"value\": \"0\", \"label\": \"Estimate.Degrees.Of.Freedom\", \"description\": \"An indication whether to estimate the degrees of freedom used for the t distribution when modelling data.\"}, {\"name\": \"Degrees.of.freedom\", \"value\": \"4\", \"label\": \"Degrees.Of.Freedom\", \"description\": \"The degrees of freedom used for the t distribution when modelling data. Provided value will be ignored if estimation is requested by the Estimate degrees of freedom parameter. Gaussian distribution will be used if Degrees of freedom are not provided and estimation is not requested.\"}, {\"name\": \"Number.of.computing.nodes\", \"value\": \"1\", \"label\": \"Number.Of.Computing.Nodes\", \"description\": \"How many nodes (e.g., processors, cores) to use if you wish to run the analysis in a parallel mode? Enter '1' if you wish do NOT want to use the parallel mode. Enter a number higher than 1 if your server/cluster has multiple computers/processors/cores and you want to utilize several of these for FlowMerge clustering.\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"Fpkm_trackingToGct\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Takes a Cufflinks FPKM_tracking file and converts it to GCT format.\", \"required_args\": [{\"name\": \"input.file\", \"value\": \"\", \"label\": \"Input.File\", \"description\": \"Input file in FPKM_TRACKING format\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"Fpkm_trackingToGct\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"row.labels\", \"value\": \"Tracking ID\", \"label\": \"Row.Labels\", \"description\": \"Select row labels to be written to GCT file\"}, {\"name\": \"row.descriptions\", \"value\": \"Locus\", \"label\": \"Row.Descriptions\", \"description\": \"Select row descriptions to be written to GCT file\"}, {\"name\": \"filter.rows\", \"value\": \"yes\", \"label\": \"Filter.Rows\", \"description\": \"Whether to filter rows with invalid row names (ie., \\\"-\\\")\"}, {\"name\": \"output.prefix\", \"value\": \"<input.file_basename>\", \"label\": \"Output.Prefix\", \"description\": \"A label that will be used to name the GCT output file.\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"FPmatching\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Matches Fluidigm fingerprint report to reference set of CCLE SNP6.0 array data.\\nTakes old Fluidigm software report or new PED style report (saved as .csv).\\nAlso backwards compatible with Sequenom reports (Combine 100k and AffySNP6.0 reports together into one file first).\", \"required_args\": [{\"name\": \"platereport\", \"value\": \"\", \"label\": \"Platereport\", \"description\": \"Should be .csv format\"}, {\"name\": \"platemapping\", \"value\": \"\", \"label\": \"Platemapping\", \"description\": \"Should be .csv format\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"FPmatching\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"format\", \"value\": \"new\", \"label\": \"Format\", \"description\": \"New PED style report OR Old Fluidigm software format OR Sequenom style report (expects combined 100k and SNP6.0 file)\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"Gct2Arff\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Description: Convert a gene pattern .gct / .cls file pair into an .arff file\", \"required_args\": [{\"name\": \"gct.file\", \"value\": \"\", \"label\": \"Gct.File\", \"description\": \"The dataset features - .gct\"}, {\"name\": \"cls.file\", \"value\": \"\", \"label\": \"Cls.File\", \"description\": \"The dataset class labels - .cls\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"Gct2Arff\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.file\", \"value\": \"<gct.file_basename>\", \"label\": \"Output.File\", \"description\": \"The output file base name\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"GctToPcl\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Convert .gct to .pcl\", \"required_args\": [{\"name\": \"input.filename\", \"value\": \"\", \"label\": \"Input.Filename\", \"description\": \"GCT file to convert\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"GctToPcl\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.filename\", \"value\": \"<input.filename_basename>.pcl\", \"label\": \"Output.Filename\", \"description\": \"PCL file that will be produced\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"GENE_E\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Launches GENE-E with input file(s)\", \"required_args\": [{\"name\": \"input.file\", \"value\": \"\", \"label\": \"Input.File\", \"description\": \"Dataset to be viewed in GENE-E\"}, {\"name\": \"sample.information.or.class.file\", \"value\": \"\", \"label\": \"Sample.Information.Or.Class.File\", \"description\": \"The cls or sample info file\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"GENE_E\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"GeneListSignificanceViewer\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Views the results of marker analysis\", \"required_args\": [{\"name\": \"inputfilename\", \"value\": \"\", \"label\": \"Inputfilename\", \"description\": \"odf format file from marker analysis (GeneNeighbors or ClassNeighbors)\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"GeneListSignificanceViewer\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"GeneNeighbors\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Select genes that most closely resemble a continuous profile (e.g., another gene)\", \"required_args\": [{\"name\": \"data.filename\", \"value\": \"\", \"label\": \"Data.Filename\", \"description\": \"data file - .gct, .res, .odf\"}, {\"name\": \"gene.accession\", \"value\": \"\", \"label\": \"Gene.Accession\", \"description\": \"reference gene accession from data file to find neighbors for\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"GeneNeighbors\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"num.neighbors\", \"value\": \"50\", \"label\": \"Num.Neighbors\", \"description\": \"number of neighbors to find\"}, {\"name\": \"marker.list.file\", \"value\": \"<data.filename_basename>.markerlist\", \"label\": \"Marker.List.File\", \"description\": \"output filename for analysis results - .odf file\"}, {\"name\": \"marker.dataset.file\", \"value\": \"<data.filename_basename>.markerdata\", \"label\": \"Marker.Dataset.File\", \"description\": \"output filename raw data for selected markers - .gct file\"}, {\"name\": \"distance.metric\", \"value\": \"3\", \"label\": \"Distance.Metric\", \"description\": \"continuous metric for finding neighbors\"}, {\"name\": \"filter.data\", \"value\": \"no\", \"label\": \"Filter.Data\", \"description\": \"if no, values below will be ignored\"}, {\"name\": \"min.threshold\", \"value\": \"20\", \"label\": \"Min.Threshold\", \"description\": \"minimum threshold for data\"}, {\"name\": \"max.threshold\", \"value\": \"20000\", \"label\": \"Max.Threshold\", \"description\": \"maximum threshold for data\"}, {\"name\": \"min.fold.diff\", \"value\": \"5\", \"label\": \"Min.Fold.Diff\", \"description\": \"minimum fold difference for filtering genes\"}, {\"name\": \"min.abs.diff\", \"value\": \"50\", \"label\": \"Min.Abs.Diff\", \"description\": \"minimum absolute difference for filtering genes\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"GenePatternDocumentExtractor\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Extracts GenePattern pipelines and other embedded GenePattern data from Word 2007 documents created with the Microsoft Word Add-In for the GenePattern Reproducible Research Document.\", \"required_args\": [{\"name\": \"input.filename\", \"value\": \"\", \"label\": \"Input.Filename\", \"description\": \"The file you want to extract the data from. Note that this must be a Word 2007 (.docx) file -- Files saved as Word 2003 or earlier (2003) may be compatible with the Add-In, but will not work with the extractor.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"GenePatternDocumentExtractor\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"GEOImporter\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Imports data from the Gene Expression Omnibus (GEO)\", \"required_args\": [{\"name\": \"GEO.accession\", \"value\": \"\", \"label\": \"GEO.Accession\", \"description\": \"GEO Series record (GSExxx) or a GEO Dataset (GDSxxx). Either a GEO accession or a GEO SOFT file must be given.\"}, {\"name\": \"GEO.SOFT.file\", \"value\": \"\", \"label\": \"GEO.SOFT.File\", \"description\": \"A previously downloaded GEO SOFT file or its gzipped representation. Either a GEO SOFT file or a GEO accession must be given.\"}, {\"name\": \"ftp.proxy.server\", \"value\": \"\", \"label\": \"Ftp.Proxy.Server\", \"description\": \"FTP Proxy server\"}, {\"name\": \"ftp.proxy.username\", \"value\": \"\", \"label\": \"Ftp.Proxy.Username\", \"description\": \"FTP Proxy username\"}, {\"name\": \"ftp.proxy.password\", \"value\": \"\", \"label\": \"Ftp.Proxy.Password\", \"description\": \"FTP Proxy password\"}, {\"name\": \"http.proxy.server\", \"value\": \"\", \"label\": \"Http.Proxy.Server\", \"description\": \"HTTP Proxy server\"}, {\"name\": \"http.proxy.username\", \"value\": \"\", \"label\": \"Http.Proxy.Username\", \"description\": \"HTTP Proxy username\"}, {\"name\": \"http.proxy.password\", \"value\": \"\", \"label\": \"Http.Proxy.Password\", \"description\": \"HTTP Proxy password\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"GEOImporter\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"data.column.name\", \"value\": \"VALUE\", \"label\": \"Data.Column.Name\", \"description\": \"The name of the column that contains the expression values\"}, {\"name\": \"output.filename\", \"value\": \"<GEO.accession><GEO.SOFT.file_basename>\", \"label\": \"Output.Filename\", \"description\": \"The name of the output file\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"GetDatasetInSilico\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"downloads a compressed .tgz file from the Insilico servers and extract it\", \"required_args\": [{\"name\": \"DownloadURL1.URL\", \"value\": \"\", \"label\": \"DownloadURL1.URL\", \"description\": \"The URL to retrieve the file from\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"GetDatasetInSilico\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"GISTIC_2.0\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Genomic Identification of Significant Targets in Cancer (version 2.0.22)\", \"required_args\": [{\"name\": \"refgene.file\", \"value\": \"\", \"label\": \"Refgene.File\", \"description\": \"The reference genome file input provides cytoband and gene location information.\"}, {\"name\": \"seg.file\", \"value\": \"\", \"label\": \"Seg.File\", \"description\": \"The segmentation file contains the segmented data for all the samples identified by GLAD, CBS, or some other segmentation algorithm. (See GLAD file format in the Genepattern file formats documentation.) It is a six column, tab-delimited file with an optional first line identifying the columns. Positions are in base pair units.\"}, {\"name\": \"markers.file\", \"value\": \"\", \"label\": \"Markers.File\", \"description\": \"The markers file identifies the marker names and positions of the markers in the original dataset (before segmentation). It is a three column, tab-delimited file with an optional header. If not already, markers are sorted by genomic position.\"}, {\"name\": \"array.list.file\", \"value\": \"\", \"label\": \"Array.List.File\", \"description\": \"The array list file is an optional file identifying the subset of samples to be used in the analysis. It is a one column file with an optional header. The sample identifiers listed in the array list file must match the sample names given in the segmentation file.\"}, {\"name\": \"cnv.file\", \"value\": \"\", \"label\": \"Cnv.File\", \"description\": \"Specifies germline CNV (or other highly variant) regions to exclude from significance analysis. There are two forms for the cnv file: one allows CNVs to be identified by marker names in a single column; the other identifies excluded regions using their genomic location.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"GISTIC_2.0\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"gene.gistic\", \"value\": \"1\", \"label\": \"Gene.Gistic\", \"description\": \"Flag indicating that the gene GISTIC algorithm should be used to calculate the significant regions of deletion using gene locations instead of a marker locations.\"}, {\"name\": \"amplifications.threshold\", \"value\": \"0.1\", \"label\": \"Amplifications.Threshold\", \"description\": \"Threshold for copy number amplifications. Regions with a log2 ratio above this value are considered amplified.\"}, {\"name\": \"deletions.threshold\", \"value\": \"0.1\", \"label\": \"Deletions.Threshold\", \"description\": \"Threshold for copy number deletions. Regions with a log2 ratio below the negative of this value are considered deletions.\"}, {\"name\": \"join.segment.size\", \"value\": \"4\", \"label\": \"Join.Segment.Size\", \"description\": \"Smallest number of markers to allow in segments from the segmented data. Segments that contain a number of markers less than or equal to this number are joined to the neighboring segment that is closest in copy number.\"}, {\"name\": \"q-value.threshold\", \"value\": \"0.25\", \"label\": \"Q-Value.Threshold\", \"description\": \"Maximum false discovery rate for a region to be considered significant.\"}, {\"name\": \"remove.X\", \"value\": \"1\", \"label\": \"Remove.X\", \"description\": \"Flag indicating whether to remove data from the X-chromosome before analysis.\"}, {\"name\": \"cap.val\", \"value\": \"1.5\", \"label\": \"Cap.Val\", \"description\": \"Minimum and maximum cap values on analyzed data. Regions with a log2 ratio greater than the cap are set to the cap value; regions with a log2 ratio less than -cap value are set to -cap.\"}, {\"name\": \"confidence.level\", \"value\": \"0.90\", \"label\": \"Confidence.Level\", \"description\": \"Confidence level used to calculate the region containing a driver.\"}, {\"name\": \"run.broad.analysis\", \"value\": \"0\", \"label\": \"Run.Broad.Analysis\", \"description\": \"Flag indicating whether an additional broad-level analysis should be performed.\"}, {\"name\": \"focal.length.cutoff\", \"value\": \"0.50\", \"label\": \"Focal.Length.Cutoff\", \"description\": \"Threshold used to distinguish broad from focal events, given in units of fraction of chromosome arm.\"}, {\"name\": \"max.sample.segs\", \"value\": \"2500\", \"label\": \"Max.Sample.Segs\", \"description\": \"Maximum number of segments allowed for a sample in the input data. Samples with more segments than this threshold are excluded from the analysis.\"}, {\"name\": \"arm.peel\", \"value\": \"0\", \"label\": \"Arm.Peel\", \"description\": \"Whether to perform arm level peel off. This helps separate peaks which cleans up noise.\"}, {\"name\": \"sample.center\", \"value\": \"median\", \"label\": \"Sample.Center\", \"description\": \"Method for centering each sample prior to the GISTIC analysis.\"}, {\"name\": \"gene.collapse.method\", \"value\": \"extreme\", \"label\": \"Gene.Collapse.Method\", \"description\": \"Method for reducing marker-level copy number data to the gene-level copy number data in the gene tables. Markers contained in the gene are used when available, otherwise the flanking marker or markers are used.\"}, {\"name\": \"output.prefix\", \"value\": \"<seg.file_basename>\", \"label\": \"Output.Prefix\", \"description\": \"The prefix for the output file name\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"GISTICPreprocess\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Prepares SNP files for GISTIC\", \"required_args\": [{\"name\": \"snp.archive\", \"value\": \"\", \"label\": \"Snp.Archive\", \"description\": \"A zipfile containing the input genotypes and signal intensities. These are .cnx files created by SNPFileCreator.\"}, {\"name\": \"sample.info.file\", \"value\": \"\", \"label\": \"Sample.Info.File\", \"description\": \"The sample info file is a tab-delimited text file (*.txt) containing information about the arrays.\"}, {\"name\": \"array.list.file\", \"value\": \"\", \"label\": \"Array.List.File\", \"description\": \"The array list file identifies the subset of samples to be used in the analysis. It is a one-column file with a header (??array??). The sample identifiers listed in the array list file must match the sample names given in the segmentation file.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"GISTICPreprocess\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"use.paired.normals\", \"value\": \"0\", \"label\": \"Use.Paired.Normals\", \"description\": \"When set to 1, normalizes tumors against paired normals. (Default = 0; best results are usually achieved under this default setting)\"}, {\"name\": \"quality.control.tumors\", \"value\": \"1\", \"label\": \"Quality.Control.Tumors\", \"description\": \"When set to 1, performs histogram quality control on tumor samples and includes only those tumors with distinct copy number peaks in the output array list to be used in the GISTIC module. (Default = 1).\"}, {\"name\": \"normal.selection.method\", \"value\": \"1\", \"label\": \"Normal.Selection.Method\", \"description\": \"Method for selecting normals to use in sample normalization. Allowed values: 0 (use all normals), 1 (use the n normals most closely matched to the tumor??s noise profile). (Default = 1)\"}, {\"name\": \"n.closest.n\", \"value\": \"5\", \"label\": \"N.Closest.N\", \"description\": \"Number of normals to choose when using closest_n normalization. (Default = 5)\"}, {\"name\": \"correct.batch.effect\", \"value\": \"0\", \"label\": \"Correct.Batch.Effect\", \"description\": \"When set to 1, runs module to correct for marker intensity variations due to batch effect. (Default = 0)\"}, {\"name\": \"batch.correct.minimum.size\", \"value\": \"5\", \"label\": \"Batch.Correct.Minimum.Size\", \"description\": \"Minimum number of samples required in a batch for the batch to be eligible for batch correction. (Default = 5)\"}, {\"name\": \"batch.correct.pvalue\", \"value\": \"0.001\", \"label\": \"Batch.Correct.Pvalue\", \"description\": \"p-value threshold for batch effect correction. For each marker, signal intensities in each batch are compared to all others using a t-test. If the p-value for this comparison is less than the p-value threshold, the signal intensities in this batch will be brought to the mean of all other batches. (Default = 0.001)\"}, {\"name\": \"verbosity\", \"value\": \"0\", \"label\": \"Verbosity\", \"description\": \"Set verbosity level (stand-alone version only). At a verbosity level of 0, no text indicating program status is written to sdout. At a level of 30, the maximum amount of text is written to stdout. (Default = 0)\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"GLAD\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Gain and Loss Analysis of DNA\", \"required_args\": [{\"name\": \"copy.number.input.file\", \"value\": \"\", \"label\": \"Copy.Number.Input.File\", \"description\": \"cn file that contains sorted SNPs and raw copy number value per probe. (note: data should not be in log2)\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"GLAD\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.filename\", \"value\": \"<copy.number.input.file_basename>.glad\", \"label\": \"Output.Filename\", \"description\": \"Name of the output file\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"GlobalAlignment\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"standard global sequence analysis\", \"required_args\": [{\"name\": \"subject.fasta.filename\", \"value\": \"\", \"label\": \"Subject.Fasta.Filename\", \"description\": \"filename of subject FASTA file\"}, {\"name\": \"match\", \"value\": \"\", \"label\": \"Match\", \"description\": \"> 0\"}, {\"name\": \"mismatch\", \"value\": \"\", \"label\": \"Mismatch\", \"description\": \"< 0\"}, {\"name\": \"gapopen\", \"value\": \"\", \"label\": \"Gapopen\", \"description\": \"< 0\"}, {\"name\": \"gapextend\", \"value\": \"\", \"label\": \"Gapextend\", \"description\": \"< 0\"}, {\"name\": \"query.filename\", \"value\": \"\", \"label\": \"Query.Filename\", \"description\": \"FASTA file\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"GlobalAlignment\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.file\", \"value\": \"<subject.fasta.filename_basename>\", \"label\": \"Output.File\", \"description\": \"filename of output\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"Golub.Slonim.1999.Nature.all.aml.pipeline\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"ALL/AML methodology, from Golub and Slonim et al., 1999\", \"required_args\": [], \"default_args\": [{\"name\": \"task_name\", \"value\": \"Golub.Slonim.1999.Nature.all.aml.pipeline\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"GSEA\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Gene Set Enrichment Analysis\", \"required_args\": [{\"name\": \"expression.dataset\", \"value\": \"\", \"label\": \"Expression.Dataset\", \"description\": \"Dataset file - .res, .gct\"}, {\"name\": \"gene.sets.database\", \"value\": \"\", \"label\": \"Gene.Sets.Database\", \"description\": \"Gene sets database from GSEA website.\"}, {\"name\": \"gene.sets.database.file\", \"value\": \"\", \"label\": \"Gene.Sets.Database.File\", \"description\": \"Gene sets database - .gmt, .gmx, .grp. Upload a gene set if your gene set is not listed as a choice for the gene sets database parameter.\"}, {\"name\": \"phenotype.labels\", \"value\": \"\", \"label\": \"Phenotype.Labels\", \"description\": \"Cls file - .cls, must be binary\"}, {\"name\": \"target.profile\", \"value\": \"\", \"label\": \"Target.Profile\", \"description\": \"Name of the target phenotype profile.  Only applicable if class file defines continuous labels (one or more phenotype profiles).  Leave blank if class file defines a discrete phenotype.\"}, {\"name\": \"chip.platform.file\", \"value\": \"\", \"label\": \"Chip.Platform.File\", \"description\": \"DNA Chip (array) annotation file from GSEA website.  Upload your own chip file if the one corresponding to your DNA Microarray platform is not listed in the drop-down menu.  A chip file is only required if collapse dataset is set to true.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"GSEA\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"number.of.permutations\", \"value\": \"1000\", \"label\": \"Number.Of.Permutations\", \"description\": \"Number of permutations to perform\"}, {\"name\": \"collapse.dataset\", \"value\": \"true\", \"label\": \"Collapse.Dataset\", \"description\": \"Select true to have GSEA collapse each probe set in the expression dataset into a single vector for the gene, which gets identified by its gene symbol.\"}, {\"name\": \"permutation.type\", \"value\": \"phenotype\", \"label\": \"Permutation.Type\", \"description\": \"Type of permutations to perform\"}, {\"name\": \"scoring.scheme\", \"value\": \"weighted\", \"label\": \"Scoring.Scheme\", \"description\": \"The statistic used to score hits (gene set members) and misses (non-members)\"}, {\"name\": \"metric.for.ranking.genes\", \"value\": \"Signal2Noise\", \"label\": \"Metric.For.Ranking.Genes\", \"description\": \"Class separation metric - gene markers are ranked using this metric to produce the gene list\"}, {\"name\": \"gene.list.sorting.mode\", \"value\": \"real\", \"label\": \"Gene.List.Sorting.Mode\", \"description\": \"Mode in which scores from the gene list should be considered\"}, {\"name\": \"gene.list.ordering.mode\", \"value\": \"descending\", \"label\": \"Gene.List.Ordering.Mode\", \"description\": \"Direction in which the gene list should be ordered\"}, {\"name\": \"max.gene.set.size\", \"value\": \"500\", \"label\": \"Max.Gene.Set.Size\", \"description\": \"Gene sets larger than this are excluded from the analysis\"}, {\"name\": \"min.gene.set.size\", \"value\": \"15\", \"label\": \"Min.Gene.Set.Size\", \"description\": \"Gene sets smaller than this are excluded from the analysis\"}, {\"name\": \"collapsing.mode.for.probe.sets.with.more.than.one.match\", \"value\": \"Max_probe\", \"label\": \"Collapsing.Mode.For.Probe.Sets.With.More.Than.One.Match\", \"description\": \"Collapsing mode\"}, {\"name\": \"normalization.mode\", \"value\": \"meandiv\", \"label\": \"Normalization.Mode\", \"description\": \"Normalization mode\"}, {\"name\": \"randomization.mode\", \"value\": \"no_balance\", \"label\": \"Randomization.Mode\", \"description\": \"Type of phenotype randomization (does not apply to gene set permutations)\"}, {\"name\": \"omit.features.with.no.symbol.match\", \"value\": \"true\", \"label\": \"Omit.Features.With.No.Symbol.Match\", \"description\": \"If there is no known gene symbol match for a probe set, omit it from the collapsed dataset\"}, {\"name\": \"make.detailed.gene.set.report\", \"value\": \"true\", \"label\": \"Make.Detailed.Gene.Set.Report\", \"description\": \"Create detailed gene set report (heat map, mountain plot, etc.) for every enriched gene set\"}, {\"name\": \"median.for.class.metrics\", \"value\": \"false\", \"label\": \"Median.For.Class.Metrics\", \"description\": \"Use the median of each class instead of the mean for the class separation metrics\"}, {\"name\": \"number.of.markers\", \"value\": \"100\", \"label\": \"Number.Of.Markers\", \"description\": \"Number of markers\"}, {\"name\": \"plot.graphs.for.the.top.sets.of.each.phenotype\", \"value\": \"20\", \"label\": \"Plot.Graphs.For.The.Top.Sets.Of.Each.Phenotype\", \"description\": \"Plot GSEA mountain and related plots for the top sets of each phenotype\"}, {\"name\": \"random.seed\", \"value\": \"timestamp\", \"label\": \"Random.Seed\", \"description\": \"Seed to use for randomization\"}, {\"name\": \"save.random.ranked.lists\", \"value\": \"false\", \"label\": \"Save.Random.Ranked.Lists\", \"description\": \"Whether to save random ranked lists\"}, {\"name\": \"output.file.name\", \"value\": \"<expression.dataset_basename>.zip\", \"label\": \"Output.File.Name\", \"description\": \"Name of the output file\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"GSEALeadingEdgeViewer\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Leading Edge Viewer for GSEA results\", \"required_args\": [{\"name\": \"zip.file\", \"value\": \"\", \"label\": \"Zip.File\", \"description\": \"zip output file from GSEA - .zip. \"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"GSEALeadingEdgeViewer\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"GSEAPreranked\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Runs the gene set enrichment analysis against a user-supplied ranked list of genes.\", \"required_args\": [{\"name\": \"ranked.list\", \"value\": \"\", \"label\": \"Ranked.List\", \"description\": \"Rank ordered gene list - .rnk\"}, {\"name\": \"gene.sets.database\", \"value\": \"\", \"label\": \"Gene.Sets.Database\", \"description\": \"Gene sets database from MSigDB website.\"}, {\"name\": \"gene.sets.database.file\", \"value\": \"\", \"label\": \"Gene.Sets.Database.File\", \"description\": \"Gene sets database - .gmt, .gmx, .grp. Upload a gene set if your gene set is not listed as a choice for the gene sets database parameter.\"}, {\"name\": \"chip.platform.file\", \"value\": \"\", \"label\": \"Chip.Platform.File\", \"description\": \"DNA Chip (array) annotation file from GSEA website.  Upload your own chip file if the one corresponding to your DNA Microarray platform is not listed in the drop-down menu.  A chip file is only required if collapse dataset is set to true.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"GSEAPreranked\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"number.of.permutations\", \"value\": \"1000\", \"label\": \"Number.Of.Permutations\", \"description\": \"Number of permutations to perform\"}, {\"name\": \"collapse.dataset\", \"value\": \"false\", \"label\": \"Collapse.Dataset\", \"description\": \"Select true to have GSEAPreranked collapse each feature set in the ranked list into a single vector for the gene, which gets identified by its gene symbol.\"}, {\"name\": \"scoring.scheme\", \"value\": \"weighted\", \"label\": \"Scoring.Scheme\", \"description\": \"The statistic used to score hits (gene set members) and misses (non-members)\"}, {\"name\": \"max.gene.set.size\", \"value\": \"500\", \"label\": \"Max.Gene.Set.Size\", \"description\": \"Gene sets larger than this are excluded from the analysis\"}, {\"name\": \"min.gene.set.size\", \"value\": \"15\", \"label\": \"Min.Gene.Set.Size\", \"description\": \"Gene sets smaller than this are excluded from the analysis\"}, {\"name\": \"normalization.mode\", \"value\": \"meandiv\", \"label\": \"Normalization.Mode\", \"description\": \"Normalization mode\"}, {\"name\": \"collapsing.mode\", \"value\": \"Max_probe\", \"label\": \"Collapsing.Mode\", \"description\": \"Collapsing mode\"}, {\"name\": \"omit.features.with.no.symbol.match\", \"value\": \"true\", \"label\": \"Omit.Features.With.No.Symbol.Match\", \"description\": \"If there is no known gene symbol match for a probe set, omit it from the collapsed dataset; only relevant when collapse dataset is set to true\"}, {\"name\": \"make.detailed.gene.set.report\", \"value\": \"true\", \"label\": \"Make.Detailed.Gene.Set.Report\", \"description\": \"Create detailed gene set report (heat map, mountain plot, etc.) for every enriched gene set\"}, {\"name\": \"num.top.sets\", \"value\": \"20\", \"label\": \"Num.Top.Sets\", \"description\": \"Plot GSEA mountain and related plots for the top sets of each phenotype\"}, {\"name\": \"random.seed\", \"value\": \"timestamp\", \"label\": \"Random.Seed\", \"description\": \"Seed to use for generating random permutations.\"}, {\"name\": \"output.file.name\", \"value\": \"<ranked.list_basename>.zip\", \"label\": \"Output.File.Name\", \"description\": \"Name of the output file\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"HAPSEG\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"[**Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] A probabilistic method to interpret bi-allelic marker data in cancer samples.\", \"required_args\": [{\"name\": \"plate.name\", \"value\": \"\", \"label\": \"Plate.Name\", \"description\": \"Name of the sample plate\"}, {\"name\": \"array.name\", \"value\": \"\", \"label\": \"Array.Name\", \"description\": \"Name of the chip that was run\"}, {\"name\": \"seg.file\", \"value\": \"\", \"label\": \"Seg.File\", \"description\": \"Segmented copy number data file for this sample. If not provided, HAPSEG will segment the data for you.\"}, {\"name\": \"snp.file\", \"value\": \"\", \"label\": \"Snp.File\", \"description\": \"SNP intensity file for this sample\"}, {\"name\": \"use.pop\", \"value\": \"\", \"label\": \"Use.Pop\", \"description\": \"Population to run with (currently supported values: CEPH, CH, JA, YOR)\"}, {\"name\": \"calls.file\", \"value\": \"\", \"label\": \"Calls.File\", \"description\": \"If using a matched normal sample, a Birdseed calls file needs to be supplied\"}, {\"name\": \"mn.sample\", \"value\": \"\", \"label\": \"Mn.Sample\", \"description\": \"If using a matched sample, the name of that matched normal sample\"}, {\"name\": \"clusters.file\", \"value\": \"\", \"label\": \"Clusters.File\", \"description\": \"If calibrate.data is 'On' the user must supply a Birdseed clusters file\"}, {\"name\": \"prev.theta.file\", \"value\": \"\", \"label\": \"Prev.Theta.File\", \"description\": \"An optional file storing the previous theta values\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"HAPSEG\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"out.file.name\", \"value\": \"<plate.name>_<array.name>.segdat.RData\", \"label\": \"Out.File.Name\", \"description\": \"The name of the output file.\"}, {\"name\": \"genome.build\", \"value\": \"ftp://gpftp.broadinstitute.org/module_support_files/HAPSEG_1.1.1/BEAGLE/phasedBGL/hg19/\", \"label\": \"Genome.Build\", \"description\": \"Which build of the human genome to use.  Only 'hg18' and 'hg19' are currently supported.  This will select a set of phased BEAGLE files.\"}, {\"name\": \"platform\", \"value\": \"SNP_6.0\", \"label\": \"Platform\", \"description\": \"The chip type used. The supported values are currently 'SNP_250K_STY' and 'SNP_6.0'\"}, {\"name\": \"impute.gt\", \"value\": \"TRUE\", \"label\": \"Impute.Gt\", \"description\": \"If TRUE will impute genotypes via BEAGLE. We recommend this be TRUE unless there's a reason to do otherwise\"}, {\"name\": \"plot.segfit\", \"value\": \"TRUE\", \"label\": \"Plot.Segfit\", \"description\": \"If TRUE will plot JPG images of the segmentations fits\"}, {\"name\": \"merge.small\", \"value\": \"TRUE\", \"label\": \"Merge.Small\", \"description\": \"If TRUE will merge small segments\"}, {\"name\": \"merge.close\", \"value\": \"TRUE\", \"label\": \"Merge.Close\", \"description\": \"if TRUE will merge close segments\"}, {\"name\": \"min.seg.size\", \"value\": \"10\", \"label\": \"Min.Seg.Size\", \"description\": \"Minimum segment size\"}, {\"name\": \"normal\", \"value\": \"FALSE\", \"label\": \"Normal\", \"description\": \"If TRUE this sample will be treated as a normal\"}, {\"name\": \"out.p\", \"value\": \"0.05\", \"label\": \"Out.P\", \"description\": \"Outlier probability\"}, {\"name\": \"seg.merge.thresh\", \"value\": \"1e-10\", \"label\": \"Seg.Merge.Thresh\", \"description\": \"Segmentation threshold\"}, {\"name\": \"use.normal\", \"value\": \"FALSE\", \"label\": \"Use.Normal\", \"description\": \"If TRUE will use a matched normal sample if one is provided\"}, {\"name\": \"drop.x\", \"value\": \"FALSE\", \"label\": \"Drop.X\", \"description\": \"If TRUE will remove the X chromosome from the calculation\"}, {\"name\": \"drop.y\", \"value\": \"TRUE\", \"label\": \"Drop.Y\", \"description\": \"If TRUE will remove the Y chromosome from the calculation\"}, {\"name\": \"calibrate.data\", \"value\": \"NA\", \"label\": \"Calibrate.Data\", \"description\": \"If 'On', will perform a calibration on the input data.  If 'Off', no calibration will be performed. If left at the default value (Inferred), the calibration status will be inferred.\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"HeatMapImage\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Creates a heat map from a dataset\", \"required_args\": [{\"name\": \"input.dataset\", \"value\": \"\", \"label\": \"Input.Dataset\", \"description\": \"The input dataset - .res, .gct, .odf\"}, {\"name\": \"rows.to.highlight\", \"value\": \"\", \"label\": \"Rows.To.Highlight\", \"description\": \"Rows to highlight in the heat map\"}, {\"name\": \"row.highlight.color\", \"value\": \"\", \"label\": \"Row.Highlight.Color\", \"description\": \"The color to highlight the rows if a rows to highlight file is provided\"}, {\"name\": \"color.palette\", \"value\": \"\", \"label\": \"Color.Palette\", \"description\": \"Color palette consisting of one r:g:b triplet or one hex value per line\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"HeatMapImage\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output\", \"value\": \"<input.dataset_basename>\", \"label\": \"Output\", \"description\": \"The output file name -.jpeg, .png, .tiff, .bmp, .eps\"}, {\"name\": \"output.format\", \"value\": \"jpeg\", \"label\": \"Output.Format\", \"description\": \"The output file format\"}, {\"name\": \"column.size\", \"value\": \"8\", \"label\": \"Column.Size\", \"description\": \"The size in pixels of an element along the horizontal axis\"}, {\"name\": \"row.size\", \"value\": \"8\", \"label\": \"Row.Size\", \"description\": \"The size in pixels of an element along the vertical axis\"}, {\"name\": \"show.grid\", \"value\": \"yes\", \"label\": \"Show.Grid\", \"description\": \"Whether to show a grid between elements\"}, {\"name\": \"grid.color\", \"value\": \"black\", \"label\": \"Grid.Color\", \"description\": \"The grid color\"}, {\"name\": \"show.row.descriptions\", \"value\": \"no\", \"label\": \"Show.Row.Descriptions\", \"description\": \"Whether to show row descriptions\"}, {\"name\": \"show.row.names\", \"value\": \"yes\", \"label\": \"Show.Row.Names\", \"description\": \"Whether to show row names\"}, {\"name\": \"color.scheme\", \"value\": \"row normalized\", \"label\": \"Color.Scheme\", \"description\": \"The color scheme\"}, {\"name\": \"use.color.gradient\", \"value\": \"no\", \"label\": \"Use.Color.Gradient\", \"description\": \"Whether to use a linear color gradient or discrete colors\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"HeatMapViewer\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"A configurable heat map viewer that provides users with several options for manipulating and visualizing array-based data\", \"required_args\": [{\"name\": \"dataset\", \"value\": \"\", \"label\": \"Dataset\", \"description\": \"The dataset to display - .gct\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"HeatMapViewer\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"helloWorld\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Generic hello world module\", \"required_args\": [], \"default_args\": [{\"name\": \"task_name\", \"value\": \"helloWorld\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"message\", \"value\": \"Hello, World!\", \"label\": \"Message\", \"description\": \"\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"helloWorld_R2.15_Rscript\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Demo R2.15_Rscript module\", \"required_args\": [], \"default_args\": [{\"name\": \"task_name\", \"value\": \"helloWorld_R2.15_Rscript\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"HierarchicalClustering\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Hierarchical Clustering\", \"required_args\": [{\"name\": \"input.filename\", \"value\": \"\", \"label\": \"Input.Filename\", \"description\": \"input data file name - .gct, .res, .pcl\"}, {\"name\": \"log.transform\", \"value\": \"\", \"label\": \"Log.Transform\", \"description\": \"log-transform the data before clustering\"}, {\"name\": \"row.center\", \"value\": \"\", \"label\": \"Row.Center\", \"description\": \"whether to center each row (gene) in the data\"}, {\"name\": \"row.normalize\", \"value\": \"\", \"label\": \"Row.Normalize\", \"description\": \"whether to normalize each row (gene) in the data\"}, {\"name\": \"column.center\", \"value\": \"\", \"label\": \"Column.Center\", \"description\": \"whether to center each column (sample) in the data\"}, {\"name\": \"column.normalize\", \"value\": \"\", \"label\": \"Column.Normalize\", \"description\": \"whether to normalize each column (sample) in the data\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"HierarchicalClustering\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"column.distance.measure\", \"value\": \"2\", \"label\": \"Column.Distance.Measure\", \"description\": \"distance measure for column (sample) clustering\"}, {\"name\": \"row.distance.measure\", \"value\": \"0\", \"label\": \"Row.Distance.Measure\", \"description\": \"distance measure for row (gene) clustering  NOTE: Filtering beforehand is recommended since row clustering is computationally intensive.\"}, {\"name\": \"clustering.method\", \"value\": \"a\", \"label\": \"Clustering.Method\", \"description\": \"hierarchical clustering method to use\"}, {\"name\": \"output.base.name\", \"value\": \"<input.filename_basename>\", \"label\": \"Output.Base.Name\", \"description\": \"base name for output files\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"HierarchicalClusteringImage\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Creates an image of the dendrogram generated from HierarchicalClustering\", \"required_args\": [{\"name\": \"cdt\", \"value\": \"\", \"label\": \"Cdt\", \"description\": \"Clustered data table, contains the original data, but reordered\"}, {\"name\": \"gtr\", \"value\": \"\", \"label\": \"Gtr\", \"description\": \"Gene Tree file, records the order in which genes (rows) were joined (at least one of gtr and atr files are required)\"}, {\"name\": \"atr\", \"value\": \"\", \"label\": \"Atr\", \"description\": \"Array Tree file, records the order in which samples (columns) were joined (at least one of gtr and atr files are required)\"}, {\"name\": \"rows.to.highlight\", \"value\": \"\", \"label\": \"Rows.To.Highlight\", \"description\": \"Rows to highlight in the heat map\"}, {\"name\": \"row.highlight.color\", \"value\": \"\", \"label\": \"Row.Highlight.Color\", \"description\": \"The color to highlight the rows if a rows to highlight file is provided\"}, {\"name\": \"color.palette\", \"value\": \"\", \"label\": \"Color.Palette\", \"description\": \"Color palette consisting of one r:g:b triplet or one hex value per line\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"HierarchicalClusteringImage\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output\", \"value\": \"<cdt_basename>\", \"label\": \"Output\", \"description\": \"The output file name - .bmp, .jpeg, .png, .tiff\"}, {\"name\": \"output.format\", \"value\": \"jpeg\", \"label\": \"Output.Format\", \"description\": \"The output file format\"}, {\"name\": \"column.size\", \"value\": \"8\", \"label\": \"Column.Size\", \"description\": \"The size in pixels of an element along the horizontal axis\"}, {\"name\": \"row.size\", \"value\": \"8\", \"label\": \"Row.Size\", \"description\": \"The size in pixels of an element along the vertical axis\"}, {\"name\": \"show.grid\", \"value\": \"yes\", \"label\": \"Show.Grid\", \"description\": \"Whether to show a grid between elements\"}, {\"name\": \"grid.color\", \"value\": \"0:0:0\", \"label\": \"Grid.Color\", \"description\": \"The grid color\"}, {\"name\": \"show.row.descriptions\", \"value\": \"no\", \"label\": \"Show.Row.Descriptions\", \"description\": \"Whether to show row descriptions\"}, {\"name\": \"show.row.names\", \"value\": \"yes\", \"label\": \"Show.Row.Names\", \"description\": \"Whether to show row names\"}, {\"name\": \"color.scheme\", \"value\": \"row normalized\", \"label\": \"Color.Scheme\", \"description\": \"The color scheme\"}, {\"name\": \"use.color.gradient\", \"value\": \"no\", \"label\": \"Use.Color.Gradient\", \"description\": \"Whether to use a linear color gradient or discrete colors\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"HierarchicalClusteringViewer\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Views results of hierarchical clustering.\", \"required_args\": [{\"name\": \"cdt.file\", \"value\": \"\", \"label\": \"Cdt.File\", \"description\": \"Clustered data table, contains the original data, but reordered\"}, {\"name\": \"gtr.file\", \"value\": \"\", \"label\": \"Gtr.File\", \"description\": \"Gene tree file, records the order in which genes (rows) were joined (at least one of gtr file and atr file is required)\"}, {\"name\": \"atr.file\", \"value\": \"\", \"label\": \"Atr.File\", \"description\": \"Array tree file, records the order in which samples (columns) were joined (at least one of gtr file and atr file is required)\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"HierarchicalClusteringViewer\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"Hu68kHu35kAtoU95\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Converts a list of Affymetrix Hu6800/Hu35KsubA probes to the corresponding Affymetrix U95Av2 probes.\", \"required_args\": [{\"name\": \"input.filename\", \"value\": \"\", \"label\": \"Input.Filename\", \"description\": \"the .pol, .tag or .grp file to be converted\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"Hu68kHu35kAtoU95\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output\", \"value\": \"<input.filename_basename>.U95\", \"label\": \"Output\", \"description\": \"name of the output file (same format as input file)\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"Identify.an.up.or.downregulated.pathway.from.expression.data\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"This pipeline is used in the GenomeSpace recipe, \\\"Identify an up- or down-regulated pathway from expression data\\\". The recipe can be found at: http://www.genomespace.org/support/guides/recipes/sections/identify-an-up-or-down-regulated-pathway-from-expression-data.\", \"required_args\": [{\"name\": \"ConvertLineEndings1.input.filename\", \"value\": \"\", \"label\": \"ConvertLineEndings1.Input.Filename\", \"description\": \"The input file (any non-binary file format)\"}, {\"name\": \"ComparativeMarkerSelection2.cls.file\", \"value\": \"\", \"label\": \"ComparativeMarkerSelection2.Cls.File\", \"description\": \"The class file - .cls\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"Identify.an.up.or.downregulated.pathway.from.expression.data\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"Identify.and.Validate.a.Consensus.Signature.Using.Gene.Expression.Data\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"This pipeline is used in the GenomeSpace recipe, \\\"Identify and Validate a Consensus Signature Using Gene Expression Data\\\". The recipe can be found at: http://genomespace.org/support/guides/recipes/sections/identify-and-validate-a-consensus-signature-using-gene-expression-data.\", \"required_args\": [{\"name\": \"SelectFeaturesColumns1.input.filename\", \"value\": \"\", \"label\": \"SelectFeaturesColumns1.Input.Filename\", \"description\": \"input file - .res, .gct, .snp, .cn, .loh, .odf, or .cls file from which the slice is extracted\"}, {\"name\": \"SelectFeaturesColumns2.input.filename\", \"value\": \"\", \"label\": \"SelectFeaturesColumns2.Input.Filename\", \"description\": \"input file - .res, .gct, .snp, .cn, .loh, .odf, or .cls file from which the slice is extracted\"}, {\"name\": \"SelectFeaturesRows3.list.filename\", \"value\": \"\", \"label\": \"SelectFeaturesRows3.List.Filename\", \"description\": \"text file containing one identifier per line that specifies the rows to select. \"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"Identify.and.Validate.a.Consensus.Signature.Using.Gene.Expression.Data\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"Identify.and.visualize.expressed.transcripts.in.RNASeq.data\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"This pipeline is used in the GenomeSpace recipe, \\\"Identify and visualize expressed transcripts in RNA-Seq data\\\". The recipe can be found at: http://www.genomespace.org/support/guides/recipes/sections/identify-and-visualize-expressed-transcripts-in-rna-seq-data.\", \"required_args\": [{\"name\": \"ConvertLineEndings1.input.filename\", \"value\": \"\", \"label\": \"ConvertLineEndings1.Input.Filename\", \"description\": \"The input file (any non-binary file format)\"}, {\"name\": \"TopHat2.reads.pair.1\", \"value\": \"\", \"label\": \"TopHat2.Reads.Pair.1\", \"description\": \"Unpaired reads file or first mate for paired reads. One or more files containing reads in FASTA or FASTQ format (bz2 and gz compressed files are supported).\"}, {\"name\": \"TopHat2.reads.pair.2\", \"value\": \"\", \"label\": \"TopHat2.Reads.Pair.2\", \"description\": \"Second mate for paired reads. Zero or more files in FASTA or FASTQ format (bz2 and gz compressed files are supported).\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"Identify.and.visualize.expressed.transcripts.in.RNASeq.data\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"IGV\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Launches IGV with input file(s)\", \"required_args\": [{\"name\": \"input.file\", \"value\": \"\", \"label\": \"Input.File\", \"description\": \"Dataset to be viewed in IGV or IGV session file\"}, {\"name\": \"index.file\", \"value\": \"\", \"label\": \"Index.File\", \"description\": \"The index file.\"}, {\"name\": \"genomeId\", \"value\": \"\", \"label\": \"GenomeId\", \"description\": \"Choose the genome for your data. IGV will launch with hg19 or the last viewed genome by default.\"}, {\"name\": \"locus\", \"value\": \"\", \"label\": \"Locus\", \"description\": \"Provide a locus or gene of interest for your data. IGV will launch on chr1 or the last viewed chromosome by default. (e.g. chr1:2000-3000 or egfr)\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"IGV\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"IGV.js\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Launches IGV.js with input file(s)\", \"required_args\": [{\"name\": \"input.file\", \"value\": \"\", \"label\": \"Input.File\", \"description\": \"Track dataset to be viewed in IGV\"}, {\"name\": \"index.file\", \"value\": \"\", \"label\": \"Index.File\", \"description\": \"The index file (i.e .bai, .idx, or .tbi).\"}, {\"name\": \"genome\", \"value\": \"\", \"label\": \"Genome\", \"description\": \"Choose the genome for your data. IGV will launch with hg19.\"}, {\"name\": \"locus\", \"value\": \"\", \"label\": \"Locus\", \"description\": \"Provide a locus or gene of interest for your data. IGV will launch on chr1.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"IGV.js\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"IlluminaConcatenator\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Concatenate normalized Illumina probe sets into a single GCT file\", \"required_args\": [{\"name\": \"input.file\", \"value\": \"\", \"label\": \"Input.File\", \"description\": \"ZIP file of normalized GCT files output by IlluminaNormalizer\"}, {\"name\": \"output.file.type\", \"value\": \"\", \"label\": \"Output.File.Type\", \"description\": \"Whether to concatenate the data or detection scores\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"IlluminaConcatenator\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"IlluminaDASLPipeline\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"creates a GenePattern gct file from raw Illumina scan data\", \"required_args\": [{\"name\": \"IlluminaScanExtractor1.sample.description.file\", \"value\": \"\", \"label\": \"IlluminaScanExtractor1.Sample.Description.File\", \"description\": \"file with one sample per line, one Sentrix Position, DAP#, and Sentrix ID per DAP\"}, {\"name\": \"IlluminaConcatenator3.output.file.type\", \"value\": \"\", \"label\": \"IlluminaConcatenator3.Output.File.Type\", \"description\": \"Whether to concatenate the data or detection scores\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"IlluminaDASLPipeline\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"IlluminaScanExtractor1.outlier.method\", \"value\": \"> 3 Std. Deviations\", \"label\": \"IlluminaScanExtractor1.Outlier.Method\", \"description\": \"outlier detection method (only applies to extraction using bead-level data)\"}, {\"name\": \"IlluminaNormalizer2.normalization.method\", \"value\": \"CubicSpline\", \"label\": \"IlluminaNormalizer2.Normalization.Method\", \"description\": \"correction method. Note that mean background subtraction does not apply when the input is a single GCT file.\"}, {\"name\": \"IlluminaNormalizer2.discard.controls\", \"value\": \"no\", \"label\": \"IlluminaNormalizer2.Discard.Controls\", \"description\": \"discard top 2 and bottom 2 negative controls (applies to mean background subtraction only)\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"IlluminaExpressionFileCreator\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Creates a GCT file from a zip of Illumina IDAT files and an Illumina manifest file\", \"required_args\": [{\"name\": \"idat.zip\", \"value\": \"\", \"label\": \"Idat.Zip\", \"description\": \"Zip of Illumina IDAT files\"}, {\"name\": \"manifest\", \"value\": \"\", \"label\": \"Manifest\", \"description\": \"Illumina manifest file in tab-delimited (TXT) format. A manifest file is required; either selected from this list or specified in custom manifest below.\"}, {\"name\": \"custom.manifest\", \"value\": \"\", \"label\": \"Custom.Manifest\", \"description\": \"Other Illumina manifest file in tab-delimited (TXT) format. A manifest file is required; either selected from the list above or specified here.\"}, {\"name\": \"chip\", \"value\": \"\", \"label\": \"Chip\", \"description\": \"CHIP file to map probes to genes.\"}, {\"name\": \"custom.chip\", \"value\": \"\", \"label\": \"Custom.Chip\", \"description\": \"Other CHIP file to map probes to genes.\"}, {\"name\": \"clm\", \"value\": \"\", \"label\": \"Clm\", \"description\": \"CLM file to map file names to sample names.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"IlluminaExpressionFileCreator\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.file\", \"value\": \"<idat.zip_basename>.gct\", \"label\": \"Output.File\", \"description\": \"Output file name\"}, {\"name\": \"background.subtraction.mode\", \"value\": \"false\", \"label\": \"Background.Subtraction.Mode\", \"description\": \"Perform background subtraction.\"}, {\"name\": \"collapse.mode\", \"value\": \"none\", \"label\": \"Collapse.Mode\", \"description\": \"Collapse probes to genes based on the manifest or CHIP file (if provided).\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"IlluminaNormalizer\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Normalize zipped Illumina scans\", \"required_args\": [{\"name\": \"zip.or.gct.file\", \"value\": \"\", \"label\": \"Zip.Or.Gct.File\", \"description\": \"a zip of GCT files output by IlluminaScanExtractor (6k DASL) or a GCT file (whole-genome DASL)\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"IlluminaNormalizer\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"normalization.method\", \"value\": \"CubicSpline\", \"label\": \"Normalization.Method\", \"description\": \"correction method. Note that mean background subtraction does not apply when the input is a single GCT file.\"}, {\"name\": \"output.file\", \"value\": \"<zip.or.gct.file_basename>_<normalization.method>\", \"label\": \"Output.File\", \"description\": \"output file name\"}, {\"name\": \"discard.controls\", \"value\": \"no\", \"label\": \"Discard.Controls\", \"description\": \"discard top 2 and bottom 2 negative controls (applies to mean background subtraction only)\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"IlluminaScanExtractor\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Extract intensity values from Illumina scans\", \"required_args\": [{\"name\": \"sample.description.file\", \"value\": \"\", \"label\": \"Sample.Description.File\", \"description\": \"file with one sample per line, one Sentrix Position, DAP#, and Sentrix ID per DAP\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"IlluminaScanExtractor\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"outlier.method\", \"value\": \"> 3 Std. Deviations\", \"label\": \"Outlier.Method\", \"description\": \"outlier detection method (only applies to extraction using bead-level data)\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"ImmPort_FLOCK_CrossSample\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"ImmPort FLOCK and CrossSample pipeline\", \"required_args\": [{\"name\": \"ImmPortFCSConvLogicleTrans1.Input.FCS.file\", \"value\": \"\", \"label\": \"ImmPortFCSConvLogicleTrans1.Input.FCS.File\", \"description\": \"An FCS file that shall be transformed with the Logicle transformation, converted and saved as tab-delimited text file.\"}, {\"name\": \"ImmPortColSelection2.Dimension.specification\", \"value\": \"\", \"label\": \"ImmPortColSelection2.Dimension.Specification\", \"description\": \"A list of dimensions/columns to extract, specified by their indexes (starting from 1), e.g., 1,4,6. There should not be any white space characters (e.g, spaces, tabs, new lines, etc.) in the parameter value.\"}, {\"name\": \"ImmPortCrossSample4.Input.file\", \"value\": \"\", \"label\": \"ImmPortCrossSample4.Input.File\", \"description\": \"A tab-delimited text file with columns corresponding to parameters and rows to events. Such a text file may be generated from an FCS file by the ImmPortFCSConvLogicleTrans module.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"ImmPort_FLOCK_CrossSample\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"ImmPort_FLOCK_Individual_FCS\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"ImmPort FLOCK and Individual FCS pipeline\", \"required_args\": [{\"name\": \"ImmPortFCSConvLogicleTrans1.Input.FCS.file\", \"value\": \"\", \"label\": \"ImmPortFCSConvLogicleTrans1.Input.FCS.File\", \"description\": \"An FCS file that shall be transformed with the Logicle transformation, converted and saved as tab-delimited text file.\"}, {\"name\": \"ImmPortColSelection2.Dimension.specification\", \"value\": \"\", \"label\": \"ImmPortColSelection2.Dimension.Specification\", \"description\": \"A list of dimensions/columns to extract, specified by their indexes (starting from 1), e.g., 1,4,6. There should not be any white space characters (e.g, spaces, tabs, new lines, etc.) in the parameter value.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"ImmPort_FLOCK_Individual_FCS\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"ImmPortColSelection\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Select a set of columns from a txt file\", \"required_args\": [{\"name\": \"Input.file\", \"value\": \"\", \"label\": \"Input.File\", \"description\": \"A tab-delimited text file with columns corresponding to parameters and rows to events, converted from an fcs file, typically by ImmPortFCSConvLogicleTrans.\"}, {\"name\": \"Dimension.specification\", \"value\": \"\", \"label\": \"Dimension.Specification\", \"description\": \"A list of dimensions/columns to extract, specified by their indexes (starting from 1), e.g., 1,4,6. There should not be any white space characters (e.g, spaces, tabs, new lines, etc.) in the parameter value.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"ImmPortColSelection\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"ImmPortCrossSample\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Applying an individual FLOCK result across another FCM sample (.txt) version 1 (1-round Kmeans)\", \"required_args\": [{\"name\": \"Input.file\", \"value\": \"\", \"label\": \"Input.File\", \"description\": \"A tab-delimited text file with columns corresponding to parameters and rows to events. Such a text file may be generated from an FCS file by the ImmPortFCSConvLogicleTrans module.\"}, {\"name\": \"Centroid.file\", \"value\": \"\", \"label\": \"Centroid.File\", \"description\": \"A text-file specifying the position of each of the cell populations. Such a text file is supposed to be generated by the ImmPortFLOCK module (file population_center.txt).\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"ImmPortCrossSample\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"Number.of.iterations\", \"value\": \"1\", \"label\": \"Number.Of.Iterations\", \"description\": \"The number of iterations that the algorithm will use to match events with the closest population; values from the [1, 10] range are allowed.\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"ImmPortCSV2TXT\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Convert a .csv file to a .txt file that FLOCK can read\", \"required_args\": [{\"name\": \"Input.CSV.data.file\", \"value\": \"\", \"label\": \"Input.CSV.Data.File\", \"description\": \"The input CSV file that will be converted to a a tab-delimited text file; the CSV shall contain a header with parameter names.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"ImmPortCSV2TXT\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"ImmPortFCSConvLogicleTrans\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Logicle transformation of an fcs file to .txt file\", \"required_args\": [{\"name\": \"Input.FCS.file\", \"value\": \"\", \"label\": \"Input.FCS.File\", \"description\": \"An FCS file that shall be transformed with the Logicle transformation, converted and saved as tab-delimited text file.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"ImmPortFCSConvLogicleTrans\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"ImmPortFLOCK\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Automated Gating with ImmPort FLOCK algorithm version 1.0\", \"required_args\": [{\"name\": \"Input.file\", \"value\": \"\", \"label\": \"Input.File\", \"description\": \"A tab-delimited text file with columns corresponding to parameters and rows to events. The first row is a heading labelling parameters. Such a text file is supposed to be generated from an FCS file by the ImmPortFCSConvLogicleTrans module optionally followed by the ImmPortColSelection module (if applicable to remove unused parameters). Alternatively, this text file may be created by the ImmPortCSV2TXT module if the original data is in the CSV format.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"ImmPortFLOCK\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"Number.of.bins\", \"value\": \"0\", \"label\": \"Number.Of.Bins\", \"description\": \"The number of bins used to partition each dimension. The default value of 0 means that the algorithm will automatically determine a suitable number of bins. Any value from the [6,29] range may be provided to manually override algorithmic estimations.\"}, {\"name\": \"Density.threshold\", \"value\": \"0\", \"label\": \"Density.Threshold\", \"description\": \"The minimal number of events in a bin (in the hyperspace of used markers) needed to mark the bin as a dense bin. The default value of 0 means that the algorithm will automatically determine a suitable density threshold. Any value from the [3,99] range may be provided to manually override algorithmic estimations.\"}, {\"name\": \"Maximum.number.of.populations\", \"value\": \"30\", \"label\": \"Maximum.Number.Of.Populations\", \"description\": \"The maximum number of populations identifiable by the clustering algorithm. The default value of 30 works well in most cases but may be overridden by any value from the [5,999] range. Larger values may increase the execution time of the module.\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"ImputeMissingValues.KNN\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Imputes missing values\", \"required_args\": [{\"name\": \"data.filename\", \"value\": \"\", \"label\": \"Data.Filename\", \"description\": \"Data file (missing values are stored as NA) - .gct\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"ImputeMissingValues.KNN\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"k\", \"value\": \"10\", \"label\": \"K\", \"description\": \"Number of neighbors to be used in the imputation\"}, {\"name\": \"rowmax\", \"value\": \"0.5\", \"label\": \"Rowmax\", \"description\": \"The maximum percent missing data allowed in any row\"}, {\"name\": \"colmax\", \"value\": \"0.8\", \"label\": \"Colmax\", \"description\": \"The maximum percent missing data allowed in any column\"}, {\"name\": \"output.file\", \"value\": \"<data.filename_basename>.imputed.gct\", \"label\": \"Output.File\", \"description\": \"The name of the output file - .gct\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"Install.Proteogenomics.R.Packages\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Installs all packages and dependencies need to run the Protegenomics GenePattern modules\", \"required_args\": [], \"default_args\": [{\"name\": \"task_name\", \"value\": \"Install.Proteogenomics.R.Packages\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"IntersectingIdentifiers\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Given 2 tab delimited files (res, gct, tab, odf) caculate which ids in the first intersect with the second. Outputs a file of intersecting, non-intersecting, in the first but not the second and in the second but not the first\", \"required_args\": [{\"name\": \"First.file\", \"value\": \"\", \"label\": \"First.File\", \"description\": \"Any tab delimited file\"}, {\"name\": \"Second.file\", \"value\": \"\", \"label\": \"Second.File\", \"description\": \"Any tab delimited file\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"IntersectingIdentifiers\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"First.file.identifier.column.index\", \"value\": \"1\", \"label\": \"First.File.Identifier.Column.Index\", \"description\": \"Index of identifiers in first file, staring at 1\"}, {\"name\": \"Second.file.identifier.column.index\", \"value\": \"1\", \"label\": \"Second.File.Identifier.Column.Index\", \"description\": \"Index of identifiers in second file, staring at 1\"}, {\"name\": \"Output.file.name.prefix\", \"value\": \"<First.file_basename>\", \"label\": \"Output.File.Name.Prefix\", \"description\": \"Name of output file\"}, {\"name\": \"First.file.skip.rows\", \"value\": \"2\", \"label\": \"First.File.Skip.Rows\", \"description\": \"Number of rows to skip in the first file to avoid headers\"}, {\"name\": \"Second.file.skip.rows\", \"value\": \"2\", \"label\": \"Second.File.Skip.Rows\", \"description\": \"Number of rows to skip in the first file to avoid headers\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"JavaTreeView\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Hierarchical clustering viewer that reads in Eisen's cdt, atr, and gtr files. A detailed description of these formats is available from http://microarray.ccgb.umn.edu/smd/html/MicroArray/help/formats.shtml\", \"required_args\": [{\"name\": \"cdt.filename\", \"value\": \"\", \"label\": \"Cdt.Filename\", \"description\": \"Clustered data table, contains the original data, but reordered\"}, {\"name\": \"gtr.filename\", \"value\": \"\", \"label\": \"Gtr.Filename\", \"description\": \"Gene Tree file, records the order in which genes (rows) were joined (at least one of gtr filename and atr filename is required)\"}, {\"name\": \"atr.filename\", \"value\": \"\", \"label\": \"Atr.Filename\", \"description\": \"Array Tree file, records the order in which samples (columns) were joined (at least one of gtr filename and atr filename is required)\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"JavaTreeView\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"KMeansClassifyFCS\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Find clusters in a flow cytometry file with k means clustering\", \"required_args\": [{\"name\": \"Input.FCS.data.file\", \"value\": \"\", \"label\": \"Input.FCS.Data.File\", \"description\": \"The FCS file\"}, {\"name\": \"K\", \"value\": \"\", \"label\": \"K\", \"description\": \"The number of populations you are looking for.\"}, {\"name\": \"Dimensions\", \"value\": \"\", \"label\": \"Dimensions\", \"description\": \"The channels to search for populations, separated by commas. Leave blank to search in all channels. E.g. FSC Lin, SS Lin, CD4 Log\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"KMeansClassifyFCS\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"CSV.file.name\", \"value\": \"<Input.FCS.data.file_basename>\", \"label\": \"CSV.File.Name\", \"description\": \"The base filename to split the results into csv files, each containing one population\"}, {\"name\": \"CLS.file.name\", \"value\": \"<Input.FCS.data.file_basename>\", \"label\": \"CLS.File.Name\", \"description\": \"A .CLS filename to store the population number for each data point\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"KMeansClustering\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Module that performs the K-Means Clustering algorithm\", \"required_args\": [{\"name\": \"input.filename\", \"value\": \"\", \"label\": \"Input.Filename\", \"description\": \"File containing data to cluster - .res, .gct, .odf\"}, {\"name\": \"number.of.clusters\", \"value\": \"\", \"label\": \"Number.Of.Clusters\", \"description\": \"Number of centroids (clusters)\"}, {\"name\": \"cluster.by\", \"value\": \"\", \"label\": \"Cluster.By\", \"description\": \"Whether to cluster by rows or columns\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"KMeansClustering\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.base.name\", \"value\": \"<input.filename_basename>_KMcluster_output\", \"label\": \"Output.Base.Name\", \"description\": \"The base output file name - .gct\"}, {\"name\": \"seed.value\", \"value\": \"12345\", \"label\": \"Seed.Value\", \"description\": \"Seed value for random number generator\"}, {\"name\": \"distance.metric\", \"value\": \"0\", \"label\": \"Distance.Metric\", \"description\": \"How to compute distance between points\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"KNN\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"K-nearest neighbors classification\", \"required_args\": [{\"name\": \"train.filename\", \"value\": \"\", \"label\": \"Train.Filename\", \"description\": \"training data file name - .gct, .res, .odf type = Dataset, ignored if a saved model (saved.model.filename) is used\"}, {\"name\": \"train.class.filename\", \"value\": \"\", \"label\": \"Train.Class.Filename\", \"description\": \"class file for training data - .cls, ignored if a saved model (saved.model.filename) is used\"}, {\"name\": \"saved.model.filename\", \"value\": \"\", \"label\": \"Saved.Model.Filename\", \"description\": \"input KNN model file - .odf type = KNN Prediction Model\"}, {\"name\": \"test.filename\", \"value\": \"\", \"label\": \"Test.Filename\", \"description\": \"test data file name - .gct, .res, .odf type = Dataset\"}, {\"name\": \"test.class.filename\", \"value\": \"\", \"label\": \"Test.Class.Filename\", \"description\": \"class file for test data - .cls\"}, {\"name\": \"min.std\", \"value\": \"\", \"label\": \"Min.Std\", \"description\": \"minimum standard deviation if feature selection statistic includes min std option\"}, {\"name\": \"feature.list.filename\", \"value\": \"\", \"label\": \"Feature.List.Filename\", \"description\": \"features to use for prediction\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"KNN\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"model.file\", \"value\": \"<train.filename_basename>.model\", \"label\": \"Model.File\", \"description\": \"name of output KNN model file  - .odf type = KNN Prediction Model\"}, {\"name\": \"num.features\", \"value\": \"10\", \"label\": \"Num.Features\", \"description\": \"number of selected features if feature list filename is not specified\"}, {\"name\": \"feature.selection.statistic\", \"value\": \"0\", \"label\": \"Feature.Selection.Statistic\", \"description\": \"statistic to use to perform feature selection\"}, {\"name\": \"num.neighbors\", \"value\": \"3\", \"label\": \"Num.Neighbors\", \"description\": \"number of neighbors for KNN\"}, {\"name\": \"weighting.type\", \"value\": \"distance\", \"label\": \"Weighting.Type\", \"description\": \"weighting type for neighbors\"}, {\"name\": \"distance.measure\", \"value\": \"1\", \"label\": \"Distance.Measure\", \"description\": \"distance measure\"}, {\"name\": \"pred.results.file\", \"value\": \"<test.filename_basename>.pred.odf\", \"label\": \"Pred.Results.File\", \"description\": \"name of output file for prediction results - .odf\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"KNNXValidation\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"KNN classification with Leave-One-Out Cross-Validation\", \"required_args\": [{\"name\": \"data.filename\", \"value\": \"\", \"label\": \"Data.Filename\", \"description\": \"data file - .gct, .res, .odf type = Dataset\"}, {\"name\": \"class.filename\", \"value\": \"\", \"label\": \"Class.Filename\", \"description\": \"class file - .cls\"}, {\"name\": \"min.std\", \"value\": \"\", \"label\": \"Min.Std\", \"description\": \"minimum standard deviation if feature selection statistic includes min std option\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"KNNXValidation\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"num.features\", \"value\": \"10\", \"label\": \"Num.Features\", \"description\": \"number of selected features\"}, {\"name\": \"feature.selection.statistic\", \"value\": \"0\", \"label\": \"Feature.Selection.Statistic\", \"description\": \"statistic to use to perform feature selection\"}, {\"name\": \"num.neighbors\", \"value\": \"3\", \"label\": \"Num.Neighbors\", \"description\": \"number of neighbors for KNN\"}, {\"name\": \"weighting.type\", \"value\": \"distance\", \"label\": \"Weighting.Type\", \"description\": \"weighting type for neighbors\"}, {\"name\": \"distance.measure\", \"value\": \"1\", \"label\": \"Distance.Measure\", \"description\": \"distance measure\"}, {\"name\": \"pred.results.file\", \"value\": \"<data.filename_basename>.pred.odf\", \"label\": \"Pred.Results.File\", \"description\": \"prediction results output file name .odf type = Prediction Results\"}, {\"name\": \"feature.summary.file\", \"value\": \"<data.filename_basename>.feat.odf\", \"label\": \"Feature.Summary.File\", \"description\": \"feature summary output file name - .odf type = Prediction Features\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"KSscore\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Computes the Kolmogorov-Smirnov score for a set of genes within an ordered list.\", \"required_args\": [{\"name\": \"query.filename\", \"value\": \"\", \"label\": \"Query.Filename\", \"description\": \"query gene set file containing one identifier per line\"}, {\"name\": \"input.filename\", \"value\": \"\", \"label\": \"Input.Filename\", \"description\": \"file containing an ordered list of genes (.pol format)\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"KSscore\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output\", \"value\": \"<query.filename_basename>\", \"label\": \"Output\", \"description\": \"output file name (a .pol extension will be appended)\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"LandmarkMatch\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"A proteomics method to propogate identifed peptides across multiple MS runs\", \"required_args\": [{\"name\": \"peakList.filename\", \"value\": \"\", \"label\": \"PeakList.Filename\", \"description\": \"Zip file containing peak lists\"}, {\"name\": \"retentionTime.filename\", \"value\": \"\", \"label\": \"RetentionTime.Filename\", \"description\": \"Zip file containing retention time (in this case from mapquant)\"}, {\"name\": \"sampleInfo.filename\", \"value\": \"\", \"label\": \"SampleInfo.Filename\", \"description\": \"CSV file with header: experiment, sample, class\"}, {\"name\": \"globalID.filename\", \"value\": \"\", \"label\": \"GlobalID.Filename\", \"description\": \"Global list of identified peptides\"}, {\"name\": \"accurateMass.filename\", \"value\": \"\", \"label\": \"AccurateMass.Filename\", \"description\": \"OPTIONAL: Specify the accurate mass table\"}, {\"name\": \"prefitCoef.filename\", \"value\": \"\", \"label\": \"PrefitCoef.Filename\", \"description\": \"OPTIONAL: Specify the prefit coefficients in a file\"}, {\"name\": \"Bootstrap\", \"value\": \"\", \"label\": \"Bootstrap\", \"description\": \"Use bootstrapping to calculate landmark match statistics\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"LandmarkMatch\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"running.on.LSF\", \"value\": \"no\", \"label\": \"Running.On.LSF\", \"description\": \"Whether you are running on an LSF cluster version of GenePattern\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"Life.Cycle.Time.Point.Correlation\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Calculate correlation to life cycle time point of p.falciparum (derisi data)\", \"required_args\": [{\"name\": \"input.file\", \"value\": \"\", \"label\": \"Input.File\", \"description\": \"Input data file name. Data in CSV format, with plasmodbID as first column \"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"Life.Cycle.Time.Point.Correlation\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.file\", \"value\": \"<input.file_basename>-correlation.csv\", \"label\": \"Output.File\", \"description\": \"Name of output file that will contain the resulting correlation table. CSV format.\"}, {\"name\": \"correlation\", \"value\": \"pearson\", \"label\": \"Correlation\", \"description\": \"Type of correlation to calculate\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"ListFiles\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"a helper module which outputs a filelist, the list of all files in the input directory, similar to the unix 'ls' command\", \"required_args\": [{\"name\": \"dir\", \"value\": \"\", \"label\": \"Dir\", \"description\": \"a directory\"}, {\"name\": \"recursive\", \"value\": \"\", \"label\": \"Recursive\", \"description\": \"\"}, {\"name\": \"includePattern\", \"value\": \"\", \"label\": \"IncludePattern\", \"description\": \"an optional list of filename patterns, e.g. *.cls,*.gct\"}, {\"name\": \"pruneDir\", \"value\": \"\", \"label\": \"PruneDir\", \"description\": \"an optional list of directory name patterns to ignore, e.g. *.svn\"}, {\"name\": \"order.by\", \"value\": \"\", \"label\": \"Order.By\", \"description\": \"an optional flag for how to order the files\"}, {\"name\": \"type\", \"value\": \"\", \"label\": \"Type\", \"description\": \"\"}, {\"name\": \"output.filename\", \"value\": \"\", \"label\": \"Output.Filename\", \"description\": \"the name of the output file, if none specified, write to stdout\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"ListFiles\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"LocatePeaks\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Locates detected peaks in a spectrum.\", \"required_args\": [{\"name\": \"peaks.output.filename\", \"value\": \"\", \"label\": \"Peaks.Output.Filename\", \"description\": \"The output from the Peaks module\"}, {\"name\": \"locate.peaks.as\", \"value\": \"\", \"label\": \"Locate.Peaks.As\", \"description\": \"max-location of maximum filtered value, area-center of area of the peak\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"LocatePeaks\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.file\", \"value\": \"<peaks.output.filename_basename>.locate.odf\", \"label\": \"Output.File\", \"description\": \"output file name\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"LogicleTransformFCS\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Logicle transformation of (selected) parameters in a list mode FCS data file.\", \"required_args\": [{\"name\": \"Input.fcs.data.file\", \"value\": \"\", \"label\": \"Input.Fcs.Data.File\", \"description\": \"FCS data file to transform.\"}, {\"name\": \"Parameters.to.transform\", \"value\": \"\", \"label\": \"Parameters.To.Transform\", \"description\": \"Comma-separated list of parameters to transform, parameters are specified by their short names ($PnN values); Optional. By default, all parameters with the \\\"FL\\\" prefix in their short name will be transformed.\"}, {\"name\": \"T\", \"value\": \"\", \"label\": \"T\", \"description\": \"T > 0; Data value at the top of the scale range to be displayed, optional. The maximum input value will be used if no explicit value for T is provided.\"}, {\"name\": \"W\", \"value\": \"\", \"label\": \"W\", \"description\": \"W >= 0; Linearization width in asymptotic decades; optional. The described calculation will be used to set W for each of the transformed FCS parameters if no explicit value for W is provided.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"LogicleTransformFCS\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"M\", \"value\": \"4.5\", \"label\": \"M\", \"description\": \"M < 0; Full width of the intended Logicle display in asymptotic decades, optional. The default value is 4.5 (decades). Note also that for the resulting Logicle transformation, Logicle (T) = M.\"}, {\"name\": \"A\", \"value\": \"0\", \"label\": \"A\", \"description\": \"A, usually <= 0; Additional negative display range in asymptotic decades; optional. The default value is 0.\"}, {\"name\": \"Use.Fast.Implementation\", \"value\": \"no\", \"label\": \"Use.Fast.Implementation\", \"description\": \"Use a fast implementation of Logicle with less precision; optional.\"}, {\"name\": \"Output.fcs.file.name\", \"value\": \"<Input.fcs.data.file_basename>.logicle_transformed.fcs\", \"label\": \"Output.Fcs.File.Name\", \"description\": \"The output FCS file name\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"LogisticRegression\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Logistic regression using clinical data and .cls file\", \"required_args\": [{\"name\": \"input.clin.data.filename\", \"value\": \"\", \"label\": \"Input.Clin.Data.Filename\", \"description\": \"input clinical data (.txt file)\"}, {\"name\": \"input.cls.filename\", \"value\": \"\", \"label\": \"Input.Cls.Filename\", \"description\": \"input class information (.cls file)\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"LogisticRegression\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.file\", \"value\": \"LogisticRegression_result\", \"label\": \"Output.File\", \"description\": \"output file name\"}, {\"name\": \"response.variable\", \"value\": \"NA\", \"label\": \"Response.Variable\", \"description\": \"\"}, {\"name\": \"variable.continuous\", \"value\": \"NA\", \"label\": \"Variable.Continuous\", \"description\": \"variable.continuous\"}, {\"name\": \"variable.category\", \"value\": \"NA\", \"label\": \"Variable.Category\", \"description\": \"(e.g. sex,cls,grade,...)\"}, {\"name\": \"variable.interaction.terms\", \"value\": \"NA\", \"label\": \"Variable.Interaction.Terms\", \"description\": \"(e.g. sex:cls,sex:grade,...)\"}, {\"name\": \"input.subgroup\", \"value\": \"NA\", \"label\": \"Input.Subgroup\", \"description\": \"(e.g. cls:2) *only one subgroup can be specified\"}, {\"name\": \"variable.selection\", \"value\": \"none\", \"label\": \"Variable.Selection\", \"description\": \"\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"LogTransformGLAD\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Log base 2 transform last column of GLAD file (for GISTIC)\", \"required_args\": [{\"name\": \"input.filename\", \"value\": \"\", \"label\": \"Input.Filename\", \"description\": \"The file to transform the last column of (glad format)\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"LogTransformGLAD\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.file\", \"value\": \"<input.filename_basename>.LogTransformed.glad\", \"label\": \"Output.File\", \"description\": \"Output filename\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"LOHPaired\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Computes LOH for paired samples\", \"required_args\": [{\"name\": \"input.filename\", \"value\": \"\", \"label\": \"Input.Filename\", \"description\": \"A .snp file that contains genotypes\"}, {\"name\": \"sample.info.filename\", \"value\": \"\", \"label\": \"Sample.Info.Filename\", \"description\": \"A sample info file with the paired information\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"LOHPaired\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.file\", \"value\": \"<input.filename_basename>.loh\", \"label\": \"Output.File\", \"description\": \"Output file\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"Lu.Getz.Miska.Nature.June.2005.mouse.lung.pipeline\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Normal/tumor classifier and kNN prediction of mouse lung samples\", \"required_args\": [], \"default_args\": [{\"name\": \"task_name\", \"value\": \"Lu.Getz.Miska.Nature.June.2005.mouse.lung.pipeline\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"MAGEMLImportViewer\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"A visualizer to import data in MAGE-ML format into GenePattern\", \"required_args\": [{\"name\": \"input.filename\", \"value\": \"\", \"label\": \"Input.Filename\", \"description\": \"The archive file containing the MAGE-ML files - .tar, .tgz, .tar.gz, .zip, .gz\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"MAGEMLImportViewer\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"MAGETABImportViewer\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"A visualizer to import data in MAGE-TAB format into GenePattern\", \"required_args\": [{\"name\": \"sdrf.file\", \"value\": \"\", \"label\": \"Sdrf.File\", \"description\": \"The sdrf file - .txt\"}, {\"name\": \"data.files\", \"value\": \"\", \"label\": \"Data.Files\", \"description\": \"A zip file containing MAGE-TAB data matrix files - .zip\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"MAGETABImportViewer\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"MapChipFeaturesGeneral\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Change (map) the features (genes) of a dataset\", \"required_args\": [{\"name\": \"input.filename\", \"value\": \"\", \"label\": \"Input.Filename\", \"description\": \"Input filename - .res\"}, {\"name\": \"lookup.table.filename\", \"value\": \"\", \"label\": \"Lookup.Table.Filename\", \"description\": \"Name of the look up table file with the feature mapping\"}, {\"name\": \"table.format\", \"value\": \"\", \"label\": \"Table.Format\", \"description\": \"Table format\"}, {\"name\": \"direction\", \"value\": \"\", \"label\": \"Direction\", \"description\": \"Direction of look up\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"MapChipFeaturesGeneral\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.prefix\", \"value\": \"<input.filename_basename>.fea\", \"label\": \"Output.Prefix\", \"description\": \"prefix for output file\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"Marker.Selection\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Marker selection, ranking and classification\", \"required_args\": [{\"name\": \"gct.file\", \"value\": \"\", \"label\": \"Gct.File\", \"description\": \"Input data file (in gct format)\"}, {\"name\": \"cls.file\", \"value\": \"\", \"label\": \"Cls.File\", \"description\": \"Class file (cls format) to match with input data file\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"Marker.Selection\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.prefix\", \"value\": \"<gct.file_basename>\", \"label\": \"Output.Prefix\", \"description\": \"Prefix for output file names\"}, {\"name\": \"fdr.threshold\", \"value\": \"0.01\", \"label\": \"Fdr.Threshold\", \"description\": \"FDR threshold for calling statistically significant markers\"}, {\"name\": \"gct.id.col\", \"value\": \"2\", \"label\": \"Gct.Id.Col\", \"description\": \"Column in data file containing (gene or protein) ID\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"MClustClusterLabel\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Model based labeling/matching of clusters (mate-clustering)\", \"required_args\": [{\"name\": \"Input.clustering.results\", \"value\": \"\", \"label\": \"Input.Clustering.Results\", \"description\": \"The input clustering results (of multiple data files) in CSV files with each clustered population stored in a separate CSV file. All these CSV files shall have the same columns (dimensions/channels) or at least, they all shall include columns specified by the dimensions parameter.\"}, {\"name\": \"No.of.labels\", \"value\": \"\", \"label\": \"No.Of.Labels\", \"description\": \"The number of distinct labels (populations) to look for. You may run the MClustClusterLabelBIC module to get the number of labels you should look for.\"}, {\"name\": \"Dimensions\", \"value\": \"\", \"label\": \"Dimensions\", \"description\": \"A comma-separated list of dimensions (parameters/channels) that shall be used for meta clustering (clustering of means of clusters). All specified dimensions shall be included in all the input clustering results CSV files. All dimensions (but Time) will be used if this parameter is not provided.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"MClustClusterLabel\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"Model\", \"value\": \"VII\", \"label\": \"Model\", \"description\": \"Specific model used for meta clustering; see documentation for details about different models.\"}, {\"name\": \"Uncertainty.threshold\", \"value\": \"65\", \"label\": \"Uncertainty.Threshold\", \"description\": \"The threshold (in percentage) for uncertainty when assigning labels. Input a value greater than 50 and lower than 100. Populations where no label can be assigned with probability greater or equal to this uncertainty threshold will be left unassigned in the results.\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"MClustClusterLabelBIC\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Bayesian information criterion (BIC) for model based labeling/matching of clusters; helps estimating the number of labels for MClustClusterLabel.\", \"required_args\": [{\"name\": \"Input.clustering.results\", \"value\": \"\", \"label\": \"Input.Clustering.Results\", \"description\": \"The input clustering results (of multiple data files) in CSV files with each clustered population stored in a separate CSV file. All these CSV files shall have the same columns (dimensions/channels) or at least, they all shall include columns specified by the dimensions parameter.\"}, {\"name\": \"Dimensions\", \"value\": \"\", \"label\": \"Dimensions\", \"description\": \"A comma-separated list of dimensions (parameters/channels) that shall be used for meta clustering (clustering of means of clusters). All specified dimensions shall be included in all the input clustering results CSV files. All dimensions (but Time) will be used if this parameter is not provided.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"MClustClusterLabelBIC\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"Range\", \"value\": \"1-10\", \"label\": \"Range\", \"description\": \"The range of distinct labels (populations, components) to calculate BIC for. Provide the range in the min-max format, where both, mix and max are integers and min is smaller than max.\"}, {\"name\": \"EII\", \"value\": \"TRUE\", \"label\": \"Eii\", \"description\": \"Include the EII model (spherical, equal volume) when calculating BIC.\"}, {\"name\": \"VII\", \"value\": \"TRUE\", \"label\": \"Vii\", \"description\": \"Include the VII model (spherical, unequal volume) when calculating BIC.\"}, {\"name\": \"EEI\", \"value\": \"FALSE\", \"label\": \"Eei\", \"description\": \"Include the EEI model (diagonal, equal volume and shape) when calculating BIC.\"}, {\"name\": \"VEI\", \"value\": \"FALSE\", \"label\": \"Vei\", \"description\": \"Include the VEI model (diagonal, varying volume, equal shape) when calculating BIC.\"}, {\"name\": \"EVI\", \"value\": \"FALSE\", \"label\": \"Evi\", \"description\": \"Include the EVI model (diagonal, equal volume, varying shape) when calculating BIC.\"}, {\"name\": \"VVI\", \"value\": \"FALSE\", \"label\": \"Vvi\", \"description\": \"Include the VVI model (diagonal, varying volume and shape) when calculating BIC.\"}, {\"name\": \"EEE\", \"value\": \"FALSE\", \"label\": \"Eee\", \"description\": \"Include the EEE model (ellipsoidal, equal volume, shape, and orientation) when calculating BIC.\"}, {\"name\": \"EEV\", \"value\": \"FALSE\", \"label\": \"Eev\", \"description\": \"Include the EEV model (ellipsoidal, equal volume and equal shape) when calculating BIC.\"}, {\"name\": \"VEV\", \"value\": \"FALSE\", \"label\": \"Vev\", \"description\": \"Include the VEV model (ellipsoidal, equal shape) when calculating BIC.\"}, {\"name\": \"VVV\", \"value\": \"FALSE\", \"label\": \"Vvv\", \"description\": \"Include the VVV model (ellipsoidal, varying volume, shape, and orientation) when calculating BIC.\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"MergeColumns\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Merge datasets by column.\", \"required_args\": [{\"name\": \"input.filename1\", \"value\": \"\", \"label\": \"Input.Filename1\", \"description\": \"first file to merge\"}, {\"name\": \"input.filename2\", \"value\": \"\", \"label\": \"Input.Filename2\", \"description\": \"second file to merge\"}, {\"name\": \"zip.filename\", \"value\": \"\", \"label\": \"Zip.Filename\", \"description\": \"zip file containing files to merge. Either a zip file or the input files above must be provided.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"MergeColumns\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output\", \"value\": \"<input.filename1_basename><zip.filename_basename>.merged\", \"label\": \"Output\", \"description\": \"name of the merged file\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"MergeFCSDataFiles\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Merge multiple FCS data files into a single FCS datasets; includes multimple subsampling options.\", \"required_args\": [{\"name\": \"Input.files\", \"value\": \"\", \"label\": \"Input.Files\", \"description\": \"The input FCS files to be merged; either a zip file or a pointer to a directory or a text file with FCS file names (full path) with one filename per line.\"}, {\"name\": \"Keep.n.events\", \"value\": \"\", \"label\": \"Keep.N.Events\", \"description\": \"Keep approximately n events from each dataset, optional, cannot be combined with subsampling rate.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"MergeFCSDataFiles\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"Subsampling.rate\", \"value\": \"100\", \"label\": \"Subsampling.Rate\", \"description\": \"Subsampling rate as percentage of events to be included from each data set, value between 0 and 100, optional, default 100%, cannot be combined with keep n events.\"}, {\"name\": \"Random.seed\", \"value\": \"1234567\", \"label\": \"Random.Seed\", \"description\": \"The random seed\"}, {\"name\": \"Output.file.name\", \"value\": \"MergedFCSDataFiles.fcs\", \"label\": \"Output.File.Name\", \"description\": \"The output FCS file name.\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"MergeHTSeqCounts\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Merges HTSeq read count data files into one file\", \"required_args\": [{\"name\": \"input.files\", \"value\": \"\", \"label\": \"Input.Files\", \"description\": \"A directory or one or more text files containing HTSeq read counts\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"MergeHTSeqCounts\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.prefix\", \"value\": \"<input.files_basename>\", \"label\": \"Output.Prefix\", \"description\": \"The base name of the output file. File extensions will be added automatically.\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"MergeRows\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Merge datasets by row.\", \"required_args\": [{\"name\": \"input.filename1\", \"value\": \"\", \"label\": \"Input.Filename1\", \"description\": \"first file to merge\"}, {\"name\": \"input.filename2\", \"value\": \"\", \"label\": \"Input.Filename2\", \"description\": \"second file to merge\"}, {\"name\": \"zip.filename\", \"value\": \"\", \"label\": \"Zip.Filename\", \"description\": \"zip file containing files to merge. Either a zip file or the input files above must be provided.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"MergeRows\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output\", \"value\": \"<input.filename1_basename><zip.filename_basename>.merged\", \"label\": \"Output\", \"description\": \"name of the merged file\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"MetageneProjection\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Projects one or more data sets onto the metagene representation of another data set\", \"required_args\": [{\"name\": \"model.gct.file\", \"value\": \"\", \"label\": \"Model.Gct.File\", \"description\": \"The model .gct data file\"}, {\"name\": \"model.cls.file\", \"value\": \"\", \"label\": \"Model.Cls.File\", \"description\": \"The model .cls data file\"}, {\"name\": \"model.preprocessing.file\", \"value\": \"\", \"label\": \"Model.Preprocessing.File\", \"description\": \"The pre-processing parameters for the model dataset\"}, {\"name\": \"test.gct.file\", \"value\": \"\", \"label\": \"Test.Gct.File\", \"description\": \"The test .gct or zip of .gct data files\"}, {\"name\": \"test.cls.file\", \"value\": \"\", \"label\": \"Test.Cls.File\", \"description\": \"The test .cls or zip of .cls data files\"}, {\"name\": \"test.preprocessing.file\", \"value\": \"\", \"label\": \"Test.Preprocessing.File\", \"description\": \"The pre-processing parameters for the test dataset\"}, {\"name\": \"phenotype.plotting.colors\", \"value\": \"\", \"label\": \"Phenotype.Plotting.Colors\", \"description\": \"Plotting colors for phenotypes\"}, {\"name\": \"phenotype.plotting.symbols\", \"value\": \"\", \"label\": \"Phenotype.Plotting.Symbols\", \"description\": \"Plotting symbols for phenotypes\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"MetageneProjection\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"num.characters\", \"value\": \"9\", \"label\": \"Num.Characters\", \"description\": \"Number of characters to use to identify classes from the .cls files\"}, {\"name\": \"identifier\", \"value\": \"<model.gct.file_basename>.output\", \"label\": \"Identifier\", \"description\": \"Prefix to name output files\"}, {\"name\": \"k.projection\", \"value\": \"3\", \"label\": \"K.Projection\", \"description\": \"Number of metagenes in projection\"}, {\"name\": \"algorithm\", \"value\": \"NMF.div\", \"label\": \"Algorithm\", \"description\": \"Algorithm for Metagene Projection\"}, {\"name\": \"number.of.iterations\", \"value\": \"2000\", \"label\": \"Number.Of.Iterations\", \"description\": \"Number of algorithm iterations\"}, {\"name\": \"seed\", \"value\": \"1234\", \"label\": \"Seed\", \"description\": \"Random seed to initialize metagene matrices (Note: results may only be reproducible on the same platform)\"}, {\"name\": \"post.projection.normalization\", \"value\": \"yes\", \"label\": \"Post.Projection.Normalization\", \"description\": \"apply post-projection normalization\"}, {\"name\": \"heatmap.row.norm\", \"value\": \"no\", \"label\": \"Heatmap.Row.Norm\", \"description\": \"row-normalize (standardize) the rows in the heat map\"}, {\"name\": \"heatmap.color.scheme\", \"value\": \"reddish color map\", \"label\": \"Heatmap.Color.Scheme\", \"description\": \"\"}, {\"name\": \"confidence.threshold\", \"value\": \"0.3\", \"label\": \"Confidence.Threshold\", \"description\": \"Confidence threshold (Brier score) to separate call from no-calls\"}, {\"name\": \"symbol.scaling\", \"value\": \"1\", \"label\": \"Symbol.Scaling\", \"description\": \"Graphical scaling for symbols in plots and plot legends\"}, {\"name\": \"kernel\", \"value\": \"radial\", \"label\": \"Kernel\", \"description\": \"Kernel function for SVM\"}, {\"name\": \"cost\", \"value\": \"1\", \"label\": \"Cost\", \"description\": \"cost function for SVM\"}, {\"name\": \"gamma\", \"value\": \"0.05\", \"label\": \"Gamma\", \"description\": \"Gamma coefficient for radial base function kernel\"}, {\"name\": \"theta\", \"value\": \"0\", \"label\": \"Theta\", \"description\": \"Smoothing parameter only used for the NSNMF.div algorithm\"}, {\"name\": \"lambda\", \"value\": \"0\", \"label\": \"Lambda\", \"description\": \"Sparse parameter used only for the SNMF algorithm\"}, {\"name\": \"model.set.refinement\", \"value\": \"yes\", \"label\": \"Model.Set.Refinement\", \"description\": \"perform model set refinement\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"MINDY\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Runs the MINDY algorithm for inferring genes that modulate the activity of a transcription factor at post-transcriptional levels \", \"required_args\": [{\"name\": \"input.file\", \"value\": \"\", \"label\": \"Input.File\", \"description\": \"The dataset file - .res, .gct\"}, {\"name\": \"modulator.list\", \"value\": \"\", \"label\": \"Modulator.List\", \"description\": \"A list of possible modulator genes to analyze - .txt\"}, {\"name\": \"dpi.transcription.factor.list\", \"value\": \"\", \"label\": \"Dpi.Transcription.Factor.List\", \"description\": \"List of transcription factors in dataset - .txt\"}, {\"name\": \"transcription.factor\", \"value\": \"\", \"label\": \"Transcription.Factor\", \"description\": \"Transcription factor\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"MINDY\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"sample.percentage\", \"value\": \"35.0\", \"label\": \"Sample.Percentage\", \"description\": \"Percentage of samples to use in analysis (must be <=50)\"}, {\"name\": \"conditional.mi.threshold\", \"value\": \"0.0\", \"label\": \"Conditional.Mi.Threshold\", \"description\": \"Conditional mutual information (MI) score threshold (number between 0 and 1)\"}, {\"name\": \"conditional.mi.p.value\", \"value\": \"1.0\", \"label\": \"Conditional.Mi.P.Value\", \"description\": \"Conditional mutual info p-value threshold (ignored if conditonal MI threshold >0)\"}, {\"name\": \"dpi.tolerance\", \"value\": \"1.0\", \"label\": \"Dpi.Tolerance\", \"description\": \"The data processing inequality tolerance (number between 0 and 1)\"}, {\"name\": \"unconditional.mi.threshold\", \"value\": \"0.0\", \"label\": \"Unconditional.Mi.Threshold\", \"description\": \"Unconditional mutual information (MI) score threshold (number between 0 and 1)\"}, {\"name\": \"unconditional.mi.p.value\", \"value\": \"1.0\", \"label\": \"Unconditional.Mi.P.Value\", \"description\": \"Unconditional mutual information p-value threshold (ignored if unconditonal MI threshold >0)\"}, {\"name\": \"output.file\", \"value\": \"<input.file_basename>.mindy\", \"label\": \"Output.File\", \"description\": \"Output file name\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"MINDYViewer\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Visualizes the results of MINDY\", \"required_args\": [{\"name\": \"dataset.file\", \"value\": \"\", \"label\": \"Dataset.File\", \"description\": \"The dataset file - .res, .gct\"}, {\"name\": \"mindy.file\", \"value\": \"\", \"label\": \"Mindy.File\", \"description\": \"Output from MINDY- .mindy\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"MINDYViewer\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"ModEcocTrainTest\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Modified error correcting output code prediction\", \"required_args\": [{\"name\": \"train.file\", \"value\": \"\", \"label\": \"Train.File\", \"description\": \"The training features and class label information in .arff format\"}, {\"name\": \"test.file\", \"value\": \"\", \"label\": \"Test.File\", \"description\": \"The testing features and possible class label information in .arff format\"}, {\"name\": \"saved.model.file\", \"value\": \"\", \"label\": \"Saved.Model.File\", \"description\": \"Input modEcocTrainTest model\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"ModEcocTrainTest\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"model.output.file\", \"value\": \"<train.file_basename>.model\", \"label\": \"Model.Output.File\", \"description\": \"Output modEcocTrainTest model\"}, {\"name\": \"base.classifier\", \"value\": \"0\", \"label\": \"Base.Classifier\", \"description\": \"The name of the base classifier to use\"}, {\"name\": \"base.multiplier\", \"value\": \"4\", \"label\": \"Base.Multiplier\", \"description\": \"The base classifier multiplier\"}, {\"name\": \"pred.results.output.file\", \"value\": \"<test.file_basename>\", \"label\": \"Pred.Results.Output.File\", \"description\": \"The name of the output file for prediction results.\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"ModEcocXValidation\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Modified error correcting output code cross validation prediction\", \"required_args\": [{\"name\": \"input.file\", \"value\": \"\", \"label\": \"Input.File\", \"description\": \"The input features and class label information in .arff format\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"ModEcocXValidation\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"num.folds\", \"value\": \"2\", \"label\": \"Num.Folds\", \"description\": \"The number of folds should be an integer greater than or equal to two.\"}, {\"name\": \"base.classifier\", \"value\": \"0\", \"label\": \"Base.Classifier\", \"description\": \"The name of the base classifier to use\"}, {\"name\": \"base.multiplier\", \"value\": \"4\", \"label\": \"Base.Multiplier\", \"description\": \"The base classifier multiplier\"}, {\"name\": \"pred.results.file\", \"value\": \"<input.file_basename>\", \"label\": \"Pred.Results.File\", \"description\": \"The name of the output file for prediction results.\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"Multiplot\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Creates customizable plots of expression data-derived data.\", \"required_args\": [{\"name\": \"input.file.name\", \"value\": \"\", \"label\": \"Input.File.Name\", \"description\": \"File created by MultiplotPreprocess (.zip)\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"Multiplot\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"number.of.plots\", \"value\": \"1\", \"label\": \"Number.Of.Plots\", \"description\": \"Smaller one-plot visualizer or bigger two-plot\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"MultiplotExtractor\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Creates customizable plots of expression data-derived data.\", \"required_args\": [{\"name\": \"input.filename\", \"value\": \"\", \"label\": \"Input.Filename\", \"description\": \"File created by MultiplotPreprocess (.zip)\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"MultiplotExtractor\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"MultiplotPreprocess\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Creates derived data from an expression dataset for use in the Multiplot Visualizer module\", \"required_args\": [{\"name\": \"input.filename\", \"value\": \"\", \"label\": \"Input.Filename\", \"description\": \"File with data to prepare for plotting (gct)\"}, {\"name\": \"class.filename\", \"value\": \"\", \"label\": \"Class.Filename\", \"description\": \"Class file for this data (cls)\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"MultiplotPreprocess\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.file\", \"value\": \"<input.filename_basename>.zip\", \"label\": \"Output.File\", \"description\": \"Name of file to create for use in Multiplot visualizer\"}, {\"name\": \"replicateFC\", \"value\": \"0\", \"label\": \"ReplicateFC\", \"description\": \"Whether to compute fold changes between each replicate (turning this on can slow your work).\"}, {\"name\": \"randomData\", \"value\": \"0\", \"label\": \"RandomData\", \"description\": \"Whether to create random expression data (and fold changes, etc for this data). Turning this on can slow your work\"}, {\"name\": \"outlierElimination\", \"value\": \"0\", \"label\": \"OutlierElimination\", \"description\": \"Remove outlier replicate values of genes from processing (see \\\"Help\\\" for details)\"}, {\"name\": \"doStats\", \"value\": \"1\", \"label\": \"DoStats\", \"description\": \"Perform t-test, Hochberg p-value adjustment, and significance comparison to randomized data\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"MultiplotStudio\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Scatterplot visualization tool for analyzing differential gene expression.\", \"required_args\": [{\"name\": \"dataFile\", \"value\": \"\", \"label\": \"DataFile\", \"description\": \"GenePattern dataset (*.gct), Multiplot Studio Workspace file (*.msdata) or CSV (see help for format requirements).\"}, {\"name\": \"classFile\", \"value\": \"\", \"label\": \"ClassFile\", \"description\": \"Class file (*.cls). Not required, but lack of a class file will result in fewer calculation options. Ignored if dataFile is a Workspace.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"MultiplotStudio\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"MutSigCV\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Identifies genes that are significantly mutated in cancer genomes, using a model with mutational covariates.\", \"required_args\": [{\"name\": \"mutation.table.file\", \"value\": \"\", \"label\": \"Mutation.Table.File\", \"description\": \"Mutation list in MAF format.\"}, {\"name\": \"coverage.table.file\", \"value\": \"\", \"label\": \"Coverage.Table.File\", \"description\": \"File containing the coverage table.  This should be a file with per-sample coverage as produced by MuTect from the samples' WIG files.  If you do not have this level of coverage information, you can instead use the territory file for the reference human exome to direct MutSigCV to assume full coverage.  This file is found using \\\"Add Path or URL\\\" under \\\"shared_data/example_files/MutSigCV_1.3/exome_full192.coverage.txt\\\".\"}, {\"name\": \"mutation.type.dictionary\", \"value\": \"\", \"label\": \"Mutation.Type.Dictionary\", \"description\": \"The mutation type dictionary to use for automatic category and effect discovery.  This is necessary only if you are using a MAF file without the columns \\\"categ\\\" and \\\"effect\\\".  A suggested dictionary can be found using \\\"Add Path or URL\\\" under \\\"shared_data/example_files/MutSigCV_1.3/mutation_type_dictionary_file.txt\\\".\"}, {\"name\": \"genome.build\", \"value\": \"\", \"label\": \"Genome.Build\", \"description\": \"Genome build to use for automatic category and effect discovery.  This is necessary only if you are using a MAF file without the columns \\\"categ\\\" and \\\"effect\\\".\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"MutSigCV\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"covariates.table.file\", \"value\": \"ftp://ftp.broadinstitute.org/pub/genepattern/example_files/MutSigCV_1.3/gene.covariates.txt\", \"label\": \"Covariates.Table.File\", \"description\": \"File containing the covariates table.  The default selection is the covariates file used in the MutSigCV paper published in Nature (see the documentation for further information and a full reference). \"}, {\"name\": \"output.filename.base\", \"value\": \"<mutation.table.file_basename>\", \"label\": \"Output.Filename.Base\", \"description\": \"Base name for the output files.\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"mzXMLToCsv\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Converts a mzXML file to a zip of csv files\", \"required_args\": [{\"name\": \"input.filename\", \"value\": \"\", \"label\": \"Input.Filename\", \"description\": \"input filename - .mzXML\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"mzXMLToCsv\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.file\", \"value\": \"<input.filename_basename>.zip\", \"label\": \"Output.File\", \"description\": \"output file containing zip of csv files\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"NearestTemplatePrediction\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Nearest neighbor prediction based on a list of marker genes\", \"required_args\": [{\"name\": \"input.exp.filename\", \"value\": \"\", \"label\": \"Input.Exp.Filename\", \"description\": \"Gene expression data set (.gct)\"}, {\"name\": \"input.features.filename\", \"value\": \"\", \"label\": \"Input.Features.Filename\", \"description\": \"List of marker genes (.txt): Probe ID, Gene name, Class (1,2,...), Weight (optional)\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"NearestTemplatePrediction\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.name\", \"value\": \"NTP\", \"label\": \"Output.Name\", \"description\": \"Name for output files\"}, {\"name\": \"distance.selection\", \"value\": \"cosine\", \"label\": \"Distance.Selection\", \"description\": \"Distance metric\"}, {\"name\": \"weight.genes\", \"value\": \"T\", \"label\": \"Weight.Genes\", \"description\": \"Weight genes? (by statistic, fold change, etc. only for 2 classes)\"}, {\"name\": \"num.resamplings\", \"value\": \"1000\", \"label\": \"Num.Resamplings\", \"description\": \"# resampling to generate null distribution for distance metric\"}, {\"name\": \"GenePattern.output\", \"value\": \"F\", \"label\": \"GenePattern.Output\", \"description\": \"Create .gct and .cls files for GenePattern\"}, {\"name\": \"random.seed\", \"value\": \"7392854\", \"label\": \"Random.Seed\", \"description\": \"Random seed\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"NMF\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Non-negative Matrix Factorization\", \"required_args\": [{\"name\": \"input.filename\", \"value\": \"\", \"label\": \"Input.Filename\", \"description\": \"Input file in GCT format with input data (V matrix) (GCT format only). e.g. ALL_vs_AML.gct\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"NMF\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.file\", \"value\": \"<input.filename_basename>.log\", \"label\": \"Output.File\", \"description\": \"Output log file. e.g. NMF.log\"}, {\"name\": \"nfact\", \"value\": \"4\", \"label\": \"Nfact\", \"description\": \"Number of NMF factors. e.g. 2, 4, 10\"}, {\"name\": \"niter\", \"value\": \"100\", \"label\": \"Niter\", \"description\": \"Number of NMF iterations. e.g. 100, 250, 500\"}, {\"name\": \"print.dump\", \"value\": \"0\", \"label\": \"Print.Dump\", \"description\": \"Print matrices in output for debug\"}, {\"name\": \"cnorm.flag\", \"value\": \"0\", \"label\": \"Cnorm.Flag\", \"description\": \"Column normalization flag\"}, {\"name\": \"rnorm.flag\", \"value\": \"0\", \"label\": \"Rnorm.Flag\", \"description\": \"Row normalization flag\"}, {\"name\": \"gshift\", \"value\": \"0\", \"label\": \"Gshift\", \"description\": \"Global additive shift after normalization. e.g. 1, 4\"}, {\"name\": \"norm.type\", \"value\": \"0\", \"label\": \"Norm.Type\", \"description\": \"Type of normalization\"}, {\"name\": \"seed\", \"value\": \"12345\", \"label\": \"Seed\", \"description\": \"Random seed used to initialize W and H matrices by the randon number generator. e.g. 4585, 4567, 5980\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"NMFConsensus\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Non-negative Matrix Factorization (NMF) Consensus Clustering\", \"required_args\": [{\"name\": \"dataset.filename\", \"value\": \"\", \"label\": \"Dataset.Filename\", \"description\": \"Input dataset (gct or res)\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"NMFConsensus\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"k.initial\", \"value\": \"2\", \"label\": \"K.Initial\", \"description\": \"Initial value of K\"}, {\"name\": \"k.final\", \"value\": \"5\", \"label\": \"K.Final\", \"description\": \"Final value of K\"}, {\"name\": \"num.clusterings\", \"value\": \"20\", \"label\": \"Num.Clusterings\", \"description\": \"Number of clusterings per value of K\"}, {\"name\": \"max.num.iterations\", \"value\": \"2000\", \"label\": \"Max.Num.Iterations\", \"description\": \"Maximum number of NMF iterations per clustering\"}, {\"name\": \"error.function\", \"value\": \"divergence\", \"label\": \"Error.Function\", \"description\": \"Error function\"}, {\"name\": \"random.seed\", \"value\": \"123456789\", \"label\": \"Random.Seed\", \"description\": \"Seed for random number generator\"}, {\"name\": \"output.file.prefix\", \"value\": \"<dataset.filename_basename>\", \"label\": \"Output.File.Prefix\", \"description\": \"Prefix to append to all output files\"}, {\"name\": \"stop.convergence\", \"value\": \"40\", \"label\": \"Stop.Convergence\", \"description\": \"How many \\\"no change\\\" checks are needed to stop NMF iterations before max iterations is reached(convergence)\"}, {\"name\": \"stop.frequency\", \"value\": \"10\", \"label\": \"Stop.Frequency\", \"description\": \"Frequency (NMF iterations) of \\\"no change\\\" checks\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"NMFConsensus_combine_lewu\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Non-negative Matrix Factorization (NMF) Consensus Clustering (combine single k result)\", \"required_args\": [{\"name\": \"param.filename\", \"value\": \"\", \"label\": \"Param.Filename\", \"description\": \"Input param file \"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"NMFConsensus_combine_lewu\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"NMFConsensus_sk_lewu\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Non-negative Matrix Factorization (NMF) Consensus Clustering\", \"required_args\": [{\"name\": \"param.filename\", \"value\": \"\", \"label\": \"Param.Filename\", \"description\": \"Input param file \"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"NMFConsensus_sk_lewu\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"NMFPartition\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"The module will output a list of parameter files\", \"required_args\": [{\"name\": \"dataset.filename\", \"value\": \"\", \"label\": \"Dataset.Filename\", \"description\": \"Input dataset (gct or res)\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"NMFPartition\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"k.initial\", \"value\": \"2\", \"label\": \"K.Initial\", \"description\": \"Initial value of K\"}, {\"name\": \"k.final\", \"value\": \"5\", \"label\": \"K.Final\", \"description\": \"Final value of K\"}, {\"name\": \"num.clusterings\", \"value\": \"20\", \"label\": \"Num.Clusterings\", \"description\": \"Number of clusterings per value of K\"}, {\"name\": \"max.num.iterations\", \"value\": \"2000\", \"label\": \"Max.Num.Iterations\", \"description\": \"Maximum number of NMF iterations per clustering\"}, {\"name\": \"error.function\", \"value\": \"divergence\", \"label\": \"Error.Function\", \"description\": \"Error function\"}, {\"name\": \"random.seed\", \"value\": \"123456789\", \"label\": \"Random.Seed\", \"description\": \"Seed for random number generator\"}, {\"name\": \"output.file.prefix\", \"value\": \"<dataset.filename_basename>\", \"label\": \"Output.File.Prefix\", \"description\": \"Prefix to append to all output files\"}, {\"name\": \"stop.convergence\", \"value\": \"40\", \"label\": \"Stop.Convergence\", \"description\": \"How many \\\"no change\\\" checks are needed to stop NMF iterations before max iterations is reached(convergence)\"}, {\"name\": \"stop.frequency\", \"value\": \"10\", \"label\": \"Stop.Frequency\", \"description\": \"Frequency (NMF iterations) of \\\"no change\\\" checks\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"NMFSingleKPipe\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"\", \"required_args\": [{\"name\": \"NMFConsensus_sk_lewu1.param.filename\", \"value\": \"\", \"label\": \"NMFConsensus Sk Lewu1.Param.Filename\", \"description\": \"Input param file \"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"NMFSingleKPipe\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"NormLines\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Normalizes RNAi hairpin scores across cell lines using ZMAD, PMAD, rank or quantile- ignores NaNs\", \"required_args\": [{\"name\": \"gct.file\", \"value\": \"\", \"label\": \"Gct.File\", \"description\": \"Input gct file to normalize by samples\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"NormLines\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"method\", \"value\": \"quantile\", \"label\": \"Method\", \"description\": \"Options are ZMAD, PMAD, quantile\"}, {\"name\": \"plotfile\", \"value\": \"NormPDF.pdf\", \"label\": \"Plotfile\", \"description\": \"\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"ParallelCBS\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Runs CBS algorithm on multiple samples in parallel\", \"required_args\": [{\"name\": \"SplitColumns1.input.file\", \"value\": \"\", \"label\": \"SplitColumns1.Input.File\", \"description\": \"A tab delimited file\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"ParallelCBS\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"ConcatenateFilelist4.output.filename\", \"value\": \"<filelist_basename>.catted.seg.txt\", \"label\": \"ConcatenateFilelist4.Output.Filename\", \"description\": \"The name of the output file\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"ParallelNMFConsensusPipe\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"A parallelized implementation of the NMFConsensus module, implemented as a GenePattern scatter-gather pipeline.\", \"required_args\": [{\"name\": \"NMFPartition1.dataset.filename\", \"value\": \"\", \"label\": \"NMFPartition1.Dataset.Filename\", \"description\": \"Input dataset (gct or res)\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"ParallelNMFConsensusPipe\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"NMFPartition1.k.initial\", \"value\": \"2\", \"label\": \"NMFPartition1.K.Initial\", \"description\": \"Initial value of K\"}, {\"name\": \"NMFPartition1.k.final\", \"value\": \"5\", \"label\": \"NMFPartition1.K.Final\", \"description\": \"Final value of K\"}, {\"name\": \"NMFPartition1.num.clusterings\", \"value\": \"20\", \"label\": \"NMFPartition1.Num.Clusterings\", \"description\": \"Number of clusterings per value of K\"}, {\"name\": \"NMFPartition1.max.num.iterations\", \"value\": \"2000\", \"label\": \"NMFPartition1.Max.Num.Iterations\", \"description\": \"Maximum number of NMF iterations per clustering\"}, {\"name\": \"NMFPartition1.error.function\", \"value\": \"divergence\", \"label\": \"NMFPartition1.Error.Function\", \"description\": \"Error function\"}, {\"name\": \"NMFPartition1.random.seed\", \"value\": \"123456789\", \"label\": \"NMFPartition1.Random.Seed\", \"description\": \"Seed for random number generator\"}, {\"name\": \"NMFPartition1.output.file.prefix\", \"value\": \"<dataset.filename_basename>\", \"label\": \"NMFPartition1.Output.File.Prefix\", \"description\": \"Prefix to append to all output files\"}, {\"name\": \"NMFPartition1.stop.convergence\", \"value\": \"40\", \"label\": \"NMFPartition1.Stop.Convergence\", \"description\": \"How many \\\"no change\\\" checks are needed to stop NMF iterations before max iterations is reached(convergence)\"}, {\"name\": \"NMFPartition1.stop.frequency\", \"value\": \"10\", \"label\": \"NMFPartition1.Stop.Frequency\", \"description\": \"Frequency (NMF iterations) of \\\"no change\\\" checks\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"PARIS\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Feature selection by class or against a continuous vector (Probablility Analysis by Ranked Information Score)\", \"required_args\": [{\"name\": \"target.profile\", \"value\": \"\", \"label\": \"Target.Profile\", \"description\": \"Target profile file (either .cls or .gct).  Used to classify cell line samples into groups, or assign a continuous target vector to the cell line samples.\"}, {\"name\": \"target\", \"value\": \"\", \"label\": \"Target\", \"description\": \"Target gene or ID from target.profile file.  Should match a row in the 'Name' or 'Description' column of a .gct file, or match one label in the 2nd row of the .cls file\"}, {\"name\": \"data\", \"value\": \"\", \"label\": \"Data\", \"description\": \"Data (feature) file to rank based on target profile\"}, {\"name\": \"prefix\", \"value\": \"\", \"label\": \"Prefix\", \"description\": \"prefix for labeling output files\"}, {\"name\": \"advanced\", \"value\": \"\", \"label\": \"Advanced\", \"description\": \"Leave blank for default.  This parameter allows the user to change the underlying default metric.  Options will be listed in the documentation.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"PARIS\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"permtype\", \"value\": \"standard\", \"label\": \"Permtype\", \"description\": \"standard; subclass.stratified (lineage-corrected)\"}, {\"name\": \"type.flag\", \"value\": \"gene\", \"label\": \"Type.Flag\", \"description\": \"perturbagen level data (shRNA) or gene level scores (gene)\"}, {\"name\": \"quality\", \"value\": \"normal\", \"label\": \"Quality\", \"description\": \"controls how many permutations are run; draft will take less time\"}, {\"name\": \"direction\", \"value\": \"positive\", \"label\": \"Direction\", \"description\": \"Direction of feature matching; positive means high values to high values, low to low; negative means high to low values\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"PCA\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Principal Component Analysis (adapted from MeV, http://www.tm4.org/m)\", \"required_args\": [{\"name\": \"input.filename\", \"value\": \"\", \"label\": \"Input.Filename\", \"description\": \"input file - .gct, .res, .odf\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"PCA\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"cluster.by\", \"value\": \"3\", \"label\": \"Cluster.By\", \"description\": \"Cluster by rows or columns\"}, {\"name\": \"output.file\", \"value\": \"<input.filename_basename>\", \"label\": \"Output.File\", \"description\": \"stub name of the output files\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"PCAViewer\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Principal Component Analysis Visualizer (adapted from MeV, http://www.tigr.org/software/tm4/mev.html)\", \"required_args\": [{\"name\": \"dataset.file\", \"value\": \"\", \"label\": \"Dataset.File\", \"description\": \"The dataset - .res, .gct\"}, {\"name\": \"s.matrix.file\", \"value\": \"\", \"label\": \"S.Matrix.File\", \"description\": \"The s matrix - .odf\"}, {\"name\": \"t.matrix.file\", \"value\": \"\", \"label\": \"T.Matrix.File\", \"description\": \"The t matrix - .odf\"}, {\"name\": \"u.matrix.file\", \"value\": \"\", \"label\": \"U.Matrix.File\", \"description\": \"The u matrix - .odf\"}, {\"name\": \"cls.or.sample.info.file\", \"value\": \"\", \"label\": \"Cls.Or.Sample.Info.File\", \"description\": \"A cls or sample info file - .cls, .txt\"}, {\"name\": \"sample.info.class\", \"value\": \"\", \"label\": \"Sample.Info.Class\", \"description\": \"The name of the class column to use in the sample info file (required if sample info file provided)\"}, {\"name\": \"featureset.file\", \"value\": \"\", \"label\": \"Featureset.File\", \"description\": \"The feature set file - .gmt, .gmx, .grp\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"PCAViewer\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"PclToGct\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Convert Stanford preclustering file (.pcl) to .gct\", \"required_args\": [{\"name\": \"input.filename\", \"value\": \"\", \"label\": \"Input.Filename\", \"description\": \"pcl file to convert\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"PclToGct\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.filename\", \"value\": \"<input.filename_basename>.gct\", \"label\": \"Output.Filename\", \"description\": \"output gct file\"}, {\"name\": \"weight.handling\", \"value\": \"3\", \"label\": \"Weight.Handling\", \"description\": \"How to handle manually edited weights (weights not equal to 1) present in PCL file\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"PeakMatch\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Perform peak matching on LC-MS data\", \"required_args\": [{\"name\": \"peakList.filename\", \"value\": \"\", \"label\": \"PeakList.Filename\", \"description\": \"Zip file containing peak lists\"}, {\"name\": \"sampleInfo.filename\", \"value\": \"\", \"label\": \"SampleInfo.Filename\", \"description\": \"CSV file with header: experiment, sample, class\"}, {\"name\": \"LandmarkMatchOutput.filename\", \"value\": \"\", \"label\": \"LandmarkMatchOutput.Filename\", \"description\": \"Zip file containing the output from landmark matching.\"}, {\"name\": \"MZ.tolerance\", \"value\": \"\", \"label\": \"MZ.Tolerance\", \"description\": \"m/z tolerance (ppm). Used if landmark match output is not supplied. Defaults to 10 ppm.\"}, {\"name\": \"RT.tolerance\", \"value\": \"\", \"label\": \"RT.Tolerance\", \"description\": \"retention time tolerance (min). Used if landmark match is not supplied. Defaults to 2 min.\"}, {\"name\": \"numberProcesses\", \"value\": \"\", \"label\": \"NumberProcesses\", \"description\": \"Number of processess when running in parallel. Defaults to 1 (sequential).\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"PeakMatch\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"outputName\", \"value\": \"PeakMatchOutput\", \"label\": \"OutputName\", \"description\": \"File name prefix for the output files\"}, {\"name\": \"running.on.LSF\", \"value\": \"no\", \"label\": \"Running.On.LSF\", \"description\": \"Whether you are running on an LSF cluster version of GenePattern\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"Peaks\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Determine peaks in the spectrum using a series of digital filters.\", \"required_args\": [{\"name\": \"spectrum.filename\", \"value\": \"\", \"label\": \"Spectrum.Filename\", \"description\": \"spectrum data file - .csv\"}, {\"name\": \"low.Da\", \"value\": \"\", \"label\": \"Low.Da\", \"description\": \"Minimum M/Z to include\"}, {\"name\": \"high.Da\", \"value\": \"\", \"label\": \"High.Da\", \"description\": \"Maximum M/Z to include\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"Peaks\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"percentile\", \"value\": \"0.65\", \"label\": \"Percentile\", \"description\": \"After applying filters, threshold above which peaks are located\"}, {\"name\": \"smoothing.size\", \"value\": \"21\", \"label\": \"Smoothing.Size\", \"description\": \"Size of the smoothing filter\"}, {\"name\": \"adaptive.background.correction.strength\", \"value\": \"0.75\", \"label\": \"Adaptive.Background.Correction.Strength\", \"description\": \"Strength of the adaptive background correction filter\"}, {\"name\": \"adaptive.background.correction.size\", \"value\": \"21\", \"label\": \"Adaptive.Background.Correction.Size\", \"description\": \"Size of the adaptive background correction filter\"}, {\"name\": \"high.pass.filter.strength\", \"value\": \"10\", \"label\": \"High.Pass.Filter.Strength\", \"description\": \"Strength of the high pass filter\"}, {\"name\": \"high.pass.filter.factor\", \"value\": \"5\", \"label\": \"High.Pass.Filter.Factor\", \"description\": \"Filter factor for the high pass filter\"}, {\"name\": \"output.file\", \"value\": \"<spectrum.filename_basename>.peaks.odf\", \"label\": \"Output.File\", \"description\": \"output file name\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"Picard.AddOrReplaceReadGroups\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Replaces all read groups in the input file with a new read group and assigns all reads to this read group in the output\", \"required_args\": [{\"name\": \"input.file\", \"value\": \"\", \"label\": \"Input.File\", \"description\": \"Input file (BAM or SAM).\"}, {\"name\": \"read.group.id\", \"value\": \"\", \"label\": \"Read.Group.Id\", \"description\": \"Read Group ID\"}, {\"name\": \"read.group.library\", \"value\": \"\", \"label\": \"Read.Group.Library\", \"description\": \"DNA preparation library. Note that this field is required by the module Picard.MarkDuplicates. It is used to determine which read groups may contain molecular duplicates, in the case where the same DNA library was sequenced on multiple lanes.\"}, {\"name\": \"read.group.platform\", \"value\": \"\", \"label\": \"Read.Group.Platform\", \"description\": \"Read Group platform (e.g., Illumina, SOLID).\"}, {\"name\": \"read.group.platform.unit\", \"value\": \"\", \"label\": \"Read.Group.Platform.Unit\", \"description\": \"Read Group platform unit (e.g., run barcode).\"}, {\"name\": \"read.group.sample.name\", \"value\": \"\", \"label\": \"Read.Group.Sample.Name\", \"description\": \"Read Group sample name.\"}, {\"name\": \"read.group.sequencing.center.name\", \"value\": \"\", \"label\": \"Read.Group.Sequencing.Center.Name\", \"description\": \"Read Group sequencing center name.\"}, {\"name\": \"read.group.description\", \"value\": \"\", \"label\": \"Read.Group.Description\", \"description\": \"Read Group description.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"Picard.AddOrReplaceReadGroups\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.prefix\", \"value\": \"<input.file_basename>\", \"label\": \"Output.Prefix\", \"description\": \"A label used to name the output file\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"Picard.BamToSam\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Converts a BAM file to a SAM file\", \"required_args\": [{\"name\": \"input.file\", \"value\": \"\", \"label\": \"Input.File\", \"description\": \"A BAM file\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"Picard.BamToSam\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.prefix\", \"value\": \"<input.file_basename>\", \"label\": \"Output.Prefix\", \"description\": \"A label used to name the output file\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"Picard.CreateSequenceDictionary\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Reads FASTA or FASTA.GZ files containing reference sequences, and writes them as a SAM file containing a sequence dictionary\", \"required_args\": [{\"name\": \"reference.sequence.file\", \"value\": \"\", \"label\": \"Reference.Sequence.File\", \"description\": \"Input reference fasta or fasta.gz \"}, {\"name\": \"genome.assembly\", \"value\": \"\", \"label\": \"Genome.Assembly\", \"description\": \"The genome assembly to put into AS field of sequence dictionary entry\"}, {\"name\": \"uri\", \"value\": \"\", \"label\": \"Uri\", \"description\": \"The uri to put into UR field of sequence dictionary entry\"}, {\"name\": \"num.sequences\", \"value\": \"\", \"label\": \"Num.Sequences\", \"description\": \"Stop after writing this many sequences\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"Picard.CreateSequenceDictionary\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"truncate.names.at.white.space\", \"value\": \"no\", \"label\": \"Truncate.Names.At.White.Space\", \"description\": \"Make sequence name the first word from the > line in the reference file\"}, {\"name\": \"output.file\", \"value\": \"<reference.sequence.file_basename>.dict\", \"label\": \"Output.File\", \"description\": \"The name of the output file\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"Picard.FastqToSam\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Extracts read sequences and qualities from the input fastq file and writes them into the output file in unaligned SAM or BAM format\", \"required_args\": [{\"name\": \"fastq.file\", \"value\": \"\", \"label\": \"Fastq.File\", \"description\": \"Input fastq file (optionally gzipped) for single end data, or first read in paired end data\"}, {\"name\": \"fastq.file.2\", \"value\": \"\", \"label\": \"Fastq.File.2\", \"description\": \"The fastq file (optionally gzipped) for second read in paired end data\"}, {\"name\": \"read.group.name\", \"value\": \"\", \"label\": \"Read.Group.Name\", \"description\": \"Read group name\"}, {\"name\": \"sample.name\", \"value\": \"\", \"label\": \"Sample.Name\", \"description\": \"Sample name to insert into the read group header\"}, {\"name\": \"library.name\", \"value\": \"\", \"label\": \"Library.Name\", \"description\": \"The library name to insert into the LB attribute in the read group header\"}, {\"name\": \"platform.unit\", \"value\": \"\", \"label\": \"Platform.Unit\", \"description\": \"The platform unit (often run_barcode.lane) to insert into the read group header\"}, {\"name\": \"platform\", \"value\": \"\", \"label\": \"Platform\", \"description\": \"The platform type (e.g., Illumina, SOLiD) to insert into the read group header\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"Picard.FastqToSam\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"quality.format\", \"value\": \"Standard\", \"label\": \"Quality.Format\", \"description\": \"A value describing how the quality values are encoded in the fastq\"}, {\"name\": \"sort.order\", \"value\": \"coordinate\", \"label\": \"Sort.Order\", \"description\": \"\"}, {\"name\": \"output.format\", \"value\": \"bam\", \"label\": \"Output.Format\", \"description\": \"Whether the file output format should be SAM or BAM\"}, {\"name\": \"output.prefix\", \"value\": \"<fastq.file_basename>\", \"label\": \"Output.Prefix\", \"description\": \"The prefix to use for the output file name\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"Picard.MarkDuplicates\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Examines aligned records in the supplied SAM or BAM file to locate duplicate reads.\", \"required_args\": [{\"name\": \"input.file\", \"value\": \"\", \"label\": \"Input.File\", \"description\": \"A SAM or BAM file\"}, {\"name\": \"read.name.regex\", \"value\": \"\", \"label\": \"Read.Name.Regex\", \"description\": \"Regular expression that can be used to parse read names in the input file\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"Picard.MarkDuplicates\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"remove.duplicates\", \"value\": \"false\", \"label\": \"Remove.Duplicates\", \"description\": \"Whether to remove duplicates\"}, {\"name\": \"max.file.handles\", \"value\": \"8000\", \"label\": \"Max.File.Handles\", \"description\": \"Maximum number of file handles to keep open when spilling read ends to disk\"}, {\"name\": \"optical.duplicate.distance\", \"value\": \"100\", \"label\": \"Optical.Duplicate.Distance\", \"description\": \"The maximum offset between two duplicate clusters in order to consider them optical duplicates. This should usually be set to some fairly small number (e.g. 5-10 pixels) unless using later versions of the Illumina pipeline that multiply pixel values by 10, in which case 50-100 is more normal\"}, {\"name\": \"sorting.collection.size.ratio\", \"value\": \"0.25\", \"label\": \"Sorting.Collection.Size.Ratio\", \"description\": \"This number, plus the maximum RAM available to the JVM, determine the memory footprint used by some of the sorting collections.  If you are running out of memory, try reducing this number\"}, {\"name\": \"output.prefix\", \"value\": \"<input.file_basename>\", \"label\": \"Output.Prefix\", \"description\": \"The prefix to use for the output file name\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"Picard.ReorderSam\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Reorders a SAM file or a BAM file to match contig ordering in a provided reference file\", \"required_args\": [{\"name\": \"input.file\", \"value\": \"\", \"label\": \"Input.File\", \"description\": \"A SAM or BAM file.\"}, {\"name\": \"reference.file\", \"value\": \"\", \"label\": \"Reference.File\", \"description\": \"Reference sequence to which to reorder reads in FASTA format.\"}, {\"name\": \"reference.sequence.dictionary\", \"value\": \"\", \"label\": \"Reference.Sequence.Dictionary\", \"description\": \"The sequence dictionary for the reference sequence file\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"Picard.ReorderSam\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"allow.partial.overlap\", \"value\": \"false\", \"label\": \"Allow.Partial.Overlap\", \"description\": \"If yes, then allows only a partial overlap of the BAM contigs with the new reference sequence contigs. This requires a corresponding contig in the new reference for each read contig.\"}, {\"name\": \"allow.contig.length.discordance\", \"value\": \"false\", \"label\": \"Allow.Contig.Length.Discordance\", \"description\": \"If yes, then permits mapping from a read contig to a new reference contig with the same name but a different length. Use with extreme caution and a full knowledge of the possible consequences.\"}, {\"name\": \"output.prefix\", \"value\": \"<input.file_basename>\", \"label\": \"Output.Prefix\", \"description\": \"The prefix of the output SAM or BAM file.\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"Picard.SamToBam\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Converts a SAM file to a BAM file\", \"required_args\": [{\"name\": \"input.file\", \"value\": \"\", \"label\": \"Input.File\", \"description\": \"A SAM file\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"Picard.SamToBam\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.prefix\", \"value\": \"<input.file_basename>\", \"label\": \"Output.Prefix\", \"description\": \"A label used to name the output file\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"Picard.SamToFastq\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Convert SAM or BAM files to FASTQ format\", \"required_args\": [{\"name\": \"input.file\", \"value\": \"\", \"label\": \"Input.File\", \"description\": \"The SAM or BAM file\"}, {\"name\": \"clipping.attribute\", \"value\": \"\", \"label\": \"Clipping.Attribute\", \"description\": \"Attribute that stores the position at which the SAM record should be clipped\"}, {\"name\": \"clipping.action.quality.value\", \"value\": \"\", \"label\": \"Clipping.Action.Quality.Value\", \"description\": \"Value to set base qualities to when clipping action is \\\"change base quality values\\\"\"}, {\"name\": \"fastq.output.prefix.1\", \"value\": \"\", \"label\": \"Fastq.Output.Prefix.1\", \"description\": \"The fastq output file name for single-end fastq or first end of a paired fastq. Only specify when \\\"per read group\\\" is \\\"no\\\".\"}, {\"name\": \"fastq.output.prefix.2\", \"value\": \"\", \"label\": \"Fastq.Output.Prefix.2\", \"description\": \"The second end of a paired fastq. Only specify when \\\"per read group\\\" is \\\"no\\\".\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"Picard.SamToFastq\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"per.read.group\", \"value\": \"no\", \"label\": \"Per.Read.Group\", \"description\": \"Whether to output a fastq file per read group\"}, {\"name\": \"re-reverse.bases\", \"value\": \"no\", \"label\": \"Re-Reverse.Bases\", \"description\": \"Whether to re-reverse bases and qualities of reads with negative strand flag set\"}, {\"name\": \"include.non.pf.reads\", \"value\": \"no\", \"label\": \"Include.Non.Pf.Reads\", \"description\": \"Whether to include non-PF reads\"}, {\"name\": \"clipping.action\", \"value\": \"none\", \"label\": \"Clipping.Action\", \"description\": \"Action to take with clipped reads when a clipping attribute is specified\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"Picard.SortSam\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Sorts a SAM or BAM file in a specified order\", \"required_args\": [{\"name\": \"input.file\", \"value\": \"\", \"label\": \"Input.File\", \"description\": \"Input file in SAM or BAM format\"}, {\"name\": \"output.format\", \"value\": \"\", \"label\": \"Output.Format\", \"description\": \"Whether the file output format should be SAM or BAM\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"Picard.SortSam\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"sort.order\", \"value\": \"coordinate\", \"label\": \"Sort.Order\", \"description\": \"Sort order\"}, {\"name\": \"output.prefix\", \"value\": \"<input.file_basename>\", \"label\": \"Output.Prefix\", \"description\": \"prefix for the name of output file\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"PlateQAFCS\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Look for consistency between wells of a single plate by computing summary statistics.\", \"required_args\": [{\"name\": \"Input.FCS.files\", \"value\": \"\", \"label\": \"Input.FCS.Files\", \"description\": \"FCS files from a single plate.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"PlateQAFCS\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"Check.cell.counts\", \"value\": \"1\", \"label\": \"Check.Cell.Counts\", \"description\": \"Count cells in each file?\"}, {\"name\": \"Check.medians\", \"value\": \"1\", \"label\": \"Check.Medians\", \"description\": \"Report the median measured values?\"}, {\"name\": \"Check.PDF\", \"value\": \"1\", \"label\": \"Check.PDF\", \"description\": \"Check the Probability Density Functions (PDF)?\"}, {\"name\": \"Check.ECDF\", \"value\": \"1\", \"label\": \"Check.ECDF\", \"description\": \"Check the Empirical Cumulative Distribution Functions (ECDF)?\"}, {\"name\": \"Check.boxplots\", \"value\": \"1\", \"label\": \"Check.Boxplots\", \"description\": \"Detect unexpected data distributions (unusual upper/lower quantiles)?\"}, {\"name\": \"Check.timeline\", \"value\": \"1\", \"label\": \"Check.Timeline\", \"description\": \"Check for unusual patterns in the acquisition of fluorescence and light scatter measurements over time?\"}, {\"name\": \"Check.timeflow\", \"value\": \"1\", \"label\": \"Check.Timeflow\", \"description\": \"Check for disturbances in the flow of cells over time?\"}, {\"name\": \"FSC.channel\", \"value\": \"FSC-H\", \"label\": \"FSC.Channel\", \"description\": \"The name of the forward scatter parameter (short name, value of the $PnN keyword).\"}, {\"name\": \"SSC.channel\", \"value\": \"SSC-H\", \"label\": \"SSC.Channel\", \"description\": \"The name of the side scatter parameter (short name, value of the $PnN keyword).\"}, {\"name\": \"Zip.output\", \"value\": \"TRUE\", \"label\": \"Zip.Output\", \"description\": \"Zip the resulting report together?\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"PlotPeaks\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Plot peaks\", \"required_args\": [{\"name\": \"peaks.output.filename\", \"value\": \"\", \"label\": \"Peaks.Output.Filename\", \"description\": \"The output from the Peaks module\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"PlotPeaks\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.file\", \"value\": \"<peaks.output.filename_basename>.pdf\", \"label\": \"Output.File\", \"description\": \"output file name\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"PredictionResultsViewer\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Visualizes prediction results\", \"required_args\": [{\"name\": \"prediction.results.file\", \"value\": \"\", \"label\": \"Prediction.Results.File\", \"description\": \"Prediction results - .odf\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"PredictionResultsViewer\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"Preprocess.and.quality.check.RNASeq.data\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"This pipeline is used in the GenomeSpace recipe, \\\"Preprocess and quality check RNA-Seq data\\\". The recipe can be found at: http://www.genomespace.org/support/guides/recipes/sections/preprocess-and-qc-rna-seq-data.\", \"required_args\": [{\"name\": \"ConvertLineEndings1.input.filename\", \"value\": \"\", \"label\": \"ConvertLineEndings1.Input.Filename\", \"description\": \"The input file (any non-binary file format)\"}, {\"name\": \"TopHat4.GTF.file\", \"value\": \"\", \"label\": \"TopHat4.GTF.File\", \"description\": \"A GTF file (v. 2.2 or higher) or GFF3 file containing a list of gene model annotations. TopHat will first extract the transcript sequences and align them to this virtual transcriptome first. Only the reads that do not fully map to the transcriptome will then be mapped on the genome. The reads that did map on the transcriptome will be converted to genomic mappings (spliced as needed) and merged with the novel mappings.\"}, {\"name\": \"TopHat4.reads.pair.1\", \"value\": \"\", \"label\": \"TopHat4.Reads.Pair.1\", \"description\": \"Unpaired reads file or first mate for paired reads. One or more files containing reads in FASTA or FASTQ format (bz2 and gz compressed files are supported).\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"Preprocess.and.quality.check.RNASeq.data\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"PreprocessDataset\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Performs several preprocessing steps on a res, gct, or odf input file\", \"required_args\": [{\"name\": \"input.filename\", \"value\": \"\", \"label\": \"Input.Filename\", \"description\": \"Input filename - .res, .gct, .odf\"}, {\"name\": \"threshold.for.removing.rows\", \"value\": \"\", \"label\": \"Threshold.For.Removing.Rows\", \"description\": \"Threshold for removing rows\"}, {\"name\": \"number.of.columns.above.threshold\", \"value\": \"\", \"label\": \"Number.Of.Columns.Above.Threshold\", \"description\": \"Remove row if this number of columns not >= given threshold\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"PreprocessDataset\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"threshold.and.filter\", \"value\": \"1\", \"label\": \"Threshold.And.Filter\", \"description\": \"Apply floor, ceiling and variation filter\"}, {\"name\": \"floor\", \"value\": \"20\", \"label\": \"Floor\", \"description\": \"Value for floor threshold\"}, {\"name\": \"ceiling\", \"value\": \"20000\", \"label\": \"Ceiling\", \"description\": \"Value for ceiling threshold\"}, {\"name\": \"min.fold.change\", \"value\": \"3\", \"label\": \"Min.Fold.Change\", \"description\": \"Minimum fold change for variation filter\"}, {\"name\": \"min.delta\", \"value\": \"100\", \"label\": \"Min.Delta\", \"description\": \"Minimum delta for variation filter\"}, {\"name\": \"num.outliers.to.exclude\", \"value\": \"0\", \"label\": \"Num.Outliers.To.Exclude\", \"description\": \"Number of outliers per row to ignore when calculating row min and max for variation filter\"}, {\"name\": \"row.normalization\", \"value\": \"0\", \"label\": \"Row.Normalization\", \"description\": \"Perform row normalization\"}, {\"name\": \"row.sampling.rate\", \"value\": \"1\", \"label\": \"Row.Sampling.Rate\", \"description\": \"Sample rows without replacement to obtain this fraction of the total number of rows\"}, {\"name\": \"log2.transform\", \"value\": \"0\", \"label\": \"Log2.Transform\", \"description\": \"Apply log2 transform after all other preprocessing steps\"}, {\"name\": \"output.file.format\", \"value\": \"3\", \"label\": \"Output.File.Format\", \"description\": \"Output file format\"}, {\"name\": \"output.file\", \"value\": \"<input.filename_basename>.preprocessed\", \"label\": \"Output.File\", \"description\": \"Output file name\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"PreprocessReadCounts\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Preprocess RNA-Seq count data in a GCT file so that it is suitable for use in GenePattern analyses.\", \"required_args\": [{\"name\": \"input.file\", \"value\": \"\", \"label\": \"Input.File\", \"description\": \"A GCT file containing raw RNA-Seq counts, such as is produced by MergeHTSeqCounts\"}, {\"name\": \"cls.file\", \"value\": \"\", \"label\": \"Cls.File\", \"description\": \"A categorical CLS file specifying the phenotype classes for the samples in the GCT file.  \"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"PreprocessReadCounts\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.file\", \"value\": \"<input.file_basename>.preprocessed.gct\", \"label\": \"Output.File\", \"description\": \"Output file name\"}, {\"name\": \"expression.value.filter.threshold\", \"value\": \"1\", \"label\": \"Expression.Value.Filter.Threshold\", \"description\": \"Threshold to use when filtering CPM expression values; rows are kept only if the values (in CPM) for all columns are greater than this threshold\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"PreviewFCS\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Allows viewing of structural metadata, parameters, and descriptive statistics from a Flow Cytometry Standard (FCS) data file\", \"required_args\": [{\"name\": \"Input.FCS.data.file\", \"value\": \"\", \"label\": \"Input.FCS.Data.File\", \"description\": \"An input FCS data file to be described.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"PreviewFCS\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"Output.file.format\", \"value\": \"HTML\", \"label\": \"Output.File.Format\", \"description\": \"Output file format, either HTML or XML, default HTML.\"}, {\"name\": \"Output.file.name\", \"value\": \"<Input.FCS.data.file_basename>\", \"label\": \"Output.File.Name\", \"description\": \"The output file name (optional).\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"Profile.Plots\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Create profile plots for samples\", \"required_args\": [{\"name\": \"input.file\", \"value\": \"\", \"label\": \"Input.File\", \"description\": \"Input data file (in gct format)\"}, {\"name\": \"cls.file\", \"value\": \"\", \"label\": \"Cls.File\", \"description\": \"File indicating type of samples (sample group) in input.file. Profiles for each group will be plotted in a different color\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"Profile.Plots\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.file\", \"value\": \"<input.file_basename>-profile-plots.pdf\", \"label\": \"Output.File\", \"description\": \"Output file name\"}, {\"name\": \"data.type\", \"value\": \"proteome\", \"label\": \"Data.Type\", \"description\": \"Indicate type of input data (proteome, phosphoprotome, etc.) for use in plots\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"Protein.mRNA.Corrlation\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Tabulate and plot correlation between proteome and mRNA\", \"required_args\": [{\"name\": \"proteome.file\", \"value\": \"\", \"label\": \"Proteome.File\", \"description\": \"Proteome data file in gct format\"}, {\"name\": \"mRNA.file\", \"value\": \"\", \"label\": \"MRNA.File\", \"description\": \"mRNA (RNA-seq) data file in gct.format\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"Protein.mRNA.Corrlation\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.prefix\", \"value\": \"<proteome.file_basename>\", \"label\": \"Output.Prefix\", \"description\": \"Filename prefix for results files\"}, {\"name\": \"fdr.threshold\", \"value\": \"0.01\", \"label\": \"Fdr.Threshold\", \"description\": \"FDR threshold for marking statistically significant correlations\"}, {\"name\": \"correlation.type\", \"value\": \"pearson\", \"label\": \"Correlation.Type\", \"description\": \"Correlation type to use for calculating protein-mRNA correlation\"}, {\"name\": \"data.type\", \"value\": \"proteome\", \"label\": \"Data.Type\", \"description\": \"Indicate type of input data (proteome, phosphoprotome, etc.) for use in plots\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"ProteinDatasetCreation\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Extract features from protein .FASTA files for use with standard prediction algorithms\", \"required_args\": [{\"name\": \"input.file\", \"value\": \"\", \"label\": \"Input.File\", \"description\": \"The sequence information in .FASTA format\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"ProteinDatasetCreation\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"feature.set.1\", \"value\": \"1\", \"label\": \"Feature.Set.1\", \"description\": \"Use amino acid sequence length (1 feature)\"}, {\"name\": \"feature.set.2\", \"value\": \"0\", \"label\": \"Feature.Set.2\", \"description\": \"Use amino acid composition (20 features)\"}, {\"name\": \"feature.set.3\", \"value\": \"0\", \"label\": \"Feature.Set.3\", \"description\": \"Use reduced alphabet composition of the n terminal region (19 features)\"}, {\"name\": \"feature.set.4\", \"value\": \"0\", \"label\": \"Feature.Set.4\", \"description\": \"Use reduced alphabet composition of the c terminal region (19 features)\"}, {\"name\": \"feature.set.5\", \"value\": \"0\", \"label\": \"Feature.Set.5\", \"description\": \"Use the reduced alphabet for the gap pairs of amino acids (165 features)\"}, {\"name\": \"output.file\", \"value\": \"<input.file_basename>\", \"label\": \"Output.File\", \"description\": \"The output files base name\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"ProteoArray\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"LC-MS proteomic data processing module\", \"required_args\": [{\"name\": \"input.zip.filename\", \"value\": \"\", \"label\": \"Input.Zip.Filename\", \"description\": \"zip file of LC-MS runs in mzXML format.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"ProteoArray\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.file\", \"value\": \"<input.zip.filename_basename>.prot\", \"label\": \"Output.File\", \"description\": \"output filename\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"ProteomicsAnalysis\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Runs the proteomics analysis on the set of input spectra\", \"required_args\": [{\"name\": \"zip.filename\", \"value\": \"\", \"label\": \"Zip.Filename\", \"description\": \"Zip file containing spectra data. Spectra data are stored as M/Z,Intensity\"}, {\"name\": \"peak.list.filename\", \"value\": \"\", \"label\": \"Peak.List.Filename\", \"description\": \"file containing one M/Z per line when peaks=input\"}, {\"name\": \"random.seed\", \"value\": \"\", \"label\": \"Random.Seed\", \"description\": \"random seed when peaks=random\"}, {\"name\": \"random.n.peaks\", \"value\": \"\", \"label\": \"Random.N.Peaks\", \"description\": \"number of random M/Z's selected when peaks=random\"}, {\"name\": \"low.Da\", \"value\": \"\", \"label\": \"Low.Da\", \"description\": \"Minimum M/Z to include\"}, {\"name\": \"high.Da\", \"value\": \"\", \"label\": \"High.Da\", \"description\": \"Maximum M/Z to include\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"ProteomicsAnalysis\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.file\", \"value\": \"<zip.filename_basename>\", \"label\": \"Output.File\", \"description\": \"The base name for output files\"}, {\"name\": \"quality.threshold\", \"value\": \"0.45\", \"label\": \"Quality.Threshold\", \"description\": \"discard spectra with quality < quality threshold\"}, {\"name\": \"fill.na\", \"value\": \"yes\", \"label\": \"Fill.Na\", \"description\": \"whether to fill missing peaks\"}, {\"name\": \"normalize\", \"value\": \"tan\", \"label\": \"Normalize\", \"description\": \"normalization strategy\"}, {\"name\": \"peak.detection.method\", \"value\": \"detect\", \"label\": \"Peak.Detection.Method\", \"description\": \"peak detection method\"}, {\"name\": \"filter.peaks\", \"value\": \"yes\", \"label\": \"Filter.Peaks\", \"description\": \"filter peaks based on peak intensity / noise \"}, {\"name\": \"filter.peaks.factor\", \"value\": \"10\", \"label\": \"Filter.Peaks.Factor\", \"description\": \"retains peaks if peak intensity >= filter.peaks.factor * stdev(noise)\"}, {\"name\": \"percentile\", \"value\": \"0.65\", \"label\": \"Percentile\", \"description\": \"After applying filters, threshold above which peaks are located\"}, {\"name\": \"smoothing.size\", \"value\": \"21\", \"label\": \"Smoothing.Size\", \"description\": \"Size of the smoothing filter\"}, {\"name\": \"adaptive.background.correction.strength\", \"value\": \"0.75\", \"label\": \"Adaptive.Background.Correction.Strength\", \"description\": \"Strength of the adaptive background correction filter\"}, {\"name\": \"adaptive.background.correction.size\", \"value\": \"21\", \"label\": \"Adaptive.Background.Correction.Size\", \"description\": \"Size of the adaptive background correction filter\"}, {\"name\": \"high.pass.filter.strength\", \"value\": \"10\", \"label\": \"High.Pass.Filter.Strength\", \"description\": \"Strength of the high pass filter\"}, {\"name\": \"high.pass.filter.factor\", \"value\": \"5\", \"label\": \"High.Pass.Filter.Factor\", \"description\": \"Filter factor for the high pass filter\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"QuaSAR\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"A suite of tools for QC, analysis and visualization of data from MRM-MS experiments including assay configuration, calibration curves and new sample analysis.\", \"required_args\": [{\"name\": \"skyline.file\", \"value\": \"\", \"label\": \"Skyline.File\", \"description\": \"This is a Skyline export in csv file format. The file has to include the following 10 columns with the exact column names given here (in no particular order): SampleName, PeptideSequence, ProteinName, ReplicateName, PrecursorCharge, ProductCharge, FragmentIon,  AverageMeasuredRetentionTime, light Area, heavy Area. The file may contain many more columns, but it has to include at least these 10 columns\"}, {\"name\": \"concentration.file\", \"value\": \"\", \"label\": \"Concentration.File\", \"description\": \"Concentration values given in csv file format. The file has to include the following 5 columns with the exact column names given here (in no particular order): SampleName, SampleID, Analyte Concentration, IS Conc, Concentration Ratio. Notice the single space between Analyte Concentration, IS Conc and Concentration Ratio. The space has to be maintained.\"}, {\"name\": \"title\", \"value\": \"\", \"label\": \"Title\", \"description\": \"The title is displayed on each calibration plot\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"QuaSAR\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"analyte\", \"value\": \"light Area\", \"label\": \"Analyte\", \"description\": \"User must specify the exact analyte column header name as it is appears in the input CSV file. The default is \\\"light Area\\\"\"}, {\"name\": \"standard\", \"value\": \"heavy Area\", \"label\": \"Standard\", \"description\": \"User must specify the exact internal standard column header name as it is appears in the input CSV file. The default is \\\"heavy Area\\\"\"}, {\"name\": \"generate.cv.table\", \"value\": \"1\", \"label\": \"Generate.Cv.Table\", \"description\": \"If \\\"yes\\\" then generate CV (coefficient of variation) table\"}, {\"name\": \"generate.calcurves\", \"value\": \"1\", \"label\": \"Generate.Calcurves\", \"description\": \"If \\\"yes\\\" then generate calibration curves\"}, {\"name\": \"generate.lodloq.table\", \"value\": \"1\", \"label\": \"Generate.Lodloq.Table\", \"description\": \"If \\\"yes\\\" then generate LOD/LOQ table\"}, {\"name\": \"generate.peak.area.plots\", \"value\": \"0\", \"label\": \"Generate.Peak.Area.Plots\", \"description\": \"If \\\"yes\\\" then generate peak area plots\"}, {\"name\": \"use.par\", \"value\": \"0\", \"label\": \"Use.Par\", \"description\": \"If \\\"yes\\\" then use peak area ratio (PAR) for analysis (instead of concentration)\"}, {\"name\": \"max.calcurve.linear.scale\", \"value\": \"150\", \"label\": \"Max.Calcurve.Linear.Scale\", \"description\": \"The maximum value for linear scale in fmols/ul\"}, {\"name\": \"max.calcurve.log.scale\", \"value\": \"150\", \"label\": \"Max.Calcurve.Log.Scale\", \"description\": \"The maximum value for log scale in fmols/ul\"}, {\"name\": \"perform.audit\", \"value\": \"1\", \"label\": \"Perform.Audit\", \"description\": \"If \\\"yes\\\" then perform AuDIT for interference detection.\"}, {\"name\": \"audit.cv.threshold\", \"value\": \"0.2\", \"label\": \"Audit.Cv.Threshold\", \"description\": \"For AuDIT the threshold for coefficient of variation below which transition is quantification-worthy. Default is 0.2.\"}, {\"name\": \"output.prefix\", \"value\": \"<skyline.file_basename>\", \"label\": \"Output.Prefix\", \"description\": \"Prefix used for all output file names\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"RankNormalize\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Normalize the rows in a data set by rank\", \"required_args\": [{\"name\": \"input.file\", \"value\": \"\", \"label\": \"Input.File\", \"description\": \"The dataset to be normalized\"}, {\"name\": \"scale.to.value\", \"value\": \"\", \"label\": \"Scale.To.Value\", \"description\": \"Result values will be scaled to this value by multiplication after normalization.  Leave this blank to scale to the number of rows in the data set.\"}, {\"name\": \"threshold\", \"value\": \"\", \"label\": \"Threshold\", \"description\": \"Minimum threshold for values in the data set.  Any data set value below this will be increased to the threshold value before normalization.  If more than one of these parameters is specified, the order of precedence is threshold, then ceiling, then shift.\"}, {\"name\": \"ceiling\", \"value\": \"\", \"label\": \"Ceiling\", \"description\": \"Maximum ceiling for values in the data set.  Any data set value above this will be decreased to the ceiling value before normalization.  If more than one of these parameters is specified, the order of precedence is threshold, then ceiling, then shift.\"}, {\"name\": \"shift\", \"value\": \"\", \"label\": \"Shift\", \"description\": \"Shift all values in the data set.  The shift value will be added to all data set values before normalization.  If more than one of these parameters is specified, the order of precedence is threshold, then ceiling, then shift.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"RankNormalize\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.file.name\", \"value\": \"<input.file_basename>.NORM.<input.file_extension>\", \"label\": \"Output.File.Name\", \"description\": \"The name to be given to the output file.\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"Read_group_trackingToGct\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"[**Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] Converts a Cufflinks read_group_tracking file into GCT format\", \"required_args\": [{\"name\": \"input.file\", \"value\": \"\", \"label\": \"Input.File\", \"description\": \"The read_group_tracking file to be converted into GCT format.\"}, {\"name\": \"gtf.file\", \"value\": \"\", \"label\": \"Gtf.File\", \"description\": \"An optional GTF file.  If provided, this should be the same file that you provided to Cuffdiff. You can use this to map tracking IDs in the read_group_tracking file to gene symbols for output in the GCT.  Make sure that you set the feature level to match your read_group_tracking file.\"}, {\"name\": \"override.attribute.name.for.lookup\", \"value\": \"\", \"label\": \"Override.Attribute.Name.For.Lookup\", \"description\": \"Use this to override the name of the attribute to search against to look up IDs when mapping with a GTF (gene_id, for example). Type in the attribute name exactly as it appears in the GTF.  The retrieval override must also be provided if this parameter is set.\"}, {\"name\": \"override.attribute.name.for.retrieval\", \"value\": \"\", \"label\": \"Override.Attribute.Name.For.Retrieval\", \"description\": \"Use this to override the name of the attribute to retrieve on match when mapping with a GTF (gene_name, for example). Type in the attribute name exactly as it appears in the GTF.  The lookup override must also be provided if this parameter is set.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"Read_group_trackingToGct\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.file.name\", \"value\": \"<input.file_basename>.gct\", \"label\": \"Output.File.Name\", \"description\": \"The name to be given to the GCT output file.\"}, {\"name\": \"expression.value.column\", \"value\": \"3\", \"label\": \"Expression.Value.Column\", \"description\": \"The column to use for extracting expression values.\"}, {\"name\": \"screen.out.errors\", \"value\": \"false\", \"label\": \"Screen.Out.Errors\", \"description\": \"Set to 'yes' to exclude from the GCT file any feature having at least one non-OK quantification status across all of the sequenced samples.\"}, {\"name\": \"output.value.for.error\", \"value\": \"NA\", \"label\": \"Output.Value.For.Error\", \"description\": \"This parameter controls what expression value to write to a GCT file in those cases where a feature's expression estimate carries a non-OK quantification status.  The parameter is only relevant to your final GCT when 'screen out errors' is set to 'no'.  We strongly recommend treating these as missing values (select 'NA' or 'blank').\"}, {\"name\": \"feature.level.for.symbol.lookup\", \"value\": \"gene\", \"label\": \"Feature.Level.For.Symbol.Lookup\", \"description\": \"Select the feature level of the read_group_tracking file for mapping gene symbols from a GTF. This is ignored if no GTF is provided.\"}, {\"name\": \"choose.gct.column.for.mapping\", \"value\": \"auto\", \"label\": \"Choose.Gct.Column.For.Mapping\", \"description\": \"Use this to explicitly set the GCT column to use when writing symbols retrieved from the GCT. This is automatic by default, meaning that retrieved symbols go into the Name column for gene-level files and into the Description column for other feature levels.\"}, {\"name\": \"report.naming.conflict.details\", \"value\": \"false\", \"label\": \"Report.Naming.Conflict.Details\", \"description\": \"Include the naming conflict details in the summary report, rather than just giving counts of the issues. \"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"ReannotateGCT\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Reannotate a GCT file's \\\"Description\\\" column with data from a CHIP file\", \"required_args\": [{\"name\": \"gct.file\", \"value\": \"\", \"label\": \"Gct.File\", \"description\": \"A GCT file to be annotated\"}, {\"name\": \"chip.file\", \"value\": \"\", \"label\": \"Chip.File\", \"description\": \"CHIP file to use for annotations\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"ReannotateGCT\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.file\", \"value\": \"<gct.file_basename>_reannot.gct\", \"label\": \"Output.File\", \"description\": \"The name of your re-annotated GCT file. \"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"RemoveCopyNumberOutliers\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Reads rows of a CN file and removes outliers on a sample-by-sample basis using improvised method.\", \"required_args\": [{\"name\": \"input.file\", \"value\": \"\", \"label\": \"Input.File\", \"description\": \"A CN file (.cn, .RData, or .txt)\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"RemoveCopyNumberOutliers\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"num.probes\", \"value\": \"5\", \"label\": \"Num.Probes\", \"description\": \"The number of trailing and leading probes to use to compute the median\"}, {\"name\": \"outlier.probe.replacement\", \"value\": \"NA\", \"label\": \"Outlier.Probe.Replacement\", \"description\": \"The replacement value for the outlier probe\"}, {\"name\": \"multiplicative.threshold\", \"value\": \"4\", \"label\": \"Multiplicative.Threshold\", \"description\": \"Multiplicative threshold\"}, {\"name\": \"additive.threshold\", \"value\": \"0.3\", \"label\": \"Additive.Threshold\", \"description\": \"Additive threshold\"}, {\"name\": \"output.prefix\", \"value\": \"<input.file_basename>\", \"label\": \"Output.Prefix\", \"description\": \"A prefix for the output file\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"RemoveMissingValues\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Removes features and/or samples containing missing (NA) values\", \"required_args\": [{\"name\": \"input.file\", \"value\": \"\", \"label\": \"Input.File\", \"description\": \"Input file (gct, res)\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"RemoveMissingValues\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"remove.opt\", \"value\": \"1\", \"label\": \"Remove.Opt\", \"description\": \"Remove rows, columns, or both containing missing values\"}, {\"name\": \"output.filename\", \"value\": \"<input.file_basename>.na-rm.gct\", \"label\": \"Output.Filename\", \"description\": \"output file\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"removeSamples\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Remove samples in dataset (.gct), using a list of samples provided\", \"required_args\": [{\"name\": \"gct.file\", \"value\": \"\", \"label\": \"Gct.File\", \"description\": \"data file to filter\"}, {\"name\": \"removelist\", \"value\": \"\", \"label\": \"Removelist\", \"description\": \"list of samples to remove\"}, {\"name\": \"extension\", \"value\": \"\", \"label\": \"Extension\", \"description\": \"filename prefix\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"removeSamples\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"RemoveSaturatedFCSEvents\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Remove saturated events from an FCS data file\", \"required_args\": [{\"name\": \"Input.FCS.data.file\", \"value\": \"\", \"label\": \"Input.FCS.Data.File\", \"description\": \"An FCS data file to remove saturated events from.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"RemoveSaturatedFCSEvents\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"Output.FCS.file.name\", \"value\": \"<Input.FCS.data.file_basename>.without_saturated.fcs\", \"label\": \"Output.FCS.File.Name\", \"description\": \"The output FCS file name.\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"RenameFile\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Creates a result file with the contents of the input but with a new name.  Where possible, this is done without copying the file but instead by simply making a link.  While it is possible to run this with a submitted file, the common use will be to change the name of a result file or a previously uploaded file.  Note that the original file will still be present in its original location. \", \"required_args\": [{\"name\": \"input.file\", \"value\": \"\", \"label\": \"Input.File\", \"description\": \"Input file to be renamed\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"RenameFile\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.filename\", \"value\": \"<input.file_basename>.cp.<input.file_extension>\", \"label\": \"Output.Filename\", \"description\": \"The name to give the output file\"}, {\"name\": \"screen.filename\", \"value\": \"yes\", \"label\": \"Screen.Filename\", \"description\": \"Automatically screen out problematic characters in the filename\"}, {\"name\": \"force.copy\", \"value\": \"no\", \"label\": \"Force.Copy\", \"description\": \"Force a copy to be made rather than a link. \"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"ReorderByClass\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Reorder the samples in an expression dataset and class file by class\", \"required_args\": [{\"name\": \"expression.data.filename\", \"value\": \"\", \"label\": \"Expression.Data.Filename\", \"description\": \"Expression data - .res, .gct, .odf\"}, {\"name\": \"class.filename\", \"value\": \"\", \"label\": \"Class.Filename\", \"description\": \"Class file - .cls\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"ReorderByClass\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"expression.data.file.output.name\", \"value\": \"<expression.data.filename_basename>_reordered\", \"label\": \"Expression.Data.File.Output.Name\", \"description\": \"Name of expression data output file\"}, {\"name\": \"class.file.output.name\", \"value\": \"<class.filename_basename>_reordered\", \"label\": \"Class.File.Output.Name\", \"description\": \"Name of class output file\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"ReplicatesQC\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Removes cell line replicate samples that fail quality control check of cell line replicate correlations and distribution scores.\", \"required_args\": [{\"name\": \"GCT.files\", \"value\": \"\", \"label\": \"GCT.Files\", \"description\": \"Please enter at least one and up to three GCT files.\"}, {\"name\": \"extension\", \"value\": \"\", \"label\": \"Extension\", \"description\": \"\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"ReplicatesQC\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"REVEALER\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"REVEALER (Repeated Evaluation of VariablEs conditionAL Entropy and Redundancy) is an analysis method specifically suited to find groups of genomic alterations that match in a complementary way, a predefined functional activation, dependency of drug response target profile. The method starts by considering, if available, already known genomic alterations (seed) that are the known causes or are known or assumed associated with the target. REVEALER starts by finding the genomic alteration that best matches the target profile conditional to the known seed profile using the conditional mutual information. The newly discovered alteration is then merged with the seed to form a new summary feature, and then the process repeats itself finding additional complementary alterations that explain more and more of the target profile.\", \"required_args\": [], \"default_args\": [{\"name\": \"task_name\", \"value\": \"REVEALER\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"ds1\", \"value\": \"https://personal.broadinstitute.org/tamayo/REVEALER/datasets/CTNBB1_transcriptional_reporter.gct\", \"label\": \"Ds1\", \"description\": \"Target profile dataset (e.g. https://personal.broadinstitute.org/tamayo/REVEALER/datasets/CTNBB1_transcriptional_reporter.gct)\"}, {\"name\": \"target.name\", \"value\": \"BETA-CATENIN-REPORTER\", \"label\": \"Target.Name\", \"description\": \"Target profile name (e.g. BETA-CATENIN-REPORTER)\"}, {\"name\": \"target.match\", \"value\": \"positive\", \"label\": \"Target.Match\", \"description\": \"Direction of match (e.g. positive)\"}, {\"name\": \"ds2\", \"value\": \"https://personal.broadinstitute.org/tamayo/REVEALER/datasets/CCLE_MUT_CNA_AMP_DEL_0.70_2fold.MC.gct\", \"label\": \"Ds2\", \"description\": \"Features Dataset (e.g. https://personal.broadinstitute.org/tamayo/REVEALER/datasets/CCLE_MUT_CNA_AMP_DEL_0.70_2fold.MC.gct)\"}, {\"name\": \"seed.names\", \"value\": \"CTNNB1.MC_MUT\", \"label\": \"Seed.Names\", \"description\": \"seed(s) names (e.g. CTNNB1.MC_MUT)\"}, {\"name\": \"max.n.iter\", \"value\": \"2\", \"label\": \"Max.N.Iter\", \"description\": \"Maximum number of iterations (e.g. 2)\"}, {\"name\": \"n.markers\", \"value\": \"30\", \"label\": \"N.Markers\", \"description\": \"Number of top hits to show in heatmap for every iteration (e.g. 30)\"}, {\"name\": \"locs.table.file\", \"value\": \"https://personal.broadinstitute.org/tamayo/REVEALER/datasets/hgnc_complete_set.Feb_20_2014.v2.txt\", \"label\": \"Locs.Table.File\", \"description\": \"Table with chromosomal location for each gene symbol (e.g. https://personal.broadinstitute.org/tamayo/REVEALER/datasets/hgnc_complete_set.Feb_20_2014.v2.txt)\"}, {\"name\": \"count.thres.low\", \"value\": \"3\", \"label\": \"Count.Thres.Low\", \"description\": \"Filter out features with less than count.thres.low 1's (e.g. 3)\"}, {\"name\": \"count.thres.high\", \"value\": \"50\", \"label\": \"Count.Thres.High\", \"description\": \"Filter out features with more than count.thres.low 1's (e.g. 50)\"}, {\"name\": \"pdf.output.file\", \"value\": \"BCAT_vs_MUT_CNA.pdf\", \"label\": \"Pdf.Output.File\", \"description\": \"Output PDF file with results (e.g. BCAT_vs_MUT_CNA.pdf)\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"RLibDemo\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"\", \"required_args\": [{\"name\": \"message\", \"value\": \"\", \"label\": \"Message\", \"description\": \"\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"RLibDemo\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"RNASeQC\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"[**Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] Calculates metrics on aligned RNA-seq data.\", \"required_args\": [{\"name\": \"bam.files\", \"value\": \"\", \"label\": \"Bam.Files\", \"description\": \"An indexed BAM file or zipped set of indexed BAM files to be analyzed. If you are supplying a single BAM file, it should be located in the same folder as its associated index file (BAI). (If SortSam is used to index the BAM file, then the BAM and BAI files are located in the same folder.)\"}, {\"name\": \"sample.info.file\", \"value\": \"\", \"label\": \"Sample.Info.File\", \"description\": \"A TXT format sample info file containing a sample ID, sample file name, and notes column in tab-delimited format. - .txt\"}, {\"name\": \"annotation.gtf\", \"value\": \"\", \"label\": \"Annotation.Gtf\", \"description\": \"A file containing annotations in GTF format. Either an annotation gtf or annotation gtf file must be provided.Note that the provided hg19 reference sequence requires contigs using in the '1' format.\"}, {\"name\": \"annotation.gtf.file\", \"value\": \"\", \"label\": \"Annotation.Gtf.File\", \"description\": \"A file containing annotations in GTF format. Either an annotation gtf file or annotation gtf must be provided.\"}, {\"name\": \"reference.sequence\", \"value\": \"\", \"label\": \"Reference.Sequence\", \"description\": \"A FASTA file containing a reference sequence. The reference sequence provided here has an index (.fai) and a sequence dictionary (.dict). Either a reference sequence or reference sequence file (with index and dictionary) must be provided, but not both. Note that the provided hg19 reference sequence requires contigs using in the '1' format.\"}, {\"name\": \"reference.sequence.file\", \"value\": \"\", \"label\": \"Reference.Sequence.File\", \"description\": \"A FASTA file containing a reference sequence. The reference sequence must have an index (.fai) and a sequence dictionary (.dict) either in the same location or specified in the reference sequence index and reference sequence dictionary parameters. Reference sequence human_assembly19 is available in the documentation. Click on the \\\"documentation\\\" link in the upper right hand corner of this page.\"}, {\"name\": \"reference.sequence.index\", \"value\": \"\", \"label\": \"Reference.Sequence.Index\", \"description\": \"A file containing the index for the reference sequence - .fai\"}, {\"name\": \"reference.sequence.dictionary\", \"value\": \"\", \"label\": \"Reference.Sequence.Dictionary\", \"description\": \"A file containing the dictionary for the reference sequence. - .dict\"}, {\"name\": \"transcript.type.field\", \"value\": \"\", \"label\": \"Transcript.Type.Field\", \"description\": \"Specifies the column of the GTF file in which the transcript type is specified. By default, the module looks for an attribute called transcript_type in the annotation gtf.\"}, {\"name\": \"rRNA.interval.file\", \"value\": \"\", \"label\": \"RRNA.Interval.File\", \"description\": \"A file containing the genomic coordinates of rRNA. - .list\"}, {\"name\": \"rRNA.reference.file.index\", \"value\": \"\", \"label\": \"RRNA.Reference.File.Index\", \"description\": \"zip of index files created by BWA for the supplied rRNA reference file. Required if rRNA reference file is supplied. -.zip\"}, {\"name\": \"transcript.level.metrics\", \"value\": \"\", \"label\": \"Transcript.Level.Metrics\", \"description\": \"Whether to calculate transcript-level metrics in addition to sample-level metrics.\"}, {\"name\": \"gc.content.file\", \"value\": \"\", \"label\": \"Gc.Content.File\", \"description\": \"A file containing GC content for each of the transcripts. The file must be tab-delimited with 2 columns containing transcript name and GC content. The transcript name must appear in the GTF file.\"}, {\"name\": \"num.downsampling.reads\", \"value\": \"\", \"label\": \"Num.Downsampling.Reads\", \"description\": \"Perform downsampling on the given number of reads. It randomly samples the specified number of reads in all experimental samples when calculating metrics.\"}, {\"name\": \"correlation.comparison.file\", \"value\": \"\", \"label\": \"Correlation.Comparison.File\", \"description\": \"A GCT file to use to calculate correlation between expression values (only uses first sample if gct file contains more than one sample. Note also that expression values must be log normalized, and the identifiers must match those of the annotation GTF file)\"}, {\"name\": \"gap.length\", \"value\": \"\", \"label\": \"Gap.Length\", \"description\": \"Whether to plot the distribution of gap lengths\"}, {\"name\": \"GATK.flags\", \"value\": \"\", \"label\": \"GATK.Flags\", \"description\": \"Pass a string of flags directly to the GATK (e.g. \\\"-DBQ 0\\\" to set missing base qualities to zero).\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"RNASeQC\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"single.end\", \"value\": \"yes\", \"label\": \"Single.End\", \"description\": \"Whether the bam file contains single end reads. Select no for paired-end reads.\"}, {\"name\": \"num.genes\", \"value\": \"1000\", \"label\": \"Num.Genes\", \"description\": \"The number of top-expressed genes for which to calculate metrics.\"}, {\"name\": \"num.rRNA.downsampling.reads\", \"value\": \"1000000\", \"label\": \"Num.RRNA.Downsampling.Reads\", \"description\": \"Downsamples to calculate rRNA rate more efficiently. Default is 1 million. Set to 0 to disable.\"}, {\"name\": \"transcript.end.length\", \"value\": \"200\", \"label\": \"Transcript.End.Length\", \"description\": \"The length of the 3' or 5' end of a transcript.\"}, {\"name\": \"gc.margin\", \"value\": \"0.375\", \"label\": \"Gc.Margin\", \"description\": \"Used in conjunction with the gc content file to specify the percent gc content to use as boundaries. E.g. 0.25 would set a lower cutoff of 25% and an upper cutoff of 75% (default is 0.375).\"}, {\"name\": \"output.prefix\", \"value\": \"<bam.files_basename>\", \"label\": \"Output.Prefix\", \"description\": \"The prefix to use for the output file name\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"RNASeQC_CEGS\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"\", \"required_args\": [{\"name\": \"Picard.CreateSequenceDictionary1.reference.sequence.file\", \"value\": \"\", \"label\": \"Picard.CreateSequenceDictionary1.Reference.Sequence.File\", \"description\": \"Input reference fasta or fasta.gz \"}, {\"name\": \"TopHat2.reads.pair.1\", \"value\": \"\", \"label\": \"TopHat2.Reads.Pair.1\", \"description\": \"Unpaired reads file or first mate for paired reads. A file or zip of files containing reads in FASTA or FASTQ format (bz2 and gz compressed files are supported).\"}, {\"name\": \"TopHat2.reads.pair.2\", \"value\": \"\", \"label\": \"TopHat2.Reads.Pair.2\", \"description\": \"Second mate for paired reads. A file or zip of files in FASTA or FASTQ format (bz2 and gz compressed files are supported).\"}, {\"name\": \"Picard.ReorderSam3.reference.file\", \"value\": \"\", \"label\": \"Picard.ReorderSam3.Reference.File\", \"description\": \"Reference sequence to which to reorder reads in FASTA format.\"}, {\"name\": \"RNASeQC7.gc.content.file\", \"value\": \"\", \"label\": \"RNASeQC7.Gc.Content.File\", \"description\": \"A file containing GC content for each of the transcripts. The file must be tab-delimited with 2 columns containing transcript name and GC content. The transcript name must appear in the GTF file.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"RNASeQC_CEGS\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"SamSPECTRALClusterFCS\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Cluster flow cytometry data using the SamSPECTRAL algorithm.\", \"required_args\": [{\"name\": \"Input.FCS.data.file\", \"value\": \"\", \"label\": \"Input.FCS.Data.File\", \"description\": \"The FCS file to be clustered.\"}, {\"name\": \"Dimensions\", \"value\": \"\", \"label\": \"Dimensions\", \"description\": \"A comma-separated list of dimensions (parameters/channels) that shall be used for clustering. Use either parameter names or indexes. All dimensions except for Time will be used if this parameter is not provided.\"}, {\"name\": \"Dimensions.to.transform\", \"value\": \"\", \"label\": \"Dimensions.To.Transform\", \"description\": \"A comma-separated list of dimensions (parameters/channels) that shall transformed as specified by previous parameter. This will be ignored if no transformation is specified above. If this parameter is not provided and transformation is specified above, the algorithm will use heuristics to identify parameters that shall be transformed. These heuristics are based on how parameters are stored in the FCS file, their resolution and their name. Again, you can use either parameter names or parameter indexes to specify dimensions to transform.\"}, {\"name\": \"Sigma\", \"value\": \"\", \"label\": \"Sigma\", \"description\": \"A scaling parameter that determines the resolution in the spectral clustering stage. By increasing it, more spectral clusters are identified. This can be usefull when small population are aimed. See documentation on more details how Sigma can be adjusted.\"}, {\"name\": \"Separation.factor\", \"value\": \"\", \"label\": \"Separation.Factor\", \"description\": \"Threshold that controls to what extend clusters should be combined or kept separate. See documentation on more details how the separation factor can be adjusted.\"}, {\"name\": \"Seed\", \"value\": \"\", \"label\": \"Seed\", \"description\": \"Seed value for the random generator; provide any consistent value to obtain reproducible results.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"SamSPECTRALClusterFCS\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"Transformation\", \"value\": \"ASinH\", \"label\": \"Transformation\", \"description\": \"Which transformation to apply on the data prior clustering. Fluorescence channels are usually better visualized and clustered using a transformation; the ASinH transformation produces good results on most data.\"}, {\"name\": \"Beta\", \"value\": \"4\", \"label\": \"Beta\", \"description\": \"The size of neighborhood that is being searched during the conductance calculation step. The default value of 4 should not be changed unless the input data set contains measurements of more that 100,000. Setting beta to zero or a low value will reduce computational time by applying an approximation to the conductance calculation step. This will allow you to cluster large data sets in reasonable time; however, the quality of the result may be affected. Setting beta to 5 is not recommended since it will increase computational time while not significantly improving clustering results in most cases.\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"SAMtools.FastaIndex\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Indexes a reference sequence in FASTA format\", \"required_args\": [{\"name\": \"fasta.file\", \"value\": \"\", \"label\": \"Fasta.File\", \"description\": \"A sequence in FASTA format.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"SAMtools.FastaIndex\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.prefix\", \"value\": \"<fasta.file_basename>\", \"label\": \"Output.Prefix\", \"description\": \"The prefix to use for the output file name\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"Scripture\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Segmentation-based transcript reconstruction algorithm\", \"required_args\": [{\"name\": \"alignment.file\", \"value\": \"\", \"label\": \"Alignment.File\", \"description\": \"Alignment file in SAM or BAM format, sorted by start position\"}, {\"name\": \"alignment.index.file\", \"value\": \"\", \"label\": \"Alignment.Index.File\", \"description\": \"Index file for alignment file, in SAI or BAI format. The index file must have the same root name as the alignment file. (Example: foo.sam and foo.sam.sai OR foo.sai) \"}, {\"name\": \"chromosome.size.file\", \"value\": \"\", \"label\": \"Chromosome.Size.File\", \"description\": \"File containing sizes of chromosomes\"}, {\"name\": \"chromosome\", \"value\": \"\", \"label\": \"Chromosome\", \"description\": \"Selected chromosome, e.g., chr19\"}, {\"name\": \"chromosome.sequence.file\", \"value\": \"\", \"label\": \"Chromosome.Sequence.File\", \"description\": \"Chromosome sequence in FASTA format\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"Scripture\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.prefix\", \"value\": \"<alignment.file_basename>\", \"label\": \"Output.Prefix\", \"description\": \"A label that will be used to name output files\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"ScripturePipeline\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Segmentation-based transcript reconstruction using single- or paired-end reads\", \"required_args\": [{\"name\": \"TopHat1.prebuilt.bowtie.index\", \"value\": \"\", \"label\": \"TopHat1.Prebuilt.Bowtie.Index\", \"description\": \"A prebuilt bowtie index. Either a prebuilt or a custom Bowtie index must be specified.\"}, {\"name\": \"TopHat1.custom.bowtie.index\", \"value\": \"\", \"label\": \"TopHat1.Custom.Bowtie.Index\", \"description\": \"A ZIP archive containing Bowtie index files. Either a prebuilt or a custom Bowtie index must be specified.\"}, {\"name\": \"TopHat1.reads.pair.1\", \"value\": \"\", \"label\": \"TopHat1.Reads.Pair.1\", \"description\": \"Unpaired reads file or first mate for paired reads. A file or zip of files containing reads in FASTA or FASTQ format.\"}, {\"name\": \"TopHat1.reads.pair.2\", \"value\": \"\", \"label\": \"TopHat1.Reads.Pair.2\", \"description\": \"Second mate for paired reads. A file or zip of files in FASTA or FASTQ format.\"}, {\"name\": \"TopHat1.mate.inner.dist\", \"value\": \"\", \"label\": \"TopHat1.Mate.Inner.Dist\", \"description\": \"The expected mean inner distance between mate pairs.\"}, {\"name\": \"TopHat1.mate.std.dev\", \"value\": \"\", \"label\": \"TopHat1.Mate.Std.Dev\", \"description\": \"The standard deviation for the distribution on inner distances between mate pairs.\"}, {\"name\": \"TopHat1.library.type\", \"value\": \"\", \"label\": \"TopHat1.Library.Type\", \"description\": \"Library type for strand specific reads.\"}, {\"name\": \"Scripture3.chromosome.size.file\", \"value\": \"\", \"label\": \"Scripture3.Chromosome.Size.File\", \"description\": \"File containing sizes of chromosomes\"}, {\"name\": \"Scripture3.chromosome\", \"value\": \"\", \"label\": \"Scripture3.Chromosome\", \"description\": \"Selected chromosome, e.g., chr19\"}, {\"name\": \"Scripture3.chromosome.sequence.file\", \"value\": \"\", \"label\": \"Scripture3.Chromosome.Sequence.File\", \"description\": \"Chromosome sequence in FASTA format\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"ScripturePipeline\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"TopHat1.integer.quals\", \"value\": \"no\", \"label\": \"TopHat1.Integer.Quals\", \"description\": \"Quality values are space-delimited integer values, this becomes default when you specify color option.\"}, {\"name\": \"Scripture3.output.prefix\", \"value\": \"<alignment.file_basename>\", \"label\": \"Scripture3.Output.Prefix\", \"description\": \"A label that will be used to name output files\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"ScripturePrealigned\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Segmentation-based transcript reconstruction using pre-aligned reads\", \"required_args\": [{\"name\": \"SortSam1.input.file\", \"value\": \"\", \"label\": \"SortSam1.Input.File\", \"description\": \"Input file in SAM or BAM format\"}, {\"name\": \"Scripture2.chromosome.size.file\", \"value\": \"\", \"label\": \"Scripture2.Chromosome.Size.File\", \"description\": \"File containing sizes of chromosomes\"}, {\"name\": \"Scripture2.chromosome\", \"value\": \"\", \"label\": \"Scripture2.Chromosome\", \"description\": \"Selected chromosome, e.g., chr19\"}, {\"name\": \"Scripture2.chromosome.sequence.file\", \"value\": \"\", \"label\": \"Scripture2.Chromosome.Sequence.File\", \"description\": \"Chromosome sequence in FASTA format\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"ScripturePrealigned\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"Scripture2.output.prefix\", \"value\": \"<alignment.file_basename>\", \"label\": \"Scripture2.Output.Prefix\", \"description\": \"A label that will be used to name output files\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"SegmentationCount\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Counts the number of segments in a SEG file.\", \"required_args\": [{\"name\": \"input.file\", \"value\": \"\", \"label\": \"Input.File\", \"description\": \"A segmentation file - .seg, .txt\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"SegmentationCount\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.file\", \"value\": \"<input.file_basename>.segcount.txt\", \"label\": \"Output.File\", \"description\": \"The name of the output file\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"SelectFeaturesColumns\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Takes a 'column slice' from a .res, .gct, .snp, .cn, .loh, .odf, or .cls file.\", \"required_args\": [{\"name\": \"input.filename\", \"value\": \"\", \"label\": \"Input.Filename\", \"description\": \"input file - .res, .gct, .snp, .cn, .loh, .odf, or .cls file from which the slice is extracted\"}, {\"name\": \"columns\", \"value\": \"\", \"label\": \"Columns\", \"description\": \"columns to select (e.g. 0-3, 5, 9). Either columns or a list file must be provided.\"}, {\"name\": \"list.filename\", \"value\": \"\", \"label\": \"List.Filename\", \"description\": \"text file containing one identifier per line that specifies the columns to select. Either columns or a list file must be provided.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"SelectFeaturesColumns\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output\", \"value\": \"<input.filename_basename>.slice\", \"label\": \"Output\", \"description\": \"output filename - .res, .gct, .snp, .cn, .loh, .odf, .cls\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"SelectFeaturesRows\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Generate a .res, .gct, .snp, .cn, .loh, or .odf file containing only specified features (i.e. rows).\", \"required_args\": [{\"name\": \"input.filename\", \"value\": \"\", \"label\": \"Input.Filename\", \"description\": \"input file - .res, .gct, .snp, .cn, .loh, or .odf file from which the slice is extracted\"}, {\"name\": \"list.filename\", \"value\": \"\", \"label\": \"List.Filename\", \"description\": \"text file containing one identifier per line that specifies the rows to select. \"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"SelectFeaturesRows\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output\", \"value\": \"<list.filename_basename>.slice\", \"label\": \"Output\", \"description\": \"output filename - .res, .gct, .snp, .cn, .loh, .odf\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"SelectFileMatrix\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"A helper module which selects a matrix out of a delimited text file.\", \"required_args\": [{\"name\": \"input.file\", \"value\": \"\", \"label\": \"Input.File\", \"description\": \"Name of the input file from which to pull the matrix.  Must be a delimited text file.\"}, {\"name\": \"start.row\", \"value\": \"\", \"label\": \"Start.Row\", \"description\": \"Starting row of the matrix (line number within the file).  Defaults to the first row.\"}, {\"name\": \"end.row\", \"value\": \"\", \"label\": \"End.Row\", \"description\": \"Ending row of the matrix (line number within the file).  Defaults to the last row.\"}, {\"name\": \"start.column\", \"value\": \"\", \"label\": \"Start.Column\", \"description\": \"Starting column of the matrix.  Defaults to the first column.\"}, {\"name\": \"end.column\", \"value\": \"\", \"label\": \"End.Column\", \"description\": \"Ending column of the matrix. Defaults to the last column.\"}, {\"name\": \"row.select\", \"value\": \"\", \"label\": \"Row.Select\", \"description\": \"Select specific rows to include in the matrix (e.g. \\\"3,5,7-12\\\").  Note that start/end row parameters may limit this selection.\"}, {\"name\": \"column.select\", \"value\": \"\", \"label\": \"Column.Select\", \"description\": \"Select specific columns to include in the matrix (e.g. \\\"1,2,4-6\\\").  Note that start/end column parameters may limit this selection\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"SelectFileMatrix\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.file.base.name\", \"value\": \"<input.file_basename>.select\", \"label\": \"Output.File.Base.Name\", \"description\": \"Base name for the output file.  The module will add an appropriate extension, either .csv for a comma-delimted file or .txt for a space- or tab-delimited file).\"}, {\"name\": \"delimiter\", \"value\": \"TAB\", \"label\": \"Delimiter\", \"description\": \"Token naming the character for the output column delimiter. Choices are TAB, COMMA, and SPACE. TAB is the default.\"}, {\"name\": \"output.rejects\", \"value\": \"false\", \"label\": \"Output.Rejects\", \"description\": \"Produce an output file containing the items not meeting the selection criteria.  The file will be named <output.file.base.name>.REJECT.<ext>\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"SetFCSKeywords\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Sets keyword/value(s) in a Flow Cytometry Standard (FCS) file.\", \"required_args\": [{\"name\": \"Input.FCS.data.file\", \"value\": \"\", \"label\": \"Input.FCS.Data.File\", \"description\": \"The input FCS data file in which to set keyword/value pairs.\"}, {\"name\": \"Keywords.input.file\", \"value\": \"\", \"label\": \"Keywords.Input.File\", \"description\": \"The file with keyword/value pairs to add (or change values of existing keywords). The keyword/value pairs can be submitted either as an CSV file with two columns: \\\"Keyword\\\" and \\\"Value\\\" (heading is optional) or as a text file with a chunk of FCS TEXT segment, e.g., /$P3S/MUC 1 APC/$P4S/Thy1 PE/BD$CUSTOM_VALUE23/12/.\"}, {\"name\": \"Output.fcs.file.name\", \"value\": \"\", \"label\": \"Output.Fcs.File.Name\", \"description\": \"The file name for the output FCS data file.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"SetFCSKeywords\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"shRNAcollapseReps\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Collapses replicate samples in a gct data file, based on a sample information file\", \"required_args\": [{\"name\": \"gct.file\", \"value\": \"\", \"label\": \"Gct.File\", \"description\": \"The input gct data file.\"}, {\"name\": \"sample.info.file\", \"value\": \"\", \"label\": \"Sample.Info.File\", \"description\": \"The cell line (sample) information file.\"}, {\"name\": \"prefix\", \"value\": \"\", \"label\": \"Prefix\", \"description\": \"The file name prefix.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"shRNAcollapseReps\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"shRNAfoldChange\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Calculate the fold change of each sample in a gct file, based on the appropriate DNA reference value (present in a gct file of the same dimension).\", \"required_args\": [{\"name\": \"gct.file\", \"value\": \"\", \"label\": \"Gct.File\", \"description\": \"The input gct data file.\"}, {\"name\": \"DNA.reference.file\", \"value\": \"\", \"label\": \"DNA.Reference.File\", \"description\": \"The values of the DNA pool reference per sample/shRNA in a gct file (same dimension as input gct file).\"}, {\"name\": \"prefix\", \"value\": \"\", \"label\": \"Prefix\", \"description\": \"The file name prefix.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"shRNAfoldChange\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"shRNAmapGenes\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Maps shRNAs in a gct data file to genes based on a mapping file.\", \"required_args\": [{\"name\": \"gct.file\", \"value\": \"\", \"label\": \"Gct.File\", \"description\": \"The input gct data file.\"}, {\"name\": \"mapping\", \"value\": \"\", \"label\": \"Mapping\", \"description\": \"The shRNA to gene mapping file.\"}, {\"name\": \"prefix\", \"value\": \"\", \"label\": \"Prefix\", \"description\": \"The file name prefix.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"shRNAmapGenes\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"shRNAremoveOverlap\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Remove shRNAs overlapping in sequence > 3 basepairs from a gct data file\", \"required_args\": [{\"name\": \"list\", \"value\": \"\", \"label\": \"List\", \"description\": \"The list of shRNAs that are not overlapping.\"}, {\"name\": \"gct.file\", \"value\": \"\", \"label\": \"Gct.File\", \"description\": \"The input gct data file.\"}, {\"name\": \"mapping\", \"value\": \"\", \"label\": \"Mapping\", \"description\": \"The shRNA to gene mapping file.\"}, {\"name\": \"prefix\", \"value\": \"\", \"label\": \"Prefix\", \"description\": \"The file name prefix.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"shRNAremoveOverlap\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"SNPFileCreator\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Process Affymetrix SNP probe-level data into an expression value\", \"required_args\": [{\"name\": \"chip.type\", \"value\": \"\", \"label\": \"Chip.Type\", \"description\": \"Specify the chip type.\"}, {\"name\": \"network.directory.name\", \"value\": \"\", \"label\": \"Network.Directory.Name\", \"description\": \"Networked directory name where the CEL ( and optionally TXT) files reside.\"}, {\"name\": \"zip.filename\", \"value\": \"\", \"label\": \"Zip.Filename\", \"description\": \"A zip file containing CEL and optionally TXT files.\"}, {\"name\": \"reference.filename\", \"value\": \"\", \"label\": \"Reference.Filename\", \"description\": \"Reference file to use. Leave blank if reference method is median.\"}, {\"name\": \"output.file\", \"value\": \"\", \"label\": \"Output.File\", \"description\": \"The name of the output file.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"SNPFileCreator\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"normalization.method\", \"value\": \"2\", \"label\": \"Normalization.Method\", \"description\": \"The normalization method to apply.\"}, {\"name\": \"reference.method\", \"value\": \"median\", \"label\": \"Reference.Method\", \"description\": \"Reference method to use in normalization\"}, {\"name\": \"model.method\", \"value\": \"2\", \"label\": \"Model.Method\", \"description\": \"The summarization method to apply.\"}, {\"name\": \"allele.specific\", \"value\": \"no\", \"label\": \"Allele.Specific\", \"description\": \"Create Allele Specific probe intensities\"}, {\"name\": \"sort.snp.file\", \"value\": \"yes\", \"label\": \"Sort.Snp.File\", \"description\": \"Sorts output file by Chromosome and location. Filters out probesets without location information\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"SNPFileSorter\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Sorts a .snp file by Chromosome and location\", \"required_args\": [{\"name\": \"snp.filename\", \"value\": \"\", \"label\": \"Snp.Filename\", \"description\": \"snp file to sort\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"SNPFileSorter\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.file\", \"value\": \"<snp.filename_basename>.sorted.snp\", \"label\": \"Output.File\", \"description\": \"output file name\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"SNPMultipleSampleAnalysis\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Determine Regions of Concordant Copy Number Aberrations\", \"required_args\": [{\"name\": \"input.filename\", \"value\": \"\", \"label\": \"Input.Filename\", \"description\": \"The input file to analyze - .cn, .xcn, .snp, .txt\"}, {\"name\": \"run.name\", \"value\": \"\", \"label\": \"Run.Name\", \"description\": \"The name of the job\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"SNPMultipleSampleAnalysis\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"java.max.memory\", \"value\": \"512m\", \"label\": \"Java.Max.Memory\", \"description\": \"\"}, {\"name\": \"save.dir\", \"value\": \"./\", \"label\": \"Save.Dir\", \"description\": \"Directory to save to\"}, {\"name\": \"permutations\", \"value\": \"500\", \"label\": \"Permutations\", \"description\": \"Number of permutations to use\"}, {\"name\": \"num.tests\", \"value\": \"9\", \"label\": \"Num.Tests\", \"description\": \"Number of MSA tests to use\"}, {\"name\": \"num.positions.bin\", \"value\": \"6\", \"label\": \"Num.Positions.Bin\", \"description\": \"Number of positions per bin\"}, {\"name\": \"resolution\", \"value\": \"250000\", \"label\": \"Resolution\", \"description\": \"The size of the plotted chromosome diagram\"}, {\"name\": \"chromosomes\", \"value\": \"chr22\", \"label\": \"Chromosomes\", \"description\": \"Comma delimited list of chromosomes to analyze\"}, {\"name\": \"save.images\", \"value\": \"./\", \"label\": \"Save.Images\", \"description\": \"Saves images to the specified directory\"}, {\"name\": \"significance.level\", \"value\": \".05\", \"label\": \"Significance.Level\", \"description\": \"The value at which a region is called significant\"}, {\"name\": \"analyze.by\", \"value\": \"false\", \"label\": \"Analyze.By\", \"description\": \"\"}, {\"name\": \"cytoband.filename\", \"value\": \"<libdir>CytoBands.txt\", \"label\": \"Cytoband.Filename\", \"description\": \"Cytoband file to use\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"SnpViewer\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Displays a heat map of snp data\", \"required_args\": [{\"name\": \"dataset.filename\", \"value\": \"\", \"label\": \"Dataset.Filename\", \"description\": \"The dataset to display - .snp, .cn, .xcn\"}, {\"name\": \"sample.info.filename\", \"value\": \"\", \"label\": \"Sample.Info.Filename\", \"description\": \"Sample information file\"}, {\"name\": \"sample.list.filename\", \"value\": \"\", \"label\": \"Sample.List.Filename\", \"description\": \"Sample list file\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"SnpViewer\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"max.memory\", \"value\": \"700m\", \"label\": \"Max.Memory\", \"description\": \"Maximum memory\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"SOMClustering\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Self-Organizing Maps algorithm\", \"required_args\": [{\"name\": \"dataset.filename\", \"value\": \"\", \"label\": \"Dataset.Filename\", \"description\": \"Dataset (res, gct, or odf dataset)\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"SOMClustering\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.stub\", \"value\": \"<dataset.filename_basename>\", \"label\": \"Output.Stub\", \"description\": \"The base output file name - .odf\"}, {\"name\": \"cluster.range\", \"value\": \"2-3\", \"label\": \"Cluster.Range\", \"description\": \"Range of values can be entered and the program will automatically run the algorithm for each number of clusters in the range.\"}, {\"name\": \"seed.range\", \"value\": \"42\", \"label\": \"Seed.Range\", \"description\": \"The seed for the random number generator, is exposed to allow the user to recreate a given session at a later time, (as opposed to allowing the program to generate a random initial seed which could result in different outcomes even if all the other parameters are identical).\"}, {\"name\": \"iterations\", \"value\": \"50000\", \"label\": \"Iterations\", \"description\": \"How many times the algorithm should try to refine the clusters. Initially, this value can be set low for faster exploration, but should be set high (e.g., 50,000) for good convergence.\"}, {\"name\": \"cluster.by\", \"value\": \"rows\", \"label\": \"Cluster.By\", \"description\": \"Whether to cluster by rows or columns\"}, {\"name\": \"som.rows\", \"value\": \"0\", \"label\": \"Som.Rows\", \"description\": \"Setting this and som.cols to a non zero value will override cluster.range and the computation will be for the specified geometry\"}, {\"name\": \"som.cols\", \"value\": \"0\", \"label\": \"Som.Cols\", \"description\": \"Setting this and som.rows to a non zero value will override cluster.range and the computation will be for the specified geometry\"}, {\"name\": \"initialization\", \"value\": \"Random_Vectors\", \"label\": \"Initialization\", \"description\": \"How to select inital random centroids\"}, {\"name\": \"neighborhood\", \"value\": \"Bubble\", \"label\": \"Neighborhood\", \"description\": \"Neighborhood function determining how centroids near the target centroid are updated\"}, {\"name\": \"alpha.initial\", \"value\": \"0.1\", \"label\": \"Alpha.Initial\", \"description\": \"Initial learning weight for centroid updates\"}, {\"name\": \"alpha.final\", \"value\": \"0.005\", \"label\": \"Alpha.Final\", \"description\": \"Final learning weight for centroid updates\"}, {\"name\": \"sigma.initial\", \"value\": \"5.0\", \"label\": \"Sigma.Initial\", \"description\": \"Initial sigma to determine update neighborhood size around the target centroid\"}, {\"name\": \"sigma.final\", \"value\": \"0.5\", \"label\": \"Sigma.Final\", \"description\": \"Final sigma to determine update neighborhood size around the target centroid\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"SOMClusterViewer\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Visualize clusters created with the SOM algorithm\", \"required_args\": [{\"name\": \"som.cluster.filename\", \"value\": \"\", \"label\": \"Som.Cluster.Filename\", \"description\": \"SOM Properties ODF file\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"SOMClusterViewer\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"SparseHierarchicalClustering\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Cluster the samples in a data matrix using an adaptively-chosen subset of the features.\", \"required_args\": [{\"name\": \"input.filename\", \"value\": \"\", \"label\": \"Input.Filename\", \"description\": \"A GCT file containing data to cluster\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"SparseHierarchicalClustering\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"method\", \"value\": \"average\", \"label\": \"Method\", \"description\": \"The type of linkage to use in the hierarchical clustering - \\\"single\\\", \\\"complete\\\", \\\"centroid\\\", or \\\"average.\\\"\"}, {\"name\": \"wbound\", \"value\": \"-1\", \"label\": \"Wbound\", \"description\": \"The sum of gene weights; this is the tuning parameter for sparse hierarchical clustering. If a non-negative number is given, then this tuning parameter is used. Otherwise, a tuning parameter is selected via a permutation approach. Note: The program runs much more quickly if wbound is specified rather than left at -1.\"}, {\"name\": \"maxnumgenes\", \"value\": \"5000\", \"label\": \"Maxnumgenes\", \"description\": \"By default, only the 5000 genes with highest variance will be used (for computational reasons). To use all the genes, set to -1. Warning: using all genes will be slow if there are many genes in the data set.\"}, {\"name\": \"cluster.features\", \"value\": \"false\", \"label\": \"Cluster.Features\", \"description\": \"If a clustering of the genes with non-zero weights is desired, set to TRUE.\"}, {\"name\": \"method.features\", \"value\": \"average\", \"label\": \"Method.Features\", \"description\": \"The type of linkage used to cluster the features (if desired) - \\\"single\\\", \\\"complete\\\", \\\"centroid\\\", or \\\"average\\\". Used only if cluster features is set to \\\"TRUE\\\"\"}, {\"name\": \"standardize.arrays\", \"value\": \"true\", \"label\": \"Standardize.Arrays\", \"description\": \"Whether or not the arrays should be standardized to have mean zero and standard deviation one before clustering is performed.\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"SplitColumns\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Creates a separate file for each column specified from the input file\", \"required_args\": [{\"name\": \"input.file\", \"value\": \"\", \"label\": \"Input.File\", \"description\": \"A tab delimited file\"}, {\"name\": \"split.columns\", \"value\": \"\", \"label\": \"Split.Columns\", \"description\": \"Specify columns to extract (i.e. 2,4, 5-, 11-17,). Leave blank to split all columns.\"}, {\"name\": \"additional.columns\", \"value\": \"\", \"label\": \"Additional.Columns\", \"description\": \"Specify subset of additional columns to include in each (i.e. 2,4, 5-, 11-17) output file.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"SplitColumns\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.prefix\", \"value\": \"<input.file_basename>\", \"label\": \"Output.Prefix\", \"description\": \"The prefix to use for the output file name.\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"SplitDatasetTrainTest\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Splits a dataset (and cls file) into a number of train and test subsets\", \"required_args\": [{\"name\": \"input.dataset.file\", \"value\": \"\", \"label\": \"Input.Dataset.File\", \"description\": \"input dataset - .gct, .res\"}, {\"name\": \"cls.file\", \"value\": \"\", \"label\": \"Cls.File\", \"description\": \"class template - .cls\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"SplitDatasetTrainTest\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"split.method\", \"value\": \"random\", \"label\": \"Split.Method\", \"description\": \"whether to split the data using a percentage split or cross validation approach\"}, {\"name\": \"stratified\", \"value\": \"T\", \"label\": \"Stratified\", \"description\": \"whether to create splits stratified with respect to the class template\"}, {\"name\": \"folds\", \"value\": \"1\", \"label\": \"Folds\", \"description\": \"number of train/test splits to generate\"}, {\"name\": \"percentage.split.proportion\", \"value\": \"0.5\", \"label\": \"Percentage.Split.Proportion\", \"description\": \"proportion of data to be allocated to train file when split method is percentage split (ignored when split method is cross-validation)\"}, {\"name\": \"seed\", \"value\": \"123456\", \"label\": \"Seed\", \"description\": \"random number generator seed\"}, {\"name\": \"output.prefix\", \"value\": \"<input.dataset.file_basename>\", \"label\": \"Output.Prefix\", \"description\": \"output saved to output.prefix.{train,test}.n.{gct|res,cls}, where n is the fold index\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"SraToFastQ\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Converts data in SRA format to FastQ using the SRAToolkit version 2.3.5-2.\", \"required_args\": [{\"name\": \"accession\", \"value\": \"\", \"label\": \"Accession\", \"description\": \"A Sequence Read Archive accession.  Either an accession or an sra.file is required.\"}, {\"name\": \"sra.file\", \"value\": \"\", \"label\": \"Sra.File\", \"description\": \"A Sequence Read Archive (.sra) file.  Either an accession or an sra.file is required.\"}, {\"name\": \"paired.end\", \"value\": \"\", \"label\": \"Paired.End\", \"description\": \"Whether the reads are paired end\"}, {\"name\": \"quality.score.offset\", \"value\": \"\", \"label\": \"Quality.Score.Offset\", \"description\": \"Offset to use for ASCII quality scores. Default is 33 (\\\"!\\\").\"}, {\"name\": \"biological.reads.only\", \"value\": \"\", \"label\": \"Biological.Reads.Only\", \"description\": \"Whether to include only biological reads and no technical reads in output\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"SraToFastQ\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"alignment.status\", \"value\": \"any\", \"label\": \"Alignment.Status\", \"description\": \"The alignment status of the reads to include in output\"}, {\"name\": \"output.compression\", \"value\": \"none\", \"label\": \"Output.Compression\", \"description\": \"Compression to apply to the output files\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"ssGSEAProjection\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Performs single sample GSEA Projection\", \"required_args\": [{\"name\": \"input.gct.file\", \"value\": \"\", \"label\": \"Input.Gct.File\", \"description\": \"Input gct file.  This module does not allow missing values.\"}, {\"name\": \"output.file.prefix\", \"value\": \"\", \"label\": \"Output.File.Prefix\", \"description\": \"Output filename prefix.  If unspecified, is set to <prefix of input.gct.file>.PROJ.\"}, {\"name\": \"gene.sets.database.files\", \"value\": \"\", \"label\": \"Gene.Sets.Database.Files\", \"description\": \"Gene sets description file(s) - .gmt, .gmx.  Upload a gene set if your gene set is not listed as a choice for the gene sets database parameter.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"ssGSEAProjection\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"gene.sets.database\", \"value\": \"c1.all.v5.2.symbols.gmt\", \"label\": \"Gene.Sets.Database\", \"description\": \"Gene sets database from GSEA website.\"}, {\"name\": \"gene.symbol.column\", \"value\": \"Name\", \"label\": \"Gene.Symbol.Column\", \"description\": \"Specifies the gct data column containing gene symbols.  In most cases this will be \\\"Column 1\\\".\"}, {\"name\": \"gene.set.selection\", \"value\": \"ALL\", \"label\": \"Gene.Set.Selection\", \"description\": \"Comma-separated list of gene set names on which to project expression data. Set to \\\"ALL\\\" if projecting to all defined gene sets.\"}, {\"name\": \"sample.normalization.method\", \"value\": \"rank\", \"label\": \"Sample.Normalization.Method\", \"description\": \"Method used to normalize gene expression data\"}, {\"name\": \"weighting.exponent\", \"value\": \"0.75\", \"label\": \"Weighting.Exponent\", \"description\": \"Module authors strongly recommend against changing from default.\"}, {\"name\": \"min.gene.set.size\", \"value\": \"10\", \"label\": \"Min.Gene.Set.Size\", \"description\": \"Gene sets with overlap smaller than this are excluded from the analysis.\"}, {\"name\": \"combine.mode\", \"value\": \"combine.add\", \"label\": \"Combine.Mode\", \"description\": \"Options for combining enrichment scores for paired *_UP and *_DN gene sets\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"StrandSpecRNAseqEval\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Evaluates strand-specific yeast (S. cerevisiae) RNA-seq libraries\", \"required_args\": [{\"name\": \"sam.file\", \"value\": \"\", \"label\": \"Sam.File\", \"description\": \"The input SAM file to read and evaluate - .sam\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"StrandSpecRNAseqEval\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"average.gene.stats\", \"value\": \"0\", \"label\": \"Average.Gene.Stats\", \"description\": \"A flag to indicate whether to run the average gene statistics. This is a long process.\"}, {\"name\": \"flip.strands\", \"value\": \"0\", \"label\": \"Flip.Strands\", \"description\": \"A flag to indicate whether the data is flipped, so a forward gene would appear as a reverse one\"}, {\"name\": \"output.prefix\", \"value\": \"<sam.file_basename>\", \"label\": \"Output.Prefix\", \"description\": \"A prefix for the output files\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"SubMap\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Maps subclasses between two data sets\", \"required_args\": [{\"name\": \"datasetA.file\", \"value\": \"\", \"label\": \"DatasetA.File\", \"description\": \"Input dataset A - .gct, should have common gene ID with dataset B\"}, {\"name\": \"datasetB.file\", \"value\": \"\", \"label\": \"DatasetB.File\", \"description\": \"Input dataset B - .gct,  should have common gene ID with dataset A\"}, {\"name\": \"classA.file\", \"value\": \"\", \"label\": \"ClassA.File\", \"description\": \"Input class label A - .cls, 3rd line should be numeric, +1 if starts from \\\"0\\\"\"}, {\"name\": \"classB.file\", \"value\": \"\", \"label\": \"ClassB.File\", \"description\": \"Input class label B - .cls, 3rd line should be numeric, +1 if starts from \\\"0\\\"\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"SubMap\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"num.marker.genes\", \"value\": \"100\", \"label\": \"Num.Marker.Genes\", \"description\": \"Number of marker genes to be mapped\"}, {\"name\": \"num.perm\", \"value\": \"100\", \"label\": \"Num.Perm\", \"description\": \"Number of random permutations for ES in each cell of SA matrix\"}, {\"name\": \"num.perm.fisher\", \"value\": \"1000\", \"label\": \"Num.Perm.Fisher\", \"description\": \"Number of random permutations for Fisher's statistics\"}, {\"name\": \"weighted.score.type\", \"value\": \"1\", \"label\": \"Weighted.Score.Type\", \"description\": \"Weight enrichment by correlation vector (SNR)\"}, {\"name\": \"null.distribution\", \"value\": \"pool\", \"label\": \"Null.Distribution\", \"description\": \"Choice of null distribution\"}, {\"name\": \"p.value.correction\", \"value\": \"Bonferroni\", \"label\": \"P.Value.Correction\", \"description\": \"p-value correction method\"}, {\"name\": \"cluster.rows\", \"value\": \"1\", \"label\": \"Cluster.Rows\", \"description\": \"Cluster dataset A's subclass in heatmap of SA matrix\"}, {\"name\": \"cluster.columns\", \"value\": \"1\", \"label\": \"Cluster.Columns\", \"description\": \"Cluster dataset B's subclass in heatmap of SA matrix\"}, {\"name\": \"nominal.p.value.matrix\", \"value\": \"T\", \"label\": \"Nominal.P.Value.Matrix\", \"description\": \"Create heatmap of each nominal-p matrix\"}, {\"name\": \"create.legend\", \"value\": \"T\", \"label\": \"Create.Legend\", \"description\": \"Create legend for heatmap\"}, {\"name\": \"random.seed\", \"value\": \"47365321\", \"label\": \"Random.Seed\", \"description\": \"Random seed\"}, {\"name\": \"output.filename\", \"value\": \"SubMap\", \"label\": \"Output.Filename\", \"description\": \"Name of output files containing heatmap of SA matrices, summary of ES matrix, nominal p-values, and corrected p-values\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"SuggestNumberOfPopulationsFCS\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"The FlowClust module requires that you specify how many populations you are looking for.  This module takes a data driven approach and attempts to discover how many populations are present in an FCS file\", \"required_args\": [{\"name\": \"Input.FCS.data.file\", \"value\": \"\", \"label\": \"Input.FCS.Data.File\", \"description\": \"The FCS file\"}, {\"name\": \"Dimensions\", \"value\": \"\", \"label\": \"Dimensions\", \"description\": \"The channels to search for populations, separated by commas. Leave blank to search in all channels. E.g. FSC Lin, SS Lin, CD4 Log\"}, {\"name\": \"Range\", \"value\": \"\", \"label\": \"Range\", \"description\": \"By default, we try between 1 and 30 populations and report the most likely number in the file. As this is a time consuming procedure, you can specify a more limited range. e.g. 1,12\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"SuggestNumberOfPopulationsFCS\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"SurvivalCurve\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Draws a survival curve based on a cls file\", \"required_args\": [{\"name\": \"input.surv.data.file\", \"value\": \"\", \"label\": \"Input.Surv.Data.File\", \"description\": \"input clinical data including survival time & censor status -.txt\"}, {\"name\": \"input.cls.file\", \"value\": \"\", \"label\": \"Input.Cls.File\", \"description\": \"Class information file -.cls\"}, {\"name\": \"cls.field\", \"value\": \"\", \"label\": \"Cls.Field\", \"description\": \"Name of class field in survival data file (optional)\"}, {\"name\": \"manual.line.type\", \"value\": \"\", \"label\": \"Manual.Line.Type\", \"description\": \"e.g. 1,2,3,... (1: solid, 2: dashed, 3: dotted, 4: dot-dashed)\"}, {\"name\": \"manual.line.color\", \"value\": \"\", \"label\": \"Manual.Line.Color\", \"description\": \"e.g. black, blue, red, ...\"}, {\"name\": \"max.time\", \"value\": \"\", \"label\": \"Max.Time\", \"description\": \"Maximum survival time (in original scale)\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"SurvivalCurve\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"time.field\", \"value\": \"time\", \"label\": \"Time.Field\", \"description\": \"Name of survival time field\"}, {\"name\": \"censor.field\", \"value\": \"status\", \"label\": \"Censor.Field\", \"description\": \"Name of survival time field (value should be \\\"event:1, censor:0\\\")\"}, {\"name\": \"print.fit.results\", \"value\": \"F\", \"label\": \"Print.Fit.Results\", \"description\": \"Output file name for survival fit results\"}, {\"name\": \"line.type.color.assign\", \"value\": \"automatic\", \"label\": \"Line.Type.Color.Assign\", \"description\": \"Assign line type/color automatically/manually?\"}, {\"name\": \"line.width\", \"value\": \"1\", \"label\": \"Line.Width\", \"description\": \"Line width\"}, {\"name\": \"time.conversion\", \"value\": \"1\", \"label\": \"Time.Conversion\", \"description\": \"Survival time conversion\"}, {\"name\": \"surv.function.lower\", \"value\": \"0\", \"label\": \"Surv.Function.Lower\", \"description\": \"Lower limit of y axis range\"}, {\"name\": \"surv.function.higher\", \"value\": \"1\", \"label\": \"Surv.Function.Higher\", \"description\": \"Upper limit of y axis range\"}, {\"name\": \"curve.type\", \"value\": \"log\", \"label\": \"Curve.Type\", \"description\": \"Type of survival curve\"}, {\"name\": \"show.conf.interval\", \"value\": \"0\", \"label\": \"Show.Conf.Interval\", \"description\": \"Show confidence inteval?\"}, {\"name\": \"add.legend\", \"value\": \"T\", \"label\": \"Add.Legend\", \"description\": \"Show legend?\"}, {\"name\": \"legend.position\", \"value\": \"left-bottom\", \"label\": \"Legend.Position\", \"description\": \"Position of legend\"}, {\"name\": \"output.filename\", \"value\": \"<input.surv.data.file_basename>\", \"label\": \"Output.Filename\", \"description\": \"Name for output files\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"SurvivalDifference\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Tests for survival difference based on cls file\", \"required_args\": [{\"name\": \"input.surv.data.file\", \"value\": \"\", \"label\": \"Input.Surv.Data.File\", \"description\": \"input clinical data including survival time & censor status  -.txt\"}, {\"name\": \"input.cls.file\", \"value\": \"\", \"label\": \"Input.Cls.File\", \"description\": \"class information file (.cls format)\"}, {\"name\": \"cls.field\", \"value\": \"\", \"label\": \"Cls.Field\", \"description\": \"Name of class field in survival data file (optional)\"}, {\"name\": \"subgroup\", \"value\": \"\", \"label\": \"Subgroup\", \"description\": \"Suggroups (classes) to be compared (e.g., A, C as in 3rd line of cls file)\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"SurvivalDifference\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"time.field\", \"value\": \"time\", \"label\": \"Time.Field\", \"description\": \"Name of survival time field\"}, {\"name\": \"censor.field\", \"value\": \"status\", \"label\": \"Censor.Field\", \"description\": \"Name of censor field (value should be \\\"event:1, no event: 0\\\")\"}, {\"name\": \"test.type\", \"value\": \"0\", \"label\": \"Test.Type\", \"description\": \"Type of test\"}, {\"name\": \"output.filename\", \"value\": \"<input.surv.data.file_basename>\", \"label\": \"Output.Filename\", \"description\": \"Output file name\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"SVM\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Classify samples using SVM\", \"required_args\": [{\"name\": \"train.data.filename\", \"value\": \"\", \"label\": \"Train.Data.Filename\", \"description\": \"The training data file - .gct, .res\"}, {\"name\": \"train.cls.filename\", \"value\": \"\", \"label\": \"Train.Cls.Filename\", \"description\": \"The training class file - .cls\"}, {\"name\": \"saved.model.filename\", \"value\": \"\", \"label\": \"Saved.Model.Filename\", \"description\": \"A saved model from a previous run of SVM\"}, {\"name\": \"test.data.filename\", \"value\": \"\", \"label\": \"Test.Data.Filename\", \"description\": \"The test data file - .gct, .res\"}, {\"name\": \"test.cls.filename\", \"value\": \"\", \"label\": \"Test.Cls.Filename\", \"description\": \"The test class file - .cls\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"SVM\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"pred.results.output.file\", \"value\": \"<test.data.filename_basename>.pred.odf\", \"label\": \"Pred.Results.Output.File\", \"description\": \"The name of the output file for prediction results\"}, {\"name\": \"model.output.file\", \"value\": \"<train.data.filename_basename>.model\", \"label\": \"Model.Output.File\", \"description\": \"The name of the file containing the saved model\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"TmapToGct\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Converts a TMAP file to GCT format\", \"required_args\": [{\"name\": \"input.file\", \"value\": \"\", \"label\": \"Input.File\", \"description\": \"Input file in TMAP format\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"TmapToGct\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"row.labels\", \"value\": \"Cufflinks gene ID\", \"label\": \"Row.Labels\", \"description\": \"Select row labels to be written to GCT file\"}, {\"name\": \"row.descriptions\", \"value\": \"Cufflinks transcript ID\", \"label\": \"Row.Descriptions\", \"description\": \"Select row descriptions to be written to GCT file\"}, {\"name\": \"filter.rows\", \"value\": \"no\", \"label\": \"Filter.Rows\", \"description\": \"Filter rows with invalid row names (ie., \\\"-\\\") \"}, {\"name\": \"output.prefix\", \"value\": \"<input.file_basename>\", \"label\": \"Output.Prefix\", \"description\": \"A label that will be used to name the GCT output file\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"TopHat\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"[**Beta Release** Contact gp-help with any issues. Check stdout.txt and stderr.txt for errors] TopHat 2.0.11 is a fast splice junction mapper for RNA-Seq reads. It aligns RNA-Seq reads to mammalian-sized genomes using the ultra high-throughput short read aligner Bowtie, and then analyzes the mapping results to identify splice junctions between exons.\", \"required_args\": [{\"name\": \"GTF.file\", \"value\": \"\", \"label\": \"GTF.File\", \"description\": \"A GTF file (v. 2.2 or higher) or GFF3 file containing a list of gene model annotations. TopHat will first extract the transcript sequences and align them to this virtual transcriptome first. Only the reads that do not fully map to the transcriptome will then be mapped on the genome. The reads that did map on the transcriptome will be converted to genomic mappings (spliced as needed) and merged with the novel mappings.\"}, {\"name\": \"transcriptome.index\", \"value\": \"\", \"label\": \"Transcriptome.Index\", \"description\": \"A directory containing a transcriptome index generated from a previous run of TopHat\"}, {\"name\": \"bowtie.index\", \"value\": \"\", \"label\": \"Bowtie.Index\", \"description\": \"A zip file or directory containing a Bowtie 2 index.\"}, {\"name\": \"reads.pair.1\", \"value\": \"\", \"label\": \"Reads.Pair.1\", \"description\": \"Unpaired reads file or first mate for paired reads. One or more files containing reads in FASTA or FASTQ format (bz2 and gz compressed files are supported).\"}, {\"name\": \"reads.pair.2\", \"value\": \"\", \"label\": \"Reads.Pair.2\", \"description\": \"Second mate for paired reads. Zero or more files in FASTA or FASTQ format (bz2 and gz compressed files are supported).\"}, {\"name\": \"mate.std.dev\", \"value\": \"\", \"label\": \"Mate.Std.Dev\", \"description\": \"The standard deviation for the distribution of inner distances between mate pairs.\"}, {\"name\": \"library.type\", \"value\": \"\", \"label\": \"Library.Type\", \"description\": \"Library type for strand specific reads.\"}, {\"name\": \"Bowtie.preset.options\", \"value\": \"\", \"label\": \"Bowtie.Preset.Options\", \"description\": \"A combination of pre-packaged options for Bowtie 2 based on speed and sensitivity/accuracy.\"}, {\"name\": \"max.transcriptome.hits\", \"value\": \"\", \"label\": \"Max.Transcriptome.Hits\", \"description\": \"The maximum number of mappings allowed for a read when it is aligned to the virtual transcriptome (provided in the GTF file parameter). Any reads found with more than this number of mappings will be discarded.\"}, {\"name\": \"prefilter.multihits\", \"value\": \"\", \"label\": \"Prefilter.Multihits\", \"description\": \"When mapping reads on the virtual transcriptome (provided in the GTF file parameter), some repetitive or low complexity reads that would be discarded in the context of the genome may appear to align to the transcript sequences and thus may end up reported as mapped to those genes only. This option directs TopHat to first align the reads to the whole genome, then exclude such multi-mapped reads.\"}, {\"name\": \"raw.junctions.file\", \"value\": \"\", \"label\": \"Raw.Junctions.File\", \"description\": \"A file containing raw junctions. Junctions are specified one per line in a tab-delimited format.\"}, {\"name\": \"find.novel.junctions\", \"value\": \"\", \"label\": \"Find.Novel.Junctions\", \"description\": \"If you select no, then the module will only look for junctions indicated in the GTF file supplied in the GTF file parameter. (This parameter is ignored when no GTF file is specified.)\"}, {\"name\": \"min.intron.length\", \"value\": \"\", \"label\": \"Min.Intron.Length\", \"description\": \"The minimum intron length. TopHat will ignore donor/acceptor pairs closer than this many bases apart.\"}, {\"name\": \"quality.value.scale\", \"value\": \"\", \"label\": \"Quality.Value.Scale\", \"description\": \"Whether to use the Solexa, Phred 33, or  Solexa v. 1.3 (Phred 64) quality value scale.\"}, {\"name\": \"microexon.search\", \"value\": \"\", \"label\": \"Microexon.Search\", \"description\": \"Attempts to find alignments incident to microexons. Works only for reads 50bp or longer.\"}, {\"name\": \"fusion.mapping\", \"value\": \"\", \"label\": \"Fusion.Mapping\", \"description\": \"Whether to enable fusion mapping\"}, {\"name\": \"read.edit.dist\", \"value\": \"\", \"label\": \"Read.Edit.Dist\", \"description\": \"Final read alignments having more than these many edit distance are discarded.\"}, {\"name\": \"read.gap.length\", \"value\": \"\", \"label\": \"Read.Gap.Length\", \"description\": \"Final read alignments having more than these many total length of gaps are discarded.\"}, {\"name\": \"additional.tophat.options\", \"value\": \"\", \"label\": \"Additional.Tophat.Options\", \"description\": \"Additional options to be passed along to the TopHat program at the command line. This parameter gives you a means to specify otherwise unavailable TopHat options and switches not supported by the module; check the TopHat manual for details. Recommended for experts only; use this at your own discretion.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"TopHat\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"mate.inner.dist\", \"value\": \"50\", \"label\": \"Mate.Inner.Dist\", \"description\": \"The expected mean inner distance between mate pairs.\"}, {\"name\": \"transcriptome.only\", \"value\": \"no\", \"label\": \"Transcriptome.Only\", \"description\": \"Whether to align the reads to the virtual transcriptome (provided in the GTF file parameter) and report only those mappings as genomic mappings.\"}, {\"name\": \"min.anchor.length\", \"value\": \"8\", \"label\": \"Min.Anchor.Length\", \"description\": \"The anchor length. This value must be at least 3.\"}, {\"name\": \"max.splice.mismatches\", \"value\": \"0\", \"label\": \"Max.Splice.Mismatches\", \"description\": \"The maximum number of mismatches that may appear in the \\\"anchor\\\" region of a spliced alignment.\"}, {\"name\": \"max.intron.length\", \"value\": \"500000\", \"label\": \"Max.Intron.Length\", \"description\": \"The maximum intron length. When searching for junctions ab initio, TopHat will ignore donor/acceptor pairs farther than this many bases apart, except when such a pair is supported by a split segment alignment of a long read.\"}, {\"name\": \"max.insertion.length\", \"value\": \"3\", \"label\": \"Max.Insertion.Length\", \"description\": \"The maximum insertion length\"}, {\"name\": \"max.deletion.length\", \"value\": \"3\", \"label\": \"Max.Deletion.Length\", \"description\": \"The maximum deletion length\"}, {\"name\": \"max.multihits\", \"value\": \"20\", \"label\": \"Max.Multihits\", \"description\": \"The maximum number of times a read can be aligned to the reference genome. If a read is aligned more than this number of times, then TopHat will choose the alignments based on their alignment scores, reporting the alignments with the best alignment scores. If there are more than this number of alignments with the same score for a read, TopHat will randomly report only this many alignments.\"}, {\"name\": \"read.mismatches\", \"value\": \"2\", \"label\": \"Read.Mismatches\", \"description\": \"Final read alignments having more than these many mismatches should be discarded.\"}, {\"name\": \"coverage.search\", \"value\": \"--no-coverage-search\", \"label\": \"Coverage.Search\", \"description\": \"Enables or disables the coverage based search for junctions. Use when coverage search is disabled by default (such as for reads 75bp or longer), for maximum sensitivity.\"}, {\"name\": \"fusion.anchor.length\", \"value\": \"20\", \"label\": \"Fusion.Anchor.Length\", \"description\": \"A \\\"supporting\\\" read must map to both sides of a fusion by at least these many bases. Only applies when fusion mapping is set to yes.\"}, {\"name\": \"fusion.read.mismatches\", \"value\": \"2\", \"label\": \"Fusion.Read.Mismatches\", \"description\": \"Reads support fusions if they map across fusion with at most these many mismatches. Only applies when fusion mapping is set to yes.\"}, {\"name\": \"output.prefix\", \"value\": \"<job_id>_<reads.pair.1_basename>\", \"label\": \"Output.Prefix\", \"description\": \"The prefix to use for the output file\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"TransposeDataset\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Transpose a Dataset - .res .gct, .odf\", \"required_args\": [{\"name\": \"input.filename\", \"value\": \"\", \"label\": \"Input.Filename\", \"description\": \"Dataset - .res .gct, .odf\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"TransposeDataset\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.file.format\", \"value\": \"gct\", \"label\": \"Output.File.Format\", \"description\": \"The output file format\"}, {\"name\": \"output.file.name\", \"value\": \"<input.filename_basename>.transposed\", \"label\": \"Output.File.Name\", \"description\": \"The name of the output file\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"Trimmomatic\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Provides a variety of options for trimming Illumina FASTQ files of adapter sequences and low-quality reads.\", \"required_args\": [{\"name\": \"input.file.1\", \"value\": \"\", \"label\": \"Input.File.1\", \"description\": \"The input FASTQ to be trimmed.  For paired-end data, this should be the forward (\\\"*_1\\\" or \\\"left\\\") input file.\"}, {\"name\": \"input.file.2\", \"value\": \"\", \"label\": \"Input.File.2\", \"description\": \"The reverse (\\\"*_2\\\" or \\\"right\\\") input FASTQ of paired-end data to be trimmed.\"}, {\"name\": \"adapter.clip.sequence.file\", \"value\": \"\", \"label\": \"Adapter.Clip.Sequence.File\", \"description\": \"A FASTA file containing the adapter sequences, PCR sequences, etc. to be clipped. This parameter is required to enable adapter clipping. Files are provided for several Illumina pipelines but you can also provide your own; see the documentation for details. Be sure to choose a PE file for paired-end data and an SE file for single-end data.\"}, {\"name\": \"adapter.clip.seed.mismatches\", \"value\": \"\", \"label\": \"Adapter.Clip.Seed.Mismatches\", \"description\": \"Specifies the maximum mismatch count which will still allow a full match to be performed. A value of 2 is recommended. This parameter is required to enable adapter clipping.\"}, {\"name\": \"adapter.clip.palindrome.clip.threshold\", \"value\": \"\", \"label\": \"Adapter.Clip.Palindrome.Clip.Threshold\", \"description\": \"Specifies how accurate the match between the two 'adapter ligated' reads must be for PE palindrome read alignment. This is the log10 probability against getting a match by random chance; values around 30 or more are recommended. This parameter is required to enable adapter clipping.\"}, {\"name\": \"adapter.clip.simple.clip.threshold\", \"value\": \"\", \"label\": \"Adapter.Clip.Simple.Clip.Threshold\", \"description\": \"Specifies how accurate the match between any adapter etc. sequence must be against a read as a log10 probability against getting a match by random chance; values between 7-15 are recommended. This parameter is required to enable adapter clipping.\"}, {\"name\": \"adapter.clip.min.length\", \"value\": \"\", \"label\": \"Adapter.Clip.Min.Length\", \"description\": \"In addition to the alignment score, palindrome mode can verify that a minimum length of adapter has been detected. If unspecified, this defaults to 8 bases, for historical reasons. However, since palindrome mode has a very low false positive rate, this can be safely reduced, even down to 1, to allow shorter adapter fragments to be removed.\"}, {\"name\": \"trim.leading.quality.threshold\", \"value\": \"\", \"label\": \"Trim.Leading.Quality.Threshold\", \"description\": \"Remove low quality bases from the beginning. As long as a base has a value below this threshold the base is removed and the next base will be investigated.\"}, {\"name\": \"trim.trailing.quality.threshold\", \"value\": \"\", \"label\": \"Trim.Trailing.Quality.Threshold\", \"description\": \"Remove low quality bases from the end. As long as a base has a value below this threshold the base is removed and the next trailing base will be investigated.\"}, {\"name\": \"max.info.target.length\", \"value\": \"\", \"label\": \"Max.Info.Target.Length\", \"description\": \"Performs a Max Info adaptive quality trim, balancing the benefits of retaining longer reads against the costs of retaining bases with errors. This parameter specifies the read length which is likely to allow the location of the read within the target sequence to be determined.\"}, {\"name\": \"max.info.strictness\", \"value\": \"\", \"label\": \"Max.Info.Strictness\", \"description\": \"This value, which should be set between 0 and 1, specifies the balance between preserving as much read length as possible vs. removal of incorrect bases. A low value of this parameter (<0.2) favors longer reads, while a high value (>0.8) favors read correctness.  Both max.info.target.length and max.info.strictness are required for the Max Info quality trim.\"}, {\"name\": \"sliding.window.size\", \"value\": \"\", \"label\": \"Sliding.Window.Size\", \"description\": \"Perform a sliding window trimming, cutting once the average quality within the window falls below a threshold. By considering multiple bases, a single poor quality base will not cause the removal of high quality data later in the read.  This parameter specifies the number of bases to average across.\"}, {\"name\": \"sliding.window.quality.threshold\", \"value\": \"\", \"label\": \"Sliding.Window.Quality.Threshold\", \"description\": \"Specifies the average quality required for the sliding window trimming. Both sliding.window.size and sliding.window.quality.threshold are required to enable the sliding window trimming.\"}, {\"name\": \"min.read.length\", \"value\": \"\", \"label\": \"Min.Read.Length\", \"description\": \"Remove reads that fall below the specified minimal length.\"}, {\"name\": \"extra.steps\", \"value\": \"\", \"label\": \"Extra.Steps\", \"description\": \"Extra steps to be performed after any other processing. These must be specified in exactly the format described in the Trimmomatic manual; see the documentation for details.  This is recommended for advanced users only.\"}, {\"name\": \"convert.phred.scores\", \"value\": \"\", \"label\": \"Convert.Phred.Scores\", \"description\": \"Convert phred scores into a particular encoding.  Leave this blank for no conversion.\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"Trimmomatic\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.filename.base\", \"value\": \"<input.file.1_basename>\", \"label\": \"Output.Filename.Base\", \"description\": \"A base name to be used for the output files.\"}, {\"name\": \"adapter.clip.keep.both.reads\", \"value\": \"true\", \"label\": \"Adapter.Clip.Keep.Both.Reads\", \"description\": \"Controls whether to keep both forward and reverse reads when trimming in palindrome mode. The reverse read is the same as the forward but in reverse complement and so carries no additional information. The default is \\\"yes\\\" (retain the reverse read) which is useful when downstream tools cannot handle a combination of paired and unpaired reads.\"}, {\"name\": \"phred.encoding\", \"value\": \"phred33\", \"label\": \"Phred.Encoding\", \"description\": \"Allows you to specify the phred quality encoding.  The default is phred33, which matches modern Illumina pipelines.\"}, {\"name\": \"create.trimlog\", \"value\": \"false\", \"label\": \"Create.Trimlog\", \"description\": \"Create a log of the trimming process.  This gives details on what operations were performed, etc. but can be quite lengthy.\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"Two.Component.Normalization\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Normalization for unimodal and bimodal samples using mixture models\", \"required_args\": [{\"name\": \"gct.file\", \"value\": \"\", \"label\": \"Gct.File\", \"description\": \"Input data file (gct format)\"}, {\"name\": \"cls.file\", \"value\": \"\", \"label\": \"Cls.File\", \"description\": \"File indicating type of samples (unimodal or bimodal) in gct.file\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"Two.Component.Normalization\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.prefix\", \"value\": \"<gct.file_basename>\", \"label\": \"Output.Prefix\", \"description\": \"Filename prefix for results files\"}, {\"name\": \"data.type\", \"value\": \"proteome\", \"label\": \"Data.Type\", \"description\": \"Indicate type of input data (proteome, phosphoprotome, etc.) for use in plots\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"UniquifyLabels\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Makes row and column labels unique\", \"required_args\": [{\"name\": \"input.filename\", \"value\": \"\", \"label\": \"Input.Filename\", \"description\": \"input filename - .res, .gct, .odf type=Dataset\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"UniquifyLabels\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.file\", \"value\": \"<input.filename_basename>.unique\", \"label\": \"Output.File\", \"description\": \"output file\"}, {\"name\": \"make.row.labels.unique\", \"value\": \"yes\", \"label\": \"Make.Row.Labels.Unique\", \"description\": \"whether to make the row labels unique\"}, {\"name\": \"make.column.labels.unique\", \"value\": \"yes\", \"label\": \"Make.Column.Labels.Unique\", \"description\": \"whether to make the column labels unique\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"untar\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"untars .tgz files\", \"required_args\": [{\"name\": \"input.filename\", \"value\": \"\", \"label\": \"Input.Filename\", \"description\": \"\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"untar\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"VennDiagram\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Displays a Venn diagram\", \"required_args\": [{\"name\": \"feature.list.one\", \"value\": \"\", \"label\": \"Feature.List.One\", \"description\": \"Feature list one - .grp, .txt\"}, {\"name\": \"feature.list.two\", \"value\": \"\", \"label\": \"Feature.List.Two\", \"description\": \"Feature list two - .grp, .txt\"}, {\"name\": \"feature.list.three\", \"value\": \"\", \"label\": \"Feature.List.Three\", \"description\": \"Feature list three - .grp, .txt\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"VennDiagram\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"WeightedVoting\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Weighted Voting classification\", \"required_args\": [{\"name\": \"train.filename\", \"value\": \"\", \"label\": \"Train.Filename\", \"description\": \"training data file name - .gct, .res, .odf type = Dataset, ignored if a saved model (saved.model.filename) is used\"}, {\"name\": \"train.class.filename\", \"value\": \"\", \"label\": \"Train.Class.Filename\", \"description\": \"class file for training data - .cls, ignored if a saved model (saved.model.filename) is used\"}, {\"name\": \"saved.model.filename\", \"value\": \"\", \"label\": \"Saved.Model.Filename\", \"description\": \"input Weighted Voting model file - .odf type = Weighted Voting Prediction Model\"}, {\"name\": \"test.filename\", \"value\": \"\", \"label\": \"Test.Filename\", \"description\": \"test data file name - .gct, .res, .odf type = Dataset\"}, {\"name\": \"test.class.filename\", \"value\": \"\", \"label\": \"Test.Class.Filename\", \"description\": \"class file for test data - .cls\"}, {\"name\": \"min.std\", \"value\": \"\", \"label\": \"Min.Std\", \"description\": \"minimum standard deviation if feature selection statistic includes min std option\"}, {\"name\": \"feature.list.filename\", \"value\": \"\", \"label\": \"Feature.List.Filename\", \"description\": \"features to use for prediction\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"WeightedVoting\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"model.file\", \"value\": \"<train.filename_basename>.model.odf\", \"label\": \"Model.File\", \"description\": \"name of output Weighted Voting model file  - .odf type = Weighted Voting Prediction Model\"}, {\"name\": \"num.features\", \"value\": \"10\", \"label\": \"Num.Features\", \"description\": \"number of selected features if feature list filename is not specified\"}, {\"name\": \"feature.selection.statistic\", \"value\": \"0\", \"label\": \"Feature.Selection.Statistic\", \"description\": \"statistic to use to perform feature selection\"}, {\"name\": \"pred.results.file\", \"value\": \"<test.filename_basename>.pred.odf\", \"label\": \"Pred.Results.File\", \"description\": \"name of prediction results output file - .odf type = Prediction Results\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"WeightedVotingXValidation\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Weighted Voting classification with Leave-One-Out Cross-Validation\", \"required_args\": [{\"name\": \"data.filename\", \"value\": \"\", \"label\": \"Data.Filename\", \"description\": \"Dataset input data file - res, gct, odf\"}, {\"name\": \"class.filename\", \"value\": \"\", \"label\": \"Class.Filename\", \"description\": \"Class input file - cls\"}, {\"name\": \"min.std\", \"value\": \"\", \"label\": \"Min.Std\", \"description\": \"minimum standard deviation if feature selection statistic includes min std option\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"WeightedVotingXValidation\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"num.features\", \"value\": \"10\", \"label\": \"Num.Features\", \"description\": \"number of selected features\"}, {\"name\": \"feature.selection.statistic\", \"value\": \"0\", \"label\": \"Feature.Selection.Statistic\", \"description\": \"statistic to use to perform feature selection\"}, {\"name\": \"pred.results.file\", \"value\": \"<data.filename_basename>.pred.odf\", \"label\": \"Pred.Results.File\", \"description\": \"Prediction results output file name - odf\"}, {\"name\": \"feature.summary.file\", \"value\": \"<data.filename_basename>.feat.odf\", \"label\": \"Feature.Summary.File\", \"description\": \"Feature summary output file name - odf\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"XChromosomeCorrect\": {\"library_path\": \"/Users/kateme/Jumis/tools/genepattern-python\", \"library_name\": \"gp.functions\", \"function_name\": \"execute\", \"description\": \"Corrects X Chromosome SNP's for male samples.\", \"required_args\": [{\"name\": \"snp.filename\", \"value\": \"\", \"label\": \"Snp.Filename\", \"description\": \"Summarized probe intensity data in .snp format\"}, {\"name\": \"sample.info.filename\", \"value\": \"\", \"label\": \"Sample.Info.Filename\", \"description\": \"A Sample information file that contains gender information\"}], \"default_args\": [{\"name\": \"task_name\", \"value\": \"XChromosomeCorrect\", \"label\": \"Task Name\", \"description\": \"No description.\"}], \"optional_args\": [{\"name\": \"output.file\", \"value\": \"<snp.filename_basename>.xcorrected.snp\", \"label\": \"Output.File\", \"description\": \"Output file name\"}], \"returns\": [{\"label\": \"Job\", \"description\": \"ONLY FOR TESTING: RETURNS FIELD WILL BE REMOVED\"}]}, \"Make Data Filepath\": {\"library_path\": \"\", \"library_name\": \"simpli.nbpackage_tasks\", \"function_name\": \"make_data_filepath\", \"description\": \"No description.\", \"required_args\": [{\"name\": \"filename\", \"value\": \"\", \"label\": \"File Name\", \"description\": \"No description.\"}], \"default_args\": [], \"optional_args\": [], \"returns\": [{\"label\": \"Variable for Data File Path\", \"description\": \"No description.\"}]}, \"Make Result Filepath\": {\"library_path\": \"\", \"library_name\": \"simpli.nbpackage_tasks\", \"function_name\": \"make_result_filepath\", \"description\": \"No description.\", \"required_args\": [{\"name\": \"filename\", \"value\": \"\", \"label\": \"File Name\", \"description\": \"No description.\"}], \"default_args\": [], \"optional_args\": [], \"returns\": [{\"label\": \"Variable for Result File Path\", \"description\": \"No description.\"}]}}\n"
     ]
    }
   ],
   "source": [
    "import sys\n",
    "sys.path.insert(0, '')\n",
    "import simpli\n",
    "m = simpli.Manager(True)\n",
    "m.print_tasks_as_json()\n",
    "simpli.default_tasks.link_json('/Users/kateme/Jumis/tools/genepattern-python/functions.json')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
