{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The autoreload extension is already loaded. To reload it, use:\n",
      "  %reload_ext autoreload\n"
     ]
    }
   ],
   "source": [
    "from pprint import pprint\n",
    "import json\n",
    "import sys\n",
    "sys.path.insert(0, '../genepattern-python/')\n",
    "from gp import gp_functions\n",
    "\n",
    "%load_ext autoreload\n",
    "%autoreload 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['ABSOLUTE',\n",
       " 'ABSOLUTE.review',\n",
       " 'ABSOLUTE.summarize',\n",
       " 'Acgh2Tab',\n",
       " 'AddFCSEventIndex',\n",
       " 'AddFCSParameter',\n",
       " 'AddNoiseToFCS',\n",
       " 'AffySTExpressionFileCreator',\n",
       " 'ApplyGatingML',\n",
       " 'ARACNE',\n",
       " 'AreaChange',\n",
       " 'Arff2Gct',\n",
       " 'ATARiS',\n",
       " 'AuDIT',\n",
       " 'BedToGtf',\n",
       " 'Beroukhim.Getz.2007.PNAS.Glioma.GISTIC',\n",
       " 'BlastTrainTest',\n",
       " 'BlastXValidation',\n",
       " 'Bowtie.aligner',\n",
       " 'Bowtie.indexer',\n",
       " 'BWA.aln',\n",
       " 'BWA.bwasw',\n",
       " 'BWA.indexer',\n",
       " 'CaArray2ImportViewer',\n",
       " 'CART',\n",
       " 'CARTXValidation',\n",
       " 'CBS',\n",
       " 'ChIPSeq.CreateHeatmap',\n",
       " 'ChIPSeq.CreatePlots',\n",
       " 'ChIPSeq.ExtractSignal',\n",
       " 'ClassNeighbors',\n",
       " 'ClsFileCreator',\n",
       " 'CNToXCN',\n",
       " 'CoClustering',\n",
       " 'CollapseDataset',\n",
       " 'ComBat',\n",
       " 'CombineOdf',\n",
       " 'ComparativeMarkerSelection',\n",
       " 'ComparativeMarkerSelectionViewer',\n",
       " 'CompareSpectra',\n",
       " 'CompensateFCS',\n",
       " 'ConcatenateFilelist',\n",
       " 'ConsensusClustering',\n",
       " 'ConstellationMap',\n",
       " 'ConvertLineEndings',\n",
       " 'ConvertToMAGEML',\n",
       " 'ConvertToMAGETAB',\n",
       " 'COPA',\n",
       " 'CopyNumberDivideByNormals',\n",
       " 'CopyNumberInferencePipeline',\n",
       " 'CopyNumberNoise',\n",
       " 'copyOfReorderGCT',\n",
       " 'CoxRegression',\n",
       " 'CreateSymlinks',\n",
       " 'CsvToFcs',\n",
       " 'Cuffcompare',\n",
       " 'Cuffdiff',\n",
       " 'Cufflinks',\n",
       " 'CufflinksCuffmergePipeline',\n",
       " 'Cuffmerge',\n",
       " 'CummeRbund.GeneSetReport',\n",
       " 'CummeRbund.QcReport',\n",
       " 'CummeRbund.SelectedConditionsReport',\n",
       " 'CummeRbund.SelectedGeneReport',\n",
       " 'CytoscapeViewer',\n",
       " 'DAPPLE',\n",
       " 'DeIdentifyFCS',\n",
       " 'DependencyBiomarkerDotplot',\n",
       " 'DESeq2',\n",
       " 'DownloadURL',\n",
       " 'ESPPredictor',\n",
       " 'ExpressionFileCreator',\n",
       " 'ExprToGct',\n",
       " 'ExtractColumnNames',\n",
       " 'ExtractComparativeMarkerResults',\n",
       " 'ExtractFCSDataset',\n",
       " 'ExtractFCSKeywords',\n",
       " 'ExtractFCSParameters',\n",
       " 'ExtractRowNames',\n",
       " 'FastQC',\n",
       " 'FCMFeatureExtraction',\n",
       " 'FCMSinglePanelQC',\n",
       " 'FCSClean',\n",
       " 'FCSNormalization',\n",
       " 'FcsToCsv',\n",
       " 'FeatureSummaryViewer',\n",
       " 'FileSplitter',\n",
       " 'FilterFilelist',\n",
       " 'FilterLowshRNAs',\n",
       " 'Find.differentially.expressed.genes.in.RNASeq.data',\n",
       " 'Find.subnetworks.of.differentially.expressed.genes.and.identify.associated.biological.functions',\n",
       " 'FLAMEChooseOptimalClusterNumber',\n",
       " 'FLAMEContourDataGenerator',\n",
       " 'FLAMEContourViewer.Pipeline',\n",
       " 'FLAMEMetacluster',\n",
       " 'FLAMEMixtureModel',\n",
       " 'FLAMEPreprocess',\n",
       " 'FLAMEPreviewTransformation',\n",
       " 'FLAMEViewer',\n",
       " 'FlowClustClassifyFCS',\n",
       " 'FlowFingerprinting',\n",
       " 'FlowMeansCluster',\n",
       " 'FlowMergeCluster',\n",
       " 'Fpkm_trackingToGct',\n",
       " 'FPmatching',\n",
       " 'Gct2Arff',\n",
       " 'GctToPcl',\n",
       " 'GENE_E',\n",
       " 'GeneListSignificanceViewer',\n",
       " 'GeneNeighbors',\n",
       " 'GenePatternDocumentExtractor',\n",
       " 'GEOImporter',\n",
       " 'GetDatasetInSilico',\n",
       " 'GISTIC_2.0',\n",
       " 'GISTICPreprocess',\n",
       " 'GLAD',\n",
       " 'GlobalAlignment',\n",
       " 'Golub.Slonim.1999.Nature.all.aml.pipeline',\n",
       " 'GSEA',\n",
       " 'GSEALeadingEdgeViewer',\n",
       " 'GSEAPreranked',\n",
       " 'HAPSEG',\n",
       " 'HeatMapImage',\n",
       " 'HeatMapViewer',\n",
       " 'helloWorld',\n",
       " 'helloWorld_R2.15_Rscript',\n",
       " 'HierarchicalClustering',\n",
       " 'HierarchicalClusteringImage',\n",
       " 'HierarchicalClusteringViewer',\n",
       " 'Hu68kHu35kAtoU95',\n",
       " 'Identify.an.up.or.downregulated.pathway.from.expression.data',\n",
       " 'Identify.and.Validate.a.Consensus.Signature.Using.Gene.Expression.Data',\n",
       " 'Identify.and.visualize.expressed.transcripts.in.RNASeq.data',\n",
       " 'IGV',\n",
       " 'IGV.js',\n",
       " 'IlluminaConcatenator',\n",
       " 'IlluminaDASLPipeline',\n",
       " 'IlluminaExpressionFileCreator',\n",
       " 'IlluminaNormalizer',\n",
       " 'IlluminaScanExtractor',\n",
       " 'ImmPort_FLOCK_CrossSample',\n",
       " 'ImmPort_FLOCK_Individual_FCS',\n",
       " 'ImmPortColSelection',\n",
       " 'ImmPortCrossSample',\n",
       " 'ImmPortCSV2TXT',\n",
       " 'ImmPortFCSConvLogicleTrans',\n",
       " 'ImmPortFLOCK',\n",
       " 'ImputeMissingValues.KNN',\n",
       " 'Install.Proteogenomics.R.Packages',\n",
       " 'IntersectingIdentifiers',\n",
       " 'JavaTreeView',\n",
       " 'KMeansClassifyFCS',\n",
       " 'KMeansClustering',\n",
       " 'KNN',\n",
       " 'KNNXValidation',\n",
       " 'KSscore',\n",
       " 'LandmarkMatch',\n",
       " 'Life.Cycle.Time.Point.Correlation',\n",
       " 'ListFiles',\n",
       " 'LocatePeaks',\n",
       " 'LogicleTransformFCS',\n",
       " 'LogisticRegression',\n",
       " 'LogTransformGLAD',\n",
       " 'LOHPaired',\n",
       " 'Lu.Getz.Miska.Nature.June.2005.mouse.lung.pipeline',\n",
       " 'MAGEMLImportViewer',\n",
       " 'MAGETABImportViewer',\n",
       " 'MapChipFeaturesGeneral',\n",
       " 'Marker.Selection',\n",
       " 'MClustClusterLabel',\n",
       " 'MClustClusterLabelBIC',\n",
       " 'MergeColumns',\n",
       " 'MergeFCSDataFiles',\n",
       " 'MergeHTSeqCounts',\n",
       " 'MergeRows',\n",
       " 'MetageneProjection',\n",
       " 'MINDY',\n",
       " 'MINDYViewer',\n",
       " 'ModEcocTrainTest',\n",
       " 'ModEcocXValidation',\n",
       " 'Multiplot',\n",
       " 'MultiplotExtractor',\n",
       " 'MultiplotPreprocess',\n",
       " 'MultiplotStudio',\n",
       " 'MutSigCV',\n",
       " 'mzXMLToCsv',\n",
       " 'NearestTemplatePrediction',\n",
       " 'NMF',\n",
       " 'NMFConsensus',\n",
       " 'NMFConsensus_combine_lewu',\n",
       " 'NMFConsensus_sk_lewu',\n",
       " 'NMFPartition',\n",
       " 'NMFSingleKPipe',\n",
       " 'NormLines',\n",
       " 'ParallelCBS',\n",
       " 'ParallelNMFConsensusPipe',\n",
       " 'PARIS',\n",
       " 'PCA',\n",
       " 'PCAViewer',\n",
       " 'PclToGct',\n",
       " 'PeakMatch',\n",
       " 'Peaks',\n",
       " 'Picard.AddOrReplaceReadGroups',\n",
       " 'Picard.BamToSam',\n",
       " 'Picard.CreateSequenceDictionary',\n",
       " 'Picard.FastqToSam',\n",
       " 'Picard.MarkDuplicates',\n",
       " 'Picard.ReorderSam',\n",
       " 'Picard.SamToBam',\n",
       " 'Picard.SamToFastq',\n",
       " 'Picard.SortSam',\n",
       " 'PlateQAFCS',\n",
       " 'PlotPeaks',\n",
       " 'PredictionResultsViewer',\n",
       " 'Preprocess.and.quality.check.RNASeq.data',\n",
       " 'PreprocessDataset',\n",
       " 'PreprocessReadCounts',\n",
       " 'PreviewFCS',\n",
       " 'Profile.Plots',\n",
       " 'Protein.mRNA.Corrlation',\n",
       " 'ProteinDatasetCreation',\n",
       " 'ProteoArray',\n",
       " 'ProteomicsAnalysis',\n",
       " 'QuaSAR',\n",
       " 'RankNormalize',\n",
       " 'Read_group_trackingToGct',\n",
       " 'ReannotateGCT',\n",
       " 'RemoveCopyNumberOutliers',\n",
       " 'RemoveMissingValues',\n",
       " 'removeSamples',\n",
       " 'RemoveSaturatedFCSEvents',\n",
       " 'RenameFile',\n",
       " 'ReorderByClass',\n",
       " 'ReplicatesQC',\n",
       " 'REVEALER',\n",
       " 'RLibDemo',\n",
       " 'RNASeQC',\n",
       " 'RNASeQC_CEGS',\n",
       " 'SamSPECTRALClusterFCS',\n",
       " 'SAMtools.FastaIndex',\n",
       " 'Scripture',\n",
       " 'ScripturePipeline',\n",
       " 'ScripturePrealigned',\n",
       " 'SegmentationCount',\n",
       " 'SelectFeaturesColumns',\n",
       " 'SelectFeaturesRows',\n",
       " 'SelectFileMatrix',\n",
       " 'SetFCSKeywords',\n",
       " 'shRNAcollapseReps',\n",
       " 'shRNAfoldChange',\n",
       " 'shRNAmapGenes',\n",
       " 'shRNAremoveOverlap',\n",
       " 'SNPFileCreator',\n",
       " 'SNPFileSorter',\n",
       " 'SNPMultipleSampleAnalysis',\n",
       " 'SnpViewer',\n",
       " 'SOMClustering',\n",
       " 'SOMClusterViewer',\n",
       " 'SparseHierarchicalClustering',\n",
       " 'SplitColumns',\n",
       " 'SplitDatasetTrainTest',\n",
       " 'SraToFastQ',\n",
       " 'ssGSEAProjection',\n",
       " 'StrandSpecRNAseqEval',\n",
       " 'SubMap',\n",
       " 'SuggestNumberOfPopulationsFCS',\n",
       " 'SurvivalCurve',\n",
       " 'SurvivalDifference',\n",
       " 'SVM',\n",
       " 'TmapToGct',\n",
       " 'TopHat',\n",
       " 'TransposeDataset',\n",
       " 'Trimmomatic',\n",
       " 'Two.Component.Normalization',\n",
       " 'UniquifyLabels',\n",
       " 'untar',\n",
       " 'VennDiagram',\n",
       " 'WeightedVoting',\n",
       " 'WeightedVotingXValidation',\n",
       " 'XChromosomeCorrect']"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gp_functions.login('https://genepattern.broadinstitute.org/gp', 'Simpli', 'Simpli')\n",
    "\n",
    "gp_functions.get_tasks()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'library_path': 'Jumis/tools/genepattern-python',\n",
      " 'tasks': [{'description': 'Performs several preprocessing steps on a res, '\n",
      "                           'gct, or odf input file',\n",
      "            'function_path': 'gp.gp_functions.execute',\n",
      "            'label': 'PreprocessDataset',\n",
      "            'optional_args': [{'arg_name': 'threshold.and.filter',\n",
      "                               'description': 'Apply floor, ceiling and '\n",
      "                                              'variation filter',\n",
      "                               'value': '1'},\n",
      "                              {'arg_name': 'floor',\n",
      "                               'description': 'Value for floor threshold',\n",
      "                               'value': '20'},\n",
      "                              {'arg_name': 'ceiling',\n",
      "                               'description': 'Value for ceiling threshold',\n",
      "                               'value': '20000'},\n",
      "                              {'arg_name': 'min.fold.change',\n",
      "                               'description': 'Minimum fold change for '\n",
      "                                              'variation filter',\n",
      "                               'value': '3'},\n",
      "                              {'arg_name': 'min.delta',\n",
      "                               'description': 'Minimum delta for variation '\n",
      "                                              'filter',\n",
      "                               'value': '100'},\n",
      "                              {'arg_name': 'num.outliers.to.exclude',\n",
      "                               'description': 'Number of outliers per row to '\n",
      "                                              'ignore when calculating row min '\n",
      "                                              'and max for variation filter',\n",
      "                               'value': '0'},\n",
      "                              {'arg_name': 'row.normalization',\n",
      "                               'description': 'Perform row normalization',\n",
      "                               'value': '0'},\n",
      "                              {'arg_name': 'row.sampling.rate',\n",
      "                               'description': 'Sample rows without replacement '\n",
      "                                              'to obtain this fraction of the '\n",
      "                                              'total number of rows',\n",
      "                               'value': '1'},\n",
      "                              {'arg_name': 'log2.transform',\n",
      "                               'description': 'Apply log2 transform after all '\n",
      "                                              'other preprocessing steps',\n",
      "                               'value': '0'},\n",
      "                              {'arg_name': 'output.file.format',\n",
      "                               'description': 'Output file format',\n",
      "                               'value': '3'},\n",
      "                              {'arg_name': 'output.file',\n",
      "                               'description': 'Output file name',\n",
      "                               'value': '<input.filename_basename>.preprocessed'}],\n",
      "            'other_information': {'lsid': 'urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00020:5.1',\n",
      "                                  'version': '5.1'},\n",
      "            'required_args': [{'arg_name': 'input.filename',\n",
      "                               'description': 'Input filename - .res, .gct, '\n",
      "                                              '.odf'},\n",
      "                              {'arg_name': 'threshold.for.removing.rows',\n",
      "                               'description': 'Threshold for removing rows'},\n",
      "                              {'arg_name': 'number.of.columns.above.threshold',\n",
      "                               'description': 'Remove row if this number of '\n",
      "                                              'columns not >= given '\n",
      "                                              'threshold'}]}]}\n"
     ]
    }
   ],
   "source": [
    "t = gp_functions.GPTask(gp_functions.SERVER, 'PreprocessDataset')\n",
    "t.param_load()\n",
    "\n",
    "required_args = []\n",
    "optional_args = []\n",
    "other_information = {}\n",
    "task_entry = {\n",
    "    \"function_path\": 'gp.gp_functions.execute',\n",
    "    \"label\": t.get_name(),\n",
    "    \"description\": t.get_description(),\n",
    "    \"required_args\": required_args,\n",
    "    \"optional_args\": optional_args,\n",
    "    \"other_information\": other_information,\n",
    "}\n",
    "tasks = []\n",
    "task_json = {\n",
    "    \"library_path\": \"Jumis/tools/genepattern-python\",\n",
    "    \"tasks\": tasks\n",
    "}\n",
    "\n",
    "ps = t.get_parameters()\n",
    "for p in ps:\n",
    "    a = {}\n",
    "    description = p.get_description()\n",
    "    default_value = p.get_default_value()\n",
    "    \n",
    "    a['arg_name'] = p.get_name()\n",
    "    a['description'] = description\n",
    "    if default_value:\n",
    "        a['value'] = default_value\n",
    "    \n",
    "    type_ = p.get_type()\n",
    "    if p.is_choice_param():\n",
    "        choices = [d['value'] for d in p.get_choices()]\n",
    "        a['description']: '{} (default={}) (choices={}) (type=)'.format(description, choices, default_value, type_,)\n",
    "    else:\n",
    "        a['description']: '{} (default={}) (type=)'.format(description, default_value, type_,)\n",
    "    \n",
    "    if default_value:\n",
    "        optional_args.append(a)\n",
    "    else:\n",
    "        required_args.append(a)\n",
    "\n",
    "other_information['lsid'] = t.get_lsid()\n",
    "other_information['version'] = t.get_version()\n",
    "\n",
    "tasks.append(task_entry)\n",
    "\n",
    "pprint(task_json)\n",
    "with open('/Users/kateme/Jumis/tools/genepattern-python/gp_functions.json', 'w') as f:\n",
    "    json.dump(task_json, f, indent=4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "js.set_parameter('input.filename',\n",
    "                 'https://software.broadinstitute.org/cancer/software/genepattern/data/all_aml/all_aml_test.gct')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "j = gp_functions.submit_job(js)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "j.get_status_message()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "j.is_finished()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "for f in j.get_output_files():\n",
    "    print(f.get_url())\n",
    "    data = f.read()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
