{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import gp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# Create a GenePattern server proxy instance\n",
    "gpserver = gp.GPServer('https://genepattern.broadinstitute.org/gp','Simpli', 'Simpli')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ABSOLUTE\n",
      "ABSOLUTE.review\n",
      "ABSOLUTE.summarize\n",
      "Acgh2Tab\n",
      "AddFCSEventIndex\n",
      "AddFCSParameter\n",
      "AddNoiseToFCS\n",
      "AffySTExpressionFileCreator\n",
      "ApplyGatingML\n",
      "ARACNE\n",
      "AreaChange\n",
      "Arff2Gct\n",
      "ATARiS\n",
      "AuDIT\n",
      "BedToGtf\n",
      "Beroukhim.Getz.2007.PNAS.Glioma.GISTIC\n",
      "BlastTrainTest\n",
      "BlastXValidation\n",
      "Bowtie.aligner\n",
      "Bowtie.indexer\n",
      "BWA.aln\n",
      "BWA.bwasw\n",
      "BWA.indexer\n",
      "CaArray2ImportViewer\n",
      "CART\n",
      "CARTXValidation\n",
      "CBS\n",
      "ChIPSeq.CreateHeatmap\n",
      "ChIPSeq.CreatePlots\n",
      "ChIPSeq.ExtractSignal\n",
      "ClassNeighbors\n",
      "ClsFileCreator\n",
      "CNToXCN\n",
      "CoClustering\n",
      "CollapseDataset\n",
      "ComBat\n",
      "CombineOdf\n",
      "ComparativeMarkerSelection\n",
      "ComparativeMarkerSelectionViewer\n",
      "CompareSpectra\n",
      "CompensateFCS\n",
      "ConcatenateFilelist\n",
      "ConsensusClustering\n",
      "ConstellationMap\n",
      "ConvertLineEndings\n",
      "ConvertToMAGEML\n",
      "ConvertToMAGETAB\n",
      "COPA\n",
      "CopyNumberDivideByNormals\n",
      "CopyNumberInferencePipeline\n",
      "CopyNumberNoise\n",
      "copyOfReorderGCT\n",
      "CoxRegression\n",
      "CreateSymlinks\n",
      "CsvToFcs\n",
      "Cuffcompare\n",
      "Cuffdiff\n",
      "Cufflinks\n",
      "CufflinksCuffmergePipeline\n",
      "Cuffmerge\n",
      "CummeRbund.GeneSetReport\n",
      "CummeRbund.QcReport\n",
      "CummeRbund.SelectedConditionsReport\n",
      "CummeRbund.SelectedGeneReport\n",
      "CytoscapeViewer\n",
      "DAPPLE\n",
      "DeIdentifyFCS\n",
      "DependencyBiomarkerDotplot\n",
      "DESeq2\n",
      "DownloadURL\n",
      "ESPPredictor\n",
      "ExpressionFileCreator\n",
      "ExprToGct\n",
      "ExtractColumnNames\n",
      "ExtractComparativeMarkerResults\n",
      "ExtractFCSDataset\n",
      "ExtractFCSKeywords\n",
      "ExtractFCSParameters\n",
      "ExtractRowNames\n",
      "FastQC\n",
      "FCMFeatureExtraction\n",
      "FCMSinglePanelQC\n",
      "FCSClean\n",
      "FCSNormalization\n",
      "FcsToCsv\n",
      "FeatureSummaryViewer\n",
      "FileSplitter\n",
      "FilterFilelist\n",
      "FilterLowshRNAs\n",
      "Find.differentially.expressed.genes.in.RNASeq.data\n",
      "Find.subnetworks.of.differentially.expressed.genes.and.identify.associated.biological.functions\n",
      "FLAMEChooseOptimalClusterNumber\n",
      "FLAMEContourDataGenerator\n",
      "FLAMEContourViewer.Pipeline\n",
      "FLAMEMetacluster\n",
      "FLAMEMixtureModel\n",
      "FLAMEPreprocess\n",
      "FLAMEPreviewTransformation\n",
      "FLAMEViewer\n",
      "FlowClustClassifyFCS\n",
      "FlowFingerprinting\n",
      "FlowMeansCluster\n",
      "FlowMergeCluster\n",
      "Fpkm_trackingToGct\n",
      "FPmatching\n",
      "Gct2Arff\n",
      "GctToPcl\n",
      "GENE_E\n",
      "GeneListSignificanceViewer\n",
      "GeneNeighbors\n",
      "GenePatternDocumentExtractor\n",
      "GEOImporter\n",
      "GetDatasetInSilico\n",
      "GISTIC_2.0\n",
      "GISTICPreprocess\n",
      "GLAD\n",
      "GlobalAlignment\n",
      "Golub.Slonim.1999.Nature.all.aml.pipeline\n",
      "GSEA\n",
      "GSEALeadingEdgeViewer\n",
      "GSEAPreranked\n",
      "HAPSEG\n",
      "HeatMapImage\n",
      "HeatMapViewer\n",
      "helloWorld\n",
      "helloWorld_R2.15_Rscript\n",
      "HierarchicalClustering\n",
      "HierarchicalClusteringImage\n",
      "HierarchicalClusteringViewer\n",
      "Hu68kHu35kAtoU95\n",
      "Identify.an.up.or.downregulated.pathway.from.expression.data\n",
      "Identify.and.Validate.a.Consensus.Signature.Using.Gene.Expression.Data\n",
      "Identify.and.visualize.expressed.transcripts.in.RNASeq.data\n",
      "IGV\n",
      "IGV.js\n",
      "IlluminaConcatenator\n",
      "IlluminaDASLPipeline\n",
      "IlluminaExpressionFileCreator\n",
      "IlluminaNormalizer\n",
      "IlluminaScanExtractor\n",
      "ImmPort_FLOCK_CrossSample\n",
      "ImmPort_FLOCK_Individual_FCS\n",
      "ImmPortColSelection\n",
      "ImmPortCrossSample\n",
      "ImmPortCSV2TXT\n",
      "ImmPortFCSConvLogicleTrans\n",
      "ImmPortFLOCK\n",
      "ImputeMissingValues.KNN\n",
      "Install.Proteogenomics.R.Packages\n",
      "IntersectingIdentifiers\n",
      "JavaTreeView\n",
      "KMeansClassifyFCS\n",
      "KMeansClustering\n",
      "KNN\n",
      "KNNXValidation\n",
      "KSscore\n",
      "LandmarkMatch\n",
      "Life.Cycle.Time.Point.Correlation\n",
      "ListFiles\n",
      "LocatePeaks\n",
      "LogicleTransformFCS\n",
      "LogisticRegression\n",
      "LogTransformGLAD\n",
      "LOHPaired\n",
      "Lu.Getz.Miska.Nature.June.2005.mouse.lung.pipeline\n",
      "MAGEMLImportViewer\n",
      "MAGETABImportViewer\n",
      "MapChipFeaturesGeneral\n",
      "Marker.Selection\n",
      "MClustClusterLabel\n",
      "MClustClusterLabelBIC\n",
      "MergeColumns\n",
      "MergeFCSDataFiles\n",
      "MergeHTSeqCounts\n",
      "MergeRows\n",
      "MetageneProjection\n",
      "MINDY\n",
      "MINDYViewer\n",
      "ModEcocTrainTest\n",
      "ModEcocXValidation\n",
      "Multiplot\n",
      "MultiplotExtractor\n",
      "MultiplotPreprocess\n",
      "MultiplotStudio\n",
      "MutSigCV\n",
      "mzXMLToCsv\n",
      "NearestTemplatePrediction\n",
      "NMF\n",
      "NMFConsensus\n",
      "NMFConsensus_combine_lewu\n",
      "NMFConsensus_sk_lewu\n",
      "NMFPartition\n",
      "NMFSingleKPipe\n",
      "NormLines\n",
      "ParallelCBS\n",
      "ParallelNMFConsensusPipe\n",
      "PARIS\n",
      "PCA\n",
      "PCAViewer\n",
      "PclToGct\n",
      "PeakMatch\n",
      "Peaks\n",
      "Picard.AddOrReplaceReadGroups\n",
      "Picard.BamToSam\n",
      "Picard.CreateSequenceDictionary\n",
      "Picard.FastqToSam\n",
      "Picard.MarkDuplicates\n",
      "Picard.ReorderSam\n",
      "Picard.SamToBam\n",
      "Picard.SamToFastq\n",
      "Picard.SortSam\n",
      "PlateQAFCS\n",
      "PlotPeaks\n",
      "PredictionResultsViewer\n",
      "Preprocess.and.quality.check.RNASeq.data\n",
      "PreprocessDataset\n",
      "PreprocessReadCounts\n",
      "PreviewFCS\n",
      "Profile.Plots\n",
      "Protein.mRNA.Corrlation\n",
      "ProteinDatasetCreation\n",
      "ProteoArray\n",
      "ProteomicsAnalysis\n",
      "QuaSAR\n",
      "RankNormalize\n",
      "Read_group_trackingToGct\n",
      "ReannotateGCT\n",
      "RemoveCopyNumberOutliers\n",
      "RemoveMissingValues\n",
      "removeSamples\n",
      "RemoveSaturatedFCSEvents\n",
      "RenameFile\n",
      "ReorderByClass\n",
      "ReplicatesQC\n",
      "REVEALER\n",
      "RLibDemo\n",
      "RNASeQC\n",
      "RNASeQC_CEGS\n",
      "SamSPECTRALClusterFCS\n",
      "SAMtools.FastaIndex\n",
      "Scripture\n",
      "ScripturePipeline\n",
      "ScripturePrealigned\n",
      "SegmentationCount\n",
      "SelectFeaturesColumns\n",
      "SelectFeaturesRows\n",
      "SelectFileMatrix\n",
      "SetFCSKeywords\n",
      "shRNAcollapseReps\n",
      "shRNAfoldChange\n",
      "shRNAmapGenes\n",
      "shRNAremoveOverlap\n",
      "SNPFileCreator\n",
      "SNPFileSorter\n",
      "SNPMultipleSampleAnalysis\n",
      "SnpViewer\n",
      "SOMClustering\n",
      "SOMClusterViewer\n",
      "SparseHierarchicalClustering\n",
      "SplitColumns\n",
      "SplitDatasetTrainTest\n",
      "SraToFastQ\n",
      "ssGSEAProjection\n",
      "StrandSpecRNAseqEval\n",
      "SubMap\n",
      "SuggestNumberOfPopulationsFCS\n",
      "SurvivalCurve\n",
      "SurvivalDifference\n",
      "SVM\n",
      "TmapToGct\n",
      "TopHat\n",
      "TransposeDataset\n",
      "Trimmomatic\n",
      "Two.Component.Normalization\n",
      "UniquifyLabels\n",
      "untar\n",
      "VennDiagram\n",
      "WeightedVoting\n",
      "WeightedVotingXValidation\n",
      "XChromosomeCorrect\n"
     ]
    }
   ],
   "source": [
    "task_list = gpserver.get_task_list()\n",
    "for t in task_list:\n",
    "    print(t.get_name())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "input.filename = None\n",
      "threshold.and.filter = 1\n",
      "floor = 20\n",
      "ceiling = 20000\n",
      "min.fold.change = 3\n",
      "min.delta = 100\n",
      "num.outliers.to.exclude = 0\n",
      "row.normalization = 0\n",
      "row.sampling.rate = 1\n",
      "threshold.for.removing.rows = None\n",
      "number.of.columns.above.threshold = None\n",
      "log2.transform = 0\n",
      "output.file.format = 3\n",
      "output.file = <input.filename_basename>.preprocessed\n"
     ]
    }
   ],
   "source": [
    "t = gp.GPTask(gpserver, 'PreprocessDataset')\n",
    "t.param_load()\n",
    "ps = t.get_parameters()\n",
    "for p in ps:\n",
    "    print('{} = {}'.format(p.get_name(), p.get_default_value()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "js = t.make_job_spec()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n"
     ]
    }
   ],
   "source": [
    "for p in t.get_parameters():\n",
    "    if p.get_default_value():\n",
    "        print(js.set_parameter(p.get_name(), p.get_default_value()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
